diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/AclAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/AclAPI.h
index aecd5722..9f731aa1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/AclAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/AclAPI.h
@@ -17,6 +17,7 @@ Abstract:
 #define __ACCESS_CONTROL_API__
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 
 #include <windows.h>
@@ -50,6 +51,7 @@ typedef VOID (CALLBACK *FN_PROGRESS) (
     _In_ BOOL                       SecuritySet     // Whether security was set
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -59,6 +61,7 @@ SetEntriesInAclA(
     _In_opt_  PACL           OldAcl,
     _Out_ PACL              * NewAcl
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -75,6 +78,7 @@ SetEntriesInAclW(
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -83,6 +87,7 @@ GetExplicitEntriesFromAclA(
     _Out_ PULONG                pcCountOfExplicitEntries,
     _Outptr_result_buffer_(*pcCountOfExplicitEntries) PEXPLICIT_ACCESS_A  * pListOfExplicitEntries
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -103,6 +108,7 @@ GetExplicitEntriesFromAclW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -111,6 +117,7 @@ GetEffectiveRightsFromAclA(
     _In_  PTRUSTEE_A    pTrustee,
     _Out_ PACCESS_MASK  pAccessRights
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -126,6 +133,7 @@ GetEffectiveRightsFromAclW(
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -135,6 +143,7 @@ GetAuditedPermissionsFromAclA(
     _Out_ PACCESS_MASK  pSuccessfulAuditedRights,
     _Out_ PACCESS_MASK  pFailedAuditRights
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -156,6 +165,7 @@ GetAuditedPermissionsFromAclW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -169,6 +179,7 @@ GetNamedSecurityInfoA(
     _Out_opt_       PACL         * ppSacl,
     _Out_ PSECURITY_DESCRIPTOR   * ppSecurityDescriptor
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -188,6 +199,7 @@ GetNamedSecurityInfoW(
 #define GetNamedSecurityInfo  GetNamedSecurityInfoA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -202,6 +214,7 @@ GetSecurityInfo(
     _Out_opt_ PSECURITY_DESCRIPTOR * ppSecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -214,6 +227,7 @@ SetNamedSecurityInfoA(
     _In_opt_ PACL              pDacl,
     _In_opt_ PACL              pSacl
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -232,6 +246,7 @@ SetNamedSecurityInfoW(
 #define SetNamedSecurityInfo  SetNamedSecurityInfoA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -251,6 +266,7 @@ SetSecurityInfo(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -266,6 +282,7 @@ GetInheritanceSourceA(
     _In_ PGENERIC_MAPPING         pGenericMapping,
     _Out_ PINHERITED_FROMA         pInheritArray
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -287,6 +304,7 @@ GetInheritanceSourceW(
 #define GetInheritanceSource  GetInheritanceSourceA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -296,6 +314,7 @@ FreeInheritedFromArray(
     _In_opt_ PFN_OBJECT_MGR_FUNCTS   pfnArray OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -312,6 +331,7 @@ TreeResetNamedSecurityInfoA(
     _In_     PROG_INVOKE_SETTING  ProgressInvokeSetting,
     _In_opt_ PVOID                Args
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -336,6 +356,7 @@ TreeResetNamedSecurityInfoW(
 
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 DWORD
 WINAPI
@@ -352,6 +373,7 @@ TreeSetNamedSecurityInfoA(
     _In_     PROG_INVOKE_SETTING  ProgressInvokeSetting,
     _In_opt_ PVOID                Args
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 DWORD
 WINAPI
@@ -381,6 +403,7 @@ TreeSetNamedSecurityInfoW(
 // implement access control on their own objects.
 //----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -395,6 +418,7 @@ BuildSecurityDescriptorA(
     _Out_ PULONG                  pSizeNewSD,
     _Outptr_result_bytebuffer_(*pSizeNewSD) PSECURITY_DESCRIPTOR  * pNewSD
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -415,6 +439,7 @@ BuildSecurityDescriptorW(
 #define BuildSecurityDescriptor  BuildSecurityDescriptorA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -427,6 +452,7 @@ LookupSecurityDescriptorPartsA(
     _Outptr_result_buffer_maybenull_(*pcCountOfAuditEntries) PEXPLICIT_ACCESS_A * ppListOfAuditEntries,
     _In_  PSECURITY_DESCRIPTOR pSD
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -451,6 +477,7 @@ LookupSecurityDescriptorPartsW(
 // access control structures.
 //----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -461,6 +488,7 @@ BuildExplicitAccessWithNameA(
     _In_       ACCESS_MODE         AccessMode,
     _In_       DWORD               Inheritance
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -505,6 +533,7 @@ BuildImpersonateExplicitAccessWithNameW(
 #define BuildImpersonateExplicitAccessWithName  BuildImpersonateExplicitAccessWithNameA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -512,6 +541,7 @@ BuildTrusteeWithNameA(
     _Inout_	PTRUSTEE_A  pTrustee,
     _In_opt_ LPSTR     pName
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -545,6 +575,7 @@ BuildImpersonateTrusteeW(
 #define BuildImpersonateTrustee  BuildImpersonateTrusteeA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -552,6 +583,7 @@ BuildTrusteeWithSidA(
     _Inout_ PTRUSTEE_A  pTrustee,
     _In_opt_ PSID        pSid
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -565,6 +597,7 @@ BuildTrusteeWithSidW(
 #define BuildTrusteeWithSid  BuildTrusteeWithSidA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -575,6 +608,7 @@ BuildTrusteeWithObjectsAndSidA(
     _In_opt_ GUID             * pInheritedObjectGuid,
     _In_opt_ PSID               pSid
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -591,6 +625,7 @@ BuildTrusteeWithObjectsAndSidW(
 #define BuildTrusteeWithObjectsAndSid  BuildTrusteeWithObjectsAndSidA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -602,6 +637,7 @@ BuildTrusteeWithObjectsAndNameA(
     _In_opt_ LPSTR             InheritedObjectTypeName,
     _In_opt_ LPSTR             Name
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -619,12 +655,14 @@ BuildTrusteeWithObjectsAndNameW(
 #define BuildTrusteeWithObjectsAndName  BuildTrusteeWithObjectsAndNameA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LPSTR
 WINAPI
 GetTrusteeNameA(
     _In_ PTRUSTEE_A  pTrustee
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LPWSTR
 WINAPI
@@ -637,12 +675,14 @@ GetTrusteeNameW(
 #define GetTrusteeName  GetTrusteeNameA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 TRUSTEE_TYPE
 WINAPI
 GetTrusteeTypeA(
     _In_opt_ PTRUSTEE_A  pTrustee
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 TRUSTEE_TYPE
 WINAPI
@@ -655,12 +695,14 @@ GetTrusteeTypeW(
 #define GetTrusteeType  GetTrusteeTypeA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 TRUSTEE_FORM
 WINAPI
 GetTrusteeFormA(
     _In_ PTRUSTEE_A  pTrustee
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 TRUSTEE_FORM
 WINAPI
