diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/advpub.h b/generation/WinSDK/RecompiledIdlHeaders/um/advpub.h
index 89687f51..b52a8e2e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/advpub.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/advpub.h
@@ -15,6 +15,7 @@
 #ifndef _ADVPUB_H_
 #define _ADVPUB_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -248,7 +249,9 @@ typedef STRTABLEA STRTABLE;
 typedef LPCSTRTABLEA LPCSTRTABLE;
 #endif // UNICODE
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT WINAPI RegInstallA(HMODULE hmod, LPCSTR pszSection, const STRTABLEA* pstTable);
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT WINAPI RegInstallW(HMODULE hmod, LPCWSTR pszSection, const STRTABLEW* pstTable);
 #ifdef UNICODE
 #define RegInstall  RegInstallW
