diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/amsi.h b/generation/WinSDK/RecompiledIdlHeaders/um/amsi.h
index 796708e1..f69c6bd6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/amsi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/amsi.h
@@ -102,6 +102,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -254,6 +255,7 @@ EXTERN_C const IID IID_IAmsiStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("3e47f2e5-81d4-4d3b-897f-545096770373")
     IAmsiStream : public IUnknown
     {
@@ -374,6 +376,7 @@ EXTERN_C const IID IID_IAntimalwareProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("b2cabfe3-fe04-42b1-a5df-08d483d4d125")
     IAntimalwareProvider : public IUnknown
     {
@@ -728,6 +731,7 @@ EXTERN_C const IID IID_IAntimalware;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("82d29c2e-f062-44e6-b5c9-3d9a2f24a2df")
     IAntimalware : public IUnknown
     {
@@ -962,21 +966,30 @@ EXTERN_C const CLSID CLSID_Antimalware;
 DECLARE_HANDLE(HAMSICONTEXT);
 DECLARE_HANDLE(HAMSISESSION);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI AmsiInitialize(
     _In_  LPCWSTR appName,
-    _Outptr_ HAMSICONTEXT* amsiContext);
+    _Outptr_ HAMSICONTEXT* amsiContext
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(AmsiUninitialize));
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI_(VOID) AmsiUninitialize(
     _In_  HAMSICONTEXT amsiContext);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI AmsiOpenSession(
     _In_  HAMSICONTEXT amsiContext,
-    _Out_ HAMSISESSION* amsiSession);
+    _Out_ HAMSISESSION* amsiSession
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(AmsiCloseSession));
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI_(VOID) AmsiCloseSession(
     _In_  HAMSICONTEXT amsiContext,
     _In_  HAMSISESSION amsiSession);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI AmsiScanBuffer(
     _In_  HAMSICONTEXT amsiContext,
     _In_reads_bytes_(length) PVOID buffer,
@@ -992,6 +1005,7 @@ STDAPI AmsiNotifyOperation(
     _In_opt_  LPCWSTR contentName,
     _Out_ AMSI_RESULT* result);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI AmsiScanString(
     _In_  HAMSICONTEXT amsiContext,
     _In_  LPCWSTR string,
