diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/amstream.h b/generation/WinSDK/RecompiledIdlHeaders/um/amstream.h
index c8aae883..92a70231 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/amstream.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/amstream.h
@@ -110,6 +110,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #include <ddraw.h>
@@ -978,6 +981,7 @@ EXTERN_C const IID IID_IDirectDrawMediaSampleAllocator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("AB6B4AFC-F6E4-11d0-900D-00C04FD9189D")
     IDirectDrawMediaSampleAllocator : public IUnknown
     {
@@ -1063,6 +1067,7 @@ EXTERN_C const IID IID_IDirectDrawMediaSample;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("AB6B4AFE-F6E4-11d0-900D-00C04FD9189D")
     IDirectDrawMediaSample : public IUnknown
     {
