diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/amvideo.h b/generation/WinSDK/RecompiledIdlHeaders/um/amvideo.h
index 906ec740..43518e5c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/amvideo.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/amvideo.h
@@ -10,6 +10,7 @@
 #ifndef __AMVIDEO__
 #define __AMVIDEO__
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -56,6 +57,7 @@ typedef LONGLONG REFERENCE_TIME;
 #undef INTERFACE
 #define INTERFACE IDirectDrawVideo
 
+_Win32_metadata_supported_os_(windows5.0)
 DECLARE_INTERFACE_(IDirectDrawVideo, IUnknown)
 {
     // IUnknown methods
@@ -89,6 +91,7 @@ DECLARE_INTERFACE_(IDirectDrawVideo, IUnknown)
 #undef INTERFACE
 #define INTERFACE IQualProp
 
+_Win32_metadata_supported_os_(windows5.0)
 DECLARE_INTERFACE_(IQualProp, IUnknown)
 {
     // IUnknown methods
@@ -169,6 +172,7 @@ DECLARE_INTERFACE_(IFullScreenVideo, IUnknown)
 #undef INTERFACE
 #define INTERFACE IFullScreenVideoEx
 
+_Win32_metadata_supported_os_(windows5.0)
 DECLARE_INTERFACE_(IFullScreenVideoEx, IFullScreenVideo)
 {
     // IUnknown methods
