diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/appmodel.h b/generation/WinSDK/RecompiledIdlHeaders/um/appmodel.h
index 5439c396..b3fd7a66 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/appmodel.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/appmodel.h
@@ -24,6 +24,9 @@
 
 #ifndef _APPMODEL_H_
 #define _APPMODEL_H_
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if defined(_CONTRACT_GEN)
 #include <nt.h>
@@ -79,6 +82,7 @@ typedef struct PACKAGE_ID {
 
 // Identity Functions
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -88,6 +92,7 @@ GetCurrentPackageId(
     _Out_writes_bytes_opt_(*bufferLength) BYTE* buffer
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -97,6 +102,7 @@ GetCurrentPackageFullName(
     _Out_writes_opt_(*packageFullNameLength) PWSTR packageFullName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -106,6 +112,7 @@ GetCurrentPackageFamilyName(
     _Out_writes_opt_(*packageFamilyNameLength) PWSTR packageFamilyName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -115,6 +122,7 @@ GetCurrentPackagePath(
     _Out_writes_opt_(*pathLength) PWSTR path
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -125,6 +133,7 @@ GetPackageId(
     _Out_writes_bytes_opt_(*bufferLength) BYTE* buffer
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -136,6 +145,7 @@ GetPackageFullName(
     );
 
 //TODO:8645770 Change 0x0101 to 0x0102 once THRESHOLD constants are available
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -147,6 +157,7 @@ GetPackageFullNameFromToken(
     _Out_writes_opt_(*packageFullNameLength) PWSTR packageFullName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -158,6 +169,7 @@ GetPackageFamilyName(
     );
 
 //TODO:8645770 Change 0x0101 to 0x0102 once THRESHOLD constants are available
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -169,6 +181,7 @@ GetPackageFamilyNameFromToken(
     _Out_writes_opt_(*packageFamilyNameLength) PWSTR packageFamilyName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -180,6 +193,7 @@ GetPackagePath(
     _Out_writes_opt_(*pathLength) PWSTR path
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -190,6 +204,7 @@ GetPackagePathByFullName(
     _Out_writes_opt_(*pathLength) PWSTR path
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -214,6 +229,7 @@ typedef enum PackagePathType
 #endif // NTDDI_VERSION >= NTDDI_WIN10_VB
 } PackagePathType;
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -225,6 +241,7 @@ GetPackagePathByFullName2(
     _Out_writes_opt_(*pathLength) PWSTR path
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -236,6 +253,7 @@ GetStagedPackagePathByFullName2(
     _Out_writes_opt_(*pathLength) PWSTR path
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -248,6 +266,7 @@ GetCurrentPackageInfo2(
     _Out_opt_ UINT32* count
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -349,6 +368,7 @@ VerifyPackageRelativeApplicationId(
 
 // Conversion Functions
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -361,6 +381,7 @@ PackageIdFromFullName(
     _Out_writes_bytes_opt_(*bufferLength) BYTE* buffer
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -372,6 +393,7 @@ PackageFullNameFromId(
     _Out_writes_opt_(*packageFullNameLength) PWSTR packageFullName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -383,6 +405,7 @@ PackageFamilyNameFromId(
     _Out_writes_opt_(*packageFamilyNameLength) PWSTR packageFamilyName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -394,6 +417,7 @@ PackageFamilyNameFromFullName(
     _Out_writes_opt_(*packageFamilyNameLength) PWSTR packageFamilyName
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -407,6 +431,7 @@ PackageNameAndPublisherIdFromFamilyName(
     _Out_writes_opt_(*packagePublisherIdLength) PWSTR packagePublisherId
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -419,6 +444,7 @@ FormatApplicationUserModelId(
     _Out_writes_opt_(*applicationUserModelIdLength) PWSTR applicationUserModelId
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -436,6 +462,7 @@ ParseApplicationUserModelId(
 
 // Lookup Functions
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -452,6 +479,7 @@ GetPackagesByPackageFamily(
     );
 
 /* Any combination of PACKAGE_FILTER_* */
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -480,6 +508,7 @@ typedef enum PackageOrigin
     PackageOrigin_LineOfBusiness    = 6
 } PackageOrigin;
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Check_return_
 _Success_(return == ERROR_SUCCESS)
@@ -544,6 +573,7 @@ typedef struct PACKAGE_INFO {
 
 // Dependency Functions
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -555,6 +585,7 @@ GetCurrentPackageInfo(
     _Out_opt_ UINT32* count
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -578,6 +609,7 @@ OpenPackageInfoByFullNameForUser(
     _Out_ PACKAGE_INFO_REFERENCE* packageInfoReference
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -586,6 +618,7 @@ ClosePackageInfo(
     _In_ PACKAGE_INFO_REFERENCE packageInfoReference
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -598,6 +631,7 @@ GetPackageInfo(
     _Out_opt_ UINT32* count
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
@@ -610,6 +644,7 @@ GetPackageApplicationIds(
     );
 
 #if NTDDI_VERSION >= NTDDI_WIN10_19H1
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 LONG
