diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/appserviceinterop.h b/generation/WinSDK/RecompiledIdlHeaders/um/appserviceinterop.h
index 33742703..c73fd381 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/appserviceinterop.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/appserviceinterop.h
@@ -75,6 +75,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS2)
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -288,14 +291,22 @@ extern RPC_IF_HANDLE __MIDL_itf_appserviceinterop_0000_0002_v0_0_s_ifspec;
 
 /* Additional Prototypes for ALL interfaces */
 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned long             __RPC_USER  HSTRING_UserSize(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserMarshal(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 void                      __RPC_USER  HSTRING_UserFree(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned long             __RPC_USER  HSTRING_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 void                      __RPC_USER  HSTRING_UserFree64(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
 /* end of Additional Prototypes */
