diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h b/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
index 7553d141..9d019384 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
@@ -66,6 +66,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #define   ENDPOINT_FORMAT_RESET_MIX_ONLY     0x00000001
@@ -85,6 +88,7 @@ EXTERN_C const IID IID_IAudioEndpointFormatControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("784CFD40-9F89-456E-A1A6-873B006A664E")
     IAudioEndpointFormatControl : public IUnknown
     {
