diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/audiomediatype.h b/generation/WinSDK/RecompiledIdlHeaders/um/audiomediatype.h
index 5f40e3c8..af2fe475 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/audiomediatype.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/audiomediatype.h
@@ -68,6 +68,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if !defined( _UNCOMPRESSEDAUDIOFORMAT_ )
@@ -221,6 +222,7 @@ EXTERN_C const IID IID_IAudioMediaType;
 //
 // CreateAudioMediaType
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI CreateAudioMediaType(
     const WAVEFORMATEX* pAudioFormat,
     UINT32 cbAudioFormatSize,
@@ -229,6 +231,7 @@ STDAPI CreateAudioMediaType(
 //
 // CreateAudioMediaTypeFromUncompressedAudioFormat
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI CreateAudioMediaTypeFromUncompressedAudioFormat(
     const UNCOMPRESSEDAUDIOFORMAT* pUncompressedAudioFormat,
     IAudioMediaType** ppIAudioMediaType
