diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/aux_ulib.h b/generation/WinSDK/RecompiledIdlHeaders/um/aux_ulib.h
index f2f2c78c..02bee3e8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/aux_ulib.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/aux_ulib.h
@@ -17,6 +17,7 @@ Abstract:
 #define _AUX_SHLD_LIB_H
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -47,6 +48,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AuxUlibInitialize (
@@ -82,6 +84,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_set_last_error_
 BOOL 
 WINAPI
 AuxUlibSetSystemFileCacheSize (
@@ -131,6 +134,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AuxUlibIsDLLSynchronizationHeld (
