diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothleapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothleapis.h
index 8d432133..beec7454 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothleapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothleapis.h
@@ -19,6 +19,7 @@ Environment:
 #pragma once
 #endif //_MSC_VER
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #include <BthLEDef.h>
 
 #pragma region Desktop Family
@@ -75,6 +76,7 @@ extern "C"{
 //      ERROR_NO_SYSTEM_RESOURCES
 //          The operation ran out of memory.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetServices(
@@ -141,6 +143,7 @@ BluetoothGATTGetServices(
 //          If both a parent service and a service handle is provided and the
 //          parent service hierarchy do not roll up to the service handle that is provided.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetIncludedServices(
@@ -198,6 +201,7 @@ BluetoothGATTGetIncludedServices(
 ///      ERROR_NO_SYSTEM_RESOURCES
 //          The operation ran out of memory.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetCharacteristics(
@@ -253,6 +257,7 @@ BluetoothGATTGetCharacteristics(
 ///      ERROR_NO_SYSTEM_RESOURCES
 //          The operation ran out of memory.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetDescriptors(
@@ -326,6 +331,7 @@ BluetoothGATTGetDescriptors(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetCharacteristicValue(
@@ -396,6 +402,7 @@ BluetoothGATTGetCharacteristicValue(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTGetDescriptorValue(
@@ -428,6 +435,7 @@ BluetoothGATTGetDescriptorValue(
 //      ERROR_INVALID_FUNCTION
 //          A reliable write operation is already presently underway.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTBeginReliableWrite(
@@ -486,6 +494,7 @@ BluetoothGATTBeginReliableWrite(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTSetCharacteristicValue(
@@ -532,6 +541,7 @@ BluetoothGATTSetCharacteristicValue(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTEndReliableWrite(
@@ -577,6 +587,7 @@ BluetoothGATTEndReliableWrite(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTAbortReliableWrite(
@@ -628,6 +639,7 @@ BluetoothGATTAbortReliableWrite(
 //          An error code returned by the device indicating a specific error has occured.
 //          Consult the Bluetooth specification for more information.
 //
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(S_OK == return)
 HRESULT WINAPI
 BluetoothGATTSetDescriptorValue(
@@ -677,6 +689,7 @@ BluetoothGATTSetDescriptorValue(
 //      ERROR_INVALID_PARAMETER
 //          One of the parameters were invalid.
 //
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT WINAPI
 BluetoothGATTRegisterEvent(
     _In_ HANDLE hService,
@@ -706,6 +719,7 @@ BluetoothGATTRegisterEvent(
 //      ERROR_INVALID_PARAMETER
 //          One of the parameters were invalid.
 //
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT WINAPI
 BluetoothGATTUnregisterEvent(
     _In_ BLUETOOTH_GATT_EVENT_HANDLE EventHandle,
