diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertExit.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertExit.h
index c656b3bb..1d8461ac 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertExit.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertExit.h
@@ -74,6 +74,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #define EXITEVENT_INVALID               ( 0x0 )  // Invalid event
@@ -102,6 +103,7 @@ EXTERN_C const IID IID_ICertExit;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("e19ae1a0-7364-11d0-8816-00a0c903b83c")
     ICertExit : public IDispatch
     {
@@ -285,6 +287,7 @@ EXTERN_C const IID IID_ICertExit2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("0abf484b-d049-464d-a7ed-552e7529b0ff")
     ICertExit2 : public ICertExit
     {
@@ -472,9 +475,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* end of Additional Prototypes */
