diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertEnc.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertEnc.h
index 63a04aab..e4cfac2e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertEnc.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertEnc.h
@@ -216,6 +216,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -234,6 +235,7 @@ EXTERN_C const IID IID_ICertEncodeStringArray;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("12a88820-7494-11d0-8816-00a0c903b83c")
     ICertEncodeStringArray : public IDispatch
     {
@@ -253,7 +255,7 @@ EXTERN_C const IID IID_ICertEncodeStringArray;
         
         virtual HRESULT STDMETHODCALLTYPE Reset( 
             /* [in] */ LONG Count,
-            /* [in] */ LONG StringType) = 0;
+            /* [in] */ CERT_RDN_ATTR_VALUE_TYPE StringType) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetValue( 
             /* [in] */ LONG Index,
@@ -352,7 +354,7 @@ EXTERN_C const IID IID_ICertEncodeStringArray;
         HRESULT ( STDMETHODCALLTYPE *Reset )( 
             __RPC__in ICertEncodeStringArray * This,
             /* [in] */ LONG Count,
-            /* [in] */ LONG StringType);
+            /* [in] */ CERT_RDN_ATTR_VALUE_TYPE StringType);
         
         DECLSPEC_XFGVIRT(ICertEncodeStringArray, SetValue)
         HRESULT ( STDMETHODCALLTYPE *SetValue )( 
@@ -546,7 +548,7 @@ EXTERN_C const IID IID_ICertEncodeStringArray2;
         HRESULT ( STDMETHODCALLTYPE *Reset )( 
             __RPC__in ICertEncodeStringArray2 * This,
             /* [in] */ LONG Count,
-            /* [in] */ LONG StringType);
+            /* [in] */ CERT_RDN_ATTR_VALUE_TYPE StringType);
         
         DECLSPEC_XFGVIRT(ICertEncodeStringArray, SetValue)
         HRESULT ( STDMETHODCALLTYPE *SetValue )( 
@@ -657,6 +659,7 @@ EXTERN_C const IID IID_ICertEncodeLongArray;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("15e2f230-a0a2-11d0-8821-00a0c903b83c")
     ICertEncodeLongArray : public IDispatch
     {
@@ -1058,6 +1061,7 @@ EXTERN_C const IID IID_ICertEncodeDateArray;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("2f9469a0-a470-11d0-8821-00a0c903b83c")
     ICertEncodeDateArray : public IDispatch
     {
@@ -1459,6 +1463,7 @@ EXTERN_C const IID IID_ICertEncodeCRLDistInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("01958640-bbff-11d0-8825-00a0c903b83c")
     ICertEncodeCRLDistInfo : public IDispatch
     {
@@ -1493,7 +1498,7 @@ EXTERN_C const IID IID_ICertEncodeCRLDistInfo;
         virtual HRESULT STDMETHODCALLTYPE SetNameEntry( 
             /* [in] */ LONG DistPointIndex,
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Encode( 
@@ -1610,7 +1615,7 @@ EXTERN_C const IID IID_ICertEncodeCRLDistInfo;
             __RPC__in ICertEncodeCRLDistInfo * This,
             /* [in] */ LONG DistPointIndex,
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName);
         
         DECLSPEC_XFGVIRT(ICertEncodeCRLDistInfo, Encode)
@@ -1826,7 +1831,7 @@ EXTERN_C const IID IID_ICertEncodeCRLDistInfo2;
             __RPC__in ICertEncodeCRLDistInfo2 * This,
             /* [in] */ LONG DistPointIndex,
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName);
         
         DECLSPEC_XFGVIRT(ICertEncodeCRLDistInfo, Encode)
@@ -1950,6 +1955,7 @@ EXTERN_C const IID IID_ICertEncodeAltName;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("1c9a8c70-1271-11d1-9bd4-00c04fb683fa")
     ICertEncodeAltName : public IDispatch
     {
@@ -1973,7 +1979,7 @@ EXTERN_C const IID IID_ICertEncodeAltName;
         
         virtual HRESULT STDMETHODCALLTYPE SetNameEntry( 
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Encode( 
@@ -2075,7 +2081,7 @@ EXTERN_C const IID IID_ICertEncodeAltName;
         HRESULT ( STDMETHODCALLTYPE *SetNameEntry )( 
             __RPC__in ICertEncodeAltName * This,
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName);
         
         DECLSPEC_XFGVIRT(ICertEncodeAltName, Encode)
@@ -2281,7 +2287,7 @@ EXTERN_C const IID IID_ICertEncodeAltName2;
         HRESULT ( STDMETHODCALLTYPE *SetNameEntry )( 
             __RPC__in ICertEncodeAltName2 * This,
             /* [in] */ LONG NameIndex,
-            /* [in] */ LONG NameChoice,
+            /* [in] */ CERT_ALT_NAME NameChoice,
             /* [in] */ __RPC__in const BSTR strName);
         
         DECLSPEC_XFGVIRT(ICertEncodeAltName, Encode)
@@ -2408,6 +2414,7 @@ EXTERN_C const IID IID_ICertEncodeBitString;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("6db525be-1278-11d1-9bd4-00c04fb683fa")
     ICertEncodeBitString : public IDispatch
     {
@@ -2855,9 +2862,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* end of Additional Prototypes */
