diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertBCli.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertBCli.h
index 9156600f..6d9bb166 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertBCli.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertBCli.h
@@ -17,6 +17,9 @@
 #define __CERTBCLI_H__
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -153,6 +156,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVISSERVERONLINEW)(
     IN  WCHAR const *pwszServerName,
     OUT BOOL *pfServerOnline);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvIsServerOnlineW(
@@ -187,6 +191,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPGETDYNAMICFILELISTW)(
     OUT WCHAR **ppwszzFileList,
     OUT DWORD *pcbSize);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupGetDynamicFileListW(
@@ -221,6 +226,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPPREPAREW)(
     IN  ULONG dwBackupFlags,	// CSBACKUP_TYPE_*
     OUT HCSBC *phbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupPrepareW(
@@ -257,6 +263,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPGETDATABASENAMESW)(
     OUT WCHAR **ppwszzAttachmentInformation,
     OUT DWORD *pcbSize);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupGetDatabaseNamesW(
@@ -290,6 +297,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPOPENFILEW)(
     IN  DWORD cbReadHintSize,
     OUT LARGE_INTEGER *pliFileSize);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupOpenFileW(
@@ -326,6 +334,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPREAD)(
     IN  DWORD cbBuffer,
     OUT DWORD *pcbRead);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupRead(
@@ -354,6 +363,7 @@ FNCERTSRVBACKUPREAD *pfnCertSrvBackupRead = CertSrvBackupRead;
 typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPCLOSE)(
     IN HCSBC hbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupClose(
@@ -388,6 +398,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPGETBACKUPLOGSW)(
     OUT WCHAR **ppwszzBackupLogFiles,
     OUT DWORD *pcbSize);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupGetBackupLogsW(
@@ -414,6 +425,7 @@ FNCERTSRVBACKUPGETBACKUPLOGSW *pfnCertSrvBackupGetBackupLogs = CertSrvBackupGetB
 typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPTRUNCATELOGS)(
     IN HCSBC hbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupTruncateLogs(
@@ -438,6 +450,7 @@ FNCERTSRVBACKUPTRUNCATELOGS *pfnCertSrvBackupTruncateLogs = CertSrvBackupTruncat
 typedef HRESULT (CERTBCLI_CALL FNCERTSRVBACKUPEND)(
     IN HCSBC hbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvBackupEnd(
@@ -461,6 +474,7 @@ FNCERTSRVBACKUPEND *pfnCertSrvBackupEnd = CertSrvBackupEnd;
 typedef VOID (CERTBCLI_CALL FNCERTSRVBACKUPFREE)(
     IN VOID *pv);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 VOID
 CERTBCLI_API
 CertSrvBackupFree(
@@ -501,6 +515,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVRESTOREGETDATABASELOCATIONSW)(
     OUT WCHAR **ppwszzDatabaseLocationList,
     OUT DWORD *pcbSize);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestoreGetDatabaseLocationsW(
@@ -534,6 +549,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVRESTOREPREPAREW)(
     IN  ULONG dwRestoreFlags,
     OUT HCSBC *phbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestorePrepareW(
@@ -576,6 +592,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVRESTOREREGISTERW)(
     IN ULONG genLow,
     IN ULONG genHigh);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestoreRegisterW(
@@ -588,6 +605,7 @@ CertSrvRestoreRegisterW(
     IN ULONG genLow,
     IN ULONG genHigh);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestoreRegisterThroughFile(
@@ -622,6 +640,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVRESTOREREGISTERCOMPLETE)(
     OPTIONAL IN HCSBC hbc,
     IN HRESULT hrRestoreState);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestoreRegisterComplete(
@@ -647,6 +666,7 @@ FNCERTSRVRESTOREREGISTERCOMPLETE *pfnCertSrvRestoreRegisterComplete = CertSrvRes
 typedef HRESULT (CERTBCLI_CALL FNCERTSRVRESTOREEND)(
     IN HCSBC hbc);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvRestoreEnd(
@@ -682,6 +702,7 @@ typedef HRESULT (CERTBCLI_CALL FNCERTSRVSERVERCONTROLW)(
     OPTIONAL OUT DWORD *pcbOut,
     OPTIONAL OUT BYTE **ppbOut);
 
+_Win32_metadata_supported_os_(windowsserver2003)
 HRESULT
 CERTBCLI_API
 CertSrvServerControlW(
