diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertAdm.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertAdm.h
index e4b56f55..089c3ee3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertAdm.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertAdm.h
@@ -157,6 +157,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #define	CA_DISP_INCOMPLETE	( 0 )
@@ -215,6 +216,7 @@ EXTERN_C const IID IID_ICertAdmin;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("34df6950-7fb6-11d0-8817-00a0c903b83c")
     ICertAdmin : public IDispatch
     {
@@ -518,6 +520,7 @@ EXTERN_C const IID IID_ICertAdmin2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("f7c3ac41-b8ce-4fb4-aa58-3d1dc0e36b39")
     ICertAdmin2 : public ICertAdmin
     {
@@ -936,6 +939,7 @@ EXTERN_C const IID IID_IOCSPProperty;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("66FB7839-5F04-4C25-AD18-9FF1A8376EE0")
     IOCSPProperty : public IDispatch
     {
@@ -1107,6 +1111,7 @@ EXTERN_C const IID IID_IOCSPPropertyCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("2597C18D-54E6-4B74-9FA9-A6BFDA99CBBE")
     IOCSPPropertyCollection : public IDispatch
     {
@@ -1330,6 +1335,7 @@ EXTERN_C const IID IID_IOCSPCAConfiguration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("AEC92B40-3D46-433F-87D1-B84D5C1E790D")
     IOCSPCAConfiguration : public IDispatch
     {
@@ -1721,6 +1727,7 @@ EXTERN_C const IID IID_IOCSPCAConfigurationCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("2BEBEA0B-5ECE-4F28-A91C-86B4BB20F0D3")
     IOCSPCAConfigurationCollection : public IDispatch
     {
@@ -1922,6 +1929,7 @@ EXTERN_C const IID IID_IOCSPAdmin;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("322E830D-67DB-4FE9-9577-4596D9F09294")
     IOCSPAdmin : public IDispatch
     {
@@ -2289,14 +2297,22 @@ unsigned char * __RPC_USER  VARIANT_UserMarshal(  __RPC__in unsigned long *, __R
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
 void                      __RPC_USER  VARIANT_UserFree(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  VARIANT_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
 /* end of Additional Prototypes */
