diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertView.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertView.h
index 5f916a65..6ecb571a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertView.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertView.h
@@ -102,6 +102,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #define	CV_OUT_BASE64HEADER	( 0 )
@@ -202,6 +203,7 @@ EXTERN_C const IID IID_IEnumCERTVIEWCOLUMN;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("9c735be2-57a5-11d1-9bdb-00c04fb683fa")
     IEnumCERTVIEWCOLUMN : public IDispatch
     {
@@ -439,6 +441,7 @@ EXTERN_C const IID IID_IEnumCERTVIEWATTRIBUTE;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("e77db656-7653-11d1-9bde-00c04fb683fa")
     IEnumCERTVIEWATTRIBUTE : public IDispatch
     {
@@ -630,6 +633,7 @@ EXTERN_C const IID IID_IEnumCERTVIEWEXTENSION;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("e7dd1466-7653-11d1-9bde-00c04fb683fa")
     IEnumCERTVIEWEXTENSION : public IDispatch
     {
@@ -836,6 +840,7 @@ EXTERN_C const IID IID_IEnumCERTVIEWROW;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("d1157f4c-5af2-11d1-9bdc-00c04fb683fa")
     IEnumCERTVIEWROW : public IDispatch
     {
@@ -1053,6 +1058,7 @@ EXTERN_C const IID IID_ICertView;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("c3fac344-1e84-11d1-9bd6-00c04fb683fa")
     ICertView : public IDispatch
     {
@@ -1282,6 +1288,7 @@ EXTERN_C const IID IID_ICertView2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2003)
     MIDL_INTERFACE("d594b282-8851-4b61-9c66-3edadf848863")
     ICertView2 : public ICertView
     {
