diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/CertPolEng.h b/generation/WinSDK/RecompiledIdlHeaders/um/CertPolEng.h
index 3473df5e..509e9847 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/CertPolEng.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/CertPolEng.h
@@ -3,6 +3,9 @@
 //
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -12,6 +15,7 @@
 extern "C" {
 #endif
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS 
 NTAPI
 PstGetTrustAnchors(
@@ -39,6 +43,7 @@ PstGetCertificateChain(
     _Out_                                   PCCERT_CHAIN_CONTEXT*               ppCertChainContext
     );
     
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS 
 NTAPI
 PstGetCertificates(
@@ -50,6 +55,7 @@ PstGetCertificates(
     _Outptr_result_buffer_(*pdwCertChainContextCount) PCCERT_CHAIN_CONTEXT**    ppCertChainContexts
     );
     
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NTAPI
 PstAcquirePrivateKey(
@@ -57,6 +63,7 @@ PstAcquirePrivateKey(
     );
 
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS 
 NTAPI
 PstValidate(
@@ -68,6 +75,7 @@ PstValidate(
     _Out_opt_                               GUID*                               pProvGUID
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS 
 NTAPI
 PstMapCertificate(
@@ -76,6 +84,7 @@ PstMapCertificate(
     _Outptr_                             PVOID*                              ppTokenInformation
   );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS 
 NTAPI
 PstGetUserNameForCertificate(
