diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h b/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
index c2a789cd..8e567d0b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
@@ -21,6 +21,7 @@
 =============================================================================*/
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or BootableSku Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_BOOTABLESKU)
@@ -929,6 +930,7 @@ extern "C"
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnEqual
 (
@@ -958,6 +960,7 @@ ClfsLsnEqual
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnLess
 (
@@ -987,6 +990,7 @@ ClfsLsnLess
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnGreater
 (
@@ -1016,6 +1020,7 @@ ClfsLsnGreater
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnNull
 (
