diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h b/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
index c2a789cd..7a197397 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/clfs.h
@@ -21,6 +21,9 @@
 =============================================================================*/
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or BootableSku Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_BOOTABLESKU)
@@ -929,6 +932,7 @@ extern "C"
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnEqual
 (
@@ -958,6 +962,7 @@ ClfsLsnEqual
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnLess
 (
@@ -987,6 +992,7 @@ ClfsLsnLess
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnGreater
 (
@@ -1016,6 +1022,7 @@ ClfsLsnGreater
 //
 //-----------------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 CLFSUSER_API BOOLEAN NTAPI
 ClfsLsnNull
 (
