diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/cryptdlg.h b/generation/WinSDK/RecompiledIdlHeaders/um/cryptdlg.h
index b6e314ea..69accb72 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/cryptdlg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/cryptdlg.h
@@ -16,6 +16,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -153,6 +154,8 @@ typedef CERT_SELECT_STRUCT_A CERT_SELECT_STRUCT;
 typedef PCERT_SELECT_STRUCT_A PCERT_SELECT_STRUCT;
 #endif // UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 CRYPTDLGAPI
 BOOL
 WINAPI
@@ -162,6 +165,8 @@ CertSelectCertificateA(
 #ifdef MAC
 #define CertSelectCertificate CertSelectCertificateA
 #else   // !MAC
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 CRYPTDLGAPI
 BOOL
 WINAPI
@@ -248,6 +253,7 @@ typedef CERT_VIEWPROPERTIES_STRUCT_A CERT_VIEWPROPERTIES_STRUCT;
 typedef PCERT_VIEWPROPERTIES_STRUCT_A PCERT_VIEWPROPERTIES_STRUCT;
 #endif // UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 CRYPTDLGAPI
 BOOL
 WINAPI
@@ -257,6 +263,7 @@ CertViewPropertiesA(
 #ifdef MAC
 #define CertViewProperties CertViewPropertiesA
 #else   // !MAC
+_Win32_metadata_supported_os_(windows5.1.2600)
 CRYPTDLGAPI
 BOOL
 WINAPI
@@ -309,6 +316,7 @@ typedef struct tagCMFLTR {
 //
 //   Get a formatted friendly name for a certificate
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 CRYPTDLGAPI
 DWORD
 WINAPI
@@ -317,6 +325,7 @@ GetFriendlyNameOfCertA(
     _Out_writes_to_opt_(cch, return) LPSTR pch,
     _In_ DWORD cch
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
 CRYPTDLGAPI
 DWORD
 WINAPI
@@ -444,6 +453,7 @@ typedef struct _CTL_MODIFY_REQUEST {
     DWORD               dwError;        // Operation error code
 } CTL_MODIFY_REQUEST, * PCTL_MODIFY_REQUEST;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 CRYPTDLGAPI
 HRESULT
 WINAPI
