diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_1.h b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_1.h
index 562f9109..0ba5280b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_1.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_1.h
@@ -5,6 +5,10 @@
 //
 // File name: D2D1_1.h
 //---------------------------------------------------------------------------
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifdef _MSC_VER
 #pragma once
 #endif // #ifdef _MSC_VER
@@ -670,6 +674,7 @@ EXTERN_C CONST IID IID_ID2D1Multithread;
 /// <summary>
 /// User-implementable interface for introspecting on a metafile.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("82237326-8111-4f7c-bcf4-b5c1175564fe") ID2D1GdiMetafileSink  : public IUnknown
 {
     
@@ -687,6 +692,7 @@ interface DX_DECLARE_INTERFACE("82237326-8111-4f7c-bcf4-b5c1175564fe") ID2D1GdiM
 /// <summary>
 /// Interface encapsulating a GDI/GDI+ metafile.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("2f543dc3-cfc1-4211-864f-cfd91c6f3395") ID2D1GdiMetafile  : public ID2D1Resource
 {
     
@@ -710,6 +716,7 @@ interface DX_DECLARE_INTERFACE("2f543dc3-cfc1-4211-864f-cfd91c6f3395") ID2D1GdiM
 /// Caller-supplied implementation of an interface to receive the recorded command
 /// list.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("54d7898a-a061-40a7-bec7-e465bcba2c4f") ID2D1CommandSink  : public IUnknown
 {
     
@@ -848,6 +855,7 @@ interface DX_DECLARE_INTERFACE("54d7898a-a061-40a7-bec7-e465bcba2c4f") ID2D1Comm
 /// <summary>
 /// The commandList interface.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("b4f34a19-2383-4d76-94f6-ec343657c3dc") ID2D1CommandList  : public ID2D1Image
 {
     
@@ -870,6 +878,7 @@ interface DX_DECLARE_INTERFACE("b4f34a19-2383-4d76-94f6-ec343657c3dc") ID2D1Comm
 /// Converts Direct2D primitives stored in an ID2D1CommandList into a fixed page
 /// representation. The print sub-system then consumes the primitives.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("2c1d867d-c290-41c8-ae7e-34a98702e9a5") ID2D1PrintControl  : public IUnknown
 {
     
@@ -890,6 +899,7 @@ interface DX_DECLARE_INTERFACE("2c1d867d-c290-41c8-ae7e-34a98702e9a5") ID2D1Prin
 /// Provides a brush that can take any effect, command list or bitmap and use it to
 /// fill a 2D shape.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("fe9e984d-3f95-407c-b5db-cb94d4e8f87c") ID2D1ImageBrush  : public ID2D1Brush
 {
     
@@ -936,6 +946,7 @@ interface DX_DECLARE_INTERFACE("fe9e984d-3f95-407c-b5db-cb94d4e8f87c") ID2D1Imag
 /// A bitmap brush allows a bitmap to be used to fill a geometry.  Interpolation
 /// mode is specified with D2D1_INTERPOLATION_MODE
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("41343a53-e41a-49a2-91cd-21793bbb62e5") ID2D1BitmapBrush1  : public ID2D1BitmapBrush
 {
     
@@ -954,6 +965,7 @@ interface DX_DECLARE_INTERFACE("41343a53-e41a-49a2-91cd-21793bbb62e5") ID2D1Bitm
 /// <summary>
 /// Extends a stroke style to allow nominal width strokes.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("10a72a66-e91c-43f4-993f-ddf4b82b0b4a") ID2D1StrokeStyle1  : public ID2D1StrokeStyle
 {
     
@@ -967,6 +979,7 @@ interface DX_DECLARE_INTERFACE("10a72a66-e91c-43f4-993f-ddf4b82b0b4a") ID2D1Stro
 /// particular, it provides the path geometry-specific
 /// ComputePointAndSegmentAtLength method.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("62baa2d2-ab54-41b7-b872-787e0106a421") ID2D1PathGeometry1  : public ID2D1PathGeometry
 {
     
@@ -1021,6 +1034,7 @@ interface DX_DECLARE_INTERFACE("62baa2d2-ab54-41b7-b872-787e0106a421") ID2D1Path
 /// Represents a set of run-time bindable and discoverable properties that allow a
 /// data-driven application to modify the state of a Direct2D effect.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("483473d7-cd46-4f9d-9d3a-3112aa80159d") ID2D1Properties  : public IUnknown
 {
     
@@ -1296,6 +1310,7 @@ interface DX_DECLARE_INTERFACE("483473d7-cd46-4f9d-9d3a-3112aa80159d") ID2D1Prop
 /// The effect interface. Properties control how the effect is rendered. The effect
 /// is Drawn with the DrawImage call.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("28211a43-7d89-476f-8181-2d6159b220ad") ID2D1Effect  : public ID2D1Properties
 {
     
@@ -1367,6 +1382,7 @@ interface DX_DECLARE_INTERFACE("28211a43-7d89-476f-8181-2d6159b220ad") ID2D1Effe
 /// Represents a bitmap that can be used as a surface for an ID2D1DeviceContext or
 /// mapped into system memory, and can contain additional color context information.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("a898a84c-3873-4588-b08b-ebbf978df041") ID2D1Bitmap1  : public ID2D1Bitmap
 {
     
@@ -1411,6 +1427,7 @@ interface DX_DECLARE_INTERFACE("a898a84c-3873-4588-b08b-ebbf978df041") ID2D1Bitm
 /// <summary>
 /// Represents a color context that can be used with an ID2D1Bitmap1 object.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("1c4820bb-5771-4518-a581-2fe4dd0ec657") ID2D1ColorContext  : public ID2D1Resource
 {
     
@@ -1440,6 +1457,7 @@ interface DX_DECLARE_INTERFACE("1c4820bb-5771-4518-a581-2fe4dd0ec657") ID2D1Colo
 /// Represents an collection of gradient stops that can then be the source resource
 /// for either a linear or radial gradient brush.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("ae1572f4-5dd0-4777-998b-9279472ae63b") ID2D1GradientStopCollection1  : public ID2D1GradientStopCollection
 {
     
@@ -1492,6 +1510,7 @@ interface DX_DECLARE_INTERFACE("ae1572f4-5dd0-4777-998b-9279472ae63b") ID2D1Grad
 /// <summary>
 /// Represents drawing state.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("689f1f85-c72e-4e33-8f19-85754efd5ace") ID2D1DrawingStateBlock1  : public ID2D1DrawingStateBlock
 {
     
@@ -1519,6 +1538,7 @@ interface DX_DECLARE_INTERFACE("689f1f85-c72e-4e33-8f19-85754efd5ace") ID2D1Draw
 /// The device context represents a set of state and a command buffer that is used
 /// to render to a target bitmap.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("e8f7fe7a-191c-466d-ad95-975678bda998") ID2D1DeviceContext  : public ID2D1RenderTarget
 {
     
@@ -2161,6 +2181,7 @@ interface DX_DECLARE_INTERFACE("e8f7fe7a-191c-466d-ad95-975678bda998") ID2D1Devi
 /// The device defines a resource domain whose objects and device contexts can be
 /// used together.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("47dd575d-ac05-4cdd-8049-9b02cd16f44c") ID2D1Device  : public ID2D1Resource
 {
     
@@ -2220,6 +2241,7 @@ interface DX_DECLARE_INTERFACE("47dd575d-ac05-4cdd-8049-9b02cd16f44c") ID2D1Devi
 /// <summary>
 /// Creates Direct2D resources.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("bb12d362-daee-4b9a-aa1d-14ba401cfa1f") ID2D1Factory1  : public ID2D1Factory
 {
     
@@ -2368,6 +2390,7 @@ interface DX_DECLARE_INTERFACE("bb12d362-daee-4b9a-aa1d-14ba401cfa1f") ID2D1Fact
 /// A locking mechanism from a Direct2D factory that Direct2D uses to control
 /// exclusive resource access in an app that is uses multiple threads.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("31e6e7bc-e0ff-4d46-8c64-a0a8c41c15d3") ID2D1Multithread  : public IUnknown
 {
     
@@ -2446,6 +2469,7 @@ extern "C"
 #endif
 
     HRESULT WINAPI 
+    _Win32_metadata_supported_os_(windows8.0)
     D2D1CreateDevice(
         _In_ IDXGIDevice *dxgiDevice,
         _In_opt_ CONST D2D1_CREATION_PROPERTIES *creationProperties,
@@ -2453,6 +2477,7 @@ extern "C"
         ); 
         
     HRESULT WINAPI 
+    _Win32_metadata_supported_os_(windows8.0)
     D2D1CreateDeviceContext(
         _In_ IDXGISurface *dxgiSurface,
         _In_opt_ CONST D2D1_CREATION_PROPERTIES *creationProperties,
