diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
index bab69516..8fe6ddcd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
@@ -5,6 +5,10 @@
 //
 // File name: D2D1_3.h
 //---------------------------------------------------------------------------
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifdef _MSC_VER
 #pragma once
 #endif // #ifdef _MSC_VER
@@ -690,6 +694,7 @@ interface DX_DECLARE_INTERFACE("b499923b-7029-478f-a8b3-432c7c5f5312") ID2D1Ink
 /// patches. Use the ID2D1DeviceContext2::CreateGradientMesh method to create an
 /// instance of ID2D1GradientMesh.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.1)
 interface DX_DECLARE_INTERFACE("f292e401-c050-4cde-83d7-04962d3b23c2") ID2D1GradientMesh  : public ID2D1Resource
 {
     
@@ -713,6 +718,7 @@ interface DX_DECLARE_INTERFACE("f292e401-c050-4cde-83d7-04962d3b23c2") ID2D1Grad
 /// <summary>
 /// Represents a producer of pixels that can fill an arbitrary 2D plane.
 /// </summary>
+_Win32_metadata_supported_os_(windows10.0.10240)
 interface DX_DECLARE_INTERFACE("c9b664e5-74a1-4378-9ac2-eefc37a3f4d8") ID2D1ImageSource  : public ID2D1Image
 {
     
@@ -793,6 +799,7 @@ interface DX_DECLARE_INTERFACE("53dd9855-a3b0-4d5b-82e1-26e25c5e5797") ID2D1Look
 /// interface, plus it enables functionality such as ink rendering, gradient mesh
 /// rendering, and improved image loading.
 /// </summary>
+_Win32_metadata_supported_os_(windows10.0.10240)
 interface DX_DECLARE_INTERFACE("394ea6a3-0c34-4321-950b-6ca20f0be6c7") ID2D1DeviceContext2  : public ID2D1DeviceContext1
 {
     
@@ -1018,6 +1025,7 @@ interface DX_DECLARE_INTERFACE("0869759f-4f00-413f-b03e-2bda45404d0f") ID2D1Fact
 /// This interface performs all the same functions as the existing ID2D1CommandSink1
 /// interface. It also enables access to ink rendering and gradient mesh rendering.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.1)
 interface DX_DECLARE_INTERFACE("3bab440e-417e-47df-a2e2-bc0be6a00916") ID2D1CommandSink2  : public ID2D1CommandSink1
 {
     
@@ -1044,6 +1052,7 @@ interface DX_DECLARE_INTERFACE("3bab440e-417e-47df-a2e2-bc0be6a00916") ID2D1Comm
 /// <summary>
 /// Interface encapsulating a GDI/GDI+ metafile.
 /// </summary>
+_Win32_metadata_supported_os_(windows10.0.10240)
 interface DX_DECLARE_INTERFACE("2e69f9e8-dd3f-4bf9-95ba-c04f49d788df") ID2D1GdiMetafile1  : public ID2D1GdiMetafile
 {
     
@@ -1068,6 +1077,7 @@ interface DX_DECLARE_INTERFACE("2e69f9e8-dd3f-4bf9-95ba-c04f49d788df") ID2D1GdiM
 /// <summary>
 /// User-implementable interface for introspecting on a metafile.
 /// </summary>
+_Win32_metadata_supported_os_(windows10.0.10240)
 interface DX_DECLARE_INTERFACE("fd0ecb6b-91e6-411e-8655-395e760f91b4") ID2D1GdiMetafileSink1  : public ID2D1GdiMetafileSink
 {
     
@@ -1933,6 +1943,7 @@ extern "C"
 
 #if NTDDI_VERSION >= NTDDI_WINTHRESHOLD
     void WINAPI 
+    _Win32_metadata_supported_os_(windows10.0.10240)
     D2D1GetGradientMeshInteriorPointsFromCoonsPatch(
         _In_ CONST D2D1_POINT_2F *pPoint0,
         _In_ CONST D2D1_POINT_2F *pPoint1,
