diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
index 9b4dc799..f0f7eb0d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
@@ -5,6 +5,10 @@
 //
 // File name: D2D1EffectAuthor.h
 //---------------------------------------------------------------------------
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifdef _MSC_VER
 #pragma once
 #endif // #ifdef _MSC_VER
@@ -422,6 +426,7 @@ EXTERN_C CONST IID IID_ID2D1EffectContext;
 /// <summary>
 /// A transform uses this interface to write new vertices to a vertex buffer.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("9b8b1336-00a5-4668-92b7-ced5d8bf9b7b") ID2D1VertexBuffer  : public IUnknown
 {
     
@@ -435,6 +440,7 @@ interface DX_DECLARE_INTERFACE("9b8b1336-00a5-4668-92b7-ced5d8bf9b7b") ID2D1Vert
 }; // interface ID2D1VertexBuffer
 
 
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("688d15c3-02b0-438d-b13a-d1b44c32c39a") ID2D1ResourceTexture  : public IUnknown
 {
     
@@ -456,6 +462,7 @@ interface DX_DECLARE_INTERFACE("688d15c3-02b0-438d-b13a-d1b44c32c39a") ID2D1Reso
 /// A transform uses this interface to specify how to render a particular pass in
 /// D2D.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("519ae1bd-d19a-420d-b849-364f594776b7") ID2D1RenderInfo  : public IUnknown
 {
     
@@ -497,6 +504,7 @@ interface DX_DECLARE_INTERFACE("519ae1bd-d19a-420d-b849-364f594776b7") ID2D1Rend
 /// A transform uses this interface to specify how to render a particular pass using
 /// pixel and vertex shaders.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("693ce632-7f2f-45de-93fe-18d88b37aa21") ID2D1DrawInfo  : public ID2D1RenderInfo
 {
     
@@ -550,6 +558,7 @@ interface DX_DECLARE_INTERFACE("693ce632-7f2f-45de-93fe-18d88b37aa21") ID2D1Draw
 /// A transform uses this interface to specify how to render a particular pass using
 /// compute shader.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("5598b14b-9fd7-48b7-9bdb-8f0964eb38bc") ID2D1ComputeInfo  : public ID2D1RenderInfo
 {
     
@@ -581,6 +590,7 @@ interface DX_DECLARE_INTERFACE("5598b14b-9fd7-48b7-9bdb-8f0964eb38bc") ID2D1Comp
 /// <summary>
 /// A base object which can be inserted into a transform graph.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("b2efe1e7-729f-4102-949f-505fa21bf666") ID2D1TransformNode  : public IUnknown
 {
     
@@ -595,6 +605,7 @@ interface DX_DECLARE_INTERFACE("b2efe1e7-729f-4102-949f-505fa21bf666") ID2D1Tran
 /// <summary>
 /// The implementation of the actual graph.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("13d29038-c3e6-4034-9081-13b53a417992") ID2D1TransformGraph  : public IUnknown
 {
     
@@ -671,6 +682,7 @@ interface DX_DECLARE_INTERFACE("13d29038-c3e6-4034-9081-13b53a417992") ID2D1Tran
 /// <summary>
 /// The interface implemented by a transform author.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("ef1a287d-342a-4f76-8fdb-da0d6ea9f92b") ID2D1Transform  : public ID2D1TransformNode
 {
     
@@ -699,6 +711,7 @@ interface DX_DECLARE_INTERFACE("ef1a287d-342a-4f76-8fdb-da0d6ea9f92b") ID2D1Tran
 /// <summary>
 /// The interface implemented by a transform author to provide a GPU-based effect.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("36bfdcb6-9739-435d-a30d-a653beff6a6f") ID2D1DrawTransform  : public ID2D1Transform
 {
     
@@ -712,6 +725,7 @@ interface DX_DECLARE_INTERFACE("36bfdcb6-9739-435d-a30d-a653beff6a6f") ID2D1Draw
 /// The interface implemented by a transform author to provide a Compute Shader
 /// based effect.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("0d85573c-01e3-4f7d-bfd9-0d60608bf3c3") ID2D1ComputeTransform  : public ID2D1Transform
 {
     
@@ -732,6 +746,7 @@ interface DX_DECLARE_INTERFACE("0d85573c-01e3-4f7d-bfd9-0d60608bf3c3") ID2D1Comp
 /// The interface implemented by a transform author to indicate that it should
 /// receive an analysis result callback.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("0359dc30-95e6-4568-9055-27720d130e93") ID2D1AnalysisTransform  : public IUnknown
 {
     
@@ -746,6 +761,7 @@ interface DX_DECLARE_INTERFACE("0359dc30-95e6-4568-9055-27720d130e93") ID2D1Anal
 /// The interface implemented by a transform author to provide a CPU based source
 /// effect.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("db1800dd-0c34-4cf9-be90-31cc0a5653e1") ID2D1SourceTransform  : public ID2D1Transform
 {
     
@@ -765,6 +781,7 @@ interface DX_DECLARE_INTERFACE("db1800dd-0c34-4cf9-be90-31cc0a5653e1") ID2D1Sour
 /// Base interface for built-in transforms on which precision and caching may be
 /// controlled.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("1a799d8a-69f7-4e4c-9fed-437ccc6684cc") ID2D1ConcreteTransform  : public ID2D1TransformNode
 {
     
@@ -788,6 +805,7 @@ interface DX_DECLARE_INTERFACE("1a799d8a-69f7-4e4c-9fed-437ccc6684cc") ID2D1Conc
 /// <summary>
 /// An effect uses this interface to configure a blending operation.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("63ac0b32-ba44-450f-8806-7f4ca1ff2f1b") ID2D1BlendTransform  : public ID2D1ConcreteTransform
 {
     
@@ -804,6 +822,7 @@ interface DX_DECLARE_INTERFACE("63ac0b32-ba44-450f-8806-7f4ca1ff2f1b") ID2D1Blen
 /// <summary>
 /// An effect uses this interface to configure border generation.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("4998735c-3a19-473c-9781-656847e3a347") ID2D1BorderTransform  : public ID2D1ConcreteTransform
 {
     
@@ -827,6 +846,7 @@ interface DX_DECLARE_INTERFACE("4998735c-3a19-473c-9781-656847e3a347") ID2D1Bord
 /// An effect uses this interface to offset an image without inserting a rendering
 /// pass.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("3fe6adea-7643-4f53-bd14-a0ce63f24042") ID2D1OffsetTransform  : public ID2D1TransformNode
 {
     
@@ -859,6 +879,7 @@ interface DX_DECLARE_INTERFACE("90f732e2-5092-4606-a819-8651970baccd") ID2D1Boun
 /// This is the interface implemented by an effect author, along with the
 /// constructor and registration information.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("a248fd3f-3e6c-4e63-9f03-7f68ecc91db9") ID2D1EffectImpl  : public IUnknown
 {
     
@@ -894,6 +915,7 @@ interface DX_DECLARE_INTERFACE("a248fd3f-3e6c-4e63-9f03-7f68ecc91db9") ID2D1Effe
 /// and any other operation tied to context which is not useful to the application
 /// facing API.
 /// </summary>
+_Win32_metadata_supported_os_(windows8.0)
 interface DX_DECLARE_INTERFACE("3d9f916b-27dc-4ad7-b4f1-64945340f563") ID2D1EffectContext  : public IUnknown
 {
     
