diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_1.h b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_1.h
index d05dd308..f6a6f628 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_1.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_1.h
@@ -9,8 +9,13 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
+
 #endif
 
 /* verify that the <rpcsal.h> version is high enough to compile this file*/
@@ -226,6 +231,7 @@ EXTERN_C const IID IID_ID3D11BlendState1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("cc86fabe-da55-401d-85e7-e3c9de2877e9")
     ID3D11BlendState1 : public ID3D11BlendState
     {
@@ -444,6 +450,7 @@ EXTERN_C const IID IID_ID3D11RasterizerState1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("1217d7a6-5039-418c-b042-9cbe256afd6e")
     ID3D11RasterizerState1 : public ID3D11RasterizerState
     {
@@ -602,6 +609,7 @@ EXTERN_C const IID IID_ID3DDeviceContextState;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("5c1e0d8a-7c23-48f9-8c59-a92958ceff11")
     ID3DDeviceContextState : public ID3D11DeviceChild
     {
@@ -723,6 +731,7 @@ EXTERN_C const IID IID_ID3D11DeviceContext1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("bb2c6faa-b5fb-4082-8e6b-388b8cfa90e1")
     ID3D11DeviceContext1 : public ID3D11DeviceContext
     {
@@ -2808,6 +2817,7 @@ EXTERN_C const IID IID_ID3D11VideoContext1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("A7F026DA-A5F8-4487-A564-15E34357651E")
     ID3D11VideoContext1 : public ID3D11VideoContext
     {
@@ -4084,6 +4094,7 @@ EXTERN_C const IID IID_ID3D11VideoDevice1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("29DA1D51-1321-4454-804B-F5FC9F861F0F")
     ID3D11VideoDevice1 : public ID3D11VideoDevice
     {
@@ -4504,6 +4515,7 @@ EXTERN_C const IID IID_ID3D11VideoProcessorEnumerator1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("465217F2-5568-43CF-B5B9-F61D54531CA1")
     ID3D11VideoProcessorEnumerator1 : public ID3D11VideoProcessorEnumerator
     {
@@ -4719,6 +4731,7 @@ EXTERN_C const IID IID_ID3D11Device1;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("a04bfb29-08ef-43d6-a49c-a9bdbdcbe686")
     ID3D11Device1 : public ID3D11Device
     {
@@ -5427,6 +5440,7 @@ EXTERN_C const IID IID_ID3DUserDefinedAnnotation;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("b2daad8b-03d4-4dbf-95eb-32ab4b63d0ab")
     ID3DUserDefinedAnnotation : public IUnknown
     {
