diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/datetimeapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/datetimeapi.h
index 57f76415..45b14b44 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/datetimeapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/datetimeapi.h
@@ -26,6 +26,7 @@ extern "C" {
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // For Windows Vista and above GetDateFormatEx is preferred
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -38,6 +39,7 @@ GetDateFormatA(
     _In_ int cchDate
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -57,6 +59,7 @@ GetDateFormatW(
 #endif // !UNICODE
 
 // For Windows Vista and above GetTimeFormatEx is preferred
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -69,6 +72,7 @@ GetTimeFormatA(
     _In_ int cchTime
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -93,6 +97,7 @@ GetTimeFormatW(
 #pragma region Application Family or OneCore or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -105,6 +110,7 @@ GetTimeFormatEx(
     _In_ int cchTime
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -126,6 +132,7 @@ GetDateFormatEx(
 
 #define GetDurationFormatEx_DEFINED
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
