diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dde.h b/generation/WinSDK/RecompiledIdlHeaders/um/dde.h
index 9e683906..c1be14a2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dde.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dde.h
@@ -17,6 +17,7 @@
 #endif
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #if _MSC_VER >= 1200
 #pragma warning(push)
@@ -177,6 +178,8 @@ typedef struct {
  * DDE SECURITY
  */
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DdeSetQualityOfService(
@@ -184,6 +187,8 @@ DdeSetQualityOfService(
     CONST SECURITY_QUALITY_OF_SERVICE *pqosNew,
     PSECURITY_QUALITY_OF_SERVICE pqosPrev);
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 ImpersonateDdeClientWindow(
@@ -193,9 +198,13 @@ ImpersonateDdeClientWindow(
 /*
  * DDE message packing APIs
  */
+_Win32_metadata_supported_os_(windows5.0)
 LPARAM APIENTRY PackDDElParam(_In_ UINT msg, _In_ UINT_PTR uiLo, _In_ UINT_PTR uiHi);
+_Win32_metadata_supported_os_(windows5.0)
 BOOL   APIENTRY UnpackDDElParam(_In_ UINT msg, _In_ LPARAM lParam, _Out_ PUINT_PTR puiLo, _Out_ PUINT_PTR puiHi);
+_Win32_metadata_supported_os_(windows5.0)
 BOOL   APIENTRY FreeDDElParam(_In_ UINT msg, _In_ LPARAM lParam);
+_Win32_metadata_supported_os_(windows5.0)
 LPARAM APIENTRY ReuseDDElParam(LPARAM lParam, UINT msgIn, UINT msgOut, UINT_PTR uiLo, UINT_PTR uiHi);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
