diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DirectML.h b/generation/WinSDK/RecompiledIdlHeaders/um/DirectML.h
index 3bd1921b..5039624e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DirectML.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DirectML.h
@@ -2,6 +2,7 @@
 
 #ifndef DIRECTML_H
 #define DIRECTML_H
+#include <win32metadata_annotations.h>
 #pragma once
 
 #ifdef _GAMING_XBOX_SCARLETT
@@ -2284,6 +2285,7 @@ enum DML_CREATE_DEVICE_FLAGS
 
 DEFINE_ENUM_FLAG_OPERATORS(DML_CREATE_DEVICE_FLAGS)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI DMLCreateDevice(
     ID3D12Device* d3d12Device,
     DML_CREATE_DEVICE_FLAGS flags,
