diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DSRole.h b/generation/WinSDK/RecompiledIdlHeaders/um/DSRole.h
index 2f506077..b287a560 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DSRole.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DSRole.h
@@ -20,6 +20,7 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -107,6 +108,7 @@ typedef struct _DSROLE_OPERATION_STATE_INFO {
 
 } DSROLE_OPERATION_STATE_INFO, *PDSROLE_OPERATION_STATE_INFO;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 DsRoleGetPrimaryDomainInformation(
@@ -115,6 +117,7 @@ DsRoleGetPrimaryDomainInformation(
     OUT PBYTE *Buffer 
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 VOID
 WINAPI
 DsRoleFreeMemory(
