diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dssec.h b/generation/WinSDK/RecompiledIdlHeaders/um/dssec.h
index d504431e..5547c8e3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dssec.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dssec.h
@@ -4,6 +4,7 @@
 #ifndef _DSSEC_H_
 #define _DSSEC_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -59,6 +60,7 @@ typedef HRESULT (WINAPI *PFNWRITEOBJECTSECURITY)(LPCWSTR, SECURITY_INFORMATION,
 //  Return:     HRESULT
 //
 //----------------------------------------------------------------------------
+_Win32_metadata_supported_os_(windowsserver2008)
 STDAPI
 DSCreateISecurityInfoObject(LPCWSTR pwszObjectPath,
                             LPCWSTR pwszObjectClass,
@@ -83,6 +85,7 @@ DSCreateISecurityInfoObject(LPCWSTR pwszObjectPath,
 // If the server is not provided, it is obtained from the object
 // path or DsGetDcName.
 //
+_Win32_metadata_supported_os_(windowsserver2008)
 STDAPI
 DSCreateISecurityInfoObjectEx(LPCWSTR pwszObjectPath,
                               LPCWSTR pwszObjectClass,
@@ -113,6 +116,7 @@ DSCreateISecurityInfoObjectEx(LPCWSTR pwszObjectPath,
 //  Return:     HRESULT
 //
 //----------------------------------------------------------------------------
+_Win32_metadata_supported_os_(windowsserver2003)
 STDAPI
 DSCreateSecurityPage(LPCWSTR pwszObjectPath,
                      LPCWSTR pwszObjectClass,
@@ -140,6 +144,7 @@ DSCreateSecurityPage(LPCWSTR pwszObjectPath,
 //  Return:     HRESULT
 //
 //----------------------------------------------------------------------------
+_Win32_metadata_supported_os_(windowsserver2008)
 STDAPI
 DSEditSecurity(HWND hwndOwner,
                LPCWSTR pwszObjectPath,
