diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DvbSiParser.h b/generation/WinSDK/RecompiledIdlHeaders/um/DvbSiParser.h
index 14c03472..b7c0645b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DvbSiParser.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DvbSiParser.h
@@ -553,6 +553,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 typedef /* [public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public][public] */ 
@@ -931,6 +932,7 @@ EXTERN_C const IID IID_IDvbSiParser2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("0AC5525F-F816-42F4-93BA-4C0F32F46E54")
     IDvbSiParser2 : public IDvbSiParser
     {
@@ -2294,6 +2296,7 @@ EXTERN_C const IID IID_IISDB_SDT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("3F3DC9A2-BB32-4FB9-AE9E-D856848927A3")
     IISDB_SDT : public IDVB_SDT
     {
@@ -2961,6 +2964,7 @@ EXTERN_C const IID IID_IDVB_EIT2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("61A389E0-9B9E-4ba0-AEEA-5DDD159820EA")
     IDVB_EIT2 : public IDVB_EIT
     {
@@ -4573,6 +4577,7 @@ EXTERN_C const IID IID_IISDB_BIT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("537CD71E-0E46-4173-9001-BA043F3E49E2")
     IISDB_BIT : public IUnknown
     {
@@ -4860,6 +4865,7 @@ EXTERN_C const IID IID_IISDB_NBIT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("1B1863EF-08F1-40B7-A559-3B1EFF8CAFA6")
     IISDB_NBIT : public IUnknown
     {
@@ -5185,6 +5191,7 @@ EXTERN_C const IID IID_IISDB_LDT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("141A546B-02FF-4FB9-A3A3-2F074B74A9A9")
     IISDB_LDT : public IUnknown
     {
@@ -5434,6 +5441,7 @@ EXTERN_C const IID IID_IISDB_SDTT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("EE60EF2D-813A-4DC7-BF92-EA13DAC85313")
     IISDB_SDTT : public IUnknown
     {
@@ -5840,6 +5848,7 @@ EXTERN_C const IID IID_IISDB_CDT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("25FA92C2-8B80-4787-A841-3A0E8F17984B")
     IISDB_CDT : public IUnknown
     {
@@ -6090,6 +6099,7 @@ EXTERN_C const IID IID_IISDB_EMM;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("0EDB556D-43AD-4938-9668-321B2FFECFD3")
     IISDB_EMM : public IUnknown
     {
@@ -6446,6 +6456,7 @@ EXTERN_C const IID IID_IDvbContentIdentifierDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("05E0C1EA-F661-4053-9FBF-D93B28359838")
     IDvbContentIdentifierDescriptor : public IUnknown
     {
@@ -6588,6 +6599,7 @@ EXTERN_C const IID IID_IDvbDefaultAuthorityDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("05EC24D1-3A31-44e7-B408-67C60A352276")
     IDvbDefaultAuthorityDescriptor : public IUnknown
     {
@@ -7664,6 +7676,7 @@ EXTERN_C const IID IID_IDvbPrivateDataSpecifierDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5660A019-E75A-4b82-9B4C-ED2256D165A2")
     IDvbPrivateDataSpecifierDescriptor : public IUnknown
     {
@@ -7924,6 +7937,7 @@ EXTERN_C const IID IID_IDvbLogicalChannelDescriptor2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("43ACA974-4BE8-4b98-BC17-9EAFD788B1D7")
     IDvbLogicalChannelDescriptor2 : public IDvbLogicalChannelDescriptor
     {
@@ -8065,6 +8079,7 @@ EXTERN_C const IID IID_IDvbLogicalChannel2Descriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("F69C3747-8A30-4980-998C-01FE7F0BA35A")
     IDvbLogicalChannel2Descriptor : public IDvbLogicalChannelDescriptor2
     {
@@ -8349,6 +8364,7 @@ EXTERN_C const IID IID_IDvbHDSimulcastLogicalChannelDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("1EA8B738-A307-4680-9E26-D0A908C824F4")
     IDvbHDSimulcastLogicalChannelDescriptor : public IDvbLogicalChannelDescriptor2
     {
@@ -8485,6 +8501,7 @@ EXTERN_C const IID IID_IDvbDataBroadcastIDDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5F26F518-65C8-4048-91F2-9290F59F7B90")
     IDvbDataBroadcastIDDescriptor : public IUnknown
     {
@@ -8615,6 +8632,7 @@ EXTERN_C const IID IID_IDvbDataBroadcastDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("D1EBC1D6-8B60-4c20-9CAF-E59382E7C400")
     IDvbDataBroadcastDescriptor : public IUnknown
     {
@@ -8837,6 +8855,7 @@ EXTERN_C const IID IID_IDvbLinkageDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("1CDF8B31-994A-46fc-ACFD-6A6BE8934DD5")
     IDvbLinkageDescriptor : public IUnknown
     {
@@ -9019,6 +9038,7 @@ EXTERN_C const IID IID_IDvbTeletextDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("9CD29D47-69C6-4f92-98A9-210AF1B7303A")
     IDvbTeletextDescriptor : public IUnknown
     {
@@ -9200,6 +9220,7 @@ EXTERN_C const IID IID_IDvbSubtitlingDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("9B25FE1D-FA23-4e50-9784-6DF8B26F8A49")
     IDvbSubtitlingDescriptor : public IUnknown
     {
@@ -9559,6 +9580,7 @@ EXTERN_C const IID IID_IDvbServiceDescriptor2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("D6C76506-85AB-487C-9B2B-36416511E4A2")
     IDvbServiceDescriptor2 : public IDvbServiceDescriptor
     {
@@ -9740,6 +9762,7 @@ EXTERN_C const IID IID_IDvbServiceListDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("05DB0D8F-6008-491a-ACD3-7090952707D0")
     IDvbServiceListDescriptor : public IUnknown
     {
@@ -9887,6 +9910,7 @@ EXTERN_C const IID IID_IDvbMultilingualServiceNameDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("2D80433B-B32C-47ef-987F-E78EBB773E34")
     IDvbMultilingualServiceNameDescriptor : public IUnknown
     {
@@ -10059,6 +10083,7 @@ EXTERN_C const IID IID_IDvbNetworkNameDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5B2A80CF-35B9-446C-B3E4-048B761DBC51")
     IDvbNetworkNameDescriptor : public IUnknown
     {
@@ -10189,6 +10214,7 @@ EXTERN_C const IID IID_IDvbShortEventDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("B170BE92-5B75-458E-9C6E-B0008231491A")
     IDvbShortEventDescriptor : public IUnknown
     {
@@ -10336,6 +10362,7 @@ EXTERN_C const IID IID_IDvbExtendedEventDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("C9B22ECA-85F4-499F-B1DB-EFA93A91EE57")
     IDvbExtendedEventDescriptor : public IUnknown
     {
@@ -10597,6 +10624,7 @@ EXTERN_C const IID IID_IDvbComponentDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("91E405CF-80E7-457F-9096-1B9D1CE32141")
     IDvbComponentDescriptor : public IUnknown
     {
@@ -10766,6 +10794,7 @@ EXTERN_C const IID IID_IDvbContentDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("2E883881-A467-412A-9D63-6F2B6DA05BF0")
     IDvbContentDescriptor : public IUnknown
     {
@@ -10921,6 +10950,7 @@ EXTERN_C const IID IID_IDvbParentalRatingDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("3AD9DDE1-FB1B-4186-937F-22E6B5A72A10")
     IDvbParentalRatingDescriptor : public IUnknown
     {
@@ -11055,6 +11085,7 @@ EXTERN_C const IID IID_IIsdbTerrestrialDeliverySystemDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("39FAE0A6-D151-44DD-A28A-765DE5991670")
     IIsdbTerrestrialDeliverySystemDescriptor : public IUnknown
     {
@@ -11224,6 +11255,7 @@ EXTERN_C const IID IID_IIsdbTSInformationDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("D7AD183E-38F5-4210-B55F-EC8D601BBD47")
     IIsdbTSInformationDescriptor : public IUnknown
     {
@@ -11422,6 +11454,7 @@ EXTERN_C const IID IID_IIsdbDigitalCopyControlDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("1A28417E-266A-4BB8-A4BD-D782BCFB8161")
     IIsdbDigitalCopyControlDescriptor : public IUnknown
     {
@@ -11593,6 +11626,7 @@ EXTERN_C const IID IID_IIsdbAudioComponentDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("679D2002-2425-4BE4-A4C7-D6632A574F4D")
     IIsdbAudioComponentDescriptor : public IUnknown
     {
@@ -11853,6 +11887,7 @@ EXTERN_C const IID IID_IIsdbDataContentDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("A428100A-E646-4BD6-AA14-6087BDC08CD5")
     IIsdbDataContentDescriptor : public IUnknown
     {
@@ -12069,6 +12104,7 @@ EXTERN_C const IID IID_IIsdbCAContractInformationDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("08E18B25-A28F-4E92-821E-4FCED5CC2291")
     IIsdbCAContractInformationDescriptor : public IUnknown
     {
@@ -12272,6 +12308,7 @@ EXTERN_C const IID IID_IIsdbEventGroupDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("94B06780-2E2A-44DC-A966-CC56FDABC6C2")
     IIsdbEventGroupDescriptor : public IUnknown
     {
@@ -12461,6 +12498,7 @@ EXTERN_C const IID IID_IIsdbComponentGroupDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("A494F17F-C592-47D8-8943-64C9A34BE7B9")
     IIsdbComponentGroupDescriptor : public IUnknown
     {
@@ -12726,6 +12764,7 @@ EXTERN_C const IID IID_IIsdbSeriesDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("07EF6370-1660-4F26-87FC-614ADAB24B11")
     IIsdbSeriesDescriptor : public IUnknown
     {
@@ -12925,6 +12964,7 @@ EXTERN_C const IID IID_IIsdbDownloadContentDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5298661E-CB88-4F5F-A1DE-5F440C185B92")
     IIsdbDownloadContentDescriptor : public IUnknown
     {
@@ -13256,6 +13296,7 @@ EXTERN_C const IID IID_IIsdbLogoTransmissionDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("E0103F49-4AE1-4F07-9098-756DB1FA88CD")
     IIsdbLogoTransmissionDescriptor : public IUnknown
     {
@@ -13425,6 +13466,7 @@ EXTERN_C const IID IID_IIsdbSIParameterDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("F837DC36-867C-426a-9111-F62093951A45")
     IIsdbSIParameterDescriptor : public IUnknown
     {
@@ -13619,6 +13661,7 @@ EXTERN_C const IID IID_IIsdbEmergencyInformationDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("BA6FA681-B973-4da1-9207-AC3E7F0341EB")
     IIsdbEmergencyInformationDescriptor : public IUnknown
     {
@@ -13804,6 +13847,7 @@ EXTERN_C const IID IID_IIsdbCADescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("0570AA47-52BC-42ae-8CA5-969F41E81AEA")
     IIsdbCADescriptor : public IUnknown
     {
@@ -13960,6 +14004,7 @@ EXTERN_C const IID IID_IIsdbCAServiceDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("39CBEB97-FF0B-42a7-9AB9-7B9CFE70A77A")
     IIsdbCAServiceDescriptor : public IUnknown
     {
@@ -14116,6 +14161,7 @@ EXTERN_C const IID IID_IIsdbHierarchicalTransmissionDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("B7B3AE90-EE0B-446d-8769-F7E2AA266AA6")
     IIsdbHierarchicalTransmissionDescriptor : public IUnknown
     {
@@ -14282,6 +14328,7 @@ EXTERN_C const IID IID_IPBDASiParser;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("9DE49A74-ABA2-4a18-93E1-21F17F95C3C3")
     IPBDASiParser : public IUnknown
     {
@@ -14411,6 +14458,7 @@ EXTERN_C const IID IID_IPBDA_EIT;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("A35F2DEA-098F-4ebd-984C-2BD4C3C8CE0A")
     IPBDA_EIT : public IUnknown
     {
@@ -14668,6 +14716,7 @@ EXTERN_C const IID IID_IPBDA_Services;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("944EAB37-EED4-4850-AFD2-77E7EFEB4427")
     IPBDA_Services : public IUnknown
     {
@@ -14789,6 +14838,7 @@ EXTERN_C const IID IID_IPBDAEntitlementDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("22632497-0DE3-4587-AADC-D8D99017E760")
     IPBDAEntitlementDescriptor : public IUnknown
     {
@@ -14906,6 +14956,7 @@ EXTERN_C const IID IID_IPBDAAttributesDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("313B3620-3263-45a6-9533-968BEFBEAC03")
     IPBDAAttributesDescriptor : public IUnknown
     {
