diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dxcore_interface.h b/generation/WinSDK/RecompiledIdlHeaders/um/dxcore_interface.h
index c3cc962f..e00c72df 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dxcore_interface.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dxcore_interface.h
@@ -6,6 +6,7 @@
 
 #ifndef __dxcore_interface_h__
 #define __dxcore_interface_h__
+#include <win32metadata_annotations.h>
 
 #ifndef COM_NO_WINDOWS_H
 #include "windows.h"
