diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dxvahd.h b/generation/WinSDK/RecompiledIdlHeaders/um/dxvahd.h
index 5835b704..7725086f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dxvahd.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dxvahd.h
@@ -73,6 +73,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if 0
@@ -510,6 +511,7 @@ EXTERN_C const IID IID_IDXVAHD_Device;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("95f12dfd-d77e-49be-815f-57d579634d6d")
     IDXVAHD_Device : public IUnknown
     {
@@ -748,6 +750,7 @@ EXTERN_C const IID IID_IDXVAHD_VideoProcessor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("95f4edf4-6e03-4cd7-be1b-3075d665aa52")
     IDXVAHD_VideoProcessor : public IUnknown
     {
@@ -1132,6 +1135,7 @@ typedef struct _DXVAHDETW_DESTROYVIDEOPROCESSOR
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT WINAPI
 DXVAHD_CreateDevice(
     _In_ IDirect3DDevice9Ex* pD3DDevice,
