diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dxva2api.h b/generation/WinSDK/RecompiledIdlHeaders/um/dxva2api.h
index e9a4cb2e..2e92a00c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dxva2api.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dxva2api.h
@@ -108,6 +108,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 //
 //   The following declarations within the 'if 0' block are dummy typedefs used to make
 //   the evr.idl file build. The actual definitions are contained in d3d9.h
@@ -670,6 +671,7 @@ EXTERN_C const IID IID_IDirect3DDeviceManager9;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("a0cade0f-06d5-4cf4-a1c7-f3cdd725aa75")
     IDirect3DDeviceManager9 : public IUnknown
     {
@@ -859,6 +861,7 @@ EXTERN_C const IID IID_IDirectXVideoAccelerationService;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fc51a550-d5e7-11d9-af55-00054e43ff02")
     IDirectXVideoAccelerationService : public IUnknown
     {
@@ -978,6 +981,7 @@ EXTERN_C const IID IID_IDirectXVideoDecoderService;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fc51a551-d5e7-11d9-af55-00054e43ff02")
     IDirectXVideoDecoderService : public IDirectXVideoAccelerationService
     {
@@ -1178,6 +1182,7 @@ EXTERN_C const IID IID_IDirectXVideoProcessorService;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fc51a552-d5e7-11d9-af55-00054e43ff02")
     IDirectXVideoProcessorService : public IDirectXVideoAccelerationService
     {
@@ -1478,6 +1483,7 @@ EXTERN_C const IID IID_IDirectXVideoDecoder;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("f2b0810a-fd00-43c9-918c-df94e2d8ef7d")
     IDirectXVideoDecoder : public IUnknown
     {
@@ -1671,6 +1677,7 @@ EXTERN_C const IID IID_IDirectXVideoProcessor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8c3a39f0-916e-4690-804f-4c8001355d25")
     IDirectXVideoProcessor : public IUnknown
     {
@@ -1875,6 +1882,7 @@ EXTERN_C const IID IID_IDirectXVideoMemoryConfiguration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("b7f916dd-db3b-49c1-84d7-e45ef99ec726")
     IDirectXVideoMemoryConfiguration : public IUnknown
     {
@@ -1971,16 +1979,19 @@ EXTERN_C const IID IID_IDirectXVideoMemoryConfiguration;
 /* interface __MIDL_itf_dxva2api_0000_0007 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI DXVA2CreateDirect3DDeviceManager9(
     _Out_ UINT* pResetToken,
     _Outptr_ IDirect3DDeviceManager9** ppDeviceManager
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI DXVA2CreateVideoService(
     _In_ IDirect3DDevice9* pDD,
     _In_ REFIID riid,
     _Outptr_ void** ppService
     );
 #endif  // _D3D9_H_
+_Win32_metadata_supported_os_(windows6.0.6000)
 __inline DXVA2_Fixed32 DXVA2FloatToFixed(_In_ const float _float_)
 {
     DXVA2_Fixed32 _fixed_;
@@ -1988,15 +1999,18 @@ __inline DXVA2_Fixed32 DXVA2FloatToFixed(_In_ const float _float_)
     _fixed_.Value = HIWORD(_float_ * 0x10000);
     return _fixed_;
 }
+_Win32_metadata_supported_os_(windows6.0.6000)
 __inline float DXVA2FixedToFloat(_In_ const DXVA2_Fixed32 _fixed_)
 {
     return (FLOAT)_fixed_.Value + (FLOAT)_fixed_.Fraction / 0x10000;
 }
+_Win32_metadata_supported_os_(windows6.0.6000)
 __inline const DXVA2_Fixed32 DXVA2_Fixed32TransparentAlpha()
 {
     DXVA2_Fixed32 _fixed_ = {0, 0};
     return _fixed_;
 }
+_Win32_metadata_supported_os_(windows6.0.6000)
 __inline const DXVA2_Fixed32 DXVA2_Fixed32OpaqueAlpha()
 {
     DXVA2_Fixed32 _fixed_ = {0, 1};
