diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/eaphostpeerconfigapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/eaphostpeerconfigapis.h
index f7e4b131..f9392d21 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/eaphostpeerconfigapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/eaphostpeerconfigapis.h
@@ -12,6 +12,9 @@
 #define EAPHOSTPEERCONFIGAPIS_H
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -28,6 +31,7 @@ extern "C"
 
     // This API is used to enumerate all the EAP Methods installed and available for use; this
     // includes legacy EAP Methods too. Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerGetMethods(
         // API fills the structure to have installed EAP Methods. Caller should free the inner pointers
         // using EapHostPeerFreeMemory starting at the inner most pointer.
@@ -37,6 +41,7 @@ extern "C"
 
     // This API is used to retrieve the method properties for a specific configuration
     // It returns non zero return code upon error.
+_Win32_metadata_supported_os_(windows6.1)
     DWORD WINAPI EapHostPeerGetMethodProperties(
         // The version number of the API.
         _In_ DWORD dwVersion,
@@ -63,6 +68,7 @@ extern "C"
 
     // This API is used to invoke configure UI of the specified EAP Method.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerInvokeConfigUI(
         // window handle of the parent window under which configuration dialog will show up
         _In_ HWND hwndParent,
@@ -85,6 +91,7 @@ extern "C"
     // This API is used to obtain the fields to be shown in Single Sign On scenario for
     // showing UI in long screen. The structures returned have details on how to show the fields.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerQueryCredentialInputFields(
         // handle to the impersonation token that is used while single sign on.
         _In_ HANDLE hUserImpersonationToken,
@@ -106,6 +113,7 @@ extern "C"
     // Supplicant up on receiving user inputs from Single Sign On UI, calls this API to obtain
     // credential blob that could used start authentication
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerQueryUserBlobFromCredentialInputFields(
         // handle to the impersonation token that is used while single sign on.
         _In_ HANDLE hUserImpersonationToken,
@@ -133,6 +141,7 @@ extern "C"
 
     // Tunnel Methods call this function to invoke Identity UI of the inner Methods.
     // This function returns the identity as well as the credentials to use to start the authentication.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerInvokeIdentityUI(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -167,6 +176,7 @@ extern "C"
     // supplicant should call EapHostPeerGetUIContext to get UI context data. After that. it then should call this
     // API from a process where UI can be brought up.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerInvokeInteractiveUI(
         // window handle of the parent window under which dialog will show up
         _In_ HWND hwndParent,
@@ -188,6 +198,7 @@ extern "C"
     // EapHostPeerResponseInvokeUi actionCode from EAPHost. If the return value from this API is
     // EAP_E_EAPHOST_METHOD_OPERATION_NOT_SUPPORTED, then supplicant should fall back to traditional model of invoking method
     // interactive UI, i.e. by calling EapHostPeerInvokeInteractiveUI() API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerQueryInteractiveUIInputFields(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -207,6 +218,7 @@ extern "C"
 
     // This API will enable supplicants to convert user information into a user-blob which can be consumed by EAPHost runtime APIs
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerQueryUIBlobFromInteractiveUIInputFields(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -233,6 +245,7 @@ extern "C"
     // Supplicant can utilize XML based EAP configuration storing, managing, editing. When it wants to call EapHostPeerConfigureUI
     // or when it has to start authentication, it calls this API to convert XML configuration into blob.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerConfigXml2Blob(
         // controls the behavior of the EAP Method
         _In_ DWORD dwFlags,
@@ -250,6 +263,7 @@ extern "C"
     // Supplicant can utilize XML based EAP credentials storing, managing, editing. When it wants to start
     // authentication, it calls this API to convert XML based credentials into blob.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerCredentialsXml2Blob(
         // controls the behavior of the EAP Method
         _In_ DWORD dwFlags,
@@ -272,6 +286,7 @@ extern "C"
     // Supplicant can utilize XML based EAP credentials storing, managing, editing. When it wants to start
     // convert configuration blob to XML, it can call this API.
     // API Returns non zero return code up on error.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapHostPeerConfigBlob2Xml(
         // controls the behavior of the EAP Method
         _In_ DWORD dwFlags,
@@ -289,9 +304,11 @@ extern "C"
 
     // Supplicant calls this API to free memory returned by Config APIs. Supplicant should not use this API
     // for freeing EAP_ERROR structure.
+_Win32_metadata_supported_os_(windows6.0.6000)
     VOID WINAPI EapHostPeerFreeMemory(BYTE* pData);
 
     // Supplicant calls this API to free EAP_ERROR memory, which will get filled when an API fails.
+_Win32_metadata_supported_os_(windows6.0.6000)
     VOID WINAPI EapHostPeerFreeErrorMemory(EAP_ERROR* pEapError);
 
 #ifdef __cplusplus
