diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodpeerapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodpeerapis.h
index 3c67fdeb..22d33921 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodpeerapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodpeerapis.h
@@ -12,6 +12,9 @@
 #define EAPPEERMETHODAPIS_H
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -218,6 +221,7 @@ extern "C"
     //
     // If any other APIs in this file returns EAP_ERROR, use this
     // function to free the memory allocated for it
+_Win32_metadata_supported_os_(windows6.0.6000)
     VOID WINAPI EapPeerFreeErrorMemory(_In_ EAP_ERROR* pEapError);
 
     // This function should be an exported function from the DLL
@@ -230,6 +234,7 @@ extern "C"
     //
     // - pEapInfo:
     //   returns all the information related for pEapType.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetInfo(_In_ EAP_TYPE* pEapType, _Out_ EAP_PEER_METHOD_ROUTINES* pEapInfo, _Out_opt_ EAP_ERROR** ppEapError);
 
     // This is the first function that EAPHost should call on this method.
@@ -237,6 +242,7 @@ extern "C"
     // which can be called at any time.
     // This function should be called only once and it should initialize
     // everything needed for this method work properly.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerInitialize(_Outptr_ EAP_ERROR** ppEapError);
 
     // EAPHost will call this function, and it would return user data
@@ -283,6 +289,7 @@ extern "C"
     // - ppwszIdentity:
     //   returned user identity. It will be included in the identity response packet
     //   and returned to the server.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetIdentity(
         _In_ DWORD dwFlags,
         _In_ DWORD dwSizeofConnectionData,
@@ -330,6 +337,7 @@ extern "C"
     //   being called, *pSessionHandle will be passed as the session handle.
     //   The method should be able to use this to find the information related
     //   the authentication session.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerBeginSession(
         _In_ DWORD dwFlags,
         _In_ const EapAttributes* const pAttributeArray,
@@ -345,6 +353,7 @@ extern "C"
     // A method exports either EapPeerGetIdentity (and EapPeerInvokeIdentityUI) or
     // exports EapPeerSetCredentials (and sets the InvokeUserNameDlg regkey). The
     // registry key controls which of the two apis will get called.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerSetCredentials(EAP_SESSION_HANDLE sessionHandle, LPWSTR pwszIdentity, LPWSTR pwszPassword, _Outptr_ EAP_ERROR** ppEapError);
 
     // EAPHost will pass the packet to the method for processing.
@@ -360,6 +369,7 @@ extern "C"
     //
     // - pEapOutput:
     //   the method should fill this struct to tell the supplicant what to do.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerProcessRequestPacket(
         _In_ EAP_SESSION_HANDLE sessionHandle,
         _In_ DWORD cbReceivedPacket,
@@ -378,6 +388,7 @@ extern "C"
     //
     // - pSendPacket:
     //   pointer to a buffer that contains the packet to send upon return
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetResponsePacket(
         _In_ EAP_SESSION_HANDLE sessionHandle,
         _Inout_ _Pre_satisfies_(*pcbSendPacket >= sizeof(EapPacket)) DWORD* pcbSendPacket,
@@ -396,6 +407,7 @@ extern "C"
     // - ppResult:
     //   Method should fill this struct with related information as defined
     //   by EapPeerMethodResult
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetResult(
         _In_ EAP_SESSION_HANDLE sessionHandle,
         _In_ EapPeerMethodResultReason reason,
@@ -415,6 +427,7 @@ extern "C"
     // - ppUIContextData:
     //   It will be passed to EapPeerInvokeInteractiveUI() as _In_ parameter
     //   pUIContextData
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetUIContext(
         _In_ EAP_SESSION_HANDLE sessionHandle,
         _Out_ DWORD* pdwSizeOfUIContextData,
@@ -436,6 +449,7 @@ extern "C"
     //
     // - pEapOutput:
     //   the method should fill this struct to tell the supplicant what to do.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerSetUIContext(
         _In_ EAP_SESSION_HANDLE sessionHandle,
         _In_ DWORD dwSizeOfUIContextData,
@@ -454,6 +468,7 @@ extern "C"
     //
     // - pAttributes:
     //   method should fill this with the attributes it need to return.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerGetResponseAttributes(_In_ EAP_SESSION_HANDLE sessionHandle, _Out_ EapAttributes* pAttribs, _Outptr_ EAP_ERROR** ppEapError);
 
     // See EapPeerGetResponseAttributes() for when this function will be called.
@@ -466,6 +481,7 @@ extern "C"
     //
     // - pEapOutput:
     //   the method should fill this struct to tell the supplicant what to do.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerSetResponseAttributes(
         _In_ EAP_SESSION_HANDLE sessionHandle, _In_ EapAttributes* pAttribs, _Out_ EapPeerMethodOutput* pEapOutput, _Outptr_ EAP_ERROR** ppEapError);
 
@@ -473,6 +489,7 @@ extern "C"
     // - sessionHandle:
     //   the value returned through *pSessionHandle from EapPeerBeginSession()
     //
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerEndSession(_In_ EAP_SESSION_HANDLE sessionHandle, _Outptr_ EAP_ERROR** ppEapError);
 
     // This is the last function that EAPHost should call on this method.
@@ -480,6 +497,7 @@ extern "C"
     // which can be called at any time.
     // This function should be called only once and it should un-initialize
     // everything for this method.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerShutdown(_Outptr_ EAP_ERROR** ppEapError);
 
     // It should raise a configuration UI so that the user can configure the method
@@ -508,6 +526,7 @@ extern "C"
     // - ppConnectionDataOut:
     //   *ppConnectionDataOut is configuration data returned from this functioin.
     //   It will be used to pass to next EapPeerInvokeConfigUI() call.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerInvokeConfigUI(
         _In_ EAP_METHOD_TYPE* pEapType,
         _In_ HWND hwndParent,
@@ -542,6 +561,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerInvokeInteractiveUI(
         _In_ EAP_METHOD_TYPE* pEapType,
         _In_ HWND hwndParent,
@@ -582,6 +602,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerInvokeIdentityUI(
         _In_ EAP_METHOD_TYPE* pEapType,
         _In_ DWORD dwFlags,
@@ -624,6 +645,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerQueryCredentialInputFields(
         _In_ HANDLE hUserImpersonationToken,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -668,6 +690,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerQueryUserBlobFromCredentialInputFields(
         _In_ HANDLE hUserImpersonationToken,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -686,6 +709,7 @@ extern "C"
     // This function returns fields to be displayed during interactive time when alternative UI is used.
     // Method should not bring up UI in this function.
     //
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerQueryInteractiveUIInputFields(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -705,6 +729,7 @@ extern "C"
 
     // This function returns interactive UI blob from the UI fields filled by the user
     //
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerQueryUIBlobFromInteractiveUIInputFields(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -751,6 +776,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerConfigXml2Blob(
         _In_ DWORD dwFlags,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -787,6 +813,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerCredentialsXml2Blob(
         _In_ DWORD dwFlags,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -820,6 +847,7 @@ extern "C"
     // - pEapError
     //   Pointer to pointer to EAP_ERROR that will be filled by the Method in the case of error. In some cases
     //  the structure might not be filled.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapPeerConfigBlob2Xml(
         _In_ DWORD dwFlags,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -863,6 +891,7 @@ extern "C"
     // - ppEapError
     //   A pointer to a pointer to an EAP_ERROR structure that contains any errors raised by
     //   EAPHost during the execution of this function call.
+_Win32_metadata_supported_os_(windows6.1)
     DWORD WINAPI EapPeerGetMethodProperties(
         _In_ DWORD dwVersion,
         _In_ DWORD dwFlags,
@@ -911,6 +940,7 @@ extern "C"
     //   A pointer to the address of an EAP_ERROR structure that contains any
     //   errors raised during the execution of this function call. After consuming
     //   the error data this memory will be freed by calling EapPeerFreeErrorMemory
+_Win32_metadata_supported_os_(windows8.0)
     DWORD WINAPI EapPeerGetConfigBlobAndUserBlob(
         _In_ DWORD dwFlags,
         _In_ EAP_METHOD_TYPE eapMethodType,
@@ -928,6 +958,7 @@ extern "C"
     // - pUIContextData:
     //   Any memory returned by configuration APIs.
     //
+_Win32_metadata_supported_os_(windows6.0.6000)
     VOID WINAPI EapPeerFreeMemory(_In_ void* pUIContextData);
 
 #ifdef __cplusplus
