diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodauthenticatorapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodauthenticatorapis.h
index a7cc11b3..b6d90882 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodauthenticatorapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/eapmethodauthenticatorapis.h
@@ -12,6 +12,7 @@
 #define EAP_METHOD_AUTHENTICATOR_APIS_H
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -215,14 +216,19 @@ extern "C"
     //
     // EAP Functions
     //
+_Win32_metadata_supported_os_(windows6.0.6000)
     void WINAPI EapMethodAuthenticatorFreeErrorMemory(_In_ EAP_ERROR* pEapError);
+_Win32_metadata_supported_os_(windows6.0.6000)
     void WINAPI EapMethodAuthenticatorFreeMemory(_In_ void* pUIContextData);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorGetInfo(
         _In_ EAP_METHOD_TYPE* pEapType, _Out_ EAP_AUTHENTICATOR_METHOD_ROUTINES* pEapInfo, _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorInitialize(_In_ EAP_METHOD_TYPE* pEapType, _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorBeginSession(
         // Flags to qualify the authentication process.
         _In_ DWORD dwFlags,
@@ -247,6 +253,7 @@ extern "C"
         // the failure. This will be null on a successful call.
         _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorUpdateInnerMethodParams(
         // context handle as returned from a successful call to
         // EapAuthenticatorBeginSession
@@ -263,6 +270,7 @@ extern "C"
     // The authenticator calls this any time it receives a packet that the eaphost
     // needs to process. This should be called only after a successful call to
     // EapAuthenticatorBeginSession.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorReceivePacket(
         // context handle as returned from a successful call to
         // EapAuthenticatorBeginSession
@@ -281,6 +289,7 @@ extern "C"
         // the failure. This will be null on a successful call.
         _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorSendPacket(
         // context handle as returned from a successful call to
         // EapHostAuthenticatorBeginSession
@@ -304,6 +313,7 @@ extern "C"
     // Returns an array of attributes that the caller needs to act on.
     // The supplicant will call this when a call to
     // EapHostAuthenticatorProcessRequestPacket returns EapHostAuthenticatorResponseRespond.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorGetAttributes(
         // context handle as returned from a successful call to
         // EapHostAuthenticatorBeginSession
@@ -314,6 +324,7 @@ extern "C"
 
     // Sets an array of attributes that the caller wants the eap method to act
     // on.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorSetAttributes(
         // context handle as returned from a successful call to
         // EapHostAuthenticatorBeginSession
@@ -335,6 +346,7 @@ extern "C"
     //    this method with a failure.
     // 2. The server can choose to terminate an authentication with a failure in the middle of
     //    an authentication.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorGetResult(
         // context handle as returned from a successful call to
         // EapHostPeerBeginSession
@@ -348,6 +360,7 @@ extern "C"
 
     // Ends the authentication session. This cleans up any state that the eap
     // method or eaphost might be keeping.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorEndSession(
         // context handle as returned from a successful call to
         // EapHostPeerBeginSession. This will be set to NULL on a successful call.
@@ -356,12 +369,14 @@ extern "C"
         // the failure. This will be null on a successful call.
         _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD EapMethodAuthenticatorShutdown(_In_ EAP_METHOD_TYPE* pEapType, _Outptr_ EAP_ERROR** ppEapError);
 
     /// Config API
 
 #define EAP_METHOD_AUTHENTICATOR_CONFIG_IS_IDENTITY_PRIVACY 0x1
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD WINAPI EapMethodAuthenticatorInvokeConfigUI(
         _In_ EAP_METHOD_TYPE* pEapMethodType,
         _In_ HWND hwndParent,
