diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/eappapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/eappapis.h
index 170f364f..75a90fa2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/eappapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/eappapis.h
@@ -12,6 +12,7 @@
 #define EAPPAPIS_H
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -68,11 +69,13 @@ extern "C"
     // So the client should not initialize a conflicting COM environment.
     // If the client really needs a different COM environment, it should use the
     // COM APIs directly instead of using this C style APIs.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerInitialize();
 
     // Must call this after it is done calling any other EapHostPeer...
     // functions.
     // Inside this function it calls CoUninitialize().
+_Win32_metadata_supported_os_(windows6.0.6000)
     void APIENTRY EapHostPeerUninitialize();
 
     // If this function succeeds, no matter the calls to other functions
@@ -136,6 +139,7 @@ extern "C"
     //
     // - pSessionId:
     //   Return session ID in it. This is used to identify the authentication session.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerBeginSession(
         _In_ DWORD dwFlags,
         _In_ EAP_METHOD_TYPE eapType,
@@ -177,6 +181,7 @@ extern "C"
     //
     // - pEapOutput:
     //   It will tell the supplicant what's the next EAPHost API that it should call.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerProcessReceivedPacket(
         _In_ EAP_SESSIONID sessionHandle,
         _In_ DWORD cbReceivePacket,
@@ -196,6 +201,7 @@ extern "C"
     //
     // - ppSendPacket:
     //   pointer to a buffer that contains the packet to send upon return
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetSendPacket(
         _In_ EAP_SESSIONID sessionHandle,
         _Out_ DWORD* pcbSendPacket,
@@ -213,6 +219,7 @@ extern "C"
     // - ppResult:
     //   EAPHost will fill this struct with related information as defined
     //   by EapHostPeerMethodResult
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetResult(
         _In_ EAP_SESSIONID sessionHandle,
         _In_ EapHostPeerMethodResultReason reason,
@@ -232,6 +239,7 @@ extern "C"
     // - ppUIContextData:
     //   It will be passed to EapHostPeerInvokeInteractiveUI() as _In_ parameter
     //   pUIContextData
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetUIContext(
         _In_ EAP_SESSIONID sessionHandle,
         _Out_ DWORD* pdwSizeOfUIContextData,
@@ -255,6 +263,7 @@ extern "C"
     //
     // - pEapOutput:
     //   It will tell the supplicant what's the next EAPHost API that it should call.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerSetUIContext(
         _In_ EAP_SESSIONID sessionHandle,
         _In_ DWORD dwSizeOfUIContextData,
@@ -272,6 +281,7 @@ extern "C"
     //
     // - pAttributes:
     //   EAP attributes returned to supplicant.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetResponseAttributes(_In_ EAP_SESSIONID sessionHandle, _Out_ EapAttributes* pAttribs, _Outptr_ EAP_ERROR** ppEapError);
 
     // See EapHostPeerGetResponseAttributes() for when this function will be called.
@@ -285,6 +295,7 @@ extern "C"
     //
     // - pAttribs:
     //   EAP attributes that the supplicant passes to EAPHost
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerSetResponseAttributes(
         _In_ EAP_SESSIONID sessionHandle,
         _In_ const EapAttributes* const pAttribs,
@@ -306,6 +317,7 @@ extern "C"
     // - ppAuthData:
     //   contains the returned authentication data. Interpretation of the data
     //   depends on the "authParam" passed in
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetAuthStatus(
         _In_ EAP_SESSIONID sessionHandle,
         _In_ EapHostPeerAuthParams authParam,
@@ -319,8 +331,10 @@ extern "C"
     // Parameters:
     // - sessionHandle:
     //   the value returned through *pSessionId from EapHostPeerBeginSession()
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerEndSession(_In_ EAP_SESSIONID sessionHandle, _Outptr_ EAP_ERROR** ppEapError);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
     DWORD APIENTRY EapHostPeerGetDataToUnplumbCredentials(
         _Out_ GUID* pConnectionIdThatLastSavedCreds,
         _Out_ __int3264* phCredentialImpersonationToken,
@@ -337,14 +351,17 @@ extern "C"
     //   the same connection GUID passed into EapHostPeerBeginSession(). If
     //   a NULL is passed to EapHostPeerBeginSession() as the connection id,
     //   then there is no need to call this API.
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerClearConnection(_In_ GUID* pConnectionId, _Outptr_ EAP_ERROR** ppEapError);
 
     // If any of the previous function is called, and an EAP_ERROR is returned,
     // this function must be called to free the memory.
+_Win32_metadata_supported_os_(windows6.0.6000)
     void APIENTRY EapHostPeerFreeEapError(_In_ EAP_ERROR* pEapError);
 
     // EAPHost will call this function, and it would return user data and user identity.
 
+_Win32_metadata_supported_os_(windows6.0.6000)
     DWORD APIENTRY EapHostPeerGetIdentity(
         // API's version number for ease of interoperability. It must be 0.
         _In_ DWORD dwVersion,
@@ -381,6 +398,7 @@ extern "C"
         _In_ DWORD dwSizeofPassword, _In_reads_bytes_(dwSizeofPassword) LPWSTR szPassword, _Outptr_ LPWSTR* ppszEncPassword);
 
     // In case of GetInentity returned memory needs to be released using this API
+_Win32_metadata_supported_os_(windows6.0.6000)
     void APIENTRY EapHostPeerFreeRuntimeMemory(_In_ BYTE* pData);
 
 #ifdef __cplusplus
