diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/EditionUpgradeHelper.h b/generation/WinSDK/RecompiledIdlHeaders/um/EditionUpgradeHelper.h
index 84e5b729..fcd82dad 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/EditionUpgradeHelper.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/EditionUpgradeHelper.h
@@ -32,6 +32,7 @@
 
 #ifndef __editionupgradehelper_h__
 #define __editionupgradehelper_h__
+#include <win32metadata_annotations.h>
 
 #if defined(_MSC_VER) && (_MSC_VER >= 1020)
 #pragma once
@@ -142,6 +143,7 @@ EXTERN_C const IID IID_IEditionUpgradeHelper;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("D3E9E342-5DEB-43B6-849E-6913B85D503A")
     IEditionUpgradeHelper : public IUnknown
     {
@@ -796,9 +798,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* end of Additional Prototypes */
