diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ElsCore.h b/generation/WinSDK/RecompiledIdlHeaders/um/ElsCore.h
index d3690cee..6ad059f8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ElsCore.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ElsCore.h
@@ -16,6 +16,9 @@ Abstract:
 #define __ELS_CORE__
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -149,6 +152,7 @@ typedef struct _MAPPING_PROPERTY_BAG
 // Public ELS APIs
 //
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 MappingGetServices(
@@ -158,6 +162,7 @@ MappingGetServices(
     );
 
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 MappingFreeServices(
@@ -165,6 +170,7 @@ MappingFreeServices(
     );
 
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 MappingRecognizeText(
@@ -182,6 +188,7 @@ MappingRecognizeText(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 MappingDoAction(
@@ -196,6 +203,7 @@ MappingDoAction(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 MappingFreePropertyBag(
