diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/enclaveapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/enclaveapi.h
index 70dab084..7b5a7c98 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/enclaveapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/enclaveapi.h
@@ -12,6 +12,7 @@
 
 #ifndef _ENCLAVEAPI_H_
 #define _ENCLAVEAPI_H_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -25,6 +26,8 @@ extern "C" {
 #pragma region Desktop Family or OneCore Or App Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -33,6 +36,8 @@ IsEnclaveTypeSupported(
     _In_ DWORD flEnclaveType
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 _Post_writable_byte_size_(dwSize)
@@ -49,6 +54,8 @@ CreateEnclave(
     _Out_opt_ LPDWORD lpEnclaveError
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -65,6 +72,8 @@ LoadEnclaveData(
     _Out_opt_ LPDWORD lpEnclaveError
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -92,6 +101,8 @@ LoadEnclaveImageA(
     _In_ LPCSTR lpImageName
     );
 
+_Win32_metadata_supported_os_(windows10.0.16299)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -107,6 +118,8 @@ LoadEnclaveImageW(
 #define LoadEnclaveImage  LoadEnclaveImageA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows10.0.16299)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -118,6 +131,8 @@ CallEnclave(
     _Out_ LPVOID* lpReturnValue
     );
 
+_Win32_metadata_supported_os_(windows10.0.16299)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -127,6 +142,8 @@ TerminateEnclave(
     _In_ BOOL fWait
     );
 
+_Win32_metadata_supported_os_(windows10.0.16299)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
