diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/evr.h b/generation/WinSDK/RecompiledIdlHeaders/um/evr.h
index 89b5fd6e..f5606b38 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/evr.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/evr.h
@@ -153,6 +153,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if defined(__midl)
@@ -254,6 +255,7 @@ EXTERN_C const IID IID_IMFVideoPositionMapper;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("1F6A9F17-E70B-4e24-8AE4-0B2C3BA7A4AE")
     IMFVideoPositionMapper : public IUnknown
     {
@@ -361,6 +363,7 @@ EXTERN_C const IID IID_IMFVideoDeviceID;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("A38D9567-5A9C-4f3c-B293-8EB415B279BA")
     IMFVideoDeviceID : public IUnknown
     {
@@ -492,6 +495,7 @@ EXTERN_C const IID IID_IMFVideoDisplayControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("a490b1e4-ab84-4d31-a1b2-181e03b1077a")
     IMFVideoDisplayControl : public IUnknown
     {
@@ -819,6 +823,7 @@ EXTERN_C const IID IID_IMFVideoPresenter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("29AFF080-182A-4a5d-AF3B-448F3A6346CB")
     IMFVideoPresenter : public IMFClockStateSink
     {
@@ -969,6 +974,7 @@ EXTERN_C const IID IID_IMFDesiredSample;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("56C294D0-753E-4260-8D61-A3D8820B1D54")
     IMFDesiredSample : public IUnknown
     {
@@ -1086,6 +1092,7 @@ EXTERN_C const IID IID_IMFVideoMixerControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("A5C6C53F-C202-4aa5-9695-175BA8C508A5")
     IMFVideoMixerControl : public IUnknown
     {
@@ -1247,6 +1254,7 @@ EXTERN_C const IID IID_IMFVideoMixerControl2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("8459616d-966e-4930-b658-54fa7e5a16d3")
     IMFVideoMixerControl2 : public IMFVideoMixerControl
     {
@@ -1392,6 +1400,7 @@ EXTERN_C const IID IID_IMFVideoRenderer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DFDFD197-A9CA-43d8-B341-6AF3503792CD")
     IMFVideoRenderer : public IUnknown
     {
@@ -1483,6 +1492,7 @@ EXTERN_C const IID IID_IEVRFilterConfig;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("83E91E85-82C1-4ea7-801D-85DC50B75086")
     IEVRFilterConfig : public IUnknown
     {
@@ -1598,6 +1608,7 @@ EXTERN_C const IID IID_IEVRFilterConfigEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("aea36028-796d-454f-beee-b48071e24304")
     IEVRFilterConfigEx : public IEVRFilterConfig
     {
@@ -1736,6 +1747,7 @@ EXTERN_C const IID IID_IMFTopologyServiceLookup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fa993889-4383-415a-a930-dd472a8cf6f7")
     IMFTopologyServiceLookup : public IUnknown
     {
@@ -1843,6 +1855,7 @@ EXTERN_C const IID IID_IMFTopologyServiceLookupClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fa99388a-4383-415a-a930-dd472a8cf6f7")
     IMFTopologyServiceLookupClient : public IUnknown
     {
@@ -1939,6 +1952,7 @@ EXTERN_C const IID IID_IEVRTrustedVideoPlugin;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("83A4CE40-7710-494b-A893-A472049AF630")
     IEVRTrustedVideoPlugin : public IUnknown
     {
