diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/FaxRoute.h b/generation/WinSDK/RecompiledIdlHeaders/um/FaxRoute.h
index e003287d..83089607 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/FaxRoute.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/FaxRoute.h
@@ -27,6 +27,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -100,12 +103,16 @@ enum FAXROUTE_ENABLE {
 // prototypes
 //
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 FaxRouteInitialize(
     IN HANDLE HeapHandle,
     IN PFAX_ROUTE_CALLBACKROUTINES FaxRouteCallbackRoutines
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 FaxRouteDeviceEnable(
     IN  LPCWSTR RoutingGuid,
@@ -113,12 +120,16 @@ FaxRouteDeviceEnable(
     IN  LONG Enabled
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 FaxRouteDeviceChangeNotification(
     IN  DWORD DeviceId,
     IN  BOOL  NewDevice
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 FaxRouteGetRoutingInfo(
     IN  LPCWSTR RoutingGuid,
@@ -127,6 +138,8 @@ FaxRouteGetRoutingInfo(
     OUT LPDWORD RoutingInfoSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 FaxRouteSetRoutingInfo(
     IN  LPCWSTR RoutingGuid,
