diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fdi.h b/generation/WinSDK/RecompiledIdlHeaders/um/fdi.h
index 180d640e..4b7fb58a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fdi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fdi.h
@@ -185,6 +185,7 @@
 
 #ifndef INCLUDED_FDI
 #define INCLUDED_FDI    1
+#include <win32metadata_annotations.h>
 
 #include <specstrings.h>
 typedef _Null_terminated_ char* LPSTR;
@@ -905,6 +906,7 @@ typedef FDISPILLFILE *PFDISPILLFILE; /* pfdisf */
 #endif
 
 
+_Win32_metadata_supported_os_(windows5.0)
 /*** cpuType values for FDICreate()
  *
  *  WARNING: For 16-bit Windows applications, the CPU detection may not
@@ -957,6 +959,7 @@ typedef FDISPILLFILE *PFDISPILLFILE; /* pfdisf */
  *      FDICreate(), and then free it up afterwards (or do all your allocation
  *      before calling FDICreate().
  */
+_Win32_metadata_supported_os_(windows5.0)
 HFDI FAR DIAMONDAPI FDICreate(__callback PFNALLOC pfnalloc,
                               __callback PFNFREE  pfnfree,
                               __callback PFNOPEN  pfnopen,
@@ -983,6 +986,7 @@ HFDI FAR DIAMONDAPI FDICreate(__callback PFNALLOC pfnalloc,
  *      Returns FALSE, file is not a cabinet;  If an error occurred,
  *          perf (passed on FDICreate call!) filled in with error.
  */
+_Win32_metadata_supported_os_(windows5.0)
 BOOL FAR DIAMONDAPI FDIIsCabinet(_In_  HFDI             hfdi,
                                  _In_  INT_PTR          hf,
                                  _Out_opt_ PFDICABINETINFO  pfdici);
@@ -1014,6 +1018,7 @@ BOOL FAR DIAMONDAPI FDIIsCabinet(_In_  HFDI             hfdi,
  *      it will not be in a valid state (since there was an error while
  *      writing it out).
  */
+_Win32_metadata_supported_os_(windows5.0)
 BOOL FAR DIAMONDAPI FDICopy(_In_ HFDI                   hfdi,
                             _In_ LPSTR                  pszCabinet,
                             _In_ LPSTR                  pszCabPath,
@@ -1034,6 +1039,7 @@ BOOL FAR DIAMONDAPI FDICopy(_In_ HFDI                   hfdi,
  *  Exit-Failure:
  *      Returns FALSE;
  */
+_Win32_metadata_supported_os_(windows5.0)
 BOOL FAR DIAMONDAPI FDIDestroy(_In_ HFDI hfdi);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM) */
