diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
index 870fbeab..3e0e7647 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
@@ -12,6 +12,7 @@
 
 #ifndef _FIBERS_H_
 #define _FIBERS_H_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -30,6 +31,8 @@ extern "C" {
 #define FLS_OUT_OF_INDEXES ((DWORD)0xFFFFFFFF)
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -37,6 +40,8 @@ FlsAlloc(
     _In_opt_ PFLS_CALLBACK_FUNCTION lpCallback
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 PVOID
 WINAPI
@@ -44,6 +49,8 @@ FlsGetValue(
     _In_ DWORD dwFlsIndex
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -52,6 +59,8 @@ FlsSetValue(
     _In_opt_ PVOID lpFlsData
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -69,6 +78,7 @@ FlsFree(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 BOOL
 WINAPI
