diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fltUser.h b/generation/WinSDK/RecompiledIdlHeaders/um/fltUser.h
index 5bcd819e..ef7c23bd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fltUser.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fltUser.h
@@ -19,6 +19,7 @@ Environment:
 #ifndef __FLTUSER_H__
 #define __FLTUSER_H__
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -234,6 +235,7 @@ FilterFindFirst (
     _In_ DWORD dwBufferSize,
     _Out_ LPDWORD lpBytesReturned,
     _Out_ LPHANDLE lpFilterFind
+        _Win32_metadata_raii_free_(FilterFindClose)
     );
 
 _Must_inspect_result_
@@ -264,6 +266,7 @@ FilterVolumeFindFirst (
     _In_ DWORD dwBufferSize,
     _Out_ LPDWORD lpBytesReturned,
     _Out_ PHANDLE lpVolumeFind
+        _Win32_metadata_raii_free_(FilterVolumeFindClose)
     );
 
 _Must_inspect_result_
@@ -297,6 +300,7 @@ FilterInstanceFindFirst (
     _In_ DWORD dwBufferSize,
     _Out_ LPDWORD lpBytesReturned,
     _Out_ LPHANDLE lpFilterInstanceFind
+        _Win32_metadata_raii_free_(FilterInstanceFindClose)
     );
 
 _Must_inspect_result_
@@ -332,6 +336,7 @@ FilterVolumeInstanceFindFirst (
     _In_ DWORD dwBufferSize,
     _Out_ LPDWORD lpBytesReturned,
     _Out_ LPHANDLE lpVolumeInstanceFind
+        _Win32_metadata_raii_free_(FilterVolumeInstanceFindClose)
     );
 
 _Must_inspect_result_
@@ -430,6 +435,7 @@ FilterGetMessage (
     _Inout_opt_ LPOVERLAPPED lpOverlapped
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Must_inspect_result_
 HRESULT
 WINAPI
