diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fsrmquota.h b/generation/WinSDK/RecompiledIdlHeaders/um/fsrmquota.h
index af5efb6b..5618a89d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fsrmquota.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fsrmquota.h
@@ -124,6 +124,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -180,6 +181,7 @@ EXTERN_C const IID IID_IFsrmQuotaBase;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("1568a795-3924-4118-b74b-68d8f0fa5daf")
     IFsrmQuotaBase : public IFsrmObject
     {
@@ -504,6 +506,7 @@ EXTERN_C const IID IID_IFsrmQuotaObject;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("42dc3511-61d5-48ae-b6dc-59fc00c0a8d6")
     IFsrmQuotaObject : public IFsrmQuotaBase
     {
@@ -859,6 +862,7 @@ EXTERN_C const IID IID_IFsrmQuota;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("377f739d-9647-4b8e-97d2-5ffce6d759cd")
     IFsrmQuota : public IFsrmQuotaObject
     {
@@ -1248,6 +1252,7 @@ EXTERN_C const IID IID_IFsrmAutoApplyQuota;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("f82e5729-6aba-4740-bfc7-c7f58f75fb7b")
     IFsrmAutoApplyQuota : public IFsrmQuotaObject
     {
@@ -1627,6 +1632,7 @@ EXTERN_C const IID IID_IFsrmQuotaManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("8bb68c7d-19d8-4ffb-809e-be4fc1734014")
     IFsrmQuotaManager : public IDispatch
     {
@@ -1967,6 +1973,7 @@ EXTERN_C const IID IID_IFsrmQuotaManagerEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("4846cb01-d430-494f-abb4-b1054999fb09")
     IFsrmQuotaManagerEx : public IFsrmQuotaManager
     {
@@ -2257,6 +2264,7 @@ EXTERN_C const IID IID_IFsrmQuotaTemplate;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("a2efab31-295e-46bb-b976-e86d58b52e8b")
     IFsrmQuotaTemplate : public IFsrmQuotaBase
     {
@@ -2594,6 +2602,7 @@ EXTERN_C const IID IID_IFsrmQuotaTemplateImported;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("9a2bf113-a329-44cc-809a-5c00fce8da40")
     IFsrmQuotaTemplateImported : public IFsrmQuotaTemplate
     {
@@ -2938,6 +2947,7 @@ EXTERN_C const IID IID_IFsrmQuotaTemplateManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("4173ac41-172d-4d52-963c-fdc7e415f717")
     IFsrmQuotaTemplateManager : public IDispatch
     {
@@ -3175,19 +3185,31 @@ unsigned char * __RPC_USER  VARIANT_UserMarshal(  __RPC__in unsigned long *, __R
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
 void                      __RPC_USER  VARIANT_UserFree(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  VARIANT_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
 /* end of Additional Prototypes */
