diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryprovider.h b/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryprovider.h
index 916dce94..9377533c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryprovider.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryprovider.h
@@ -118,6 +118,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -140,6 +143,7 @@ EXTERN_C const IID IID_IFunctionDiscoveryProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("dcde394f-1478-4813-a402-f6fb10657222")
     IFunctionDiscoveryProvider : public IUnknown
     {
@@ -374,6 +378,7 @@ EXTERN_C const IID IID_IProviderProperties;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("cf986ea6-3b5f-4c5f-b88a-2f8b20ceef17")
     IProviderProperties : public IUnknown
     {
@@ -544,6 +549,7 @@ EXTERN_C const IID IID_IProviderPublishing;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("CD1B9A04-206C-4a05-A0C8-1635A21A2B7C")
     IProviderPublishing : public IUnknown
     {
@@ -664,6 +670,7 @@ EXTERN_C const IID IID_IFunctionDiscoveryProviderFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("86443ff0-1ad5-4e68-a45a-40c2c329de3b")
     IFunctionDiscoveryProviderFactory : public IUnknown
     {
@@ -797,6 +804,7 @@ EXTERN_C const IID IID_IFunctionDiscoveryProviderQuery;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6876ea98-baec-46db-bc20-75a76e267a3a")
     IFunctionDiscoveryProviderQuery : public IUnknown
     {
@@ -931,6 +939,7 @@ EXTERN_C const IID IID_IProviderQueryConstraintCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9c243e11-3261-4bcd-b922-84a873d460ae")
     IProviderQueryConstraintCollection : public IUnknown
     {
@@ -1091,6 +1100,7 @@ EXTERN_C const IID IID_IProviderPropertyConstraintCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("f4fae42f-5778-4a13-8540-b5fd8c1398dd")
     IProviderPropertyConstraintCollection : public IUnknown
     {
@@ -1263,6 +1273,7 @@ EXTERN_C const IID IID_IFunctionDiscoveryServiceProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("4c81ed02-1b04-43f2-a451-69966cbcd1c2")
     IFunctionDiscoveryServiceProvider : public IUnknown
     {
@@ -1369,14 +1380,22 @@ unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal(  __RPC__in unsigned long *,
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
 void                      __RPC_USER  LPSAFEARRAY_UserFree(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
 /* end of Additional Prototypes */
