diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryproviderhelper.h b/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryproviderhelper.h
index a080bf20..a7fbeee1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryproviderhelper.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/functiondiscoveryproviderhelper.h
@@ -4,6 +4,7 @@
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -38,6 +39,7 @@ public:
     //    pIUnk: Interface proxy to enable callbacks on
     //
     // Returns: S_OK on success, appropriate HRESULT otherwise
+_Win32_metadata_supported_os_(windows5.0)
     static HRESULT CoSetProxyBlanket( IUnknown *pIUnk )
     {
         HRESULT hr = S_OK;
