diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fwpmu.h b/generation/WinSDK/RecompiledIdlHeaders/um/fwpmu.h
index 207e490f..2ce86b9d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fwpmu.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fwpmu.h
@@ -6,6 +6,7 @@
      Declares the management portion of the FWP API.
 */
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or AppRuntime Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_APPRUNTIME)
@@ -3330,6 +3331,7 @@ static const IPSEC_CIPHER_TRANSFORM_ID0 IPSEC_CIPHER_TRANSFORM_ID_GCM_AES_256 =
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 void WINAPI FwpmFreeMemory0(_Inout_ void** p);
 
 
@@ -3339,6 +3341,7 @@ void WINAPI FwpmFreeMemory0(_Inout_ void** p);
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineOpen0(
@@ -3349,10 +3352,12 @@ FwpmEngineOpen0(
    _Out_ HANDLE* engineHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineClose0(_Inout_ HANDLE engineHandle);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineGetOption0(
@@ -3361,6 +3366,7 @@ FwpmEngineGetOption0(
    _Outptr_ FWP_VALUE0** value
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineSetOption0(
@@ -3369,6 +3375,7 @@ FwpmEngineSetOption0(
    _In_ const FWP_VALUE0* newValue
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineGetSecurityInfo0(
@@ -3381,6 +3388,7 @@ FwpmEngineGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmEngineSetSecurityInfo0(
@@ -3392,6 +3400,7 @@ FwpmEngineSetSecurityInfo0(
    _In_opt_ const ACL* sacl
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSessionCreateEnumHandle0(
@@ -3400,6 +3409,7 @@ FwpmSessionCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSessionEnum0(
@@ -3410,6 +3420,7 @@ FwpmSessionEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSessionDestroyEnumHandle0(
@@ -3426,6 +3437,7 @@ FwpmSessionDestroyEnumHandle0(
 
 #define FWPM_TXN_READ_ONLY (0x00000001)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmTransactionBegin0(
@@ -3433,10 +3445,12 @@ FwpmTransactionBegin0(
    _In_ UINT32 flags
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmTransactionCommit0(_In_ HANDLE engineHandle);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmTransactionAbort0(_In_ HANDLE engineHandle);
@@ -3448,6 +3462,7 @@ FwpmTransactionAbort0(_In_ HANDLE engineHandle);
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderAdd0(
@@ -3456,6 +3471,7 @@ FwpmProviderAdd0(
    _In_opt_ PSECURITY_DESCRIPTOR sd
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderDeleteByKey0(
@@ -3463,6 +3479,7 @@ FwpmProviderDeleteByKey0(
    _In_ const GUID* key
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderGetByKey0(
@@ -3471,6 +3488,7 @@ FwpmProviderGetByKey0(
    _Outptr_ FWPM_PROVIDER0** provider
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderCreateEnumHandle0(
@@ -3479,6 +3497,7 @@ FwpmProviderCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderEnum0(
@@ -3489,6 +3508,7 @@ FwpmProviderEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderDestroyEnumHandle0(
@@ -3496,6 +3516,7 @@ FwpmProviderDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderGetSecurityInfoByKey0(
@@ -3509,6 +3530,7 @@ FwpmProviderGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderSetSecurityInfoByKey0(
@@ -3526,6 +3548,7 @@ typedef void (CALLBACK *FWPM_PROVIDER_CHANGE_CALLBACK0)(
                            _In_ const FWPM_PROVIDER_CHANGE0* change
                            );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderSubscribeChanges0(
@@ -3536,6 +3559,7 @@ FwpmProviderSubscribeChanges0(
    _Out_ HANDLE* changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderUnsubscribeChanges0(
@@ -3543,6 +3567,7 @@ FwpmProviderUnsubscribeChanges0(
    _Inout_ HANDLE changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderSubscriptionsGet0(
@@ -3559,6 +3584,7 @@ FwpmProviderSubscriptionsGet0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextAdd0(
@@ -3569,6 +3595,7 @@ FwpmProviderContextAdd0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmProviderContextAdd1(
@@ -3580,6 +3607,7 @@ FwpmProviderContextAdd1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmProviderContextAdd2(
@@ -3602,6 +3630,7 @@ FwpmProviderContextAdd3(
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextDeleteById0(
@@ -3609,6 +3638,7 @@ FwpmProviderContextDeleteById0(
    _In_ UINT64 id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextDeleteByKey0(
@@ -3616,6 +3646,7 @@ FwpmProviderContextDeleteByKey0(
    _In_ const GUID* key
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextGetById0(
@@ -3625,6 +3656,7 @@ FwpmProviderContextGetById0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmProviderContextGetById1(
@@ -3635,6 +3667,7 @@ FwpmProviderContextGetById1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmProviderContextGetById2(
@@ -3655,6 +3688,7 @@ FwpmProviderContextGetById3(
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextGetByKey0(
@@ -3664,6 +3698,7 @@ FwpmProviderContextGetByKey0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmProviderContextGetByKey1(
@@ -3674,6 +3709,7 @@ FwpmProviderContextGetByKey1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmProviderContextGetByKey2(
@@ -3693,6 +3729,7 @@ FwpmProviderContextGetByKey3(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextCreateEnumHandle0(
@@ -3701,6 +3738,7 @@ FwpmProviderContextCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextEnum0(
@@ -3713,6 +3751,7 @@ FwpmProviderContextEnum0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmProviderContextEnum1(
@@ -3726,6 +3765,7 @@ FwpmProviderContextEnum1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmProviderContextEnum2(
@@ -3752,6 +3792,7 @@ FwpmProviderContextEnum3(
 
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextDestroyEnumHandle0(
@@ -3759,6 +3800,7 @@ FwpmProviderContextDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextGetSecurityInfoByKey0(
@@ -3772,6 +3814,7 @@ FwpmProviderContextGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextSetSecurityInfoByKey0(
@@ -3789,6 +3832,7 @@ typedef void (CALLBACK *FWPM_PROVIDER_CONTEXT_CHANGE_CALLBACK0)(
                            _In_ const FWPM_PROVIDER_CONTEXT_CHANGE0* change
                            );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextSubscribeChanges0(
@@ -3799,6 +3843,7 @@ FwpmProviderContextSubscribeChanges0(
    _Out_ HANDLE* changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextUnsubscribeChanges0(
@@ -3806,6 +3851,7 @@ FwpmProviderContextUnsubscribeChanges0(
    _Inout_ HANDLE changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmProviderContextSubscriptionsGet0(
@@ -3822,6 +3868,7 @@ FwpmProviderContextSubscriptionsGet0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerAdd0(
@@ -3830,6 +3877,7 @@ FwpmSubLayerAdd0(
    _In_opt_ PSECURITY_DESCRIPTOR sd
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerDeleteByKey0(
@@ -3837,6 +3885,7 @@ FwpmSubLayerDeleteByKey0(
    _In_ const GUID* key
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerGetByKey0(
@@ -3845,6 +3894,7 @@ FwpmSubLayerGetByKey0(
    _Outptr_ FWPM_SUBLAYER0** subLayer
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerCreateEnumHandle0(
@@ -3853,6 +3903,7 @@ FwpmSubLayerCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerEnum0(
@@ -3863,6 +3914,7 @@ FwpmSubLayerEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerDestroyEnumHandle0(
@@ -3870,6 +3922,7 @@ FwpmSubLayerDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerGetSecurityInfoByKey0(
@@ -3883,6 +3936,7 @@ FwpmSubLayerGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerSetSecurityInfoByKey0(
@@ -3900,6 +3954,7 @@ typedef void (CALLBACK *FWPM_SUBLAYER_CHANGE_CALLBACK0)(
                            _In_ const FWPM_SUBLAYER_CHANGE0* change
                            );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerSubscribeChanges0(
@@ -3910,6 +3965,7 @@ FwpmSubLayerSubscribeChanges0(
    _Out_ HANDLE* changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerUnsubscribeChanges0(
@@ -3917,6 +3973,7 @@ FwpmSubLayerUnsubscribeChanges0(
    _Inout_ HANDLE changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmSubLayerSubscriptionsGet0(
@@ -3933,6 +3990,7 @@ FwpmSubLayerSubscriptionsGet0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerGetById0(
@@ -3941,6 +3999,7 @@ FwpmLayerGetById0(
    _Outptr_ FWPM_LAYER0** layer
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerGetByKey0(
@@ -3949,6 +4008,7 @@ FwpmLayerGetByKey0(
    _Outptr_ FWPM_LAYER0** layer
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerCreateEnumHandle0(
@@ -3957,6 +4017,7 @@ FwpmLayerCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerEnum0(
@@ -3967,6 +4028,7 @@ FwpmLayerEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerDestroyEnumHandle0(
@@ -3974,6 +4036,7 @@ FwpmLayerDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerGetSecurityInfoByKey0(
@@ -3987,6 +4050,7 @@ FwpmLayerGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmLayerSetSecurityInfoByKey0(
@@ -4006,6 +4070,7 @@ FwpmLayerSetSecurityInfoByKey0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutAdd0(
@@ -4015,6 +4080,7 @@ FwpmCalloutAdd0(
    _Out_opt_ UINT32* id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutDeleteById0(
@@ -4022,6 +4088,7 @@ FwpmCalloutDeleteById0(
    _In_ UINT32 id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutDeleteByKey0(
@@ -4029,6 +4096,7 @@ FwpmCalloutDeleteByKey0(
    _In_ const GUID* key
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutGetById0(
@@ -4037,6 +4105,7 @@ FwpmCalloutGetById0(
    _Outptr_ FWPM_CALLOUT0** callout
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutGetByKey0(
@@ -4045,6 +4114,7 @@ FwpmCalloutGetByKey0(
    _Outptr_ FWPM_CALLOUT0** callout
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutCreateEnumHandle0(
@@ -4053,6 +4123,7 @@ FwpmCalloutCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutEnum0(
@@ -4063,6 +4134,7 @@ FwpmCalloutEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutDestroyEnumHandle0(
@@ -4070,6 +4142,7 @@ FwpmCalloutDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutGetSecurityInfoByKey0(
@@ -4083,6 +4156,7 @@ FwpmCalloutGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutSetSecurityInfoByKey0(
@@ -4100,6 +4174,7 @@ typedef void (CALLBACK *FWPM_CALLOUT_CHANGE_CALLBACK0)(
                            _In_ const FWPM_CALLOUT_CHANGE0* change
                            );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutSubscribeChanges0(
@@ -4110,6 +4185,7 @@ FwpmCalloutSubscribeChanges0(
    _Out_ HANDLE* changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutUnsubscribeChanges0(
@@ -4117,6 +4193,7 @@ FwpmCalloutUnsubscribeChanges0(
    _Inout_ HANDLE changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmCalloutSubscriptionsGet0(
@@ -4133,6 +4210,7 @@ FwpmCalloutSubscriptionsGet0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterAdd0(
@@ -4142,6 +4220,7 @@ FwpmFilterAdd0(
    _Out_opt_ UINT64* id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterDeleteById0(
@@ -4149,6 +4228,7 @@ FwpmFilterDeleteById0(
    _In_ UINT64 id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterDeleteByKey0(
@@ -4156,6 +4236,7 @@ FwpmFilterDeleteByKey0(
    _In_ const GUID* key
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterGetById0(
@@ -4164,6 +4245,7 @@ FwpmFilterGetById0(
    _Outptr_ FWPM_FILTER0** filter
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterGetByKey0(
@@ -4172,6 +4254,7 @@ FwpmFilterGetByKey0(
    _Outptr_ FWPM_FILTER0** filter
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterCreateEnumHandle0(
@@ -4180,6 +4263,7 @@ FwpmFilterCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterEnum0(
@@ -4190,6 +4274,7 @@ FwpmFilterEnum0(
    _Out_ UINT32* numEntriesReturned
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterDestroyEnumHandle0(
@@ -4197,6 +4282,7 @@ FwpmFilterDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterGetSecurityInfoByKey0(
@@ -4210,6 +4296,7 @@ FwpmFilterGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterSetSecurityInfoByKey0(
@@ -4227,6 +4314,7 @@ typedef void (CALLBACK *FWPM_FILTER_CHANGE_CALLBACK0)(
                            _In_ const FWPM_FILTER_CHANGE0* change
                            );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterSubscribeChanges0(
@@ -4237,6 +4325,7 @@ FwpmFilterSubscribeChanges0(
    _Out_ HANDLE* changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterUnsubscribeChanges0(
@@ -4244,6 +4333,7 @@ FwpmFilterUnsubscribeChanges0(
    _Inout_ HANDLE changeHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmFilterSubscriptionsGet0(
@@ -4253,6 +4343,7 @@ FwpmFilterSubscriptionsGet0(
    _Out_ UINT32* numEntries
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmGetAppIdFromFileName0(
@@ -4278,6 +4369,7 @@ FwpmGetAppIdFromFileName0(
 #define FWPM_TUNNEL_FLAG_RESERVED0 (0x00000004)
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmIPsecTunnelAdd0(
@@ -4292,6 +4384,7 @@ FwpmIPsecTunnelAdd0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmIPsecTunnelAdd1(
@@ -4308,6 +4401,7 @@ FwpmIPsecTunnelAdd1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmIPsecTunnelAdd2(
@@ -4340,6 +4434,7 @@ FwpmIPsecTunnelAdd3(
 
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmIPsecTunnelDeleteByKey0(
@@ -4355,6 +4450,7 @@ FwpmIPsecTunnelDeleteByKey0(
 ///////////////////////////////////////////////////////////////////////////////
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecGetStatistics0(
@@ -4363,6 +4459,7 @@ IPsecGetStatistics0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecGetStatistics1(
@@ -4371,6 +4468,7 @@ IPsecGetStatistics1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextCreate0(
@@ -4381,6 +4479,7 @@ IPsecSaContextCreate0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextCreate1(
@@ -4392,6 +4491,7 @@ IPsecSaContextCreate1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextDeleteById0(
@@ -4399,6 +4499,7 @@ IPsecSaContextDeleteById0(
    _In_ UINT64 id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextGetById0(
@@ -4408,6 +4509,7 @@ IPsecSaContextGetById0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextGetById1(
@@ -4417,6 +4519,7 @@ IPsecSaContextGetById1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextGetSpi0(
@@ -4427,6 +4530,7 @@ IPsecSaContextGetSpi0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextGetSpi1(
@@ -4436,6 +4540,7 @@ IPsecSaContextGetSpi1(
    _Out_ IPSEC_SA_SPI* inboundSpi
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextSetSpi0(
@@ -4446,6 +4551,7 @@ IPsecSaContextSetSpi0(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextAddInbound0(
@@ -4454,6 +4560,7 @@ IPsecSaContextAddInbound0(
    _In_ const IPSEC_SA_BUNDLE0* inboundBundle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextAddOutbound0(
@@ -4463,6 +4570,7 @@ IPsecSaContextAddOutbound0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextAddInbound1(
@@ -4471,6 +4579,7 @@ IPsecSaContextAddInbound1(
    _In_ const IPSEC_SA_BUNDLE1* inboundBundle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextAddOutbound1(
@@ -4480,6 +4589,7 @@ IPsecSaContextAddOutbound1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextExpire0(
@@ -4499,6 +4609,7 @@ IPsecSaContextExpire0(
 #define IPSEC_SA_BUNDLE_UPDATE_PEER_V4_PRIVATE_ADDRESS (0x20ui64)
 #define IPSEC_SA_BUNDLE_UPDATE_MM_SA_ID                (0x40ui64)
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextUpdate0(
@@ -4508,6 +4619,7 @@ IPsecSaContextUpdate0(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextCreateEnumHandle0(
@@ -4516,6 +4628,7 @@ IPsecSaContextCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextEnum0(
@@ -4527,6 +4640,7 @@ IPsecSaContextEnum0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaContextEnum1(
@@ -4538,6 +4652,7 @@ IPsecSaContextEnum1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaContextDestroyEnumHandle0(
@@ -4553,6 +4668,7 @@ typedef void (CALLBACK *IPSEC_SA_CONTEXT_CALLBACK0)(
   _In_ const IPSEC_SA_CONTEXT_CHANGE0 *change
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecSaContextSubscribe0(
@@ -4563,6 +4679,7 @@ IPsecSaContextSubscribe0(
   _Out_ HANDLE* eventsHandle
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecSaContextUnsubscribe0(
@@ -4570,6 +4687,7 @@ IPsecSaContextUnsubscribe0(
   _Inout_ HANDLE eventsHandle
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecSaContextSubscriptionsGet0(
@@ -4582,6 +4700,7 @@ IPsecSaContextSubscriptionsGet0(
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaCreateEnumHandle0(
@@ -4590,6 +4709,7 @@ IPsecSaCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaEnum0(
@@ -4601,6 +4721,7 @@ IPsecSaEnum0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecSaEnum1(
@@ -4612,6 +4733,7 @@ IPsecSaEnum1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaDestroyEnumHandle0(
@@ -4619,6 +4741,7 @@ IPsecSaDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaDbGetSecurityInfo0(
@@ -4631,6 +4754,7 @@ IPsecSaDbGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IPsecSaDbSetSecurityInfo0(
@@ -4650,6 +4774,7 @@ IPsecSaDbSetSecurityInfo0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospGetStatistics0(
@@ -4657,6 +4782,7 @@ IPsecDospGetStatistics0(
    _Out_ IPSEC_DOSP_STATISTICS0* idpStatistics
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospStateCreateEnumHandle0(
@@ -4665,6 +4791,7 @@ IPsecDospStateCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospStateEnum0(
@@ -4675,6 +4802,7 @@ IPsecDospStateEnum0(
    _Out_ UINT32* numEntries
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospStateDestroyEnumHandle0(
@@ -4682,6 +4810,7 @@ IPsecDospStateDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospGetSecurityInfo0(
@@ -4694,6 +4823,7 @@ IPsecDospGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IPsecDospSetSecurityInfo0(
@@ -4741,6 +4871,7 @@ typedef struct _IPSEC_KEY_MANAGER_CALLBACKS0 {
    IPSEC_KEY_MANAGER_NOTIFY_KEY0 keyNotify;
 } IPSEC_KEY_MANAGER_CALLBACKS0;
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecKeyManagerAddAndRegister0(
@@ -4750,6 +4881,7 @@ IPsecKeyManagerAddAndRegister0(
    _Out_ HANDLE* keyMgmtHandle
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecKeyManagerUnregisterAndDelete0(
@@ -4757,6 +4889,7 @@ IPsecKeyManagerUnregisterAndDelete0(
   _In_ HANDLE keyMgmtHandle
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecKeyManagersGet0(
@@ -4765,6 +4898,7 @@ IPsecKeyManagersGet0(
    _Out_ UINT32* numEntries
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecKeyManagerGetSecurityInfoByKey0(
@@ -4778,6 +4912,7 @@ IPsecKeyManagerGetSecurityInfoByKey0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IPsecKeyManagerSetSecurityInfoByKey0(
@@ -4798,6 +4933,7 @@ IPsecKeyManagerSetSecurityInfoByKey0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextGetStatistics0(
@@ -4806,6 +4942,7 @@ IkeextGetStatistics0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IkeextGetStatistics1(
@@ -4814,6 +4951,7 @@ IkeextGetStatistics1(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaDeleteById0(
@@ -4821,6 +4959,7 @@ IkeextSaDeleteById0(
    _In_ UINT64 id
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaGetById0(
@@ -4830,6 +4969,7 @@ IkeextSaGetById0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IkeextSaGetById1(
@@ -4841,6 +4981,7 @@ IkeextSaGetById1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IkeextSaGetById2(
@@ -4851,6 +4992,7 @@ IkeextSaGetById2(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaCreateEnumHandle0(
@@ -4859,6 +5001,7 @@ IkeextSaCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaEnum0(
@@ -4870,6 +5013,7 @@ IkeextSaEnum0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 IkeextSaEnum1(
@@ -4882,6 +5026,7 @@ IkeextSaEnum1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 IkeextSaEnum2(
@@ -4894,6 +5039,7 @@ IkeextSaEnum2(
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaDestroyEnumHandle0(
@@ -4901,6 +5047,7 @@ IkeextSaDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaDbGetSecurityInfo0(
@@ -4913,6 +5060,7 @@ IkeextSaDbGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 IkeextSaDbSetSecurityInfo0(
@@ -4930,6 +5078,7 @@ IkeextSaDbSetSecurityInfo0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmNetEventCreateEnumHandle0(
@@ -4938,6 +5087,7 @@ FwpmNetEventCreateEnumHandle0(
    _Out_ HANDLE* enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmNetEventEnum0(
@@ -4949,6 +5099,7 @@ FwpmNetEventEnum0(
    );
 
 #if (NTDDI_VERSION >= NTDDI_WIN7)
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmNetEventEnum1(
@@ -4961,6 +5112,7 @@ FwpmNetEventEnum1(
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmNetEventEnum2(
@@ -4973,6 +5125,7 @@ FwpmNetEventEnum2(
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 #if (NTDDI_VERSION >= NTDDI_WINTHRESHOLD)
+_Win32_metadata_supported_os_(windows10.0.14393)
 DWORD
 WINAPI
 FwpmNetEventEnum3(
@@ -5008,6 +5161,7 @@ FwpmNetEventEnum5(
    );
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS5)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmNetEventDestroyEnumHandle0(
@@ -5015,6 +5169,7 @@ FwpmNetEventDestroyEnumHandle0(
    _Inout_ HANDLE enumHandle
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmNetEventsGetSecurityInfo0(
@@ -5027,6 +5182,7 @@ FwpmNetEventsGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 WINAPI
 FwpmNetEventsSetSecurityInfo0(
@@ -5044,6 +5200,7 @@ typedef void (CALLBACK *FWPM_NET_EVENT_CALLBACK0)(
                            _In_ const FWPM_NET_EVENT1* event
                            );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmNetEventSubscribe0(
@@ -5054,6 +5211,7 @@ FwpmNetEventSubscribe0(
    _Out_ HANDLE* eventsHandle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmNetEventUnsubscribe0(
@@ -5061,6 +5219,7 @@ FwpmNetEventUnsubscribe0(
    _Inout_ HANDLE eventsHandle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmNetEventSubscriptionsGet0(
@@ -5078,6 +5237,7 @@ typedef void (CALLBACK *FWPM_NET_EVENT_CALLBACK1)(
                            _In_ const FWPM_NET_EVENT2* event
                            );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmNetEventSubscribe1(
@@ -5097,6 +5257,7 @@ typedef void (CALLBACK *FWPM_NET_EVENT_CALLBACK2)(
                            _In_ const FWPM_NET_EVENT3* event
                            );
 
+_Win32_metadata_supported_os_(windows10.0.14393)
 DWORD
 WINAPI
 FwpmNetEventSubscribe2(
@@ -5189,6 +5350,7 @@ FwpmDynamicKeywordUnsubscribe0(
 //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmSystemPortsGet0(
@@ -5201,6 +5363,7 @@ typedef void (CALLBACK *FWPM_SYSTEM_PORTS_CALLBACK0)(
                            _In_ const FWPM_SYSTEM_PORTS0* sysPorts
                            );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmSystemPortsSubscribe0(
@@ -5211,6 +5374,7 @@ FwpmSystemPortsSubscribe0(
    _Out_ HANDLE* sysPortsHandle
    );
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 FwpmSystemPortsUnsubscribe0(
@@ -5222,6 +5386,7 @@ FwpmSystemPortsUnsubscribe0(
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionGetById0(
@@ -5230,6 +5395,7 @@ FwpmConnectionGetById0(
    _Outptr_ FWPM_CONNECTION0** connection
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionEnum0(
@@ -5240,6 +5406,7 @@ FwpmConnectionEnum0(
     _Out_ UINT32 *numEntriesReturned
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionCreateEnumHandle0(
@@ -5248,6 +5415,7 @@ FwpmConnectionCreateEnumHandle0(
    _Out_ HANDLE *enumHandle
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionDestroyEnumHandle0(
@@ -5257,6 +5425,7 @@ FwpmConnectionDestroyEnumHandle0(
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionGetSecurityInfo0(
@@ -5269,6 +5438,7 @@ FwpmConnectionGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionSetSecurityInfo0(
@@ -5289,6 +5459,7 @@ typedef void (CALLBACK *FWPM_CONNECTION_CALLBACK0)(
    _In_ const FWPM_CONNECTION0* connection
 );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionSubscribe0(
@@ -5299,6 +5470,7 @@ FwpmConnectionSubscribe0(
    _Out_ HANDLE* eventsHandle
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionUnsubscribe0(
@@ -5306,6 +5478,7 @@ FwpmConnectionUnsubscribe0(
     _Inout_ HANDLE eventsHandle
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmConnectionSubscriptionsGet0(
@@ -5326,6 +5499,7 @@ typedef DWORD (CALLBACK *FWPM_VSWITCH_EVENT_CALLBACK0)(
                            _In_ const FWPM_VSWITCH_EVENT0* vSwitchEvent
                            );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmvSwitchEventSubscribe0(
@@ -5336,6 +5510,7 @@ FwpmvSwitchEventSubscribe0(
    _Out_ HANDLE* subscriptionHandle
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmvSwitchEventUnsubscribe0(
@@ -5346,6 +5521,7 @@ FwpmvSwitchEventUnsubscribe0(
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmvSwitchEventsGetSecurityInfo0(
@@ -5358,6 +5534,7 @@ FwpmvSwitchEventsGetSecurityInfo0(
    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
    );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 FwpmvSwitchEventsSetSecurityInfo0(
