diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fxsutility.h b/generation/WinSDK/RecompiledIdlHeaders/um/fxsutility.h
index 41bb83ad..6ff0dc7c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fxsutility.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fxsutility.h
@@ -23,6 +23,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -36,7 +39,9 @@ typedef enum {
 	  SEND_TO_FAX_RECIPIENT_ATTACHMENT
 } SendToMode;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 BOOL WINAPI CanSendToFaxRecipient();
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI SendToFaxRecipient(SendToMode sndMode, LPCWSTR lpFileName);
 
 #ifdef __cplusplus
