diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gb18030.h b/generation/WinSDK/RecompiledIdlHeaders/um/gb18030.h
index 1ef27e11..7d29d15a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gb18030.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gb18030.h
@@ -1,6 +1,7 @@
 #ifndef __GB18030_H
 #define __GB18030_H
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -95,6 +96,8 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 STDAPI_(DWORD) NlsDllCodePageTranslation(
     DWORD CodePage,
     DWORD dwFlags,
