diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
index a9750e89..6b4bf3e9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
@@ -19,6 +19,9 @@
 #ifndef _GDIPLUSHEADERS_H
 #define _GDIPLUSHEADERS_H
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -432,6 +435,7 @@ public:
     REAL GetVerticalResolution();
     UINT GetFlags();
     Status GetRawFormat(OUT GUID *format);
+_Win32_metadata_set_last_error_
     PixelFormat GetPixelFormat();
 
     INT GetPaletteSize();
