diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
index a9750e89..c85eaaf5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplusheaders.h
@@ -19,6 +19,7 @@
 #ifndef _GDIPLUSHEADERS_H
 #define _GDIPLUSHEADERS_H
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -432,6 +433,8 @@ public:
     REAL GetVerticalResolution();
     UINT GetFlags();
     Status GetRawFormat(OUT GUID *format);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
     PixelFormat GetPixelFormat();
 
     INT GetPaletteSize();
@@ -562,9 +565,11 @@ public:
                     IN PixelFormat format,
                     OUT BitmapData* lockedBitmapData);
     Status UnlockBits(IN BitmapData* lockedBitmapData);
+_Win32_metadata_supported_os_(windows5.0)
     Status GetPixel(IN INT x,
                     IN INT y,
                     OUT Color *color);
+_Win32_metadata_supported_os_(windows5.0)
     Status SetPixel(IN INT x,
                     IN INT y,
                     IN const Color &color);
