diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluspen.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluspen.h
index 06a7edf9..f8fbabe5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluspen.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluspen.h
@@ -14,6 +14,7 @@
 #ifndef _GDIPLUSPEN_H
 #define _GDIPLUSPEN_H
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -188,12 +189,14 @@ public:
                                                     &(customCap->nativeCap)));
     }
 
+_Win32_metadata_supported_os_(windows5.0)
     Status SetMiterLimit(IN REAL miterLimit)
     {
         return SetStatus(DllExports::GdipSetPenMiterLimit(nativePen, 
                                                     miterLimit));
     }
 
+_Win32_metadata_supported_os_(windows5.0)
     REAL GetMiterLimit() const
     {
         REAL miterLimit;
