diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplustypes.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplustypes.h
index c4560742..8b0c9398 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdiplustypes.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdiplustypes.h
@@ -15,6 +15,9 @@
 #ifndef _GDIPLUSTYPES_H
 #define _GDIPLUSTYPES_H
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -140,6 +143,7 @@ _Return_type_success_(return == 0) enum Status
 
 class SizeF
 {
+_Win32_metadata_supported_os_(windows6.1)
 public:
     SizeF()
     {
@@ -356,6 +360,7 @@ class RectF
 {
 public:
 
+_Win32_metadata_supported_os_(windows6.1)
     RectF()
     {
         X = Y = Width = Height = 0.0f;
