diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/GPEdit.h b/generation/WinSDK/RecompiledIdlHeaders/um/GPEdit.h
index e9da7395..0bc1e2a4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/GPEdit.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/GPEdit.h
@@ -162,6 +162,7 @@ DEFINE_GUID(IID_IRSOPInformation, 0x9a5a81b5, 0xd9c7, 0x49ef, 0x9d, 0x11, 0xdd,
 #ifndef _GPEDIT_H_
 #define _GPEDIT_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -228,6 +229,7 @@ typedef enum _GROUP_POLICY_HINT_TYPE {
 
 #undef INTERFACE
 #define INTERFACE   IGPEInformation
+_Win32_metadata_supported_os_(windows6.0.6000)
 DECLARE_INTERFACE_(IGPEInformation, IUnknown)
 {
     // *** IUnknown methods ***
@@ -384,6 +386,7 @@ typedef IGPEInformation *LPGPEINFORMATION;
 
 #undef INTERFACE
 #define INTERFACE   IGroupPolicyObject
+_Win32_metadata_supported_os_(windows6.0.6000)
 DECLARE_INTERFACE_(IGroupPolicyObject, IUnknown)
 {
     // *** IUnknown methods ***
@@ -644,6 +647,7 @@ typedef IGroupPolicyObject *LPGROUPPOLICYOBJECT;
 
 #undef INTERFACE
 #define INTERFACE   IRSOPInformation
+_Win32_metadata_supported_os_(windows6.0.6000)
 DECLARE_INTERFACE_(IRSOPInformation, IUnknown)
 {
     // *** IUnknown methods ***
@@ -706,6 +710,7 @@ typedef IRSOPInformation *LPRSOPINFORMATION;
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
@@ -728,6 +733,7 @@ CreateGPOLink(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
@@ -748,6 +754,7 @@ DeleteGPOLink(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
@@ -796,6 +803,7 @@ typedef struct tag_GPOBROWSEINFO
 } GPOBROWSEINFO, *LPGPOBROWSEINFO;
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
@@ -819,6 +827,7 @@ BrowseForGPO(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
@@ -844,6 +853,7 @@ ImportRSoPData(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 GPEDITAPI
 HRESULT
 WINAPI
