diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gpmgmt.h b/generation/WinSDK/RecompiledIdlHeaders/um/gpmgmt.h
index 5de600ad..339d5116 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gpmgmt.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gpmgmt.h
@@ -755,6 +755,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 typedef /* [public][public][public][public][public][public] */ 
@@ -929,6 +932,7 @@ EXTERN_C const IID IID_IGPM;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F5FAE809-3BD6-4DA9-A65E-17665B41D763")
     IGPM : public IDispatch
     {
@@ -1285,6 +1289,7 @@ EXTERN_C const IID IID_IGPMDomain;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6B21CC14-5A00-4F44-A738-FEEC8A94C7E3")
     IGPMDomain : public IDispatch
     {
@@ -1587,6 +1592,7 @@ EXTERN_C const IID IID_IGPMBackupDir;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("B1568BED-0A93-4ACC-810F-AFE7081019B9")
     IGPMBackupDir : public IDispatch
     {
@@ -1770,6 +1776,7 @@ EXTERN_C const IID IID_IGPMSitesContainer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("4725A899-2782-4D27-A6BB-D499246FFD72")
     IGPMSitesContainer : public IDispatch
     {
@@ -1979,6 +1986,7 @@ EXTERN_C const IID IID_IGPMSearchCriteria;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D6F11C42-829B-48D4-83F5-3615B67DFC22")
     IGPMSearchCriteria : public IDispatch
     {
@@ -2136,6 +2144,7 @@ EXTERN_C const IID IID_IGPMTrustee;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("3B466DA8-C1A4-4B2A-999A-BEFCDD56CEFB")
     IGPMTrustee : public IDispatch
     {
@@ -2337,6 +2346,7 @@ EXTERN_C const IID IID_IGPMPermission;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("35EBCA40-E1A1-4A02-8905-D79416FB464A")
     IGPMPermission : public IDispatch
     {
@@ -2538,6 +2548,7 @@ EXTERN_C const IID IID_IGPMSecurityInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("B6C31ED4-1C93-4D3E-AE84-EB6D61161B60")
     IGPMSecurityInfo : public IDispatch
     {
@@ -2754,6 +2765,7 @@ EXTERN_C const IID IID_IGPMBackup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D8A16A35-3B0D-416B-8D02-4DF6F95A7119")
     IGPMBackup : public IDispatch
     {
@@ -3032,6 +3044,7 @@ EXTERN_C const IID IID_IGPMBackupCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("C786FC0F-26D8-4BAB-A745-39CA7E800CAC")
     IGPMBackupCollection : public IDispatch
     {
@@ -3217,6 +3230,7 @@ EXTERN_C const IID IID_IGPMSOM;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("C0A7F09E-05A1-4F0C-8158-9E5C33684F6B")
     IGPMSOM : public IDispatch
     {
@@ -3491,6 +3505,7 @@ EXTERN_C const IID IID_IGPMSOMCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("ADC1688E-00E4-4495-ABBA-BED200DF0CAB")
     IGPMSOMCollection : public IDispatch
     {
@@ -3668,6 +3683,7 @@ EXTERN_C const IID IID_IGPMWMIFilter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("EF2FF9B4-3C27-459A-B979-038305CEC75D")
     IGPMWMIFilter : public IDispatch
     {
@@ -3908,6 +3924,7 @@ EXTERN_C const IID IID_IGPMWMIFilterCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5782D582-1A36-4661-8A94-C3C32551945B")
     IGPMWMIFilterCollection : public IDispatch
     {
@@ -4085,6 +4102,7 @@ EXTERN_C const IID IID_IGPMRSOP;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("49ED785A-3237-4FF2-B1F0-FDF5A8D5A1EE")
     IGPMRSOP : public IDispatch
     {
@@ -4684,6 +4702,7 @@ EXTERN_C const IID IID_IGPMGPO;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("58CC4352-1CA3-48E5-9864-1DA4D6E0D60F")
     IGPMGPO : public IDispatch
     {
@@ -5255,6 +5274,7 @@ EXTERN_C const IID IID_IGPMGPOCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F0F0D5CF-70CA-4C39-9E29-B642F8726C01")
     IGPMGPOCollection : public IDispatch
     {
@@ -5434,6 +5454,7 @@ EXTERN_C const IID IID_IGPMGPOLink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("434B99BD-5DE7-478A-809C-C251721DF70C")
     IGPMGPOLink : public IDispatch
     {
@@ -5683,6 +5704,7 @@ EXTERN_C const IID IID_IGPMGPOLinksCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("189D7B68-16BD-4D0D-A2EC-2E6AA2288C7F")
     IGPMGPOLinksCollection : public IDispatch
     {
@@ -5862,6 +5884,7 @@ EXTERN_C const IID IID_IGPMCSECollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2E52A97D-0A4A-4A6F-85DB-201622455DA0")
     IGPMCSECollection : public IDispatch
     {
@@ -6041,6 +6064,7 @@ EXTERN_C const IID IID_IGPMClientSideExtension;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("69DA7488-B8DB-415E-9266-901BE4D49928")
     IGPMClientSideExtension : public IDispatch
     {
@@ -6229,6 +6253,7 @@ EXTERN_C const IID IID_IGPMAsyncCancel;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DDC67754-BE67-4541-8166-F48166868C9C")
     IGPMAsyncCancel : public IDispatch
     {
@@ -6374,6 +6399,7 @@ EXTERN_C const IID IID_IGPMAsyncProgress;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6AAC29F8-5948-4324-BF70-423818942DBC")
     IGPMAsyncProgress : public IDispatch
     {
@@ -6539,6 +6565,7 @@ EXTERN_C const IID IID_IGPMStatusMsgCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9B6E1AF0-1A92-40F3-A59D-F36AC1F728B7")
     IGPMStatusMsgCollection : public IDispatch
     {
@@ -6718,6 +6745,7 @@ EXTERN_C const IID IID_IGPMStatusMessage;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8496C22F-F3DE-4A1F-8F58-603CAAA93D7B")
     IGPMStatusMessage : public IDispatch
     {
@@ -6924,6 +6952,7 @@ EXTERN_C const IID IID_IGPMConstants;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("50EF73E6-D35C-4C8D-BE63-7EA5D2AAC5C4")
     IGPMConstants : public IDispatch
     {
@@ -7860,6 +7889,7 @@ EXTERN_C const IID IID_IGPMResult;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("86DFF7E9-F76F-42AB-9570-CEBC6BE8A52D")
     IGPMResult : public IDispatch
     {
@@ -8031,6 +8061,7 @@ EXTERN_C const IID IID_IGPMMapEntryCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("BB0BF49B-E53F-443F-B807-8BE22BFB6D42")
     IGPMMapEntryCollection : public IDispatch
     {
@@ -8210,6 +8241,7 @@ EXTERN_C const IID IID_IGPMMapEntry;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8E79AD06-2381-4444-BE4C-FF693E6E6F2B")
     IGPMMapEntry : public IDispatch
     {
@@ -8398,6 +8430,7 @@ EXTERN_C const IID IID_IGPMMigrationTable;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("48F823B1-EFAF-470B-B6ED-40D14EE1A4EC")
     IGPMMigrationTable : public IDispatch
     {
@@ -8684,6 +8717,7 @@ EXTERN_C const IID IID_IGPMBackupDirEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F8DC55ED-3BA0-4864-AAD4-D365189EE1D5")
     IGPMBackupDirEx : public IDispatch
     {
@@ -8880,6 +8914,7 @@ EXTERN_C const IID IID_IGPMStarterGPOBackupCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("C998031D-ADD0-4bb5-8DEA-298505D8423B")
     IGPMStarterGPOBackupCollection : public IDispatch
     {
@@ -9057,6 +9092,7 @@ EXTERN_C const IID IID_IGPMStarterGPOBackup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("51D98EDA-A87E-43dd-B80A-0B66EF1938D6")
     IGPMStarterGPOBackup : public IDispatch
     {
@@ -9348,6 +9384,7 @@ EXTERN_C const IID IID_IGPM2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("00238F8A-3D86-41ac-8F5E-06A6638A634A")
     IGPM2 : public IGPM
     {
@@ -9663,6 +9700,7 @@ EXTERN_C const IID IID_IGPMStarterGPO;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DFC3F61B-8880-4490-9337-D29C7BA8C2F0")
     IGPMStarterGPO : public IDispatch
     {
@@ -10128,6 +10166,7 @@ EXTERN_C const IID IID_IGPMStarterGPOCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2E522729-2219-44ad-933A-64DFD650C423")
     IGPMStarterGPOCollection : public IDispatch
     {
@@ -10305,6 +10344,7 @@ EXTERN_C const IID IID_IGPMDomain2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7CA6BB8B-F1EB-490a-938D-3C4E51C768E6")
     IGPMDomain2 : public IGPMDomain
     {
@@ -10660,6 +10700,7 @@ EXTERN_C const IID IID_IGPMConstants2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("05AE21B0-AC09-4032-A26F-9E7DA786DC19")
     IGPMConstants2 : public IGPMConstants
     {
@@ -11542,6 +11583,7 @@ EXTERN_C const IID IID_IGPMGPO2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8A66A210-B78B-4d99-88E2-C306A817C925")
     IGPMGPO2 : public IGPMGPO
     {
@@ -13150,14 +13192,22 @@ unsigned char * __RPC_USER  VARIANT_UserMarshal(  __RPC__in unsigned long *, __R
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
 void                      __RPC_USER  VARIANT_UserFree(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  VARIANT_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
 /* end of Additional Prototypes */
