diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/handleapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/handleapi.h
index fe2d256b..c313b2fe 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/handleapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/handleapi.h
@@ -17,6 +17,9 @@
 #include <apisetcconv.h>
 #include <minwindef.h>
 
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -32,6 +35,8 @@ extern "C" {
 //
 // Prototypes
 //
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -39,6 +44,8 @@ CloseHandle(
     _In_ _Post_ptr_invalid_ HANDLE hObject
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -58,6 +65,7 @@ DuplicateHandle(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 BOOL
 WINAPI
@@ -72,6 +80,8 @@ CompareObjectHandles(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -80,6 +90,8 @@ GetHandleInformation(
     _Out_ LPDWORD lpdwFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
