diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
index 54c1445d..b104d858 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
@@ -15,6 +15,9 @@
 
 #include <apiset.h>
 #include <apisetcconv.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #include <minwindef.h>
 #include <minwinbase.h>
 
@@ -53,6 +56,7 @@ typedef PHEAP_SUMMARY LPHEAP_SUMMARY;
 // Prototypes
 //
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -63,6 +67,7 @@ HeapCreate(
     _In_ SIZE_T dwMaximumSize
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -96,6 +101,7 @@ HeapReAlloc(
     _In_ SIZE_T dwBytes
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -115,6 +121,7 @@ HeapSize(
     _In_ LPCVOID lpMem
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -122,6 +129,7 @@ GetProcessHeap(
     VOID
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 SIZE_T
 WINAPI
@@ -130,6 +138,7 @@ HeapCompact(
     _In_ DWORD dwFlags
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -175,6 +184,7 @@ HeapSummary(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -183,6 +193,7 @@ GetProcessHeaps(
     _Out_writes_to_(NumberOfHeaps,return) PHANDLE ProcessHeaps
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -190,6 +201,7 @@ HeapLock(
     _In_ HANDLE hHeap
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -203,6 +215,7 @@ HeapUnlock(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -211,6 +224,7 @@ HeapWalk(
     _Inout_ LPPROCESS_HEAP_ENTRY lpEntry
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
