diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/httpserv.h b/generation/WinSDK/RecompiledIdlHeaders/um/httpserv.h
index 54b2ef39..9b822ff1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/httpserv.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/httpserv.h
@@ -4,6 +4,7 @@
 #ifndef _HTTPSERV_H_
 #define _HTTPSERV_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -1548,6 +1549,7 @@ IHttpContext
         VOID
     ) const = 0;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     virtual
     HRESULT
     CloneContext(
@@ -1593,6 +1595,7 @@ IHttpContext
         _Out_   DWORD **    ppDiffedVariableIndices
     ) = 0;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     virtual
     HRESULT
     CancelIo(
@@ -1944,6 +1947,7 @@ public:
         _Out_ DWORD *           pSecondsToLive
     ) const = 0;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     virtual
     HRESULT
     AccessCheck(
@@ -2472,6 +2476,7 @@ public:
         VOID
     ) const = 0;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     virtual
     VOID
     SetFlags(
