diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/httpcompression.h b/generation/WinSDK/RecompiledIdlHeaders/um/httpcompression.h
index 0b699332..cf7e510f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/httpcompression.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/httpcompression.h
@@ -2,6 +2,7 @@
 #define _HTTP_COMPRESSION_H_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -77,6 +78,7 @@ CreateCompression(
 //
 // Compress data
 //
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT WINAPI Compress(
     IN  OUT PVOID           context,            // compression context
     IN      CONST BYTE *    input_buffer,       // input buffer
