diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/IcmpAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/IcmpAPI.h
index d864f1eb..6ef0623a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/IcmpAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/IcmpAPI.h
@@ -28,6 +28,7 @@ Notes:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 //
 // Establish DLL function linkage if supported by the current build
@@ -75,12 +76,15 @@ extern "C" {
 //
 //--
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 HANDLE
 WINAPI
 IcmpCreateFile(
     VOID
-    );
+    )
+    _Win32_metadata_raii_free_(IcmpCloseHandle);
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
 //++
@@ -104,12 +108,15 @@ IcmpCreateFile(
 //
 //--
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 HANDLE
 WINAPI
 Icmp6CreateFile(
     VOID
-    );
+    )
+    _Win32_metadata_raii_free_(IcmpCloseHandle);
 #endif
 
 //++
@@ -133,6 +140,8 @@ Icmp6CreateFile(
 //
 //--
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 BOOL
 WINAPI
@@ -194,6 +203,8 @@ IcmpCloseHandle(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -283,6 +294,8 @@ IcmpSendEcho(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -312,6 +325,8 @@ IcmpSendEcho2(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 #if (NTDDI_VERSION >= NTDDI_VISTASP1)
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -343,6 +358,8 @@ IcmpSendEcho2Ex(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -397,6 +414,8 @@ Icmp6SendEcho2(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -407,6 +426,8 @@ IcmpParseReplies(
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
