diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/inkrenderer.h b/generation/WinSDK/RecompiledIdlHeaders/um/inkrenderer.h
index eadc6a38..36376942 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/inkrenderer.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/inkrenderer.h
@@ -87,6 +87,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 #if (NTDDI_VERSION >= NTDDI_WINTHRESHOLD)
@@ -114,6 +115,7 @@ EXTERN_C const IID IID_IInkD2DRenderer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("407fb1de-f85a-4150-97cf-b7fb274fb4f8")
     IInkD2DRenderer : public IUnknown
     {
