diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/inputpanelconfiguration.h b/generation/WinSDK/RecompiledIdlHeaders/um/inputpanelconfiguration.h
index 5d57c288..4d4801b4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/inputpanelconfiguration.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/inputpanelconfiguration.h
@@ -9,6 +9,10 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
 #endif
@@ -105,6 +109,7 @@ EXTERN_C const IID IID_IInputPanelConfiguration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("41C81592-514C-48BD-A22E-E6AF638521A6")
     IInputPanelConfiguration : public IUnknown
     {
@@ -188,6 +193,7 @@ EXTERN_C const IID IID_IInputPanelInvocationConfiguration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("A213F136-3B45-4362-A332-EFB6547CD432")
     IInputPanelInvocationConfiguration : public IUnknown
     {
