diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/InteractionContext.h b/generation/WinSDK/RecompiledIdlHeaders/um/InteractionContext.h
index 6046390a..943fc7f5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/InteractionContext.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/InteractionContext.h
@@ -7,6 +7,9 @@
 #define INTERACTION_CONTEXT_H
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -308,16 +311,19 @@ DECLARE_HANDLE(HINTERACTIONCONTEXT);
 
 #define INERTIA_PARAMETER_INVALID_VALUE    FLT_MAX
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 CreateInteractionContext(
     _Out_writes_(1) HINTERACTIONCONTEXT *interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 DestroyInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 RegisterOutputCallbackInteractionContext(
@@ -334,6 +340,7 @@ RegisterOutputCallbackInteractionContext2(
     _In_opt_ void *clientData);
 #endif
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetInteractionConfigurationInteractionContext(
@@ -341,6 +348,7 @@ SetInteractionConfigurationInteractionContext(
     _In_ UINT32 configurationCount,
     _In_reads_(configurationCount) const INTERACTION_CONTEXT_CONFIGURATION *configuration);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 GetInteractionConfigurationInteractionContext(
@@ -348,6 +356,7 @@ GetInteractionConfigurationInteractionContext(
     _In_ UINT32 configurationCount,
     _Inout_updates_(configurationCount) INTERACTION_CONTEXT_CONFIGURATION *configuration);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetPropertyInteractionContext(
@@ -355,6 +364,7 @@ SetPropertyInteractionContext(
     _In_ INTERACTION_CONTEXT_PROPERTY contextProperty,
     _In_ UINT32 value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 GetPropertyInteractionContext(
@@ -362,6 +372,7 @@ GetPropertyInteractionContext(
     _In_ INTERACTION_CONTEXT_PROPERTY contextProperty,
     _Out_writes_(1) UINT32 *value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetInertiaParameterInteractionContext(
@@ -369,6 +380,7 @@ SetInertiaParameterInteractionContext(
     _In_ INERTIA_PARAMETER inertiaParameter,
     _In_ float value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 GetInertiaParameterInteractionContext(
@@ -376,6 +388,7 @@ GetInertiaParameterInteractionContext(
     _In_ INERTIA_PARAMETER inertiaParameter,
     _Out_writes_(1) float *value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetCrossSlideParametersInteractionContext(
@@ -383,6 +396,7 @@ SetCrossSlideParametersInteractionContext(
     _In_ UINT32 parameterCount,
     _In_reads_(parameterCount) CROSS_SLIDE_PARAMETER *crossSlideParameters);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
     GetCrossSlideParameterInteractionContext(
@@ -434,6 +448,7 @@ GetTranslationParameterInteractionContext(
     _Out_ float *value);
 #endif
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetMouseWheelParameterInteractionContext(
@@ -441,6 +456,7 @@ SetMouseWheelParameterInteractionContext(
     _In_ MOUSE_WHEEL_PARAMETER parameter,
     _In_ float value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 GetMouseWheelParameterInteractionContext(
@@ -448,11 +464,13 @@ GetMouseWheelParameterInteractionContext(
     _In_ MOUSE_WHEEL_PARAMETER parameter,
     _Out_writes_(1) float *value);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 ResetInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 GetStateInteractionContext(
@@ -460,18 +478,21 @@ GetStateInteractionContext(
     _In_opt_ const POINTER_INFO *pointerInfo,
     _Out_writes_(1) INTERACTION_STATE *state);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 AddPointerInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext,
     _In_ UINT32 pointerId);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 RemovePointerInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext,
     _In_ UINT32 pointerId);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 ProcessPointerFramesInteractionContext(
@@ -480,6 +501,7 @@ ProcessPointerFramesInteractionContext(
     _In_ UINT32 pointerCount,
     _In_reads_(entriesCount * pointerCount) const POINTER_INFO *pointerInfo);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 BufferPointerPacketsInteractionContext(
@@ -487,21 +509,25 @@ BufferPointerPacketsInteractionContext(
     _In_ UINT32 entriesCount,
     _In_reads_(entriesCount) const POINTER_INFO *pointerInfo);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 ProcessBufferedPacketsInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 ProcessInertiaInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 StopInteractionContext(
     _In_ HINTERACTIONCONTEXT interactionContext);
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT
 WINAPI
 SetPivotInteractionContext(
