diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/intsafe.h b/generation/WinSDK/RecompiledIdlHeaders/shared/intsafe.h
index 412182be..a5577c3f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/intsafe.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/intsafe.h
@@ -10,6 +10,7 @@
 #define _INTSAFE_H_INCLUDED_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 
 #if (_MSC_VER > 1000)
@@ -374,6 +375,7 @@ C_ASSERT(sizeof(UINT_PTR) == sizeof(ULONG_PTR));
 //
 // INT8 -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -400,6 +402,7 @@ Int8ToUChar(
 //
 // INT8 -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -431,6 +434,7 @@ Int8ToUInt8(
 //
 // INT8 -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -467,6 +471,7 @@ Int8ToUShort(
 //
 // INT8 -> UINT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -498,6 +503,7 @@ Int8ToUInt(
 //
 // INT8 -> UINT_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -524,6 +530,7 @@ Int8ToUIntPtr(
 //
 // INT8 -> ULONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -550,6 +557,7 @@ Int8ToULong(
 //
 // INT8 -> ULONG_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -586,6 +594,7 @@ Int8ToULongPtr(
 //
 // INT8 -> ULONGLONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -642,6 +651,7 @@ Int8ToULongLong(
 //
 // UINT8 -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -668,6 +678,7 @@ UInt8ToInt8(
 //
 // UINT8 -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 UInt8ToChar(
@@ -685,6 +696,7 @@ UInt8ToChar(
 //
 // BYTE -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -711,6 +723,7 @@ ByteToInt8(
 //
 // BYTE -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 ByteToChar(
@@ -728,6 +741,7 @@ ByteToChar(
 //
 // SHORT -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -796,6 +810,7 @@ ShortToChar(
 //
 // SHORT -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -863,6 +878,7 @@ ShortToUShort(
 //
 // SHORT -> UINT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -894,6 +910,7 @@ ShortToUInt(
 //
 // SHORT -> UINT_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -920,6 +937,7 @@ ShortToUIntPtr(
 //
 // SHORT -> ULONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -946,6 +964,7 @@ ShortToULong(
 //
 // SHORT -> ULONG_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -977,6 +996,7 @@ ShortToULongPtr(
 //
 // SHORT -> DWORD_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1003,6 +1023,7 @@ ShortToDWordPtr(
 //
 // SHORT -> ULONGLONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1169,6 +1190,7 @@ ShortToULongLong(
 //
 // USHORT -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1237,6 +1259,7 @@ UShortToChar(
 //
 // USHORT -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1369,6 +1392,7 @@ UShortToShort(
 //
 // INT -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1442,6 +1466,7 @@ IntToChar(
 //
 // INT -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1791,6 +1816,7 @@ IntToULongLong(
 //
 // INT_PTR -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1817,6 +1843,7 @@ IntPtrToInt8(
 //
 // INT_PTR -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1843,6 +1870,7 @@ IntPtrToUChar(
 //
 // INT_PTR -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 IntPtrToChar(
@@ -1859,6 +1887,7 @@ IntPtrToChar(
 //
 // INT_PTR -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1890,6 +1919,7 @@ IntPtrToUInt8(
 //
 // INT_PTR -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -1921,6 +1951,7 @@ IntPtrToShort(
 //
 // INT_PTR -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2063,6 +2094,7 @@ IntPtrToLong(
 //
 // INT_PTR -> LONG_PTR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2207,6 +2239,7 @@ IntPtrToULongLong(
 //
 // UINT -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2275,6 +2308,7 @@ UIntToChar(
 //
 // UINT -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2561,6 +2595,7 @@ UIntToLongPtr(
 //
 // UINT_PTR -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2587,6 +2622,7 @@ UIntPtrToInt8(
 //
 // UINT_PTR -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2613,6 +2649,7 @@ UIntPtrToUChar(
 //
 // UINT_PTR -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 UIntPtrToChar(
@@ -2629,6 +2666,7 @@ UIntPtrToChar(
 //
 // UINT_PTR -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2660,6 +2698,7 @@ UIntPtrToUInt8(
 //
 // UINT_PTR -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2686,6 +2725,7 @@ UIntPtrToShort(
 //
 // UINT_PTR -> INT16 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2712,6 +2752,7 @@ UIntPtrToInt16(
 //
 // UINT_PTR -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2738,6 +2779,7 @@ UIntPtrToUShort(
 //
 // UINT_PTR -> UINT16 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2927,6 +2969,7 @@ UIntPtrToULong(
 #ifdef _WIN64
 #define UIntPtrToLongLong   ULongLongToLongLong
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -2962,6 +3005,7 @@ UIntPtrToLongLong(
 //
 // LONG -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3030,6 +3074,7 @@ LongToChar(
 //
 // LONG -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3354,6 +3399,7 @@ LongToULongLong(
 //
 // LONG_PTR -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3380,6 +3426,7 @@ LongPtrToInt8(
 //
 // LONG_PTR -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3406,6 +3453,7 @@ LongPtrToUChar(
 //
 // LONG_PTR -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 LongPtrToChar(
@@ -3422,6 +3470,7 @@ LongPtrToChar(
 //
 // LONG_PTR -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3453,6 +3502,7 @@ LongPtrToUInt8(
 //
 // LONG_PTR -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3484,6 +3534,7 @@ LongPtrToShort(
 //
 // LONG_PTR -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3760,6 +3811,7 @@ LongPtrToULongLong(
 //
 // ULONG -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -3828,6 +3880,7 @@ ULongToChar(
 //
 // ULONG -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4069,6 +4122,7 @@ ULongToLongPtr(
 //
 // ULONG_PTR -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4095,6 +4149,7 @@ ULongPtrToInt8(
 //
 // ULONG_PTR -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4121,6 +4176,7 @@ ULongPtrToUChar(
 //
 // ULONG_PTR -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 ULongPtrToChar(
@@ -4137,6 +4193,7 @@ ULongPtrToChar(
 //
 // ULONG_PTR -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4168,6 +4225,7 @@ ULongPtrToUInt8(
 //
 // ULONG_PTR -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4199,6 +4257,7 @@ ULongPtrToShort(
 //
 // ULONG_PTR -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4668,6 +4727,7 @@ ULongPtrToLongLong(
 //
 // LONGLONG -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4694,6 +4754,7 @@ LongLongToInt8(
 //
 // LONGLONG -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4720,6 +4781,7 @@ LongLongToUChar(
 //
 // LONGLONG -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 LongLongToChar(
@@ -4736,6 +4798,7 @@ LongLongToChar(
 //
 // LONGLONG -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4767,6 +4830,7 @@ LongLongToUInt8(
 //
 // LONGLONG -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4798,6 +4862,7 @@ LongLongToShort(
 //
 // LONGLONG -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4834,6 +4899,7 @@ LongLongToUShort(
 //
 // LONGLONG -> INT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4866,6 +4932,7 @@ LongLongToInt(
 // LONGLONG -> INT_PTR conversion
 //
 #ifdef _WIN64
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4883,6 +4950,7 @@ LongLongToIntPtr(
 //
 // LONGLONG -> UINT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4923,6 +4991,7 @@ LongLongToUInt(
 //
 // LONGLONG -> LONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4950,6 +5019,7 @@ LongLongToLong(
 // LONGLONG -> LONG_PTR conversion
 //
 #ifdef _WIN64
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -4967,6 +5037,7 @@ LongLongToLongPtr(
 //
 // LONGLONG -> ULONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5012,6 +5083,7 @@ LongLongToULong(
 //
 // LONGLONG -> ULONGLONG conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5368,6 +5440,7 @@ LongLongToULongLong(
 //
 // ULONGLONG -> INT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5394,6 +5467,7 @@ ULongLongToInt8(
 //
 // ULONGLONG -> UCHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5420,6 +5494,7 @@ ULongLongToUChar(
 //
 // ULONGLONG -> CHAR conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 __forceinline
 HRESULT
 ULongLongToChar(
@@ -5436,6 +5511,7 @@ ULongLongToChar(
 //
 // ULONGLONG -> UINT8 conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5467,6 +5543,7 @@ ULongLongToUInt8(
 //
 // ULONGLONG -> SHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -5498,6 +5575,7 @@ ULongLongToShort(
 //
 // ULONGLONG -> USHORT conversion
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -6896,6 +6974,7 @@ ULongLongToLongLong(
 //
 // UINT8 addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -7227,6 +7306,7 @@ ULongLongAdd(
 //
 // UINT8 subtraction
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -7560,6 +7640,7 @@ ULongLongSub(
 //
 // UINT8 multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -7956,6 +8037,7 @@ __Multiply128(
 //
 // INT8 Addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -7972,6 +8054,7 @@ Int8Add(
 //
 // SHORT Addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -7993,6 +8076,7 @@ ShortAdd(
 //
 // INT Addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8017,6 +8101,7 @@ IntAdd(
 #ifdef _WIN64
 #define IntPtrAdd   LongLongAdd
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8034,6 +8119,7 @@ IntPtrAdd(
 //
 // LONG Addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8058,6 +8144,7 @@ LongAdd(
 #ifdef _WIN64
 #define LongPtrAdd   LongLongAdd
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8075,6 +8162,7 @@ LongPtrAdd(
 //
 // LONGLONG Addition
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8132,6 +8220,7 @@ LongLongAdd(
 #ifdef _WIN64
 #define PtrdiffTAdd LongLongAdd
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8152,6 +8241,7 @@ PtrdiffTAdd(
 #ifdef _WIN64
 #define SSIZETAdd   LongLongAdd
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8174,6 +8264,7 @@ SSIZETAdd(
 //
 // INT8 Subtraction
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8190,6 +8281,7 @@ Int8Sub(
 //
 // SHORT Subtraction
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8211,6 +8303,7 @@ ShortSub(
 //
 // INT Subtraction
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8235,6 +8328,7 @@ IntSub(
 #ifdef _WIN64
 #define IntPtrSub  LongLongSub
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8252,6 +8346,7 @@ IntPtrSub(
 //
 // LONG Subtraction
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8276,6 +8371,7 @@ LongSub(
 #ifdef _WIN64
 #define LongPtrSub  LongLongSub
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8293,6 +8389,7 @@ LongPtrSub(
 //
 // LongLongSub
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8351,6 +8448,7 @@ LongLongSub(
 #ifdef _WIN64
 #define PtrdiffTSub LongLongSub
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8371,6 +8469,7 @@ PtrdiffTSub(
 #ifdef _WIN64
 #define SSIZETSub  LongLongSub
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8392,6 +8491,7 @@ SSIZETSub(
 //
 // INT8 multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8408,6 +8508,7 @@ Int8Mult(
 //
 // SHORT multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8429,6 +8530,7 @@ ShortMult(
 //
 // INT multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8453,6 +8555,7 @@ IntMult(
 #ifdef _WIN64
 #define IntPtrMult   LongLongMult
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8470,6 +8573,7 @@ IntPtrMult(
 //
 // LONG multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8494,6 +8598,7 @@ LongMult(
 #ifdef _WIN64
 #define LongPtrMult LongLongMult
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8511,6 +8616,7 @@ LongPtrMult(
 //
 // LONGLONG multiplication
 //
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8626,6 +8732,7 @@ LongLongMult(
 #ifdef _WIN64
 #define PtrdiffTMult    LongLongMult
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
@@ -8646,6 +8753,7 @@ PtrdiffTMult(
 #ifdef _WIN64
 #define SSIZETMult  LongLongMult
 #else
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 __inline
 HRESULT
