diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/ip2string.h b/generation/WinSDK/RecompiledIdlHeaders/shared/ip2string.h
index 700cd58e..84c6ff79 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/ip2string.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/ip2string.h
@@ -31,6 +31,9 @@ Environment:
 #define __IP2STRING__
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -50,6 +53,7 @@ struct in6_addr;
 union _DL_EUI48;
 typedef union _DL_EUI48 DL_EUI48, *PDL_EUI48;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PSTR
 NTAPI
@@ -58,6 +62,7 @@ RtlIpv4AddressToStringA (
     _Out_writes_(16) PSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PSTR
 NTAPI
@@ -66,6 +71,7 @@ RtlIpv6AddressToStringA (
     _Out_writes_(46) PSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 PSTR
 NTAPI
@@ -95,6 +101,7 @@ RtlIpv6AddressToStringExA(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -103,6 +110,7 @@ RtlIpv4AddressToStringW (
     _Out_writes_(16) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -111,6 +119,7 @@ RtlIpv6AddressToStringW (
     _Out_writes_(46) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -119,6 +128,7 @@ RtlEthernetAddressToStringW (
     _Out_writes_(18) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 NTSTATUS
 NTAPI
@@ -129,6 +139,7 @@ RtlIpv4AddressToStringExW(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 NTSTATUS
 NTAPI
@@ -140,6 +151,7 @@ RtlIpv6AddressToStringExW(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -151,6 +163,7 @@ RtlIpv4StringToAddressA (
     _Out_ struct in_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -161,6 +174,7 @@ RtlIpv6StringToAddressA (
     _Out_ struct in6_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -193,6 +207,7 @@ RtlIpv6StringToAddressExA (
     _Out_ PUSHORT Port
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -204,6 +219,7 @@ RtlIpv4StringToAddressW (
     _Out_ struct in_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -214,6 +230,7 @@ RtlIpv6StringToAddressW (
     _Out_ struct in6_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -224,6 +241,7 @@ RtlEthernetStringToAddressW (
     _Out_ DL_EUI48 *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
@@ -235,6 +253,7 @@ RtlIpv4StringToAddressExW (
     _Out_ PUSHORT Port
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Must_inspect_result_
 NTSYSAPI
 NTSTATUS
