diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/iphlpapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/iphlpapi.h
index 52c0937e..af547876 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/iphlpapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/iphlpapi.h
@@ -19,6 +19,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 //////////////////////////////////////////////////////////////////////////////
 //                                                                          //
@@ -78,6 +81,7 @@ extern "C" {
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -93,6 +97,7 @@ GetNumberOfInterfaces(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -106,6 +111,7 @@ GetIfEntry(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -121,6 +127,7 @@ GetIfTable(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -136,6 +143,7 @@ GetIpAddrTable(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -151,6 +159,7 @@ GetIpNetTable(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -172,6 +181,7 @@ GetIpForwardTable(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -181,6 +191,7 @@ GetTcpTable(
     _In_                             BOOL          Order
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -211,6 +222,7 @@ GetExtendedTcpTable(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -227,6 +239,7 @@ GetOwnerModuleFromTcpEntry(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -236,6 +249,7 @@ GetUdpTable(
     _In_                             BOOL          Order
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -254,6 +268,7 @@ GetExtendedUdpTable(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -265,6 +280,7 @@ GetOwnerModuleFromUdpEntry(
     );
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -286,6 +302,7 @@ GetTcpTable2(
 // Deprecated APIs, Added for documentation.
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD
 AllocateAndGetTcpExTableFromStack(
     _Outptr_ PVOID         *ppTcpTable,
@@ -295,6 +312,7 @@ AllocateAndGetTcpExTableFromStack(
     _In_        DWORD         dwFamily
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD
 AllocateAndGetUdpExTableFromStack(
     _Outptr_ PVOID         *ppUdpTable,
@@ -318,6 +336,7 @@ AllocateAndGetUdpExTableFromStack(
 //
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -327,6 +346,7 @@ GetTcp6Table(
     _In_                         BOOL           Order
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -340,6 +360,7 @@ GetTcp6Table2(
 
 #ifdef WINAPI
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -357,6 +378,7 @@ GetPerTcpConnectionEStats(
     _In_ ULONG RodSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -371,6 +393,7 @@ SetPerTcpConnectionEStats(
 
 #ifdef _WS2IPDEF_
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -388,6 +411,7 @@ GetPerTcp6ConnectionEStats(
     _In_  ULONG RodSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -404,6 +428,7 @@ SetPerTcp6ConnectionEStats(
 
 #endif // WINAPI
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -414,6 +439,7 @@ GetOwnerModuleFromTcp6Entry(
     _Inout_                   PDWORD                        pdwSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -423,6 +449,7 @@ GetUdp6Table(
     _In_                         BOOL           Order
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -455,6 +482,7 @@ GetOwnerModuleFromPidAndInfo(
 //////////////////////////////////////////////////////////////////////////////
 
 #if (NTDDI_VERSION >= NTDDI_WIN2K)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -470,6 +498,7 @@ GetIpStatistics(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if (NTDDI_VERSION >= NTDDI_WIN2K)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -485,6 +514,7 @@ GetIcmpStatistics(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if (NTDDI_VERSION >= NTDDI_WIN2K)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -492,6 +522,7 @@ GetTcpStatistics(
     _Out_ PMIB_TCPSTATS Statistics
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -501,6 +532,7 @@ GetUdpStatistics(
 #endif
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -517,6 +549,7 @@ SetIpStatisticsEx(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -525,6 +558,7 @@ GetIpStatisticsEx(
     _In_  ULONG Family
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -533,6 +567,7 @@ GetIcmpStatisticsEx(
     _In_  ULONG Family
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -541,6 +576,7 @@ GetTcpStatisticsEx(
     _In_  ULONG Family
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -551,6 +587,7 @@ GetUdpStatisticsEx(
 #endif
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS3)
+_Win32_metadata_supported_os_(windows10.0.16299)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -559,6 +596,7 @@ GetTcpStatisticsEx2(
     _In_  ULONG Family
     );
 
+_Win32_metadata_supported_os_(windows10.0.16299)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -583,6 +621,7 @@ GetUdpStatisticsEx2(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -600,6 +639,7 @@ SetIfEntry(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -607,6 +647,7 @@ CreateIpForwardEntry(
     _In_ PMIB_IPFORWARDROW pRoute
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -614,6 +655,7 @@ SetIpForwardEntry(
     _In_ PMIB_IPFORWARDROW pRoute
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -632,6 +674,7 @@ DeleteIpForwardEntry(
 //////////////////////////////////////////////////////////////////////////////
 
 #if (NTDDI_VERSION >= NTDDI_WIN2K)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -646,6 +689,7 @@ SetIpStatistics(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -661,6 +705,7 @@ SetIpTTL(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -668,6 +713,7 @@ CreateIpNetEntry(
     _In_ PMIB_IPNETROW    pArpEntry
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -675,6 +721,7 @@ SetIpNetEntry(
     _In_ PMIB_IPNETROW    pArpEntry
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -682,6 +729,7 @@ DeleteIpNetEntry(
     _In_ PMIB_IPNETROW    pArpEntry
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -698,6 +746,7 @@ FlushIpNetTable(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -707,6 +756,7 @@ CreateProxyArpEntry(
     _In_  DWORD   dwIfIndex
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -724,6 +774,7 @@ DeleteProxyArpEntry(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -731,6 +782,7 @@ SetTcpEntry(
     _In_ PMIB_TCPROW pTcpRow
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -751,6 +803,7 @@ GetUniDirectionalAdapterInfo(
 #ifndef NHPALLOCATEANDGETINTERFACEINFOFROMSTACK_DEFINED
 #define NHPALLOCATEANDGETINTERFACEINFOFROMSTACK_DEFINED
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -771,6 +824,7 @@ NhpAllocateAndGetInterfaceInfoFromStack(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -787,6 +841,7 @@ GetBestInterface(
 
 #pragma warning(push)
 #pragma warning(disable:4115)
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -812,6 +867,7 @@ GetBestInterfaceEx(
 //                                                                          //
 //////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -821,6 +877,7 @@ GetBestRoute(
     _Out_     PMIB_IPFORWARDROW   pBestRoute
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -829,6 +886,7 @@ NotifyAddrChange(
     _In_  LPOVERLAPPED overlapped
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -837,6 +895,7 @@ NotifyRouteChange(
     _In_  LPOVERLAPPED overlapped
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 BOOL
 WINAPI
@@ -844,6 +903,7 @@ CancelIPChangeNotify(
     _In_  LPOVERLAPPED notifyOverlapped
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -852,6 +912,7 @@ GetAdapterIndex(
     _Inout_ PULONG IfIndex
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -863,6 +924,7 @@ AddIPAddress(
     _Out_ PULONG  NTEInstance
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -877,6 +939,7 @@ DeleteIPAddress(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if (NTDDI_VERSION >= NTDDI_WIN2KSP1)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -892,6 +955,7 @@ GetNetworkParams(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -900,6 +964,7 @@ GetAdaptersInfo(
     _Inout_                         PULONG           SizePointer
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 PIP_ADAPTER_ORDER_MAP
 WINAPI
@@ -920,6 +985,7 @@ GetAdapterOrderMap(
 // The following functions require Winsock2.
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -934,6 +1000,7 @@ GetAdaptersAddresses(
 #endif
 
 #if (NTDDI_VERSION >= NTDDI_WIN2KSP1)
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1178,6 +1245,7 @@ CancelIfTimestampConfigChange(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1185,6 +1253,7 @@ IpReleaseAddress(
     _In_ PIP_ADAPTER_INDEX_MAP  AdapterInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1192,6 +1261,7 @@ IpRenewAddress(
     _In_ PIP_ADAPTER_INDEX_MAP  AdapterInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1202,6 +1272,8 @@ SendARP(
     _Inout_ PULONG  PhyAddrLen
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 IPHLPAPI_DLL_LINKAGE
 BOOL
 WINAPI
@@ -1212,6 +1284,7 @@ GetRTTAndHopCount(
     _Out_ PULONG RTT
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1219,6 +1292,7 @@ GetFriendlyIfIndex(
     _In_ DWORD IfIndex
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1227,6 +1301,7 @@ EnableRouter(
     _Out_ OVERLAPPED* pOverlapped
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1235,6 +1310,7 @@ UnenableRouter(
     _Out_opt_ LPDWORD lpdwEnableCount
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1243,6 +1319,7 @@ DisableMediaSense(
     _In_  OVERLAPPED *pOverLapped
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1253,6 +1330,7 @@ RestoreMediaSense(
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
@@ -1264,6 +1342,7 @@ GetIpErrorString(
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
 #ifdef _WS2DEF_
+_Win32_metadata_supported_os_(windows5.1.2600)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1279,6 +1358,7 @@ ResolveNeighbor(
 // Port reservation API routines.
 //
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1288,6 +1368,7 @@ CreatePersistentTcpPortReservation(
     _Out_ PULONG64 Token
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1297,6 +1378,7 @@ CreatePersistentUdpPortReservation(
     _Out_ PULONG64 Token
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1305,6 +1387,7 @@ DeletePersistentTcpPortReservation(
     _In_ USHORT NumberOfPorts
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1313,6 +1396,7 @@ DeletePersistentUdpPortReservation(
     _In_ USHORT NumberOfPorts
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1322,6 +1406,7 @@ LookupPersistentTcpPortReservation(
     _Out_ PULONG64 Token
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 ULONG
 WINAPI
@@ -1427,6 +1512,7 @@ typedef struct NET_ADDRESS_INFO_
 
 } NET_ADDRESS_INFO, *PNET_ADDRESS_INFO;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 DWORD
 WINAPI
