diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/jobapi2.h b/generation/WinSDK/RecompiledIdlHeaders/um/jobapi2.h
index 984429f9..0c89a283 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/jobapi2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/jobapi2.h
@@ -12,6 +12,9 @@
 
 #include <apiset.h>
 #include <apisetcconv.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #include <minwindef.h>
 #include <minwinbase.h>
 
@@ -32,6 +35,7 @@ typedef struct JOBOBJECT_IO_RATE_CONTROL_INFORMATION {
     ULONG ControlFlags;
 } JOBOBJECT_IO_RATE_CONTROL_INFORMATION;
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -47,6 +51,7 @@ FreeMemoryJobObject(
     _In_ _Frees_ptr_ VOID* Buffer
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -56,6 +61,7 @@ OpenJobObjectW(
     _In_ LPCWSTR lpName
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -64,6 +70,7 @@ AssignProcessToJobObject(
     _In_ HANDLE hProcess
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -72,6 +79,7 @@ TerminateJobObject(
     _In_ UINT uExitCode
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -82,6 +90,7 @@ SetInformationJobObject(
     _In_ DWORD cbJobObjectInformationLength
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -90,6 +99,7 @@ SetIoRateControlInformationJobObject(
     _In_ JOBOBJECT_IO_RATE_CONTROL_INFORMATION* IoRateControlInfo
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -101,6 +111,7 @@ QueryInformationJobObject(
     _Out_opt_ LPDWORD lpReturnLength
     );
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
