diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
index 105eee55..f6744ed0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
@@ -12,6 +12,9 @@
 
 #ifndef _APISETLIBLOADER2_
 #define _APISETLIBLOADER2_
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -27,6 +30,7 @@ extern "C" {
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -46,6 +50,8 @@ LoadPackagedLibrary(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
