diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi.h
index d4b98341..d5632cc0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi.h
@@ -18,6 +18,7 @@
 #include <minwindef.h>
 #include <minwinbase.h>
 
+#include <win32metadata_annotations.h>
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -127,6 +128,8 @@ typedef ENUMRESLANGPROCA ENUMRESLANGPROC;
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -151,6 +154,8 @@ FindResourceExW(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -165,6 +170,8 @@ FindStringOrdinal(
 
 #endif // (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -172,6 +179,7 @@ FreeLibrary(
     _In_ HMODULE hLibModule
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 DECLSPEC_NORETURN
 VOID
@@ -187,6 +195,7 @@ FreeLibraryAndExitThread(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 BOOL
 WINAPI
@@ -200,6 +209,8 @@ FreeResource(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0)
 _Ret_range_(1,nSize)
@@ -211,6 +222,8 @@ GetModuleFileNameA(
     _In_ DWORD nSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0)
 _Ret_range_(1,nSize)
@@ -228,6 +241,8 @@ GetModuleFileNameW(
 #define GetModuleFileName  GetModuleFileNameA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _When_(lpModuleName == NULL,_Ret_notnull_)
 _When_(lpModuleName != NULL,_Ret_maybenull_)
@@ -237,6 +252,8 @@ GetModuleHandleA(
     _In_opt_ LPCSTR lpModuleName
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _When_(lpModuleName == NULL,_Ret_notnull_)
 _When_(lpModuleName != NULL,_Ret_maybenull_)
@@ -280,6 +297,8 @@ PGET_MODULE_HANDLE_EXW)(
 #define PGET_MODULE_HANDLE_EX  PGET_MODULE_HANDLE_EXA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -289,6 +308,8 @@ GetModuleHandleExA(
     _Out_ HMODULE* phModule
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -306,6 +327,8 @@ GetModuleHandleExW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 FARPROC
 WINAPI
@@ -332,6 +355,8 @@ typedef struct _REDIRECTION_DESCRIPTOR {
 
 typedef const REDIRECTION_DESCRIPTOR *PCREDIRECTION_DESCRIPTOR;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -342,6 +367,8 @@ LoadLibraryExA(
     _In_ DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -395,6 +422,8 @@ LoadLibraryExW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HGLOBAL
@@ -404,6 +433,8 @@ LoadResource(
     _In_ HRSRC hResInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -414,6 +445,8 @@ LoadStringA(
     _In_ int cchBufferMax
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -430,6 +463,7 @@ LoadStringW(
 #define LoadString  LoadStringA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 LPVOID
 WINAPI
@@ -437,6 +471,8 @@ LockResource(
     _In_ HGLOBAL hResData
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -453,6 +489,8 @@ SizeofResource(
 
 typedef PVOID DLL_DIRECTORY_COOKIE, *PDLL_DIRECTORY_COOKIE;
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DLL_DIRECTORY_COOKIE
 WINAPI
@@ -460,6 +498,8 @@ AddDllDirectory(
     _In_ PCWSTR NewDirectory
     );
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -467,6 +507,8 @@ RemoveDllDirectory(
     _In_ DLL_DIRECTORY_COOKIE Cookie
     );
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -482,6 +524,8 @@ SetDefaultDllDirectories(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 APIENTRY
@@ -495,6 +539,8 @@ EnumResourceLanguagesExA(
     LANGID LangId
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 APIENTRY
@@ -514,6 +560,8 @@ EnumResourceLanguagesExW(
 #define EnumResourceLanguagesEx  EnumResourceLanguagesExA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -526,6 +574,8 @@ EnumResourceNamesExA(
     LANGID LangId
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -544,6 +594,8 @@ EnumResourceNamesExW(
 #define EnumResourceNamesEx  EnumResourceNamesExA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -555,6 +607,8 @@ EnumResourceTypesExA(
     LANGID LangId
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -594,6 +648,8 @@ FindResourceW(
 #define FindResource  FindResourceW
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -602,6 +658,8 @@ LoadLibraryA(
     _In_ LPCSTR lpLibFileName
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -632,6 +690,8 @@ EnumResourceNamesW(
     _In_ LONG_PTR lParam
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
