diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMaccess.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMaccess.h
index ae8bb839..79945c28 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMaccess.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMaccess.h
@@ -23,6 +23,9 @@ Notes:
 --*/
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -49,6 +52,7 @@ extern "C" {
 // Function Prototypes - User
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserAdd (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -61,6 +65,7 @@ NetUserAdd (
     _Out_opt_ LPDWORD    parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserEnum (
     _In_opt_    LPCWSTR     servername OPTIONAL,
@@ -73,6 +78,7 @@ NetUserEnum (
     _Inout_opt_ PDWORD resume_handle OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NET_API_STATUS NET_API_FUNCTION
 NetUserGetInfo (
     _In_opt_  LPCWSTR     servername OPTIONAL,
@@ -81,6 +87,7 @@ NetUserGetInfo (
     _Outptr_result_buffer_(_Inexpressible_("varies")) LPBYTE     *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserSetInfo (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -90,12 +97,14 @@ NetUserSetInfo (
     _Out_opt_ LPDWORD   parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserDel (
     _In_opt_  LPCWSTR    servername OPTIONAL,
     _In_      LPCWSTR    username
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserGetGroups (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -107,6 +116,7 @@ NetUserGetGroups (
     _Out_     LPDWORD   totalentries
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserSetGroups (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -116,6 +126,7 @@ NetUserSetGroups (
     _In_      DWORD     num_entries
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserGetLocalGroups (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -128,6 +139,7 @@ NetUserGetLocalGroups (
     _Out_     LPDWORD   totalentries
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserModalsGet (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -135,6 +147,7 @@ NetUserModalsGet (
     _Outptr_result_buffer_(_Inexpressible_("varies")) LPBYTE    *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetUserModalsSet (
     _In_opt_  LPCWSTR    servername OPTIONAL,
@@ -143,6 +156,8 @@ NetUserModalsSet (
     _Out_opt_ LPDWORD   parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 NET_API_STATUS NET_API_FUNCTION
 NetUserChangePassword (
     _In_opt_ IN  LPCWSTR   domainname OPTIONAL,
@@ -823,6 +838,7 @@ typedef struct _MSA_INFO_1 {
 // Function Prototypes
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupAdd (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -831,6 +847,7 @@ NetGroupAdd (
     _Out_opt_ LPDWORD  parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupAddUser (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -838,6 +855,7 @@ NetGroupAddUser (
     _In_      LPCWSTR   username
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupEnum (
     _In_opt_    LPCWSTR      servername OPTIONAL,
@@ -849,6 +867,7 @@ NetGroupEnum (
     _Inout_opt_ PDWORD_PTR resume_handle OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupGetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -857,6 +876,7 @@ NetGroupGetInfo (
     _Outptr_result_buffer_(_Inexpressible_("varies")) LPBYTE   *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupSetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -866,12 +886,14 @@ NetGroupSetInfo (
     _Out_opt_ LPDWORD  parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupDel (
     _In_opt_  LPCWSTR   servername OPTIONAL,
     _In_      LPCWSTR   groupname
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupDelUser (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -879,6 +901,7 @@ NetGroupDelUser (
     _In_      LPCWSTR   Username
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupGetUsers (
     _In_opt_    LPCWSTR     servername OPTIONAL,
@@ -891,6 +914,7 @@ NetGroupGetUsers (
     _Inout_opt_ PDWORD_PTR ResumeHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGroupSetUsers (
     _In_opt_  LPCWSTR     servername OPTIONAL,
@@ -999,6 +1023,7 @@ typedef struct _GROUP_USERS_INFO_1 {
 // Function Prototypes
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupAdd (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1014,6 +1039,7 @@ NetLocalGroupAddMember (
     _In_ PSID     membersid
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupEnum (
     _In_opt_    LPCWSTR      servername OPTIONAL,
@@ -1025,6 +1051,7 @@ NetLocalGroupEnum (
     _Inout_opt_ PDWORD_PTR resumehandle OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupGetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1033,6 +1060,7 @@ NetLocalGroupGetInfo (
     _Outptr_result_buffer_(_Inexpressible_("varies")) LPBYTE   *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupSetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1042,6 +1070,7 @@ NetLocalGroupSetInfo (
     _Out_opt_ LPDWORD  parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupDel (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1055,6 +1084,7 @@ NetLocalGroupDelMember (
     _In_      PSID     membersid
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupGetMembers (
     _In_opt_    LPCWSTR     servername OPTIONAL,
@@ -1067,6 +1097,7 @@ NetLocalGroupGetMembers (
     _Inout_opt_ PDWORD_PTR resumehandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupSetMembers (
     _In_opt_  LPCWSTR     servername OPTIONAL,
@@ -1076,6 +1107,7 @@ NetLocalGroupSetMembers (
     _In_      DWORD      totalentries
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupAddMembers (
     _In_opt_  LPCWSTR     servername OPTIONAL,
@@ -1085,6 +1117,7 @@ NetLocalGroupAddMembers (
     _In_      DWORD      totalentries
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetLocalGroupDelMembers (
     _In_opt_  LPCWSTR     servername OPTIONAL,
@@ -1148,6 +1181,7 @@ typedef struct _LOCALGROUP_USERS_INFO_0 {
 // Display Information APIs
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetQueryDisplayInformation(
     IN LPCWSTR ServerName OPTIONAL,
@@ -1158,6 +1192,7 @@ NetQueryDisplayInformation(
     OUT LPDWORD ReturnedEntryCount,
     OUT PVOID   *SortedBuffer );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetGetDisplayInformationIndex(
     IN LPCWSTR ServerName OPTIONAL,
@@ -1213,6 +1248,7 @@ typedef struct _NET_DISPLAY_GROUP {
 // Function Prototypes - Access
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessAdd (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1221,6 +1257,7 @@ NetAccessAdd (
     _Out_opt_ LPDWORD  parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessEnum (
     _In_opt_    LPCWSTR     servername OPTIONAL,
@@ -1234,6 +1271,7 @@ NetAccessEnum (
     _Inout_opt_ LPDWORD resume_handle OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessGetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1242,6 +1280,7 @@ NetAccessGetInfo (
     _Outptr_result_buffer_(_Inexpressible_("varies")) LPBYTE   *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessSetInfo (
     _In_opt_  LPCWSTR   servername OPTIONAL,
@@ -1251,12 +1290,14 @@ NetAccessSetInfo (
     _Out_opt_ LPDWORD  parm_err OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessDel (
     IN  LPCWSTR   servername OPTIONAL,
     IN  LPCWSTR   resource
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAccessGetUserPerms (
     IN  LPCWSTR   servername OPTIONAL,
@@ -1514,6 +1555,7 @@ typedef struct _NET_VALIDATE_PASSWORD_RESET_INPUT_ARG{
 //    Password Checking API structures end here
 //
 
+_Win32_metadata_supported_os_(windowsserver2003)
 NET_API_STATUS NET_API_FUNCTION
 NetValidatePasswordPolicy(
     IN LPCWSTR ServerName,
@@ -1523,6 +1565,7 @@ NetValidatePasswordPolicy(
     OUT LPVOID *OutputArg
     );
 
+_Win32_metadata_supported_os_(windowsserver2003)
 NET_API_STATUS NET_API_FUNCTION
 NetValidatePasswordPolicyFree(
     IN LPVOID *OutputArg
@@ -1547,6 +1590,7 @@ NetValidatePasswordPolicyFree(
 // Function Prototypes - Domain
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return == ERROR_SUCCESS)
 NET_API_STATUS NET_API_FUNCTION
 NetGetDCName (
@@ -1556,6 +1600,7 @@ NetGetDCName (
         LPBYTE  *Buffer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return == ERROR_SUCCESS)
 NET_API_STATUS NET_API_FUNCTION
 NetGetAnyDCName (
@@ -1711,6 +1756,7 @@ DEFINE_GUID( //262e99c9-6160-4871-acec-4e61736b6f21
     0x4871,
     0xAC, 0xEC, 0x4E, 0x61, 0x73, 0x6B, 0x6F, 0x21 );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NetAddServiceAccount(
     _In_opt_ LPWSTR ServerName,
@@ -1736,12 +1782,14 @@ NetAddServiceAccount(
 // attempt to unlink or delete the account
 #define SERVICE_ACCOUNT_FLAG_REMOVE_OFFLINE       0x00000002L
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NetRemoveServiceAccount(
     _In_opt_ LPWSTR ServerName,
     _In_ LPWSTR AccountName,
     _In_ DWORD Flags);
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NetEnumerateServiceAccounts(
     _In_opt_ LPWSTR ServerName,
@@ -1749,6 +1797,7 @@ NetEnumerateServiceAccounts(
     _Out_ DWORD* AccountsCount,
     _Outptr_result_buffer_(*AccountsCount) PZPWSTR* Accounts);
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NetIsServiceAccount(
     _In_opt_ LPWSTR ServerName,
@@ -1762,6 +1811,7 @@ NetIsServiceAccount2(
     _Out_ BOOL * IsService,
     _Out_ MSA_INFO_ACCOUNT_TYPE * AccountType);
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSTATUS
 NetQueryServiceAccount(
     _In_opt_ LPWSTR ServerName,
