diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMalert.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMalert.h
index 1237911f..02324ee9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMalert.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMalert.h
@@ -33,6 +33,7 @@ Notes:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -46,6 +47,7 @@ extern "C" {
 // Function Prototypes
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAlertRaise(
     _In_ LPCWSTR AlertType,
@@ -53,6 +55,7 @@ NetAlertRaise(
     _In_ DWORD   BufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetAlertRaiseEx(
     _In_ LPCWSTR AlertType,
