diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
index 2e59e296..6b9ae6ed 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
@@ -33,6 +33,7 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -108,6 +109,7 @@ typedef struct _AT_ENUM {
     LPWSTR  Command;
 } AT_ENUM, *PAT_ENUM, *LPAT_ENUM;
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobAdd(
     IN      LPCWSTR         Servername  OPTIONAL,
@@ -115,6 +117,7 @@ NetScheduleJobAdd(
     OUT     LPDWORD         JobId
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobDel(
     IN      LPCWSTR         Servername  OPTIONAL,
@@ -122,6 +125,7 @@ NetScheduleJobDel(
     IN      DWORD           MaxJobId
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobEnum(
     IN      LPCWSTR         Servername              OPTIONAL,
@@ -132,6 +136,7 @@ NetScheduleJobEnum(
     IN OUT  LPDWORD         ResumeHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobGetInfo(
     IN      LPCWSTR         Servername              OPTIONAL,
