diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMJoin.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMJoin.h
index 0bb4c7b5..1f4ebf9a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMJoin.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMJoin.h
@@ -28,6 +28,9 @@ Notes:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -173,6 +176,7 @@ typedef struct _DSREG_JOIN_INFO
 //
 // Joins a machine to the domain.
 //
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetJoinDomain(
@@ -184,6 +188,7 @@ NetJoinDomain(
     _In_ DWORD   fJoinOptions
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetUnjoinDomain(
@@ -193,6 +198,7 @@ NetUnjoinDomain(
     _In_ DWORD   fUnjoinOptions
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetRenameMachineInDomain(
@@ -206,6 +212,7 @@ NetRenameMachineInDomain(
 //
 // Determine the validity of a name
 //
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetValidateName(
@@ -219,6 +226,7 @@ NetValidateName(
 //
 // Determines the list of OUs that the client can create a machine account in
 //
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetGetJoinableOUs(
@@ -238,6 +246,7 @@ NetGetJoinableOUs(
 
 #define NET_IGNORE_UNSUPPORTED_FLAGS  0x01
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NET_API_STATUS
 NET_API_FUNCTION
 NetAddAlternateComputerName(
@@ -248,6 +257,7 @@ NetAddAlternateComputerName(
     _In_ ULONG Reserved
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NET_API_STATUS
 NET_API_FUNCTION
 NetRemoveAlternateComputerName(
@@ -258,6 +268,7 @@ NetRemoveAlternateComputerName(
     _In_ ULONG Reserved
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NET_API_STATUS
 NET_API_FUNCTION
 NetSetPrimaryComputerName(
@@ -281,6 +292,7 @@ typedef enum _NET_COMPUTER_NAME_TYPE {
     NetComputerNameTypeMax
 } NET_COMPUTER_NAME_TYPE, *PNET_COMPUTER_NAME_TYPE;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NET_API_STATUS
 NET_API_FUNCTION
 NetEnumerateComputerNames(
@@ -344,6 +356,7 @@ NetEnumerateComputerNames(
 //
 #define NETSETUP_PROVISION_CHECK_PWD_ONLY         0x80000000
 
+_Win32_metadata_supported_os_(windows6.1)
 NET_API_STATUS
 NET_API_FUNCTION
 NetProvisionComputerAccount(
@@ -359,6 +372,7 @@ NetProvisionComputerAccount(
                    LPWSTR *pProvisionTextData
 );
 
+_Win32_metadata_supported_os_(windows6.1)
 NET_API_STATUS
 NET_API_FUNCTION
 NetRequestOfflineDomainJoin(
@@ -398,6 +412,7 @@ typedef struct _NETSETUP_PROVISIONING_PARAMS
     LPWSTR  lpPrimaryDNSDomain;     //  _In_
 } NETSETUP_PROVISIONING_PARAMS, *PNETSETUP_PROVISIONING_PARAMS;
 
+_Win32_metadata_supported_os_(windows8.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetCreateProvisioningPackage(
@@ -408,6 +423,7 @@ NetCreateProvisioningPackage(
     _Outptr_opt_result_maybenull_z_ LPWSTR *ppPackageTextData
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetRequestProvisioningPackageInstall(
@@ -428,6 +444,7 @@ NetRequestProvisioningPackageInstall(
 
 #if(_WIN32_WINNT >= _WIN32_WINNT_WIN10)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT
 NET_API_FUNCTION
 NetGetAadJoinInformation(
@@ -435,6 +452,7 @@ NetGetAadJoinInformation(
     _Outptr_result_maybenull_ PDSREG_JOIN_INFO *ppJoinInfo
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 VOID
 NET_API_FUNCTION
 NetFreeAadJoinInformation(
@@ -464,6 +482,7 @@ typedef enum _NETSETUP_JOIN_STATUS {
 //
 // Determines whether a workstation is joined to a domain or not
 //
+_Win32_metadata_supported_os_(windows5.0)
 NET_API_STATUS
 NET_API_FUNCTION
 NetGetJoinInformation(
