diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMShare.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMShare.h
index 6347b27d..2ec4bba8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMShare.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMShare.h
@@ -26,6 +26,9 @@ Notes:
 --*/
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -51,6 +54,7 @@ extern "C" {
 // Function Prototypes - Share
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetShareAdd(
@@ -63,6 +67,7 @@ NetShareAdd(
     _Out_opt_ LPDWORD parm_err
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0 || return == ERROR_MORE_DATA )
 NET_API_STATUS NET_API_FUNCTION
@@ -99,6 +104,7 @@ NetShareEnumSticky(
     _Inout_opt_ LPDWORD resume_handle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0 || return == ERROR_MORE_DATA )
 NET_API_STATUS NET_API_FUNCTION
@@ -116,6 +122,7 @@ NetShareGetInfo(
     LPBYTE *bufptr
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetShareSetInfo(
@@ -134,6 +141,7 @@ NetShareSetInfo(
     _Out_opt_ LPDWORD parm_err
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetShareDel(
@@ -150,6 +158,7 @@ NetShareDelSticky(
     _Reserved_ DWORD reserved
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetShareCheck(
@@ -158,6 +167,7 @@ NetShareCheck(
     _Out_ LPDWORD type
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetShareDelEx(
@@ -412,6 +422,7 @@ typedef struct _SERVER_ALIAS_INFO_0 {
 //
 // Function Prototypes Session
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0 || return == ERROR_MORE_DATA )
 NET_API_STATUS NET_API_FUNCTION
@@ -432,6 +443,7 @@ NetSessionEnum(
     _Inout_opt_ LPDWORD resume_handle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetSessionDel(
@@ -440,6 +452,7 @@ NetSessionDel(
     _In_opt_ LMSTR username
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0 )
 NET_API_STATUS NET_API_FUNCTION
@@ -528,6 +541,7 @@ typedef struct _SESSION_INFO_502 {
 // Function Prototypes - CONNECTION
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0  || return == ERROR_MORE_DATA )
 NET_API_STATUS NET_API_FUNCTION
@@ -575,6 +589,7 @@ typedef struct _CONNECTION_INFO_1 {
 // Function Prototypes - FILE
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 NET_API_STATUS NET_API_FUNCTION
 NetFileClose(
@@ -582,6 +597,7 @@ NetFileClose(
     _In_ DWORD fileid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0 || return == ERROR_MORE_DATA )
 NET_API_STATUS NET_API_FUNCTION
@@ -599,6 +615,7 @@ NetFileEnum(
     _Inout_opt_ PDWORD_PTR resume_handle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Check_return_
 _Success_( return == 0)
 NET_API_STATUS NET_API_FUNCTION
