diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/metahost.h b/generation/WinSDK/RecompiledIdlHeaders/um/metahost.h
index 10515e83..e8d05be3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/metahost.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/metahost.h
@@ -174,7 +174,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_import_library_("MSCorEE.dll")
 STDAPI CLRCreateInstance(REFCLSID clsid, REFIID riid, /*iid_is(riid)*/ LPVOID *ppInterface);
 EXTERN_GUID(CLSID_CLRStrongName, 0xB79B0ACD, 0xF5CD, 0x409b, 0xB5, 0xA5, 0xA1, 0x62, 0x44, 0x61, 0x0B, 0x92);
 EXTERN_GUID(IID_ICLRMetaHost, 0xD332DB9E, 0xB9B3, 0x4125, 0x82, 0x07, 0xA1, 0x48, 0x84, 0xF5, 0x32, 0x16);
