diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mfapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/mfapi.h
index bf17436e..c4ba13d3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mfapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mfapi.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 //*@@@+++@@@@******************************************************************
 //
@@ -67,6 +68,7 @@ extern "C" {
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFStartup( ULONG Version, DWORD dwFlags = MFSTARTUP_FULL );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
@@ -77,6 +79,7 @@ STDAPI MFStartup( ULONG Version, DWORD dwFlags = MFSTARTUP_FULL );
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFStartup( ULONG Version, DWORD dwFlags );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
@@ -93,6 +96,7 @@ STDAPI MFStartup( ULONG Version, DWORD dwFlags );
 // Application should call MFShutdown the same number of times as MFStartup
 // Application should not call MFStartup / MFShutdown from workqueue threads
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFShutdown();
 
 
@@ -104,7 +108,9 @@ STDAPI MFShutdown();
 // These functions can be used to keep the MF platform object in place.
 // Every call to MFLockPlatform should have a matching call to MFUnlockPlatform
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFLockPlatform();
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFUnlockPlatform();
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -120,6 +126,7 @@ typedef unsigned __int64 MFWORKITEM_KEY;
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFPutWorkItem(
             DWORD dwQueue,
             IMFAsyncCallback * pCallback,
@@ -131,6 +138,7 @@ STDAPI MFPutWorkItem(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFPutWorkItem2(
             DWORD dwQueue,
             LONG Priority,
@@ -143,6 +151,7 @@ STDAPI MFPutWorkItem2(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFPutWorkItemEx(
             DWORD dwQueue,
             IMFAsyncResult * pResult);
@@ -153,11 +162,13 @@ STDAPI MFPutWorkItemEx(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFPutWorkItemEx2(
             DWORD dwQueue,
             LONG Priority,
             _In_ IMFAsyncResult * pResult);
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFPutWaitingWorkItem (
             HANDLE hEvent,
             LONG Priority,
@@ -165,10 +176,12 @@ STDAPI MFPutWaitingWorkItem (
             _Out_opt_ MFWORKITEM_KEY * pKey
             );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFAllocateSerialWorkQueue (
             _In_ DWORD dwWorkQueue,
             _Out_ OUT DWORD * pdwWorkQueue);
             
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFScheduleWorkItemEx(
             IMFAsyncResult * pResult,
             INT64 Timeout,
@@ -180,6 +193,7 @@ STDAPI MFScheduleWorkItemEx(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFScheduleWorkItem(
             IMFAsyncCallback * pCallback,
             IUnknown * pState,
@@ -197,6 +211,7 @@ STDAPI MFScheduleWorkItem(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCancelWorkItem(
             MFWORKITEM_KEY Key);
 
@@ -211,16 +226,19 @@ STDAPI MFCancelWorkItem(
 //
 // MF periodic callbacks
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetTimerPeriodicity(
             _Out_ DWORD * Periodicity);
 
 typedef void (*MFPERIODICCALLBACK)(IUnknown* pContext);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFAddPeriodicCallback(
             MFPERIODICCALLBACK Callback,
             IUnknown * pContext,
             _Out_opt_ DWORD * pdwKey);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFRemovePeriodicCallback(
             DWORD dwKey);
 
@@ -249,6 +267,7 @@ typedef enum
     MF_MULTITHREADED_WORKQUEUE = 2, // common MT threadpool
 }   MFASYNC_WORKQUEUE_TYPE;
 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFAllocateWorkQueueEx(
             _In_ MFASYNC_WORKQUEUE_TYPE WorkQueueType,
             _Out_ OUT DWORD * pdwWorkQueue);
@@ -258,6 +277,7 @@ STDAPI MFAllocateWorkQueueEx(
 // Allocate a standard work queue. the behaviour is the same with:
 // MFAllocateWorkQueueEx( MF_STANDARD_WORKQUEUE, pdwWorkQueue )
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFAllocateWorkQueue(
             _Out_ OUT DWORD * pdwWorkQueue);
 
@@ -268,9 +288,11 @@ STDAPI MFAllocateWorkQueue(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFLockWorkQueue(
             _In_ DWORD dwWorkQueue);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFUnlockWorkQueue(
             _In_ DWORD dwWorkQueue);
 
@@ -280,6 +302,7 @@ STDAPI MFUnlockWorkQueue(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFBeginRegisterWorkQueueWithMMCSS(
             DWORD dwWorkQueueId,
             _In_ LPCWSTR wszClass,
@@ -287,6 +310,7 @@ STDAPI MFBeginRegisterWorkQueueWithMMCSS(
             _In_ IMFAsyncCallback * pDoneCallback,
             _In_ IUnknown * pDoneState );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFBeginRegisterWorkQueueWithMMCSSEx(
             DWORD dwWorkQueueId,
             _In_ LPCWSTR wszClass,
@@ -295,32 +319,39 @@ STDAPI MFBeginRegisterWorkQueueWithMMCSSEx(
             _In_ IMFAsyncCallback * pDoneCallback,
             _In_ IUnknown * pDoneState );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFEndRegisterWorkQueueWithMMCSS(
             _In_ IMFAsyncResult * pResult,
             _Out_ DWORD * pdwTaskId );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFBeginUnregisterWorkQueueWithMMCSS(
             DWORD dwWorkQueueId,
             _In_ IMFAsyncCallback * pDoneCallback,
             _In_ IUnknown * pDoneState );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFEndUnregisterWorkQueueWithMMCSS(
             _In_ IMFAsyncResult * pResult );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetWorkQueueMMCSSClass(
             DWORD dwWorkQueueId,
             _Out_writes_to_opt_(*pcchClass,*pcchClass)  LPWSTR pwszClass,
             _Inout_  DWORD *pcchClass );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetWorkQueueMMCSSTaskId(
             DWORD dwWorkQueueId,
             _Out_ LPDWORD pdwTaskId );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFRegisterPlatformWithMMCSS(
     _In_ PCWSTR wszClass,
     _Inout_ DWORD* pdwTaskId,
     _In_ LONG lPriority );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFUnregisterPlatformFromMMCSS();
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES) */
@@ -329,6 +360,7 @@ STDAPI MFUnregisterPlatformFromMMCSS();
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFLockSharedWorkQueue(
     _In_ PCWSTR wszClass,
     _In_ LONG BasePriority,
@@ -341,6 +373,7 @@ STDAPI MFLockSharedWorkQueue(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFGetWorkQueueMMCSSPriority(
             DWORD dwWorkQueueId,
             _Out_ LONG* lPriority );
@@ -359,6 +392,7 @@ STDAPI MFGetWorkQueueMMCSSPriority(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateAsyncResult(
     IUnknown * punkObject,
     IMFAsyncCallback * pCallback,
@@ -368,6 +402,7 @@ STDAPI MFCreateAsyncResult(
 //
 // Helper for calling IMFAsyncCallback::Invoke
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFInvokeCallback(
     IMFAsyncResult * pAsyncResult );
 
@@ -411,6 +446,7 @@ typedef struct tagMFASYNCRESULT
 // Regardless of the access mode with which the file is opened, the sharing
 // permissions will allow shared reading and deleting.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateFile(
     MF_FILE_ACCESSMODE  AccessMode,
     MF_FILE_OPENMODE    OpenMode,
@@ -418,12 +454,14 @@ STDAPI MFCreateFile(
     LPCWSTR             pwszFileURL,
     _Out_ IMFByteStream       **ppIByteStream );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateTempFile(
     MF_FILE_ACCESSMODE  AccessMode,
     MF_FILE_OPENMODE    OpenMode,
     MF_FILE_FLAGS       fFlags,
     _Out_ IMFByteStream       **ppIByteStream );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFBeginCreateFile(
     MF_FILE_ACCESSMODE  AccessMode,
     MF_FILE_OPENMODE    OpenMode,
@@ -433,10 +471,12 @@ STDAPI MFBeginCreateFile(
     IUnknown *          pState,
     _Out_ IUnknown ** ppCancelCookie);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFEndCreateFile(
     IMFAsyncResult * pResult,
     _Out_ IMFByteStream **ppFile );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCancelCreateFile(
     IUnknown * pCancelCookie);
 
@@ -454,6 +494,7 @@ STDAPI MFCancelCreateFile(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateMemoryBuffer(
     _In_ DWORD                      cbMaxLength,
     _Out_ IMFMediaBuffer **         ppBuffer );
@@ -462,6 +503,7 @@ STDAPI MFCreateMemoryBuffer(
 // Creates an IMFMediaBuffer wrapper at the given offset and length
 // within an existing IMFMediaBuffer
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateMediaBufferWrapper(
     _In_ IMFMediaBuffer *           pBuffer,
     _In_ DWORD                      cbOffset,
@@ -482,6 +524,7 @@ STDAPI MFCreateMediaBufferWrapper(
 // QueryInterface for IID_IMFSample, from which the original sample's attributes
 // can be obtained
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateLegacyMediaBufferOnMFMediaBuffer(
     _In_opt_ IMFSample *            pSample,
     _In_ IMFMediaBuffer *           pMFMediaBuffer,
@@ -492,7 +535,9 @@ STDAPI MFCreateLegacyMediaBufferOnMFMediaBuffer(
 // Create a DirectX surface buffer
 //
 #include <dxgiformat.h>
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI_(DXGI_FORMAT) MFMapDX9FormatToDXGIFormat( _In_ DWORD dx9 );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI_(DWORD) MFMapDXGIFormatToDX9Format( _In_ DXGI_FORMAT dx11 );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES) */
@@ -501,11 +546,13 @@ STDAPI_(DWORD) MFMapDXGIFormatToDX9Format( _In_ DXGI_FORMAT dx11 );
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFLockDXGIDeviceManager(
     _Out_opt_ UINT* pResetToken,
     _Outptr_ IMFDXGIDeviceManager** ppManager
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFUnlockDXGIDeviceManager();
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
 #pragma endregion
@@ -513,12 +560,14 @@ STDAPI MFUnlockDXGIDeviceManager();
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateDXSurfaceBuffer(
     _In_ REFIID                     riid,
     _In_ IUnknown *                 punkSurface,
     _In_ BOOL                       fBottomUpWhenLinear,
     _Outptr_ IMFMediaBuffer **   ppBuffer );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateWICBitmapBuffer(
     _In_ REFIID                     riid,
     _In_ IUnknown *                 punkSurface,
@@ -531,6 +580,7 @@ STDAPI MFCreateWICBitmapBuffer(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateDXGISurfaceBuffer(
     _In_ REFIID riid,
@@ -552,11 +602,13 @@ STDAPI MFCreateDXGICrossAdapterBuffer(
 
 #endif /*(NTDDI_VERSION >= NTDDI_WIN11_GE)*/
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateVideoSampleAllocatorEx(
     _In_   REFIID riid,
     _Outptr_  void** ppSampleAllocator
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateDXGIDeviceManager(
     _Out_ UINT* resetToken,
@@ -599,6 +651,7 @@ MFGetDXGIDeviceManageMode(
 #define MF_4096_BYTE_ALIGNMENT    0x00000fff
 #define MF_8192_BYTE_ALIGNMENT    0x00001fff
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateAlignedMemoryBuffer(
     _In_ DWORD                      cbMaxLength,
     _In_ DWORD                      cbAligment, 
@@ -630,6 +683,7 @@ EXTERN_C const GUID MR_BUFFER_SERVICE;
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateMediaEvent(
     _In_ MediaEventType met,
     _In_ REFGUID guidExtendedType,
@@ -646,6 +700,7 @@ STDAPI MFCreateMediaEvent(
 // EndGetEvent), and the various IMFMediaEventQueue::QueueEventXXX methods
 // can be used to queue events that the caller will consume.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateEventQueue(
     _Out_ IMFMediaEventQueue **ppMediaEventQueue );
 
@@ -894,6 +949,7 @@ DEFINE_GUID(MF_EVENT_STREAM_METADATA_SYSTEMID,
 
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateSample( _Out_ IMFSample **ppIMFSample );
 
 
@@ -1826,12 +1882,14 @@ typedef struct tagHistogramDataHeader
 ////////////////////////////////////////////////////////////////////////////////
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateAttributes(
     _Out_   IMFAttributes** ppMFAttributes,
     _In_    UINT32          cInitialSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitAttributesFromBlob(
     _In_                    IMFAttributes*  pAttributes,
@@ -1839,12 +1897,14 @@ MFInitAttributesFromBlob(
     _In_                    UINT            cbBufSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFGetAttributesAsBlobSize(
     _In_    IMFAttributes*  pAttributes,
     _Out_   UINT32*         pcbBufSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFGetAttributesAsBlob(
     _In_                    IMFAttributes*  pAttributes,
@@ -1926,6 +1986,7 @@ DEFINE_GUID(MFT_CATEGORY_VIDEO_RENDERER_EFFECT,
 //
 // "Flags" is for future expansion - for now must be 0
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFTRegister(
     _In_                            CLSID                   clsidMFT,
@@ -1939,6 +2000,7 @@ MFTRegister(
     _In_opt_                        IMFAttributes*          pAttributes
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFTUnregister(
     _In_    CLSID   clsidMFT
@@ -1946,6 +2008,7 @@ MFTUnregister(
 
 #if (WINVER >= _WIN32_WINNT_WIN7)
 //  Register an MFT class in-process
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFTRegisterLocal(
    _In_                        IClassFactory*          pClassFactory,
@@ -1960,12 +2023,14 @@ MFTRegisterLocal(
 
 //  Unregister locally registered MFT
 //  If pClassFactory is NULL all local MFTs are unregistered
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFTUnregisterLocal(
     _In_opt_    IClassFactory *   pClassFactory
     );
 
 // Register an MFT class in-process, by CLSID
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFTRegisterLocalByCLSID(
    _In_                        REFCLSID                clisdMFT,
@@ -1979,6 +2044,7 @@ MFTRegisterLocalByCLSID(
     );
 
 // Unregister locally registered MFT by CLSID
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFTUnregisterLocalByCLSID(
     _In_    CLSID   clsidMFT
@@ -1988,6 +2054,7 @@ MFTUnregisterLocalByCLSID(
 //
 // result *ppclsidMFT must be freed with CoTaskMemFree.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFTEnum(
     _In_                    GUID                    guidCategory,
@@ -2028,6 +2095,7 @@ enum _MFT_ENUM_FLAG
 // buffer should be released.
 //
 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFTEnumEx(
     _In_                                GUID                            guidCategory,
@@ -2064,6 +2132,7 @@ DEFINE_GUID(MFT_ENUM_VIDEO_RENDERER_EXTENSION_PROFILE,
 DEFINE_GUID( MFT_ENUM_ADAPTER_LUID,
     0x1d39518c, 0xe220, 0x4da8, 0xa0, 0x7f, 0xba, 0x17, 0x25, 0x52, 0xd6, 0xb1 );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI
 MFTEnum2(
     _In_                                GUID                            guidCategory,
@@ -2082,6 +2151,7 @@ MFTEnum2(
 // results *pszName, *ppInputTypes, and *ppOutputTypes must be freed with CoTaskMemFree.
 // *ppAttributes must be released.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFTGetInfo(
     _In_                                   CLSID                       clsidMFT,
@@ -2099,6 +2169,7 @@ MFTGetInfo(
 //
 //  Get the plugin control API
 //
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFGetPluginControl(
     _Out_ IMFPluginControl **ppPluginControl
@@ -2107,6 +2178,7 @@ MFGetPluginControl(
 //
 //  Get MFT's merit - checking that is has a valid certificate
 //
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFGetMFTMerit(
     _Inout_ IUnknown *pMFT,
@@ -2125,12 +2197,14 @@ MFGetMFTMerit(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFRegisterLocalSchemeHandler(
     _In_        PCWSTR          szScheme,
     _In_        IMFActivate*    pActivate
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFRegisterLocalByteStreamHandler(
     _In_        PCWSTR          szFileExtension,
@@ -2144,6 +2218,7 @@ MFRegisterLocalByteStreamHandler(
 // original bytestream can then be passed to another
 // media source for instance.
 //
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateMFByteStreamWrapper(
     _In_        IMFByteStream*  pStream,
@@ -2160,6 +2235,7 @@ MFCreateMFByteStreamWrapper(
 // Create a MF activate object that can instantiate media extension objects.
 // The activate object supports both IMFActivate and IClassFactory.
 //
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateMediaExtensionActivate(
     _In_        PCWSTR          szActivatableClassId,
@@ -2244,6 +2320,7 @@ DEFINE_GUID(MFT_SUPPORT_DYNAMIC_FORMAT_CHANGE,
 #define LOCAL_D3DFMT_DEFINES 1
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_Base,      0x00000000 );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_RGB32,     D3DFMT_X8R8G8B8 );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_ARGB32,    D3DFMT_A8R8G8B8 );
@@ -2300,6 +2377,7 @@ DEFINE_MEDIATYPE_GUID( MFVideoFormat_DV25,      FCC('dv25') );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_DV50,      FCC('dv50') );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_DVH1,      FCC('dvh1') );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_DVC,       FCC('dvc ') );
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_H264,      FCC('H264') );  // assume MFVideoFormat_H264 is frame aligned. that is, each input sample has one complete compressed frame (one frame picture, two field pictures or a single unpaired field picture)
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_H265,      FCC('H265') );  
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_MJPG,      FCC('MJPG') );
@@ -2311,19 +2389,23 @@ DEFINE_MEDIATYPE_GUID( MFVideoFormat_VP90,      FCC('VP90') );
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_ORAW,      FCC('ORAW') );
 
 #if (WINVER >= _WIN32_WINNT_WIN8)
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFVideoFormat_H263,      FCC('H263') );
 #endif // (WINVER >= _WIN32_WINNT_WIN8)
 
 #if (WDK_NTDDI_VERSION >= NTDDI_WIN10)
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_A2R10G10B10, D3DFMT_A2B10G10R10);
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_A16B16G16R16F, D3DFMT_A16B16G16R16F);
 #endif
 
 #if (WDK_NTDDI_VERSION >= NTDDI_WIN10_RS3)
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_VP10,       FCC('VP10'));
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_AV1,        FCC('AV01'));
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_Theora,     FCC('theo')); // {6F656874-0000-0010-8000-00AA00389B71}
 DEFINE_MEDIATYPE_GUID(MFVideoFormat_APV,        FCC('APV '));
 
@@ -2385,6 +2467,7 @@ DEFINE_GUID(MFVideoFormat_MPEG2,
 //
 // audio media types
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_Base,              0x00000000 );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_PCM,               WAVE_FORMAT_PCM );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_Float,             WAVE_FORMAT_IEEE_FLOAT );
@@ -2397,10 +2480,12 @@ DEFINE_MEDIATYPE_GUID( MFAudioFormat_Float,             WAVE_FORMAT_IEEE_FLOAT )
 // If DTS extension substreams may be present, use MFAudioFormat_DTS_HD (same as MEDIASUBTYPE_DTS_HD) for Master Audio,
 // and MEDIASUBTYPE_DTS_HD_HRA for High Resolution Audio and other extension substream variants.
 // (KSDATAFORMAT_SUBTYPE_IEC61937_DTS_HD is the S/PDIF media subtype for MEDIASUBTYPE_DTS_HD and MEDIASUBTYPE_DTS_HD_HRA.)
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_DTS,               WAVE_FORMAT_DTS );
 
 // MFAudioFormat_Dolby_AC3_SPDIF is for S/PDIF-encapsulated AC-3. It is the same as KSDATAFORMAT_SUBTYPE_IEC61937_DOLBY_DIGITAL in ksmedia.h.
 // Use MFAudioFormat_Dolby_AC3 (MEDIASUBTYPE_DOLBY_AC3 in wmcodecdsp.h) for raw AC-3 streams.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_Dolby_AC3_SPDIF,   WAVE_FORMAT_DOLBY_AC3_SPDIF );
 
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_DRM,               WAVE_FORMAT_DRM );
@@ -2417,10 +2502,12 @@ DEFINE_MEDIATYPE_GUID( MFAudioFormat_AMR_NB,            WAVE_FORMAT_AMR_NB );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_AMR_WB,            WAVE_FORMAT_AMR_WB );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_AMR_WP,            WAVE_FORMAT_AMR_WP );
 #if (WINVER >= _WIN32_WINNT_WINTHRESHOLD)
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_FLAC,              WAVE_FORMAT_FLAC );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_ALAC,              WAVE_FORMAT_ALAC );
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_Opus,              WAVE_FORMAT_OPUS );
 #endif
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_MEDIATYPE_GUID( MFAudioFormat_Dolby_AC4,         WAVE_FORMAT_DOLBY_AC4 );
 
 // These audio types are not derived from an existing wFormatTag 
@@ -2727,16 +2814,19 @@ DEFINE_GUID(MFSampleExtension_MULTIPLEXED_MANAGER,
 0x8dcdee79, 0x6b5a, 0x4c45, 0x8d, 0xb9, 0x20, 0xb3, 0x95, 0xf0, 0x2f, 0xcf);
 
 
+_Win32_metadata_supported_os_(windows10.0.15063)
 STDAPI MFCreateMuxStreamAttributes(
     _In_ IMFCollection *pAttributesToMux,
     _COM_Outptr_ IMFAttributes**ppMuxAttribs
 );
 
+_Win32_metadata_supported_os_(windows10.0.15063)
 STDAPI MFCreateMuxStreamMediaType(
     _In_ IMFCollection *pMediaTypesToMux,
     _COM_Outptr_ IMFMediaType**ppMuxMediaType
 );
 
+_Win32_metadata_supported_os_(windows10.0.15063)
 STDAPI MFCreateMuxStreamSample(
     _In_ IMFCollection *pSamplesToMux,
     _COM_Outptr_ IMFSample**ppMuxSample
@@ -3781,6 +3871,7 @@ typedef struct tagMPEG2VIDEOINFO MPEG2VIDEOINFO;
 struct _AMMediaType;
 typedef struct _AMMediaType AM_MEDIA_TYPE;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFValidateMediaTypeSize(
     _In_                    GUID    FormatType,
@@ -3794,6 +3885,7 @@ MFValidateMediaTypeSize(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateMediaType(
     _Outptr_ IMFMediaType**  ppMFType
@@ -3805,6 +3897,7 @@ MFCreateMediaType(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateMFVideoFormatFromMFMediaType(
     _In_        IMFMediaType*           pMFType,
@@ -3833,6 +3926,7 @@ typedef enum _MFWaveFormatExConvertFlags {
 
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateWaveFormatExFromMFMediaType(
     _In_        IMFMediaType*   pMFType,
@@ -3847,6 +3941,7 @@ MFCreateWaveFormatExFromMFMediaType(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromVideoInfoHeader(
     _In_                    IMFMediaType*           pMFType,
@@ -3855,6 +3950,7 @@ MFInitMediaTypeFromVideoInfoHeader(
     _In_opt_                const GUID*             pSubtype = NULL
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromVideoInfoHeader2(
     _In_                    IMFMediaType*           pMFType,
@@ -3863,6 +3959,7 @@ MFInitMediaTypeFromVideoInfoHeader2(
     _In_opt_                const GUID*             pSubtype = NULL
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromMPEG1VideoInfo(
     _In_                    IMFMediaType*           pMFType,
@@ -3871,6 +3968,7 @@ MFInitMediaTypeFromMPEG1VideoInfo(
     _In_opt_                const GUID*             pSubtype = NULL
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromMPEG2VideoInfo(
     _In_                    IMFMediaType*           pMFType,
@@ -3880,6 +3978,7 @@ MFInitMediaTypeFromMPEG2VideoInfo(
     );
 
 #ifndef NOBITMAP
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCalculateBitmapImageSize(
     _In_reads_bytes_(cbBufSize)  const BITMAPINFOHEADER* pBMIH,
@@ -3900,6 +3999,7 @@ MFCalculateBitmapImageSize(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateWaveFormatExFromMFMediaType(
     _In_        IMFMediaType*   pMFType,
@@ -3912,6 +4012,7 @@ MFCreateWaveFormatExFromMFMediaType(
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromVideoInfoHeader(
     _In_                    IMFMediaType*           pMFType,
@@ -3920,6 +4021,7 @@ MFInitMediaTypeFromVideoInfoHeader(
     _In_opt_                const GUID*             pSubtype
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromVideoInfoHeader2(
     _In_                    IMFMediaType*           pMFType,
@@ -3928,6 +4030,7 @@ MFInitMediaTypeFromVideoInfoHeader2(
     _In_opt_                const GUID*             pSubtype
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromMPEG1VideoInfo(
     _In_                    IMFMediaType*           pMFType,
@@ -3936,6 +4039,7 @@ MFInitMediaTypeFromMPEG1VideoInfo(
     _In_opt_                const GUID*             pSubtype
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromMPEG2VideoInfo(
     _In_                    IMFMediaType*           pMFType,
@@ -3945,6 +4049,7 @@ MFInitMediaTypeFromMPEG2VideoInfo(
     );
 
 #ifndef NOBITMAP
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCalculateBitmapImageSize(
     _In_reads_bytes_(cbBufSize)  const BITMAPINFOHEADER* pBMIH,
@@ -3961,6 +4066,7 @@ MFCalculateBitmapImageSize(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCalculateImageSize(
     _In_                    REFGUID                 guidSubtype,
@@ -3969,6 +4075,7 @@ MFCalculateImageSize(
     _Out_                   UINT32*                 pcbImageSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFFrameRateToAverageTimePerFrame(
     _In_                    UINT32                  unNumerator,
@@ -3976,6 +4083,7 @@ MFFrameRateToAverageTimePerFrame(
     _Out_                   UINT64*                 punAverageTimePerFrame
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFAverageTimePerFrameToFrameRate(
     _In_                    UINT64                  unAverageTimePerFrame,
@@ -3983,6 +4091,7 @@ MFAverageTimePerFrameToFrameRate(
     _Out_                   UINT32*                 punDenominator
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromWaveFormatEx(
     _In_                    IMFMediaType*           pMFType,
@@ -3994,6 +4103,7 @@ MFInitMediaTypeFromWaveFormatEx(
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromMFVideoFormat(
     _In_                    IMFMediaType*           pMFType,
@@ -4001,12 +4111,14 @@ MFInitMediaTypeFromMFVideoFormat(
     _In_                    UINT32                  cbBufSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitMediaTypeFromAMMediaType(
     _In_    IMFMediaType*           pMFType,
     _In_    const AM_MEDIA_TYPE*    pAMType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitAMMediaTypeFromMFMediaType(
     _In_    IMFMediaType*           pMFType,
@@ -4014,6 +4126,7 @@ MFInitAMMediaTypeFromMFMediaType(
     _Inout_ AM_MEDIA_TYPE*          pAMType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateAMMediaTypeFromMFMediaType(
     _In_    IMFMediaType*           pMFType,
@@ -4037,6 +4150,7 @@ MFCreateAMMediaTypeFromMFMediaType(
 // and all of the attributes in the partial type exist in the full type and are set to
 // the same value.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI_(BOOL)
 MFCompareFullToPartialMediaType(
     _In_    IMFMediaType*   pMFTypeFull,
@@ -4049,6 +4163,7 @@ MFCompareFullToPartialMediaType(
 
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFWrapMediaType(
     _In_    IMFMediaType*    pOrig,
@@ -4059,12 +4174,14 @@ MFWrapMediaType(
 
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFUnwrapMediaType(
     _In_    IMFMediaType*    pWrap,
     _Out_   IMFMediaType **  ppOrig
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetStrideForBitmapInfoHeader(
     DWORD format,
     DWORD dwWidth,
@@ -4082,6 +4199,7 @@ STDAPI MFGetStrideForBitmapInfoHeader(
 //
 
 #ifdef _KSMEDIA_
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateVideoMediaTypeFromVideoInfoHeader(
     _In_ const KS_VIDEOINFOHEADER* pVideoInfoHeader,
     DWORD cbVideoInfoHeader,
@@ -4093,6 +4211,7 @@ STDAPI MFCreateVideoMediaTypeFromVideoInfoHeader(
     _Out_ IMFVideoMediaType** ppIVideoMediaType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateVideoMediaTypeFromVideoInfoHeader2(
     _In_ const KS_VIDEOINFOHEADER2* pVideoInfoHeader,
     DWORD cbVideoInfoHeader,
@@ -4103,16 +4222,19 @@ STDAPI MFCreateVideoMediaTypeFromVideoInfoHeader2(
 
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateVideoMediaType(
     _In_ const MFVIDEOFORMAT* pVideoFormat,
     _Out_ IMFVideoMediaType** ppIVideoMediaType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateVideoMediaTypeFromSubtype(
     _In_ const GUID * pAMSubtype,
     _Out_ IMFVideoMediaType  **ppIVideoMediaType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI_(BOOL)
 MFIsFormatYUV(
     DWORD Format
@@ -4122,6 +4244,7 @@ MFIsFormatYUV(
 //  These depend on BITMAPINFOHEADER being defined
 //
 #ifndef NOBITMAP
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateVideoMediaTypeFromBitMapInfoHeader(
     _In_ const BITMAPINFOHEADER* pbmihBitMapInfoHeader,
     DWORD dwPixelAspectRatioX,
@@ -4135,6 +4258,7 @@ STDAPI MFCreateVideoMediaTypeFromBitMapInfoHeader(
     );
 #endif // NOBITMAP
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetPlaneSize(
     DWORD format,
     DWORD dwWidth,
@@ -4148,6 +4272,7 @@ STDAPI MFGetPlaneSize(
 // MFCreateVideoMediaTypeFromBitMapInfoHeaderEx
 //
 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateVideoMediaTypeFromBitMapInfoHeaderEx(
     _In_reads_bytes_(cbBitMapInfoHeader) const BITMAPINFOHEADER* pbmihBitMapInfoHeader,
     _In_    UINT32                  cbBitMapInfoHeader,
@@ -4167,6 +4292,7 @@ STDAPI MFCreateVideoMediaTypeFromBitMapInfoHeaderEx(
 // MFCreateMediaTypeFromRepresentation
 //
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateMediaTypeFromRepresentation(
     GUID guidRepresentation,
     _In_ LPVOID pvRepresentation,
@@ -4179,24 +4305,28 @@ STDAPI MFCreateMediaTypeFromRepresentation(
 //
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateAudioMediaType(
     _In_    const WAVEFORMATEX* pAudioFormat,
     _Out_   IMFAudioMediaType** ppIAudioMediaType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD
 STDMETHODCALLTYPE
 MFGetUncompressedVideoFormat(
     _In_    const MFVIDEOFORMAT* pVideoFormat
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI 
 MFInitVideoFormat(
     _In_    MFVIDEOFORMAT*          pVideoFormat,
     _In_    MFStandardVideoFormat   type
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFInitVideoFormat_RGB(
     _In_    MFVIDEOFORMAT*  pVideoFormat,
@@ -4205,11 +4335,13 @@ MFInitVideoFormat_RGB(
     _In_    DWORD           D3Dfmt /* 0 indicates sRGB */
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI 
 MFConvertColorInfoToDXVA(
     _Out_ DWORD* pdwToDXVA,
     _In_  const MFVIDEOFORMAT* pFromFormat
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFConvertColorInfoFromDXVA(
     _Inout_ MFVIDEOFORMAT* pToFormat,
@@ -4225,6 +4357,7 @@ MFConvertColorInfoFromDXVA(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCopyImage(
     _Out_writes_bytes_(_Inexpressible_(abs(lDestStride) *  dwLines)) BYTE* pDest,
     LONG    lDestStride,
@@ -4240,12 +4373,14 @@ STDAPI MFCopyImage(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFConvertFromFP16Array(
     _Out_writes_(dwCount) float* pDest,
     _In_reads_(dwCount) const WORD* pSrc,
     DWORD dwCount
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFConvertToFP16Array(
     _Out_writes_(dwCount) WORD* pDest,
     _In_reads_(dwCount) const float* pSrc,
@@ -4258,6 +4393,7 @@ STDAPI MFConvertToFP16Array(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreate2DMediaBuffer( 
     _In_ DWORD dwWidth,
     _In_ DWORD dwHeight,
@@ -4270,6 +4406,7 @@ STDAPI MFCreate2DMediaBuffer(
 //
 // Creates an optimal system memory media buffer from a media type
 //
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateMediaBufferFromMediaType(
     _In_ IMFMediaType* pMediaType,
     _In_ LONGLONG llDuration,   // Sample Duration, needed for audio
@@ -4307,6 +4444,7 @@ LO32(UINT64 unPacked)
     return (UINT32)unPacked;
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 UINT64
 Pack2UINT32AsUINT64(UINT32 unHigh, UINT32 unLow)
@@ -4314,6 +4452,7 @@ Pack2UINT32AsUINT64(UINT32 unHigh, UINT32 unLow)
     return ((UINT64)unHigh << 32) | unLow;
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 void
 Unpack2UINT32AsUINT64(UINT64 unPacked, _Out_ UINT32* punHigh, _Out_ UINT32* punLow)
@@ -4322,6 +4461,7 @@ Unpack2UINT32AsUINT64(UINT64 unPacked, _Out_ UINT32* punHigh, _Out_ UINT32* punL
     *punLow = LO32(unPacked);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 inline
 UINT64
 PackSize(UINT32 unWidth, UINT32 unHeight)
@@ -4329,6 +4469,7 @@ PackSize(UINT32 unWidth, UINT32 unHeight)
     return Pack2UINT32AsUINT64(unWidth, unHeight);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 inline
 void
 UnpackSize(UINT64 unPacked, _Out_ UINT32* punWidth, _Out_ UINT32* punHeight)
@@ -4336,6 +4477,7 @@ UnpackSize(UINT64 unPacked, _Out_ UINT32* punWidth, _Out_ UINT32* punHeight)
     Unpack2UINT32AsUINT64(unPacked, punWidth, punHeight);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 inline
 UINT64
 PackRatio(INT32 nNumerator, UINT32 unDenominator)
@@ -4343,6 +4485,7 @@ PackRatio(INT32 nNumerator, UINT32 unDenominator)
     return Pack2UINT32AsUINT64((UINT32)nNumerator, unDenominator);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 inline
 void
 UnpackRatio(UINT64 unPacked, _Out_ INT32* pnNumerator, _Out_ UINT32* punDenominator)
@@ -4354,6 +4497,7 @@ UnpackRatio(UINT64 unPacked, _Out_ INT32* pnNumerator, _Out_ UINT32* punDenomina
 //
 // "failsafe" inline get methods - return the stored value or return a default
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 UINT32
 MFGetAttributeUINT32(
@@ -4369,6 +4513,7 @@ MFGetAttributeUINT32(
     return unRet;
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 UINT64
 MFGetAttributeUINT64(
@@ -4384,6 +4529,7 @@ MFGetAttributeUINT64(
     return unRet;
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 double
 MFGetAttributeDouble(
@@ -4403,6 +4549,7 @@ MFGetAttributeDouble(
 // helpers for getting/setting ratios and sizes
 //
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFGetAttribute2UINT32asUINT64(
@@ -4424,6 +4571,7 @@ MFGetAttribute2UINT32asUINT64(
     return hr;
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFSetAttribute2UINT32asUINT64(
@@ -4436,6 +4584,7 @@ MFSetAttribute2UINT32asUINT64(
     return pAttributes->SetUINT64(guidKey, Pack2UINT32AsUINT64(unHigh32, unLow32));
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFGetAttributeRatio(
@@ -4448,6 +4597,7 @@ MFGetAttributeRatio(
     return MFGetAttribute2UINT32asUINT64(pAttributes, guidKey, punNumerator, punDenominator);
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFGetAttributeSize(
@@ -4460,6 +4610,7 @@ MFGetAttributeSize(
     return MFGetAttribute2UINT32asUINT64(pAttributes, guidKey, punWidth, punHeight);
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFSetAttributeRatio(
@@ -4472,6 +4623,7 @@ MFSetAttributeRatio(
     return MFSetAttribute2UINT32asUINT64(pAttributes, guidKey, unNumerator, unDenominator);
 }
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 inline
 HRESULT
 MFSetAttributeSize(
@@ -4485,6 +4637,7 @@ MFSetAttributeSize(
 }
 
 #ifdef _INTSAFE_H_INCLUDED_
+_Win32_metadata_supported_os_(windows8.0)
 inline
 HRESULT
 MFGetAttributeString(
@@ -4532,6 +4685,7 @@ MFGetAttributeString(
 //
 // Instantiates the MF-provided IMFCollection implementation
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateCollection(
     _Out_ IMFCollection **ppIMFCollection );
 
@@ -4559,11 +4713,13 @@ typedef enum _EAllocationType
     eAllocationTypeIgnore
 }   EAllocationType;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 EXTERN_C void* WINAPI MFHeapAlloc( size_t nSize,
                             ULONG dwFlags,
                             _In_opt_ char *pszFile,
                             int line,
                             EAllocationType eat);
+_Win32_metadata_supported_os_(windows6.0.6000)
 EXTERN_C void WINAPI MFHeapFree( void * pv );
 
 //////////////////////////       SourceResolver     ////////////////////////////
@@ -4584,6 +4740,7 @@ DEFINE_GUID(CLSID_MFSourceResolver,
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.1)
 LONGLONG WINAPI MFllMulDiv(LONGLONG a, LONGLONG b, LONGLONG c, LONGLONG d);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
@@ -4598,6 +4755,7 @@ LONGLONG WINAPI MFllMulDiv(LONGLONG a, LONGLONG b, LONGLONG c, LONGLONG d);
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFGetContentProtectionSystemCLSID(
     _In_ REFGUID guidProtectionSystemID,
     _Out_ CLSID *pclsid );
@@ -4642,6 +4800,7 @@ DEFINE_GUID(MF_DEVICESTREAM_ATTRIBUTE_SECURE_CAPABILITY,
 //  pSampleToAdd - the sample to append 
 //  dwMaxMergedDurationInMS - indicates the maximum duration that the combined sample should be allowed to occupy 
 //  pCombined - indicates that pSampleToAdd was successfully added to the base sample
+_Win32_metadata_supported_os_(windows10.0.19041)
 STDAPI MFCombineSamples( _In_ IMFSample *pSample, _In_ IMFSample* pSampleToAdd, _In_ DWORD dwMaxMergedDurationInMS, _Out_ BOOL* pMerged );
 
 // MFSplitSample
@@ -4650,6 +4809,7 @@ STDAPI MFCombineSamples( _In_ IMFSample *pSample, _In_ IMFSample* pSampleToAdd,
 //  dwOutputSampleMaxCount - maximum array size (use the BufferCount on pSample to find out an upper bound) 
 //  pdwOutputSampleCount - actual number of output samples produce
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 STDAPI MFSplitSample( _In_ IMFSample *pSample, _Out_writes_to_(dwOutputSampleMaxCount, *pdwOutputSampleCount) IMFSample** pOutputSamples, _In_ DWORD dwOutputSampleMaxCount, _Out_ DWORD* pdwOutputSampleCount );
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_VB)
 
