diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mfidl.h b/generation/WinSDK/RecompiledIdlHeaders/um/mfidl.h
index 42f91684..a7284c21 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mfidl.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mfidl.h
@@ -1047,6 +1047,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #include <windef.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -1114,6 +1115,7 @@ EXTERN_C const IID IID_IMFMediaSession;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("90377834-21D0-4dee-8214-BA2E3E6C1127")
     IMFMediaSession : public IMFMediaEventGenerator
     {
@@ -1363,6 +1365,7 @@ EXTERN_GUID( MF_SESSION_SERVER_CONTEXT, 0xafe5b291, 0x50fa, 0x46e8, 0xb9, 0xbe,
 EXTERN_GUID( MF_SESSION_REMOTE_SOURCE_MODE, 0xf4033ef4, 0x9bb3, 0x4378, 0x94, 0x1f, 0x85, 0xa0, 0x85, 0x6b, 0xc2, 0x44);
 EXTERN_GUID( MF_SESSION_APPROX_EVENT_OCCURRENCE_TIME, 0x190e852f, 0x6238, 0x42d1, 0xb5, 0xaf, 0x69, 0xea, 0x33, 0x8e, 0xf8, 0x50);
 EXTERN_GUID( MF_PMP_SERVER_CONTEXT, 0x2f00c910, 0xd2cf, 0x4278, 0x8b, 0x6a, 0xd0, 0x77, 0xfa, 0xc3, 0xa2, 0x5f);
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateMediaSession(
     IMFAttributes* pConfiguration,
     _Outptr_ IMFMediaSession** ppMediaSession
@@ -1371,6 +1374,7 @@ STDAPI MFCreateMediaSession(
 #pragma endregion
 #pragma region PC Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreatePMPMediaSession(
     DWORD dwCreationFlags,
     IMFAttributes *pConfiguration,
@@ -1448,6 +1452,7 @@ EXTERN_C const IID IID_IMFSourceResolver;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("FBE5A32D-A497-4b61-BB85-97B1A848A6E3")
     IMFSourceResolver : public IUnknown
     {
@@ -1775,16 +1780,20 @@ void __RPC_STUB IMFSourceResolver_RemoteEndCreateObjectFromByteStream_Stub(
 /* interface __MIDL_itf_mfidl_0000_0002 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateSourceResolver( 
         /* out */     _Outptr_ IMFSourceResolver     **ppISourceResolver);
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI CreatePropertyStore( 
         /* out */     _Outptr_ IPropertyStore        **ppStore);
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetSupportedSchemes(                      
        _Out_ PROPVARIANT* pPropVarSchemeArray  );  
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetSupportedMimeTypes(                      
        _Out_ PROPVARIANT* pPropVarMimeTypeArray  );  
 EXTERN_C const DECLSPEC_SELECTANY PROPERTYKEY MFPKEY_SourceOpenMonitor = { { 0x074d4637, 0xb5ae, 0x465d, 0xaf, 0x17, 0x1a, 0x53, 0x8d, 0x28, 0x59, 0xdd}, 0x02 }; 
@@ -1865,6 +1874,7 @@ EXTERN_C const IID IID_IMFMediaSource;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("279a808d-aec7-40c8-9c6b-a6b492c78a66")
     IMFMediaSource : public IMFMediaEventGenerator
     {
@@ -2092,6 +2102,7 @@ EXTERN_C const IID IID_IMFMediaSourceEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("3C9B2EB9-86D5-4514-A394-F56664F9F0D8")
     IMFMediaSourceEx : public IMFMediaSource
     {
@@ -2327,6 +2338,7 @@ EXTERN_C const IID IID_IMFClockConsumer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("6ef2a662-47c0-4666-b13d-cbb717f2fa2c")
     IMFClockConsumer : public IUnknown
     {
@@ -2440,6 +2452,7 @@ EXTERN_C const IID IID_IMFMediaStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D182108F-4EC6-443f-AA42-A71106EC825F")
     IMFMediaStream : public IMFMediaEventGenerator
     {
@@ -2651,6 +2664,7 @@ EXTERN_C const IID IID_IMFMediaSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6ef2a660-47c0-4666-b13d-cbb717f2fa2c")
     IMFMediaSink : public IUnknown
     {
@@ -2872,6 +2886,7 @@ EXTERN_C const IID IID_IMFStreamSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("0A97B3CF-8E7C-4a3d-8F8C-0C843DC247FB")
     IMFStreamSink : public IMFMediaEventGenerator
     {
@@ -3077,6 +3092,7 @@ EXTERN_C const IID IID_IMFVideoSampleAllocator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("86cbc910-e533-4751-8e3b-f19b5b806a03")
     IMFVideoSampleAllocator : public IUnknown
     {
@@ -3212,6 +3228,7 @@ EXTERN_C const IID IID_IMFVideoSampleAllocatorNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("A792CDBE-C374-4e89-8335-278E7B9956A4")
     IMFVideoSampleAllocatorNotify : public IUnknown
     {
@@ -3295,6 +3312,7 @@ EXTERN_C const IID IID_IMFVideoSampleAllocatorNotifyEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("3978AA1A-6D5B-4B7F-A340-90899189AE34")
     IMFVideoSampleAllocatorNotifyEx : public IMFVideoSampleAllocatorNotify
     {
@@ -3388,6 +3406,7 @@ EXTERN_C const IID IID_IMFVideoSampleAllocatorCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("992388B4-3372-4f67-8B6F-C84C071F4751")
     IMFVideoSampleAllocatorCallback : public IUnknown
     {
@@ -3488,6 +3507,7 @@ EXTERN_C const IID IID_IMFVideoSampleAllocatorEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("545b3a48-3283-4f62-866f-a62d8f598f9f")
     IMFVideoSampleAllocatorEx : public IMFVideoSampleAllocator
     {
@@ -3634,6 +3654,7 @@ EXTERN_C const IID IID_IMFDXGIDeviceManagerSource;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("20bc074b-7a8d-4609-8c3b-64a0a3b5d7ce")
     IMFDXGIDeviceManagerSource : public IUnknown
     {
@@ -3746,6 +3767,7 @@ EXTERN_C const IID IID_IMFVideoProcessorControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("A3F675D5-6119-4f7f-A100-1D8B280F0EFB")
     IMFVideoProcessorControl : public IUnknown
     {
@@ -3907,6 +3929,7 @@ EXTERN_C const IID IID_IMFVideoProcessorControl2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("BDE633D3-E1DC-4a7f-A693-BBAE399C4A20")
     IMFVideoProcessorControl2 : public IMFVideoProcessorControl
     {
@@ -4449,6 +4472,7 @@ EXTERN_C const IID IID_IMFTopology;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("83CF873A-F6DA-4bc8-823F-BACFD55DC433")
     IMFTopology : public IMFAttributes
     {
@@ -4990,6 +5014,7 @@ EXTERN_GUID( MF_LOCAL_PLUGIN_CONTROL_POLICY, 0xd91b0085, 0xc86d, 0x4f81, 0x88, 0
 #endif // (WINVER >= _WIN32_WINNT_WIN8) 
 #pragma region PC Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateTopology(
     _Outptr_ IMFTopology ** ppTopo );
 typedef 
@@ -5022,6 +5047,7 @@ EXTERN_C const IID IID_IMFTopologyNode;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("83CF873A-F6DA-4bc8-823F-BACFD55DC430")
     IMFTopologyNode : public IMFAttributes
     {
@@ -5716,6 +5742,7 @@ EXTERN_GUID( MF_TOPONODE_PRIMARYOUTPUT, 0x6304ef99, 0x16b2, 0x4ebe, 0x9d, 0x67,
 #pragma endregion
 #pragma region PC Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateTopologyNode(
     MF_TOPOLOGY_TYPE NodeType,
     _Outptr_ IMFTopologyNode ** ppNode );
@@ -5724,6 +5751,7 @@ STDAPI MFCreateTopologyNode(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if (WINVER >= _WIN32_WINNT_WIN7) 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFGetTopoNodeCurrentType(
     IMFTopologyNode* pNode,
     DWORD dwStreamIndex,
@@ -5750,6 +5778,7 @@ EXTERN_C const IID IID_IMFGetService;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("fa993888-4383-415a-a930-dd472a8cf6f7")
     IMFGetService : public IUnknown
     {
@@ -5837,6 +5866,7 @@ EXTERN_C const IID IID_IMFGetService;
 /* interface __MIDL_itf_mfidl_0000_0021 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFGetService(
     IUnknown* punkObject,
     REFGUID guidService,
@@ -5906,6 +5936,7 @@ EXTERN_C const IID IID_IMFClock;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2eb1e945-18b8-4139-9b1a-d5d584818530")
     IMFClock : public IUnknown
     {
@@ -6049,6 +6080,7 @@ EXTERN_C const IID IID_IMFClock;
 /* interface __MIDL_itf_mfidl_0000_0022 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI_(MFTIME) 
 MFGetSystemTime(
     );
@@ -6071,6 +6103,7 @@ EXTERN_C const IID IID_IMFPresentationClock;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("868CE85C-8EA9-4f55-AB82-B009A910A805")
     IMFPresentationClock : public IMFClock
     {
@@ -6289,6 +6322,7 @@ EXTERN_C const IID IID_IMFPresentationClock;
 #pragma endregion
 #pragma region PC Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreatePresentationClock(
     _Outptr_ IMFPresentationClock** ppPresentationClock
     );
@@ -6312,6 +6346,7 @@ EXTERN_C const IID IID_IMFPresentationTimeSource;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7FF12CCE-F76F-41c2-863B-1666C8E5E139")
     IMFPresentationTimeSource : public IMFClock
     {
@@ -6443,6 +6478,7 @@ EXTERN_C const IID IID_IMFPresentationTimeSource;
 /* interface __MIDL_itf_mfidl_0000_0024 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateSystemTimeSource(
     _Outptr_ IMFPresentationTimeSource** ppSystemTimeSource
@@ -6463,6 +6499,7 @@ EXTERN_C const IID IID_IMFClockStateSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F6696E82-74F7-4f3d-A178-8A5E09C3659F")
     IMFClockStateSink : public IUnknown
     {
@@ -6643,6 +6680,7 @@ EXTERN_C const IID IID_IMFPresentationDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("03cb2711-24d7-4db6-a17f-f3a7a479a536")
     IMFPresentationDescriptor : public IMFAttributes
     {
@@ -7077,6 +7115,7 @@ EXTERN_C const IID IID_IMFPresentationDescriptor;
 /* interface __MIDL_itf_mfidl_0000_0026 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreatePresentationDescriptor(
     DWORD cStreamDescriptors,
     _In_reads_opt_( cStreamDescriptors ) IMFStreamDescriptor** apStreamDescriptors,
@@ -7087,15 +7126,18 @@ STDAPI MFCreatePresentationDescriptor(
 #pragma endregion
 #pragma region Desktop or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFRequireProtectedEnvironment(
      _In_ IMFPresentationDescriptor* pPresentationDescriptor
      );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFSerializePresentationDescriptor(
     _In_ IMFPresentationDescriptor * pPD,
     _Out_ DWORD * pcbData,
     _Outptr_result_bytebuffer_to_(*pcbData, *pcbData) BYTE ** ppbData);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFDeserializePresentationDescriptor(
     _In_ DWORD cbData,
     _In_reads_( cbData ) BYTE * pbData,
@@ -7126,6 +7168,7 @@ EXTERN_C const IID IID_IMFStreamDescriptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("56c03d9c-9dbb-45f5-ab4b-d80f47c05938")
     IMFStreamDescriptor : public IMFAttributes
     {
@@ -7513,6 +7556,7 @@ EXTERN_C const IID IID_IMFStreamDescriptor;
 /* interface __MIDL_itf_mfidl_0000_0027 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateStreamDescriptor(
     DWORD dwStreamIdentifier,
     DWORD cMediaTypes,
@@ -7536,6 +7580,7 @@ EXTERN_C const IID IID_IMFMediaTypeHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e93dcf6c-4b07-4e1e-8123-aa16ed6eadf5")
     IMFMediaTypeHandler : public IUnknown
     {
@@ -7758,6 +7803,7 @@ void __RPC_STUB IMFMediaTypeHandler_RemoteGetCurrentMediaType_Stub(
 #pragma endregion
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateSimpleTypeHandler(
     _Outptr_ IMFMediaTypeHandler ** ppHandler );
 typedef 
@@ -7782,6 +7828,7 @@ EXTERN_C const IID IID_IMFTimer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e56e4cbd-8f70-49d8-a0f8-edb3d6ab9bf2")
     IMFTimer : public IUnknown
     {
@@ -7935,6 +7982,7 @@ EXTERN_C const IID IID_IMFShutdown;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("97ec2ea4-0e42-4937-97ac-9d6d328824e1")
     IMFShutdown : public IUnknown
     {
@@ -8027,9 +8075,11 @@ EXTERN_C const IID IID_IMFShutdown;
 #pragma endregion
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI 
 MFShutdownObject(
     IUnknown * pUnk );
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateAudioRenderer(
     IMFAttributes* pAudioAttributes,
@@ -8039,6 +8089,7 @@ MFCreateAudioRenderer(
 #pragma endregion
 #pragma region PC Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateAudioRendererActivate( 
     _Outptr_ IMFActivate ** ppActivate 
@@ -8057,12 +8108,14 @@ EXTERN_GUID( MF_AUDIO_RENDERER_ATTRIBUTE_SESSION_ID, 0xede4b5e3, 0xf805, 0x4d6c,
 EXTERN_GUID( MF_AUDIO_RENDERER_ATTRIBUTE_ENDPOINT_ID, 0xb10aaec3, 0xef71, 0x4cc3, 0xb8, 0x73, 0x5, 0xa9, 0xa0, 0x8b, 0x9f, 0x8e);
 EXTERN_GUID( MF_AUDIO_RENDERER_ATTRIBUTE_ENDPOINT_ROLE, 0x6ba644ff, 0x27c5, 0x4d02, 0x98, 0x87, 0xc2, 0x86, 0x19, 0xfd, 0xb9, 0x1b);
 EXTERN_GUID( MF_AUDIO_RENDERER_ATTRIBUTE_STREAM_CATEGORY, 0xa9770471, 0x92ec, 0x4df4, 0x94, 0xfe, 0x81, 0xc3, 0x6f, 0xc, 0x3a, 0x7a);
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateVideoRendererActivate(
     _In_ HWND hwndVideo, 
     _Outptr_ IMFActivate ** ppActivate 
     );
 #if (WINVER >= _WIN32_WINNT_WIN7) 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFCreateMPEG4MediaSink(
     _In_ IMFByteStream* pIByteStream, 
@@ -8070,6 +8123,7 @@ MFCreateMPEG4MediaSink(
     _In_opt_ IMFMediaType* pAudioMediaType, 
     _Outptr_ IMFMediaSink** ppIMediaSink 
     );
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFCreate3GPMediaSink(
     _In_ IMFByteStream* pIByteStream, 
@@ -8077,6 +8131,7 @@ MFCreate3GPMediaSink(
     _In_opt_ IMFMediaType* pAudioMediaType, 
     _Outptr_ IMFMediaSink** ppIMediaSink 
     );
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI
 MFCreateMP3MediaSink(
     _In_ IMFByteStream* pTargetByteStream, 
@@ -8084,18 +8139,21 @@ MFCreateMP3MediaSink(
     );
 #endif // (WINVER >= _WIN32_WINNT_WIN7) 
 #if (WINVER >= _WIN32_WINNT_WIN8) 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateAC3MediaSink(
     _In_ IMFByteStream* pTargetByteStream, 
     _In_ IMFMediaType* pAudioMediaType, 
      _Outptr_ IMFMediaSink** ppMediaSink 
     );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateADTSMediaSink(
     _In_ IMFByteStream* pTargetByteStream, 
     _In_ IMFMediaType* pAudioMediaType, 
      _Outptr_ IMFMediaSink** ppMediaSink 
     );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateMuxSink(
     _In_ GUID guidOutputSubType, 
@@ -8103,6 +8161,7 @@ MFCreateMuxSink(
     _In_opt_ IMFByteStream* pOutputByteStream, 
     _Outptr_ IMFMediaSink** ppMuxSink 
     );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI
 MFCreateFMPEG4MediaSink(
     _In_ IMFByteStream* pIByteStream, 
@@ -8112,6 +8171,7 @@ MFCreateFMPEG4MediaSink(
     );
 #endif // (WINVER >= _WIN32_WINNT_WIN8) 
 #if (WINVER >= _WIN32_WINNT_WINBLUE) 
+_Win32_metadata_supported_os_(windows8.1)
 STDAPI
 MFCreateAVIMediaSink(
     _In_ IMFByteStream* pIByteStream, 
@@ -8121,6 +8181,7 @@ MFCreateAVIMediaSink(
     );
 #endif // (WINVER >= _WIN32_WINNT_WINBLUE) 
 #if (WINVER >= _WIN32_WINNT_WINBLUE) 
+_Win32_metadata_supported_os_(windows8.1)
 STDAPI
 MFCreateWAVEMediaSink(
     _In_ IMFByteStream* pTargetByteStream, 
@@ -8144,6 +8205,7 @@ EXTERN_C const IID IID_IMFTopoLoader;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DE9A6157-F660-4643-B56A-DF9F7998C7CD")
     IMFTopoLoader : public IUnknown
     {
@@ -8231,6 +8293,7 @@ EXTERN_C const IID IID_IMFTopoLoader;
 /* interface __MIDL_itf_mfidl_0000_0031 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateTopoLoader(
     _Outptr_ IMFTopoLoader ** ppObj );
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
@@ -8253,6 +8316,7 @@ EXTERN_C const IID IID_IMFContentProtectionManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("ACF92459-6A61-42bd-B57C-B43E51203CB0")
     IMFContentProtectionManager : public IUnknown
     {
@@ -8413,6 +8477,7 @@ EXTERN_C const IID IID_IMFContentEnabler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D3C4EF59-49CE-4381-9071-D5BCD044C770")
     IMFContentEnabler : public IUnknown
     {
@@ -8694,6 +8759,7 @@ EXTERN_C const IID IID_IMFMetadata;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F88CFB8C-EF16-4991-B450-CB8C69E51704")
     IMFMetadata : public IUnknown
     {
@@ -8867,6 +8933,7 @@ EXTERN_C const IID IID_IMFMetadataProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("56181D2D-E221-4adb-B1C8-3CEE6A53F76F")
     IMFMetadataProvider : public IUnknown
     {
@@ -8993,6 +9060,7 @@ EXTERN_C const IID IID_IMFRateSupport;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("0a9ccdbc-d797-4563-9667-94ec5d79292d")
     IMFRateSupport : public IUnknown
     {
@@ -9139,6 +9207,7 @@ EXTERN_C const IID IID_IMFRateControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("88ddcd21-03c3-4275-91ed-55ee3929328f")
     IMFRateControl : public IUnknown
     {
@@ -9260,6 +9329,7 @@ EXTERN_C const IID IID_IMFTimecodeTranslate;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("ab9d8661-f7e8-4ef4-9861-89f334f94e74")
     IMFTimecodeTranslate : public IUnknown
     {
@@ -9425,6 +9495,7 @@ EXTERN_C const IID IID_IMFSeekInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("26AFEA53-D9ED-42B5-AB80-E64F9EE34779")
     IMFSeekInfo : public IUnknown
     {
@@ -9540,6 +9611,7 @@ EXTERN_C const IID IID_IMFSimpleAudioVolume;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("089EDF13-CF71-4338-8D13-9E569DBDC319")
     IMFSimpleAudioVolume : public IUnknown
     {
@@ -9678,6 +9750,7 @@ EXTERN_C const IID IID_IMFAudioStreamVolume;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("76B1BBDB-4EC8-4f36-B106-70A9316DF593")
     IMFAudioStreamVolume : public IUnknown
     {
@@ -9842,6 +9915,7 @@ EXTERN_C const IID IID_IMFAudioPolicy;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("a0638c2b-6465-4395-9ae7-a321a9fd2856")
     IMFAudioPolicy : public IUnknown
     {
@@ -10003,6 +10077,7 @@ EXTERN_C const IID IID_IMFSampleGrabberSinkCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8C7B80BF-EE42-4b59-B1DF-55668E1BDCA8")
     IMFSampleGrabberSinkCallback : public IMFClockStateSink
     {
@@ -10174,6 +10249,7 @@ EXTERN_C const IID IID_IMFSampleGrabberSinkCallback;
 /* interface __MIDL_itf_mfidl_0000_0043 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI
 MFCreateSampleGrabberSinkActivate(
     IMFMediaType *pIMFMediaType,
@@ -10199,6 +10275,7 @@ EXTERN_C const IID IID_IMFSampleGrabberSinkCallback2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("ca86aa50-c46e-429e-ab27-16d6ac6844cb")
     IMFSampleGrabberSinkCallback2 : public IMFSampleGrabberSinkCallback
     {
@@ -10414,6 +10491,7 @@ EXTERN_C const IID IID_IMFWorkQueueServices;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("35FE1BB8-A3A9-40fe-BBEC-EB569C9CCCA3")
     IMFWorkQueueServices : public IUnknown
     {
@@ -10827,6 +10905,7 @@ EXTERN_C const IID IID_IMFWorkQueueServicesEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("96bf961b-40fe-42f1-ba9d-320238b49700")
     IMFWorkQueueServicesEx : public IMFWorkQueueServices
     {
@@ -11180,6 +11259,7 @@ EXTERN_C const IID IID_IMFQualityManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8D009D86-5B9F-4115-B1FC-9F80D52AB8AB")
     IMFQualityManager : public IUnknown
     {
@@ -11340,6 +11420,7 @@ EXTERN_C const IID IID_IMFQualityManager;
 /* interface __MIDL_itf_mfidl_0000_0047 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateStandardQualityManager(
     _Outptr_ IMFQualityManager **ppQualityManager );
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
@@ -11368,6 +11449,7 @@ EXTERN_C const IID IID_IMFQualityAdvise;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("EC15E2E9-E36B-4f7c-8758-77D452EF4CE7")
     IMFQualityAdvise : public IUnknown
     {
@@ -11520,6 +11602,7 @@ EXTERN_C const IID IID_IMFQualityAdvise2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("F3706F0D-8EA2-4886-8000-7155E9EC2EAE")
     IMFQualityAdvise2 : public IMFQualityAdvise
     {
@@ -11661,6 +11744,7 @@ EXTERN_C const IID IID_IMFQualityAdviseLimits;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("dfcd8e4d-30b5-4567-acaa-8eb5b7853dc9")
     IMFQualityAdviseLimits : public IUnknown
     {
@@ -11774,6 +11858,7 @@ EXTERN_C const IID IID_IMFRealTimeClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2347D60B-3FB5-480c-8803-8DF3ADCD3EF0")
     IMFRealTimeClient : public IUnknown
     {
@@ -11900,6 +11985,7 @@ EXTERN_C const IID IID_IMFRealTimeClientEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("03910848-AB16-4611-B100-17B88AE2F248")
     IMFRealTimeClientEx : public IUnknown
     {
@@ -12044,6 +12130,7 @@ EXTERN_C const IID IID_IMFSequencerSource;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("197CD219-19CB-4de1-A64C-ACF2EDCBE59E")
     IMFSequencerSource : public IUnknown
     {
@@ -12200,10 +12287,12 @@ EXTERN_C const IID IID_IMFSequencerSource;
 /* [local] */ 
 
 EXTERN_GUID( MF_TIME_FORMAT_SEGMENT_OFFSET, 0xc8b8be77, 0x869c, 0x431d, 0x81, 0x2e, 0x16, 0x96, 0x93, 0xf6, 0x5a, 0x39 );
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateSequencerSource(
     IUnknown *pReserved,
     _Outptr_ IMFSequencerSource **ppSequencerSource
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateSequencerSegmentOffset(
     MFSequencerElementId dwId,
     MFTIME hnsOffset,
@@ -12214,6 +12303,7 @@ STDAPI MFCreateSequencerSegmentOffset(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 #if (WINVER >= _WIN32_WINNT_WIN7) 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateAggregateSource(
     _In_ IMFCollection *pSourceCollection,
     _Outptr_ IMFMediaSource **ppAggSource
@@ -12239,6 +12329,7 @@ EXTERN_C const IID IID_IMFMediaSourceTopologyProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("0E1D6009-C9F3-442d-8C51-A42D2D49452F")
     IMFMediaSourceTopologyProvider : public IUnknown
     {
@@ -12330,6 +12421,7 @@ EXTERN_C const IID IID_IMFMediaSourcePresentationProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("0E1D600a-C9F3-442d-8C51-A42D2D49452F")
     IMFMediaSourcePresentationProvider : public IUnknown
     {
@@ -12447,6 +12539,7 @@ EXTERN_C const IID IID_IMFTopologyNodeAttributeEditor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("676aa6dd-238a-410d-bb99-65668d01605a")
     IMFTopologyNodeAttributeEditor : public IUnknown
     {
@@ -12596,6 +12689,7 @@ EXTERN_C const IID IID_IMFByteStreamBuffering;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6d66d782-1d4f-4db7-8c63-cb8c77f1ef5e")
     IMFByteStreamBuffering : public IUnknown
     {
@@ -12705,6 +12799,7 @@ EXTERN_C const IID IID_IMFByteStreamCacheControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("F5042EA4-7A96-4a75-AA7B-2BE1EF7F88D5")
     IMFByteStreamCacheControl : public IUnknown
     {
@@ -12788,6 +12883,7 @@ EXTERN_C const IID IID_IMFByteStreamTimeSeek;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("64976BFA-FB61-4041-9069-8C9A5F659BEB")
     IMFByteStreamTimeSeek : public IUnknown
     {
@@ -12924,6 +13020,7 @@ EXTERN_C const IID IID_IMFByteStreamCacheControl2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("71CE469C-F34B-49EA-A56B-2D2A10E51149")
     IMFByteStreamCacheControl2 : public IMFByteStreamCacheControl
     {
@@ -13062,6 +13159,7 @@ EXTERN_C const IID IID_IMFNetCredential;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5b87ef6a-7ed8-434f-ba0e-184fac1628d1")
     IMFNetCredential : public IUnknown
     {
@@ -13245,6 +13343,7 @@ EXTERN_C const IID IID_IMFNetCredentialManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5b87ef6b-7ed8-434f-ba0e-184fac1628d1")
     IMFNetCredentialManager : public IUnknown
     {
@@ -13397,6 +13496,7 @@ EXTERN_C const IID IID_IMFNetCredentialCache;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5b87ef6c-7ed8-434f-ba0e-184fac1628d1")
     IMFNetCredentialCache : public IUnknown
     {
@@ -13526,6 +13626,7 @@ EXTERN_C const IID IID_IMFNetCredentialCache;
 /* interface __MIDL_itf_mfidl_0000_0063 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI 
 MFCreateCredentialCache(
     _Outptr_ IMFNetCredentialCache ** ppCache);
@@ -13550,6 +13651,7 @@ EXTERN_C const IID IID_IMFSSLCertificateManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("61f7d887-1230-4a8b-aeba-8ad434d1a64d")
     IMFSSLCertificateManager : public IUnknown
     {
@@ -13742,6 +13844,7 @@ EXTERN_C const IID IID_IMFNetResourceFilter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("091878a3-bf11-4a5c-bc9f-33995b06ef2d")
     IMFNetResourceFilter : public IUnknown
     {
@@ -13860,6 +13963,7 @@ EXTERN_C const IID IID_IMFSourceOpenMonitor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("059054B3-027C-494C-A27D-9113291CF87F")
     IMFSourceOpenMonitor : public IUnknown
     {
@@ -13947,6 +14051,7 @@ EXTERN_C const IID IID_IMFNetProxyLocator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e9cd0383-a268-4bb4-82de-658d53574d41")
     IMFNetProxyLocator : public IUnknown
     {
@@ -14086,6 +14191,7 @@ EXTERN_C const IID IID_IMFNetProxyLocator;
 /* interface __MIDL_itf_mfidl_0000_0067 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateProxyLocator(
     LPCWSTR pszProtocol,
     IPropertyStore* pProxyConfig, 
@@ -14106,6 +14212,7 @@ EXTERN_C const IID IID_IMFNetProxyLocatorFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e9cd0384-a268-4bb4-82de-658d53574d41")
     IMFNetProxyLocatorFactory : public IUnknown
     {
@@ -14197,6 +14304,7 @@ EXTERN_C const IID IID_IMFSaveJob;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e9931663-80bf-4c6e-98af-5dcf58747d1f")
     IMFSaveJob : public IUnknown
     {
@@ -14346,6 +14454,7 @@ EXTERN_C const IID IID_IMFNetSchemeHandlerConfig;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7BE19E73-C9BF-468a-AC5A-A5E8653BEC87")
     IMFNetSchemeHandlerConfig : public IUnknown
     {
@@ -14451,6 +14560,7 @@ EXTERN_C const IID IID_IMFNetSchemeHandlerConfig;
 /* interface __MIDL_itf_mfidl_0000_0070 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateNetSchemePlugin(
     REFIID riid, 
     LPVOID *ppvHandler );
@@ -14597,6 +14707,7 @@ EXTERN_C const IID IID_IMFSchemeHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6D4C7B74-52A0-4bb7-B0DB-55F29F47A668")
     IMFSchemeHandler : public IUnknown
     {
@@ -14749,6 +14860,7 @@ EXTERN_C const IID IID_IMFByteStreamHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("BB420AA4-765B-4a1f-91FE-D6A8A143924C")
     IMFByteStreamHandler : public IUnknown
     {
@@ -14916,6 +15028,7 @@ EXTERN_C const IID IID_IMFTrustedInput;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("542612C4-A1B8-4632-B521-DE11EA64A0B0")
     IMFTrustedInput : public IUnknown
     {
@@ -15054,6 +15167,7 @@ EXTERN_C const IID IID_IMFInputTrustAuthority;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D19F8E98-B126-4446-890C-5DCB7AD71453")
     IMFInputTrustAuthority : public IUnknown
     {
@@ -15223,6 +15337,7 @@ EXTERN_C const IID IID_IMFTrustedOutput;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D19F8E95-B126-4446-890C-5DCB7AD71453")
     IMFTrustedOutput : public IUnknown
     {
@@ -15340,6 +15455,7 @@ EXTERN_C const IID IID_IMFOutputTrustAuthority;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("D19F8E94-B126-4446-890C-5DCB7AD71453")
     IMFOutputTrustAuthority : public IUnknown
     {
@@ -15461,6 +15577,7 @@ EXTERN_C const IID IID_IMFOutputPolicy;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7F00F10A-DAED-41AF-AB26-5FDFA4DFBA3C")
     IMFOutputPolicy : public IMFAttributes
     {
@@ -15929,6 +16046,7 @@ EXTERN_C const IID IID_IMFOutputSchema;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7BE0FC5B-ABD9-44FB-A5C8-F50136E71599")
     IMFOutputSchema : public IMFAttributes
     {
@@ -16471,6 +16589,7 @@ EXTERN_C const IID IID_IMFSecureChannel;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("d0ae555d-3b12-4d97-b060-0990bc5aeb67")
     IMFSecureChannel : public IUnknown
     {
@@ -16600,6 +16719,7 @@ EXTERN_C const IID IID_IMFSampleProtection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("8e36395f-c7b9-43c4-a54d-512b4af63c95")
     IMFSampleProtection : public IUnknown
     {
@@ -16779,6 +16899,7 @@ EXTERN_C const IID IID_IMFMediaSinkPreroll;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5dfd4b2a-7674-4110-a4e6-8a68fd5f3688")
     IMFMediaSinkPreroll : public IUnknown
     {
@@ -16878,6 +16999,7 @@ EXTERN_C const IID IID_IMFFinalizableMediaSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("EAECB74A-9A50-42ce-9541-6A7F57AA4AD7")
     IMFFinalizableMediaSink : public IMFMediaSink
     {
@@ -17083,6 +17205,7 @@ EXTERN_C const IID IID_IMFStreamingSinkConfig;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("9db7aa41-3cc5-40d4-8509-555804ad34cc")
     IMFStreamingSinkConfig : public IUnknown
     {
@@ -17187,6 +17310,7 @@ EXTERN_C const IID IID_IMFRemoteProxy;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("994e23ad-1cc2-493c-b9fa-46f1cb040fa4")
     IMFRemoteProxy : public IUnknown
     {
@@ -17304,6 +17428,7 @@ EXTERN_C const IID IID_IMFObjectReferenceStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("09EF5BE3-C8A7-469e-8B70-73BF25BB193F")
     IMFObjectReferenceStream : public IUnknown
     {
@@ -17421,6 +17546,7 @@ EXTERN_C const IID IID_IMFPMPHost;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("F70CA1A9-FDC7-4782-B994-ADFFB1C98606")
     IMFPMPHost : public IUnknown
     {
@@ -17570,6 +17696,7 @@ EXTERN_C const IID IID_IMFPMPClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6C4E655D-EAD8-4421-B6B9-54DCDBBDF820")
     IMFPMPClient : public IUnknown
     {
@@ -17657,6 +17784,7 @@ EXTERN_C const IID IID_IMFPMPServer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("994e23af-1cc2-493c-b9fa-46f1cb040fa4")
     IMFPMPServer : public IUnknown
     {
@@ -17762,6 +17890,7 @@ EXTERN_C const IID IID_IMFPMPServer;
 /* interface __MIDL_itf_mfidl_0000_0088 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreatePMPServer(
     DWORD dwCreationFlags,
     _Outptr_ IMFPMPServer** ppPMPServer
@@ -17782,6 +17911,7 @@ EXTERN_C const IID IID_IMFRemoteDesktopPlugin;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("1cde6309-cae0-4940-907e-c1ec9c3d1d4a")
     IMFRemoteDesktopPlugin : public IUnknown
     {
@@ -17861,8 +17991,10 @@ EXTERN_C const IID IID_IMFRemoteDesktopPlugin;
 /* interface __MIDL_itf_mfidl_0000_0089 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFCreateRemoteDesktopPlugin(
     _Outptr_ IMFRemoteDesktopPlugin** ppPlugin );
+_Win32_metadata_supported_os_(windows6.0.6000)
 EXTERN_C HRESULT STDAPICALLTYPE CreateNamedPropertyStore(
         _Outptr_ INamedPropertyStore **ppStore
         );
@@ -17883,6 +18015,7 @@ EXTERN_C const IID IID_IMFSAMIStyle;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("A7E025DD-5303-4a62-89D6-E747E1EFAC73")
     IMFSAMIStyle : public IUnknown
     {
@@ -18009,6 +18142,7 @@ EXTERN_GUID( MF_SD_SAMI_LANGUAGE, 0x36fcb98a, 0x6cd0, 0x44cb, 0xac, 0xb9, 0xa8,
 #if (WINVER >= _WIN32_WINNT_WIN7) 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateSampleCopierMFT(_Outptr_ IMFTransform** ppCopierMFT);
 
 
@@ -18026,6 +18160,7 @@ EXTERN_C const IID IID_IMFTranscodeProfile;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("4ADFDBA3-7AB0-4953-A62B-461E7FF3DA1E")
     IMFTranscodeProfile : public IUnknown
     {
@@ -18238,9 +18373,11 @@ EXTERN_GUID( MF_XVP_CALLER_ALLOCATES_OUTPUT, 0x4a2cabc, 0xcab, 0x40b1, 0xa1, 0xb
 EXTERN_GUID(CLSID_VideoProcessorMFT, 0x88753b26, 0x5b24, 0x49bd, 0xb2, 0xe7, 0xc, 0x44, 0x5c, 0x78, 0xc9, 0x82);
 #endif // (WINVER < _WIN32_WINNT_WINTHRESHOLD) 
 #endif // (WINVER >= _WIN32_WINNT_WINBLUE) 
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateTranscodeProfile(
     _Outptr_ IMFTranscodeProfile** ppTranscodeProfile
     );
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateTranscodeTopology(
     _In_ IMFMediaSource* pSrc,
     _In_ LPCWSTR pwszOutputFilePath,
@@ -18252,6 +18389,7 @@ STDAPI MFCreateTranscodeTopology(
 #if (WINVER >= _WIN32_WINNT_WIN8) 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateTranscodeTopologyFromByteStream(
     _In_ IMFMediaSource* pSrc,
     _In_ IMFByteStream* pOutputStream,
@@ -18263,6 +18401,7 @@ STDAPI MFCreateTranscodeTopologyFromByteStream(
 #endif // (WINVER >= _WIN32_WINNT_WIN8) 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFTranscodeGetAudioOutputAvailableTypes(
  _In_ REFGUID guidSubType,
  _In_ DWORD dwMFTFlags,
@@ -18292,6 +18431,7 @@ EXTERN_C const IID IID_IMFTranscodeSinkInfoProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("8CFFCD2E-5A03-4a3a-AFF7-EDCD107C620E")
     IMFTranscodeSinkInfoProvider : public IUnknown
     {
@@ -18410,6 +18550,7 @@ EXTERN_C const IID IID_IMFTranscodeSinkInfoProvider;
 /* interface __MIDL_itf_mfidl_0000_0092 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.1)
  STDAPI MFCreateTranscodeSinkActivate( 
  _Outptr_ IMFActivate** ppActivate ); 
 
@@ -18428,6 +18569,7 @@ EXTERN_C const IID IID_IMFFieldOfUseMFTUnlock;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("508E71D3-EC66-4fc3-8775-B4B9ED6BA847")
     IMFFieldOfUseMFTUnlock : public IUnknown
     {
@@ -18536,6 +18678,7 @@ EXTERN_C const IID IID_IMFLocalMFTRegistration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("149c4d73-b4be-4f8d-8b87-079e926b6add")
     IMFLocalMFTRegistration : public IUnknown
     {
@@ -18751,6 +18894,7 @@ EXTERN_C const IID IID_IMFPMPHostApp;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("84d2054a-3aa1-4728-a3b0-440a418cf49c")
     IMFPMPHostApp : public IUnknown
     {
@@ -18868,6 +19012,7 @@ EXTERN_C const IID IID_IMFPMPClientApp;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("c004f646-be2c-48f3-93a2-a0983eba1108")
     IMFPMPClientApp : public IUnknown
     {
@@ -18965,6 +19110,7 @@ EXTERN_C const IID IID_IMFMediaStreamSourceSampleRequest;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("380b9af9-a85b-4e78-a2af-ea5ce645c6b4")
     IMFMediaStreamSourceSampleRequest : public IUnknown
     {
@@ -19061,6 +19207,7 @@ EXTERN_C const IID IID_IMFTrackedSample;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("245BF8E9-0755-40f7-88A5-AE0F18D55E17")
     IMFTrackedSample : public IUnknown
     {
@@ -19144,15 +19291,18 @@ EXTERN_C const IID IID_IMFTrackedSample;
 /* interface __MIDL_itf_mfidl_0000_0099 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateTrackedSample(
     _Outptr_ IMFTrackedSample** ppMFSample);
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 #pragma region Desktop or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateMFByteStreamOnStream(
     IStream*        pStream,
     _Outptr_ IMFByteStream** ppByteStream);
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateStreamOnMFByteStream(
     _In_ IMFByteStream* pByteStream,
     _Outptr_ IStream** ppStream);
@@ -19160,16 +19310,20 @@ STDAPI MFCreateStreamOnMFByteStream(
 #pragma endregion
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateMFByteStreamOnStreamEx(
     _In_ IUnknown* punkStream,
     _Outptr_ IMFByteStream** ppByteStream );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateStreamOnMFByteStreamEx(
     _In_ IMFByteStream* pByteStream,
     _In_ REFIID riid,
     _Outptr_ void **ppv );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateMediaTypeFromProperties(
     _In_ IUnknown* punkStream,
     _Outptr_ IMFMediaType** ppMediaType );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreatePropertiesFromMediaType(
     _In_ IMFMediaType* pMediaType,
     _In_ REFIID riid,
@@ -19194,15 +19348,18 @@ EXTERN_GUID(CLSID_NetSchemePlugin, 0xe9f4ebab, 0xd97b, 0x463e, 0xa2, 0xb1, 0xc5,
 #pragma endregion
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFEnumDeviceSources(
     _In_                IMFAttributes*                      pAttributes,
     _Outptr_result_buffer_(*pcSourceActivate) IMFActivate***    pppSourceActivate,
     _Out_               UINT32*                             pcSourceActivate
 );
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateDeviceSource(
     _In_  IMFAttributes*   pAttributes,
     _Outptr_ IMFMediaSource** ppSource
 );
+_Win32_metadata_supported_os_(windows6.1)
 STDAPI MFCreateDeviceSourceActivate( 
     _In_  IMFAttributes*   pAttributes,
     _Outptr_ IMFActivate**   ppActivate
@@ -19273,6 +19430,7 @@ EXTERN_C const IID IID_IMFProtectedEnvironmentAccess;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("ef5dc845-f0d9-4ec9-b00c-cb5183d38434")
     IMFProtectedEnvironmentAccess : public IUnknown
     {
@@ -19389,6 +19547,7 @@ EXTERN_C const IID IID_IMFSignedLibrary;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("4a724bca-ff6a-4c07-8e0d-7a358421cf06")
     IMFSignedLibrary : public IUnknown
     {
@@ -19480,6 +19639,7 @@ EXTERN_C const IID IID_IMFSystemId;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("fff4af3a-1fc1-4ef9-a29b-d26c49e2f31a")
     IMFSystemId : public IUnknown
     {
@@ -19588,16 +19748,20 @@ EXTERN_C const IID IID_IMFSystemId;
 /* interface __MIDL_itf_mfidl_0000_0102 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFCreateProtectedEnvironmentAccess( 
     _Outptr_ IMFProtectedEnvironmentAccess**   ppAccess
 );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFLoadSignedLibrary( 
     _In_ LPCWSTR pszName,
     _Outptr_ IMFSignedLibrary**   ppLib
 );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFGetSystemId( 
     _Outptr_ IMFSystemId** ppId
 );
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI MFGetLocalId( 
     _In_reads_bytes_(size) const BYTE *verifier,
     _In_ UINT32 size,
@@ -19653,6 +19817,7 @@ EXTERN_C const IID IID_IMFContentProtectionDevice;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("E6257174-A060-4C9A-A088-3B1B471CAD28")
     IMFContentProtectionDevice : public IUnknown
     {
@@ -19765,9 +19930,11 @@ EXTERN_C const IID IID_IMFContentProtectionDevice;
 /* interface __MIDL_itf_mfidl_0000_0103 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI MFCreateContentProtectionDevice( 
    _In_ REFGUID ProtectionSystemId, 
    _Outptr_ IMFContentProtectionDevice **ContentProtectionDevice); 
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI MFIsContentProtectionDeviceSupported( 
    _In_ REFGUID ProtectionSystemId, 
    _Out_ BOOL *isSupported); 
@@ -19787,6 +19954,7 @@ EXTERN_C const IID IID_IMFContentDecryptorContext;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("7EC4B1BD-43FB-4763-85D2-64FCB5C5F4CB")
     IMFContentDecryptorContext : public IUnknown
     {
@@ -19876,6 +20044,7 @@ EXTERN_C const IID IID_IMFContentDecryptorContext;
 
 EXTERN_GUID( MF_CONTENT_DECRYPTOR_SERVICE, 0x68a72927, 0xfc7b, 0x44ee, 0x85, 0xf4, 0x7c, 0x51, 0xbd, 0x55, 0xa6, 0x59);
 EXTERN_GUID( MF_CONTENT_PROTECTION_DEVICE_SERVICE, 0xff58436f, 0x76a0, 0x41fe, 0xb5, 0x66, 0x10, 0xcc, 0x53, 0x96, 0x2e, 0xdd);
+_Win32_metadata_supported_os_(windows10.0.10240)
 STDAPI MFCreateContentDecryptorContext( 
    _In_ REFGUID guidMediaProtectionSystemId, 
    _In_opt_ IMFDXGIDeviceManager *pD3DManager, 
@@ -19969,6 +20138,7 @@ EXTERN_C const IID IID_IMFNetCrossOriginSupport;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("bc2b7d44-a72d-49d5-8376-1480dee58b22")
     IMFNetCrossOriginSupport : public IUnknown
     {
@@ -20095,6 +20265,7 @@ EXTERN_C const IID IID_IMFHttpDownloadRequest;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("F779FDDF-26E7-4270-8A8B-B983D1859DE0")
     IMFHttpDownloadRequest : public IUnknown
     {
@@ -20425,6 +20596,7 @@ EXTERN_C const IID IID_IMFHttpDownloadSession;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("71FA9A2C-53CE-4662-A132-1A7E8CBF62DB")
     IMFHttpDownloadSession : public IUnknown
     {
@@ -20558,6 +20730,7 @@ EXTERN_C const IID IID_IMFHttpDownloadSessionProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("1B4CF4B9-3A16-4115-839D-03CC5C99DF01")
     IMFHttpDownloadSessionProvider : public IUnknown
     {
@@ -21129,6 +21302,7 @@ EXTERN_C const IID IID_IMFSensorDevice;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.14393)
     MIDL_INTERFACE("FB9F48F2-2A18-4E28-9730-786F30F04DC4")
     IMFSensorDevice : public IUnknown
     {
@@ -21340,6 +21514,7 @@ EXTERN_C const IID IID_IMFSensorGroup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.14393)
     MIDL_INTERFACE("4110243A-9757-461F-89F1-F22345BCAB4E")
     IMFSensorGroup : public IUnknown
     {
@@ -21530,6 +21705,7 @@ EXTERN_C const IID IID_IMFSensorStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("E9A42171-C56E-498A-8B39-EDA5A070B7FC")
     IMFSensorStream : public IMFAttributes
     {
@@ -21942,6 +22118,7 @@ EXTERN_C const IID IID_IMFSensorTransformFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("EED9C2EE-66B4-4F18-A697-AC7D3960215C")
     IMFSensorTransformFactory : public IUnknown
     {
@@ -22101,12 +22278,14 @@ EXTERN_C const IID IID_IMFSensorTransformFactory;
 /* interface __MIDL_itf_mfidl_0000_0114 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows10.0.14393)
 STDAPI
 MFCreateSensorGroup(
     _In_z_ LPCWSTR SensorGroupSymbolicLink,
     _COM_Outptr_ IMFSensorGroup** ppSensorGroup
     );
 
+_Win32_metadata_supported_os_(windows10.0.15063)
 STDAPI
 MFCreateSensorStream(
     _In_ DWORD StreamId,
@@ -22138,6 +22317,7 @@ EXTERN_C const IID IID_IMFSensorProfile;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.17134)
     MIDL_INTERFACE("22F765D1-8DAB-4107-846D-56BAF72215E7")
     IMFSensorProfile : public IUnknown
     {
@@ -22276,6 +22456,7 @@ EXTERN_C const IID IID_IMFSensorProfileCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.17134)
     MIDL_INTERFACE("C95EA55B-0187-48BE-9353-8D2507662351")
     IMFSensorProfileCollection : public IUnknown
     {
@@ -22424,6 +22605,7 @@ EXTERN_C const IID IID_IMFSensorProfileCollection;
 /* interface __MIDL_itf_mfidl_0000_0116 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows10.0.17134)
 STDAPI
 MFCreateSensorProfile(
     _In_ REFGUID ProfileType,
@@ -22431,6 +22613,7 @@ MFCreateSensorProfile(
     _In_opt_z_ LPCWSTR Constraints,
     _COM_Outptr_ IMFSensorProfile** ppProfile
     );
+_Win32_metadata_supported_os_(windows10.0.15063)
 STDAPI
 MFCreateSensorProfileCollection(
     _COM_Outptr_ IMFSensorProfileCollection** ppSensorProfile
@@ -22453,6 +22636,7 @@ EXTERN_C const IID IID_IMFSensorProcessActivity;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("39DC7F4A-B141-4719-813C-A7F46162A2B8")
     IMFSensorProcessActivity : public IUnknown
     {
@@ -22579,6 +22763,7 @@ EXTERN_C const IID IID_IMFSensorActivityReport;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("3E8C4BE1-A8C2-4528-90DE-2851BDE5FEAD")
     IMFSensorActivityReport : public IUnknown
     {
@@ -22725,6 +22910,7 @@ EXTERN_C const IID IID_IMFSensorActivitiesReport;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("683F7A5E-4A19-43CD-B1A9-DBF4AB3F7777")
     IMFSensorActivitiesReport : public IUnknown
     {
@@ -22846,6 +23032,7 @@ EXTERN_C const IID IID_IMFSensorActivitiesReportCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("DE5072EE-DBE3-46DC-8A87-B6F631194751")
     IMFSensorActivitiesReportCallback : public IUnknown
     {
@@ -22933,6 +23120,7 @@ EXTERN_C const IID IID_IMFSensorActivityMonitor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("D0CEF145-B3F4-4340-A2E5-7A5080CA05CB")
     IMFSensorActivityMonitor : public IUnknown
     {
@@ -23822,6 +24010,7 @@ EXTERN_C const IID IID_IMFRelativePanelReport;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.19041)
     MIDL_INTERFACE("F25362EA-2C0E-447F-81E2-755914CDC0C3")
     IMFRelativePanelReport : public IUnknown
     {
@@ -23909,6 +24098,7 @@ EXTERN_C const IID IID_IMFRelativePanelWatcher;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.19041)
     MIDL_INTERFACE("421AF7F6-573E-4AD0-8FDA-2E57CEDB18C6")
     IMFRelativePanelWatcher : public IMFShutdown
     {
@@ -24040,6 +24230,7 @@ EXTERN_C const IID IID_IMFRelativePanelWatcher;
 /* [local] */ 
 
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 STDAPI
 MFCreateRelativePanelWatcher(
     _In_ PCWSTR videoDeviceId,
@@ -24062,6 +24253,7 @@ EXTERN_C const IID IID_IMFVideoCaptureSampleAllocator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.19041)
     MIDL_INTERFACE("725B77C7-CA9F-4FE5-9D72-9946BF9B3C70")
     IMFVideoCaptureSampleAllocator : public IMFVideoSampleAllocator
     {
@@ -24222,6 +24414,7 @@ EXTERN_C const IID IID_IMFSampleAllocatorControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.19041)
     MIDL_INTERFACE("DA62B958-3A38-4A97-BD27-149C640C0771")
     IMFSampleAllocatorControl : public IUnknown
     {
@@ -25882,14 +26075,22 @@ unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal(  __RPC__in unsigned long *,
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
 void                      __RPC_USER  LPSAFEARRAY_UserFree(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
 /* [local] */ HRESULT STDMETHODCALLTYPE IMFSourceResolver_BeginCreateObjectFromURL_Proxy( 
