diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mfmediaengine.h b/generation/WinSDK/RecompiledIdlHeaders/um/mfmediaengine.h
index d1b6f45a..2429a6a8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mfmediaengine.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mfmediaengine.h
@@ -438,6 +438,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #if (WINVER >= _WIN32_WINNT_WIN8) 
 
 // Prevent a name collision
@@ -473,6 +476,7 @@ EXTERN_C const IID IID_IMFMediaError;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("fc0e10d2-ab2a-4501-a951-06bb1075184c")
     IMFMediaError : public IUnknown
     {
@@ -591,6 +595,7 @@ EXTERN_C const IID IID_IMFMediaTimeRange;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("db71a2fc-078a-414e-9df9-8c2531b0aa6c")
     IMFMediaTimeRange : public IUnknown
     {
@@ -799,6 +804,7 @@ EXTERN_C const IID IID_IMFMediaEngineNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("fee7c112-e776-42b5-9bbf-0048524e2bd5")
     IMFMediaEngineNotify : public IUnknown
     {
@@ -894,6 +900,7 @@ EXTERN_C const IID IID_IMFMediaEngineSrcElements;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("7a5e5354-b114-4c72-b991-3131d75032ea")
     IMFMediaEngineSrcElements : public IUnknown
     {
@@ -1114,6 +1121,7 @@ EXTERN_C const IID IID_IMFMediaEngine;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("98a1b0bb-03eb-4935-ae7c-93c1fa0e1c93")
     IMFMediaEngine : public IUnknown
     {
@@ -1710,6 +1718,7 @@ EXTERN_C const IID IID_IMFMediaEngineEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("83015ead-b1e6-40d0-a98a-37145ffe1ad1")
     IMFMediaEngineEx : public IMFMediaEngine
     {
@@ -2757,6 +2766,7 @@ EXTERN_C const IID IID_IMFMediaEngineExtension;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("2f69d622-20b5-41e9-afdf-89ced1dda04e")
     IMFMediaEngineExtension : public IUnknown
     {
@@ -2930,6 +2940,7 @@ EXTERN_C const IID IID_IMFMediaEngineProtectedContent;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("9f8021e8-9c8c-487e-bb5c-79aa4779938c")
     IMFMediaEngineProtectedContent : public IUnknown
     {
@@ -3351,6 +3362,7 @@ EXTERN_C const IID IID_IMFMediaSourceExtensionNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("a7901327-05dd-4469-a7b7-0e01979e361d")
     IMFMediaSourceExtensionNotify : public IUnknown
     {
@@ -3452,6 +3464,7 @@ EXTERN_C const IID IID_IMFBufferListNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("24cd47f7-81d8-4785-adb2-af697a963cd2")
     IMFBufferListNotify : public IUnknown
     {
@@ -3544,6 +3557,7 @@ EXTERN_C const IID IID_IMFSourceBufferNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("87e47623-2ceb-45d6-9b88-d8520c4dcbbc")
     IMFSourceBufferNotify : public IUnknown
     {
@@ -3667,6 +3681,7 @@ EXTERN_C const IID IID_IMFSourceBuffer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("e2cd3a4b-af25-4d3d-9110-da0e6f8ee877")
     IMFSourceBuffer : public IUnknown
     {
@@ -4010,6 +4025,7 @@ EXTERN_C const IID IID_IMFSourceBufferList;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("249981f8-8325-41f3-b80c-3b9e3aad0cbe")
     IMFSourceBufferList : public IUnknown
     {
@@ -4131,6 +4147,7 @@ EXTERN_C const IID IID_IMFMediaSourceExtension;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("e467b94e-a713-4562-a802-816a42e9008a")
     IMFMediaSourceExtension : public IUnknown
     {
@@ -4440,6 +4457,7 @@ EXTERN_C const IID IID_IMFMediaEngineEME;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("50dc93e4-ba4f-4275-ae66-83e836e57469")
     IMFMediaEngineEME : public IUnknown
     {
@@ -4540,6 +4558,7 @@ EXTERN_C const IID IID_IMFMediaEngineSrcElementsEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("654a6bb3-e1a3-424a-9908-53a43a0dfda0")
     IMFMediaEngineSrcElementsEx : public IMFMediaEngineSrcElements
     {
@@ -4727,6 +4746,7 @@ EXTERN_C const IID IID_IMFMediaEngineNeedKeyNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("46a30204-a696-4b18-8804-246b8f031bb1")
     IMFMediaEngineNeedKeyNotify : public IUnknown
     {
@@ -4818,6 +4838,7 @@ EXTERN_C const IID IID_IMFMediaKeys;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("5cb31c05-61ff-418f-afda-caaf41421a38")
     IMFMediaKeys : public IUnknown
     {
@@ -4983,6 +5004,7 @@ EXTERN_C const IID IID_IMFMediaKeySession;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("24fa67d5-d1d0-4dc5-995c-c0efdc191fb5")
     IMFMediaKeySession : public IUnknown
     {
@@ -5126,6 +5148,7 @@ EXTERN_C const IID IID_IMFMediaKeySessionNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("6a0083f9-8947-4c1d-9ce0-cdee22b23135")
     IMFMediaKeySessionNotify : public IUnknown
     {
@@ -5247,6 +5270,7 @@ EXTERN_C const IID IID_IMFCdmSuspendNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("7a5645d2-43bd-47fd-87b7-dcd24cc7d692")
     IMFCdmSuspendNotify : public IUnknown
     {
@@ -5478,6 +5502,7 @@ EXTERN_C const IID IID_IMFMediaEngineOPMInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("765763e6-6c01-4b01-bb0f-b829f60ed28c")
     IMFMediaEngineOPMInfo : public IUnknown
     {
@@ -5663,6 +5688,7 @@ EXTERN_C const IID IID_IMFMediaEngineClassFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("4D645ACE-26AA-4688-9BE1-DF3516990B93")
     IMFMediaEngineClassFactory : public IUnknown
     {
@@ -5796,6 +5822,7 @@ EXTERN_C const IID IID_IMFMediaEngineClassFactoryEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("c56156c6-ea5b-48a5-9df8-fbe035d0929e")
     IMFMediaEngineClassFactoryEx : public IMFMediaEngineClassFactory
     {
@@ -5965,6 +5992,7 @@ EXTERN_C const IID IID_IMFMediaEngineClassFactory2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("09083cef-867f-4bf6-8776-dee3a7b42fca")
     IMFMediaEngineClassFactory2 : public IUnknown
     {
@@ -6159,6 +6187,7 @@ EXTERN_C const IID IID_IMFMediaEngineSupportsSourceTransfer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("a724b056-1b2e-4642-a6f3-db9420c52908")
     IMFMediaEngineSupportsSourceTransfer : public IUnknown
     {
@@ -6561,6 +6590,7 @@ EXTERN_C const IID IID_IMFTimedText;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("1f2a94c9-a3df-430d-9d0f-acd85ddc29af")
     IMFTimedText : public IUnknown
     {
@@ -6869,6 +6899,7 @@ EXTERN_C const IID IID_IMFTimedTextNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("df6b87b6-ce12-45db-aba7-432fe054e57d")
     IMFTimedTextNotify : public IUnknown
     {
@@ -7050,6 +7081,7 @@ EXTERN_C const IID IID_IMFTimedTextTrack;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("8822c32d-654e-4233-bf21-d7f2e67d30d4")
     IMFTimedTextTrack : public IUnknown
     {
@@ -7265,6 +7297,7 @@ EXTERN_C const IID IID_IMFTimedTextTrackList;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("23ff334c-442c-445f-bccc-edc438aa11e2")
     IMFTimedTextTrackList : public IUnknown
     {
@@ -7382,6 +7415,7 @@ EXTERN_C const IID IID_IMFTimedTextCue;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("1e560447-9a2b-43e1-a94c-b0aaabfbfbc9")
     IMFTimedTextCue : public IUnknown
     {
@@ -7579,6 +7613,7 @@ EXTERN_C const IID IID_IMFTimedTextFormattedText;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("e13af3c1-4d47-4354-b1f5-e83ae0ecae60")
     IMFTimedTextFormattedText : public IUnknown
     {
@@ -7700,6 +7735,7 @@ EXTERN_C const IID IID_IMFTimedTextStyle;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("09b2455d-b834-4f01-a347-9052e21c450e")
     IMFTimedTextStyle : public IUnknown
     {
@@ -7955,6 +7991,7 @@ EXTERN_C const IID IID_IMFTimedTextRegion;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("c8d22afc-bc47-4bdf-9b04-787e49ce3f58")
     IMFTimedTextRegion : public IUnknown
     {
@@ -8221,6 +8258,7 @@ EXTERN_C const IID IID_IMFTimedTextBinary;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("4ae3a412-0545-43c4-bf6f-6b97a5c6c432")
     IMFTimedTextBinary : public IUnknown
     {
