diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mfobjects.h b/generation/WinSDK/RecompiledIdlHeaders/um/mfobjects.h
index 2ff32920..734aa3d2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mfobjects.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mfobjects.h
@@ -33,6 +33,8 @@
 #ifndef __mfobjects_h__
 #define __mfobjects_h__
 
+#include <win32metadata_annotations.h>
+
 #if defined(_MSC_VER) && (_MSC_VER >= 1020)
 #pragma once
 #endif
@@ -307,6 +309,7 @@ EXTERN_C const IID IID_IMFAttributes;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2cd2d921-c447-44a7-a13c-4adabfc247e3")
     IMFAttributes : public IUnknown
     {
@@ -816,10 +819,12 @@ enum MF_ATTRIBUTE_SERIALIZE_OPTIONS
     {
         MF_ATTRIBUTE_SERIALIZE_UNKNOWN_BYREF	= 0x1
     } ;
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFSerializeAttributesToStream(
   IMFAttributes * pAttr, 
   DWORD dwOptions, 
   IStream * pStm);
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI MFDeserializeAttributesFromStream(
   IMFAttributes * pAttr, 
   DWORD dwOptions, 
@@ -840,6 +845,7 @@ EXTERN_C const IID IID_IMFMediaBuffer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("045FA593-8799-42b8-BC8D-8968C6453507")
     IMFMediaBuffer : public IUnknown
     {
@@ -983,6 +989,7 @@ EXTERN_C const IID IID_IMFSample;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("c40a00f2-b93a-4d80-ae8c-5a1c634f58e4")
     IMFSample : public IMFAttributes
     {
@@ -1534,6 +1541,7 @@ EXTERN_C const IID IID_IMF2DBuffer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7DC9D5F9-9ED9-44ec-9BBF-0600BB589FBB")
     IMF2DBuffer : public IUnknown
     {
@@ -1729,6 +1737,7 @@ EXTERN_C const IID IID_IMF2DBuffer2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("33ae5ea6-4316-436f-8ddd-d73d22f829ec")
     IMF2DBuffer2 : public IMF2DBuffer
     {
@@ -1915,6 +1924,7 @@ EXTERN_C const IID IID_IMFDXGIBuffer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("e7174cfa-1c9e-48b1-8866-626226bfc258")
     IMFDXGIBuffer : public IUnknown
     {
@@ -2215,6 +2225,7 @@ EXTERN_C const IID IID_IMFMediaType;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("44ae0fa8-ea31-4109-8d2e-4cae4997c555")
     IMFMediaType : public IMFAttributes
     {
@@ -2678,6 +2689,7 @@ EXTERN_C const IID IID_IMFAudioMediaType;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("26a0adc3-ce26-4672-9304-69552edd3faf")
     IMFAudioMediaType : public IMFMediaType
     {
@@ -3447,6 +3459,7 @@ EXTERN_C const IID IID_IMFVideoMediaType;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("b99f381f-a8f9-47a2-a5af-ca3a225a3890")
     IMFVideoMediaType : public IMFMediaType
     {
@@ -3910,6 +3923,7 @@ EXTERN_C const IID IID_IMFAsyncResult;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("ac6b7889-0740-4d51-8619-905994a55cc6")
     IMFAsyncResult : public IUnknown
     {
@@ -4041,6 +4055,7 @@ EXTERN_C const IID IID_IMFAsyncCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("a27003cf-2354-4f2a-8d6a-ab7cff15437e")
     IMFAsyncCallback : public IUnknown
     {
@@ -4157,6 +4172,7 @@ EXTERN_C const IID IID_IMFAsyncCallbackLogging;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("c7a4dca1-f5f0-47b6-b92b-bf0106d25791")
     IMFAsyncCallbackLogging : public IMFAsyncCallback
     {
@@ -4422,6 +4438,7 @@ EXTERN_C const IID IID_IMFMediaEvent;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DF598932-F10C-4E39-BBA2-C308F101DAA3")
     IMFMediaEvent : public IMFAttributes
     {
@@ -4837,6 +4854,19 @@ EXTERN_C const IID IID_IMFMediaEvent;
 
 #define MF_EVENT_FLAG_NO_WAIT 0x00000001
 
+#ifdef WIN32METADATA
+#pragma push_macro("MF_EVENT_FLAG_NO_WAIT")
+#undef MF_EVENT_FLAG_NO_WAIT
+enum class MEDIA_EVENT_GENERATOR_GET_EVENT_FLAGS : int
+{
+    MF_EVENT_FLAG_NONE = 0,
+    MF_EVENT_FLAG_NO_WAIT = 0x00000001,
+};
+#pragma pop_macro("MF_EVENT_FLAG_NO_WAIT")
+#else
+typedef DWORD MEDIA_EVENT_GENERATOR_GET_EVENT_FLAGS;
+#endif
+
 
 
 extern RPC_IF_HANDLE __MIDL_itf_mfobjects_0000_0014_v0_0_c_ifspec;
@@ -4853,13 +4883,14 @@ EXTERN_C const IID IID_IMFMediaEventGenerator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2CD0BD52-BCD5-4B89-B62C-EADC0C031E7D")
     IMFMediaEventGenerator : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetEvent( 
             /* [annotation][in] */ 
-            _In_  DWORD dwFlags,
+            _In_  MEDIA_EVENT_GENERATOR_GET_EVENT_FLAGS dwFlags,
             /* [annotation][out] */ 
             _Out_  IMFMediaEvent **ppEvent) = 0;
         
@@ -4914,7 +4945,7 @@ EXTERN_C const IID IID_IMFMediaEventGenerator;
         HRESULT ( STDMETHODCALLTYPE *GetEvent )( 
             __RPC__in IMFMediaEventGenerator * This,
             /* [annotation][in] */ 
-            _In_  DWORD dwFlags,
+            _In_  MEDIA_EVENT_GENERATOR_GET_EVENT_FLAGS dwFlags,
             /* [annotation][out] */ 
             _Out_  IMFMediaEvent **ppEvent);
         
@@ -5045,6 +5076,7 @@ EXTERN_C const IID IID_IMFRemoteAsyncCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("a27003d0-2354-4f2a-8d6a-ab7cff15437e")
     IMFRemoteAsyncCallback : public IUnknown
     {
@@ -5155,6 +5187,7 @@ EXTERN_C const IID IID_IMFByteStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("ad4c1b00-4bf7-422f-9175-756693d9130d")
     IMFByteStream : public IUnknown
     {
@@ -5575,6 +5608,7 @@ EXTERN_C const IID IID_IMFByteStreamProxyClassFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("a6b43f84-5c0a-42e8-a44d-b1857a76992f")
     IMFByteStreamProxyClassFactory : public IUnknown
     {
@@ -5712,6 +5746,7 @@ EXTERN_C const IID IID_IMFSampleOutputStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("8feed468-6f7e-440d-869a-49bdd283ad0d")
     IMFSampleOutputStream : public IUnknown
     {
@@ -5829,6 +5864,7 @@ EXTERN_C const IID IID_IMFCollection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("5BC8A76B-869A-46a3-9B03-FA218A66AEBE")
     IMFCollection : public IUnknown
     {
@@ -5989,6 +6025,7 @@ EXTERN_C const IID IID_IMFMediaEventQueue;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("36f846fc-2256-48b6-b58e-e2b638316581")
     IMFMediaEventQueue : public IUnknown
     {
@@ -6186,6 +6223,7 @@ EXTERN_C const IID IID_IMFActivate;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7FEE9E9A-4A89-47a6-899C-B6A53A70FB67")
     IMFActivate : public IMFAttributes
     {
@@ -6612,6 +6650,7 @@ EXTERN_C const IID IID_IMFPluginControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5c6c44bf-1db6-435b-9249-e8cd10fdec96")
     IMFPluginControl : public IUnknown
     {
@@ -6807,6 +6846,7 @@ EXTERN_C const IID IID_IMFPluginControl2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("C6982083-3DDC-45CB-AF5E-0F7A8CE4DE77")
     IMFPluginControl2 : public IMFPluginControl
     {
@@ -6974,6 +7014,7 @@ EXTERN_C const IID IID_IMFDXGIDeviceManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("eb533d5d-2db6-40f8-97a9-494692014f07")
     IMFDXGIDeviceManager : public IUnknown
     {
@@ -7197,6 +7238,7 @@ EXTERN_C const IID IID_IMFMuxStreamAttributesManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("CE8BD576-E440-43B3-BE34-1E53F565F7E8")
     IMFMuxStreamAttributesManager : public IUnknown
     {
@@ -7301,6 +7343,7 @@ EXTERN_C const IID IID_IMFMuxStreamMediaTypeManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("505A2C72-42F7-4690-AEAB-8F513D0FFDB8")
     IMFMuxStreamMediaTypeManager : public IUnknown
     {
@@ -7461,6 +7504,7 @@ EXTERN_C const IID IID_IMFMuxStreamSampleManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("74ABBC19-B1CC-4E41-BB8B-9D9B86A8F6CA")
     IMFMuxStreamSampleManager : public IUnknown
     {
@@ -7673,14 +7717,22 @@ unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal(  __RPC__in unsigned long *,
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
 void                      __RPC_USER  LPSAFEARRAY_UserFree(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
 /* [local] */ HRESULT STDMETHODCALLTYPE IMFMediaEventGenerator_BeginGetEvent_Proxy( 
@@ -7797,4 +7849,3 @@ void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsi
 
 #endif
 
-
