diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MgmtAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/MgmtAPI.h
index 61d8256b..874f1db3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MgmtAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MgmtAPI.h
@@ -19,6 +19,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -74,6 +77,8 @@ typedef PVOID LPSNMP_MGR_SESSION;
 //                                                                           //
 ///////////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 LPSNMP_MGR_SESSION
 SNMP_FUNC_TYPE
 SnmpMgrOpen(
@@ -83,6 +88,8 @@ SnmpMgrOpen(
     IN INT   nRetries                       // Comm time-out/retry count
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrCtl(
@@ -95,12 +102,15 @@ SnmpMgrCtl(
     LPDWORD            lpcbBytesReturned    // space used from lpvOutBuffer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrClose(
     IN LPSNMP_MGR_SESSION session           // SNMP session pointer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 SNMPAPI
 SNMP_FUNC_TYPE
 SnmpMgrRequest(
@@ -111,6 +121,7 @@ SnmpMgrRequest(
        OUT AsnInteger         *errorIndex        // Result error index
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrStrToOid(
@@ -118,6 +129,7 @@ SnmpMgrStrToOid(
     OUT AsnObjectIdentifier *oid            // OID internal representation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrOidToStr(
@@ -125,12 +137,15 @@ SnmpMgrOidToStr(
     _Outptr_opt_ LPSTR   *string         // OID string representation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrTrapListen(
     OUT HANDLE *phTrapAvailable             // Event indicating trap available
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrGetTrap(
@@ -142,6 +157,7 @@ SnmpMgrGetTrap(
     OUT RFC1157VarBindList  *variableBindings    // Variable bindings
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 SNMP_FUNC_TYPE
 SnmpMgrGetTrapEx(
