diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mi.h b/generation/WinSDK/RecompiledIdlHeaders/um/mi.h
index 61d31af9..29803910 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mi.h
@@ -14,6 +14,7 @@
 */
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or WinMgmt Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WINMGMT)
@@ -1682,6 +1683,7 @@ struct _MI_Server
  * return: MI_RESULT_OK, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_Result MI_CALL MI_Server_GetVersion(MI_Uint32* version);
 
 /**
@@ -1697,6 +1699,7 @@ MI_Result MI_CALL MI_Server_GetVersion(MI_Uint32* version);
  * return: MI_RESULT_OK, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_Result MI_CALL MI_Server_GetSystemName(const MI_Char** systemName);
 
 /*
@@ -1741,6 +1744,7 @@ struct _MI_Filter
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Filter_Evaluate(
         _In_ const MI_Filter* self,
         _In_ const MI_Instance* instance,
@@ -1767,6 +1771,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Filter_Evaluate(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Filter_GetExpression(
         _In_ const MI_Filter* self,
         _Outptr_result_maybenull_z_ const MI_Char** queryLang,
@@ -1850,6 +1855,7 @@ struct _MI_PropertySet
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_GetElementCount(
     _In_ const MI_PropertySet* self,
     _Out_ MI_Uint32* count)
@@ -1874,6 +1880,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_GetElementCount(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_ContainsElement(
     _In_ const MI_PropertySet* self,
     _In_z_ const MI_Char* name,
@@ -1898,6 +1905,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_ContainsElement(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_ALREADY_EXISTS, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_AddElement(
     _Inout_ MI_PropertySet* self,
     _In_z_ const MI_Char* name)
@@ -1923,6 +1931,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_AddElement(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_GetElementAt(
     _In_ const MI_PropertySet* self,
     MI_Uint32 index,
@@ -1948,6 +1957,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_GetElementAt(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Clear(
     _Inout_ MI_PropertySet* self)
 {
@@ -1970,6 +1980,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Clear(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Destruct(
     _Inout_ MI_PropertySet* self)
 {
@@ -1992,6 +2003,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Destruct(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Delete(
     _Inout_ MI_PropertySet* self)
 {
@@ -2017,6 +2029,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Delete(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_PropertySet_Clone(
     _In_ const MI_PropertySet* self,
     _Outptr_ MI_PropertySet** newPropertySet)
@@ -3256,6 +3269,7 @@ struct _MI_Instance
  * See also: MI_Instance_Delete()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Clone(
     _In_ const MI_Instance* self,
     _Outptr_ MI_Instance** newInstance)
@@ -3283,6 +3297,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Clone(
  * See also: MI_Instance_Delete()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Destruct(
     _Inout_ MI_Instance* self)
 {
@@ -3306,6 +3321,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Destruct(
  *
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Delete(
     _Inout_ MI_Instance* self)
 {
@@ -3332,6 +3348,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Delete(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_IsA(
     _In_ const MI_Instance* self,
     _In_ const MI_ClassDecl* classDecl,
@@ -3356,6 +3373,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_IsA(
  * return: MI_RESULT_OK, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetClassName(
     _In_ const MI_Instance* self,
     _Outptr_result_maybenull_z_ const MI_Char** className)
@@ -3386,6 +3404,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetClassName(
  *         MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetNameSpace(
     _Inout_ MI_Instance* self,
     _In_z_ const MI_Char* nameSpace)
@@ -3409,6 +3428,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetNameSpace(
  * return: MI_RESULT_OK, MI_RESULT_INVALID_PARAMETER
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetNameSpace(
     _In_ const MI_Instance* self,
     _Outptr_result_maybenull_z_ const MI_Char** nameSpace)
@@ -3434,6 +3454,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetNameSpace(
  * See also: MI_Instance_GetAt(), MI_Instance_SetAt()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElementCount(
     _In_ const MI_Instance* self,
     _Out_ MI_Uint32* count)
@@ -3466,6 +3487,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElementCount(
  *
  * return: MI_RESULT_OK, MI_RESULT_INVALID_PARAMETER
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_AddElement(
     _Inout_ MI_Instance* self,
     _In_z_ const MI_Char* name,
@@ -3501,6 +3523,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_AddElement(
  * See also: MI_Instance_GetElementCount()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetElementAt(
     _Inout_ MI_Instance* self,
     MI_Uint32 index,
@@ -3539,6 +3562,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetElementAt(
  * See also: MI_Instance_GetElement()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetElement(
     _Inout_ MI_Instance* self,
     _In_z_ const MI_Char* name,
@@ -3572,6 +3596,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetElement(
  *     MI_RESULT_NOT_FOUND, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElement(
     _In_ const MI_Instance* self,
     _In_z_ const MI_Char* name,
@@ -3608,6 +3633,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElement(
  * See also: MI_Instance_GetElementCount()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElementAt(
     _In_ const MI_Instance* self,
     MI_Uint32 index,
@@ -3636,6 +3662,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetElementAt(
  *
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_ClearElement(
     _Inout_ MI_Instance* self,
     _In_z_ const MI_Char* name)
@@ -3661,6 +3688,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_ClearElement(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_ClearElementAt(
     _Inout_ MI_Instance* self,
     MI_Uint32 index)
@@ -3687,6 +3715,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_ClearElementAt(
  * return: MI_RESULT_OK, MI_RESULT_FAILED, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetServerName(
     _In_ const MI_Instance* self,
     _Outptr_result_maybenull_z_ const MI_Char** name)
@@ -3701,6 +3730,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetServerName(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetServerName(
     _Inout_ MI_Instance* self,
     _In_z_ const MI_Char* name)
@@ -3715,6 +3745,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_SetServerName(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetClass(
     _In_ const MI_Instance* self,
     _Outptr_ MI_Class** instanceClass)
@@ -3729,6 +3760,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_GetClass(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Instance_Normalize(
     _Inout_ MI_Instance** self)
 {
@@ -4039,6 +4071,7 @@ struct _MI_Context
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostResult(
     _In_ MI_Context* context,
     MI_Result result)
@@ -4062,6 +4095,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostResult(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostCimError(
     _In_ MI_Context* context,
     _In_ const MI_Instance *error)
@@ -4087,6 +4121,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostCimError(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostError(
     _In_ MI_Context* context,
     MI_Uint32 resultCode,
@@ -4115,6 +4150,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostError(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostInstance(
     _In_ MI_Context* context,
     _In_ const MI_Instance* instance)
@@ -4142,6 +4178,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostInstance(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostIndication(
     _In_ MI_Context* context,
     _In_ const MI_Instance* indication,
@@ -4172,6 +4209,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PostIndication(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ConstructInstance(
     _In_ MI_Context* context,
     _In_ const MI_ClassDecl* classDecl,
@@ -4200,6 +4238,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ConstructInstance(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ConstructParameters(
     _In_ MI_Context* context,
     _In_ const MI_MethodDecl* methodDecl,
@@ -4227,6 +4266,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ConstructParameters(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewInstance(
     _In_ MI_Context* context,
     _In_ const MI_ClassDecl* classDecl,
@@ -4255,6 +4295,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewInstance(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewDynamicInstance(
     _In_ MI_Context* context,
     _In_ const MI_Char* className,
@@ -4283,6 +4324,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewDynamicInstance(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewParameters(
     _In_ MI_Context* context,
     _In_ const MI_MethodDecl* methodDecl,
@@ -4310,6 +4352,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_NewParameters(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_Canceled(
     _In_ const MI_Context* context,
     _Out_ MI_Boolean* flag)
@@ -4334,6 +4377,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_Canceled(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetLocale(
     _In_ const MI_Context* context,
     MI_LocaleType localeType,
@@ -4365,6 +4409,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetLocale(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RegisterCancel(
     _In_ MI_Context* context,
     _In_ MI_CancelCallback callback,
@@ -4390,6 +4435,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RegisterCancel(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RequestUnload(
     _In_ MI_Context* context)
 {
@@ -4415,6 +4461,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RequestUnload(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RefuseUnload(
     _In_ MI_Context* context)
 {
@@ -4441,6 +4488,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_RefuseUnload(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetLocalSession(
     _In_ const MI_Context* context,
     _Out_ MI_Session* session)
@@ -4492,6 +4540,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetLocalSession(
  *   WBEM_RIGHT_SUBSCRIBE permission, the SDDL must contain this flag, please
  *   see http://msdn.microsoft.com/en-us/library/aa390415.aspx for details.
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_SetStringOption(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* name,
@@ -4517,6 +4566,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_SetStringOption(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetStringOption(
     _In_  MI_Context* context,
     _In_z_ const MI_Char* name,
@@ -4542,6 +4592,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetStringOption(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetNumberOption(
     _In_  MI_Context* context,
     _In_z_ const MI_Char* name,
@@ -4567,6 +4618,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetNumberOption(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOption(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* name,
@@ -4593,6 +4645,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOption(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOptionCount(
     _In_  MI_Context* context,
     _Out_opt_ MI_Uint32* count)
@@ -4617,6 +4670,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOptionCount(
  * return: MI_RESULT_OK, MI_RESULT_FAILED
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOptionAt(
     _In_  MI_Context* context,
     _In_ MI_Uint32 index,
@@ -4645,6 +4699,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_GetCustomOptionAt(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ShouldProcess(
     _In_ MI_Context* context,
     _In_z_ const MI_Char *target,
@@ -4671,6 +4726,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ShouldProcess(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ShouldContinue(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* message,
@@ -4698,6 +4754,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_ShouldContinue(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PromptUser(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* message,
@@ -4727,6 +4784,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_PromptUser(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteError(
     _In_ MI_Context* context,
     MI_Uint32 resultCode,
@@ -4754,6 +4812,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteError(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteCimError(
     _In_ MI_Context* context,
     _In_ const MI_Instance *error,
@@ -4782,6 +4841,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteCimError(
  * return: MI_TRUE or MI_FALSE.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteMessage(
     _In_ MI_Context* context,
     MI_Uint32 channel,
@@ -4810,6 +4870,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteMessage(
  * return: MI_TRUE or MI_FALSE.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteProgress(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* activity,
@@ -4844,6 +4905,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteProgress(
  * return: MI_RESULT_OK, MI_RESULT_FAILED.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteStreamParameter(
     _In_ MI_Context* self,
     _In_z_ const MI_Char* name,
@@ -4870,6 +4932,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteStreamParameter(
  * return: MI_TRUE or MI_FALSE.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteWarning(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* message)
@@ -4893,6 +4956,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteWarning(
  * return: MI_TRUE or MI_FALSE.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteVerbose(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* message)
@@ -4916,6 +4980,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteVerbose(
  * return: MI_TRUE or MI_FALSE.
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Context_WriteDebug(
     _In_ MI_Context* context,
     _In_z_ const MI_Char* message)
@@ -6712,6 +6777,7 @@ __declspec(dllimport) const MI_ClientFT_V1 *mi_clientFT_V1;
 ** Return: MI_RESULT_OK success, other errors indicate specific failure
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_Result MI_MAIN_CALL MI_Application_InitializeV1(
              MI_Uint32 flags,
     _In_opt_z_ const MI_Char *applicationID,
@@ -6734,6 +6800,7 @@ MI_Result MI_MAIN_CALL MI_Application_InitializeV1(
 ** application: Handle returned from MI_Application_Initialize.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_Close(
     _Inout_ MI_Application *application)
 {
@@ -6764,6 +6831,7 @@ MI_INLINE MI_Result MI_Application_Close(
 ** Return: MI_RESULT_OK success, other errors indicate specific failure
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewInstance(
     _In_     MI_Application *application,
     _In_z_ const MI_Char *className,
@@ -6783,6 +6851,7 @@ MI_INLINE MI_Result MI_Application_NewInstance(
         return MI_RESULT_INVALID_PARAMETER;
     }
 }
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewInstanceFromClass(
     _In_     MI_Application *application,
     _In_z_ const MI_Char *className,
@@ -6820,6 +6889,7 @@ MI_INLINE MI_Result MI_Application_NewInstanceFromClass(
 ** Return: MI_RESULT_OK success, other errors indicate specific failure
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows10.0.10240)
 MI_INLINE MI_Result MI_Application_NewClass(
         _In_ MI_Application *application,
         _In_ const MI_ClassDecl* classDecl,
@@ -6842,6 +6912,7 @@ MI_INLINE MI_Result MI_Application_NewClass(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewParameterSet(
     _In_     MI_Application *application,
     _In_opt_ const MI_ClassDecl *classRTTI,
@@ -6876,6 +6947,7 @@ MI_INLINE MI_Result MI_Application_NewParameterSet(
 ** Return: MI_RESULT_OK success, other errors indicate specific failure
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewDestinationOptions(
     _In_  MI_Application *application,
     _Out_ MI_DestinationOptions *options)
@@ -6909,6 +6981,7 @@ MI_INLINE MI_Result MI_Application_NewDestinationOptions(
 ** Return: MI_RESULT_OK success, other errors indicate specific failure
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewOperationOptions(
     _In_  MI_Application *application,
           MI_Boolean mustUnderstand,
@@ -6928,6 +7001,7 @@ MI_INLINE MI_Result MI_Application_NewOperationOptions(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewSubscriptionDeliveryOptions(
     _In_  MI_Application *application,
     _In_  MI_SubscriptionDeliveryType deliveryType,
@@ -6958,6 +7032,7 @@ MI_INLINE MI_Result MI_Application_NewSubscriptionDeliveryOptions(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewSession(
     _In_     MI_Application *application,
     _In_opt_z_ const MI_Char *protocol,
@@ -6988,6 +7063,7 @@ MI_INLINE MI_Result MI_Application_NewSession(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewHostedProvider(
     _In_  MI_Application *application,
     _In_z_  const MI_Char *namespaceName,
@@ -7010,6 +7086,7 @@ MI_INLINE MI_Result MI_Application_NewHostedProvider(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewSerializer(
     _Inout_ MI_Application *application,
     MI_Uint32 flags,
@@ -7030,6 +7107,7 @@ MI_INLINE MI_Result MI_Application_NewSerializer(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Application_NewDeserializer(
     _Inout_ MI_Application *application,
     MI_Uint32 flags,
@@ -7059,6 +7137,7 @@ MI_INLINE MI_Result MI_Application_NewDeserializer(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_HostedProvider_Close(
     _Inout_ MI_HostedProvider *hostedProvider)
 {
@@ -7079,6 +7158,7 @@ MI_INLINE MI_Result MI_HostedProvider_Close(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_HostedProvider_GetApplication(
     _In_  MI_HostedProvider *hostedProvider,
     _Out_ MI_Application *application)
@@ -7110,6 +7190,7 @@ MI_INLINE MI_Result MI_HostedProvider_GetApplication(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Session_Close(
     _Inout_     MI_Session *session,
     _In_opt_ void *completionContext,
@@ -7137,6 +7218,7 @@ MI_INLINE MI_Result MI_Session_Close(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Session_GetApplication(
     _In_  MI_Session *session,
     _Out_ MI_Application *application)
@@ -7165,6 +7247,7 @@ MI_INLINE MI_Result MI_Session_GetApplication(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_GetInstance(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7204,6 +7287,7 @@ MI_INLINE void MI_Session_GetInstance(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_ModifyInstance(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7243,6 +7327,7 @@ MI_INLINE void MI_Session_ModifyInstance(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_CreateInstance(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7281,6 +7366,7 @@ MI_INLINE void MI_Session_CreateInstance(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_DeleteInstance(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7320,6 +7406,7 @@ MI_INLINE void MI_Session_DeleteInstance(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_Invoke(
     _In_       MI_Session *session,
                MI_Uint32 flags,
@@ -7364,6 +7451,7 @@ MI_INLINE void MI_Session_Invoke(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_EnumerateInstances(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7391,6 +7479,7 @@ MI_INLINE void MI_Session_EnumerateInstances(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_QueryInstances(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7433,6 +7522,7 @@ MI_INLINE void MI_Session_QueryInstances(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_AssociatorInstances(
     _In_       MI_Session *session,
                MI_Uint32 flags,
@@ -7479,6 +7569,7 @@ MI_INLINE void MI_Session_AssociatorInstances(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_ReferenceInstances(
     _In_     MI_Session *session,
              MI_Uint32 flags,
@@ -7524,6 +7615,7 @@ MI_INLINE void MI_Session_ReferenceInstances(
 **=============================================================================
 */
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_Subscribe(
     _In_       MI_Session *session,
                MI_Uint32 flags,
@@ -7559,6 +7651,7 @@ MI_INLINE void MI_Session_Subscribe(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_GetClass(
     _In_       MI_Session *session,
                MI_Uint32 flags,
@@ -7592,6 +7685,7 @@ MI_INLINE void MI_Session_GetClass(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_EnumerateClasses(
     _In_       MI_Session *session,
                MI_Uint32 flags,
@@ -7619,6 +7713,7 @@ MI_INLINE void MI_Session_EnumerateClasses(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_Session_TestConnection(
         _In_     MI_Session *session,
                  MI_Uint32 flags,
@@ -7658,6 +7753,7 @@ MI_INLINE void MI_Session_TestConnection(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_GetInstance(
     _In_        MI_Operation *operation,
     _Outptr_result_maybenull_       const MI_Instance **instance,
@@ -7689,6 +7785,7 @@ MI_INLINE MI_Result MI_Operation_GetInstance(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_GetIndication(
     _In_        MI_Operation *operation,
     _Outptr_result_maybenull_       const MI_Instance **instance,
@@ -7715,6 +7812,7 @@ MI_INLINE MI_Result MI_Operation_GetIndication(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_GetClass(
     _In_        MI_Operation *operation,
     _Outptr_result_maybenull_       const MI_Class **classResult,
@@ -7745,6 +7843,7 @@ MI_INLINE MI_Result MI_Operation_GetClass(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_Close(
     _Inout_ MI_Operation *operation)
 {
@@ -7764,6 +7863,7 @@ MI_INLINE MI_Result MI_Operation_Close(
 ** returns MI_FALSE.  The operation also must still be closed with MI_Operation_Close.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_Cancel(
     _Inout_ MI_Operation *operation,
          MI_CancellationReason reason)
@@ -7782,6 +7882,7 @@ MI_INLINE MI_Result MI_Operation_Cancel(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Operation_GetSession(
     _In_  MI_Operation *operation,
     _Out_ MI_Session *session)
@@ -7805,6 +7906,7 @@ MI_INLINE MI_Result MI_Operation_GetSession(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_DestinationOptions_Delete(
     _Inout_ MI_DestinationOptions *options)
 {
@@ -7821,6 +7923,7 @@ MI_INLINE void MI_DestinationOptions_Delete(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetTimeout(
     _Inout_ MI_DestinationOptions *options,
     _In_ const MI_Interval *timeout)
@@ -7842,6 +7945,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetTimeout(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetTimeout(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Interval *timeout)
@@ -7864,6 +7968,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetTimeout(
 **=============================================================================
 */
 /* Check/skip CA check when doing SSL, default TRUE, check*/
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetCertCACheck(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean check)
@@ -7888,6 +7993,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetCertCACheck(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCertCACheck(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *check)
@@ -7915,6 +8021,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCertCACheck(
 **=============================================================================
 */
 /* Check/skip CN check when doing SSL, default TRUE, check*/
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetCertCNCheck(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean check)
@@ -7939,6 +8046,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetCertCNCheck(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCertCNCheck(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *check)
@@ -7965,6 +8073,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCertCNCheck(
 **=============================================================================
 */
 /* Check/skip cert revocation check when doing SSL, default TRUE, check*/
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetCertRevocationCheck(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean check)
@@ -7989,6 +8098,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetCertRevocationCheck(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCertRevocationCheck(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *check)
@@ -8015,6 +8125,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCertRevocationCheck(
 ** Ensure packet privacy (encryption), default TRUE
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetPacketPrivacy(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean privacy)
@@ -8039,6 +8150,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetPacketPrivacy(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetPacketPrivacy(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *privacy)
@@ -8065,6 +8177,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetPacketPrivacy(
 **=============================================================================
 */
 /* Ensure packet integrity (sign), default TRUE*/
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetPacketIntegrity(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean integrity)
@@ -8089,6 +8202,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetPacketIntegrity(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetPacketIntegrity(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *integrity)
@@ -8124,6 +8238,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetPacketIntegrity(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetPacketEncoding(
     _Inout_   MI_DestinationOptions *options,
     _In_z_ const MI_Char *encoding)
@@ -8145,6 +8260,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetPacketEncoding(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetPacketEncoding(
     _In_   MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **encoding)
@@ -8178,6 +8294,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetPacketEncoding(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetDataLocale(
     _Inout_   MI_DestinationOptions *options,
     _In_z_ const MI_Char *locale)
@@ -8207,6 +8324,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetDataLocale(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetDataLocale(
     _In_   MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **locale)
@@ -8239,6 +8357,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetDataLocale(
 **    locale:  Valid UI locale string.  Example: en-us
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetUILocale(
     _Inout_   MI_DestinationOptions *options,
     _In_z_ const MI_Char *locale)
@@ -8267,6 +8386,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetUILocale(
 **    locale:  UI locale string has been set.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetUILocale(
     _In_   MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **locale)
@@ -8288,6 +8408,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetUILocale(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetMaxEnvelopeSize(
     _Inout_ MI_DestinationOptions *options,
          MI_Uint32 sizeInKB)
@@ -8309,6 +8430,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetMaxEnvelopeSize(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetMaxEnvelopeSize(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Uint32 *sizeInKB)
@@ -8331,6 +8453,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetMaxEnvelopeSize(
 ** default MI_FALSE = don't, MI_TRUE = do encode, WinRM transport specific
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetEncodePortInSPN(
     _Inout_ MI_DestinationOptions *options,
          MI_Boolean encodePort)
@@ -8353,6 +8476,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetEncodePortInSPN(
 ** default MI_FALSE = don't, MI_TRUE = do encode, WinRM transport specific
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetEncodePortInSPN(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Boolean *encodePort)
@@ -8379,6 +8503,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetEncodePortInSPN(
 **Specific to HTTP/HTTP based transports, WinRM default is /wsman, other transports are transports/protocols specific
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetHttpUrlPrefix(
     _Inout_   MI_DestinationOptions *options,
     _In_z_ const MI_Char *prefix)
@@ -8401,6 +8526,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetHttpUrlPrefix(
 **Specific to HTTP/HTTP based transports, WinRM default is /wsman, other transports are transports/protocols specific
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetHttpUrlPrefix(
     _In_   MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **prefix)
@@ -8423,6 +8549,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetHttpUrlPrefix(
 ** Transport specific, Default port for transport.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetDestinationPort(
     _Inout_ MI_DestinationOptions *options,
          MI_Uint32 port)
@@ -8445,6 +8572,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetDestinationPort(
 ** Transport specific, Default port for transport.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetDestinationPort(
     _In_ MI_DestinationOptions *options,
     _Out_ MI_Uint32 *port)
@@ -8478,6 +8606,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetDestinationPort(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetTransport(
     _Inout_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *transport)
@@ -8499,6 +8628,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetTransport(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetTransport(
     _In_ MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **transport)
@@ -8532,6 +8662,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetTransport(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetProxyType(
     _Inout_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *proxyType)
@@ -8553,6 +8684,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetProxyType(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetProxyType(
     _In_ MI_DestinationOptions *options,
     _Outptr_result_z_ const MI_Char **proxyType)
@@ -8577,6 +8709,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetProxyType(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_AddProxyCredentials(
     _Inout_ MI_DestinationOptions *options,
     _In_ const MI_UserCredentials *credentials)
@@ -8600,6 +8733,7 @@ MI_INLINE MI_Result MI_DestinationOptions_AddProxyCredentials(
 ** can be combined, not all transport handles support dual auth.
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_AddDestinationCredentials(
     _Inout_     MI_DestinationOptions *options,
     _In_ const MI_UserCredentials *credentials)
@@ -8623,6 +8757,7 @@ typedef enum _MI_DestinationOptions_ImpersonationType
     MI_DestinationOptions_ImpersonationType_Delegate = 4
 } MI_DestinationOptions_ImpersonationType;
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetImpersonationType(
     _Inout_ MI_DestinationOptions *options,
     _In_ MI_DestinationOptions_ImpersonationType impersonationType)
@@ -8637,6 +8772,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetImpersonationType(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetImpersonationType(
     _Inout_ MI_DestinationOptions *options,
     _Out_ MI_DestinationOptions_ImpersonationType * impersonationType)
@@ -8662,6 +8798,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetImpersonationType(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetString(
     _Inout_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -8684,6 +8821,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetString(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetString(
     _In_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -8707,6 +8845,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetString(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_SetNumber(
     _Inout_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -8729,6 +8868,7 @@ MI_INLINE MI_Result MI_DestinationOptions_SetNumber(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetNumber(
     _In_ MI_DestinationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -8753,6 +8893,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetNumber(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetOptionCount(
     _In_   MI_DestinationOptions *options,
     _Out_ MI_Uint32 *count)
@@ -8775,6 +8916,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetOptionCount(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetOptionAt(
     _In_   MI_DestinationOptions *options,
     MI_Uint32 index,
@@ -8800,6 +8942,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetOptionAt(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetOption(
     _In_   MI_DestinationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -8826,6 +8969,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetOption(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsCount(
     _In_   MI_DestinationOptions *options,
     _Out_ MI_Uint32 *count)
@@ -8848,6 +8992,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsCount(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsAt(
     _In_   MI_DestinationOptions *options,
     MI_Uint32 index,
@@ -8873,6 +9018,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsAt(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsPasswordAt(
     _In_   MI_DestinationOptions *options,
     MI_Uint32 index,
@@ -8911,6 +9057,7 @@ MI_INLINE MI_Result MI_DestinationOptions_GetCredentialsPasswordAt(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_DestinationOptions_Clone(
     _In_ const MI_DestinationOptions* self,
     _Out_ MI_DestinationOptions* newDestinationOptions)
@@ -8932,6 +9079,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_DestinationOptions_Clone(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE void MI_OperationOptions_Delete(
     _Inout_ MI_OperationOptions *options)
 {
@@ -8948,6 +9096,7 @@ MI_INLINE void MI_OperationOptions_Delete(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetWriteErrorMode(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_CallbackMode mode)
@@ -8969,6 +9118,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetWriteErrorMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetWriteErrorMode(
     _In_ MI_OperationOptions *options,
     _Out_ MI_CallbackMode *mode)
@@ -8994,6 +9144,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetWriteErrorMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetPromptUserMode(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_CallbackMode mode)
@@ -9018,6 +9169,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetPromptUserMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetPromptUserMode(
     _In_ MI_OperationOptions *options,
     _Out_ MI_CallbackMode *mode)
@@ -9043,6 +9195,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetPromptUserMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetPromptUserRegularMode(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_CallbackMode mode,
@@ -9069,6 +9222,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetPromptUserRegularMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetPromptUserRegularMode(
     _Inout_ MI_OperationOptions *options,
     _Out_ MI_CallbackMode *mode,
@@ -9105,6 +9259,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetPromptUserRegularMode(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetProviderArchitecture(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_ProviderArchitecture architecture,
@@ -9131,6 +9286,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetProviderArchitecture(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetProviderArchitecture(
     _In_ MI_OperationOptions *options,
     _Out_ MI_ProviderArchitecture *architecture,
@@ -9165,6 +9321,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetProviderArchitecture(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_EnableChannel(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_Uint32 channel)
@@ -9186,6 +9343,7 @@ MI_INLINE MI_Result MI_OperationOptions_EnableChannel(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_DisableChannel(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_Uint32 channel)
@@ -9207,6 +9365,7 @@ MI_INLINE MI_Result MI_OperationOptions_DisableChannel(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetEnabledChannels(
     _In_   MI_OperationOptions *options,
     _Out_writes_to_opt_(bufferLength, *channelCount) MI_Uint32 *channels,
@@ -9231,6 +9390,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetEnabledChannels(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetTimeout(
     _Inout_ MI_OperationOptions *options,
     _In_ const MI_Interval *timeout)
@@ -9252,6 +9412,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetTimeout(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetTimeout(
     _In_ MI_OperationOptions *options,
     _Out_ MI_Interval *timeout)
@@ -9273,6 +9434,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetTimeout(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetResourceUriPrefix(
     _Inout_   MI_OperationOptions *options,
     _In_z_ const MI_Char *ruriPrefix)
@@ -9294,6 +9456,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetResourceUriPrefix(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetResourceUriPrefix(
     _In_   MI_OperationOptions *options,
     _Outptr_result_z_ const MI_Char **ruriPrefix)
@@ -9315,6 +9478,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetResourceUriPrefix(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetResourceUri(
     _Inout_   MI_OperationOptions *options,
     _In_z_ const MI_Char *rUri)
@@ -9336,6 +9500,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetResourceUri(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetResourceUri(
     _In_   MI_OperationOptions *options,
     _Outptr_result_z_ const MI_Char **rUri)
@@ -9358,6 +9523,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetResourceUri(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetUseMachineID(
     _Inout_ MI_OperationOptions *options,
     _In_ MI_Boolean machineID)
@@ -9379,6 +9545,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetUseMachineID(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetUseMachineID(
     _In_ MI_OperationOptions *options,
     _Out_ MI_Boolean *machineID)
@@ -9404,6 +9571,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetUseMachineID(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetCustomOption(
     _Inout_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9428,6 +9596,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetCustomOption(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetOptionCount(
     _In_   MI_OperationOptions *options,
     _Out_ MI_Uint32 *count)
@@ -9449,6 +9618,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetOptionCount(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetOptionAt(
     _In_   MI_OperationOptions *options,
     MI_Uint32 index,
@@ -9474,6 +9644,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetOptionAt(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetString(
     _Inout_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9497,6 +9668,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetString(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetString(
     _In_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9521,6 +9693,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetString(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_SetNumber(
     _Inout_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9544,6 +9717,7 @@ MI_INLINE MI_Result MI_OperationOptions_SetNumber(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetNumber(
     _In_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9568,6 +9742,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetNumber(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_OperationOptions_GetOption(
     _In_   MI_OperationOptions *options,
     _In_z_ const MI_Char *optionName,
@@ -9605,6 +9780,7 @@ MI_INLINE MI_Result MI_OperationOptions_GetOption(
 **
 **=============================================================================
 */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_OperationOptions_Clone(
     _In_ const MI_OperationOptions* self,
     _Out_ MI_OperationOptions* newOperationOptions)
@@ -9620,6 +9796,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_OperationOptions_Clone(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetClassName(
         _In_              const MI_Class* self,
         _Outptr_result_maybenull_z_ const MI_Char** className)
@@ -9634,6 +9811,7 @@ MI_INLINE MI_Result MI_Class_GetClassName(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetNameSpace(
         _In_              const MI_Class* self,
         _Outptr_result_maybenull_z_ const MI_Char** nameSpace)
@@ -9649,6 +9827,7 @@ MI_INLINE MI_Result MI_Class_GetNameSpace(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetServerName(
         _In_              const MI_Class* self,
         _Outptr_result_maybenull_z_ const MI_Char** serverName)
@@ -9664,6 +9843,7 @@ MI_INLINE MI_Result MI_Class_GetServerName(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetElementCount(
         _In_  const MI_Class* self,
         _Out_ MI_Uint32* count)
@@ -9679,6 +9859,7 @@ MI_INLINE MI_Result MI_Class_GetElementCount(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetElement(
         _In_      const MI_Class* self,
         _In_z_    const MI_Char* name,
@@ -9701,6 +9882,7 @@ MI_INLINE MI_Result MI_Class_GetElement(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetElementAt(
         _In_              const MI_Class* self,
                           MI_Uint32 index,
@@ -9722,6 +9904,7 @@ MI_INLINE MI_Result MI_Class_GetElementAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetClassQualifierSet(
         _In_            const MI_Class* self,
         _Out_opt_ MI_QualifierSet *qualifierSet
@@ -9737,6 +9920,7 @@ MI_INLINE MI_Result MI_Class_GetClassQualifierSet(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetMethodCount(
         _In_  const MI_Class* self,
         _Out_ MI_Uint32* count)
@@ -9751,6 +9935,7 @@ MI_INLINE MI_Result MI_Class_GetMethodCount(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetMethodAt(
         _In_ const MI_Class *self,
         MI_Uint32 index,
@@ -9768,6 +9953,7 @@ MI_INLINE MI_Result MI_Class_GetMethodAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetMethod(
         _In_ const MI_Class *self,
         _In_z_ const MI_Char *name,
@@ -9785,6 +9971,7 @@ MI_INLINE MI_Result MI_Class_GetMethod(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetParentClassName(
         _In_ const MI_Class *self,
         _Outptr_result_maybenull_z_ const MI_Char **name)
@@ -9799,6 +9986,7 @@ MI_INLINE MI_Result MI_Class_GetParentClassName(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_GetParentClass(
         _In_ const MI_Class *self,
         _Outptr_ MI_Class **parentClass)
@@ -9813,6 +10001,7 @@ MI_INLINE MI_Result MI_Class_GetParentClass(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Class_Delete(
         _Inout_ MI_Class* self)
 {
@@ -9826,6 +10015,7 @@ MI_INLINE MI_Result MI_Class_Delete(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_Class_Clone(
     _In_ const MI_Class* self,
     _Outptr_ MI_Class** newClass)
@@ -9841,6 +10031,7 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_Class_Clone(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_ParameterSet_GetMethodReturnType(
     _In_  const MI_ParameterSet *self,
     _Out_ MI_Type *returnType,
@@ -9856,6 +10047,7 @@ MI_INLINE MI_Result MI_ParameterSet_GetMethodReturnType(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_ParameterSet_GetParameterCount(
     _In_ const MI_ParameterSet *self,
     _Out_ MI_Uint32 *count)
@@ -9870,6 +10062,7 @@ MI_INLINE MI_Result MI_ParameterSet_GetParameterCount(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_ParameterSet_GetParameterAt(
     _In_ const MI_ParameterSet *self,
     MI_Uint32 index,
@@ -9888,6 +10081,7 @@ MI_INLINE MI_Result MI_ParameterSet_GetParameterAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_ParameterSet_GetParameter(
     _In_ const MI_ParameterSet *self,
     _In_z_ const MI_Char *name,
@@ -9906,6 +10100,7 @@ MI_INLINE MI_Result MI_ParameterSet_GetParameter(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_QualifierSet_GetQualifierCount(
     _In_ const MI_QualifierSet *self,
     _Out_ MI_Uint32 *count)
@@ -9920,6 +10115,7 @@ MI_INLINE MI_Result MI_QualifierSet_GetQualifierCount(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_QualifierSet_GetQualifierAt(
     _In_ const MI_QualifierSet *self,
     MI_Uint32 index,
@@ -9938,6 +10134,7 @@ MI_INLINE MI_Result MI_QualifierSet_GetQualifierAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_QualifierSet_GetQualifier(
     _In_ const MI_QualifierSet *self,
     _In_z_ const MI_Char *name,
@@ -9956,6 +10153,7 @@ MI_INLINE MI_Result MI_QualifierSet_GetQualifier(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetMaximumLatency(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   MI_Interval *value)
@@ -9970,6 +10168,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetMaximumLatency(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetMaximumLatency(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Interval *value)
@@ -9984,6 +10183,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetMaximumLatency(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetHeartbeatInterval(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   MI_Interval *value)
@@ -9998,6 +10198,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetHeartbeatInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetHeartbeatInterval(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Interval *value)
@@ -10012,6 +10213,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetHeartbeatInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetExpirationTime(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   MI_Datetime *value)
@@ -10026,6 +10228,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetExpirationTime(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetExpirationTime(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Datetime *value)
@@ -10051,6 +10254,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetExpirationTime(
 #define MI_SUBSCRIBE_BOOKMARK_OLDEST L"MI_SUBSCRIBE_BOOKMARK_OLDEST"
 #define MI_SUBSCRIBE_BOOKMARK_NEWEST L"MI_SUBSCRIBE_BOOKMARK_NEWEST"
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetBookmark(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_   const MI_Char *value)
@@ -10065,6 +10269,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetBookmark(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetBookmark(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Outptr_result_z_ const MI_Char **value)
@@ -10079,6 +10284,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetBookmark(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryDestination(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *value)
@@ -10093,6 +10299,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryDestination(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryDestination(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Outptr_result_z_ const MI_Char **value)
@@ -10107,6 +10314,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryDestination(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryPortNumber(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   MI_Uint32 value)
@@ -10121,6 +10329,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryPortNumber(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryPortNumber(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_   MI_Uint32 *value)
@@ -10135,6 +10344,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryPortNumber(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_AddDeliveryCredentials(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   const MI_UserCredentials *value)
@@ -10149,6 +10359,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_AddDeliveryCredentials(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryRetryInterval(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   const MI_Interval *value)
@@ -10163,6 +10374,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryRetryInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryRetryInterval(
     _In_  MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Interval *value)
@@ -10177,6 +10389,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryRetryInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryRetryAttempts(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_   MI_Uint32 value)
@@ -10191,6 +10404,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDeliveryRetryAttempts(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryRetryAttempts(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Uint32 *value)
@@ -10206,6 +10420,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDeliveryRetryAttempts(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_Delete(
         _Inout_ MI_SubscriptionDeliveryOptions* self)
 {
@@ -10219,6 +10434,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_Delete(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetString(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10235,6 +10451,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetString(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetNumber(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10251,6 +10468,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetNumber(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDateTime(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10267,6 +10485,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetDateTime(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetInterval(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10283,6 +10502,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_SetInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetString(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10300,6 +10520,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetString(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetNumber(
     _Inout_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10317,6 +10538,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetNumber(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDateTime(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10334,6 +10556,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetDateTime(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetInterval(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10351,6 +10574,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetInterval(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOptionCount(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_opt_ MI_Uint32 *count)
@@ -10365,6 +10589,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOptionCount(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOptionAt(
     _In_   MI_SubscriptionDeliveryOptions *self,
     MI_Uint32 index,
@@ -10383,6 +10608,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOptionAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOption(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _In_z_ const MI_Char *optionName,
@@ -10401,6 +10627,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetOption(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsCount(
     _In_   MI_SubscriptionDeliveryOptions *self,
     _Out_ MI_Uint32 *count)
@@ -10415,6 +10642,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsCount(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsAt(
     _In_   MI_SubscriptionDeliveryOptions *self,
     MI_Uint32 index,
@@ -10432,6 +10660,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsAt(
     }
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsPasswordAt(
     _In_   MI_SubscriptionDeliveryOptions *self,
     MI_Uint32 index,
@@ -10465,6 +10694,7 @@ MI_INLINE MI_Result MI_SubscriptionDeliveryOptions_GetCredentialsPasswordAt(
  * See also: MI_SubscriptionDeliveryOptions_Delete()
  *
  */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_INLINE_CALL MI_SubscriptionDeliveryOptions_Clone(
     _In_ const MI_SubscriptionDeliveryOptions* self,
     _Out_ MI_SubscriptionDeliveryOptions* newSubscriptionDeliveryOptions)
@@ -10482,12 +10712,14 @@ MI_INLINE MI_Result MI_INLINE_CALL MI_SubscriptionDeliveryOptions_Clone(
 #define MI_SERIALIZER_FLAGS_CLASS_DEEP 1
 #define MI_SERIALIZER_FLAGS_INSTANCE_WITH_CLASS 1
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Serializer_Close(
     _Inout_ MI_Serializer *serializer)
 {
     return mi_clientFT->serializerFT->Close(serializer);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Serializer_SerializeClass(
     _Inout_ MI_Serializer *serializer,
     MI_Uint32 flags,
@@ -10499,6 +10731,7 @@ MI_INLINE MI_Result MI_Serializer_SerializeClass(
     return mi_clientFT->serializerFT->SerializeClass(serializer, flags, classObject, clientBuffer, clientBufferLength, clientBufferNeeded);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Serializer_SerializeInstance(
    _Inout_ MI_Serializer *serializer,
    MI_Uint32 flags,
@@ -10511,6 +10744,7 @@ MI_INLINE MI_Result MI_Serializer_SerializeInstance(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_Close(
     _Inout_ MI_Deserializer *deserializer)
 {
@@ -10519,6 +10753,7 @@ MI_INLINE MI_Result MI_Deserializer_Close(
     return deserializerFT->Close(deserializer);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_DeserializeClass(
     _Inout_ MI_Deserializer *deserializer,
     MI_Uint32 flags,
@@ -10536,6 +10771,7 @@ MI_INLINE MI_Result MI_Deserializer_DeserializeClass(
     return mi_clientFT->deserializerFT->DeserializeClass(deserializer, flags, serializedBuffer, serializedBufferLength, parentClass, serverName, namespaceName, classObjectNeeded, classObjectNeededContext, serializedBufferRead, classObject, cimErrorDetails);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_Class_GetClassName(
     _Inout_ MI_Deserializer *deserializer,
     _In_reads_(serializedBufferLength) MI_Uint8 *serializedBuffer,
@@ -10548,6 +10784,7 @@ MI_INLINE MI_Result MI_Deserializer_Class_GetClassName(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_Class_GetParentClassName(
     _Inout_ MI_Deserializer *deserializer,
     _In_reads_(serializedBufferLength) MI_Uint8 *serializedBuffer,
@@ -10559,6 +10796,7 @@ MI_INLINE MI_Result MI_Deserializer_Class_GetParentClassName(
     return mi_clientFT->deserializerFT->Class_GetParentClassName(deserializer, serializedBuffer, serializedBufferLength, parentClassName, parentClassNameLength, cimErrorDetails);
 }
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_DeserializeInstance(
     _Inout_ MI_Deserializer *deserializer,
     MI_Uint32 flags,
@@ -10576,6 +10814,7 @@ MI_INLINE MI_Result MI_Deserializer_DeserializeInstance(
 }
 
 
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Deserializer_Instance_GetClassName(
     _Inout_ MI_Deserializer *deserializer,
     _In_reads_(serializedBufferLength) MI_Uint8 *serializedBuffer,
@@ -10589,6 +10828,7 @@ MI_INLINE MI_Result MI_Deserializer_Instance_GetClassName(
 
 
 /* Map an OS specific error code to an error category */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_ErrorCategory MI_Utilities_MapErrorToMiErrorCategory(
         _In_z_ MI_Char *errorType,
         MI_Uint32 error)
@@ -10600,6 +10840,7 @@ MI_INLINE MI_ErrorCategory MI_Utilities_MapErrorToMiErrorCategory(
     * Call MI_Instance_Delete on the CIM when finished
     * using it.
     */
+_Win32_metadata_supported_os_(windows8.0)
 MI_INLINE MI_Result MI_Utilities_CimErrorFromErrorCode(
     MI_Uint32 error,
     _In_z_ const MI_Char *errorType,
