diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Midles.h b/generation/WinSDK/RecompiledIdlHeaders/um/Midles.h
index 557f8b48..c7d79929 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Midles.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Midles.h
@@ -20,6 +20,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
@@ -74,6 +77,7 @@ typedef struct _MIDL_TYPE_PICKLING_INFO
 } MIDL_TYPE_PICKLING_INFO,  * PMIDL_TYPE_PICKLING_INFO;
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesEncodeIncrementalHandleCreate(
     void          * UserState,
@@ -81,6 +85,7 @@ MesEncodeIncrementalHandleCreate(
     MIDL_ES_WRITE   WriteFn,
     handle_t      * pHandle );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesDecodeIncrementalHandleCreate(
     void         *  UserState,
@@ -88,6 +93,7 @@ MesDecodeIncrementalHandleCreate(
     handle_t     *  pHandle );
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesIncrementalHandleReset(
     handle_t        Handle,
@@ -98,6 +104,7 @@ MesIncrementalHandleReset(
     MIDL_ES_CODE    Operation );
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesEncodeFixedBufferHandleCreate(
     _Out_writes_bytes_(BufferSize) char *pBuffer,
@@ -106,6 +113,7 @@ MesEncodeFixedBufferHandleCreate(
     _Out_ handle_t *pHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesEncodeDynBufferHandleCreate(
     _Outptr_result_bytebuffer_(*pEncodedSize) char **pBuffer,
@@ -113,6 +121,7 @@ MesEncodeDynBufferHandleCreate(
     _Out_ handle_t *pHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesDecodeBufferHandleCreate(
     _In_reads_bytes_(BufferSize ) char * Buffer,
@@ -121,6 +130,7 @@ MesDecodeBufferHandleCreate(
     );
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesBufferHandleReset(
     _In_ handle_t Handle,
@@ -132,9 +142,11 @@ MesBufferHandleReset(
     );
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesHandleFree( handle_t  Handle );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS  RPC_ENTRY
 MesInqProcEncodingId(
     handle_t                    Handle,
@@ -219,6 +231,7 @@ NdrMesProcEncodeDecode(
     PFORMAT_STRING          pFormatString,
     ... );
 
+_Win32_metadata_supported_os_(windows5.0)
 CLIENT_CALL_RETURN  RPC_VAR_ENTRY
 NdrMesProcEncodeDecode2(
     handle_t                Handle,
