diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mmiscapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/mmiscapi.h
index f91b63b4..a8faf165 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mmiscapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mmiscapi.h
@@ -12,6 +12,7 @@
 
 #ifndef _MMISCAPI_H_
 #define _MMISCAPI_H_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -95,6 +96,7 @@ typedef struct tagDRVCONFIGINFO {
 
 typedef LRESULT (CALLBACK* DRIVERPROC)(DWORD_PTR, HDRVR, UINT, LPARAM, LPARAM);
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LRESULT
 WINAPI
@@ -104,6 +106,7 @@ CloseDriver(
     _In_ LPARAM lParam2
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 HDRVR
 WINAPI
@@ -113,6 +116,7 @@ OpenDriver(
     _In_ LPARAM lParam2
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LRESULT
 WINAPI
@@ -123,6 +127,7 @@ SendDriverMessage(
     _In_ LPARAM lParam2
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 HMODULE
 WINAPI
@@ -130,6 +135,7 @@ DrvGetModuleHandle(
     _In_ HDRVR hDriver
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 HMODULE
 WINAPI
@@ -137,6 +143,7 @@ GetDriverModuleHandle(
     _In_ HDRVR hDriver
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LRESULT
 WINAPI
@@ -152,7 +159,9 @@ DefDriverProc(
 LRESULT   WINAPI DrvClose(HDRVR hdrvr, LPARAM lParam1, LPARAM lParam2);
 HDRVR     WINAPI DrvOpen(LPCSTR szDriverName, LPCSTR szSectionName, LPARAM lParam2);
 LRESULT   WINAPI DrvSendMessage(HDRVR hdrvr, UINT uMsg, LPARAM lParam1, LPARAM lParam2);
+_Win32_metadata_supported_os_(windows5.0)
 HINSTANCE WINAPI DrvGetModuleHandle(HDRVR hdrvr);
+_Win32_metadata_supported_os_(windows5.0)
 LRESULT   WINAPI DrvDefDriverProc(DWORD dwDriverIdentifier, HDRVR hdrvr, UINT uMsg, LPARAM lParam1, LPARAM lParam2);
 #define DefDriverProc DrvDefDriverProc
 #endif /* ifdef _WIN32 */
@@ -175,6 +184,7 @@ LRESULT   WINAPI DrvDefDriverProc(DWORD dwDriverIdentifier, HDRVR hdrvr, UINT uM
                       Driver Helper function moved from mmddk.h
 
 ***************************************************************************/
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 APIENTRY
 DriverCallback(
@@ -192,6 +202,7 @@ DriverCallback(
   Sound schemes
 
 ****************************************************************************/LONG
+_Win32_metadata_supported_os_(windows8.1)
 WINAPI
 sndOpenSound(
     _In_ LPCWSTR EventName,
@@ -382,6 +393,7 @@ typedef const MMCKINFO *LPCMMCKINFO;
 /* MMIO function prototypes */
 #ifdef _WIN32
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 FOURCC
 WINAPI
@@ -390,6 +402,7 @@ mmioStringToFOURCCA(
     _In_ UINT uFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 FOURCC
 WINAPI
@@ -403,6 +416,7 @@ mmioStringToFOURCCW(
 #else
 #define mmioStringToFOURCC  mmioStringToFOURCCA
 #endif // !UNICODE
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LPMMIOPROC
 WINAPI
@@ -412,6 +426,7 @@ mmioInstallIOProcA(
     _In_ DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LPMMIOPROC
 WINAPI
@@ -426,6 +441,7 @@ mmioInstallIOProcW(
 #else
 #define mmioInstallIOProc  mmioInstallIOProcA
 #endif // !UNICODE
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 HMMIO
 WINAPI
@@ -435,6 +451,7 @@ mmioOpenA(
     _In_ DWORD fdwOpen
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 HMMIO
 WINAPI
@@ -449,6 +466,7 @@ mmioOpenW(
 #else
 #define mmioOpen  mmioOpenA
 #endif // !UNICODE
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -459,6 +477,7 @@ mmioRenameA(
     _In_ DWORD fdwRename
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -475,14 +494,19 @@ mmioRenameW(
 #define mmioRename  mmioRenameA
 #endif // !UNICODE
 #else
+_Win32_metadata_supported_os_(windows5.0)
 FOURCC WINAPI mmioStringToFOURCC( LPCSTR sz, UINT uFlags);
+_Win32_metadata_supported_os_(windows5.0)
 LPMMIOPROC WINAPI mmioInstallIOProc( FOURCC fccIOProc, LPMMIOPROC pIOProc, DWORD dwFlags);
+_Win32_metadata_supported_os_(windows5.0)
 HMMIO WINAPI mmioOpen(_Inout_opt_ LPSTR pszFileName, LPMMIOINFO pmmioinfo, DWORD fdwOpen);
 #if (WINVER >= 0x030a)
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT WINAPI mmioRename( _In_ LPCSTR pszFileName, _In_ LPCSTR pszNewFileName, _In_opt_ const MMIOINFO FAR* pmmioinfo, _In_ DWORD fdwRename);
 #endif /* ifdef WINVER >= 0x030a */
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -491,6 +515,7 @@ mmioClose(
     _In_ UINT fuClose
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LONG
 WINAPI
@@ -500,6 +525,7 @@ mmioRead(
     _In_ LONG cch
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LONG
 WINAPI
@@ -509,6 +535,7 @@ mmioWrite(
     _In_ LONG cch
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LONG
 WINAPI
@@ -518,6 +545,7 @@ mmioSeek(
     _In_ int iOrigin
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -527,6 +555,7 @@ mmioGetInfo(
     _In_ UINT fuInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -536,6 +565,7 @@ mmioSetInfo(
     _In_ UINT fuInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -546,6 +576,7 @@ mmioSetBuffer(
     _In_ UINT fuBuffer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -554,6 +585,7 @@ mmioFlush(
     _In_ UINT fuFlush
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -563,6 +595,7 @@ mmioAdvance(
     _In_ UINT fuAdvance
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 LRESULT
 WINAPI
@@ -573,6 +606,7 @@ mmioSendMessage(
     _In_opt_ LPARAM lParam2
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -583,6 +617,7 @@ mmioDescend(
     _In_ UINT fuDescend
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
@@ -592,6 +627,7 @@ mmioAscend(
     _In_ UINT fuAscend
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINMMAPI
 MMRESULT
 WINAPI
