diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mpconfig.h b/generation/WinSDK/RecompiledIdlHeaders/um/mpconfig.h
index 34fe1ded..3ae6000c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mpconfig.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mpconfig.h
@@ -11,6 +11,7 @@
 #define __IMPConfig__
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -28,6 +29,7 @@ typedef enum _AM_ASPECT_RATIO_MODE
 } AM_ASPECT_RATIO_MODE;
 
 
+_Win32_metadata_supported_os_(windows5.0)
 DECLARE_INTERFACE_(IMixerPinConfig, IUnknown)
 {
     // this function sets the position of the stream in the display window, assuming 
@@ -126,6 +128,7 @@ DECLARE_INTERFACE_(IMixerPinConfig, IUnknown)
 };
 
 
+_Win32_metadata_supported_os_(windows5.0)
 DECLARE_INTERFACE_(IMixerPinConfig2, IMixerPinConfig)
 {
     // this functions sets the color-controls, if the vga chip supports it.
