diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Mprapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/Mprapi.h
index 7a04178b..8be9d35d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Mprapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Mprapi.h
@@ -130,6 +130,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or CmdLineTools Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_CMDTOOLS)
@@ -2882,6 +2885,7 @@ DWORD APIENTRY
 MprConfigServerConnect(
     _In_opt_  LPWSTR                  lpwsServerName,
     _Out_     HANDLE*                 phMprConfig
+        _Win32_metadata_raii_free_(MprConfigServerDisconnect)
 );
 
 VOID APIENTRY
