diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h b/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
index c48a1f20..954ebdb5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
@@ -21,6 +21,9 @@ Abstract:
 #endif
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -2344,6 +2347,7 @@ APIENTRY
 MQCreateCursor(
     _In_ QUEUEHANDLE hQueue,
     _Out_ PHANDLE phCursor
+        _Win32_metadata_raii_free_(MQCloseCursor)
     );
 
 HRESULT
