diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MSAcm.h b/generation/WinSDK/RecompiledIdlHeaders/um/MSAcm.h
index f1e137c9..47f64d16 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MSAcm.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MSAcm.h
@@ -18,6 +18,9 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -210,6 +213,7 @@ typedef UINT                MMRESULT;
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 DWORD ACMAPI acmGetVersion
 (
     void
@@ -223,6 +227,7 @@ DWORD ACMAPI acmGetVersion
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmMetrics
 (
     HACMOBJ                 hao,
@@ -272,6 +277,7 @@ typedef BOOL (CALLBACK *ACMDRIVERENUMCB)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverEnum
 (
     ACMDRIVERENUMCB         fnCallback,
@@ -289,6 +295,7 @@ MMRESULT ACMAPI acmDriverEnum
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverID
 (
     HACMOBJ                 hao,
@@ -305,6 +312,7 @@ MMRESULT ACMAPI acmDriverID
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverAddA
 (
     LPHACMDRIVERID          phadid,
@@ -314,6 +322,7 @@ MMRESULT ACMAPI acmDriverAddA
     DWORD                   fdwAdd
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverAddW
 (
     LPHACMDRIVERID          phadid,
@@ -329,6 +338,7 @@ MMRESULT ACMAPI acmDriverAddW
 #define acmDriverAdd        acmDriverAddA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverAdd
 (
     LPHACMDRIVERID          phadid,
@@ -360,6 +370,7 @@ typedef ACMDRIVERPROC FAR *LPACMDRIVERPROC;
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverRemove
 (
     HACMDRIVERID            hadid,
@@ -374,6 +385,7 @@ MMRESULT ACMAPI acmDriverRemove
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverOpen
 (
     LPHACMDRIVER            phad, 
@@ -389,6 +401,7 @@ MMRESULT ACMAPI acmDriverOpen
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverClose
 (
     HACMDRIVER              had,
@@ -403,6 +416,7 @@ MMRESULT ACMAPI acmDriverClose
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 LRESULT ACMAPI acmDriverMessage
 (
     HACMDRIVER              had,
@@ -432,6 +446,7 @@ LRESULT ACMAPI acmDriverMessage
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverPriority
 (
     HACMDRIVERID            hadid,
@@ -625,6 +640,7 @@ typedef struct tACMDRIVERDETAILS
 
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverDetailsA
 (
     HACMDRIVERID            hadid,
@@ -632,6 +648,7 @@ MMRESULT ACMAPI acmDriverDetailsA
     DWORD                   fdwDetails
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverDetailsW
 (
     HACMDRIVERID            hadid,
@@ -645,6 +662,7 @@ MMRESULT ACMAPI acmDriverDetailsW
 #define acmDriverDetails    acmDriverDetailsA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmDriverDetails
 (
     HACMDRIVERID            hadid,
@@ -722,6 +740,7 @@ typedef struct tACMFORMATTAGDETAILS
 #endif
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagDetailsA
 (
     HACMDRIVER              had,
@@ -729,6 +748,7 @@ MMRESULT ACMAPI acmFormatTagDetailsA
     DWORD                   fdwDetails
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagDetailsW
 (
     HACMDRIVER              had,
@@ -742,6 +762,7 @@ MMRESULT ACMAPI acmFormatTagDetailsW
 #define acmFormatTagDetails     acmFormatTagDetailsA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagDetails
 (
     HACMDRIVER              had,
@@ -773,6 +794,7 @@ typedef BOOL (CALLBACK *ACMFORMATTAGENUMCBA)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagEnumA
 (
     HACMDRIVER              had,
@@ -789,6 +811,7 @@ typedef BOOL (CALLBACK *ACMFORMATTAGENUMCBW)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagEnumW
 (
     HACMDRIVER              had,
@@ -814,6 +837,7 @@ typedef BOOL (CALLBACK *ACMFORMATTAGENUMCB)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatTagEnum
 (
     HACMDRIVER              had,
@@ -893,6 +917,7 @@ typedef struct tACMFORMATDETAILS
 
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatDetailsA
 (
     HACMDRIVER              had,
@@ -900,6 +925,7 @@ MMRESULT ACMAPI acmFormatDetailsA
     DWORD                   fdwDetails
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatDetailsW
 (
     HACMDRIVER              had,
@@ -913,6 +939,7 @@ MMRESULT ACMAPI acmFormatDetailsW
 #define acmFormatDetails    acmFormatDetailsA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatDetails
 (
     HACMDRIVER              had,
@@ -942,6 +969,7 @@ typedef BOOL (CALLBACK *ACMFORMATENUMCBA)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatEnumA
 (
     HACMDRIVER              had,
@@ -959,6 +987,7 @@ typedef BOOL (CALLBACK *ACMFORMATENUMCBW)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatEnumW
 (
     HACMDRIVER              had,
@@ -984,6 +1013,7 @@ typedef BOOL (CALLBACK *ACMFORMATENUMCB)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatEnum
 (
     HACMDRIVER              had,
@@ -1013,6 +1043,7 @@ MMRESULT ACMAPI acmFormatEnum
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatSuggest
 (
     HACMDRIVER          had,
@@ -1212,11 +1243,13 @@ typedef struct tACMFORMATCHOOSE
 #define ACMFORMATCHOOSE_STYLEF_CONTEXTHELP           0x00000080L
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatChooseA
 (
     LPACMFORMATCHOOSEA      pafmtc
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatChooseW
 (
     LPACMFORMATCHOOSEW      pafmtc
@@ -1228,6 +1261,7 @@ MMRESULT ACMAPI acmFormatChooseW
 #define acmFormatChoose     acmFormatChooseA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFormatChoose
 (
     LPACMFORMATCHOOSE       pafmtc
@@ -1302,6 +1336,7 @@ typedef struct tACMFILTERTAGDETAILS
 #endif
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagDetailsA
 (
     HACMDRIVER              had,
@@ -1309,6 +1344,7 @@ MMRESULT ACMAPI acmFilterTagDetailsA
     DWORD                   fdwDetails
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagDetailsW
 (
     HACMDRIVER              had,
@@ -1322,6 +1358,7 @@ MMRESULT ACMAPI acmFilterTagDetailsW
 #define acmFilterTagDetails     acmFilterTagDetailsA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagDetails
 (
     HACMDRIVER              had,
@@ -1353,6 +1390,7 @@ typedef BOOL (CALLBACK *ACMFILTERTAGENUMCBA)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagEnumA
 (
     HACMDRIVER              had,
@@ -1370,6 +1408,7 @@ typedef BOOL (CALLBACK *ACMFILTERTAGENUMCBW)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagEnumW
 (
     HACMDRIVER              had,
@@ -1395,6 +1434,7 @@ typedef BOOL (CALLBACK *ACMFILTERTAGENUMCB)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterTagEnum
 (
     HACMDRIVER              had,
@@ -1473,6 +1513,7 @@ typedef struct tACMFILTERDETAILS
 #endif
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterDetailsA
 (
     HACMDRIVER              had,
@@ -1480,6 +1521,7 @@ MMRESULT ACMAPI acmFilterDetailsA
     DWORD                   fdwDetails
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterDetailsW
 (
     HACMDRIVER              had,
@@ -1492,6 +1534,7 @@ MMRESULT ACMAPI acmFilterDetailsW
 #define acmFilterDetails    acmFilterDetailsA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterDetails
 (
     HACMDRIVER              had,
@@ -1521,6 +1564,7 @@ typedef BOOL (CALLBACK *ACMFILTERENUMCBA)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterEnumA
 (
     HACMDRIVER              had,
@@ -1538,6 +1582,7 @@ typedef BOOL (CALLBACK *ACMFILTERENUMCBW)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterEnumW
 (
     HACMDRIVER              had,
@@ -1563,6 +1608,7 @@ typedef BOOL (CALLBACK *ACMFILTERENUMCB)
     DWORD                   fdwSupport
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterEnum
 (
     HACMDRIVER              had,
@@ -1736,11 +1782,13 @@ typedef struct tACMFILTERCHOOSE
 #define ACMFILTERCHOOSE_STYLEF_CONTEXTHELP           0x00000080L
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterChooseA
 (
     LPACMFILTERCHOOSEA      pafltrc
 );
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterChooseW
 (
     LPACMFILTERCHOOSEW      pafltrc
@@ -1752,6 +1800,7 @@ MMRESULT ACMAPI acmFilterChooseW
 #define acmFilterChoose     acmFilterChooseA
 #endif
 #else
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmFilterChoose
 (
     LPACMFILTERCHOOSE       pafltrc
@@ -1806,6 +1855,7 @@ typedef struct tACMSTREAMHEADER
 #define ACMSTREAMHEADER_STATUSF_INQUEUE     0x00100000L
 
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamOpen
 (
     LPHACMSTREAM            phas,       // pointer to stream handle
@@ -1829,6 +1879,7 @@ MMRESULT ACMAPI acmStreamOpen
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamClose
 (
     HACMSTREAM              has,
@@ -1843,6 +1894,7 @@ MMRESULT ACMAPI acmStreamClose
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamSize
 (
     HACMSTREAM              has,
@@ -1863,6 +1915,7 @@ MMRESULT ACMAPI acmStreamSize
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamReset
 (
     HACMSTREAM              has,
@@ -1877,6 +1930,7 @@ MMRESULT ACMAPI acmStreamReset
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamMessage
 (
     HACMSTREAM              has,
@@ -1894,6 +1948,7 @@ MMRESULT ACMAPI acmStreamMessage
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamConvert
 (
     HACMSTREAM              has, 
@@ -1912,6 +1967,7 @@ MMRESULT ACMAPI acmStreamConvert
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamPrepareHeader
 (
     HACMSTREAM          has,
@@ -1927,6 +1983,7 @@ MMRESULT ACMAPI acmStreamPrepareHeader
 //
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ;
 
+_Win32_metadata_supported_os_(windows5.0)
 MMRESULT ACMAPI acmStreamUnprepareHeader
 (
     HACMSTREAM          has,
