diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MsCtfMonitorApi.h b/generation/WinSDK/RecompiledIdlHeaders/um/MsCtfMonitorApi.h
index 54e963eb..b9c139fc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MsCtfMonitorApi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MsCtfMonitorApi.h
@@ -15,6 +15,7 @@
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -59,6 +60,7 @@ EXTERN_C BOOL WINAPI DoMsCtfMonitor(DWORD dwFlags, HANDLE hEventForServiceStop);
 // InitLocalMsCtfMonitor
 //
 ////////////////////////////////////////////////////////////////////////////
+_Win32_metadata_supported_os_(windows6.0.6000)
 EXTERN_C HRESULT WINAPI InitLocalMsCtfMonitor(DWORD dwFlags);
 
 ////////////////////////////////////////////////////////////////////////////
@@ -66,6 +68,7 @@ EXTERN_C HRESULT WINAPI InitLocalMsCtfMonitor(DWORD dwFlags);
 // UninitLocalMsCtfMonitor
 //
 ////////////////////////////////////////////////////////////////////////////
+_Win32_metadata_supported_os_(windows6.0.6000)
 EXTERN_C HRESULT WINAPI UninitLocalMsCtfMonitor();
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
