diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MsHTML.h b/generation/WinSDK/RecompiledIdlHeaders/um/MsHTML.h
index 8b415ce8..55b9fbd4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MsHTML.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MsHTML.h
@@ -9677,6 +9677,7 @@ extern "C"{
 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #include <dxgitype.h>
 #include <olectl.h>
@@ -184435,6 +184436,7 @@ EXTERN_C const IID IID_IRangeException;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("3051072d-98b5-11cf-bb82-00aa00bdce0b")
     IRangeException : public IDispatch
     {
@@ -215430,9 +215432,13 @@ unsigned char * __RPC_USER  VARIANT_UserMarshal(  __RPC__in unsigned long *, __R
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
 void                      __RPC_USER  VARIANT_UserFree(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  VARIANT_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
 /* end of Additional Prototypes */
