diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Msi.h b/generation/WinSDK/RecompiledIdlHeaders/um/Msi.h
index 6722861f..50c37bde 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Msi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Msi.h
@@ -16,6 +16,7 @@
 #define _MSI_H_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -98,6 +99,7 @@ extern "C" {
 // All handles obtained from API calls must be closed when no longer needed
 // Normally succeeds, returning TRUE. 
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiCloseHandle(MSIHANDLE hAny);
 
 // Close all handles open in the process, a diagnostic call
@@ -105,6 +107,7 @@ UINT WINAPI MsiCloseHandle(MSIHANDLE hAny);
 // Can be called at termination to assure that all handles have been closed
 // Returns 0 if all handles have been close, else number of open handles
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiCloseAllHandles();
 
 #ifdef __cplusplus
@@ -121,9 +124,11 @@ class PMSIHANDLE
  public:
 	PMSIHANDLE():m_h(0){}
 	PMSIHANDLE(MSIHANDLE h):m_h(h){}
+_Win32_metadata_supported_os_(windows8.0)
   ~PMSIHANDLE(){if (m_h!=0) MsiCloseHandle(m_h);}
 	void operator =(MSIHANDLE h) {if (m_h) MsiCloseHandle(m_h); m_h=h;}
 	operator MSIHANDLE() {return m_h;}
+_Win32_metadata_supported_os_(windows8.0)
 	MSIHANDLE* operator &() {if (m_h) MsiCloseHandle(m_h); m_h = 0; return &m_h;}
 };
 #endif  //__cplusplus
@@ -528,6 +533,7 @@ extern "C" {
 
 // Enable internal UI
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLUILEVEL WINAPI MsiSetInternalUI(
 	_In_ INSTALLUILEVEL  dwUILevel,     // UI level
 	_Inout_opt_ HWND  *phWnd);          // handle of owner window
@@ -535,6 +541,7 @@ INSTALLUILEVEL WINAPI MsiSetInternalUI(
 // Enable external UI handling, returns any previous handler or NULL if none.
 // Messages are designated with a combination of bits from INSTALLLOGMODE enum.
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLUI_HANDLERA WINAPI MsiSetExternalUIA(
 	_In_opt_ INSTALLUI_HANDLERA puiHandler,  // for progress and error handling 
 	_In_ DWORD              dwMessageFilter, // bit flags designating messages to handle
@@ -551,6 +558,7 @@ INSTALLUI_HANDLERW WINAPI MsiSetExternalUIW(
 
 #if (_WIN32_MSI >= 310)
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSetExternalUIRecord(
 	_In_opt_ INSTALLUI_HANDLER_RECORD puiHandler,         // for progress and error handling 
 	_In_ DWORD              dwMessageFilter,              // bit flags designating messages to handle
@@ -563,6 +571,7 @@ UINT WINAPI MsiSetExternalUIRecord(
 // with control over which log messages are passed to the specified log file.
 // Messages are designated with a combination of bits from INSTALLLOGMODE enum.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnableLogA(
 	_In_ DWORD     dwLogMode,           // bit flags designating operations to report
 	_In_opt_ LPCSTR  szLogFile,       // log file, or NULL to disable logging
@@ -583,8 +592,10 @@ UINT WINAPI MsiEnableLogW(
 
 // Return the installed state for a product
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiQueryProductStateA(
 	_In_ LPCSTR  szProduct);
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiQueryProductStateW(
 	_In_ LPCWSTR  szProduct);
 #ifdef UNICODE
@@ -595,6 +606,7 @@ INSTALLSTATE WINAPI MsiQueryProductStateW(
 
 // Return product info
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetProductInfoA(
 	_In_ LPCSTR   szProduct,                              // product code
 	_In_ LPCSTR   szAttribute,                            // attribute name, case-sensitive
@@ -614,6 +626,7 @@ UINT WINAPI MsiGetProductInfoW(
 #if (_WIN32_MSI >= 300)
 // Retrieve product info for a particular product instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetProductInfoExA(
 		_In_ LPCSTR szProductCode,                  // Information is queried on this product
 		_In_opt_ LPCSTR szUserSid,                  // Account of this product instance
@@ -638,6 +651,7 @@ UINT WINAPI MsiGetProductInfoExW(
 // Install a new product.
 // Either may be NULL, but the DATABASE property must be specfied
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiInstallProductA(
 	_In_ LPCSTR      szPackagePath,        // location of package to install
 	_In_opt_ LPCSTR      szCommandLine);   // command line <property settings>
@@ -653,6 +667,7 @@ UINT WINAPI MsiInstallProductW(
 // Install/uninstall an advertised or installed product
 // No action if installed and INSTALLSTATE_DEFAULT specified
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiConfigureProductA(
 	_In_ LPCSTR      szProduct,       // product code
 	_In_ int          iInstallLevel,    // how much of the product to install
@@ -670,6 +685,7 @@ UINT WINAPI MsiConfigureProductW(
 // Install/uninstall an advertised or installed product
 // No action if installed and INSTALLSTATE_DEFAULT specified
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiConfigureProductExA(
 	_In_ LPCSTR      szProduct,            // product code
 	_In_ int          iInstallLevel,         // how much of the product to install
@@ -688,6 +704,7 @@ UINT WINAPI MsiConfigureProductExW(
 
 // Reinstall product, used to validate or correct problems
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiReinstallProductA(
 	_In_ LPCSTR      szProduct,        // product code
 	_In_ DWORD         szReinstallMode); // one or more REINSTALLMODE modes
@@ -709,6 +726,7 @@ UINT WINAPI MsiReinstallProductW(
 // If dwOptions includes MSIADVERTISEOPTIONFLAGS_INSTANCE, then a new instance is advertised. Use of
 //    this option requires that szTransforms include the instance transform that changes the product code
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiAdvertiseProductExA(
 	_In_ LPCSTR	szPackagePath,      // location of package
 	_In_opt_ LPCSTR    szScriptfilePath,   // if NULL, product is locally advertised
@@ -735,6 +753,7 @@ UINT WINAPI MsiAdvertiseProductExW(
 
 // Output reg and shortcut info to script file for Assign or Publish
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiAdvertiseProductA(
 	_In_ LPCSTR      szPackagePath,    // location of package
 	_In_opt_ LPCSTR      szScriptfilePath,  // if NULL, product is locally advertised
@@ -761,6 +780,7 @@ UINT WINAPI MsiAdvertiseProductW(
 //    returned by SHGetSpecialFolderLocation(?), it will hold the shortcuts.
 // if fRemoveItems is TRUE, items that are present will be removed
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiProcessAdvertiseScriptA(
 	_In_ LPCSTR      szScriptFile,  // path to script from MsiAdvertiseProduct
 	_In_opt_ LPCSTR      szIconFolder,  // optional path to folder for icon files and transforms
@@ -785,6 +805,7 @@ UINT WINAPI MsiProcessAdvertiseScriptW(
 // Process advertise script file using the supplied dwFlags control flags
 // if fRemoveItems is TRUE, items that are present will be removed
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiAdvertiseScriptA(
 	_In_ LPCSTR      szScriptFile,  // path to script from MsiAdvertiseProduct
 	_In_ DWORD         dwFlags,       // the SCRIPTFLAGS bit flags that control the script execution
@@ -805,6 +826,7 @@ UINT WINAPI MsiAdvertiseScriptW(
 //   product code, language, version, readable name, path to package
 // Returns TRUE is success, FALSE if szScriptFile is not a valid script file
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetProductInfoFromScriptA(
 	_In_                               LPCSTR  szScriptFile,    // path to installer script file
 	_Out_writes_opt_(MAX_GUID_CHARS+1) LPSTR   lpProductBuf39,  // buffer for product code string GUID, 39 chars
@@ -831,6 +853,7 @@ UINT WINAPI MsiGetProductInfoFromScriptW(
 
 // Return the product code for a registered component, called once by apps
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetProductCodeA(
 	_In_ LPCSTR   szComponent,   // component Id registered for this product
 	_Out_writes_(MAX_GUID_CHARS+1) LPSTR lpBuf39);   // returned string GUID, sized for 39 characters
@@ -845,6 +868,7 @@ UINT WINAPI MsiGetProductCodeW(
 
 // Return the registered user information for an installed product
 
+_Win32_metadata_supported_os_(windows8.0)
 USERINFOSTATE WINAPI MsiGetUserInfoA(
 	_In_                                   LPCSTR  szProduct,    // product code, string GUID
 	_Out_writes_opt_(*pcchUserNameBuf) LPSTR   lpUserNameBuf,    // return user name           
@@ -869,6 +893,7 @@ USERINFOSTATE WINAPI MsiGetUserInfoW(
 
 // Obtain and store user info and PID from installation wizard (first run)
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiCollectUserInfoA(
 	_In_ LPCSTR  szProduct);     // product code, string GUID
 UINT WINAPI MsiCollectUserInfoW(
@@ -885,6 +910,7 @@ UINT WINAPI MsiCollectUserInfoW(
 
 // Patch all possible installed products.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiApplyPatchA(
 	_In_ LPCSTR szPatchPackage,       // location of patch package
 	_In_opt_ LPCSTR szInstallPackage, // location of package for install to patch <optional>
@@ -903,6 +929,7 @@ UINT WINAPI MsiApplyPatchW(
 
 // Return patch info for first visible patch
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetPatchInfoA(
 	_In_ LPCSTR szPatch,                              // patch code
 	_In_ LPCSTR szAttribute,                          // attribute name, case-sensitive
@@ -921,12 +948,14 @@ UINT WINAPI MsiGetPatchInfoW(
 
 // Enumerate all patches for a product in contexts visible to caller
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumPatchesA(
 	_In_ LPCSTR szProduct,
 	_In_ DWORD iPatchIndex,
 	_Out_writes_(MAX_GUID_CHARS+1)   LPSTR  lpPatchBuf,
 	_Out_writes_(*pcchTransformsBuf) LPSTR  lpTransformsBuf,
 	_Inout_ LPDWORD  pcchTransformsBuf);
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumPatchesW(
 	_In_ LPCWSTR szProduct,
 	_In_ DWORD iPatchIndex,
@@ -942,6 +971,7 @@ UINT WINAPI MsiEnumPatchesW(
 #if (_WIN32_MSI >= 300)
 // Remove one or more patches from the specified product.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiRemovePatchesA(
 	_In_ LPCSTR szPatchList,         // semi-colon delimited list of patches to remove; patch can be referenced by patch package path or Patch GUID
 	_In_ LPCSTR szProductCode,       // ProductCode GUID of product with patch to remove
@@ -960,6 +990,7 @@ UINT WINAPI MsiRemovePatchesW(
 
 // Extract XML data from the patch
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 UINT WINAPI MsiExtractPatchXMLDataA(
 	_In_ LPCSTR szPatchPath,                         // Patch file to open
 	_Reserved_ DWORD dwReserved,                       // Reserved
@@ -979,6 +1010,7 @@ UINT WINAPI MsiExtractPatchXMLDataW(
 // Retrieve extended patch info for a particular patch applied
 // to a particular product instance.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetPatchInfoExA(
         _In_ LPCSTR szPatchCode,                    // target patch to query
         _In_ LPCSTR szProductCode,                  // target product of patch application
@@ -1004,6 +1036,7 @@ UINT WINAPI MsiGetPatchInfoExW(
 // Apply multiple patches to the specified product or to all eligible products
 // on the machine
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiApplyMultiplePatchesA(
         _In_ LPCSTR szPatchPackages,       // Patches to apply
         _In_opt_ LPCSTR szProductCode,     // target product of patch application
@@ -1022,6 +1055,7 @@ UINT WINAPI MsiApplyMultiplePatchesW(
 // patch files, XML files, and/or XML blobs. Patches already applied to the product
 // are taken into account
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiDeterminePatchSequenceA(
 	_In_ LPCSTR szProductCode,                                   // Product code GUID of an installed product
 	_In_opt_ LPCSTR szUserSid,                                   // User account we're interested in
@@ -1044,6 +1078,7 @@ UINT WINAPI MsiDeterminePatchSequenceW(
 // XML files, and/or XML blobs. Does not take into account either products or patches
 // already installed on the machine.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiDetermineApplicablePatchesA(
 	_In_ LPCSTR szProductPackagePath,                            // Fully qualified path to a product MSI database
 	_In_ DWORD cPatchInfo,                                         // Number of patches in the array
@@ -1061,6 +1096,7 @@ UINT WINAPI MsiDetermineApplicablePatchesW(
 // Enumerates all patches applied for a single product or all products based upon
 // enumeration context and filter.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumPatchesExA(
 	_In_opt_ LPCSTR szProductCode,                                   // Enumerate patches on instances of this product
 	_In_opt_ LPCSTR szUserSid,                                       // Account for enumeration
@@ -1097,9 +1133,11 @@ UINT WINAPI MsiEnumPatchesExW(
 
 // Return the installed state for a product feature
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiQueryFeatureStateA(
 	_In_ LPCSTR  szProduct,
 	_In_ LPCSTR  szFeature);
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiQueryFeatureStateW(
 	_In_ LPCWSTR  szProduct,
 	_In_ LPCWSTR  szFeature);
@@ -1112,6 +1150,7 @@ INSTALLSTATE WINAPI MsiQueryFeatureStateW(
 #if (_WIN32_MSI >= 300)	
 // Retrieve the installed state for a product feature in a particular product instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiQueryFeatureStateExA(
 		_In_ LPCSTR szProductCode,        // Information is queried on this product
 		_In_opt_ LPCSTR szUserSid,        // Account of this product instance
@@ -1134,9 +1173,11 @@ UINT WINAPI MsiQueryFeatureStateExW(
 // Indicate intent to use a product feature, increments usage count
 // Prompts for CD if not loaded, does not install feature
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiUseFeatureA(
 	_In_ LPCSTR  szProduct,
 	_In_ LPCSTR  szFeature);
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiUseFeatureW(
 	_In_ LPCWSTR  szProduct,
 	_In_ LPCWSTR  szFeature);
@@ -1150,6 +1191,7 @@ INSTALLSTATE WINAPI MsiUseFeatureW(
 // Prompts for CD if not loaded, does not install feature
 // Allows for bypassing component detection where performance is critical
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiUseFeatureExA(
 	_In_ LPCSTR  szProduct,          // product code
 	_In_ LPCSTR  szFeature,          // feature ID
@@ -1168,6 +1210,7 @@ INSTALLSTATE WINAPI MsiUseFeatureExW(
 
 // Return the usage metrics for a product feature
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetFeatureUsageA(
 	_In_ LPCSTR      szProduct,            // product code
 	_In_ LPCSTR      szFeature,            // feature ID
@@ -1186,6 +1229,7 @@ UINT WINAPI MsiGetFeatureUsageW(
 
 // Force the installed state for a product feature
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiConfigureFeatureA(
 	_In_ LPCSTR  szProduct,
 	_In_ LPCSTR  szFeature,
@@ -1203,6 +1247,7 @@ UINT WINAPI MsiConfigureFeatureW(
 
 // Reinstall feature, used to validate or correct problems
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiReinstallFeatureA(
 	_In_ LPCSTR      szProduct,        // product code
 	_In_ LPCSTR      szFeature,        // feature ID, NULL for entire product
@@ -1228,6 +1273,7 @@ UINT WINAPI MsiReinstallFeatureW(
 // then calls MsiConfigureFeature if any of its components are uninstalled
 // then calls MsiLocateComponent to obtain the path the its key file
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiProvideComponentA(
 	_In_ LPCSTR     szProduct,                               // product code in case install required
 	_In_ LPCSTR     szFeature,                               // feature ID in case install required
@@ -1251,6 +1297,7 @@ UINT WINAPI MsiProvideComponentW(
 // Return full component path for a qualified component, performing any necessary installation. 
 // Prompts for source if necessary and increments the usage count for the feature.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiProvideQualifiedComponentA(
 	_In_ LPCSTR     szCategory,                           // component category ID
 	_In_ LPCSTR     szQualifier,                          // specifies which component to access
@@ -1274,6 +1321,7 @@ UINT WINAPI MsiProvideQualifiedComponentW(
 // The szProduct parameter specifies the product to match that has published the qualified
 // component. If null, this API works the same as MsiProvideQualifiedComponent. 
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiProvideQualifiedComponentExA(
 	_In_ LPCSTR     szCategory,                           // component category ID
 	_In_ LPCSTR     szQualifier,                          // specifies which component to access
@@ -1300,6 +1348,7 @@ UINT WINAPI MsiProvideQualifiedComponentExW(
 
 // Return full path to an installed component
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiGetComponentPathA(
 	_In_ LPCSTR   szProduct,                        // product code for client product
 	_In_ LPCSTR   szComponent,                      // component Id, string GUID
@@ -1320,6 +1369,7 @@ INSTALLSTATE WINAPI MsiGetComponentPathW(
 // Return full path to an installed component for a given user
 // and context
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiGetComponentPathExA(		
 	_In_					LPCSTR		szProductCode,		// ProductCode
 	_In_					LPCSTR		szComponentCode,	// ComponentCode
@@ -1352,6 +1402,7 @@ INSTALLSTATE WINAPI MsiGetComponentPathExW(
 // The szAppContext is the full path to the .cfg file or the app exe to which the assembly being requested 
 // has been privatised to, which is null for global assemblies
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiProvideAssemblyA(
 	_In_ LPCSTR    szAssemblyName,                       // stringized assembly name
 	_In_opt_ LPCSTR    szAppContext,                     // specifies the full path to the parent asm's .cfg file, null for global assemblies
@@ -1377,6 +1428,7 @@ UINT WINAPI MsiProvideAssemblyW(
 #if (_WIN32_MSI >= 300)
 // Query the installed state of a component in the specified product instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiQueryComponentStateA(
 		_In_ LPCSTR szProductCode,       // Information is queried on this product
 		_In_opt_ LPCSTR szUserSid,       // Account of this product instance
@@ -1404,6 +1456,7 @@ UINT WINAPI MsiQueryComponentStateW(
 
 // Enumerate the registered products, either installed or advertised
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumProductsA(
 	_In_ DWORD     iProductIndex,                               // 0-based index into registered products
 	_Out_writes_(MAX_GUID_CHARS+1)  LPSTR   lpProductBuf);    // buffer of char count: 39 (size of string GUID)
@@ -1419,6 +1472,7 @@ UINT WINAPI MsiEnumProductsW(
 #if (_WIN32_MSI >= 300)
 // Enumerate the registered products, either installed or advertised in
 // the specified context
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumProductsExA(
 		_In_opt_ LPCSTR szProductCode,                                      // Enumerate instances of this product
 		_In_opt_ LPCSTR szUserSid,                                          // Account for enumeration
@@ -1430,6 +1484,7 @@ UINT WINAPI MsiEnumProductsExA(
 		_Inout_opt_ LPDWORD pcchSid);                                         // in/out character count of szSid 
 // Enumerate the registered products, either installed or advertised in
 // the specified context
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumProductsExW(
 		_In_opt_ LPCWSTR szProductCode,                                      // Enumerate instances of this product
 		_In_opt_ LPCWSTR szUserSid,                                          // Account for enumeration
@@ -1450,6 +1505,7 @@ UINT WINAPI MsiEnumProductsExW(
 
 // Enumerate products with given upgrade code
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumRelatedProductsA(
 	_In_ LPCSTR  lpUpgradeCode,                               // upgrade code of products to enumerate
 	_Reserved_ DWORD     dwReserved,                            // reserved, must be 0
@@ -1471,6 +1527,7 @@ UINT WINAPI MsiEnumRelatedProductsW(
 // Enumerate the advertised features for a given product.
 // If parent is not required, supplying NULL will improve performance.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumFeaturesA(
 	_In_ LPCSTR  szProduct,
 	_In_ DWORD     iFeatureIndex,                                    // 0-based index into published features
@@ -1489,6 +1546,7 @@ UINT WINAPI MsiEnumFeaturesW(
 
 // Enumerate the installed components for all products
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumComponentsA(
 	_In_ DWORD    iComponentIndex,                              // 0-based index into installed components
 	_Out_writes_(MAX_GUID_CHARS+1)  LPSTR   lpComponentBuf);  // buffer of char count: 39 (size of string GUID)
@@ -1505,6 +1563,7 @@ UINT WINAPI MsiEnumComponentsW(
 // Enumerate installed components for all products for a given users 
 // and context
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumComponentsExA(	
 	_In_opt_				LPCSTR		szUserSid,			// Account for enumeration
 	_In_					DWORD			dwContext,			// Contexts for enumeration
@@ -1531,6 +1590,7 @@ UINT WINAPI MsiEnumComponentsExW(
 
 // Enumerate the client products for a component
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumClientsA(
 	_In_ LPCSTR  szComponent,
 	_In_ DWORD     iProductIndex,                               // 0-based index into client products
@@ -1549,6 +1609,7 @@ UINT WINAPI MsiEnumClientsW(
 // Enumerate the client products for a component for a given users 
 // and context
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumClientsExA (	
 	_In_					LPCSTR		szComponent,		// ComponentCode for enumeration
 	_In_opt_				LPCSTR		szUserSid,		// Account for enumeration
@@ -1576,6 +1637,7 @@ UINT WINAPI MsiEnumClientsExW (
 
 // Enumerate the qualifiers for an advertised component.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEnumComponentQualifiersA(
 	_In_                                       LPCSTR   szComponent,             // generic component ID that is qualified
 	_In_                                       DWORD      iIndex,	               // 0-based index into qualifiers
@@ -1602,6 +1664,7 @@ UINT WINAPI MsiEnumComponentQualifiersW(
 
 // Open the installation for a product to obtain detailed information
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiOpenProductA(
 	_In_ LPCSTR   szProduct,     // product code
 	_Out_ MSIHANDLE  *hProduct);   // returned product handle, must be closed
@@ -1616,6 +1679,7 @@ UINT WINAPI MsiOpenProductW(
 
 // Open a product package in order to access product properties
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiOpenPackageA(
 	_In_ LPCSTR    szPackagePath,      // path to package, or database handle: #nnnn
 	_Out_ MSIHANDLE  *hProduct);         // returned product handle, must be closed
@@ -1634,6 +1698,7 @@ UINT WINAPI MsiOpenPackageW(
 // Option to create a "safe" engine that does not look at machine state
 //  and does not allow for modification of machine state
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiOpenPackageExA(
 	_In_ LPCSTR   szPackagePath,  // path to package, or database handle: #nnnn
 	_In_ DWORD      dwOptions,      // options flags to indicate whether or not to ignore machine state
@@ -1652,11 +1717,13 @@ UINT WINAPI MsiOpenPackageExW(
 
 #if (_WIN32_MSI >= 400)
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetPatchFileListA(
 	_In_  LPCSTR    szProductCode,
 	_In_  LPCSTR    szPatchPackages,
 	_Out_ LPDWORD      pcFiles,
 	_Outptr_result_buffer_maybenull_(*pcFiles)MSIHANDLE** pphFileRecords);
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetPatchFileListW(
 	_In_  LPCWSTR    szProductCode,
 	_In_  LPCWSTR    szPatchPackages,
@@ -1672,6 +1739,7 @@ UINT WINAPI MsiGetPatchFileListW(
 
 // Provide the value for an installation property.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetProductPropertyA(
 	_In_ MSIHANDLE   hProduct,                                   // product handle obtained from MsiOpenProduct
 	_In_ LPCSTR    szProperty,                                 // property name, case-sensitive
@@ -1692,6 +1760,7 @@ UINT WINAPI MsiGetProductPropertyW(
 // Determine whether a file is a package
 // Returns ERROR_SUCCESS if file is a package.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiVerifyPackageA(
 	_In_ LPCSTR      szPackagePath);   // location of package
 UINT WINAPI MsiVerifyPackageW(
@@ -1709,6 +1778,7 @@ UINT WINAPI MsiVerifyPackageW(
 //   1 = highest priority, feature installed if parent is installed
 //  >1 = decreasing priority, feature installation based on InstallLevel property
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetFeatureInfoA(
 	_In_                             MSIHANDLE   hProduct,      // product handle obtained from MsiOpenProduct
 	_In_                             LPCSTR    szFeature,     // feature name
@@ -1740,6 +1810,7 @@ UINT WINAPI MsiGetFeatureInfoW(
 // This would typically occur due to failue to establish feature availability
 // The product feature having the smallest incremental cost is installed
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiInstallMissingComponentA(
 	_In_ LPCSTR      szProduct,        // product code
 	_In_ LPCSTR      szComponent,      // component Id, string GUID
@@ -1759,6 +1830,7 @@ UINT WINAPI MsiInstallMissingComponentW(
 // The missing component is determined from the product's File table, then
 // the product feature having the smallest incremental cost is installed
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiInstallMissingFileA(
 	_In_ LPCSTR      szProduct,        // product code
 	_In_ LPCSTR      szFile);          // file name, without path
@@ -1776,6 +1848,7 @@ UINT WINAPI MsiInstallMissingFileW(
 // but is not guaranteed to find the correct product for the caller.
 // MsiGetComponentPath should always be called when possible.
 
+_Win32_metadata_supported_os_(windows8.0)
 INSTALLSTATE WINAPI MsiLocateComponentA(
 	_In_ LPCSTR szComponent,                         // component Id, string GUID
 	_Out_writes_opt_(*pcchBuf)  LPSTR  lpPathBuf,    // returned path
@@ -1800,6 +1873,7 @@ INSTALLSTATE WINAPI MsiLocateComponentW(
 // and removes all network sources from the list. A NULL or empty value for
 // the user name indicates the per-machine install.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListClearAllA(
 	_In_ LPCSTR szProduct,          // product code
 	_In_opt_ LPCSTR szUserName,     // user name or NULL/empty for per-machine
@@ -1818,6 +1892,7 @@ UINT WINAPI MsiSourceListClearAllW(
 // and adds the provided source as a new network source. A NULL or empty 
 // value for the user name indicates the per-machine install.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListAddSourceA(
 	_In_ LPCSTR szProduct,          // product code
 	_In_opt_ LPCSTR szUserName,     // user name or NULL/empty for per-machine
@@ -1837,6 +1912,7 @@ UINT WINAPI MsiSourceListAddSourceW(
 // Forces the installer to reevaluate the list of sources the next time that
 // the specified product needs a source.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListForceResolutionA(
 	_In_ LPCSTR szProduct,          // product code
 	_In_opt_ LPCSTR szUserName,     // user name or NULL/empty for per-machine
@@ -1857,6 +1933,7 @@ UINT WINAPI MsiSourceListForceResolutionW(
 // Adds a source or moves a source within the list of sources for the 
 // specified product or patch instance. The source can be network/url type.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListAddSourceExA(
 	_In_ LPCSTR szProductCodeOrPatchCode,// Source will be added to this product or patch
 	_In_opt_ LPCSTR szUserSid,           // Account of this product/patch instance
@@ -1879,6 +1956,7 @@ UINT WINAPI MsiSourceListAddSourceExW(
 
 // Add or update a media source for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListAddMediaDiskA(
 	_In_ LPCSTR szProductCodeOrPatchCode,// Disk will be added to this product or patch
 	_In_opt_ LPCSTR szUserSid,           // Account of this product/patch instance
@@ -1903,6 +1981,7 @@ UINT WINAPI MsiSourceListAddMediaDiskW(
 
 // Remove an existing source for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListClearSourceA(
 	_In_ LPCSTR szProductCodeOrPatchCode, // Source will be cleared for this product or patch
 	_In_opt_ LPCSTR szUserSid,            // Account of this product/patch instance
@@ -1924,6 +2003,7 @@ UINT WINAPI MsiSourceListClearSourceW(
 // Remove an existing registered disk under the media source for the specified product or
 // patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListClearMediaDiskA(
 	_In_ LPCSTR szProductCodeOrPatchCode, // Disk will be cleared for this product or patch
 	_In_opt_ LPCSTR szUserSid,            // Account of this product/patch instance
@@ -1945,6 +2025,7 @@ UINT WINAPI MsiSourceListClearMediaDiskW(
 // Remove all existing sources of the given source type for the specified product or patch
 // instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListClearAllExA(
 	_In_ LPCSTR szProductCodeOrPatchCode, // All sources will be cleared for this product or patch
 	_In_opt_ LPCSTR szUserSid,            // Account of this product/patch instance
@@ -1964,6 +2045,7 @@ UINT WINAPI MsiSourceListClearAllExW(
 // Forces the installer to reevaluate the list of sources the next time that
 // the specified product or patch instance needs a source.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListForceResolutionExA(
 	_In_ LPCSTR szProductCodeOrPatchCode,  // Source Resolution will be forced for this product or patch
 	_In_opt_ LPCSTR szUserSid,             // Account of this product/patch instance
@@ -1982,6 +2064,7 @@ UINT WINAPI MsiSourceListForceResolutionExW(
 
 // Set the source list property to the provided value for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListSetInfoA(
 	_In_ LPCSTR szProductCodeOrPatchCode, // Source Info will be set for this product or patch
 	_In_opt_ LPCSTR szUserSid,            // Account of this product/patch instance
@@ -2004,6 +2087,7 @@ UINT WINAPI MsiSourceListSetInfoW(
 
 // Query the given source list property for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListGetInfoA(
 	_In_ LPCSTR szProductCodeOrPatchCode,        // Source Info will be queried for this produt or patch
 	_In_opt_ LPCSTR szUserSid,                   // Account of this product/patch instance
@@ -2028,6 +2112,7 @@ UINT WINAPI MsiSourceListGetInfoW(
 
 // Enumerate all sources for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListEnumSourcesA(
 	_In_ LPCSTR szProductCodeOrPatchCode,             // Sources will be enumerated for this product or patch
 	_In_opt_ LPCSTR szUserSid,                        // Account of this product/patch instance
@@ -2052,6 +2137,7 @@ UINT WINAPI MsiSourceListEnumSourcesW(
 
 // Enumerate all registered media disks for the specified product or patch instance
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiSourceListEnumMediaDisksA(
 	_In_ LPCSTR szProductCodeOrPatchCode,                      // Disks will be enumerated for this product or patch
 	_In_opt_ LPCSTR szUserSid,                                 // Account of this product/patch instance
@@ -2088,6 +2174,7 @@ UINT WINAPI MsiSourceListEnumMediaDisksW(
 
 // Give the version string and language for a specified file
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetFileVersionA(
 	_In_                               LPCSTR    szFilePath,       // path to the file
 	_Out_writes_opt_(*pcchVersionBuf)  LPSTR     lpVersionBuf,     // returned version string
@@ -2109,6 +2196,7 @@ UINT WINAPI MsiGetFileVersionW(
 
 #if (_WIN32_MSI >=  150)
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetFileHashA(
 	_In_ LPCSTR         szFilePath,  // path to the file
 	_In_ DWORD            dwOptions,   // options
@@ -2128,6 +2216,7 @@ UINT WINAPI MsiGetFileHashW(
 #if (_WIN32_MSI >= 150)
 #ifndef _MSI_NO_CRYPTO
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT WINAPI MsiGetFileSignatureInformationA(
 	_In_ LPCSTR szSignedObjectPath,                     // path to the signed object
 	_In_ DWORD dwFlags,                                   // special extra error case flags
@@ -2159,6 +2248,7 @@ HRESULT WINAPI MsiGetFileSignatureInformationW(
 // examine a shortcut, and retrieve its descriptor information 
 // if available.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiGetShortcutTargetA(
 	_In_                                   LPCSTR    szShortcutPath,    // full file path for the shortcut
 	_Out_writes_opt_(MAX_GUID_CHARS+1)     LPSTR     szProductCode,     // returned product code   - GUID
@@ -2182,12 +2272,14 @@ UINT WINAPI MsiGetShortcutTargetW(
 // checks to see if a product is managed
 // checks per-machine if called from system context, per-user if from
 // user context
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiIsProductElevatedA(
 	_In_ LPCSTR szProduct,  // product code
 	_Out_ BOOL *pfElevated);  // result   
 // checks to see if a product is managed
 // checks per-machine if called from system context, per-user if from
 // user context
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiIsProductElevatedW(
 	_In_ LPCWSTR szProduct,  // product code
 	_Out_ BOOL *pfElevated);  // result   
@@ -2204,8 +2296,10 @@ UINT WINAPI MsiIsProductElevatedW(
 
 // Caller notifies us of a user who's been moved and results in a sid change.
 
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiNotifySidChangeA(_In_ LPCSTR pOldSid,
                                 _In_ LPCSTR pNewSid);
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiNotifySidChangeW(_In_ LPCWSTR pOldSid,
                                 _In_ LPCWSTR pNewSid);
 #ifdef UNICODE
@@ -2223,12 +2317,14 @@ UINT WINAPI MsiNotifySidChangeW(_In_ LPCWSTR pOldSid,
 #if (_WIN32_MSI >=  450)
 
 // Start a multi package transaction
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiBeginTransactionA(
 	_In_ LPCSTR szName,                         // A name to identify the transaction
 	_In_ DWORD dwTransactionAttributes,           // Attributes to control the nature of the transaction
 	_Out_ MSIHANDLE* phTransactionHandle,         // Transaction handle
 	_Out_ HANDLE* phChangeOfOwnerEvent);          // Change Of Owner Event handle
 // Start a multi package transaction
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiBeginTransactionW(
 	_In_ LPCWSTR szName,                         // A name to identify the transaction
 	_In_ DWORD dwTransactionAttributes,           // Attributes to control the nature of the transaction
@@ -2241,10 +2337,12 @@ UINT WINAPI MsiBeginTransactionW(
 #endif // !UNICODE
 
 // End a multi package transaction
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiEndTransaction(
 	_In_ DWORD dwTransactionState);      // Flags to indicate how to end a transaction
 
 // Join an already created multi package transaction
+_Win32_metadata_supported_os_(windows8.0)
 UINT WINAPI MsiJoinTransaction(
 	_In_ MSIHANDLE hTransactionHandle,   //Transaction handle returned by MsiBeginTransaction
 	_In_ DWORD dwTransactionAttributes,  //Transaction attributes
