diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MspEnum.h b/generation/WinSDK/RecompiledIdlHeaders/um/MspEnum.h
index b8b83f12..41c39115 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MspEnum.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MspEnum.h
@@ -7,6 +7,7 @@
 #ifndef _MSPENUM_H_
 #define _MSPENUM_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
