diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/mstcpip.h b/generation/WinSDK/RecompiledIdlHeaders/shared/mstcpip.h
index 013e9001..ac93da8e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/mstcpip.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/mstcpip.h
@@ -64,6 +64,7 @@ DEFINE_GUID(
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -1907,6 +1908,7 @@ INET_UNCANONICALIZE_SCOPE_ID(
 #pragma region Desktop Family or Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PSTR
 NTAPI
@@ -1925,6 +1927,7 @@ RtlIpv4AddressToStringExA(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -1933,6 +1936,7 @@ RtlIpv4AddressToStringW (
     _Out_writes_(16) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -1943,6 +1947,7 @@ RtlIpv4AddressToStringExW(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -1963,6 +1968,7 @@ RtlIpv4StringToAddressExA (
     _Out_ PUSHORT Port
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -1973,6 +1979,7 @@ RtlIpv4StringToAddressW (
     _Out_ struct in_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -2004,6 +2011,7 @@ RtlIpv4StringToAddressExW (
 #pragma region Desktop Family or Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PSTR
 NTAPI
@@ -2023,6 +2031,7 @@ RtlIpv6AddressToStringExA(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -2031,6 +2040,7 @@ RtlIpv6AddressToStringW (
     _Out_writes_(46) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -2042,6 +2052,7 @@ RtlIpv6AddressToStringExW(
     _Inout_ PULONG AddressStringLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -2061,6 +2072,7 @@ RtlIpv6StringToAddressExA (
     _Out_ PUSHORT Port
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -2070,6 +2082,7 @@ RtlIpv6StringToAddressW (
     _Out_ struct in6_addr *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 NTSYSAPI
 LONG
 NTAPI
@@ -2107,6 +2120,7 @@ RtlIpv6StringToAddressExW (
 union _DL_EUI48;
 typedef union _DL_EUI48 DL_EUI48, *PDL_EUI48;
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 PSTR
 NTAPI
@@ -2115,6 +2129,7 @@ RtlEthernetAddressToStringA (
     _Out_writes_(18) PSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 PWSTR
 NTAPI
@@ -2123,6 +2138,7 @@ RtlEthernetAddressToStringW (
     _Out_writes_(18) PWSTR S
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 LONG
 NTAPI
@@ -2132,6 +2148,7 @@ RtlEthernetStringToAddressA (
     _Out_ DL_EUI48 *Addr
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 NTSYSAPI
 LONG
 NTAPI
