diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
index 848fdbd9..9c191b86 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
@@ -22,6 +22,9 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if !defined(_WINSOCK_DEPRECATED_BY)
 #if ((defined(_WINSOCK_DEPRECATED_NO_WARNINGS) || defined(BUILD_WINDOWS)) && !defined(_WINSOCK_DEPRECATE_WARNINGS)) || defined(MIDL_PASS)
@@ -125,6 +128,8 @@ extern "C" {
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 _WINSOCK_DEPRECATED_BY("WSARecv()")
 #if(_WIN32_WINNT < 0x0600)
 int
@@ -136,6 +141,7 @@ WSARecvEx(
     _Inout_ int FAR *flags
     );
 #else //(_WIN32_WINNT < 0x0600)
+_Win32_metadata_supported_os_(windows5.0)
 INT
 PASCAL FAR
 WSARecvEx(
@@ -162,6 +168,7 @@ typedef struct _TRANSMIT_FILE_BUFFERS {
 #define TF_USE_SYSTEM_THREAD  0x10
 #define TF_USE_KERNEL_APC     0x20
 
+_Win32_metadata_supported_os_(windows8.1)
 BOOL
 PASCAL FAR
 TransmitFile (
@@ -174,6 +181,7 @@ TransmitFile (
     _In_  DWORD dwReserved
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 BOOL
 PASCAL FAR
 AcceptEx (
@@ -187,6 +195,7 @@ AcceptEx (
     _Inout_ LPOVERLAPPED lpOverlapped
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 VOID
 PASCAL FAR
 GetAcceptExSockaddrs (
