diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
index 7134e305..3a4fcb1a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
@@ -34,7 +34,8 @@ CreatePrivateNamespaceW(
     _In_opt_ LPSECURITY_ATTRIBUTES lpPrivateNamespaceAttributes,
     _In_ LPVOID lpBoundaryDescriptor,
     _In_ LPCWSTR lpAliasPrefix
-    );
+    )
+    _Win32_metadata_raii_free_(ClosePrivateNamespace);
 
 WINBASEAPI
 HANDLE
@@ -42,8 +43,10 @@ WINAPI
 OpenPrivateNamespaceW(
     _In_ LPVOID lpBoundaryDescriptor,
     _In_ LPCWSTR lpAliasPrefix
-    );
+    )
+    _Win32_metadata_raii_free_(ClosePrivateNamespace);
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOLEAN
 WINAPI
@@ -58,8 +61,10 @@ WINAPI
 CreateBoundaryDescriptorW(
     _In_ LPCWSTR Name,
     _In_ ULONG Flags
-    );
+    )
+    _Win32_metadata_raii_free_(DeleteBoundaryDescriptor);
 
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
