diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
index 7134e305..4df235f0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/namespaceapi.h
@@ -12,6 +12,7 @@
 
 #ifndef _APISETNAMESPACE_
 #define _APISETNAMESPACE_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -34,7 +35,8 @@ CreatePrivateNamespaceW(
     _In_opt_ LPSECURITY_ATTRIBUTES lpPrivateNamespaceAttributes,
     _In_ LPVOID lpBoundaryDescriptor,
     _In_ LPCWSTR lpAliasPrefix
-    );
+    )
+    _Win32_metadata_raii_free_(ClosePrivateNamespace);
 
 WINBASEAPI
 HANDLE
@@ -42,8 +44,11 @@ WINAPI
 OpenPrivateNamespaceW(
     _In_ LPVOID lpBoundaryDescriptor,
     _In_ LPCWSTR lpAliasPrefix
-    );
+    )
+    _Win32_metadata_raii_free_(ClosePrivateNamespace);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOLEAN
 WINAPI
@@ -58,8 +63,11 @@ WINAPI
 CreateBoundaryDescriptorW(
     _In_ LPCWSTR Name,
     _In_ ULONG Flags
-    );
+    )
+    _Win32_metadata_raii_free_(DeleteBoundaryDescriptor);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -68,6 +76,7 @@ AddSIDToBoundaryDescriptor(
     _In_ PSID RequiredSid
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
