diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h b/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
index 7e75679f..ee5a56db 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
@@ -267,6 +267,9 @@ NCryptCreateProtectionDescriptor(
     _In_            LPCWSTR                             pwszDescriptorString,
                     DWORD                               dwFlags,
     _Out_           NCRYPT_DESCRIPTOR_HANDLE            *phDescriptor
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(NCryptCloseProtectionDescriptor)
     );
 
 //
@@ -501,7 +504,10 @@ NCryptProtectSecret(
 SECURITY_STATUS
 WINAPI
 NCryptUnprotectSecret(
-    _Out_opt_       NCRYPT_DESCRIPTOR_HANDLE            *phDescriptor,
+    _Out_opt_       NCRYPT_DESCRIPTOR_HANDLE            *phDescriptor
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(NCryptCloseProtectionDescriptor),
      _In_            NCRYPT_FLAGS                               dwFlags,
     _In_reads_bytes_(cbProtectedBlob)
                     const BYTE                          *pbProtectedBlob,
@@ -635,6 +641,9 @@ NCryptStreamOpenToProtect(
     _In_opt_        HWND                    hWnd,
     _In_            NCRYPT_PROTECT_STREAM_INFO          *pStreamInfo,
     _Out_           NCRYPT_STREAM_HANDLE                *phStream
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(NCryptStreamClose)
     );
 
 /****************************************************************************
@@ -679,6 +688,9 @@ NCryptStreamOpenToUnprotect(
                     DWORD                               dwFlags,
     _In_opt_        HWND                hWnd,
     _Out_           NCRYPT_STREAM_HANDLE                *phStream
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(NCryptStreamClose)
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINTHRESHOLD)
@@ -796,6 +808,9 @@ NCryptStreamOpenToUnprotectEx(
                     DWORD                               dwFlags,
     _In_opt_        HWND                                hWnd,
     _Out_           NCRYPT_STREAM_HANDLE                *phStream
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(NCryptStreamClose)
 );
 
 #endif
