diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h b/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
index 2d30728caaa48445ec42a06fcb7a09e3b991f06b..7e75679f1e978226a143dd3b3015bb4bf3d679d0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ncryptprotect.h
@@ -502,7 +502,7 @@ SECURITY_STATUS
 WINAPI
 NCryptUnprotectSecret(
     _Out_opt_       NCRYPT_DESCRIPTOR_HANDLE            *phDescriptor,
-     _In_            DWORD                               dwFlags,
+     _In_            NCRYPT_FLAGS                               dwFlags,
     _In_reads_bytes_(cbProtectedBlob)
                     const BYTE                          *pbProtectedBlob,
                     ULONG                               cbProtectedBlob,
