diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h b/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
index 9c44a5b0..98c69969 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
@@ -55,6 +55,7 @@ Notes:
 #define _NETIOAPI_H_
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
@@ -223,6 +224,7 @@ typedef struct _MIB_IF_TABLE2 {
     MIB_IF_ROW2 Table[ANY_SIZE];
 } MIB_IF_TABLE2, *PMIB_IF_TABLE2;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -270,6 +272,7 @@ typedef enum _MIB_IF_ENTRY_LEVEL {
     MibIfEntryNormalWithoutStatistics = 2
 } MIB_IF_ENTRY_LEVEL, *PMIB_IF_ENTRY_LEVEL;
 
+_Win32_metadata_supported_os_(windows10.0.15063)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -312,6 +315,7 @@ Notes:
 --*/
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -350,6 +354,7 @@ typedef enum _MIB_IF_TABLE_LEVEL {
     #endif
 } MIB_IF_TABLE_LEVEL, *PMIB_IF_TABLE_LEVEL;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -496,6 +501,7 @@ typedef struct _MIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES {
 } MIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES,
   *PMIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -504,6 +510,7 @@ GetIfStackTable(
     _Outptr_ PMIB_IFSTACK_TABLE *Table
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -512,6 +519,7 @@ GetInvertedIfStackTable(
     _Outptr_ PMIB_INVERTEDIFSTACK_TABLE *Table
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -547,6 +555,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -586,6 +595,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
@@ -615,6 +625,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -667,6 +678,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -703,6 +715,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows8.0)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -793,6 +806,7 @@ VOID
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -829,6 +843,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -860,6 +875,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -900,6 +916,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -946,6 +963,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
@@ -983,6 +1001,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1042,6 +1061,7 @@ VOID
     _In_ PMIB_UNICASTIPADDRESS_TABLE AddressTable
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1060,6 +1080,7 @@ NotifyStableUnicastIpAddressTable(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1119,6 +1140,7 @@ typedef struct _MIB_ANYCASTIPADDRESS_TABLE {
     MIB_ANYCASTIPADDRESS_ROW Table[ANY_SIZE];
 } MIB_ANYCASTIPADDRESS_TABLE, *PMIB_ANYCASTIPADDRESS_TABLE;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1150,6 +1172,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1181,6 +1204,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1215,6 +1239,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1278,6 +1303,7 @@ typedef struct _MIB_MULTICASTIPADDRESS_TABLE {
     MIB_MULTICASTIPADDRESS_ROW Table[ANY_SIZE];
 } MIB_MULTICASTIPADDRESS_TABLE, *PMIB_MULTICASTIPADDRESS_TABLE;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1312,6 +1338,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1416,6 +1443,7 @@ VOID
     _In_ MIB_NOTIFICATION_TYPE NotificationType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1453,6 +1481,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1491,6 +1520,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1548,6 +1578,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1590,6 +1621,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1635,6 +1667,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
@@ -1664,6 +1697,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1716,6 +1750,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1809,6 +1844,7 @@ typedef struct _MIB_IPPATH_TABLE {
 } MIB_IPPATH_TABLE, *PMIB_IPPATH_TABLE;
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1840,6 +1876,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1874,6 +1911,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1955,6 +1993,7 @@ typedef struct _MIB_IPNET_TABLE2 {
     MIB_IPNET_ROW2 Table[ANY_SIZE];
 } MIB_IPNET_TABLE2, *PMIB_IPNET_TABLE2;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -1987,6 +2026,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2019,6 +2059,7 @@ Notes:
 --*/
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2056,6 +2097,7 @@ Return Value:
 --*/
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2090,6 +2132,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2129,6 +2172,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2170,6 +2214,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2221,6 +2266,7 @@ VOID
     _Inout_ MIB_NOTIFICATION_TYPE NotificationType
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2232,6 +2278,7 @@ NotifyTeredoPortChange(
     _Inout_ HANDLE *NotificationHandle
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2279,6 +2326,7 @@ Return Value:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
@@ -2308,6 +2356,7 @@ Notes:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 VOID
 NETIOAPI_API_
@@ -2336,6 +2385,7 @@ Return Value:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2457,6 +2507,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2485,6 +2536,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2513,6 +2565,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2544,6 +2597,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2575,6 +2629,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2603,6 +2658,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2631,6 +2687,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2662,6 +2719,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2690,6 +2748,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2718,6 +2777,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -2748,6 +2808,7 @@ Return Value:
 
 #define IF_NAMESIZE NDIS_IF_MAX_STRING_SIZE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 NET_IFINDEX
@@ -2771,6 +2832,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 PCHAR
@@ -3021,6 +3083,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 NET_IF_COMPARTMENT_ID
@@ -3111,6 +3174,7 @@ Return Value:
 
 #pragma warning(pop)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
 ConvertLengthToIpv4Mask(
@@ -3137,6 +3201,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
 ConvertIpv4MaskToLength(
@@ -3518,6 +3583,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -3544,6 +3610,7 @@ Return Value:
 
 --*/
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
@@ -3582,6 +3649,7 @@ VOID
     _In_ NL_NETWORK_CONNECTIVITY_HINT ConnectivityHint
     );
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
