diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/netlistmgr.h b/generation/WinSDK/RecompiledIdlHeaders/um/netlistmgr.h
index b1a28136..b649d4e6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/netlistmgr.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/netlistmgr.h
@@ -179,6 +179,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -326,6 +327,7 @@ EXTERN_C const IID IID_INetworkListManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00000-570F-4A9B-8D69-199FDBA5723B")
     INetworkListManager : public IDispatch
     {
@@ -587,6 +589,7 @@ EXTERN_C const IID IID_INetworkListManagerEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00001-570F-4A9B-8D69-199FDBA5723B")
     INetworkListManagerEvents : public IUnknown
     {
@@ -682,6 +685,7 @@ EXTERN_C const IID IID_INetwork;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00002-570F-4A9B-8D69-199FDBA5723B")
     INetwork : public IDispatch
     {
@@ -1276,6 +1280,7 @@ EXTERN_C const IID IID_IEnumNetworks;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00003-570F-4A9B-8D69-199FDBA5723B")
     IEnumNetworks : public IDispatch
     {
@@ -1489,6 +1494,7 @@ EXTERN_C const IID IID_INetworkEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00004-570F-4A9B-8D69-199FDBA5723B")
     INetworkEvents : public IUnknown
     {
@@ -1623,6 +1629,7 @@ EXTERN_C const IID IID_INetworkConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00005-570F-4A9B-8D69-199FDBA5723B")
     INetworkConnection : public IDispatch
     {
@@ -2065,6 +2072,7 @@ EXTERN_C const IID IID_IEnumNetworkConnections;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00006-570F-4A9B-8D69-199FDBA5723B")
     IEnumNetworkConnections : public IDispatch
     {
@@ -2274,6 +2282,7 @@ EXTERN_C const IID IID_INetworkConnectionEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("DCB00007-570F-4A9B-8D69-199FDBA5723B")
     INetworkConnectionEvents : public IUnknown
     {
@@ -2382,6 +2391,7 @@ EXTERN_C const IID IID_INetworkCostManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("DCB00008-570F-4A9B-8D69-199FDBA5723B")
     INetworkCostManager : public IUnknown
     {
@@ -2509,6 +2519,7 @@ EXTERN_C const IID IID_INetworkCostManagerEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("DCB00009-570F-4A9B-8D69-199FDBA5723B")
     INetworkCostManagerEvents : public IUnknown
     {
@@ -2613,6 +2624,7 @@ EXTERN_C const IID IID_INetworkConnectionCost;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("DCB0000a-570F-4A9B-8D69-199FDBA5723B")
     INetworkConnectionCost : public IUnknown
     {
@@ -2713,6 +2725,7 @@ EXTERN_C const IID IID_INetworkConnectionCostEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("DCB0000b-570F-4A9B-8D69-199FDBA5723B")
     INetworkConnectionCostEvents : public IUnknown
     {
@@ -2844,9 +2857,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* end of Additional Prototypes */
