diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/NetSh.h b/generation/WinSDK/RecompiledIdlHeaders/um/NetSh.h
index c15c101c..257e6c31 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/NetSh.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/NetSh.h
@@ -18,6 +18,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -310,6 +313,7 @@ DWORD
 
 typedef NS_DLL_INIT_FN *PNS_DLL_INIT_FN;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD WINAPI MatchEnumTag(
     IN  HANDLE             hModule,
     IN  LPCWSTR            pwcArg,
@@ -318,11 +322,13 @@ DWORD WINAPI MatchEnumTag(
     OUT PDWORD             pdwValue
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL WINAPI MatchToken(
     IN  LPCWSTR  pwszUserToken,
     IN  LPCWSTR  pwszCmdToken
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD WINAPI PreprocessCommand(
     _In_opt_                   HANDLE    hModule,
     _Inout_updates_(dwArgCount) LPWSTR   *ppwcArguments,
@@ -335,27 +341,32 @@ DWORD WINAPI PreprocessCommand(
     _Out_writes_opt_(dwArgCount-dwCurrentIndex)    DWORD    *pdwTagType
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD PrintError(
     IN  HANDLE  hModule, OPTIONAL
     IN  DWORD   dwErrId,
     ...
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD PrintMessageFromModule(
     IN  HANDLE  hModule,
     IN  DWORD   dwMsgId,
     ...
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD PrintMessage(
     IN  LPCWSTR  pwszFormat,
     ...
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD WINAPI RegisterContext(
     IN    CONST NS_CONTEXT_ATTRIBUTES *pChildContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD WINAPI RegisterHelper(
     IN    CONST GUID                 *pguidParentContext,
     IN    CONST NS_HELPER_ATTRIBUTES *pfnRegisterSubContext
