diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Npapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/Npapi.h
index 3e99aab7..3fb478b1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Npapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Npapi.h
@@ -25,6 +25,9 @@ Environment:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -43,6 +46,7 @@ extern "C" {
 //  CONNECTIONS
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPAddConnection (
     _In_     LPNETRESOURCEW lpNetResource,
@@ -57,6 +61,7 @@ typedef DWORD (APIENTRY *PF_NPAddConnection) (
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPAddConnection3 (
     _In_opt_ HWND           hwndOwner,
@@ -95,6 +100,7 @@ typedef DWORD (APIENTRY *PF_NPAddConnection4) (
     _In_     DWORD          cbUseOptions
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPCancelConnection (
     _In_ LPWSTR  lpName,
@@ -119,6 +125,7 @@ typedef DWORD (APIENTRY *PF_NPCancelConnection2) (
     _In_ DWORD   dwFlags
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetConnection (
     _In_ LPWSTR     lpLocalName,
@@ -136,6 +143,7 @@ typedef DWORD (APIENTRY *PF_NPGetConnection) (
 #define WNGETCON_CONNECTED      0x00000000
 #define WNGETCON_DISCONNECTED   0x00000001
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetConnection3 (
     _In_ LPCWSTR    lpLocalName,
@@ -152,6 +160,7 @@ typedef DWORD (APIENTRY *PF_NPGetConnection3) (
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetUniversalName (
     _In_ LPCWSTR    lpLocalPath,
@@ -167,6 +176,7 @@ typedef DWORD (APIENTRY *PF_NPGetUniversalName) (
     _Inout_ LPDWORD lpnBufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetConnectionPerformance (
     _In_  LPCWSTR                lpRemoteName,
@@ -179,6 +189,7 @@ typedef DWORD (APIENTRY *PF_NPGetConnectionPerformance) (
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPOpenEnum (
     _In_     DWORD          dwScope,
@@ -196,6 +207,7 @@ typedef DWORD (APIENTRY *PF_NPOpenEnum) (
     _Out_    LPHANDLE       lphEnum
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPEnumResource (
     _In_    HANDLE  hEnum,
@@ -211,6 +223,7 @@ typedef DWORD (APIENTRY *PF_NPEnumResource) (
     _Inout_ LPDWORD lpBufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPCloseEnum (
     _In_ HANDLE   hEnum
@@ -284,6 +297,7 @@ typedef DWORD (APIENTRY *PF_NPCloseEnum) (
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetCaps (
     _In_ DWORD ndex
@@ -297,6 +311,7 @@ typedef DWORD (APIENTRY *PF_NPGetCaps) (
 //  OTHER
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetUser (
     _In_    LPWSTR  lpName,
@@ -346,6 +361,7 @@ typedef DWORD (APIENTRY *PF_NPGetPersistentUseOptionsForConnection) (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPDeviceMode(
     _In_ HWND hParent
@@ -358,6 +374,7 @@ typedef DWORD (APIENTRY *PF_NPDeviceMode) (
 // flag for search dialog
 #define WNSRCH_REFRESH_FIRST_LEVEL 0x00000001
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPSearchDialog(
     _In_     HWND   hwndParent,
@@ -381,6 +398,7 @@ typedef DWORD (APIENTRY *PF_NPSearchDialog) (
 #pragma region Desktop Family or RemoteFS Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_REMOTEFS)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetResourceParent(
     _In_ LPNETRESOURCEW lpNetResource,
@@ -394,6 +412,7 @@ typedef DWORD (APIENTRY *PF_NPGetResourceParent) (
     _Inout_ LPDWORD lpBufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY NPGetResourceInformation(
     _In_ LPNETRESOURCEW lpNetResource,
     _Out_writes_bytes_(*lpBufferSize) LPVOID lpBuffer,
@@ -408,6 +427,7 @@ typedef DWORD (APIENTRY *PF_NPGetResourceInformation) (
     _Outptr_ LPWSTR *lplpSystem
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPFormatNetworkName(
     _In_    LPWSTR   lpRemoteName,
@@ -431,6 +451,7 @@ typedef DWORD (APIENTRY *PF_NPFormatNetworkName) (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetPropertyText(
     _In_ DWORD  iButton,
@@ -450,6 +471,7 @@ typedef DWORD (APIENTRY *PF_NPGetPropertyText) (
     _In_ DWORD  nType
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPPropertyDialog(
     _In_ HWND   hwndParent,
@@ -479,6 +501,7 @@ typedef DWORD (APIENTRY *PF_NPPropertyDialog) (
 #define WNDN_RMDIR    2
 #define WNDN_MVDIR    3
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPGetDirectoryType (
     _In_ LPWSTR  lpName,
@@ -492,6 +515,7 @@ typedef DWORD (APIENTRY *PF_NPGetDirectoryType) (
     _In_ BOOL    bFlushCache
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPDirectoryNotify (
     _In_ HWND    hwnd,
@@ -511,6 +535,7 @@ typedef DWORD (APIENTRY *PF_NPDirectoryNotify) (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 VOID
 WNetSetLastErrorA(
     _In_ DWORD   err,
@@ -551,6 +576,7 @@ WNetSetLastErrorW(
 #define WN_VALID_LOGON_ACCOUNT      0x00000001
 #define WN_NT_PASSWORD_CHANGED      0x00000002
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPLogonNotify (
     _In_     PLUID               lpLogonId,
@@ -574,6 +600,7 @@ typedef DWORD (APIENTRY *PF_NPLogonNotify) (
     _Out_    LPWSTR              *lpLogonScript
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPPasswordChangeNotify (
     _In_     LPCWSTR             lpAuthentInfoType,
@@ -626,6 +653,7 @@ typedef struct _NOTIFYCANCEL {
 } NOTIFYCANCEL, *LPNOTIFYCANCEL;
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 AddConnectNotify (
     _Inout_ LPNOTIFYINFO lpNotifyInfo,
@@ -637,6 +665,7 @@ typedef DWORD (APIENTRY *PF_AddConnectNotify) (
     _In_    LPNOTIFYADD  lpAddInfo
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 CancelConnectNotify (
     _Inout_ LPNOTIFYINFO   lpNotifyInfo,
@@ -659,6 +688,7 @@ typedef DWORD (APIENTRY *PF_CancelConnectNotify) (
 #define WNPERMC_AUDIT 0x00000002
 #define WNPERMC_OWNER 0x00000004
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPFMXGetPermCaps (
     _In_ LPWSTR lpDriveName
@@ -675,6 +705,7 @@ typedef DWORD (APIENTRY *PF_NPFMXGetPermCaps) (
 #define WNPERM_DLG_AUDIT  1
 #define WNPERM_DLG_OWNER  2
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPFMXEditPerm (
     _In_ LPWSTR lpDriveName,
@@ -688,6 +719,7 @@ typedef DWORD (APIENTRY *PF_NPFMXEditPerm) (
     _In_ DWORD  nDialogType
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 DWORD APIENTRY
 NPFMXGetPermHelp (
     _In_    LPWSTR  lpDriveName,
