diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
index 3bbc196f..7f7a88a8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
@@ -15,6 +15,9 @@ Revision History:
 --*/
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if _MSC_VER >= 1200
 #pragma warning(push)
@@ -5479,6 +5482,7 @@ typedef const PAUDIT_POLICY_INFORMATION PCAUDIT_POLICY_INFORMATION, LPCAUDIT_POL
 #define AUDIT_GENERIC_EXECUTE        (STANDARD_RIGHTS_EXECUTE)
 
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditSetSystemPolicy(
@@ -5486,6 +5490,7 @@ AuditSetSystemPolicy(
     _In_ ULONG dwPolicyCount
     );
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditSetPerUserPolicy(
@@ -5494,6 +5499,7 @@ AuditSetPerUserPolicy(
     _In_ ULONG dwPolicyCount
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5504,6 +5510,7 @@ AuditQuerySystemPolicy(
     _When_(return != 0, __drv_allocatesMem(Mem)) PAUDIT_POLICY_INFORMATION* ppAuditPolicy
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5515,6 +5522,7 @@ AuditQueryPerUserPolicy(
     _When_(return != 0, __drv_allocatesMem(Mem)) PAUDIT_POLICY_INFORMATION* ppAuditPolicy
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5522,6 +5530,7 @@ AuditEnumeratePerUserPolicy(
     _Out_ _When_(return != 0, __drv_allocatesMem(Mem)) PPOLICY_AUDIT_SID_ARRAY* ppAuditSidArray
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5533,6 +5542,7 @@ AuditComputeEffectivePolicyBySid(
     _When_(return != 0, __drv_allocatesMem(Mem)) PAUDIT_POLICY_INFORMATION* ppAuditPolicy
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5544,6 +5554,7 @@ AuditComputeEffectivePolicyByToken(
     _When_(return != 0, __drv_allocatesMem(Mem)) PAUDIT_POLICY_INFORMATION* ppAuditPolicy
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5553,6 +5564,7 @@ AuditEnumerateCategories(
     _Out_ PULONG pdwCountReturned
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5564,6 +5576,7 @@ AuditEnumerateSubCategories(
     _Out_ PULONG pdwCountReturned
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5572,6 +5585,7 @@ AuditLookupCategoryNameW(
     _Outptr_ _When_(return != 0, __drv_allocatesMem(Mem)) PWSTR* ppszCategoryName
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5586,6 +5600,7 @@ AuditLookupCategoryNameA(
 #define AuditLookupCategoryName  AuditLookupCategoryNameA
 #endif
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5594,6 +5609,7 @@ AuditLookupSubCategoryNameW(
     _Outptr_ _When_(return != 0, __drv_allocatesMem(Mem)) PWSTR* ppszSubCategoryName
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5608,6 +5624,7 @@ AuditLookupSubCategoryNameA(
 #define AuditLookupSubCategoryName  AuditLookupSubCategoryNameA
 #endif
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditLookupCategoryIdFromCategoryGuid(
@@ -5615,6 +5632,7 @@ AuditLookupCategoryIdFromCategoryGuid(
     _Out_ PPOLICY_AUDIT_EVENT_TYPE pAuditCategoryId
     );
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditLookupCategoryGuidFromCategoryId(
@@ -5622,6 +5640,7 @@ AuditLookupCategoryGuidFromCategoryId(
     _Out_ GUID* pAuditCategoryGuid
     );
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditSetSecurity(
@@ -5629,6 +5648,7 @@ AuditSetSecurity(
     _In_ PSECURITY_DESCRIPTOR pSecurityDescriptor
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5637,6 +5657,7 @@ AuditQuerySecurity(
     _Outptr_ _When_(return != 0, __drv_allocatesMem(Mem)) PSECURITY_DESCRIPTOR *ppSecurityDescriptor
     );
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditSetGlobalSaclW(
@@ -5644,6 +5665,7 @@ AuditSetGlobalSaclW(
     _In_opt_ PACL Acl
     );
 
+_Win32_metadata_set_last_error_
 BOOLEAN
 NTAPI
 AuditSetGlobalSaclA(
@@ -5657,6 +5679,7 @@ AuditSetGlobalSaclA(
 #define AuditSetGlobalSacl AuditSetGlobalSaclA
 #endif
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
@@ -5665,6 +5688,7 @@ AuditQueryGlobalSaclW(
     _Out_ _When_(return != 0, __drv_allocatesMem(Mem)) PACL *Acl
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 BOOLEAN
 NTAPI
