diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
index e076b65d..c4a771d3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecAPI.h
@@ -1685,7 +1685,10 @@ NTSTATUS
 NTAPI
 LsaRegisterLogonProcess (
     _In_ PLSA_STRING LogonProcessName,
-    _Out_ PHANDLE LsaHandle,
+    _Out_ PHANDLE LsaHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(LsaDeregisterLogonProcess),
     _Out_ PLSA_OPERATIONAL_MODE SecurityMode
     );
 
@@ -1710,7 +1713,10 @@ LsaLogonUser (
     _Out_       PVOID           *ProfileBuffer,
     _Out_       PULONG          ProfileBufferLength,
     _Inout_     PLUID           LogonId,
-    _Out_       PHANDLE         Token,
+    _Out_       PHANDLE         Token
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(CloseHandle),
     _Out_       PQUOTA_LIMITS   Quotas,
     _Out_       PNTSTATUS       SubStatus
     );
@@ -1766,6 +1772,9 @@ NTSTATUS
 NTAPI
 LsaConnectUntrusted (
     _Out_ PHANDLE LsaHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(LsaDeregisterLogonProcess)
     );
 
 __drv_sameIRQL
@@ -2972,6 +2981,9 @@ LsaOpenPolicy(
     _In_ PLSA_OBJECT_ATTRIBUTES ObjectAttributes,
     _In_ ACCESS_MASK DesiredAccess,
     _Out_ PLSA_HANDLE PolicyHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(LsaClose)
     );
 
 
@@ -3286,6 +3298,9 @@ LsaOpenTrustedDomainByName(
     _In_ PLSA_UNICODE_STRING TrustedDomainName,
     _In_ ACCESS_MASK DesiredAccess,
     _Out_ PLSA_HANDLE TrustedDomainHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(LsaClose)
     );
 
 
@@ -3350,6 +3365,9 @@ LsaCreateTrustedDomainEx(
     _In_ PTRUSTED_DOMAIN_AUTH_INFORMATION AuthenticationInformation,
     _In_ ACCESS_MASK DesiredAccess,
     _Out_ PLSA_HANDLE TrustedDomainHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(LsaClose)
     );
 
 #if (_WIN32_WINNT >= 0x0501)
