diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/oobenotification.h b/generation/WinSDK/RecompiledIdlHeaders/um/oobenotification.h
index da7a62e9..d1317bca 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/oobenotification.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/oobenotification.h
@@ -35,6 +35,7 @@
 #include <ntrtl.h>
 #include <nturtl.h>
 #include <windows.h>
+#include <win32metadata_annotations.h>
 #endif // defined(_CONTRACT_GEN)
 
 #ifdef __cplusplus
@@ -52,6 +53,7 @@ VOID
     _In_opt_ PVOID CallbackContext
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 _Success_(return != FALSE)
 BOOL
@@ -60,6 +62,7 @@ OOBEComplete(
     _Out_ PBOOL isOOBEComplete
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 _Success_(return != FALSE)
 BOOL
@@ -70,6 +73,7 @@ RegisterWaitUntilOOBECompleted(
     _Out_ PVOID* WaitHandle
     );
 
+_Win32_metadata_set_last_error_
 _Check_return_
 _Success_(return != FALSE)
 BOOL
