diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/opmapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/opmapi.h
index ea341a5e..9489e6a2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/opmapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/opmapi.h
@@ -67,6 +67,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #ifndef OPM_API_DEFINED
 #define OPM_API_DEFINED
 #pragma region Desktop Family
@@ -441,6 +442,7 @@ EXTERN_C const IID IID_IOPMVideoOutput;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("0A15159D-41C7-4456-93E1-284CD61D4E8D")
     IOPMVideoOutput : public IUnknown
     {
@@ -618,12 +620,14 @@ typedef enum {
 #define GetBusType( ulBusTypeAndImplementation )                        ((ulBusTypeAndImplementation) & OPM_BUS_TYPE_MASK)
 #define GetBusImplementation( ulBusTypeAndImplementation )              (((ulBusTypeAndImplementation) & OPM_BUS_IMPLEMENTATION_MODIFIER_MASK) >> 16) 
 #define IsNonStandardBusImplementation( ulBusTypeAndImplementation )    ((ulBusTypeAndImplementation) & OPM_BUS_IMPLEMENTATION_MODIFIER_NON_STANDARD) 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI OPMGetVideoOutputsFromHMONITOR(                                                      
     _In_ HMONITOR hMonitor,                                                                 
     _In_ OPM_VIDEO_OUTPUT_SEMANTICS vos,                                                    
     _Out_ ULONG* pulNumVideoOutputs,                                                        
     _Outptr_result_buffer_(*pulNumVideoOutputs) IOPMVideoOutput*** pppOPMVideoOutputArray       
     );                                                                                      
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI OPMGetVideoOutputForTarget(                                                          
     _In_ LUID* pAdapterLuid,                                                                
     _In_ ULONG VidPnTarget,                                                                 
@@ -631,6 +635,7 @@ STDAPI OPMGetVideoOutputForTarget(
     _COM_Outptr_ IOPMVideoOutput** ppOPMVideoOutput                                         
     );                                                                                      
 #ifndef DO_NOT_USE_DIRECTX_OR_DXVA2
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI OPMGetVideoOutputsFromIDirect3DDevice9Object(                                        
     _In_ IDirect3DDevice9* pDirect3DDevice9,                                                
     _In_ OPM_VIDEO_OUTPUT_SEMANTICS vos,                                                    
