diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Pdh.h b/generation/WinSDK/RecompiledIdlHeaders/um/Pdh.h
index 9058afda..d00da4c6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Pdh.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Pdh.h
@@ -18,6 +18,9 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family and or PerfCounter Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_PERFCOUNTER)
@@ -310,6 +313,7 @@ typedef struct _PDH_LOG_SERVICE_QUERY_INFO_W {
 
 // function definitions
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDllVersion(
     _Out_opt_ LPDWORD lpdwVersion
@@ -319,6 +323,7 @@ PdhGetDllVersion(
 //  Query Functions
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION   
 PdhOpenQueryW(
     _In_opt_ LPCWSTR      szDataSource,
@@ -326,6 +331,7 @@ PdhOpenQueryW(
     _Out_    PDH_HQUERY * phQuery
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhOpenQueryA(
     _In_opt_ LPCSTR       szDataSource,
@@ -333,6 +339,7 @@ PdhOpenQueryA(
     _Out_    PDH_HQUERY * phQuery
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhAddCounterW(
     _In_  PDH_HQUERY     hQuery,
@@ -341,6 +348,7 @@ PdhAddCounterW(
     _Out_ PDH_HCOUNTER * phCounter
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhAddCounterA(
     _In_  PDH_HQUERY     hQuery,
@@ -350,6 +358,7 @@ PdhAddCounterA(
 );
 
 #if (_WIN32_WINDOWS >= 0x0600 || _WIN32_WINNT >= 0x0600 || (defined(NTDDI_VERSION) && NTDDI_VERSION >= NTDDI_VISTA))
+_Win32_metadata_supported_os_(windows6.0.6000)
 PDH_FUNCTION
 PdhAddEnglishCounterW(
     _In_  PDH_HQUERY     hQuery,
@@ -358,6 +367,7 @@ PdhAddEnglishCounterW(
     _Out_ PDH_HCOUNTER * phCounter
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 PDH_FUNCTION
 PdhAddEnglishCounterA(
     _In_  PDH_HQUERY     hQuery,
@@ -366,18 +376,21 @@ PdhAddEnglishCounterA(
     _Out_ PDH_HCOUNTER * phCounter
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 PDH_FUNCTION
 PdhCollectQueryDataWithTime(
     _Inout_ PDH_HQUERY   hQuery,
     _Out_   LONGLONG   * pllTimeStamp
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 PDH_FUNCTION
 PdhValidatePathExW(
     _In_opt_ PDH_HLOG hDataSource,
     _In_     LPCWSTR  szFullPathBuffer
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 PDH_FUNCTION
 PdhValidatePathExA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -385,16 +398,19 @@ PdhValidatePathExA(
 );
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhRemoveCounter(
     _In_ PDH_HCOUNTER hCounter
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhCollectQueryData(
     _Inout_ PDH_HQUERY hQuery
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhCloseQuery(
     _Inout_ PDH_HQUERY hQuery
@@ -404,6 +420,7 @@ PdhCloseQuery(
 //  Counter Functions
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetFormattedCounterValue(
     _In_      PDH_HCOUNTER          hCounter,
@@ -412,6 +429,7 @@ PdhGetFormattedCounterValue(
     _Out_     PPDH_FMT_COUNTERVALUE pValue
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetFormattedCounterArrayA(
@@ -422,6 +440,7 @@ PdhGetFormattedCounterArrayA(
     _Out_writes_bytes_opt_(* lpdwBufferSize) PPDH_FMT_COUNTERVALUE_ITEM_A ItemBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetFormattedCounterArrayW(
@@ -447,6 +466,7 @@ PdhGetFormattedCounterArrayW(
 #define PERF_DETAIL_COSTLY   ((DWORD) 0x00010000)
 #define PERF_DETAIL_STANDARD ((DWORD) 0x0000FFFF)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetRawCounterValue(
     _In_      PDH_HCOUNTER      hCounter,
@@ -454,6 +474,7 @@ PdhGetRawCounterValue(
     _Out_     PPDH_RAW_COUNTER  pValue
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetRawCounterArrayA(
@@ -463,6 +484,7 @@ PdhGetRawCounterArrayA(
     _Out_writes_bytes_opt_(* lpdwBufferSize) PPDH_RAW_COUNTER_ITEM_A ItemBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetRawCounterArrayW(
@@ -472,6 +494,7 @@ PdhGetRawCounterArrayW(
     _Out_writes_bytes_opt_(* lpdwBufferSize) PPDH_RAW_COUNTER_ITEM_W ItemBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhCalculateCounterFromRawValue(
     _In_  PDH_HCOUNTER          hCounter,
@@ -481,6 +504,7 @@ PdhCalculateCounterFromRawValue(
     _Out_ PPDH_FMT_COUNTERVALUE fmtValue
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhComputeCounterStatistics(
     _In_  PDH_HCOUNTER     hCounter,
@@ -491,6 +515,7 @@ PdhComputeCounterStatistics(
     _Out_ PPDH_STATISTICS  data
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetCounterInfoW(
@@ -500,6 +525,7 @@ PdhGetCounterInfoW(
     _Out_writes_bytes_opt_(* pdwBufferSize) PPDH_COUNTER_INFO_W lpBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetCounterInfoA(
@@ -512,6 +538,7 @@ PdhGetCounterInfoA(
 #define PDH_MAX_SCALE  (7L)
 #define PDH_MIN_SCALE (-7L)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhSetCounterScaleFactor(
     _Inout_ PDH_HCOUNTER hCounter,
@@ -520,16 +547,19 @@ PdhSetCounterScaleFactor(
 //
 //   Browsing and enumeration functions
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhConnectMachineW(
     _In_opt_ LPCWSTR szMachineName
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhConnectMachineA(
     _In_opt_ LPCSTR szMachineName
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumMachinesW(
     _In_opt_ LPCWSTR szDataSource,
@@ -537,6 +567,7 @@ PdhEnumMachinesW(
     _Inout_ LPDWORD pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumMachinesA(
     _In_opt_ LPCSTR  szDataSource,
@@ -544,6 +575,7 @@ PdhEnumMachinesA(
     _Inout_ LPDWORD pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectsW(
     _In_opt_ LPCWSTR szDataSource,
@@ -554,6 +586,7 @@ PdhEnumObjectsW(
     _In_     BOOL    bRefresh
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectsA(
     _In_opt_ LPCSTR  szDataSource,
@@ -564,6 +597,7 @@ PdhEnumObjectsA(
     _In_     BOOL    bRefresh
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectItemsW(
     _In_opt_ LPCWSTR szDataSource,
@@ -577,6 +611,7 @@ PdhEnumObjectItemsW(
     _In_     DWORD   dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectItemsA(
     _In_opt_ LPCSTR  szDataSource,
@@ -590,6 +625,7 @@ PdhEnumObjectItemsA(
     _In_     DWORD   dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhMakeCounterPathW(
     _In_    PPDH_COUNTER_PATH_ELEMENTS_W pCounterPathElements,
@@ -598,6 +634,7 @@ PdhMakeCounterPathW(
     _In_    DWORD                        dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhMakeCounterPathA(
     _In_    PPDH_COUNTER_PATH_ELEMENTS_A pCounterPathElements,
@@ -606,6 +643,7 @@ PdhMakeCounterPathA(
     _In_    DWORD                        dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhParseCounterPathW(
@@ -615,6 +653,7 @@ PdhParseCounterPathW(
     _In_    DWORD                     dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhParseCounterPathA(
@@ -629,6 +668,7 @@ PdhParseCounterPathA(
 
 #define PDH_PATH_LANG_FLAGS(LangId, Flags)  ((DWORD)(((LangId & 0x0000FFFF) << 16) | (Flags & 0x0000FFFF)))
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhParseInstanceNameW(
     _In_    LPCWSTR szInstanceString,
@@ -639,6 +679,7 @@ PdhParseInstanceNameW(
     _Out_   LPDWORD lpIndex
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhParseInstanceNameA(
     _In_    LPCSTR  szInstanceString,
@@ -649,16 +690,19 @@ PdhParseInstanceNameA(
     _Out_   LPDWORD lpIndex
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhValidatePathW(
     _In_ LPCWSTR szFullPathBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhValidatePathA(
     _In_ LPCSTR szFullPathBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfObjectW(
     _In_opt_ LPCWSTR szDataSource,
@@ -667,6 +711,7 @@ PdhGetDefaultPerfObjectW(
     _Inout_  LPDWORD pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfObjectA(
     _In_opt_ LPCSTR  szDataSource,
@@ -675,6 +720,7 @@ PdhGetDefaultPerfObjectA(
     _Inout_  LPDWORD pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfCounterW(
     _In_opt_ LPCWSTR szDataSource,
@@ -684,6 +730,7 @@ PdhGetDefaultPerfCounterW(
     _Inout_  LPDWORD pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfCounterA(
     _In_opt_ LPCSTR  szDataSource,
@@ -857,16 +904,19 @@ typedef struct _BrowseDlgConfig_A {
 #pragma warning ( default : 4214 )
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBrowseCountersW(
     _In_ PPDH_BROWSE_DLG_CONFIG_W pBrowseDlgData
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBrowseCountersA(
     _In_ PPDH_BROWSE_DLG_CONFIG_A pBrowseDlgData
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandCounterPathW(
     _In_    LPCWSTR szWildCardPath,
@@ -874,6 +924,7 @@ PdhExpandCounterPathW(
     _Inout_ LPDWORD pcchPathListLength
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandCounterPathA(
     _In_    LPCSTR  szWildCardPath,
@@ -884,6 +935,7 @@ PdhExpandCounterPathA(
 //
 //  v2.0 functions
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhLookupPerfNameByIndexW(
     _In_opt_ LPCWSTR szMachineName,
@@ -892,6 +944,7 @@ PdhLookupPerfNameByIndexW(
     _Inout_  LPDWORD pcchNameBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhLookupPerfNameByIndexA(
     _In_opt_ LPCSTR  szMachineName,
@@ -900,6 +953,7 @@ PdhLookupPerfNameByIndexA(
     _Inout_  LPDWORD pcchNameBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhLookupPerfIndexByNameW(
     _In_opt_ LPCWSTR szMachineName,
@@ -907,6 +961,7 @@ PdhLookupPerfIndexByNameW(
     _Out_    LPDWORD pdwIndex
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhLookupPerfIndexByNameA(
     _In_opt_ LPCSTR  szMachineName,
@@ -920,6 +975,7 @@ PdhLookupPerfIndexByNameA(
 #define PDH_REFRESHCOUNTERS     4
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandWildCardPathA(
     _In_opt_ LPCSTR  szDataSource,
@@ -929,6 +985,7 @@ PdhExpandWildCardPathA(
     _In_     DWORD   dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandWildCardPathW(
     _In_opt_ LPCWSTR szDataSource,
@@ -975,6 +1032,7 @@ PdhExpandWildCardPathW(
 #define PDH_LOG_TYPE_BINARY         3   // Obsolete - not supported.
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhOpenLogW(
     _In_     LPCWSTR     szLogFileName,
@@ -986,6 +1044,7 @@ PdhOpenLogW(
     _Out_    PDH_HLOG  * phLog
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhOpenLogA(
     _In_     LPCSTR       szLogFileName,
@@ -997,29 +1056,34 @@ PdhOpenLogA(
     _Out_    PDH_HLOG   * phLog
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhUpdateLogW(
     _In_     PDH_HLOG hLog,
     _In_opt_ LPCWSTR  szUserString
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhUpdateLogA(
     _In_     PDH_HLOG hLog,
     _In_opt_ LPCSTR   szUserString
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhUpdateLogFileCatalog(
     _In_ PDH_HLOG hLog
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetLogFileSize(
     _In_  PDH_HLOG   hLog,
     _Out_ LONGLONG * llSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhCloseLog(
     _In_ PDH_HLOG hLog,
@@ -1032,6 +1096,7 @@ PdhCloseLog(
 //
 #define PDH_FLAGS_FILE_BROWSER_ONLY ((DWORD) 0x00000001)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhSelectDataSourceW(
     _In_    HWND    hWndOwner,
@@ -1040,6 +1105,7 @@ PdhSelectDataSourceW(
     _Inout_ LPDWORD pcchBufferLength
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhSelectDataSourceA(
     _In_    HWND    hWndOwner,
@@ -1048,18 +1114,21 @@ PdhSelectDataSourceA(
     _Inout_ LPDWORD pcchBufferLength
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 __stdcall
 PdhIsRealTimeQuery(
     _In_ PDH_HQUERY hQuery
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhSetQueryTimeRange(
     _In_ PDH_HQUERY     hQuery,
     _In_ PPDH_TIME_INFO pInfo
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetDataSourceTimeRangeW(
@@ -1069,6 +1138,7 @@ PdhGetDataSourceTimeRangeW(
     _Inout_  LPDWORD  pdwBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetDataSourceTimeRangeA(
@@ -1078,6 +1148,7 @@ PdhGetDataSourceTimeRangeA(
     _Inout_  LPDWORD  pdwBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhCollectQueryDataEx(
     _In_ PDH_HQUERY hQuery,
@@ -1085,6 +1156,7 @@ PdhCollectQueryDataEx(
     _In_ HANDLE     hNewDataEvent
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhFormatFromRawValue(
     _In_  DWORD                   dwCounterType,
@@ -1095,12 +1167,14 @@ PdhFormatFromRawValue(
     _Out_ PPDH_FMT_COUNTERVALUE   pFmtValue
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetCounterTimeBase(
     _In_  PDH_HCOUNTER   hCounter,
     _Out_ LONGLONG     * pTimeBase
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhReadRawLogRecord(
@@ -1115,6 +1189,7 @@ PdhReadRawLogRecord(
 #define DATA_SOURCE_LOGFILE  ((DWORD) 0x00000002)
 #define DATA_SOURCE_WBEM     ((DWORD) 0x00000004)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhSetDefaultRealTimeDataSource(
     _In_ DWORD dwDataSourceId
@@ -1123,18 +1198,21 @@ PdhSetDefaultRealTimeDataSource(
 #if (_WIN32_WINDOWS >= 0x0501 || _WIN32_WINNT >= 0x0501 || (defined(NTDDI_VERSION) && NTDDI_VERSION >= NTDDI_WINXP))
 // Extended API for WMI event trace logfile format
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBindInputDataSourceW(
     _Out_    PDH_HLOG * phDataSource,
     _In_opt_ LPCWSTR    LogFileNameList
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBindInputDataSourceA(
     _Out_    PDH_HLOG * phDataSource,
     _In_opt_ LPCSTR     LogFileNameList
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhOpenQueryH(
     _In_opt_  PDH_HLOG     hDataSource,
@@ -1142,6 +1220,7 @@ PdhOpenQueryH(
     _Out_     PDH_HQUERY * phQuery
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumMachinesHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1149,6 +1228,7 @@ PdhEnumMachinesHW(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumMachinesHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1156,6 +1236,7 @@ PdhEnumMachinesHA(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectsHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1166,6 +1247,7 @@ PdhEnumObjectsHW(
     _In_     BOOL     bRefresh
 ) ;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectsHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1176,6 +1258,7 @@ PdhEnumObjectsHA(
     _In_     BOOL     bRefresh
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectItemsHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1189,6 +1272,7 @@ PdhEnumObjectItemsHW(
     _In_     DWORD    dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumObjectItemsHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1202,6 +1286,7 @@ PdhEnumObjectItemsHA(
     _In_     DWORD    dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandWildCardPathHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1211,6 +1296,7 @@ PdhExpandWildCardPathHW(
     _In_    DWORD    dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhExpandWildCardPathHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1220,6 +1306,7 @@ PdhExpandWildCardPathHA(
     _In_    DWORD    dwFlags
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return == ERROR_SUCCESS)
 PDH_FUNCTION
 PdhGetDataSourceTimeRangeH(
@@ -1229,6 +1316,7 @@ PdhGetDataSourceTimeRangeH(
     _Inout_  LPDWORD  pdwBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfObjectHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1237,6 +1325,7 @@ PdhGetDefaultPerfObjectHW(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfObjectHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1245,6 +1334,7 @@ PdhGetDefaultPerfObjectHA(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfCounterHW(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1254,6 +1344,7 @@ PdhGetDefaultPerfCounterHW(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhGetDefaultPerfCounterHA(
     _In_opt_ PDH_HLOG hDataSource,
@@ -1263,11 +1354,13 @@ PdhGetDefaultPerfCounterHA(
     _Inout_  LPDWORD  pcchBufferSize
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBrowseCountersHW(
     _In_ PPDH_BROWSE_DLG_CONFIG_HW pBrowseDlgData
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhBrowseCountersHA(
     _In_ PPDH_BROWSE_DLG_CONFIG_HA pBrowseDlgData
@@ -1297,6 +1390,7 @@ PdhCreateSQLTablesA(
 );     
 
 //Return the list of Log set names in the database pointed to by the DSN.
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumLogSetNamesW(
     _In_    LPCWSTR szDataSource,
@@ -1304,6 +1398,7 @@ PdhEnumLogSetNamesW(
     _Inout_ LPDWORD pcchBufferLength
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 PDH_FUNCTION
 PdhEnumLogSetNamesA(
     _In_    LPCSTR  szDataSource,
