diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h b/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
index 5b0cae8d..5179c49d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
@@ -15,6 +15,9 @@ Revision History:
 --*/
 #ifndef _PRINTASYNCNOTIFY_H_
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -170,6 +173,7 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyCallback, IUnknown)
 
 #undef INTERFACE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT
 WINAPI
 RegisterForPrintAsyncNotifications(
@@ -179,14 +183,17 @@ RegisterForPrintAsyncNotifications(
              PrintAsyncNotifyConversationStyle eConversationStyle,
     _In_     IPrintAsyncNotifyCallback*        pCallback,
     _Out_    HANDLE*                           phNotify
+        _Win32_metadata_raii_free_(UnRegisterForPrintAsyncNotifications)
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT
 WINAPI
 UnRegisterForPrintAsyncNotifications(
     _In_     HANDLE
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT
 WINAPI
 CreatePrintAsyncNotifyChannel(
