diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Provider.h b/generation/WinSDK/RecompiledIdlHeaders/um/Provider.h
index a6986599..460b6d62 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Provider.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Provider.h
@@ -16,6 +16,7 @@
 #define _PROVIDER_H__
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -67,12 +68,14 @@ protected:
     // you will be given an object with the key properties filled in
     // you need to fill in all of the rest of the properties, or
     // return WBEM_E_NOT_FOUND if the object doesn't exist.
+_Win32_metadata_supported_os_(windows5.0)
     virtual HRESULT GetObject(_Inout_ CInstance *pInstance, long lFlags = 0L);
 
     // You will be given an object with the key properties filled in.
     // You can either fill in all the properties, or check the Query object
     // to see what properties are required.  If you don't implement this method, the
     // GetObject(CInstance, long) method will be called instead.
+_Win32_metadata_supported_os_(windows5.0)
     virtual HRESULT GetObject(_Inout_ CInstance *pInstance, long lFlags, _In_ CFrameworkQuery &Query);
 
     // If a provider wants to process queries, they should override this
@@ -153,6 +156,7 @@ private:
     // by the framework entrypoint functions
 
     /* Interfaces For Use by the Framework         */
+_Win32_metadata_supported_os_(windows5.0)
     HRESULT GetObject(  _In_ ParsedObjectPath *pParsedObjectPath, 
         _In_ MethodContext *pContext, long lFlags = 0L );
 
