diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h b/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
index a075757e..766d1c7d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
@@ -585,12 +585,17 @@ typedef LPCPROPSHEETHEADERW_V2      LPCPROPSHEETHEADERW;
 #define PSCB_BUTTONPRESSED 3
 #endif  // NTDDI_VERSION >= NTDDI_WINXP
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI HPROPSHEETPAGE WINAPI CreatePropertySheetPageA(LPCPROPSHEETPAGEA constPropSheetPagePointer);
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI HPROPSHEETPAGE WINAPI CreatePropertySheetPageW(LPCPROPSHEETPAGEW constPropSheetPagePointer);
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI BOOL           WINAPI DestroyPropertySheetPage(HPROPSHEETPAGE);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI INT_PTR        WINAPI PropertySheetA(LPCPROPSHEETHEADERA);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI INT_PTR        WINAPI PropertySheetW(LPCPROPSHEETHEADERW);
 
 #ifdef UNICODE
