diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h b/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
index 94df403e..42f7f779 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prsht.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 /*****************************************************************************\
 *                                                                             *
@@ -582,12 +583,19 @@ typedef LPCPROPSHEETHEADERW_V2      LPCPROPSHEETHEADERW;
 #define PSCB_BUTTONPRESSED 3
 #endif  // NTDDI_VERSION >= NTDDI_WINXP
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI HPROPSHEETPAGE WINAPI CreatePropertySheetPageA(LPCPROPSHEETPAGEA constPropSheetPagePointer);
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI HPROPSHEETPAGE WINAPI CreatePropertySheetPageW(LPCPROPSHEETPAGEW constPropSheetPagePointer);
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI BOOL           WINAPI DestroyPropertySheetPage(HPROPSHEETPAGE);
 
+_Win32_metadata_set_last_error_
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI INT_PTR        WINAPI PropertySheetA(LPCPROPSHEETHEADERA);
 
+_Win32_metadata_set_last_error_
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINCOMMCTRLAPI INT_PTR        WINAPI PropertySheetW(LPCPROPSHEETHEADERW);
 
 #ifdef UNICODE
@@ -600,6 +608,7 @@ WINCOMMCTRLAPI INT_PTR        WINAPI PropertySheetW(LPCPROPSHEETHEADERW);
 
 
 
+_Win32_metadata_canonical_name_(LPFNSVADDPROPSHEETPAGE)
 typedef BOOL (CALLBACK *LPFNADDPROPSHEETPAGE)(HPROPSHEETPAGE, LPARAM);
 typedef BOOL (CALLBACK *LPFNADDPROPSHEETPAGES)(LPVOID, LPFNADDPROPSHEETPAGE, LPARAM);
 
