diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/qos2.h b/generation/WinSDK/RecompiledIdlHeaders/um/qos2.h
index 55a0ce89..57d25c52 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/qos2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/qos2.h
@@ -14,6 +14,7 @@ Abstract:
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -503,12 +504,14 @@ QOS_SUBTRACT_OVERHEAD(
 //          exception for the qWave service. Please see the developer guidelines 
 //          and MSFT firewall documentation for more details.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
 QOSCreateHandle(
     _In_    PQOS_VERSION    Version,
     _Out_   PHANDLE         QOSHandle
+        _Win32_metadata_raii_free_(QOSCloseHandle)
 );
 
 //
@@ -538,6 +541,7 @@ QOSCreateHandle(
 //  If the function fails, the return value is zero. To get extended error 
 //  information, call GetLastError.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -595,6 +599,7 @@ QOSCloseHandle(
 //          you've specified. It could be that the other host does does not 
 //          have the required components or that it is not on the same link.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -629,6 +634,7 @@ QOSStartTrackingClient(
 //  If the function fails, the return value is zero. To get extended error 
 //  information, call GetLastError.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -674,6 +680,7 @@ QOSStopTrackingClient(
 //          The caller does not have the administrator rights required 
 //          to perform this call.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -817,6 +824,7 @@ QOSEnumerateFlows(
 //          you've specified. It could be that the other host does does not 
 //          have the required components or that it is not on the same link.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -880,6 +888,7 @@ QOSAddSocketToFlow(
 //      ERROR_NOT_FOUND
 //          Invalid FlowId specified
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -984,6 +993,7 @@ QOSRemoveSocketFromFlow(
 //          The caller does not have the administrator rights required 
 //          to perform this call.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -1106,6 +1116,7 @@ QOSSetFlow(
 //          qWave has erred on the side of caution as it waits for more data
 //          before ascertaining the state of the network.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -1227,6 +1238,7 @@ QOSQueryFlow(
 //          network which the qWave subsystem does not have. Obtaining this
 //          information on your network is currently not supported.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
@@ -1273,6 +1285,7 @@ QOSNotifyFlow(
 //  If the function fails, the return value is zero. To get extended error 
 //  information, call GetLastError.
 //
+_Win32_metadata_supported_os_(windows6.0.6000)
 ExternC
 BOOL
 WINAPI
