diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Ratings.h b/generation/WinSDK/RecompiledIdlHeaders/um/Ratings.h
index 0f5f2d60..a42c7331 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Ratings.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Ratings.h
@@ -8,6 +8,7 @@
 #define _RATINGS_H_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -29,10 +30,12 @@ STDAPI RatingFreeDetails(_In_opt_ void *pRatingDetails);
 STDAPI RatingObtainCancel(HANDLE hRatingObtainQuery);
 STDAPI RatingObtainQuery(LPCSTR pszTargetUrl, DWORD dwUserData,
     void (*fCallback)(DWORD dwUserData, HRESULT hr, LPCSTR pszRating, void *lpvRatingDetails),
-    _Out_opt_ HANDLE *phRatingObtainQuery);
+    _Out_opt_ HANDLE *phRatingObtainQuery
+        _Win32_metadata_raii_free_(RatingObtainCancel));
 STDAPI RatingObtainQueryW(LPCWSTR pszTargetUrl, DWORD dwUserData,
     void (*fCallback)(DWORD dwUserData, HRESULT hr, LPCWSTR pszRating, void *lpvRatingDetails),
-    _Out_opt_ HANDLE *phRatingObtainQuery);
+    _Out_opt_ HANDLE *phRatingObtainQuery
+        _Win32_metadata_raii_free_(RatingObtainCancel));
 STDAPI RatingSetupUI(HWND hDlg, LPCSTR pszUsername);
 STDAPI RatingSetupUIW(HWND hDlg, LPCWSTR pszUsername);
 #ifdef _INC_COMMCTRL
