diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/remotesystemadditionalinfo.h b/generation/WinSDK/RecompiledIdlHeaders/um/remotesystemadditionalinfo.h
index 035bb3de..d1917fb0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/remotesystemadditionalinfo.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/remotesystemadditionalinfo.h
@@ -67,6 +67,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS4)
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
@@ -183,14 +186,22 @@ extern RPC_IF_HANDLE __MIDL_itf_remotesystemadditionalinfo_0000_0001_v0_0_s_ifsp
 
 /* Additional Prototypes for ALL interfaces */
 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned long             __RPC_USER  HSTRING_UserSize(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserMarshal(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 void                      __RPC_USER  HSTRING_UserFree(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned long             __RPC_USER  HSTRING_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Win32_metadata_supported_os_(windows8.0)
 void                      __RPC_USER  HSTRING_UserFree64(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
 /* end of Additional Prototypes */
