diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RestartManager.h b/generation/WinSDK/RecompiledIdlHeaders/um/RestartManager.h
index 0c832073..e998ee80 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RestartManager.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RestartManager.h
@@ -8,6 +8,7 @@
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -147,6 +148,7 @@ typedef void (*RM_WRITE_STATUS_CALLBACK)( _In_ UINT nPercentComplete );
 //            Read/write operation failed.
 //        ERROR_OUTOFMEMORY
 //            Could not allocate memory or instantiate class object.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmStartSession(
     _Out_ DWORD *pSessionHandle,
@@ -164,6 +166,7 @@ RmStartSession(
 //            Read/write operation failed.
 //        ERROR_OUTOFMEMORY
 //            Could not allocate memory or instantiate class object.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmJoinSession(
     _Out_ DWORD *pSessionHandle,
@@ -182,6 +185,7 @@ RmJoinSession(
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmEndSession( _In_ DWORD dwSessionHandle );
 
@@ -203,6 +207,7 @@ RmEndSession( _In_ DWORD dwSessionHandle );
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmRegisterResources(
     _In_ DWORD dwSessionHandle,
@@ -236,6 +241,7 @@ RmRegisterResources(
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmGetList(
     _In_ DWORD dwSessionHandle,
@@ -264,6 +270,7 @@ RmGetList(
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmShutdown(
     _In_ DWORD dwSessionHandle,
@@ -291,6 +298,7 @@ RmShutdown(
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmRestart(
         _In_ DWORD dwSessionHandle,
@@ -307,6 +315,7 @@ RmRestart(
 //            Could not allocate memory or instantiate class object.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmCancelCurrentTask(
         _In_ DWORD dwSessionHandle);
@@ -320,6 +329,7 @@ RmCancelCurrentTask(
 //            An invalid argument was supplied to the function.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmAddFilter(
         _In_ DWORD dwSessionHandle,
@@ -338,6 +348,7 @@ RmAddFilter(
 //            An invalid argument was supplied to the function.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmRemoveFilter(
         _In_ DWORD dwSessionHandle,
@@ -353,6 +364,7 @@ RmRemoveFilter(
 //            An invalid argument was supplied to the function.
 //        ERROR_INVALID_HANDLE
 //            No session exists for the supplied handle.
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 RmGetFilterList(
         _In_ DWORD dwSessionHandle,
