diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RestrictedErrorInfo.h b/generation/WinSDK/RecompiledIdlHeaders/um/RestrictedErrorInfo.h
index ccafa8ca..f46d0cc0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RestrictedErrorInfo.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RestrictedErrorInfo.h
@@ -94,6 +94,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
@@ -116,6 +117,7 @@ EXTERN_C const IID IID_IRestrictedErrorInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("82BA7092-4C88-427D-A7BC-16DD93FEB67E")
     IRestrictedErrorInfo : public IUnknown
     {
@@ -228,6 +230,7 @@ EXTERN_C const IID IID_ILanguageExceptionErrorInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.1)
     MIDL_INTERFACE("04a2dbf3-df83-116c-0946-0812abf6e07d")
     ILanguageExceptionErrorInfo : public IUnknown
     {
@@ -315,6 +318,7 @@ EXTERN_C const IID IID_ILanguageExceptionTransform;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("FEB5A271-A6CD-45CE-880A-696706BADC65")
     ILanguageExceptionTransform : public IUnknown
     {
@@ -402,6 +406,7 @@ EXTERN_C const IID IID_ILanguageExceptionStackBackTrace;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("CBE53FB5-F967-4258-8D34-42F5E25833DE")
     ILanguageExceptionStackBackTrace : public IUnknown
     {
@@ -495,6 +500,7 @@ EXTERN_C const IID IID_ILanguageExceptionErrorInfo2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.15063)
     MIDL_INTERFACE("5746E5C4-5B97-424C-B620-2822915734DD")
     ILanguageExceptionErrorInfo2 : public ILanguageExceptionErrorInfo
     {
@@ -624,9 +630,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* end of Additional Prototypes */
