diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcasync.h b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcasync.h
index 060112fb..87600728 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcasync.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcasync.h
@@ -1,4 +1,7 @@
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 /*++
 
@@ -178,6 +181,7 @@ typedef struct _RPC_ASYNC_STATE {
 #define RpcAsyncGetCallHandle(pAsync) (((PRPC_ASYNC_STATE) pAsync)->RuntimeInfo)
 
 #if !defined(_KRPCENV_)
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -187,6 +191,7 @@ RpcAsyncRegisterInfo (
     ) ;
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -196,6 +201,7 @@ RpcAsyncInitializeHandle (
     _In_ unsigned int     Size
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -204,6 +210,7 @@ RpcAsyncGetCallStatus (
     _In_ PRPC_ASYNC_STATE pAsync
     ) ;
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -213,6 +220,7 @@ RpcAsyncCompleteCall (
     _Out_opt_ void *Reply
     ) ;
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -222,6 +230,7 @@ RpcAsyncAbortCall (
     _In_ unsigned long ExceptionCode
     ) ;
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -308,6 +317,7 @@ typedef struct tagRPC_ERROR_ENUM_HANDLE
     void *Head;
 } RPC_ERROR_ENUM_HANDLE;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -315,6 +325,7 @@ RpcErrorStartEnumeration (
     _Out_ RPC_ERROR_ENUM_HANDLE *EnumHandle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -324,6 +335,7 @@ RpcErrorGetNextRecord (
     _Out_ RPC_EXTENDED_ERROR_INFO *ErrorInfo
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -331,6 +343,7 @@ RpcErrorEndEnumeration (
     _Inout_ RPC_ERROR_ENUM_HANDLE *EnumHandle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -338,6 +351,7 @@ RpcErrorResetEnumeration (
     _Inout_ RPC_ERROR_ENUM_HANDLE *EnumHandle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -346,6 +360,7 @@ RpcErrorGetNumberOfRecords (
     _Out_ int *Records
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -355,6 +370,7 @@ RpcErrorSaveErrorInfo (
     _Out_ size_t *BlobSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -364,6 +380,7 @@ RpcErrorLoadErrorInfo (
     _Out_ RPC_ERROR_ENUM_HANDLE *EnumHandle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -371,6 +388,7 @@ RpcErrorAddRecord (
     _In_ RPC_EXTENDED_ERROR_INFO *ErrorInfo
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 void 
 RPC_ENTRY
@@ -395,6 +413,7 @@ RpcAsyncCleanupThread (
     _In_ DWORD dwTimeout
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -409,6 +428,7 @@ RpcGetAuthorizationContextForClient (
     _Out_ PVOID *pAuthzClientContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY
@@ -416,6 +436,7 @@ RpcFreeAuthorizationContext (
     _Inout_ PVOID *pAuthzClientContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -424,6 +445,7 @@ RpcSsContextLockExclusive (
     _In_ PVOID UserContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -615,6 +637,7 @@ typedef struct tagRPC_CALL_ATTRIBUTES_V3_A
 
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -623,6 +646,7 @@ RpcServerInqCallAttributesW (
     _Inout_ void *RpcCallAttributes
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -670,6 +694,7 @@ typedef enum _RPC_NOTIFICATIONS
 
 #define RpcNotificationCallStatusChange    RpcNotificationClientDisconnect
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY 
@@ -680,6 +705,7 @@ RpcServerSubscribeForNotification (
     _In_ RPC_ASYNC_NOTIFICATION_INFO *NotificationInfo
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY 
@@ -700,6 +726,7 @@ RpcServerUnsubscribeForNotification (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY 
@@ -709,6 +736,7 @@ RpcBindingBind (
     _In_ RPC_IF_HANDLE IfSpec    
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY 
