diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdcep.h b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdcep.h
index 2c1cb293..399769cf 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdcep.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdcep.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -549,6 +550,7 @@ I_RpcBindingInqMarshalledTargetInfo (
 #endif // (NTDDI_VERSION >= NTDDI_WS03)
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
