diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdce.h b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdce.h
index d5597957..d0d52c6d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdce.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/rpcdce.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -290,6 +291,7 @@ typedef struct
 } RPC_IF_ID_VECTOR;
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -300,6 +302,7 @@ RpcBindingCopy (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -308,6 +311,7 @@ RpcBindingFree (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -319,6 +323,7 @@ RpcBindingSetOption (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -335,6 +340,7 @@ RpcBindingInqOption (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -344,6 +350,7 @@ RpcBindingFromStringBindingA (
     _Out_ RPC_BINDING_HANDLE __RPC_FAR * Binding
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -361,6 +368,7 @@ RpcBindingFromStringBindingW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -404,6 +412,7 @@ RpcBindingInqMaxCalls (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -414,6 +423,7 @@ RpcBindingInqObject (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -426,6 +436,7 @@ RpcBindingReset (
 /* RpcBindingSetAuthInfo */
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -442,6 +453,7 @@ RpcBindingSetObject (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -462,6 +474,7 @@ RpcMgmtInqDefaultProtectLevel (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -471,6 +484,7 @@ RpcBindingToStringBindingA (
     _Outptr_ RPC_CSTR __RPC_FAR * StringBinding
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -491,6 +505,7 @@ RpcBindingToStringBindingW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -506,6 +521,7 @@ RpcBindingToStringBinding (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -526,6 +542,7 @@ RpcBindingVectorFree (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -539,6 +556,7 @@ RpcStringBindingComposeA (
     _Outptr_opt_ RPC_CSTR __RPC_FAR * StringBinding
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -566,6 +584,7 @@ RpcStringBindingComposeW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -591,6 +610,7 @@ RpcStringBindingCompose (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -604,6 +624,7 @@ RpcStringBindingParseA (
     _Outptr_opt_ RPC_CSTR __RPC_FAR * NetworkOptions
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -631,6 +652,7 @@ RpcStringBindingParseW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -657,6 +679,7 @@ RpcStringBindingParse (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -664,6 +687,7 @@ RpcStringFreeA (
     _Inout_ RPC_CSTR __RPC_FAR * String    
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -685,6 +709,7 @@ RpcStringFreeW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -698,6 +723,7 @@ RpcStringFree (
 #endif /* RPC_UNICODE_SUPPORTED */
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -715,6 +741,7 @@ RpcIfInqId (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -723,6 +750,7 @@ RpcNetworkIsProtseqValidA (
     _In_ RPC_CSTR Protseq     
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -745,6 +773,7 @@ RpcNetworkIsProtseqValidW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -758,6 +787,7 @@ RpcNetworkIsProtseqValid (
 #endif /* RPC_UNICODE_SUPPORTED */
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -772,6 +802,7 @@ RpcMgmtInqComTimeout (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -782,6 +813,7 @@ RpcMgmtSetComTimeout (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -800,6 +832,7 @@ RpcMgmtSetCancelTimeout(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -808,6 +841,7 @@ RpcNetworkInqProtseqsA (
     _Outptr_ RPC_PROTSEQ_VECTORA __RPC_FAR * __RPC_FAR * ProtseqVector    
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -824,6 +858,7 @@ RpcNetworkInqProtseqsW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -834,6 +869,7 @@ RpcNetworkInqProtseqs (
 #endif /* RPC_UNICODE_SUPPORTED */
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -844,6 +880,7 @@ RpcObjectInqType (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -853,6 +890,7 @@ RpcObjectSetInqFn (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -866,6 +904,7 @@ RpcObjectSetType (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -873,6 +912,7 @@ RpcProtseqVectorFreeA (
     _Inout_ RPC_PROTSEQ_VECTORA __RPC_FAR * __RPC_FAR * ProtseqVector
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -888,6 +928,7 @@ RpcProtseqVectorFreeW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -898,6 +939,7 @@ RpcProtseqVectorFree (
 #endif /* RPC_UNICODE_SUPPORTED */
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -917,6 +959,7 @@ RpcServerInqBindingsEx (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -928,6 +971,7 @@ RpcServerInqIf (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -939,6 +983,7 @@ RpcServerListen (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -950,6 +995,7 @@ RpcServerRegisterIf (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -964,6 +1010,7 @@ RpcServerRegisterIfEx (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -979,6 +1026,7 @@ RpcServerRegisterIf2 (
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 /* server */
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -995,6 +1043,7 @@ RpcServerRegisterIf3 (
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1007,6 +1056,7 @@ RpcServerUnregisterIf (
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
 /* server */
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1018,6 +1068,7 @@ RpcServerUnregisterIfEx (
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1028,6 +1079,7 @@ RpcServerUseAllProtseqs (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1039,6 +1091,7 @@ RpcServerUseAllProtseqsEx (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1050,6 +1103,7 @@ RpcServerUseAllProtseqsIf (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1066,6 +1120,7 @@ RpcServerUseAllProtseqsIfEx (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1076,6 +1131,7 @@ RpcServerUseProtseqA (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1087,6 +1143,7 @@ RpcServerUseProtseqExA (
     _In_ PRPC_POLICY Policy
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1097,6 +1154,7 @@ RpcServerUseProtseqW (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1118,6 +1176,7 @@ RpcServerUseProtseqExW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1127,6 +1186,7 @@ RpcServerUseProtseq (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1143,6 +1203,7 @@ RpcServerUseProtseqEx (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1154,6 +1215,7 @@ RpcServerUseProtseqEpA (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1166,6 +1228,7 @@ RpcServerUseProtseqEpExA (
     _In_ PRPC_POLICY Policy
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1177,6 +1240,7 @@ RpcServerUseProtseqEpW (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1199,6 +1263,7 @@ RpcServerUseProtseqEpExW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1209,6 +1274,7 @@ RpcServerUseProtseqEp (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1226,6 +1292,7 @@ RpcServerUseProtseqEpEx (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1237,6 +1304,7 @@ RpcServerUseProtseqIfA (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1249,6 +1317,7 @@ RpcServerUseProtseqIfExA (
     _In_ PRPC_POLICY Policy
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1260,6 +1329,7 @@ RpcServerUseProtseqIfW (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1282,6 +1352,7 @@ RpcServerUseProtseqIfExW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1292,6 +1363,7 @@ RpcServerUseProtseqIf (
     _In_opt_ void __RPC_FAR * SecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1313,6 +1385,7 @@ RpcServerYield (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1327,6 +1400,7 @@ RpcMgmtStatsVectorFree (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1337,6 +1411,7 @@ RpcMgmtInqStats (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1352,6 +1427,7 @@ RpcMgmtIsServerListening (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1361,6 +1437,7 @@ RpcMgmtStopServerListening (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1370,6 +1447,7 @@ RpcMgmtWaitServerListen (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1379,6 +1457,7 @@ RpcMgmtSetServerStackSize (
     );
 
 /* server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -1387,6 +1466,7 @@ RpcSsDontSerializeContext (
     );
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1395,6 +1475,7 @@ RpcMgmtEnableIdleCleanup (
     void
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1404,6 +1485,7 @@ RpcMgmtInqIfIds (
     _Outptr_ RPC_IF_ID_VECTOR __RPC_FAR * __RPC_FAR * IfIdVector
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1419,6 +1501,7 @@ RpcIfIdVectorFree (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1429,6 +1512,7 @@ RpcMgmtInqServerPrincNameA (
     _Outptr_ RPC_CSTR __RPC_FAR * ServerPrincName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1453,6 +1537,7 @@ RpcMgmtInqServerPrincNameW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1472,6 +1557,7 @@ RpcMgmtInqServerPrincName (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1481,6 +1567,7 @@ RpcServerInqDefaultPrincNameA (
     _Outptr_ RPC_CSTR __RPC_FAR * PrincName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1498,6 +1585,7 @@ RpcServerInqDefaultPrincNameW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -1515,6 +1603,7 @@ RpcServerInqDefaultPrincName (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 /* client */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1534,6 +1623,7 @@ RpcEpResolveBinding (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1544,6 +1634,7 @@ RpcNsBindingInqEntryNameA (
     _Outptr_ RPC_CSTR __RPC_FAR * EntryName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -1562,6 +1653,7 @@ RpcNsBindingInqEntryNameW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2127,6 +2219,7 @@ typedef struct _RPC_BINDING_HANDLE_OPTIONS_V1 {
 
 #ifdef _AUTH_IDENTITY_A_DEFINED
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2139,6 +2232,7 @@ RpcBindingCreateA (
 
 #endif // _AUTH_IDENTITY_A_DEFINED
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2169,6 +2263,7 @@ RpcBindingGetTrainingContextHandle (
     _Outptr_ void ** ContextHandle
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 RPC_STATUS 
 RPC_ENTRY 
@@ -2256,6 +2351,7 @@ typedef void
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2271,6 +2367,7 @@ RpcImpersonateClient2 (
     _In_opt_ RPC_BINDING_HANDLE BindingHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2279,6 +2376,7 @@ RpcRevertToSelfEx (
     _In_opt_ RPC_BINDING_HANDLE BindingHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2287,6 +2385,7 @@ RpcRevertToSelf (
     void
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2295,6 +2394,7 @@ RpcImpersonateClientContainer (
     _In_opt_ RPC_BINDING_HANDLE BindingHandle
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2313,6 +2413,7 @@ RpcRevertContainerImpersonation (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2326,6 +2427,7 @@ RpcBindingInqAuthClientA (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2339,6 +2441,7 @@ RpcBindingInqAuthClientW (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2352,6 +2455,7 @@ RpcBindingInqAuthClientExA (
     _In_  unsigned long             Flags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2371,6 +2475,7 @@ RpcBindingInqAuthClientExW (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2384,6 +2489,7 @@ RpcBindingInqAuthInfoA (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2397,6 +2503,7 @@ RpcBindingInqAuthInfoW (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2410,6 +2517,7 @@ RpcBindingSetAuthInfoA (
     _In_ unsigned long AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2424,6 +2532,7 @@ RpcBindingSetAuthInfoExA (
     _In_opt_ RPC_SECURITY_QOS * SecurityQos
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2437,6 +2546,7 @@ RpcBindingSetAuthInfoW (
     _In_ unsigned long AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2451,6 +2561,7 @@ RpcBindingSetAuthInfoExW (
     _In_opt_ RPC_SECURITY_QOS * SecurityQOS
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2466,6 +2577,7 @@ RpcBindingInqAuthInfoExA (
     _Out_opt_ RPC_SECURITY_QOS *SecurityQOS
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2496,6 +2608,7 @@ typedef void
     _Out_ RPC_STATUS __RPC_FAR * Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY 
@@ -2504,6 +2617,7 @@ RpcServerCompleteSecurityCallback(
     _In_ RPC_STATUS Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2515,6 +2629,7 @@ RpcServerRegisterAuthInfoA (
     _In_opt_ void __RPC_FAR * Arg
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2558,6 +2673,7 @@ RpcServerRegisterAuthInfoW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2570,6 +2686,7 @@ RpcBindingInqAuthClient (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2583,6 +2700,7 @@ RpcBindingInqAuthClientEx (
     _In_  unsigned long             Flags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2595,6 +2713,7 @@ RpcBindingInqAuthInfo (
     _Out_opt_ unsigned long __RPC_FAR * AuthzSvc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2616,6 +2735,7 @@ typedef void
     _Out_ RPC_STATUS __RPC_FAR * Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2645,6 +2765,7 @@ typedef struct {
 #endif 
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2660,6 +2781,7 @@ RpcBindingServerFromClient (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 DECLSPEC_NORETURN
 void
@@ -2668,6 +2790,7 @@ RpcRaiseException (
     _In_ RPC_STATUS exception
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2682,6 +2805,7 @@ RpcTestCancel(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2690,6 +2814,7 @@ RpcServerTestCancel (
     _In_opt_ RPC_BINDING_HANDLE BindingHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2698,6 +2823,7 @@ RpcCancelThread(
     _In_ void * Thread
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2713,6 +2839,7 @@ RpcCancelThreadEx(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2722,6 +2849,7 @@ UuidCreate (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2732,6 +2860,7 @@ UuidCreateSequential (
 #ifdef RPC_UNICODE_SUPPORTED
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2742,6 +2871,7 @@ UuidToStringA (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2752,6 +2882,7 @@ UuidFromStringA (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2762,6 +2893,7 @@ UuidToStringW (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2782,6 +2914,7 @@ UuidFromStringW (
 #else /* RPC_UNICODE_SUPPORTED */
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2791,6 +2924,7 @@ UuidToString (
     );
 
 /* client/server */
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2801,6 +2935,7 @@ UuidFromString (
 
 #endif /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 signed int
 RPC_ENTRY
@@ -2810,6 +2945,7 @@ UuidCompare (
     _Out_ RPC_STATUS __RPC_FAR * Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2818,6 +2954,7 @@ UuidCreateNil (
     _Out_ UUID __RPC_FAR * NilUuid
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 int
 RPC_ENTRY
@@ -2827,6 +2964,7 @@ UuidEqual (
     _Out_ RPC_STATUS __RPC_FAR * Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 unsigned short
 RPC_ENTRY
@@ -2835,6 +2973,7 @@ UuidHash (
     _Out_ RPC_STATUS __RPC_FAR * Status
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 int
 RPC_ENTRY
@@ -2851,6 +2990,7 @@ UuidIsNil (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2862,6 +3002,7 @@ RpcEpRegisterNoReplaceA (
     _In_opt_ RPC_CSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2873,6 +3014,7 @@ RpcEpRegisterNoReplaceW (
     _In_opt_ RPC_WSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2884,6 +3026,7 @@ RpcEpRegisterA (
     _In_opt_ RPC_CSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2905,6 +3048,7 @@ RpcEpRegisterW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2915,6 +3059,7 @@ RpcEpRegisterNoReplace (
     _In_opt_ RPC_CSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -2928,6 +3073,7 @@ RpcEpRegister (
 #endif /* RPC_UNICODE_SUPPORTED */
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2948,6 +3094,7 @@ RpcEpUnregister(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2957,6 +3104,7 @@ DceErrorInqTextA (
     _Out_writes_(DCE_C_ERROR_STRING_LEN) RPC_CSTR ErrorText
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -2980,6 +3128,7 @@ DceErrorInqTextW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3009,6 +3158,7 @@ typedef I_RPC_HANDLE * RPC_EP_INQ_HANDLE;
 #define  RPC_C_VERS_MAJOR_ONLY    4
 #define  RPC_C_VERS_UPTO          5
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -3022,6 +3172,7 @@ RpcMgmtEpEltInqBegin (
     _Out_ RPC_EP_INQ_HANDLE __RPC_FAR * InquiryContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -3032,6 +3183,7 @@ RpcMgmtEpEltInqDone (
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -3044,6 +3196,7 @@ RpcMgmtEpEltInqNextA (
     _Outptr_opt_ RPC_CSTR __RPC_FAR * Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -3064,6 +3217,7 @@ RpcMgmtEpEltInqNextW (
 
 #else /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3100,6 +3254,7 @@ typedef int
 #define RPC_C_MGMT_IS_SERVER_LISTEN   3
 #define RPC_C_MGMT_STOP_SERVER_LISTEN 4
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 _Must_inspect_result_
 RPC_STATUS
@@ -3115,6 +3270,7 @@ RpcMgmtSetAuthorizationFn (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
+_Win32_metadata_supported_os_(windows6.0.6000)
 RPCRTAPI
 int
 RPC_ENTRY
@@ -3260,6 +3416,7 @@ RPC_INTERFACE_GROUP_IDLE_CALLBACK_FN (
 #ifdef RPC_UNICODE_SUPPORTED
 
 /* server */
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3276,6 +3433,7 @@ RpcServerInterfaceGroupCreateW (
 
 
 /* server */
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3299,6 +3457,7 @@ RpcServerInterfaceGroupCreateA (
 #else /* RPC_UNICODE_SUPPORTED */
 
 /* server */
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3315,6 +3474,7 @@ RpcServerInterfaceGroupCreate (
 
 #endif /* RPC_UNICODE_SUPPORTED */
 
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3322,6 +3482,7 @@ RpcServerInterfaceGroupClose (
     _In_ RPC_INTERFACE_GROUP IfGroup
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3329,6 +3490,7 @@ RpcServerInterfaceGroupActivate (
     _In_ RPC_INTERFACE_GROUP IfGroup
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
@@ -3337,6 +3499,7 @@ RpcServerInterfaceGroupDeactivate (
     _In_ unsigned long ForceDeactivation
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 RPCRTAPI
 RPC_STATUS
 RPC_ENTRY
