diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/rpcnsi.h b/generation/WinSDK/RecompiledIdlHeaders/um/rpcnsi.h
index aaafec15..0721224a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/rpcnsi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/rpcnsi.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -41,6 +42,7 @@ typedef void __RPC_FAR * RPC_NS_HANDLE;
 
 /* Server APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingExportA(
     _In_ unsigned long EntryNameSyntax,
@@ -51,6 +53,7 @@ RpcNsBindingExportA(
     );
 
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingUnexportA(
     _In_ unsigned long EntryNameSyntax,
@@ -61,6 +64,7 @@ RpcNsBindingUnexportA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingExportW(
     _In_ unsigned long EntryNameSyntax,
@@ -70,6 +74,7 @@ RpcNsBindingExportW(
     _In_opt_ UUID_VECTOR __RPC_FAR *ObjectUuidVec
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingUnexportW(
     _In_ unsigned long EntryNameSyntax,
@@ -82,6 +87,7 @@ RpcNsBindingUnexportW(
 
 /* Server PnP APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS RPC_ENTRY
 RpcNsBindingExportPnPA(
     _In_ unsigned long EntryNameSyntax,
@@ -90,6 +96,7 @@ RpcNsBindingExportPnPA(
     _In_opt_ UUID_VECTOR *ObjectVector
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS RPC_ENTRY
 RpcNsBindingUnexportPnPA(
     _In_ unsigned long EntryNameSyntax,
@@ -100,6 +107,7 @@ RpcNsBindingUnexportPnPA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS RPC_ENTRY
 RpcNsBindingExportPnPW(
     _In_ unsigned long EntryNameSyntax,
@@ -108,6 +116,7 @@ RpcNsBindingExportPnPW(
     _In_opt_ UUID_VECTOR *ObjectVector
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPC_STATUS RPC_ENTRY
 RpcNsBindingUnexportPnPW(
     _In_ unsigned long EntryNameSyntax,
@@ -120,6 +129,7 @@ RpcNsBindingUnexportPnPW(
 
 /* Client APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingLookupBeginA(
     _In_ unsigned long EntryNameSyntax,
@@ -132,6 +142,7 @@ RpcNsBindingLookupBeginA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingLookupBeginW(
     _In_ unsigned long EntryNameSyntax,
@@ -143,12 +154,14 @@ RpcNsBindingLookupBeginW(
     );
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingLookupNext(
     IN  RPC_NS_HANDLE LookupContext,
     OUT RPC_BINDING_VECTOR __RPC_FAR * __RPC_FAR * BindingVec
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingLookupDone(
     IN OUT RPC_NS_HANDLE __RPC_FAR * LookupContext
@@ -156,12 +169,14 @@ RpcNsBindingLookupDone(
 
 /* Group APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupDeleteA(
     _In_ unsigned long GroupNameSyntax,
     _In_opt_ RPC_CSTR GroupName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrAddA(
     _In_ unsigned long GroupNameSyntax,
@@ -170,6 +185,7 @@ RpcNsGroupMbrAddA(
     _In_ RPC_CSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrRemoveA(
     _In_ unsigned long GroupNameSyntax,
@@ -178,6 +194,7 @@ RpcNsGroupMbrRemoveA(
     _In_ RPC_CSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrInqBeginA(
     _In_ unsigned long GroupNameSyntax,
@@ -186,6 +203,7 @@ RpcNsGroupMbrInqBeginA(
     _Out_ RPC_NS_HANDLE __RPC_FAR *InquiryContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrInqNextA(
     _Inout_ RPC_NS_HANDLE InquiryContext,
@@ -194,12 +212,14 @@ RpcNsGroupMbrInqNextA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupDeleteW(
     _In_ unsigned long GroupNameSyntax,
     _In_opt_ RPC_WSTR GroupName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrAddW(
     _In_ unsigned long GroupNameSyntax,
@@ -208,6 +228,7 @@ RpcNsGroupMbrAddW(
     _In_ RPC_WSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrRemoveW(
     _In_ unsigned long GroupNameSyntax,
@@ -216,6 +237,7 @@ RpcNsGroupMbrRemoveW(
     _In_ RPC_WSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrInqBeginW(
     _In_ unsigned long GroupNameSyntax,
@@ -224,6 +246,7 @@ RpcNsGroupMbrInqBeginW(
     _Out_ RPC_NS_HANDLE __RPC_FAR *InquiryContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrInqNextW(
     _Inout_ RPC_NS_HANDLE InquiryContext,
@@ -232,6 +255,7 @@ RpcNsGroupMbrInqNextW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsGroupMbrInqDone(
     IN OUT RPC_NS_HANDLE __RPC_FAR * InquiryContext
@@ -239,12 +263,14 @@ RpcNsGroupMbrInqDone(
 
 /* Profile APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileDeleteA(
     _In_ unsigned long ProfileNameSyntax,
     _In_ RPC_CSTR ProfileName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltAddA(
     _In_ unsigned long ProfileNameSyntax,
@@ -256,6 +282,7 @@ RpcNsProfileEltAddA(
     _In_opt_ RPC_CSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltRemoveA(
     _In_ unsigned long ProfileNameSyntax,
@@ -265,6 +292,7 @@ RpcNsProfileEltRemoveA(
     _In_ RPC_CSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltInqBeginA(
     _In_ unsigned long ProfileNameSyntax,
@@ -277,6 +305,7 @@ RpcNsProfileEltInqBeginA(
     _Out_ RPC_NS_HANDLE __RPC_FAR *InquiryContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltInqNextA(
     _In_ RPC_NS_HANDLE InquiryContext,
@@ -288,12 +317,14 @@ RpcNsProfileEltInqNextA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileDeleteW(
     _In_ unsigned long ProfileNameSyntax,
     _In_ RPC_WSTR ProfileName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltAddW(
     _In_ unsigned long ProfileNameSyntax,
@@ -305,6 +336,7 @@ RpcNsProfileEltAddW(
     _In_opt_ RPC_WSTR Annotation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltRemoveW(
     _In_ unsigned long ProfileNameSyntax,
@@ -314,6 +346,7 @@ RpcNsProfileEltRemoveW(
     _In_ RPC_WSTR MemberName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltInqBeginW(
     _In_ unsigned long ProfileNameSyntax,
@@ -326,6 +359,7 @@ RpcNsProfileEltInqBeginW(
     _Out_ RPC_NS_HANDLE __RPC_FAR *InquiryContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltInqNextW(
     _In_ RPC_NS_HANDLE InquiryContext,
@@ -337,6 +371,7 @@ RpcNsProfileEltInqNextW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsProfileEltInqDone(
     IN OUT RPC_NS_HANDLE __RPC_FAR * InquiryContext
@@ -344,6 +379,7 @@ RpcNsProfileEltInqDone(
 
 /* Entry object APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryObjectInqBeginA(
     _In_ unsigned long EntryNameSyntax,
@@ -353,6 +389,7 @@ RpcNsEntryObjectInqBeginA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryObjectInqBeginW(
     _In_ unsigned long EntryNameSyntax,
@@ -362,12 +399,14 @@ RpcNsEntryObjectInqBeginW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryObjectInqNext(
     IN  RPC_NS_HANDLE InquiryContext,
     OUT UUID __RPC_FAR * ObjUuid
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryObjectInqDone(
     IN OUT RPC_NS_HANDLE __RPC_FAR * InquiryContext
@@ -375,6 +414,7 @@ RpcNsEntryObjectInqDone(
 
 /* Management and MISC APIs */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryExpandNameA(
     _In_ unsigned long EntryNameSyntax,
@@ -382,6 +422,7 @@ RpcNsEntryExpandNameA(
     _Out_ RPC_CSTR __RPC_FAR *ExpandedName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtBindingUnexportA(
     _In_ unsigned long EntryNameSyntax,
@@ -391,18 +432,21 @@ RpcNsMgmtBindingUnexportA(
     _In_opt_ UUID_VECTOR __RPC_FAR *ObjectUuidVec
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryCreateA(
     _In_ unsigned long EntryNameSyntax,
     _In_ RPC_CSTR EntryName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryDeleteA(
     _In_ unsigned long EntryNameSyntax,
     _In_ RPC_CSTR EntryName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryInqIfIdsA(
     _In_ unsigned long EntryNameSyntax,
@@ -410,17 +454,20 @@ RpcNsMgmtEntryInqIfIdsA(
     _Out_ RPC_IF_ID_VECTOR __RPC_FAR * __RPC_FAR *IfIdVec
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtHandleSetExpAge(
     IN RPC_NS_HANDLE NsHandle,
     IN unsigned long ExpirationAge
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtInqExpAge(
     OUT unsigned long __RPC_FAR * ExpirationAge
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtSetExpAge(
     IN unsigned long ExpirationAge
@@ -428,6 +475,7 @@ RpcNsMgmtSetExpAge(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsEntryExpandNameW(
     _In_ unsigned long EntryNameSyntax,
@@ -435,6 +483,7 @@ RpcNsEntryExpandNameW(
     _Out_ RPC_WSTR __RPC_FAR *ExpandedName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtBindingUnexportW(
     _In_ unsigned long EntryNameSyntax,
@@ -444,18 +493,21 @@ RpcNsMgmtBindingUnexportW(
     _In_opt_ UUID_VECTOR __RPC_FAR *ObjectUuidVec
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryCreateW(
     _In_ unsigned long EntryNameSyntax,
     _In_ RPC_WSTR EntryName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryDeleteW(
     _In_ unsigned long EntryNameSyntax,
     _In_ RPC_WSTR EntryName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsMgmtEntryInqIfIdsW(
     _In_ unsigned long EntryNameSyntax,
@@ -467,6 +519,7 @@ RpcNsMgmtEntryInqIfIdsW(
 
 /* Client API's implemented in wrappers. */
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingImportBeginA(
     _In_ unsigned long EntryNameSyntax,
@@ -478,6 +531,7 @@ RpcNsBindingImportBeginA(
 
 #ifdef RPC_UNICODE_SUPPORTED
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingImportBeginW(
     _In_ unsigned long EntryNameSyntax,
@@ -489,17 +543,20 @@ RpcNsBindingImportBeginW(
 
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingImportNext(
     IN RPC_NS_HANDLE ImportContext,
     OUT RPC_BINDING_HANDLE  __RPC_FAR * Binding
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingImportDone(
     IN OUT RPC_NS_HANDLE __RPC_FAR * ImportContext
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCNSAPI RPC_STATUS RPC_ENTRY
 RpcNsBindingSelect(
     IN OUT RPC_BINDING_VECTOR __RPC_FAR * BindingVec,
