diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RpcProxy.h b/generation/WinSDK/RecompiledIdlHeaders/um/RpcProxy.h
index e7e3e64b..9562a385 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RpcProxy.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RpcProxy.h
@@ -41,6 +41,7 @@ Compiler switches:
 #endif // __RPCPROXY_H_VERSION__
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #if !defined(_KRPCENV_)
 
@@ -282,6 +283,7 @@ typedef struct tagCStdPSFactoryBuffer
 #pragma region Desktop or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -292,6 +294,7 @@ NdrProxyInitialize(
     PMIDL_STUB_DESC     pStubDescriptor,
     unsigned int        ProcNum );
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -299,6 +302,7 @@ NdrProxyGetBuffer(
     void *              This,
     PMIDL_STUB_MESSAGE pStubMsg);
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -306,6 +310,7 @@ NdrProxySendReceive(
     void                *This,
     MIDL_STUB_MESSAGE   *pStubMsg);
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -320,6 +325,7 @@ NdrProxyFreeBuffer(
 #pragma region Application or OneCore or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 HRESULT
 RPC_ENTRY
@@ -332,6 +338,7 @@ NdrProxyErrorHandler(
 #pragma region Desktop or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -357,6 +364,7 @@ NdrStubInitializePartial(
     IRpcChannelBuffer * pRpcChannelBuffer,
     ULONG       RequestedBufferSize);
 
+_Win32_metadata_supported_os_(windows5.0)
 void __RPC_STUB NdrStubForwardingFunction(
     IN  IRpcStubBuffer *    This,
     IN  IRpcChannelBuffer * pChannel,
@@ -369,6 +377,7 @@ void __RPC_STUB NdrStubForwardingFunction(
 #pragma region Desktop or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 void
 RPC_ENTRY
@@ -389,12 +398,14 @@ RPC_ENTRY
 NdrStubErrorHandler(
     DWORD               dwExceptionCode);
 
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer_QueryInterface(
     IRpcStubBuffer  *   This,
     REFIID              riid,
     void            **  ppvObject);
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE
 CStdStubBuffer_AddRef(
     IRpcStubBuffer *    This);
@@ -403,45 +414,54 @@ ULONG STDMETHODCALLTYPE
 CStdStubBuffer_Release(
     IRpcStubBuffer *    This);
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE
 NdrCStdStubBuffer_Release(
     IRpcStubBuffer *    This,
     IPSFactoryBuffer *  pPSF);
 
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer_Connect(
     IRpcStubBuffer *    This,
     IUnknown *          pUnkServer);
 
+_Win32_metadata_supported_os_(windows5.0)
 void STDMETHODCALLTYPE
 CStdStubBuffer_Disconnect(
     IRpcStubBuffer *    This);
 
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer_Invoke(
     IRpcStubBuffer  *   This,
     RPCOLEMESSAGE   *   pRpcMsg,
     IRpcChannelBuffer * pRpcChannelBuffer);
 
+_Win32_metadata_supported_os_(windows5.0)
 IRpcStubBuffer * STDMETHODCALLTYPE
 CStdStubBuffer_IsIIDSupported(
     IRpcStubBuffer  *   This,
     REFIID riid);
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE
 CStdStubBuffer_CountRefs(
     IRpcStubBuffer  *   This);
 
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer_DebugServerQueryInterface(
     IRpcStubBuffer  *   This,
     void **ppv);
 
+_Win32_metadata_supported_os_(windows5.0)
 void STDMETHODCALLTYPE
 CStdStubBuffer_DebugServerRelease(
     IRpcStubBuffer  *   This,
     void *pv);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer2_QueryInterface(
     IRpcStubBuffer  *   This,
@@ -452,62 +472,75 @@ ULONG STDMETHODCALLTYPE
 CStdStubBuffer2_Release(
     IRpcStubBuffer *This);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdStubBuffer2_Connect(
     IRpcStubBuffer *pthis,
     IUnknown *pUnkServer);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 void STDMETHODCALLTYPE
 CStdStubBuffer2_Disconnect(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 ULONG STDMETHODCALLTYPE
 CStdStubBuffer2_CountRefs(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE
 NdrCStdStubBuffer2_Release(
     IRpcStubBuffer *This,
     IPSFactoryBuffer * pPSF);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdAsyncStubBuffer_QueryInterface(
     IRpcStubBuffer *pthis,
     REFIID riid,
     void **ppvObject);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 ULONG STDMETHODCALLTYPE
 CStdAsyncStubBuffer_AddRef(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 ULONG STDMETHODCALLTYPE
 CStdAsyncStubBuffer_Release(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdAsyncStubBuffer_Connect(
     IRpcStubBuffer *pthis,
     IUnknown *pUnkServer);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 void STDMETHODCALLTYPE
 CStdAsyncStubBuffer_Disconnect(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdAsyncStubBuffer_Invoke(
     IRpcStubBuffer *pthis,
     RPCOLEMESSAGE *_prpcmsg,
     IRpcChannelBuffer *_pRpcChannelBuffer);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 ULONG STDMETHODCALLTYPE
 CStdAsyncStubBuffer2_Release(
     IRpcStubBuffer *pthis);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 HRESULT STDMETHODCALLTYPE
 CStdAsyncStubBuffer2_Connect(
     IRpcStubBuffer *pthis,
     IUnknown *pUnkServer);
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 void STDMETHODCALLTYPE
 CStdAsyncStubBuffer2_Disconnect(
     IRpcStubBuffer *pthis);
@@ -774,6 +807,7 @@ NdrProxyForwardingFunction32(void);
  * Proxy Dll APIs
  ****************************************************************************/
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 HRESULT
 RPC_ENTRY
@@ -785,6 +819,7 @@ NdrDllGetClassObject (
     IN const CLSID *            pclsid,
     IN CStdPSFactoryBuffer *    pPSFactoryBuffer);
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 HRESULT
 RPC_ENTRY
@@ -831,6 +866,7 @@ NdrDllCanUnloadNow(
 The following new functions were added after the Windows NT 3.5 release.
 *************************************************************************/
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 HRESULT
 RPC_ENTRY
@@ -839,6 +875,7 @@ NdrDllRegisterProxy(
     IN const ProxyFileInfo **   pProxyFileList,
     IN const CLSID *            pclsid);
 
+_Win32_metadata_supported_os_(windows5.0)
 RPCRTAPI
 HRESULT
 RPC_ENTRY
