diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RtmV2.h b/generation/WinSDK/RecompiledIdlHeaders/um/RtmV2.h
index ee4c143d..19d2e1b1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RtmV2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RtmV2.h
@@ -21,6 +21,9 @@ Revision History:
 
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -513,6 +516,7 @@ RTM_EVENT_TYPE, *PRTM_EVENT_TYPE;
 // new entities registering,
 // or entities deregistering
 //
+_Win32_metadata_canonical_name_(RTM_EVENT_CALLBACK)
 typedef
 DWORD
 (WINAPI * _EVENT_CALLBACK) (
@@ -572,6 +576,7 @@ RTM_ENTITY_METHOD_OUTPUT, *PRTM_ENTITY_METHOD_OUTPUT;
 //
 // Common prototype for entity methods
 //
+_Win32_metadata_canonical_name_(RTM_ENTITY_EXPORT_METHOD)
 typedef
 VOID
 (WINAPI * _ENTITY_METHOD) (
@@ -692,6 +697,7 @@ typedef DWORD          RTM_NOTIFY_FLAGS,
 // Registration API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmRegisterEntity (
@@ -703,12 +709,14 @@ RtmRegisterEntity (
     OUT     PRTM_ENTITY_HANDLE              RtmRegHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeregisterEntity (
     IN      RTM_ENTITY_HANDLE               RtmRegHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetRegisteredEntities (
@@ -718,6 +726,7 @@ RtmGetRegisteredEntities (
     OUT     PRTM_ENTITY_INFO                EntityInfos OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseEntities (
@@ -730,6 +739,7 @@ RtmReleaseEntities (
 // Opaque Ptr APIs
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmLockDestination(
@@ -739,6 +749,7 @@ RtmLockDestination(
     IN      BOOL                            LockDest
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetOpaqueInformationPointer (
@@ -751,6 +762,7 @@ RtmGetOpaqueInformationPointer (
 // Export Method API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetEntityMethods (
@@ -760,6 +772,7 @@ RtmGetEntityMethods (
     OUT     PRTM_ENTITY_EXPORT_METHOD       ExptMethods
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmInvokeMethod (
@@ -770,6 +783,7 @@ RtmInvokeMethod (
     OUT     PRTM_ENTITY_METHOD_OUTPUT       Output
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmBlockMethods (
@@ -783,6 +797,7 @@ RtmBlockMethods (
 // Handle to Info Structures
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetEntityInfo (
@@ -791,6 +806,7 @@ RtmGetEntityInfo (
     OUT     PRTM_ENTITY_INFO                EntityInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetDestInfo (
@@ -801,6 +817,7 @@ RtmGetDestInfo (
     OUT     PRTM_DEST_INFO                  DestInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetRouteInfo (
@@ -810,6 +827,7 @@ RtmGetRouteInfo (
     OUT     PRTM_NET_ADDRESS                DestAddress OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetNextHopInfo (
@@ -818,6 +836,7 @@ RtmGetNextHopInfo (
     OUT     PRTM_NEXTHOP_INFO               NextHopInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseEntityInfo (
@@ -825,6 +844,7 @@ RtmReleaseEntityInfo (
     IN      PRTM_ENTITY_INFO               EntityInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseDestInfo (
@@ -832,6 +852,7 @@ RtmReleaseDestInfo (
     IN      PRTM_DEST_INFO                 DestInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseRouteInfo (
@@ -839,6 +860,7 @@ RtmReleaseRouteInfo (
     IN      PRTM_ROUTE_INFO                RouteInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseNextHopInfo (
@@ -851,6 +873,7 @@ RtmReleaseNextHopInfo (
 // RIB Insert/Delete API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmAddRouteToDest (
@@ -865,6 +888,7 @@ RtmAddRouteToDest (
     IN OUT  PRTM_ROUTE_CHANGE_FLAGS         ChangeFlags
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeleteRouteToDest (
@@ -873,6 +897,7 @@ RtmDeleteRouteToDest (
     OUT     PRTM_ROUTE_CHANGE_FLAGS         ChangeFlags
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmHoldDestination (
@@ -882,6 +907,7 @@ RtmHoldDestination (
     IN      ULONG                           HoldTime
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetRoutePointer (
@@ -890,6 +916,7 @@ RtmGetRoutePointer (
     OUT     PRTM_ROUTE_INFO                *RoutePointer
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmLockRoute(
@@ -900,6 +927,7 @@ RtmLockRoute(
     OUT     PRTM_ROUTE_INFO                *RoutePointer OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmUpdateAndUnlockRoute(
@@ -916,6 +944,7 @@ RtmUpdateAndUnlockRoute(
 // RIB Query API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetExactMatchDestination (
@@ -926,6 +955,7 @@ RtmGetExactMatchDestination (
     OUT     PRTM_DEST_INFO                  DestInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetMostSpecificDestination (
@@ -936,6 +966,7 @@ RtmGetMostSpecificDestination (
     OUT     PRTM_DEST_INFO                  DestInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetLessSpecificDestination (
@@ -946,6 +977,7 @@ RtmGetLessSpecificDestination (
     OUT     PRTM_DEST_INFO                  DestInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetExactMatchRoute (
@@ -958,6 +990,7 @@ RtmGetExactMatchRoute (
     OUT     PRTM_ROUTE_HANDLE               RouteHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmIsBestRoute (
@@ -970,6 +1003,7 @@ RtmIsBestRoute (
 // NextHop Object API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmAddNextHop (
@@ -979,6 +1013,7 @@ RtmAddNextHop (
     OUT     PRTM_NEXTHOP_CHANGE_FLAGS       ChangeFlags
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmFindNextHop (
@@ -988,6 +1023,7 @@ RtmFindNextHop (
     OUT     PRTM_NEXTHOP_INFO              *NextHopPointer OPTIONAL
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeleteNextHop (
@@ -996,6 +1032,7 @@ RtmDeleteNextHop (
     IN      PRTM_NEXTHOP_INFO               NextHopInfo
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetNextHopPointer (
@@ -1004,6 +1041,7 @@ RtmGetNextHopPointer (
     OUT     PRTM_NEXTHOP_INFO              *NextHopPointer
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmLockNextHop(
@@ -1019,6 +1057,7 @@ RtmLockNextHop(
 // Enumeration API Prototypes
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmCreateDestEnum (
@@ -1030,6 +1069,7 @@ RtmCreateDestEnum (
     OUT     PRTM_ENUM_HANDLE                RtmEnumHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetEnumDests (
@@ -1039,6 +1079,7 @@ RtmGetEnumDests (
     OUT     PRTM_DEST_INFO                  DestInfos
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseDests (
@@ -1047,6 +1088,7 @@ RtmReleaseDests (
     IN      PRTM_DEST_INFO                  DestInfos
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmCreateRouteEnum (
@@ -1061,6 +1103,7 @@ RtmCreateRouteEnum (
     OUT     PRTM_ENUM_HANDLE                RtmEnumHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetEnumRoutes (
@@ -1070,6 +1113,7 @@ RtmGetEnumRoutes (
     OUT     PRTM_ROUTE_HANDLE               RouteHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseRoutes (
@@ -1078,6 +1122,7 @@ RtmReleaseRoutes (
     IN      PRTM_ROUTE_HANDLE               RouteHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmCreateNextHopEnum (
@@ -1087,6 +1132,7 @@ RtmCreateNextHopEnum (
     OUT     PRTM_ENUM_HANDLE                RtmEnumHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetEnumNextHops (
@@ -1096,6 +1142,7 @@ RtmGetEnumNextHops (
     OUT     PRTM_NEXTHOP_HANDLE             NextHopHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseNextHops (
@@ -1104,6 +1151,7 @@ RtmReleaseNextHops (
     IN      PRTM_NEXTHOP_HANDLE             NextHopHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeleteEnumHandle (
@@ -1116,6 +1164,7 @@ RtmDeleteEnumHandle (
 // Change Notification APIs
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmRegisterForChangeNotification (
@@ -1126,6 +1175,7 @@ RtmRegisterForChangeNotification (
     OUT     PRTM_NOTIFY_HANDLE              NotifyHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetChangedDests (
@@ -1135,6 +1185,7 @@ RtmGetChangedDests (
     OUT     PRTM_DEST_INFO                  ChangedDests
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReleaseChangedDests (
@@ -1144,6 +1195,7 @@ RtmReleaseChangedDests (
     IN      PRTM_DEST_INFO                  ChangedDests
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmIgnoreChangedDests (
@@ -1153,6 +1205,7 @@ RtmIgnoreChangedDests (
     IN      PRTM_DEST_HANDLE                ChangedDests
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetChangeStatus (
@@ -1162,6 +1215,7 @@ RtmGetChangeStatus (
     OUT     PBOOL                           ChangeStatus
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmMarkDestForChangeNotification (
@@ -1171,6 +1225,7 @@ RtmMarkDestForChangeNotification (
     IN      BOOL                            MarkDest
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmIsMarkedForChangeNotification (
@@ -1180,6 +1235,7 @@ RtmIsMarkedForChangeNotification (
     OUT     PBOOL                           DestMarked
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeregisterFromChangeNotification (
@@ -1192,6 +1248,7 @@ RtmDeregisterFromChangeNotification (
 // Entity Specific List APIs
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmCreateRouteList (
@@ -1199,6 +1256,7 @@ RtmCreateRouteList (
     OUT     PRTM_ROUTE_LIST_HANDLE          RouteListHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmInsertInRouteList (
@@ -1208,6 +1266,7 @@ RtmInsertInRouteList (
     IN      PRTM_ROUTE_HANDLE               RouteHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmCreateRouteListEnum (
@@ -1216,6 +1275,7 @@ RtmCreateRouteListEnum (
     OUT     PRTM_ENUM_HANDLE                RtmEnumHandle
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmGetListEnumRoutes (
@@ -1225,6 +1285,7 @@ RtmGetListEnumRoutes (
     _Out_writes_(*NumRoutes) OUT     PRTM_ROUTE_HANDLE               RouteHandles
     );
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmDeleteRouteList (
@@ -1236,6 +1297,7 @@ RtmDeleteRouteList (
 // Handle Management APIs
 //
 
+_Win32_metadata_supported_os_(windowsserver2000)
 DWORD
 WINAPI
 RtmReferenceHandles (
