diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/scesvc.h b/generation/WinSDK/RecompiledIdlHeaders/um/scesvc.h
index 3779e9e5..a4d72b45 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/scesvc.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/scesvc.h
@@ -15,6 +15,9 @@ Revision History:
 --*/
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -250,6 +253,7 @@ typedef ISceSvcAttachmentPersistInfo __RPC_FAR *LPSCESVCATTACHMENTPERSISTINFO;
 
     MIDL_INTERFACE("6d90e0d0-200d-11d1-affb-00c04fb984f9")
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     ISceSvcAttachmentPersistInfo : public IUnknown
     {
     public:
@@ -349,6 +353,7 @@ typedef ISceSvcAttachmentData __RPC_FAR *LPSCESVCATTACHMENTDATA;
 
     MIDL_INTERFACE("17c35fde-200d-11d1-affb-00c04fb984f9")
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     ISceSvcAttachmentData  : public IUnknown
     {
     public:
