diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/securitybaseapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/securitybaseapi.h
index e62d6e9e..92731981 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/securitybaseapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/securitybaseapi.h
@@ -15,6 +15,7 @@
 
 #include <apiset.h>
 #include <apisetcconv.h>
+#include <win32metadata_annotations.h>
 #include <minwindef.h>
 #include <minwinbase.h>
 
@@ -30,6 +31,8 @@ extern "C" {
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -65,6 +68,8 @@ AccessCheckAndAuditAlarmW(
 #define AccessCheckAndAuditAlarm  AccessCheckAndAuditAlarmW
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -82,6 +87,8 @@ AccessCheckByType(
     _Out_ LPBOOL AccessStatus
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -184,6 +191,8 @@ AccessCheckByTypeResultListAndAuditAlarmByHandleW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -194,6 +203,8 @@ AddAccessAllowedAce(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -211,6 +222,8 @@ AddAccessAllowedAceEx(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -224,6 +237,8 @@ AddAccessAllowedObjectAce(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -234,6 +249,8 @@ AddAccessDeniedAce(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -245,6 +262,8 @@ AddAccessDeniedAceEx(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -264,6 +283,8 @@ AddAccessDeniedObjectAce(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -281,6 +302,8 @@ AddAce(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -293,6 +316,8 @@ AddAuditAccessAce(
     _In_ BOOL bAuditFailure
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -306,6 +331,8 @@ AddAuditAccessAceEx(
     _In_ BOOL bAuditFailure
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -329,6 +356,8 @@ AddAuditAccessObjectAce(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -350,6 +379,8 @@ AddMandatoryAce(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -363,6 +394,8 @@ AddResourceAttributeAce(
     _Out_ PDWORD pReturnLength
     );
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -382,6 +415,8 @@ AddScopedPolicyIDAce(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -400,6 +435,8 @@ AdjustTokenGroups(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -418,6 +455,8 @@ AdjustTokenPrivileges(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -435,6 +474,8 @@ AllocateAndInitializeSid(
     _Outptr_ PSID* pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -448,6 +489,7 @@ AllocateLocallyUniqueId(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -456,6 +498,7 @@ AreAllAccessesGranted(
     _In_ DWORD DesiredAccess
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -470,6 +513,8 @@ AreAnyAccessesGranted(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -487,6 +532,8 @@ CheckTokenMembership(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -512,6 +559,8 @@ GetAppContainerAce(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -530,6 +579,8 @@ CheckTokenMembershipEx(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -548,6 +599,8 @@ ConvertToAutoInheritPrivateObjectSecurity(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -563,6 +616,8 @@ CopySid(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -575,6 +630,8 @@ CreatePrivateObjectSecurity(
     _In_ PGENERIC_MAPPING GenericMapping
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -589,6 +646,8 @@ CreatePrivateObjectSecurityEx(
     _In_ PGENERIC_MAPPING GenericMapping
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -604,6 +663,8 @@ CreatePrivateObjectSecurityWithMultipleInheritance(
     _In_ PGENERIC_MAPPING GenericMapping
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -617,6 +678,9 @@ CreateRestrictedToken(
     _In_ DWORD RestrictedSidCount,
     _In_reads_opt_(RestrictedSidCount) PSID_AND_ATTRIBUTES SidsToRestrict,
     _Outptr_ PHANDLE NewTokenHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(CloseHandle)
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM) */
@@ -627,6 +691,8 @@ CreateRestrictedToken(
 
 #if (_WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -638,6 +704,8 @@ CreateWellKnownSid(
     _Inout_ DWORD* cbSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -650,6 +718,8 @@ EqualDomainSid(
 
 #endif //(_WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -664,6 +734,8 @@ DeleteAce(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -677,6 +749,8 @@ DestroyPrivateObjectSecurity(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -684,8 +758,13 @@ DuplicateToken(
     _In_ HANDLE ExistingTokenHandle,
     _In_ SECURITY_IMPERSONATION_LEVEL ImpersonationLevel,
     _Outptr_ PHANDLE DuplicateTokenHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(CloseHandle)
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -696,6 +775,9 @@ DuplicateTokenEx(
     _In_ SECURITY_IMPERSONATION_LEVEL ImpersonationLevel,
     _In_ TOKEN_TYPE TokenType,
     _Outptr_ PHANDLE phNewToken
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(CloseHandle)
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM) */
@@ -704,6 +786,8 @@ DuplicateTokenEx(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -712,6 +796,8 @@ EqualPrefixSid(
     _In_ PSID pSid2
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -720,6 +806,8 @@ EqualSid(
     _In_ PSID pSid2
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -734,6 +822,7 @@ FindFirstFreeAce(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 PVOID
 WINAPI
@@ -741,6 +830,8 @@ FreeSid(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -750,6 +841,8 @@ GetAce(
     _Outptr_ LPVOID* pAce
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -766,6 +859,7 @@ GetAclInformation(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -787,6 +881,8 @@ GetFileSecurityW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -798,6 +894,7 @@ GetKernelObjectSecurity(
     _Out_ LPDWORD lpnLengthNeeded
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 _Post_satisfies_(return >= 8 && return <= SECURITY_MAX_SID_SIZE)
 _Success_(1)
@@ -813,6 +910,8 @@ GetLengthSid(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -831,6 +930,8 @@ GetPrivateObjectSecurity(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -840,6 +941,8 @@ GetSecurityDescriptorControl(
     _Out_ LPDWORD lpdwRevision
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -850,6 +953,8 @@ GetSecurityDescriptorDacl(
     _Out_ LPBOOL lpbDaclDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -859,6 +964,7 @@ GetSecurityDescriptorGroup(
     _Out_ LPBOOL lpbGroupDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -866,6 +972,8 @@ GetSecurityDescriptorLength(
     _In_ PSECURITY_DESCRIPTOR pSecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -875,6 +983,7 @@ GetSecurityDescriptorOwner(
     _Out_ LPBOOL lpbOwnerDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -883,6 +992,8 @@ GetSecurityDescriptorRMControl(
     _Out_ PUCHAR RMControl
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -893,6 +1004,8 @@ GetSecurityDescriptorSacl(
     _Out_ LPBOOL lpbSaclDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 PSID_IDENTIFIER_AUTHORITY
 WINAPI
@@ -900,6 +1013,7 @@ GetSidIdentifierAuthority(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -907,6 +1021,8 @@ GetSidLengthRequired(
     _In_ UCHAR nSubAuthorityCount
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 PDWORD
 WINAPI
@@ -915,6 +1031,8 @@ GetSidSubAuthority(
     _In_ DWORD nSubAuthority
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 PUCHAR
 WINAPI
@@ -922,6 +1040,8 @@ GetSidSubAuthorityCount(
     _In_ PSID pSid
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -935,6 +1055,8 @@ GetTokenInformation(
 
 #if (_WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -953,6 +1075,8 @@ GetWindowsAccountDomainSid(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -960,6 +1084,8 @@ ImpersonateAnonymousToken(
     _In_ HANDLE ThreadHandle
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 _Must_inspect_result_
 WINADVAPI
 BOOL
@@ -968,6 +1094,8 @@ ImpersonateLoggedOnUser(
     _In_ HANDLE hToken
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 _Must_inspect_result_
 WINADVAPI
 BOOL
@@ -982,6 +1110,8 @@ ImpersonateSelf(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -991,6 +1121,8 @@ InitializeAcl(
     _In_ DWORD dwAclRevision
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -999,6 +1131,8 @@ InitializeSecurityDescriptor(
     _In_ DWORD dwRevision
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1014,6 +1148,8 @@ InitializeSid(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1027,6 +1163,7 @@ IsTokenRestricted(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -1034,6 +1171,7 @@ IsValidAcl(
     _In_ PACL pAcl
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -1041,6 +1179,7 @@ IsValidSecurityDescriptor(
     _In_ PSECURITY_DESCRIPTOR pSecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -1050,6 +1189,7 @@ IsValidSid(
 
 #if (_WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 BOOL
 WINAPI
@@ -1060,6 +1200,8 @@ IsWellKnownSid(
 
 #endif // (_WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -1078,6 +1220,8 @@ MakeAbsoluteSD(
     _Inout_ LPDWORD lpdwPrimaryGroupSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 _Success_(return != FALSE)
 BOOL
@@ -1094,6 +1238,7 @@ MakeSelfRelativeSD(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -1166,6 +1311,8 @@ ObjectPrivilegeAuditAlarmW(
 #define ObjectPrivilegeAuditAlarm  ObjectPrivilegeAuditAlarmW
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1192,6 +1339,7 @@ PrivilegedServiceAuditAlarmW(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 VOID
 WINAPI
@@ -1202,6 +1350,8 @@ QuerySecurityAccessMask(
 
 #endif // (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1209,6 +1359,8 @@ RevertToSelf(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1238,6 +1390,8 @@ SetFileSecurityW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1253,6 +1407,8 @@ SetKernelObjectSecurity(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1264,6 +1420,8 @@ SetPrivateObjectSecurity(
     _In_opt_ HANDLE Token
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1278,6 +1436,7 @@ SetPrivateObjectSecurityEx(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 VOID
 WINAPI
@@ -1294,6 +1453,8 @@ SetSecurityAccessMask(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1303,6 +1464,8 @@ SetSecurityDescriptorControl(
     _In_ SECURITY_DESCRIPTOR_CONTROL ControlBitsToSet
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1313,6 +1476,8 @@ SetSecurityDescriptorDacl(
     _In_ BOOL bDaclDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1322,6 +1487,8 @@ SetSecurityDescriptorGroup(
     _In_ BOOL bGroupDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1331,6 +1498,7 @@ SetSecurityDescriptorOwner(
     _In_ BOOL bOwnerDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -1339,6 +1507,8 @@ SetSecurityDescriptorRMControl(
     _In_opt_ PUCHAR RMControl
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1349,6 +1519,8 @@ SetSecurityDescriptorSacl(
     _In_ BOOL bSaclDefaulted
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -1403,6 +1575,7 @@ GetCachedSigningLevel(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINADVAPI
 LONG
 WINAPI
@@ -1420,6 +1593,7 @@ CveEventWrite(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
