diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/SensEvts.h b/generation/WinSDK/RecompiledIdlHeaders/um/SensEvts.h
index 0d202e98..e9793ed0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/SensEvts.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/SensEvts.h
@@ -95,6 +95,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -133,6 +134,7 @@ EXTERN_C const IID IID_ISensNetwork;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("d597bab1-5b9f-11d1-8dd2-00aa004abd5e")
     ISensNetwork : public IDispatch
     {
@@ -370,6 +372,7 @@ EXTERN_C const IID IID_ISensOnNow;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("d597bab2-5b9f-11d1-8dd2-00aa004abd5e")
     ISensOnNow : public IDispatch
     {
@@ -541,6 +544,7 @@ EXTERN_C const IID IID_ISensLogon;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("d597bab3-5b9f-11d1-8dd2-00aa004abd5e")
     ISensLogon : public IDispatch
     {
@@ -768,6 +772,7 @@ EXTERN_C const IID IID_ISensLogon2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("d597bab4-5b9f-11d1-8dd2-00aa004abd5e")
     ISensLogon2 : public IDispatch
     {
