diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/SensAPI.h b/generation/WinSDK/RecompiledIdlHeaders/um/SensAPI.h
index 52940c24..bb75bebb 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/SensAPI.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/SensAPI.h
@@ -32,6 +32,7 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -67,18 +68,24 @@ typedef struct tagQOCINFO
 
 #if !defined(__midl)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 BOOL APIENTRY
 IsDestinationReachableA(
     LPCSTR lpszDestination,
     LPQOCINFO lpQOCInfo
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 BOOL APIENTRY
 IsDestinationReachableW(
     LPCWSTR lpszDestination,
     LPQOCINFO lpQOCInfo
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 BOOL APIENTRY
 IsNetworkAlive(
     LPDWORD lpdwFlags
