diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Sfc.h b/generation/WinSDK/RecompiledIdlHeaders/um/Sfc.h
index 8707a75a..9f196980 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Sfc.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Sfc.h
@@ -24,6 +24,7 @@ Revision History:
 #ifndef _SFC_
 #define _SFC_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -57,6 +58,8 @@ typedef struct _PROTECTED_FILE_DATA {
 } PROTECTED_FILE_DATA, *PPROTECTED_FILE_DATA;
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SfcGetNextProtectedFile(
@@ -64,6 +67,7 @@ SfcGetNextProtectedFile(
     IN PPROTECTED_FILE_DATA ProtFileData
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 SfcIsFileProtected(
@@ -71,6 +75,7 @@ SfcIsFileProtected(
     IN LPCWSTR ProtFileName
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 BOOL
 WINAPI
 SfcIsKeyProtected(
