diff --git a/generation/WinSDK/RecompiledIdlHeaders/winrt/shcore.h b/generation/WinSDK/RecompiledIdlHeaders/winrt/shcore.h
index c15ca96f..c1084155 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/winrt/shcore.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/winrt/shcore.h
@@ -6,10 +6,12 @@
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
 #include <objidl.h> // for IStream
+#include <win32metadata_annotations.h>
 
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI CreateRandomAccessStreamOnFile(_In_ PCWSTR filePath, _In_ DWORD accessMode, _In_ REFIID riid, _COM_Outptr_ void **ppv);
 
 
@@ -19,7 +21,9 @@ typedef enum
     BSOS_PREFERDESTINATIONSTREAM  // in addition, utilize IDestinationStreamFactory::GetDestinationStream.
 } BSOS_OPTIONS;
 
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI CreateRandomAccessStreamOverStream(_In_ IStream *stream, _In_ BSOS_OPTIONS options, _In_ REFIID riid, _COM_Outptr_ void **ppv);
+_Win32_metadata_supported_os_(windows8.0)
 STDAPI CreateStreamOverRandomAccessStream(_In_ IUnknown *randomAccessStream, _In_ REFIID riid, _COM_Outptr_ void **ppv);
 
 #ifdef __cplusplus_winrt
@@ -32,6 +36,7 @@ namespace Windows {
 } /* end namespace */
 
 // Helper that avoids the need to cast IRandomAccessStream^ to IUnknown*.
+_Win32_metadata_supported_os_(windows8.0)
 __inline HRESULT CreateStreamOverRandomAccessStream(_In_ ::Windows::Storage::Streams::IRandomAccessStream ^randomAccessStream, _In_ REFIID riid, _COM_Outptr_ void **ppv)
 {
     return CreateStreamOverRandomAccessStream(reinterpret_cast<IUnknown *>(randomAccessStream), riid, ppv);
