diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/smbclnt.h b/generation/WinSDK/RecompiledIdlHeaders/um/smbclnt.h
index cecff05f..69a09212 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/smbclnt.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/smbclnt.h
@@ -27,6 +27,9 @@ Revision History:
 #endif  // _MSC_VER
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family or RemoteFS Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_REMOTEFS)
@@ -37,6 +40,7 @@ Revision History:
 extern "C" {
 #endif
 
+_Win32_metadata_supported_os_(windowsserver2012)
 DWORD
 WINAPI
 RegisterAppInstance(
@@ -107,6 +111,7 @@ typedef DWORD
 //
 #define SET_APPINSTANCE_CSV_FLAGS_VALID_ONLY_IF_CSV_COORDINATOR 0x00000001
 
+_Win32_metadata_supported_os_(windowsserver2016)
 DWORD
 WINAPI
 SetAppInstanceCsvFlags(
