diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/spellcheck.h b/generation/WinSDK/RecompiledIdlHeaders/um/spellcheck.h
index d4fbec56..91ae68a6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/spellcheck.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/spellcheck.h
@@ -133,6 +133,9 @@ extern "C"{
 #endif
 #if NTDDI_VERSION >= MIN_SPELLING_NTDDI
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 typedef /* [v1_enum] */ 
@@ -169,6 +172,7 @@ EXTERN_C const IID IID_ISpellingError;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("B7C82D61-FBE8-4B47-9B27-6C0D2E0DE0A3")
     ISpellingError : public IUnknown
     {
@@ -295,6 +299,7 @@ EXTERN_C const IID IID_IEnumSpellingError;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("803E3BD4-2828-4410-8290-418D1D73C762")
     IEnumSpellingError : public IUnknown
     {
@@ -382,6 +387,7 @@ EXTERN_C const IID IID_IOptionDescription;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("432E5F85-35CF-4606-A801-6F70277E1D7A")
     IOptionDescription : public IUnknown
     {
@@ -517,6 +523,7 @@ EXTERN_C const IID IID_ISpellCheckerChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("0B83A5B0-792F-4EAB-9799-ACF52C5ED08A")
     ISpellCheckerChangedEventHandler : public IUnknown
     {
@@ -604,6 +611,7 @@ EXTERN_C const IID IID_ISpellChecker;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("B6FD0B71-E2BC-4653-8D05-F197E412770B")
     ISpellChecker : public IUnknown
     {
@@ -888,6 +896,7 @@ EXTERN_C const IID IID_ISpellChecker2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows10.0.10240)
     MIDL_INTERFACE("E7ED1C71-87F7-4378-A840-C9200DACEE47")
     ISpellChecker2 : public ISpellChecker
     {
@@ -1116,6 +1125,7 @@ EXTERN_C const IID IID_ISpellCheckerFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("8E018A9D-2415-4677-BF08-794EA61F94BB")
     ISpellCheckerFactory : public IUnknown
     {
@@ -1249,6 +1259,7 @@ EXTERN_C const IID IID_IUserDictionariesRegistrar;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("AA176B85-0E12-4844-8E1A-EEF1DA77F586")
     IUserDictionariesRegistrar : public IUnknown
     {
