diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/speventq.h b/generation/WinSDK/RecompiledIdlHeaders/um/speventq.h
index 201f44a3..229de27a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/speventq.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/speventq.h
@@ -10,6 +10,9 @@
 #define SPEventQ_h
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -216,6 +219,7 @@ public:
                 pSerEvent->SerializedlParam = sizeof(T);
                 break;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
             case SPET_LPARAM_IS_STRING:
                 StringCchCopyW ((WCHAR *)(pSerEvent + 1), wcslen ((WCHAR*) lParam) + 1, (WCHAR*)lParam);
 
