diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/SrRestorePtApi.h b/generation/WinSDK/RecompiledIdlHeaders/um/SrRestorePtApi.h
index 3a782d47..87140746 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/SrRestorePtApi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/SrRestorePtApi.h
@@ -14,6 +14,7 @@ Abstract:
 #define _SRRESTOREPTAPI_H
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -137,6 +138,7 @@ extern "C" {
 //
 // ERROR_ALREADY_EXISTS       If this is a nested restore point
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL __stdcall
 SRSetRestorePointA(
                   _In_ PRESTOREPOINTINFOA  pRestorePtSpec,// [in] Restore Point specification
@@ -144,6 +146,7 @@ SRSetRestorePointA(
                   );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL __stdcall
 SRSetRestorePointW(
                   _In_ PRESTOREPOINTINFOW pRestorePtSpec,
@@ -157,6 +160,8 @@ SRSetRestorePointInternal(
                   _In_ BOOL           fForceSurrogate
                   );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_import_library_("SrClient.dll")
 DWORD __stdcall
 SRRemoveRestorePoint(DWORD dwRPNum);
 
