diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/stringapiset.h b/generation/WinSDK/RecompiledIdlHeaders/um/stringapiset.h
index 40a9752b..a68c0d63 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/stringapiset.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/stringapiset.h
@@ -12,6 +12,7 @@
 
 #ifndef _APISETSTRING_
 #define _APISETSTRING_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -27,6 +28,8 @@ extern "C" {
 
 #if (WINVER >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -42,6 +45,8 @@ CompareStringEx(
     _Reserved_ LPARAM lParam
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -55,6 +60,7 @@ CompareStringOrdinal(
 
 #endif //(WINVER >= 0x0600)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 int
 WINAPI
@@ -71,6 +77,8 @@ CompareStringW(
 #define CompareString  CompareStringW
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 int
 WINAPI
@@ -86,6 +94,8 @@ FoldStringW(
 #define FoldString  FoldStringW
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -101,6 +111,8 @@ GetStringTypeExW(
 #define GetStringTypeEx  GetStringTypeExW
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -115,6 +127,8 @@ GetStringTypeW(
 //  NLS Code Page Dependent APIs.
 //
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0)
          _When_((cbMultiByte == -1) && (cchWideChar != 0), _Post_equal_to_(_String_length_(lpWideCharStr)+1))
@@ -129,6 +143,8 @@ MultiByteToWideChar(
     _In_ int cchWideChar
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0)
          _When_((cchWideChar == -1) && (cbMultiByte != 0), _Post_equal_to_(_String_length_(lpMultiByteStr)+1))
