diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/strsafe.h b/generation/WinSDK/RecompiledIdlHeaders/shared/strsafe.h
index bb675c4c..414a7235 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/strsafe.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/strsafe.h
@@ -20,6 +20,9 @@
 #include <stdarg.h>         // for va_start, etc.
 #include <specstrings.h>    // for _In_, etc.
 #include <winapifamily.h>   // for WINAPI_FAMILY_PARTITION()
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if !defined(_W64)
 #if !defined(__midl) && (defined(_X86_) || defined(_M_IX86) || defined(_ARM_) || defined(_M_ARM)) && (_MSC_VER >= 1300)
@@ -534,6 +537,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -562,6 +566,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -673,6 +678,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -705,6 +711,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -846,6 +853,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -968,6 +976,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -1200,6 +1209,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyExA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -1327,6 +1337,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyExW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -1537,6 +1548,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyNA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -1575,6 +1587,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyNW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -1699,6 +1712,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyNA(
             _Out_writes_bytes_(cbDest) STRSAFE_LPSTR pszDest,
@@ -1739,6 +1753,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POTENTIAL_BUFFER_OVERFLOW_HIGH_PRIORITY)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyNW(
             _Out_writes_bytes_(cbDest) STRSAFE_LPWSTR pszDest,
@@ -1897,6 +1912,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyNExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -2020,6 +2036,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCopyNExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -2260,6 +2277,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyNExA(
             _Out_writes_bytes_(cbDest) STRSAFE_LPSTR pszDest,
@@ -2385,6 +2403,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POTENTIAL_BUFFER_OVERFLOW_HIGH_PRIORITY)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCopyNExW(
             _Out_writes_bytes_(cbDest) STRSAFE_LPWSTR pszDest,
@@ -2584,6 +2603,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatA(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -2612,6 +2632,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatW(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -2717,6 +2738,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatA(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -2746,6 +2768,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatW(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -2888,6 +2911,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatExA(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -2999,6 +3023,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatExW(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -3223,6 +3248,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatExA(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -3331,6 +3357,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatExW(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -3520,6 +3547,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatNA(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -3556,6 +3584,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatNW(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -3675,6 +3704,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatNA(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -3714,6 +3744,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatNW(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -3869,6 +3900,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatNExA(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -3981,6 +4013,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchCatNExW(
             _Inout_updates_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -4209,6 +4242,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatNExA(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -4322,6 +4356,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POTENTIAL_BUFFER_OVERFLOW_HIGH_PRIORITY)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbCatNExW(
             _Inout_updates_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -4516,6 +4551,7 @@ return value of this function
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchVPrintfA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -4550,6 +4586,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchVPrintfW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -4619,6 +4656,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchVPrintf_lA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -4651,6 +4689,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchVPrintf_lW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -4762,6 +4801,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbVPrintfA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -4795,6 +4835,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbVPrintfW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -4865,6 +4906,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbVPrintf_lA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -4895,6 +4937,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbVPrintf_lW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5006,6 +5049,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchPrintfA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -5041,6 +5085,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchPrintfW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5114,6 +5159,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchPrintf_lA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -5152,6 +5198,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchPrintf_lW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5269,6 +5316,7 @@ return value of this function
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbPrintfA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -5308,6 +5356,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbPrintfW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5384,6 +5433,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbPrintf_lA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -5420,6 +5470,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbPrintf_lW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5571,6 +5622,7 @@ return value of this function
 #pragma warning(push)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchPrintfExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -5704,6 +5756,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchPrintfExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -5875,6 +5928,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchPrintf_lExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -6006,6 +6060,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchPrintf_lExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -6251,6 +6306,7 @@ return value of this function
 #pragma warning(push)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbPrintfExA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -6387,6 +6443,7 @@ STRSAFEAPI
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbPrintfExW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -6558,6 +6615,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbPrintf_lExA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -6686,6 +6744,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbPrintf_lExW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -6930,6 +6989,7 @@ return value of this function
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchVPrintfExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -7059,6 +7119,7 @@ STRSAFEAPI
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchVPrintfExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -7225,6 +7286,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchVPrintf_lExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -7351,6 +7413,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCchVPrintf_lExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -7591,6 +7654,7 @@ return value of this function
 #pragma warning(push)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbVPrintfExA(
             _Out_writes_bytes_(cbDest) STRSAFE_LPSTR pszDest,
@@ -7717,6 +7781,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable : __WARNING_INCORRECT_ANNOTATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbVPrintfExW(
             _Out_writes_bytes_(cbDest) STRSAFE_LPWSTR pszDest,
@@ -7876,6 +7941,7 @@ STRSAFEAPI
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbVPrintf_lExA(
             _Out_writes_bytes_(cbDest) STRSAFE_LPSTR pszDest,
@@ -7997,6 +8063,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STRSAFEAPI
     StringCbVPrintf_lExW(
             _Out_writes_bytes_(cbDest) STRSAFE_LPWSTR pszDest,
@@ -8193,6 +8260,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchGetsA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -8216,6 +8284,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchGetsW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -8315,6 +8384,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbGetsA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -8339,6 +8409,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbGetsW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -8466,6 +8537,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchGetsExA(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -8569,6 +8641,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCchGetsExW(
             _Out_writes_(cchDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -8775,6 +8848,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbGetsExA(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPSTR pszDest,
@@ -8878,6 +8952,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable: __WARNING_POSTCONDITION_NULLTERMINATION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 STRSAFEAPI
     StringCbGetsExW(
             _Out_writes_bytes_(cbDest) _Always_(_Post_z_) STRSAFE_LPWSTR pszDest,
@@ -9048,6 +9123,7 @@ return value of this function.
 #pragma warning(push)
 #pragma warning(disable:__WARNING_POTENTIAL_RANGE_POSTCONDITION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Must_inspect_result_
 STRSAFEAPI
     StringCchLengthA(
@@ -9081,6 +9157,7 @@ STRSAFEAPI
 #pragma warning(push)
 #pragma warning(disable:__WARNING_POTENTIAL_RANGE_POSTCONDITION_VIOLATION)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Must_inspect_result_
 STRSAFEAPI
     StringCchLengthW(
@@ -9177,6 +9254,7 @@ return value of this function.
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Must_inspect_result_
 STRSAFEAPI
     StringCbLengthA(
@@ -9215,6 +9293,7 @@ STRSAFEAPI
 
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Must_inspect_result_
 STRSAFEAPI
     StringCbLengthW(
