diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/SubAuth.h b/generation/WinSDK/RecompiledIdlHeaders/um/SubAuth.h
index 486a3f47..d50cd074 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/SubAuth.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/SubAuth.h
@@ -18,6 +18,9 @@ Revision History:
 #define _NTSUBAUTH_
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -302,6 +305,7 @@ typedef struct _NETLOGON_TARGET_INFO {
 #define MSV1_0_GUEST_LOGON  0x02
 #define MSV1_0_KERBEROS_LOGON  0x04
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NTSTATUS NTAPI
 Msv1_0SubAuthenticationRoutine(
     IN NETLOGON_LOGON_INFO_CLASS LogonLevel,
@@ -347,6 +351,7 @@ typedef struct _MSV1_0_VALIDATION_INFO {
 #define MSV1_0_SUBAUTH_ACCOUNT_TYPE            0x00000040
 #define MSV1_0_SUBAUTH_LOCKOUT                 0x00000080
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NTSTATUS NTAPI
 Msv1_0SubAuthenticationRoutineEx(
     IN NETLOGON_LOGON_INFO_CLASS LogonLevel,
@@ -358,6 +363,7 @@ Msv1_0SubAuthenticationRoutineEx(
     OUT PULONG ActionsPerformed
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NTSTATUS NTAPI
 Msv1_0SubAuthenticationRoutineGeneric(
     IN PVOID SubmitBuffer,
@@ -366,6 +372,7 @@ Msv1_0SubAuthenticationRoutineGeneric(
     OUT PVOID *ReturnBuffer
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 NTSTATUS NTAPI
 Msv1_0SubAuthenticationFilter(
     IN NETLOGON_LOGON_INFO_CLASS LogonLevel,
@@ -373,7 +380,7 @@ Msv1_0SubAuthenticationFilter(
     IN ULONG Flags,
     IN PUSER_ALL_INFORMATION UserAll,
     OUT PULONG WhichFields,
-    OUT PULONG UserFlags,
+    OUT MSV_SUB_AUTHENTICATION_FILTER * UserFlags,
     OUT PBOOLEAN Authoritative,
     OUT PLARGE_INTEGER LogoffTime,
     OUT PLARGE_INTEGER KickoffTime
