diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/synchapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/synchapi.h
index e7866bfd..2e167930 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/synchapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/synchapi.h
@@ -18,6 +18,7 @@
 #include <minwindef.h>
 #include <minwinbase.h>
 
+#include <win32metadata_annotations.h>
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -35,6 +36,7 @@ typedef RTL_SRWLOCK SRWLOCK, *PSRWLOCK;
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -42,6 +44,7 @@ InitializeSRWLock(
     _Out_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 _Releases_exclusive_lock_(*SRWLock)
 _Releases_nonreentrant_lock_(*SRWLock)
@@ -51,6 +54,7 @@ ReleaseSRWLockExclusive(
     _Inout_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 _Releases_shared_lock_(*SRWLock)
 _Releases_nonreentrant_lock_(*SRWLock)
@@ -60,6 +64,7 @@ ReleaseSRWLockShared(
     _Inout_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 _Acquires_exclusive_lock_(*SRWLock)
 _Acquires_nonreentrant_lock_(*SRWLock)
@@ -69,6 +74,7 @@ AcquireSRWLockExclusive(
     _Inout_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 _Acquires_shared_lock_(*SRWLock)
 _Acquires_nonreentrant_lock_(*SRWLock)
@@ -78,6 +84,7 @@ AcquireSRWLockShared(
     _Inout_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 WINBASEAPI
 _When_(return!=0, _Acquires_exclusive_lock_(*SRWLock))
 _When_(return!=0, _Acquires_nonreentrant_lock_(*SRWLock))
@@ -87,6 +94,7 @@ TryAcquireSRWLockExclusive(
     _Inout_ PSRWLOCK SRWLock
     );
 
+_Win32_metadata_supported_os_(windows6.1)
 WINBASEAPI
 _When_(return!=0, _Acquires_shared_lock_(*SRWLock))
 _When_(return!=0, _Acquires_nonreentrant_lock_(*SRWLock))
@@ -100,6 +108,7 @@ TryAcquireSRWLockShared(
 
 #if (_WIN32_WINNT < 0x0600)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Maybe_raises_SEH_exception_
 WINBASEAPI
 VOID
@@ -110,6 +119,7 @@ InitializeCriticalSection(
 
 #else
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -119,6 +129,7 @@ InitializeCriticalSection(
 
 #endif  // (_WIN32_WINNT < 0x0600)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -126,6 +137,7 @@ EnterCriticalSection(
     _Inout_ LPCRITICAL_SECTION lpCriticalSection
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -133,6 +145,8 @@ LeaveCriticalSection(
     _Inout_ LPCRITICAL_SECTION lpCriticalSection
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 BOOL
@@ -144,6 +158,8 @@ InitializeCriticalSectionAndSpinCount(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -155,6 +171,7 @@ InitializeCriticalSectionEx(
 
 #endif // (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 DWORD
 WINAPI
@@ -165,6 +182,7 @@ SetCriticalSectionSpinCount(
 
 #if (_WIN32_WINNT >= 0x0400)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 BOOL
 WINAPI
@@ -174,6 +192,7 @@ TryEnterCriticalSection(
 
 #endif /* _WIN32_WINNT >= 0x0400 */
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -216,6 +235,7 @@ BOOL
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -223,6 +243,8 @@ InitOnceInitialize(
     _Out_ PINIT_ONCE InitOnce
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -233,6 +255,8 @@ InitOnceExecuteOnce(
     _Outptr_opt_result_maybenull_ LPVOID* Context
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -243,6 +267,8 @@ InitOnceBeginInitialize(
     _Outptr_opt_result_maybenull_ LPVOID* lpContext
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -274,6 +300,7 @@ typedef RTL_CONDITION_VARIABLE CONDITION_VARIABLE, *PCONDITION_VARIABLE;
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -281,6 +308,7 @@ InitializeConditionVariable(
     _Out_ PCONDITION_VARIABLE ConditionVariable
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -288,6 +316,7 @@ WakeConditionVariable(
     _Inout_ PCONDITION_VARIABLE ConditionVariable
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -295,6 +324,8 @@ WakeAllConditionVariable(
     _Inout_ PCONDITION_VARIABLE ConditionVariable
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -304,6 +335,8 @@ SleepConditionVariableCS(
     _In_ DWORD dwMilliseconds
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -316,6 +349,8 @@ SleepConditionVariableSRW(
 
 #endif // (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -323,6 +358,8 @@ SetEvent(
     _In_ HANDLE hEvent
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -330,6 +367,8 @@ ResetEvent(
     _In_ HANDLE hEvent
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -339,6 +378,8 @@ ReleaseSemaphore(
     _Out_opt_ LPLONG lpPreviousCount
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -346,6 +387,8 @@ ReleaseMutex(
     _In_ HANDLE hMutex
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -354,6 +397,7 @@ WaitForSingleObject(
     _In_ DWORD dwMilliseconds
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 DWORD
 WINAPI
@@ -362,6 +406,8 @@ SleepEx(
     _In_ BOOL bAlertable
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -371,6 +417,8 @@ WaitForSingleObjectEx(
     _In_ BOOL bAlertable
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -389,6 +437,8 @@ WaitForMultipleObjectsEx(
 #define MUTEX_MODIFY_STATE  MUTANT_QUERY_STATE
 #define MUTEX_ALL_ACCESS    MUTANT_ALL_ACCESS
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -399,6 +449,8 @@ CreateMutexA(
     _In_opt_ LPCSTR lpName
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -415,6 +467,8 @@ CreateMutexW(
 #define CreateMutex  CreateMutexA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -429,6 +483,8 @@ OpenMutexW(
 #define OpenMutex  OpenMutexW
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -440,6 +496,8 @@ CreateEventA(
     _In_opt_ LPCSTR lpName
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -457,6 +515,8 @@ CreateEventW(
 #define CreateEvent  CreateEventA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -467,6 +527,8 @@ OpenEventA(
     _In_ LPCSTR lpName
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -483,6 +545,8 @@ OpenEventW(
 #define OpenEvent  OpenEventA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -507,6 +571,8 @@ VOID
     _In_     DWORD dwTimerHighValue
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -523,6 +589,8 @@ OpenWaitableTimerW(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetWaitableTimerEx(
@@ -537,6 +605,8 @@ SetWaitableTimerEx(
 
 #endif // (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -549,6 +619,8 @@ SetWaitableTimer(
     _In_ BOOL fResume
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -560,6 +632,8 @@ CancelWaitableTimer(
 
 #define CREATE_MUTEX_INITIAL_OWNER  0x00000001
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -571,6 +645,8 @@ CreateMutexExA(
     _In_ DWORD dwDesiredAccess
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -591,6 +667,8 @@ CreateMutexExW(
 #define CREATE_EVENT_MANUAL_RESET   0x00000001
 #define CREATE_EVENT_INITIAL_SET    0x00000002
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -602,6 +680,8 @@ CreateEventExA(
     _In_ DWORD dwDesiredAccess
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -619,6 +699,8 @@ CreateEventExW(
 #define CreateEventEx  CreateEventExA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -641,6 +723,8 @@ CreateSemaphoreExW(
 #define CREATE_WAITABLE_TIMER_HIGH_RESOLUTION 0x00000002
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -668,6 +752,7 @@ typedef PRTL_BARRIER LPSYNCHRONIZATION_BARRIER;
 #define SYNCHRONIZATION_BARRIER_FLAGS_BLOCK_ONLY 0x02
 #define SYNCHRONIZATION_BARRIER_FLAGS_NO_DELETE  0x04
 
+_Win32_metadata_supported_os_(windows8.0)
 BOOL
 WINAPI
 EnterSynchronizationBarrier(
@@ -675,6 +760,8 @@ EnterSynchronizationBarrier(
     _In_ DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 InitializeSynchronizationBarrier(
@@ -683,12 +770,14 @@ InitializeSynchronizationBarrier(
     _In_ LONG lSpinCount
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 BOOL
 WINAPI
 DeleteSynchronizationBarrier(
     _Inout_ LPSYNCHRONIZATION_BARRIER lpBarrier
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -696,6 +785,8 @@ Sleep(
     _In_ DWORD dwMilliseconds
     );
 
+_Win32_metadata_supported_os_(windows8.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WaitOnAddress(
@@ -705,12 +796,14 @@ WaitOnAddress(
     _In_opt_ DWORD dwMilliseconds
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 VOID
 WINAPI
 WakeByAddressSingle(
     _In_ PVOID Address
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 VOID
 WINAPI
 WakeByAddressAll(
@@ -725,6 +818,8 @@ WakeByAddressAll(
 
 #ifndef MIDL_PASS
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -743,6 +838,8 @@ SignalObjectAndWait(
 #pragma region Application or OneCore Family or Games Partition
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 DWORD
 WINAPI
@@ -753,6 +850,8 @@ WaitForMultipleObjects(
     _In_ DWORD dwMilliseconds
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -767,6 +866,8 @@ CreateSemaphoreW(
 #define CreateSemaphore  CreateSemaphoreW
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
