diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/sysinfoapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/sysinfoapi.h
index 8cc6aa8c..7b445e2f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/sysinfoapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/sysinfoapi.h
@@ -37,6 +37,7 @@
 # define NOT_BUILD_WINDOWS_DEPRECATE __declspec(deprecated)
 #endif
 
+#include <win32metadata_annotations.h>
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -80,6 +81,8 @@ typedef struct _MEMORYSTATUSEX {
 #define USER_CET_ENVIRONMENT_VBS_ENCLAVE          0x00000010
 #define USER_CET_ENVIRONMENT_VBS_BASIC_ENCLAVE    0x00000011
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -87,6 +90,7 @@ GlobalMemoryStatusEx(
     _Out_ LPMEMORYSTATUSEX lpBuffer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 VOID
 WINAPI
@@ -94,6 +98,7 @@ GetSystemInfo(
     _Out_ LPSYSTEM_INFO lpSystemInfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 VOID
 WINAPI
@@ -101,6 +106,7 @@ GetSystemTime(
     _Out_ LPSYSTEMTIME lpSystemTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 VOID
 WINAPI
@@ -108,6 +114,7 @@ GetSystemTimeAsFileTime(
     _Out_ LPFILETIME lpSystemTimeAsFileTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINBASEAPI
 VOID
 WINAPI
@@ -150,6 +157,7 @@ GetSystemLeapSecondInformation(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 NOT_BUILD_WINDOWS_DEPRECATE
 WINBASEAPI
 __drv_preferredFunction("IsWindows*", "Deprecated. Use VerifyVersionInfo* or IsWindows* macros from VersionHelpers.")
@@ -159,6 +167,8 @@ GetVersion(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -172,6 +182,7 @@ SetLocalTime(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 __drv_preferredFunction("GetTickCount64", "GetTickCount overflows roughly every 49 days.  Code that does not take that into account can loop indefinitely.  GetTickCount64 operates on 64 bit values and does not have that problem")
 WINBASEAPI
 DWORD
@@ -182,6 +193,7 @@ GetTickCount(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 ULONGLONG
 WINAPI
@@ -197,6 +209,8 @@ GetTickCount64(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -213,6 +227,8 @@ GetSystemTimeAdjustment(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -229,6 +245,8 @@ GetSystemTimeAdjustmentPrecise(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0 && return < uSize)
 UINT
@@ -238,6 +256,8 @@ GetSystemDirectoryA(
     _In_ UINT uSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0 && return < uSize)
 UINT
@@ -259,6 +279,8 @@ GetSystemDirectoryW(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -269,6 +291,8 @@ GetWindowsDirectoryA(
     _In_ UINT uSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -291,6 +315,8 @@ GetWindowsDirectoryW(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0 && return < uSize)
 UINT
@@ -300,6 +326,8 @@ GetSystemWindowsDirectoryA(
     _In_ UINT uSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != 0 && return < uSize)
 UINT
@@ -333,6 +361,8 @@ typedef enum _COMPUTER_NAME_FORMAT {
     ComputerNameMax
 } COMPUTER_NAME_FORMAT ;
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -343,6 +373,8 @@ GetComputerNameExA(
     _Inout_ LPDWORD nSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -365,6 +397,8 @@ GetComputerNameExW(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -377,6 +411,8 @@ SetComputerNameExW(
 #define SetComputerNameEx SetComputerNameExW
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -390,7 +426,9 @@ SetSystemTime(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 NOT_BUILD_WINDOWS_DEPRECATE
+_Win32_metadata_set_last_error_
 WINBASEAPI
 __drv_preferredFunction("IsWindows*", "Deprecated. Use VerifyVersionInfo* or IsWindows* macros from VersionHelpers.")
 BOOL
@@ -399,7 +437,9 @@ GetVersionExA(
     _Inout_ LPOSVERSIONINFOA lpVersionInformation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 NOT_BUILD_WINDOWS_DEPRECATE
+_Win32_metadata_set_last_error_
 WINBASEAPI
 __drv_preferredFunction("IsWindows*", "Deprecated. Use VerifyVersionInfo* or IsWindows* macros from VersionHelpers.")
 BOOL
@@ -414,6 +454,8 @@ GetVersionExW(
 #define GetVersionEx  GetVersionExA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -424,6 +466,8 @@ GetLogicalProcessorInformation(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -437,6 +481,7 @@ GetLogicalProcessorInformationEx(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINXP)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINBASEAPI
 VOID
 WINAPI
@@ -448,6 +493,7 @@ GetNativeSystemInfo(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 VOID
 WINAPI
@@ -465,6 +511,7 @@ GetSystemTimePreciseAsFileTime(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 BOOL
 WINAPI
@@ -486,6 +533,7 @@ GetProductInfo(
 
 #if (NTDDI_VERSION >= NTDDI_WIN2K)
 
+_Win32_metadata_supported_os_(windows5.0)
 NTSYSAPI
 ULONGLONG
 NTAPI
@@ -514,6 +562,8 @@ GetOsSafeBootMode(
 #pragma region OneCore Family or App Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 UINT
 WINAPI
@@ -523,6 +573,8 @@ EnumSystemFirmwareTables(
     _In_ DWORD BufferSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 UINT
 WINAPI
@@ -549,6 +601,8 @@ DnsHostnameToComputerNameExW(
     _Inout_ LPDWORD nSize
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -572,6 +626,8 @@ SetComputerNameEx2W(
 #define SetComputerNameEx2 SetComputerNameEx2W
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -581,6 +637,8 @@ SetSystemTimeAdjustment(
     _In_ BOOL bTimeAdjustmentDisabled
     );
 
+_Win32_metadata_supported_os_(windows10.0.10240)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -590,6 +648,8 @@ SetSystemTimeAdjustmentPrecise(
     _In_ BOOL bTimeAdjustmentDisabled
     );
 
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -602,6 +662,8 @@ InstallELAMCertificateInfo(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -635,6 +697,7 @@ GetOsManufacturingMode(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINTHRESHOLD)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 WINBASEAPI
 HRESULT
 WINAPI
@@ -650,6 +713,8 @@ GetIntegratedDisplaySize(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -657,6 +722,8 @@ SetComputerNameA(
     _In_ LPCSTR lpComputerName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -670,6 +737,8 @@ SetComputerNameW(
 #define SetComputerName  SetComputerNameA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
