diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h b/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
index 4340cf3e..ff2f70e9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
@@ -18,6 +18,7 @@ Contents:   Definitions and prototypes for the TBS (TPM Base Services) for
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or SecureStartup Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_SECURESTARTUP)
@@ -94,20 +95,27 @@ typedef struct tdTBS_CONTEXT_PARAMS
 } TBS_CONTEXT_PARAMS, *PTBS_CONTEXT_PARAMS;
 typedef const TBS_CONTEXT_PARAMS *PCTBS_CONTEXT_PARAMS;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 TBS_RESULT WINAPI
 Tbsi_Context_Create(
     _In_ PCTBS_CONTEXT_PARAMS pContextParams,
-    _Out_ PTBS_HCONTEXT phContext);
+    _Out_ PTBS_HCONTEXT phContext
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(Tbsip_Context_Close));
 
 TBS_RESULT WINAPI
 Tbsi_Tpm_Vendor_Maintenance_Mode(
     _In_ PCTBS_CONTEXT_PARAMS pContextParams,
-    _Out_ PTBS_HCONTEXT phContext);
+    _Out_ PTBS_HCONTEXT phContext
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(Tbsip_Context_Close));
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 TBS_RESULT WINAPI
 Tbsip_Context_Close(
     _In_ TBS_HCONTEXT   hContext);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 TBS_RESULT WINAPI
 Tbsip_Submit_Command(
     _In_ TBS_HCONTEXT hContext,
@@ -118,10 +126,12 @@ Tbsip_Submit_Command(
     _Out_writes_bytes_(*pcbResult) PBYTE pabResult,
     _Inout_ PUINT32 pcbResult);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 TBS_RESULT WINAPI
 Tbsip_Cancel_Commands(
     _In_ TBS_HCONTEXT hContext);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 TBS_RESULT WINAPI
 Tbsi_Physical_Presence_Command(
     _In_ TBS_HCONTEXT hContext,
@@ -134,6 +144,7 @@ Tbsi_Physical_Presence_Command(
 
 #if (NTDDI_VERSION >= NTDDI_VISTASP1)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
 Tbsi_Get_TCG_Log(
@@ -213,11 +224,13 @@ typedef struct _TPM_DEVICE_INFO
 typedef const TPM_DEVICE_INFO *PCTPM_DEVICE_INFO;
 #endif //TPM_VERSION_UNKNOWN
 
+_Win32_metadata_supported_os_(windows8.0)
 TBS_RESULT WINAPI
 Tbsi_GetDeviceInfo(
     _In_ UINT32 Size,
     _Out_writes_bytes_(Size) PVOID Info);
 
+_Win32_metadata_supported_os_(windows8.0)
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
 Tbsi_Get_OwnerAuth(
@@ -226,6 +239,7 @@ Tbsi_Get_OwnerAuth(
     _Out_writes_bytes_to_opt_(*pOutputBufLen, *pOutputBufLen) PBYTE pOutputBuf,
     _Inout_  PUINT32 pOutputBufLen);
 
+_Win32_metadata_supported_os_(windows8.0)
 TBS_RESULT WINAPI
 Tbsi_Revoke_Attestation();
 
@@ -282,6 +296,7 @@ Tbsi_Create_Windows_Key(
 #define TBS_TCGLOG_DRTM_RESUME      5
 // (User mode only) Returns the DRTM log associated with the current resume from hibernation
 
+_Win32_metadata_supported_os_(windows10.0.17134)
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
 Tbsi_Get_TCG_Log_Ex(
