diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/threadpoolapiset.h b/generation/WinSDK/RecompiledIdlHeaders/um/threadpoolapiset.h
index c0e871bb..e74d0189 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/threadpoolapiset.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/threadpoolapiset.h
@@ -12,6 +12,7 @@
 
 #ifndef _THREADPOOLAPISET_H_
 #define _THREADPOOLAPISET_H_
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -42,6 +43,8 @@ VOID
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_POOL
@@ -50,6 +53,7 @@ CreateThreadpool(
     _Reserved_ PVOID reserved
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -58,6 +62,8 @@ SetThreadpoolThreadMaximum(
     _In_ DWORD cthrdMost
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -66,6 +72,8 @@ SetThreadpoolThreadMinimum(
     _In_ DWORD cthrdMic
     );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -74,6 +82,8 @@ SetThreadpoolStackInformation(
     _In_ PTP_POOL_STACK_INFORMATION ptpsi
     );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -82,6 +92,7 @@ QueryThreadpoolStackInformation(
     _Out_ PTP_POOL_STACK_INFORMATION ptpsi
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -89,6 +100,8 @@ CloseThreadpool(
     _Inout_ PTP_POOL ptpp
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_CLEANUP_GROUP
@@ -97,6 +110,7 @@ CreateThreadpoolCleanupGroup(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -106,6 +120,7 @@ CloseThreadpoolCleanupGroupMembers(
     _Inout_opt_ PVOID pvCleanupContext
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -113,6 +128,7 @@ CloseThreadpoolCleanupGroup(
     _Inout_ PTP_CLEANUP_GROUP ptpcg
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -121,6 +137,7 @@ SetEventWhenCallbackReturns(
     _In_ HANDLE evt
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -130,6 +147,7 @@ ReleaseSemaphoreWhenCallbackReturns(
     _In_ DWORD crel
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -138,6 +156,7 @@ ReleaseMutexWhenCallbackReturns(
     _In_ HANDLE mut
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -146,6 +165,7 @@ LeaveCriticalSectionWhenCallbackReturns(
     _Inout_ PCRITICAL_SECTION pcs
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -154,6 +174,7 @@ FreeLibraryWhenCallbackReturns(
     _In_ HMODULE mod
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 BOOL
 WINAPI
@@ -161,6 +182,7 @@ CallbackMayRunLong(
     _Inout_ PTP_CALLBACK_INSTANCE pci
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -168,6 +190,8 @@ DisassociateCurrentThreadFromCallback(
     _Inout_ PTP_CALLBACK_INSTANCE pci
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 BOOL
@@ -178,6 +202,8 @@ TrySubmitThreadpoolCallback(
     _In_opt_ PTP_CALLBACK_ENVIRON pcbe
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_WORK
@@ -188,6 +214,7 @@ CreateThreadpoolWork(
     _In_opt_ PTP_CALLBACK_ENVIRON pcbe
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -195,6 +222,7 @@ SubmitThreadpoolWork(
     _Inout_ PTP_WORK pwk
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -203,6 +231,7 @@ WaitForThreadpoolWorkCallbacks(
     _In_ BOOL fCancelPendingCallbacks
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -210,6 +239,8 @@ CloseThreadpoolWork(
     _Inout_ PTP_WORK pwk
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_TIMER
@@ -220,6 +251,7 @@ CreateThreadpoolTimer(
     _In_opt_ PTP_CALLBACK_ENVIRON pcbe
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -230,6 +262,7 @@ SetThreadpoolTimer(
     _In_opt_ DWORD msWindowLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 BOOL
 WINAPI
@@ -237,6 +270,7 @@ IsThreadpoolTimerSet(
     _Inout_ PTP_TIMER pti
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -245,6 +279,7 @@ WaitForThreadpoolTimerCallbacks(
     _In_ BOOL fCancelPendingCallbacks
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -252,6 +287,8 @@ CloseThreadpoolTimer(
     _Inout_ PTP_TIMER pti
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_WAIT
@@ -262,6 +299,7 @@ CreateThreadpoolWait(
     _In_opt_ PTP_CALLBACK_ENVIRON pcbe
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -271,6 +309,7 @@ SetThreadpoolWait(
     _In_opt_ PFILETIME pftTimeout
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -279,6 +318,7 @@ WaitForThreadpoolWaitCallbacks(
     _In_ BOOL fCancelPendingCallbacks
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -286,6 +326,8 @@ CloseThreadpoolWait(
     _Inout_ PTP_WAIT pwa
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Must_inspect_result_
 PTP_IO
@@ -297,6 +339,7 @@ CreateThreadpoolIo(
     _In_opt_ PTP_CALLBACK_ENVIRON pcbe
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -304,6 +347,7 @@ StartThreadpoolIo(
     _Inout_ PTP_IO pio
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -311,6 +355,7 @@ CancelThreadpoolIo(
     _Inout_ PTP_IO pio
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -319,6 +364,7 @@ WaitForThreadpoolIoCallbacks(
     _In_ BOOL fCancelPendingCallbacks
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINBASEAPI
 VOID
 WINAPI
@@ -332,6 +378,7 @@ CloseThreadpoolIo(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 BOOL
 WINAPI
@@ -342,6 +389,7 @@ SetThreadpoolTimerEx(
     _In_opt_ DWORD msWindowLength
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 BOOL
 WINAPI
