diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/timezoneapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/timezoneapi.h
index 0a3847b6..555f8848 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/timezoneapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/timezoneapi.h
@@ -19,6 +19,7 @@
 #include <minwindef.h>
 #include <minwinbase.h>
 
+#include <win32metadata_annotations.h>
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -54,6 +55,8 @@ typedef struct _TIME_DYNAMIC_ZONE_INFORMATION {
     BOOLEAN DynamicDaylightTimeDisabled;
 } DYNAMIC_TIME_ZONE_INFORMATION, *PDYNAMIC_TIME_ZONE_INFORMATION;
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -64,6 +67,8 @@ SystemTimeToTzSpecificLocalTime(
     _Out_ LPSYSTEMTIME lpLocalTime
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -74,6 +79,8 @@ TzSpecificLocalTimeToSystemTime(
     _Out_ LPSYSTEMTIME lpUniversalTime
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -83,6 +90,8 @@ FileTimeToSystemTime(
     _Out_ LPSYSTEMTIME lpSystemTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -92,6 +101,8 @@ SystemTimeToFileTime(
     _Out_ LPFILETIME lpFileTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != TIME_ZONE_ID_INVALID)
 DWORD
@@ -100,6 +111,8 @@ GetTimeZoneInformation(
     _Out_ LPTIME_ZONE_INFORMATION lpTimeZoneInformation
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -109,6 +122,8 @@ SetTimeZoneInformation(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 BOOL
 WINAPI
@@ -120,6 +135,8 @@ SetDynamicTimeZoneInformation(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != TIME_ZONE_ID_INVALID)
 DWORD
@@ -132,6 +149,8 @@ GetDynamicTimeZoneInformation(
 
 #if (_WIN32_WINNT >= 0x0601)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -147,6 +166,7 @@ GetTimeZoneInformationForYear(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -156,6 +176,7 @@ EnumDynamicTimeZoneInformation(
     _Out_ PDYNAMIC_TIME_ZONE_INFORMATION lpTimeZoneInformation
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 WINBASEAPI
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -166,6 +187,8 @@ GetDynamicTimeZoneInformationEffectiveYears(
     _Out_ LPDWORD LastYear
     );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -176,6 +199,8 @@ SystemTimeToTzSpecificLocalTimeEx(
     _Out_ LPSYSTEMTIME lpLocalTime
     );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
