diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/tpmvscmgr.h b/generation/WinSDK/RecompiledIdlHeaders/um/tpmvscmgr.h
index 535aba72..d19e39b2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/tpmvscmgr.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/tpmvscmgr.h
@@ -112,6 +112,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if (NTDDI_VERSION >= NTDDI_WIN8)
@@ -174,6 +175,7 @@ EXTERN_C const IID IID_ITpmVirtualSmartCardManagerStatusCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("1A1BB35F-ABB8-451C-A1AE-33D98F1BEF4A")
     ITpmVirtualSmartCardManagerStatusCallback : public IUnknown
     {
@@ -290,6 +292,7 @@ EXTERN_C const IID IID_ITpmVirtualSmartCardManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("112B1DFF-D9DC-41F7-869F-D67FEE7CB591")
     ITpmVirtualSmartCardManager : public IUnknown
     {
