diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/TraceLoggingActivity.h b/generation/WinSDK/RecompiledIdlHeaders/shared/TraceLoggingActivity.h
index 3fbb5324..eaf9f255 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/TraceLoggingActivity.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/TraceLoggingActivity.h
@@ -18,6 +18,7 @@ Environment:
 --*/
 
 #pragma once
+#include <win32metadata_annotations.h>
 #ifndef RC_INVOKED
 
 #include "TraceLoggingProvider.h"
