diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/tvratings.h b/generation/WinSDK/RecompiledIdlHeaders/um/tvratings.h
index 5f8a6807..5406cf37 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/tvratings.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/tvratings.h
@@ -99,6 +99,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -117,6 +118,7 @@ EXTERN_C const IID IID_IXDSToRat;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("C5C5C5B0-3ABC-11D6-B25B-00C04FA0C026")
     IXDSToRat : public IDispatch
     {
@@ -291,6 +293,7 @@ EXTERN_C const IID IID_IEvalRat;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("C5C5C5B1-3ABC-11D6-B25B-00C04FA0C026")
     IEvalRat : public IDispatch
     {
