diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Unknwn.h b/generation/WinSDK/RecompiledIdlHeaders/um/Unknwn.h
index 4c533f04..9eaeaf1c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Unknwn.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Unknwn.h
@@ -89,6 +89,7 @@ extern "C"{
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
@@ -113,6 +114,7 @@ typedef /* [unique] */ IUnknown *LPUNKNOWN;
     EXTERN_C const IID IID_IUnknown;
     extern "C++"
     {
+_Win32_metadata_supported_os_(windows5.0)
         MIDL_INTERFACE("00000000-0000-0000-C000-000000000046")
         IUnknown
         {
@@ -141,6 +143,7 @@ typedef /* [unique] */ IUnknown *LPUNKNOWN;
             END_INTERFACE
         };
     } // extern C++
+_Win32_metadata_supported_os_(windows5.0)
     HRESULT STDMETHODCALLTYPE IUnknown_QueryInterface_Proxy(
         IUnknown __RPC_FAR * This,
         /* [in] */ REFIID riid,
@@ -152,6 +155,7 @@ typedef /* [unique] */ IUnknown *LPUNKNOWN;
         PRPC_MESSAGE _pRpcMessage,
         DWORD *_pdwStubPhase);
     
+_Win32_metadata_supported_os_(windows5.0)
     ULONG STDMETHODCALLTYPE IUnknown_AddRef_Proxy(
         IUnknown __RPC_FAR * This);
     
@@ -161,6 +165,7 @@ typedef /* [unique] */ IUnknown *LPUNKNOWN;
         PRPC_MESSAGE _pRpcMessage,
         DWORD *_pdwStubPhase);
     
+_Win32_metadata_supported_os_(windows5.0)
     ULONG STDMETHODCALLTYPE IUnknown_Release_Proxy(
         IUnknown __RPC_FAR * This);
     
@@ -175,6 +180,7 @@ EXTERN_C const IID IID_IUnknown;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("00000000-0000-0000-C000-000000000046")
     IUnknown
     {
@@ -245,6 +251,7 @@ EXTERN_C const IID IID_IUnknown;
 
 
 
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE IUnknown_QueryInterface_Proxy( 
     IUnknown * This,
     /* [annotation][in] */ 
@@ -260,6 +267,7 @@ void __RPC_STUB IUnknown_QueryInterface_Stub(
     DWORD *_pdwStubPhase);
 
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE IUnknown_AddRef_Proxy( 
     IUnknown * This);
 
@@ -271,6 +279,7 @@ void __RPC_STUB IUnknown_AddRef_Stub(
     DWORD *_pdwStubPhase);
 
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG STDMETHODCALLTYPE IUnknown_Release_Proxy( 
     IUnknown * This);
 
@@ -460,6 +469,7 @@ EXTERN_C const IID IID_IClassFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("00000001-0000-0000-C000-000000000046")
     IClassFactory : public IUnknown
     {
