diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h b/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
index dc9d2dad..6636b0be 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
@@ -13,6 +13,7 @@
 #define _INC_USERENV
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -70,12 +71,16 @@ extern "C" {
 #define PI_NOUI         0x00000001      // Prevents displaying of messages
 #define PI_APPLYPOLICY  0x00000002      // Apply NT4 style policy
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
 LoadUserProfileA(
     _In_    HANDLE          hToken,
     _Inout_ LPPROFILEINFOA  lpProfileInfo);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -104,6 +109,8 @@ LoadUserProfileW(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -130,12 +137,16 @@ UnloadUserProfile(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
 GetProfilesDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -176,6 +187,8 @@ GetProfilesDirectoryW(
 #define PT_MANDATORY                 0x00000004      // The loaded profile is mandatory.
 #define PT_ROAMING_PREEXISTING       0x00000008      // The loaded roaming profile is not a brand new one - it was obtained from roamed location
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -202,6 +215,8 @@ GetProfileType(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -209,6 +224,8 @@ DeleteProfileA (
     _In_        LPCSTR    lpSidString,
     _In_opt_    LPCSTR    lpProfilePath,
     _In_opt_    LPCSTR    lpComputerName);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -249,6 +266,7 @@ DeleteProfileW (
 
 #if(WINVER >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 HRESULT
 WINAPI
@@ -281,12 +299,16 @@ CreateProfile(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
 GetDefaultUserProfileDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -322,12 +344,16 @@ GetDefaultUserProfileDirectoryW(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
 GetAllUsersProfileDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -362,6 +388,8 @@ GetAllUsersProfileDirectoryW(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -369,6 +397,8 @@ GetUserProfileDirectoryA(
     _In_                            HANDLE  hToken,
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -409,6 +439,8 @@ GetUserProfileDirectoryW(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 CreateEnvironmentBlock(
@@ -429,6 +461,8 @@ CreateEnvironmentBlock(
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DestroyEnvironmentBlock(
@@ -456,6 +490,8 @@ DestroyEnvironmentBlock(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -464,6 +500,8 @@ ExpandEnvironmentStringsForUserA(
     _In_                    LPCSTR    lpSrc,
     _Out_writes_(dwSize)    LPSTR     lpDest,
     _In_                    DWORD       dwSize);
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -495,6 +533,8 @@ ExpandEnvironmentStringsForUserW(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -522,6 +562,8 @@ RefreshPolicy(
 #define RP_FORCE            1      // Refresh policies without any optimisations.
 #define RP_SYNC             2      // Refresh Policy sync; the call does not return till the time policy processing is completed
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -560,6 +602,8 @@ RefreshPolicyEx(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 HANDLE
 WINAPI
@@ -586,6 +630,8 @@ EnterCriticalPolicySection(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -621,6 +667,8 @@ LeaveCriticalPolicySection(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -645,6 +693,8 @@ RegisterGPNotification(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -833,6 +883,8 @@ typedef PGROUP_POLICY_OBJECTA PGROUP_POLICY_OBJECT;
 #define GPO_LIST_FLAG_NO_WMIFILTERS  0x00000004  // Ignore WMI filters when filtering GPO's
 #define GPO_LIST_FLAG_NO_SECURITYFILTERS 0x00000008 // Ignore security filters
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -843,6 +895,8 @@ GetGPOListA (
     _In_opt_    LPCSTR    lpComputerName,
     _In_        DWORD       dwFlags,
     _Outptr_ PGROUP_POLICY_OBJECTA * pGPOList);
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -878,11 +932,15 @@ GetGPOListW (
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
 FreeGPOListA (
     _In_        PGROUP_POLICY_OBJECTA   pGPOList);
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 BOOL
 WINAPI
@@ -924,6 +982,7 @@ FreeGPOListW (
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 DWORD
 WINAPI
@@ -933,6 +992,7 @@ GetAppliedGPOListA (
     _In_opt_    PSID        pSidUser,
     _In_        GUID        *pGuidExtension,
     _Outptr_ PGROUP_POLICY_OBJECTA   *ppGPOList);
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 DWORD
 WINAPI
@@ -1103,6 +1163,7 @@ typedef DWORD(*PFNGENERATEGROUPPOLICY)(
 
 typedef GUID *REFGPEXTENSIONID;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 DWORD
 WINAPI
@@ -1130,6 +1191,7 @@ ProcessGroupPolicyCompleted(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 DWORD
 WINAPI
@@ -1172,6 +1234,8 @@ ProcessGroupPolicyCompletedEx(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 USERENVAPI
 HRESULT 
 WINAPI
@@ -1214,6 +1278,7 @@ RsopAccessCheckByType(
 
 #if(WINVER >= 0x0500)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 HRESULT 
 WINAPI
@@ -1278,6 +1343,7 @@ typedef struct _POLICYSETTINGSTATUSINFO
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1303,6 +1369,7 @@ RsopSetPolicySettingStatus( _In_ DWORD                       dwFlags,
 //
 //=============================================================================
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1392,6 +1459,7 @@ GenerateGPNotification (
 
 #if(WINVER >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1423,6 +1491,7 @@ CreateAppContainerProfile(
 
 #if(WINVER >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1451,6 +1520,7 @@ DeleteAppContainerProfile(
 
 #if(WINVER >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1480,6 +1550,7 @@ GetAppContainerRegistryLocation(
 
 #if(WINVER >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1508,6 +1579,7 @@ GetAppContainerFolderPath(
 
 #if(WINVER >= 0x0603)
 
+_Win32_metadata_supported_os_(windows10.0.10240)
 USERENVAPI
 HRESULT
 WINAPI
@@ -1544,6 +1616,7 @@ DeriveRestrictedAppContainerSidFromAppContainerSidAndRestrictedName(
 
 #if(WINVER >= 0x0602)
 
+_Win32_metadata_supported_os_(windows8.0)
 USERENVAPI
 HRESULT
 WINAPI
