diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Vfw.h b/generation/WinSDK/RecompiledIdlHeaders/um/Vfw.h
index bdfa6417..34d05bc8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Vfw.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Vfw.h
@@ -41,6 +41,9 @@
 #define _INC_VFW
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -543,6 +546,7 @@ typedef struct {
 	
 ************************************************************************/
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 ICInfo(
@@ -551,6 +555,7 @@ ICInfo(
     _Out_ ICINFO FAR * lpicinfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 ICInstall(
@@ -561,6 +566,7 @@ ICInstall(
     _In_ UINT wFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 ICRemove(
@@ -569,6 +575,7 @@ ICRemove(
     _In_ UINT wFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return != 0)
 LRESULT
 VFWAPI
@@ -578,6 +585,7 @@ ICGetInfo(
     _In_ DWORD cb
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 HIC
 VFWAPI
 ICOpen(
@@ -586,6 +594,7 @@ ICOpen(
     _In_ UINT wMode
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 HIC
 VFWAPI
 ICOpenFunction(
@@ -595,12 +604,14 @@ ICOpenFunction(
     _In_ FARPROC lpfnHandler
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LRESULT
 VFWAPI
 ICClose(
     _In_ HIC hic
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LRESULT
 VFWAPI
 ICSendMessage(
@@ -691,6 +702,7 @@ static DWORD dwICValue;
  *  compress a single frame
  *
  */
+_Win32_metadata_supported_os_(windows5.0)
 DWORD
 VFWAPIV
 ICCompress(
@@ -768,6 +780,7 @@ ICCompress(
  */
 #define ICDECOMPRESS_HURRYUP    0x80000000L     // don't draw just buffer (hurry up!)
 
+_Win32_metadata_supported_os_(windows5.0)
 DWORD
 VFWAPIV
 ICDecompress(
@@ -847,6 +860,7 @@ ICDecompress(
  *  decompress a single frame
  *
  */
+_Win32_metadata_supported_os_(windows5.0)
 static __inline LRESULT VFWAPI_INLINE
 ICDecompressEx(
             _In_ HIC hic,
@@ -893,6 +907,7 @@ ICDecompressEx(
  *  format (lpbiOutput) is supported.
  *
  */
+_Win32_metadata_supported_os_(windows5.0)
 static __inline LRESULT VFWAPI_INLINE
 ICDecompressExBegin(
             _In_      HIC hic,
@@ -935,6 +950,7 @@ ICDecompressExBegin(
  *  ICDecompressExQuery()
  *
  */
+_Win32_metadata_supported_os_(windows5.0)
 static __inline LRESULT VFWAPI_INLINE
 ICDecompressExQuery(
             _In_       HIC hic,
@@ -996,6 +1012,7 @@ ICDecompressExQuery(
 #define ICDRAW_FULLSCREEN   0x00000002L   // draw to full screen
 #define ICDRAW_HDC          0x00000004L   // draw to a HDC/HWND
 
+_Win32_metadata_supported_os_(windows5.0)
 DWORD
 VFWAPIV
 ICDrawBegin(
@@ -1027,6 +1044,7 @@ ICDrawBegin(
 #define ICDRAW_HURRYUP      0x80000000L   // don't draw just buffer (hurry up!)
 #define ICDRAW_UPDATE       0x40000000L   // don't draw just update screen
 
+_Win32_metadata_supported_os_(windows5.0)
 DWORD
 VFWAPIV
 ICDraw(
@@ -1041,6 +1059,7 @@ ICDraw(
 
 // ICMessage is not supported on Win32, so provide a static inline function
 // to do the same job
+_Win32_metadata_supported_os_(windows5.0)
 static __inline LRESULT VFWAPI_INLINE
 ICDrawSuggestFormat(
             _In_  HIC hic,
@@ -1128,6 +1147,7 @@ ICDrawSuggestFormat(
 
 
 // ICMessage is not supported on NT
+_Win32_metadata_supported_os_(windows5.0)
 static __inline LRESULT VFWAPI_INLINE
 ICSetStatusProc(
             _In_ HIC hic,
@@ -1159,6 +1179,7 @@ helper routines for DrawDib and MCIAVI...
 #define ICDrawOpen(fccType, fccHandler, lpbiIn) \
     ICLocate(fccType, fccHandler, lpbiIn, NULL, ICMODE_DRAW)
 
+_Win32_metadata_supported_os_(windows5.0)
 HIC
 VFWAPI
 ICLocate(
@@ -1169,6 +1190,7 @@ ICLocate(
     _In_     WORD wFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 HIC
 VFWAPI
 ICGetDisplayFormat(
@@ -1184,6 +1206,7 @@ ICGetDisplayFormat(
 Higher level functions
 ************************************************************************/
 
+_Win32_metadata_supported_os_(windows5.0)
 HANDLE
 VFWAPI
 ICImageCompress(
@@ -1196,6 +1219,7 @@ ICImageCompress(
     _Inout_opt_  LONG FAR *          plSize      // compress to this size (0=whatever)
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 HANDLE
 VFWAPI
 ICImageDecompress(
@@ -1238,6 +1262,7 @@ typedef struct {
 //
 //  allows user to choose compressor, quality etc...
 //
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 ICCompressorChoose(
@@ -1257,6 +1282,7 @@ ICCompressorChoose(
 						// can handle the input format
 						// or input data
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 ICSeqCompressFrameStart(
@@ -1264,12 +1290,14 @@ ICSeqCompressFrameStart(
     _In_ LPBITMAPINFO lpbiIn
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 void
 VFWAPI
 ICSeqCompressFrameEnd(
     _In_ PCOMPVARS pc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LPVOID
 VFWAPI
 ICSeqCompressFrame(
@@ -1280,6 +1308,7 @@ ICSeqCompressFrame(
     _Inout_opt_ LONG FAR               *plSize     // size to compress to/of returned image
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 void
 VFWAPI
 ICCompressorFree(
@@ -1337,12 +1366,14 @@ extern BOOL VFWAPI DrawDibInit(void);			// ;Internal
 **  DrawDibOpen()
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern HDRAWDIB VFWAPI DrawDibOpen(void);
 
 /*
 **  DrawDibClose()
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1354,6 +1385,7 @@ DrawDibClose(
 ** DrawDibGetBuffer()
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 LPVOID
 VFWAPI
@@ -1375,6 +1407,7 @@ extern UINT VFWAPI DrawDibError(HDRAWDIB hdd);		// ;Internal
 **  get the palette used for drawing DIBs
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 HPALETTE
 VFWAPI
@@ -1389,6 +1422,7 @@ DrawDibGetPalette(
 **  get the palette used for drawing DIBs
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1400,6 +1434,7 @@ DrawDibSetPalette(
 /*
 **  DrawDibChangePalette()
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1416,6 +1451,7 @@ DrawDibChangePalette(
 **  realize the palette in a HDD
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 UINT
 VFWAPI
@@ -1431,6 +1467,7 @@ DrawDibRealize(
 **  start of streaming playback
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1445,6 +1482,7 @@ DrawDibStart(
 **  start of streaming playback
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1458,6 +1496,7 @@ DrawDibStop(
 **  prepare to draw
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1478,6 +1517,7 @@ DrawDibBegin(
 **  actualy draw a DIB to the screen.
 **
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1508,6 +1548,7 @@ DrawDibDraw(
 /*
 **  DrawDibEnd()
 */
+_Win32_metadata_supported_os_(windows5.0)
 extern
 BOOL
 VFWAPI
@@ -1528,6 +1569,7 @@ typedef struct {
     LONG    timeSetDIBits;
 }   DRAWDIBTIME, FAR *LPDRAWDIBTIME;
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 VFWAPI
 DrawDibTime(
@@ -1542,6 +1584,7 @@ DrawDibTime(
 #define PD_STRETCHDIB_1_2_OK    0x0008      /* ... */
 #define PD_STRETCHDIB_1_N_OK    0x0010      /* ... */
 
+_Win32_metadata_supported_os_(windows5.0)
 LRESULT
 VFWAPI
 DrawDibProfileDisplay(
@@ -1550,6 +1593,7 @@ DrawDibProfileDisplay(
 
 
 #ifdef DRAWDIB_INCLUDE_STRETCHDIB
+_Win32_metadata_supported_os_(windows5.0)
 void WINAPI StretchDIB(
 	_In_  LPBITMAPINFOHEADER biDst,
 	_Out_ LPVOID	lpDst,		
@@ -2261,15 +2305,20 @@ DEFINE_AVIGUID(CLSID_AVIFile,           0x00020000, 0, 0);
 // functions
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(void) AVIFileInit(void);   // Call this first!
 STDAPI_(void) AVIFileExit(void);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(ULONG) AVIFileAddRef       (PAVIFILE pfile);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(ULONG) AVIFileRelease      (PAVIFILE pfile);
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileOpenA       (_Outptr_ PAVIFILE FAR * ppfile, _In_ LPCSTR szFile,
 			  UINT uMode, _In_opt_ LPCLSID lpHandler);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileOpenW       (_Outptr_ PAVIFILE FAR * ppfile, _In_ LPCWSTR szFile,
 			  UINT uMode, _In_opt_ LPCLSID lpHandler);
 #ifdef UNICODE
@@ -2278,13 +2327,16 @@ STDAPI AVIFileOpenW       (_Outptr_ PAVIFILE FAR * ppfile, _In_ LPCWSTR szFile,
 #define AVIFileOpen	  AVIFileOpenA	
 #endif
 #else // win16
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileOpen       (PAVIFILE FAR * ppfile, LPCSTR szFile,
 			  UINT uMode, LPCLSID lpHandler);
 #define AVIFileOpenW	AVIFileOpen	    /* ;Internal */
 #endif
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileInfoW (_In_ PAVIFILE pfile, _Out_writes_bytes_(lSize) LPAVIFILEINFOW pfi, _In_ LONG lSize);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileInfoA (_In_ PAVIFILE pfile, _Out_writes_bytes_(lSize) LPAVIFILEINFOA pfi, _In_ LONG lSize);
 #ifdef UNICODE
 #define AVIFileInfo	AVIFileInfoW
@@ -2292,16 +2344,20 @@ STDAPI AVIFileInfoA (_In_ PAVIFILE pfile, _Out_writes_bytes_(lSize) LPAVIFILEINF
 #define AVIFileInfo	AVIFileInfoA
 #endif
 #else //win16 version
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileInfo (PAVIFILE pfile, LPAVIFILEINFO pfi, LONG lSize);
 #define AVIFileInfoW AVIFileInfo	    /* ;Internal */
 #endif
 
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileGetStream     (_In_ PAVIFILE pfile, _Outptr_ PAVISTREAM FAR * ppavi, _In_ DWORD fccType, _In_ LONG lParam);
 
 
 #ifdef _WIN32
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileCreateStreamW (_In_ PAVIFILE pfile, _Outptr_ PAVISTREAM FAR *ppavi, _In_ AVISTREAMINFOW FAR * psi);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileCreateStreamA (_In_ PAVIFILE pfile, _Outptr_ PAVISTREAM FAR *ppavi, _In_ AVISTREAMINFOA FAR * psi);
 #ifdef UNICODE
 #define AVIFileCreateStream	AVIFileCreateStreamW
@@ -2309,24 +2365,32 @@ STDAPI AVIFileCreateStreamA (_In_ PAVIFILE pfile, _Outptr_ PAVISTREAM FAR *ppavi
 #define AVIFileCreateStream	AVIFileCreateStreamA
 #endif
 #else //win16 version
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileCreateStream(PAVIFILE pfile, PAVISTREAM FAR *ppavi, AVISTREAMINFO FAR * psi);
 #define AVIFileCreateStreamW AVIFileCreateStream	    /* ;Internal */
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileWriteData	(_In_ PAVIFILE pfile,
 					 _In_ DWORD ckid,
 					 _In_reads_bytes_(cbData) LPVOID lpData,
 					 _In_ LONG cbData);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileReadData	(_In_ PAVIFILE pfile,
 					 _In_ DWORD ckid,
 					 _Out_writes_bytes_to_(*lpcbData, *lpcbData) LPVOID lpData,
 					 _Inout_ LONG FAR *lpcbData);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIFileEndRecord	(_In_ PAVIFILE pfile);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(ULONG) AVIStreamAddRef       (PAVISTREAM pavi);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(ULONG) AVIStreamRelease      (PAVISTREAM pavi);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamInfoW (_In_ PAVISTREAM pavi, _Out_writes_bytes_(lSize) LPAVISTREAMINFOW psi, _In_ LONG lSize);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamInfoA (_In_ PAVISTREAM pavi, _Out_writes_bytes_(lSize) LPAVISTREAMINFOA psi, _In_ LONG lSize);
 #ifdef UNICODE
 #define AVIStreamInfo	AVIStreamInfoW
@@ -2334,12 +2398,18 @@ STDAPI AVIStreamInfoA (_In_ PAVISTREAM pavi, _Out_writes_bytes_(lSize) LPAVISTRE
 #define AVIStreamInfo	AVIStreamInfoA
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LONG) AVIStreamFindSample(_In_ PAVISTREAM pavi, _In_ LONG lPos, _In_ LONG lFlags);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamReadFormat   (_In_ PAVISTREAM pavi, _In_ LONG lPos,_Out_writes_bytes_to_opt_(*lpcbFormat,*lpcbFormat) LPVOID lpFormat,_Inout_ LONG FAR *lpcbFormat);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamSetFormat    (_In_ PAVISTREAM pavi, _In_ LONG lPos,_In_reads_bytes_(cbFormat) LPVOID lpFormat, _In_ LONG cbFormat);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamReadData     (_In_ PAVISTREAM pavi, _In_ DWORD fcc, _Out_writes_bytes_to_opt_(*lpcb, *lpcb) LPVOID lp, _Inout_ LONG FAR *lpcb);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamWriteData    (_In_ PAVISTREAM pavi, _In_ DWORD fcc, _In_reads_bytes_(cb) LPVOID lp, _In_ LONG cb);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamRead         (_In_ PAVISTREAM pavi,
 			      _In_ LONG lStart,
 			      _In_ LONG lSamples,
@@ -2349,6 +2419,7 @@ STDAPI AVIStreamRead         (_In_ PAVISTREAM pavi,
 			      _Out_opt_ LONG FAR * plSamples);
 #define AVISTREAMREAD_CONVENIENT	(-1L)
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamWrite        (_In_ PAVISTREAM pavi,
 			      _In_ LONG lStart, _In_ LONG lSamples,
 			      _In_reads_bytes_(cbBuffer) LPVOID lpBuffer, _In_ LONG cbBuffer, _In_ DWORD dwFlags,
@@ -2357,21 +2428,30 @@ STDAPI AVIStreamWrite        (_In_ PAVISTREAM pavi,
 
 // Right now, these just use AVIStreamInfo() to get information, then
 // return some of it.  Can they be more efficient?
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LONG) AVIStreamStart        (_In_ PAVISTREAM pavi);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LONG) AVIStreamLength       (_In_ PAVISTREAM pavi);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LONG) AVIStreamTimeToSample (_In_ PAVISTREAM pavi, _In_ LONG lTime);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LONG) AVIStreamSampleToTime (_In_ PAVISTREAM pavi, _In_ LONG lSample);
 
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamBeginStreaming(_In_ PAVISTREAM pavi, _In_ LONG lStart, _In_ LONG lEnd, _In_ LONG lRate);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamEndStreaming(_In_ PAVISTREAM pavi);
 
 //
 // helper functions for using IGetFrame
 //
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(PGETFRAME) AVIStreamGetFrameOpen(_In_ PAVISTREAM pavi,
 					 _In_opt_ LPBITMAPINFOHEADER lpbiWanted);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(LPVOID) AVIStreamGetFrame(_In_ PGETFRAME pg, _In_ LONG lPos);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamGetFrameClose(_In_ PGETFRAME pg);
 
 
@@ -2381,9 +2461,11 @@ STDAPI AVIStreamGetFrameClose(_In_ PGETFRAME pg);
 
 
 // Shortcut function
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamOpenFromFileA(_Outptr_ PAVISTREAM FAR *ppavi, _In_ LPCSTR szFile,
 			     _In_ DWORD fccType, _In_ LONG lParam,
 			     _In_ UINT mode, _In_opt_ CLSID FAR *pclsidHandler);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamOpenFromFileW(_Outptr_ PAVISTREAM FAR *ppavi, LPCWSTR szFile,
 			     _In_ DWORD fccType, _In_ LONG lParam,
 			     _In_ UINT mode, _In_opt_ CLSID FAR *pclsidHandler);
@@ -2394,6 +2476,7 @@ STDAPI AVIStreamOpenFromFileW(_Outptr_ PAVISTREAM FAR *ppavi, LPCWSTR szFile,
 #endif
 
 // Use to create disembodied streams
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIStreamCreate(_Outptr_ PAVISTREAM FAR *ppavi, _In_ LONG lParam1, _In_ LONG lParam2,
 		       _In_opt_ CLSID FAR *pclsidHandler);
 
@@ -2516,12 +2599,14 @@ STDAPI AVIStreamCreate(_Outptr_ PAVISTREAM FAR *ppavi, _In_ LONG lParam1, _In_ L
 #define comptypeDIB         mmioFOURCC('D', 'I', 'B', ' ')
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIMakeCompressedStream(
 		_Outptr_ PAVISTREAM FAR *	    ppsCompressed,
 		_In_ PAVISTREAM		    ppsSource,
 		_In_ AVICOMPRESSOPTIONS FAR *    lpOptions,
 		_In_opt_ CLSID FAR *pclsidHandler);
 
+_Win32_metadata_supported_os_(windows5.0)
 EXTERN_C HRESULT CDECL AVISaveA (LPCSTR               szFile,
 		_In_opt_ CLSID FAR *pclsidHandler,
 		_In_ AVISAVECALLBACK     lpfnCallback,
@@ -2530,12 +2615,14 @@ EXTERN_C HRESULT CDECL AVISaveA (LPCSTR               szFile,
 		_In_ LPAVICOMPRESSOPTIONS lpOptions,
 		...);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVISaveVA(LPCSTR               szFile,
 		_In_opt_ CLSID FAR *pclsidHandler,
 		_In_ AVISAVECALLBACK     lpfnCallback,
 		_In_ int                 nStreams,
 		_In_reads_(nStreams) PAVISTREAM FAR *    ppavi,
 		_In_reads_(nStreams) LPAVICOMPRESSOPTIONS FAR *plpOptions);
+_Win32_metadata_supported_os_(windows5.0)
 EXTERN_C HRESULT CDECL AVISaveW (LPCWSTR               szFile,
 		_In_opt_ CLSID FAR *pclsidHandler,
 		_In_ AVISAVECALLBACK     lpfnCallback,
@@ -2544,6 +2631,7 @@ EXTERN_C HRESULT CDECL AVISaveW (LPCWSTR               szFile,
 		_In_ LPAVICOMPRESSOPTIONS lpOptions,
 		...);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVISaveVW(LPCWSTR               szFile,
 		_In_opt_ CLSID FAR *pclsidHandler,
 		_In_ AVISAVECALLBACK     lpfnCallback,
@@ -2560,12 +2648,14 @@ STDAPI AVISaveVW(LPCWSTR               szFile,
 
 
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI_(INT_PTR) AVISaveOptions(_In_ HWND hwnd,
 			     _In_ UINT	uiFlags,
 			     _In_ int	nStreams,
 			     _In_reads_(nStreams) PAVISTREAM FAR *ppavi,
 			     _Inout_updates_(nStreams) LPAVICOMPRESSOPTIONS FAR *plpOptions);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVISaveOptionsFree(int nStreams,
 			     _In_reads_(nStreams) LPAVICOMPRESSOPTIONS FAR *plpOptions);
 
@@ -2575,17 +2665,21 @@ STDAPI AVISaveOptionsFree(int nStreams,
 // These determine what the compression options dialog for video streams
 // will look like.
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIBuildFilterW(_Out_writes_(cbFilter) LPWSTR lpszFilter, _In_ LONG cbFilter, _In_ BOOL fSaving);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIBuildFilterA(_Out_writes_(cbFilter) LPSTR lpszFilter, _In_ LONG cbFilter, _In_ BOOL fSaving);
 #ifdef UNICODE
 #define AVIBuildFilter	AVIBuildFilterW
 #else
 #define AVIBuildFilter	AVIBuildFilterA
 #endif
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIMakeFileFromStreams(_Outptr_ PAVIFILE FAR *	ppfile,
 			       _In_ int		nStreams,
 			       _In_reads_(nStreams) PAVISTREAM FAR *	papStreams);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIMakeStreamFromClipboard(UINT cfFormat, _In_ HANDLE hGlobal, _Outptr_ PAVISTREAM FAR *ppstream);
 
 /****************************************************************************
@@ -2594,10 +2688,13 @@ STDAPI AVIMakeStreamFromClipboard(UINT cfFormat, _In_ HANDLE hGlobal, _Outptr_ P
  *
  ***************************************************************************/
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIPutFileOnClipboard(_In_ PAVIFILE pf);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIGetFromClipboard(_Outptr_ PAVIFILE FAR * lppf);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI AVIClearClipboard(void);
 
 /****************************************************************************
@@ -2605,22 +2702,31 @@ STDAPI AVIClearClipboard(void);
  *  Editing routines
  *
  ***************************************************************************/
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI CreateEditableStream(
 		_Outptr_ PAVISTREAM FAR *	    ppsEditable,
 		_In_ PAVISTREAM		    psSource);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamCut(_In_ PAVISTREAM pavi, _Inout_ LONG FAR *plStart, _Inout_ LONG FAR *plLength, _Outptr_ PAVISTREAM FAR * ppResult);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamCopy(_In_ PAVISTREAM pavi, _Inout_ LONG FAR *plStart, _Inout_ LONG FAR *plLength, _Outptr_ PAVISTREAM FAR * ppResult);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamPaste(_In_ PAVISTREAM pavi, _Inout_ LONG FAR *plPos, _Inout_ LONG FAR *plLength, _In_ PAVISTREAM pstream, _In_ LONG lStart, _In_ LONG lEnd);
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamClone(_In_ PAVISTREAM pavi, _Outptr_ PAVISTREAM FAR *ppResult);
 
 
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamSetNameA(_In_ PAVISTREAM pavi, _In_ LPCSTR lpszName);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamSetNameW(_In_ PAVISTREAM pavi, _In_ LPCWSTR lpszName);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamSetInfoW(_In_ PAVISTREAM pavi, _In_reads_bytes_(cbInfo) LPAVISTREAMINFOW lpInfo, _In_ LONG cbInfo);
+_Win32_metadata_supported_os_(windows5.0)
 STDAPI EditStreamSetInfoA(_In_ PAVISTREAM pavi, _In_reads_bytes_(cbInfo) LPAVISTREAMINFOA lpInfo, _In_ LONG cbInfo);
 #ifdef UNICODE
 #define EditStreamSetInfo	EditStreamSetInfoW
@@ -2750,6 +2856,7 @@ STDAPI EditStreamSetInfoA(_In_ PAVISTREAM pavi, _In_reads_bytes_(cbInfo) LPAVIST
 #define MCIWndSM SendMessage    /* SendMessage in C */
 #endif  /* __cplusplus */
 #define MCIWND_WINDOW_CLASS TEXT("MCIWndClass")
+_Win32_metadata_supported_os_(windows5.0)
 HWND
 VFWAPIV
 MCIWndCreateA(
@@ -2759,6 +2866,7 @@ MCIWndCreateA(
     _In_opt_ LPCSTR szFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 HWND
 VFWAPIV
 MCIWndCreateW(
@@ -2773,6 +2881,7 @@ MCIWndCreateW(
 #else
 #define MCIWndCreate    MCIWndCreateA
 #endif
+_Win32_metadata_supported_os_(windows5.0)
 BOOL VFWAPIV MCIWndRegisterClass(void);
 
 // Flags for the MCIWndOpen command
@@ -3647,22 +3756,26 @@ typedef LRESULT (CALLBACK* CAPCONTROLCALLBACK)(_In_ HWND hWnd, _In_ int nState);
 //  The only exported functions from AVICAP.DLL
 // ------------------------------------------------------------------
 
+_Win32_metadata_supported_os_(windows5.0)
 HWND VFWAPI capCreateCaptureWindowA (
         LPCSTR lpszWindowName,
         _In_ DWORD dwStyle,
         _In_ int x, _In_ int y, _In_ int nWidth, _In_ int nHeight,
         _In_opt_ HWND hwndParent, _In_ int nID);
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL VFWAPI capGetDriverDescriptionA (UINT wDriverIndex,
         _Out_writes_(cbName) LPSTR lpszName, _In_ int cbName,
         _Out_writes_(cbVer) LPSTR lpszVer, _In_ int cbVer);
 
+_Win32_metadata_supported_os_(windows5.0)
 HWND VFWAPI capCreateCaptureWindowW (
         LPCWSTR lpszWindowName,
         DWORD dwStyle,
         _In_ int x, _In_ int y, _In_ int nWidth, _In_ int nHeight,
         _In_opt_ HWND hwndParent, _In_ int nID);
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL VFWAPI capGetDriverDescriptionW (UINT wDriverIndex,
         _Out_writes_(cbName) LPWSTR lpszName, _In_ int cbName,
         _Out_writes_(cbVer) LPWSTR lpszVer, _In_ int cbVer);
@@ -3779,24 +3892,28 @@ extern "C" {            /* Assume C declarations for C++ */
  ***************************************************************************/
 #ifdef OFN_READONLY
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL
     VFWAPI
     GetOpenFileNamePreviewA(
         _Inout_ LPOPENFILENAMEA lpofn
         );
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL
     VFWAPI
     GetSaveFileNamePreviewA(
         _Inout_ LPOPENFILENAMEA lpofn
         );
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL
     VFWAPI
     GetOpenFileNamePreviewW(
         _Inout_ LPOPENFILENAMEW lpofn
         );
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL
     VFWAPI
     GetSaveFileNamePreviewW(
