diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/vsadmin.h b/generation/WinSDK/RecompiledIdlHeaders/um/vsadmin.h
index 6bb390bd..03042687 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/vsadmin.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/vsadmin.h
@@ -87,6 +87,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -105,6 +106,7 @@ EXTERN_C const IID IID_IVssAdmin;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("77ED5996-2F63-11d3-8A39-00C04F72D8E3")
     IVssAdmin : public IUnknown
     {
