diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h b/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
index e95397d4..c350f9b1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
@@ -15,6 +15,7 @@ Abstract:
 #ifndef _VSBACKUP_H_
 #define _VSBACKUP_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or VSS Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_VSS)
@@ -791,11 +792,13 @@ public:
 #define VSS_SW_BOOTABLE_STATE   (1 << 0)
 
 extern "C"  {
+_Win32_metadata_supported_os_(windows5.1.2600)
     HRESULT STDAPICALLTYPE CreateVssBackupComponentsInternal
         (
         _Out_ IVssBackupComponents **ppBackup
         );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     HRESULT STDAPICALLTYPE CreateVssExamineWriterMetadataInternal
         (
         _In_ BSTR bstrXML,
@@ -803,6 +806,7 @@ extern "C"  {
         );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     HRESULT APIENTRY IsVolumeSnapshottedInternal
         (
         _In_z_ VSS_PWSZ  pwszVolumeName,
@@ -813,6 +817,7 @@ extern "C"  {
 /////////////////////////////////////////////////////////////////////
 // Life-management methods for structure members
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     void APIENTRY VssFreeSnapshotPropertiesInternal(
         _In_ VSS_SNAPSHOT_PROP*  pProp
         );
@@ -831,6 +836,7 @@ extern "C"  {
         );
 }
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 inline HRESULT  CreateVssBackupComponents
     (
     _Out_ IVssBackupComponents **ppBackup
@@ -839,6 +845,7 @@ inline HRESULT  CreateVssBackupComponents
     return CreateVssBackupComponentsInternal(ppBackup);
 }
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 inline HRESULT CreateVssExamineWriterMetadata
     (
     _In_ BSTR bstrXML,
@@ -848,6 +855,7 @@ inline HRESULT CreateVssExamineWriterMetadata
     return CreateVssExamineWriterMetadataInternal(bstrXML, ppMetadata);
 }
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 inline HRESULT IsVolumeSnapshotted
     (
     _In_z_ VSS_PWSZ  pwszVolumeName,
@@ -858,6 +866,7 @@ inline HRESULT IsVolumeSnapshotted
     return IsVolumeSnapshottedInternal(pwszVolumeName, pbSnapshotsPresent, plSnapshotCapability);
 }
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 inline void VssFreeSnapshotProperties
     (
     _In_ VSS_SNAPSHOT_PROP* pProp
@@ -876,6 +885,7 @@ inline HRESULT GetProviderMgmtInterface
     return GetProviderMgmtInterfaceInternal(ProviderId, InterfaceId, ppItf);
 }
 
+_Win32_metadata_supported_os_(windowsserver2008)
 inline HRESULT ShouldBlockRevert
     (
     _In_ LPCWSTR wszVolumeName,
