diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WbemCli.h b/generation/WinSDK/RecompiledIdlHeaders/um/WbemCli.h
index 286236ce..13952da9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WbemCli.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WbemCli.h
@@ -430,6 +430,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family or WinMgmt Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WINMGMT)
 
@@ -893,6 +896,7 @@ EXTERN_C const IID IID_IWbemClassObject;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("dc12a681-737f-11cf-884d-00aa004b2e24")
     IWbemClassObject : public IUnknown
     {
@@ -1375,6 +1379,7 @@ EXTERN_C const IID IID_IWbemObjectAccess;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("49353c9a-516b-11d1-aea6-00c04fb68820")
     IWbemObjectAccess : public IWbemClassObject
     {
@@ -1896,6 +1901,7 @@ EXTERN_C const IID IID_IWbemQualifierSet;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("dc12a680-737f-11cf-884d-00aa004b2e24")
     IWbemQualifierSet : public IUnknown
     {
@@ -2083,6 +2089,7 @@ EXTERN_C const IID IID_IWbemServices;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9556dc99-828c-11cf-a37e-00aa003240c7")
     IWbemServices : public IUnknown
     {
@@ -2736,6 +2743,7 @@ EXTERN_C const IID IID_IWbemLocator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("dc12a687-737f-11cf-884d-00aa004b2e24")
     IWbemLocator : public IUnknown
     {
@@ -2851,6 +2859,7 @@ EXTERN_C const IID IID_IWbemObjectSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("7c857801-7381-11cf-884d-00aa004b2e24")
     IWbemObjectSink : public IUnknown
     {
@@ -2967,6 +2976,7 @@ EXTERN_C const IID IID_IEnumWbemClassObject;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("027947e1-d731-11ce-a357-000000000001")
     IEnumWbemClassObject : public IUnknown
     {
@@ -3122,6 +3132,7 @@ EXTERN_C const IID IID_IWbemCallResult;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("44aca675-e8fc-11d0-a07c-00c04fb68820")
     IWbemCallResult : public IUnknown
     {
@@ -3264,6 +3275,7 @@ EXTERN_C const IID IID_IWbemContext;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("44aca674-e8fc-11d0-a07c-00c04fb68820")
     IWbemContext : public IUnknown
     {
@@ -3479,6 +3491,7 @@ EXTERN_C const IID IID_IUnsecuredApartment;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("1cfaba8c-1523-11d1-ad79-00c04fd8fdff")
     IUnsecuredApartment : public IUnknown
     {
@@ -3570,6 +3583,7 @@ EXTERN_C const IID IID_IWbemUnsecuredApartment;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("31739d04-3471-4cf4-9a7c-57a44ae71956")
     IWbemUnsecuredApartment : public IUnsecuredApartment
     {
@@ -3681,6 +3695,7 @@ EXTERN_C const IID IID_IWbemStatusCodeText;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("eb87e1bc-3233-11d2-aec9-00c04fb68820")
     IWbemStatusCodeText : public IUnknown
     {
@@ -3805,6 +3820,7 @@ EXTERN_C const IID IID_IWbemBackupRestore;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("C49E32C7-BC8B-11d2-85D4-00105A1F8304")
     IWbemBackupRestore : public IUnknown
     {
@@ -3913,6 +3929,7 @@ EXTERN_C const IID IID_IWbemBackupRestoreEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("A359DEC5-E813-4834-8A2A-BA7F1D777D76")
     IWbemBackupRestoreEx : public IWbemBackupRestore
     {
@@ -4028,6 +4045,7 @@ EXTERN_C const IID IID_IWbemRefresher;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("49353c99-516b-11d1-aea6-00c04fb68820")
     IWbemRefresher : public IUnknown
     {
@@ -4115,6 +4133,7 @@ EXTERN_C const IID IID_IWbemHiPerfEnum;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("2705C288-79AE-11d2-B348-00105A1F8177")
     IWbemHiPerfEnum : public IUnknown
     {
@@ -4273,6 +4292,7 @@ EXTERN_C const IID IID_IWbemConfigureRefresher;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("49353c92-516b-11d1-aea6-00c04fb68820")
     IWbemConfigureRefresher : public IUnknown
     {
@@ -4576,6 +4596,7 @@ EXTERN_C const IID IID_IWbemObjectSinkEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e7d35cfa-348b-485e-b524-252725d697ca")
     IWbemObjectSinkEx : public IWbemObjectSink
     {
@@ -4786,6 +4807,7 @@ EXTERN_C const IID IID_IWbemShutdown;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("b7b31df9-d515-11d3-a11c-00105a1f515a")
     IWbemShutdown : public IUnknown
     {
@@ -4907,6 +4929,7 @@ EXTERN_C const IID IID_IWbemObjectTextSrc;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("bfbf883a-cad7-11d3-a11b-00105a1f515a")
     IWbemObjectTextSrc : public IUnknown
     {
@@ -5077,6 +5100,7 @@ EXTERN_C const IID IID_IMofCompiler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("6daf974e-2e37-11d2-aec9-00c04fb68820")
     IMofCompiler : public IUnknown
     {
@@ -5324,14 +5348,22 @@ unsigned char * __RPC_USER  VARIANT_UserMarshal(  __RPC__in unsigned long *, __R
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
 void                      __RPC_USER  VARIANT_UserFree(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  VARIANT_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  VARIANT_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out VARIANT * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned long *, __RPC__in VARIANT * ); 
 
 /* end of Additional Prototypes */
