diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wcmapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/wcmapi.h
index f65e2bfb..db56ad2b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wcmapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wcmapi.h
@@ -7,6 +7,9 @@
 #ifndef _WCMAPI_H
 #define _WCMAPI_H
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -135,6 +138,7 @@ extern "C"
 
 #ifndef __midl
 
+_Win32_metadata_supported_os_(windows8.0)
     DWORD WINAPI WcmQueryProperty(
         _In_opt_ const GUID* pInterface, // NULL for global properties
         _In_opt_ LPCWSTR strProfileName,
@@ -143,6 +147,7 @@ extern "C"
         _Out_ PDWORD pdwDataSize,
         _Outptr_result_buffer_maybenull_(*pdwDataSize) PBYTE* ppData);
 
+_Win32_metadata_supported_os_(windows8.0)
     DWORD WINAPI WcmSetProperty(
         _In_opt_ const GUID* pInterface, // NULL for global properties
         _In_opt_ LPCWSTR strProfileName,
@@ -151,12 +156,15 @@ extern "C"
         _In_ DWORD dwDataSize,
         _In_reads_opt_(dwDataSize) const BYTE* pbData);
 
+_Win32_metadata_supported_os_(windows8.0)
     DWORD WINAPI WcmGetProfileList(_Reserved_ void* pReserved, _Outptr_ WCM_PROFILE_INFO_LIST** ppProfileList);
 
     // Ethernet pseudo-profiles should not be included in pProfileList
+_Win32_metadata_supported_os_(windows8.0)
     DWORD WINAPI WcmSetProfileList(
         _In_ WCM_PROFILE_INFO_LIST* pProfileList, _In_ DWORD dwPosition, _In_ BOOL fIgnoreUnknownProfiles, _Reserved_ void* pReserved);
 
+_Win32_metadata_supported_os_(windows8.0)
     void WINAPI WcmFreeMemory(__deallocate(Mem) void* pMemory);
 
 #endif // not __midl
