diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WcnDevice.h b/generation/WinSDK/RecompiledIdlHeaders/um/WcnDevice.h
index 5cf1abc0..9f2a3c84 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WcnDevice.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WcnDevice.h
@@ -99,6 +99,7 @@ Abstract:
 
 --*/
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if NTDDI_VERSION >= NTDDI_WIN7
@@ -158,6 +159,7 @@ EXTERN_C const IID IID_IWCNDevice;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("C100BE9C-D33A-4a4b-BF23-BBEF4663D017")
     IWCNDevice : public IUnknown
     {
@@ -392,6 +394,7 @@ EXTERN_C const IID IID_IWCNConnectNotify;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("C100BE9F-D33A-4a4b-BF23-BBEF4663D017")
     IWCNConnectNotify : public IUnknown
     {
