diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wct.h b/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
index d3441432..b61cba80 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
@@ -21,6 +21,9 @@ History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -105,13 +108,15 @@ typedef VOID (CALLBACK *PWAITCHAINCALLBACK) (
 #define WCT_ASYNC_OPEN_FLAG 0x1
 #define WCTP_OPEN_ALL_FLAGS (WCT_ASYNC_OPEN_FLAG)
 
+_Win32_metadata_set_last_error_
 _Check_return_
 WINADVAPI
 HWCT WINAPI
 OpenThreadWaitChainSession (
     _In_ DWORD Flags,
     _In_opt_ PWAITCHAINCALLBACK callback
-    );
+    )
+    _Win32_metadata_raii_free_(CloseThreadWaitChainSession);
 
 WINADVAPI
 VOID WINAPI
@@ -125,6 +130,7 @@ CloseThreadWaitChainSession (
 #define WCT_NETWORK_IO_FLAG      0x8
 #define WCTP_GETINFO_ALL_FLAGS (WCT_OUT_OF_PROC_FLAG|WCT_OUT_OF_PROC_COM_FLAG|WCT_OUT_OF_PROC_CS_FLAG)
 
+_Win32_metadata_set_last_error_
 _Check_return_
 WINADVAPI
 BOOL WINAPI
