diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wia_lh.h b/generation/WinSDK/RecompiledIdlHeaders/um/wia_lh.h
index 5dee0e1f..611c4420 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wia_lh.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wia_lh.h
@@ -265,6 +265,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -330,6 +331,7 @@ EXTERN_C const IID IID_IWiaDevMgr;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("5eb2502a-8cf1-11d1-bf92-0060081ed811")
     IWiaDevMgr : public IUnknown
     {
@@ -680,6 +682,7 @@ EXTERN_C const IID IID_IEnumWIA_DEV_INFO;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("5e38b83c-8cf1-11d1-bf92-0060081ed811")
     IEnumWIA_DEV_INFO : public IUnknown
     {
@@ -824,6 +827,7 @@ EXTERN_C const IID IID_IWiaEventCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("ae6287b0-0084-11d2-973b-00a0c9068f2e")
     IWiaEventCallback : public IUnknown
     {
@@ -940,6 +944,7 @@ EXTERN_C const IID IID_IWiaDataCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("a558a866-a5b0-11d2-a08f-00c04f72dc3c")
     IWiaDataCallback : public IUnknown
     {
@@ -1089,6 +1094,7 @@ EXTERN_C const IID IID_IWiaDataTransfer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("a6cef998-a5b0-11d2-a08f-00c04f72dc3c")
     IWiaDataTransfer : public IUnknown
     {
@@ -1247,6 +1253,7 @@ EXTERN_C const IID IID_IWiaItem;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("4db1ad10-3391-11d2-9a33-00c04fa36145")
     IWiaItem : public IUnknown
     {
@@ -1531,6 +1538,7 @@ EXTERN_C const IID IID_IWiaPropertyStorage;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("98B5E8A0-29CC-491a-AAC0-E6DB4FDCCEB6")
     IWiaPropertyStorage : public IUnknown
     {
@@ -1842,6 +1850,7 @@ EXTERN_C const IID IID_IEnumWiaItem;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("5e8383fc-3391-11d2-9a33-00c04fa36145")
     IEnumWiaItem : public IUnknown
     {
@@ -2010,6 +2019,7 @@ EXTERN_C const IID IID_IEnumWIA_DEV_CAPS;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("1fcc4287-aca6-11d2-a093-00c04f72dc3c")
     IEnumWIA_DEV_CAPS : public IUnknown
     {
@@ -2154,6 +2164,7 @@ EXTERN_C const IID IID_IEnumWIA_FORMAT_INFO;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("81BEFC5B-656D-44f1-B24C-D41D51B4DC81")
     IEnumWIA_FORMAT_INFO : public IUnknown
     {
@@ -2298,6 +2309,7 @@ EXTERN_C const IID IID_IWiaLog;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("A00C10B6-82A1-452f-8B6C-86062AAD6890")
     IWiaLog : public IUnknown
     {
@@ -2410,6 +2422,7 @@ EXTERN_C const IID IID_IWiaLogEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("AF1F22AC-7A40-4787-B421-AEb47A1FBD0B")
     IWiaLogEx : public IUnknown
     {
@@ -2554,6 +2567,7 @@ EXTERN_C const IID IID_IWiaNotifyDevMgr;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("70681EA0-E7BF-4291-9FB1-4E8813A3F78E")
     IWiaNotifyDevMgr : public IUnknown
     {
@@ -2636,6 +2650,7 @@ EXTERN_C const IID IID_IWiaItemExtras;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("6291ef2c-36ef-4532-876a-8e132593778d")
     IWiaItemExtras : public IUnknown
     {
@@ -4485,9 +4500,13 @@ unsigned char * __RPC_USER  STGMEDIUM_UserMarshal(  __RPC__in unsigned long *, _
 unsigned char * __RPC_USER  STGMEDIUM_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out STGMEDIUM * ); 
 void                      __RPC_USER  STGMEDIUM_UserFree(     __RPC__in unsigned long *, __RPC__in STGMEDIUM * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 unsigned long             __RPC_USER  HWND_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in HWND * ); 
@@ -4495,9 +4514,13 @@ unsigned char * __RPC_USER  HWND_UserMarshal64(  __RPC__in unsigned long *, __RP
 unsigned char * __RPC_USER  HWND_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HWND * ); 
 void                      __RPC_USER  HWND_UserFree64(     __RPC__in unsigned long *, __RPC__in HWND * ); 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Win32_metadata_supported_os_(windows5.1.2600)
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
 unsigned long             __RPC_USER  STGMEDIUM_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in STGMEDIUM * ); 
