diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winbio_adapter.h b/generation/WinSDK/RecompiledIdlHeaders/um/winbio_adapter.h
index b025bf4d..bc1ce09b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winbio_adapter.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winbio_adapter.h
@@ -22,6 +22,7 @@ Environment:
 #ifndef _WINBIO_ADAPTER_H_2C0E14E1_5330_4f60_9B4F_836CFFD7452A_
 #define _WINBIO_ADAPTER_H_2C0E14E1_5330_4f60_9B4F_836CFFD7452A_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -542,6 +543,7 @@ typedef struct _WINBIO_SENSOR_INTERFACE {
 // Interface-retrieval function exported by the Sensor Adapter
 // plug-in DLL. It *MUST* be called 'WbioQuerySensorInterface'.
 //
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 WbioQuerySensorInterface(
@@ -1839,6 +1841,7 @@ typedef struct _WINBIO_ENGINE_INTERFACE {
 // Interface-retrieval function exported by the Engine Adapter
 // plug-in DLL. It *MUST* be called 'WbioQueryEngineInterface'.
 //
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 WbioQueryEngineInterface(
@@ -3359,6 +3362,7 @@ typedef struct _WINBIO_STORAGE_INTERFACE {
 // Interface-retrieval function exported by the Storage Adapter
 // plug-in DLL. It *MUST* be called 'WbioQueryStorageInterface'.
 //
+_Win32_metadata_supported_os_(windows6.1)
 HRESULT
 WINAPI
 WbioQueryStorageInterface(
