diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winconp.h b/generation/WinSDK/RecompiledIdlHeaders/um/winconp.h
index d761cec6..2ceedb70 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winconp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winconp.h
@@ -15,6 +15,7 @@ extern "C" {
 #endif
 
 #include <wincontypes.h>
+#include <win32metadata_annotations.h>
 
 // ...
 
@@ -379,7 +380,8 @@ OpenConsoleW(
     _In_ LPWSTR lpConsoleDevice,
     _In_ DWORD dwDesiredAccess,
     _In_ BOOL bInheritHandle,
-    _In_ DWORD dwShareMode);
+    _In_ DWORD dwShareMode)
+    _Win32_metadata_raii_free_(CloseConsoleHandle);
 
 HANDLE
 APIENTRY
@@ -387,7 +389,8 @@ DuplicateConsoleHandle(
     _In_ HANDLE hSourceHandle,
     _In_ DWORD dwDesiredAccess,
     _In_ BOOL bInheritHandle,
-    _In_ DWORD dwOptions);
+    _In_ DWORD dwOptions)
+    _Win32_metadata_raii_free_(CloseConsoleHandle);
 
 // ...
 
@@ -424,6 +427,7 @@ GetConsoleInputWaitHandle(
 #define PID_CONSOLE_TERMINALSCROLLING       13
 
 #define CONSOLE_PROPKEY(name, id) \
+_Win32_metadata_supported_os_(windows6.0.6000)
 DEFINE_PROPERTYKEY(name, 0x0C570607, 0x0396, 0x43DE, 0x9D, 0x61, 0xE3, 0x21, 0xD7, 0xDF, 0x50, 0x26, id);
 
 CONSOLE_PROPKEY(PKEY_Console_ForceV2,                   PID_CONSOLE_FORCEV2);
