diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winddi.h b/generation/WinSDK/RecompiledIdlHeaders/um/winddi.h
index 98820f16..b6db947a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winddi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winddi.h
@@ -18,6 +18,7 @@ Abstract:
 #define _WINDDI_
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -1747,22 +1748,26 @@ typedef struct _XLATEOBJ
  * BRUSHOBJ callbacks
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PVOID APIENTRY BRUSHOBJ_pvAllocRbrush(
     BRUSHOBJ *pbo,
     ULONG     cj
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PVOID APIENTRY BRUSHOBJ_pvGetRbrush(
     BRUSHOBJ *pbo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY BRUSHOBJ_ulGetBrushColor(
     BRUSHOBJ *pbo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY BRUSHOBJ_hGetColorTransform(
     BRUSHOBJ *pbo
@@ -1802,6 +1807,7 @@ typedef struct _ENUMRECTS
     RECTL       arcl[1];
 } ENUMRECTS;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY CLIPOBJ_cEnumStart(
     CLIPOBJ *pco,
@@ -1811,6 +1817,7 @@ ULONG APIENTRY CLIPOBJ_cEnumStart(
     __in_data_source(USER_MODE) ULONG    cLimit
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY CLIPOBJ_bEnum(
     CLIPOBJ *pco,
@@ -1818,6 +1825,7 @@ BOOL APIENTRY CLIPOBJ_bEnum(
     ULONG   *pul
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 PATHOBJ* APIENTRY CLIPOBJ_ppoGetPath(
     CLIPOBJ* pco
     );
@@ -1932,17 +1940,20 @@ typedef struct _FONTINFO /* fi */
     ULONG   cjMaxGlyph32;
 } FONTINFO, *PFONTINFO;
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG APIENTRY FONTOBJ_cGetAllGlyphHandles(
     FONTOBJ *pfo,
     HGLYPH  *phg
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 VOID APIENTRY FONTOBJ_vGetInfo(
     FONTOBJ  *pfo,
     __in_data_source(USER_MODE) ULONG     cjSize,
     FONTINFO *pfi
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG APIENTRY FONTOBJ_cGetGlyphs(
     FONTOBJ *pfo,
     __in_data_source(USER_MODE) ULONG    iMode,
@@ -1956,29 +1967,35 @@ GAMMA_TABLES* APIENTRY FONTOBJ_pGetGammaTables(
     FONTOBJ *pfo);
 #endif // (NTDDI_VERSION <= NTDDI_WINXP)
 
+_Win32_metadata_supported_os_(windows5.0)
 XFORMOBJ * APIENTRY FONTOBJ_pxoGetXform(
     _In_ FONTOBJ *pfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 IFIMETRICS * APIENTRY FONTOBJ_pifi(
     _In_ FONTOBJ *pfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 FD_GLYPHSET * APIENTRY FONTOBJ_pfdg(
     FONTOBJ *pfo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 PVOID APIENTRY FONTOBJ_pvTrueTypeFontFile(
     FONTOBJ *pfo,
     ULONG   *pcjFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 PBYTE APIENTRY FONTOBJ_pjOpenTypeTablePointer (
     FONTOBJ *pfo,
     ULONG    ulTag,
     ULONG   *pcjTable
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LPWSTR APIENTRY FONTOBJ_pwszFontFilePaths (
     FONTOBJ *pfo,
     ULONG   *pcwc
@@ -1988,6 +2005,7 @@ LPWSTR APIENTRY FONTOBJ_pwszFontFilePaths (
 
 #define FO_ATTR_MODE_ROTATE         1
 
+_Win32_metadata_supported_os_(windows5.0)
 PFD_GLYPHATTR APIENTRY FONTOBJ_pQueryGlyphAttrs(
     FONTOBJ       *pfo,
     __in_data_source(USER_MODE) ULONG          iMode
@@ -2003,6 +2021,7 @@ PFD_GLYPHATTR APIENTRY FONTOBJ_pQueryGlyphAttrs(
 #define PAL_BGR           0x00000008
 #define PAL_CMYK          0x00000010
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG APIENTRY PALOBJ_cGetColors(
     PALOBJ *ppalo,
     ULONG   iStart,
@@ -2055,17 +2074,20 @@ typedef struct  _CLIPLINE
     RUN     arun[1];
 } CLIPLINE, *PCLIPLINE;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY PATHOBJ_vEnumStart(
     PATHOBJ *ppo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bEnum(
     PATHOBJ  *ppo,
     PATHDATA *ppd
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY PATHOBJ_vEnumStartClipLines(
     PATHOBJ   *ppo,
@@ -2074,6 +2096,7 @@ VOID APIENTRY PATHOBJ_vEnumStartClipLines(
     LINEATTRS *pla
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bEnumClipLines(
     PATHOBJ  *ppo,
@@ -2081,12 +2104,14 @@ BOOL APIENTRY PATHOBJ_bEnumClipLines(
     CLIPLINE *pcl
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bMoveTo(
     _In_ PATHOBJ    *ppo,
     _In_ POINTFIX    ptfx
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bPolyLineTo(
     _In_ PATHOBJ      *ppo,
@@ -2094,6 +2119,7 @@ BOOL APIENTRY PATHOBJ_bPolyLineTo(
     _In_ ULONG         cptfx
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bPolyBezierTo(
     _In_ PATHOBJ   *ppo,
@@ -2101,11 +2127,13 @@ BOOL APIENTRY PATHOBJ_bPolyBezierTo(
     _In_ ULONG      cptfx
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY PATHOBJ_bCloseFigure(
     _In_ PATHOBJ *ppo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY PATHOBJ_vGetBounds(
     PATHOBJ *ppo,
@@ -2116,11 +2144,13 @@ VOID APIENTRY PATHOBJ_vGetBounds(
  * STROBJ callbacks
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY STROBJ_vEnumStart(
     STROBJ *pstro
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY STROBJ_bEnum(
     STROBJ    *pstro,
@@ -2128,6 +2158,7 @@ BOOL APIENTRY STROBJ_bEnum(
     PGLYPHPOS *ppgpos
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY STROBJ_bEnumPositionsOnly(
     STROBJ    *pstro,
@@ -2135,21 +2166,25 @@ BOOL APIENTRY STROBJ_bEnumPositionsOnly(
     PGLYPHPOS *ppgpos
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 DWORD APIENTRY STROBJ_dwGetCodePage(
     STROBJ  *pstro
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 FIX APIENTRY STROBJ_fxCharacterExtra(
     STROBJ  *pstro
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 FIX APIENTRY STROBJ_fxBreakExtra(
     STROBJ  *pstro
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY STROBJ_bGetAdvanceWidths(
     STROBJ   *pso,
@@ -2227,12 +2262,14 @@ BOOL APIENTRY STROBJ_bGetAdvanceWidths(
 #define XF_LTOFX        2L
 #define XF_INV_FXTOL    3L
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY XFORMOBJ_iGetXform(
     _In_      XFORMOBJ *pxo,
     _Out_opt_ XFORML   *pxform
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY XFORMOBJ_bApplyXform(
     XFORMOBJ *pxo,
@@ -2265,17 +2302,20 @@ HANDLE APIENTRY XFORMOBJ_cmGetTransform(
 #define XO_SRCBITFIELDS  4
 #define XO_DESTBITFIELDS 5
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY XLATEOBJ_iXlate(
     XLATEOBJ *pxlo,
     ULONG iColor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG * APIENTRY XLATEOBJ_piVector(
     XLATEOBJ *pxlo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY XLATEOBJ_cGetPalette(
     XLATEOBJ *pxlo,
@@ -2284,6 +2324,7 @@ ULONG APIENTRY XLATEOBJ_cGetPalette(
     ULONG    *pPal
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY XLATEOBJ_hGetColorTransform(
     XLATEOBJ *pxlo
@@ -2300,9 +2341,11 @@ HANDLE APIENTRY XLATEOBJ_hGetColorTransform(
 
 #else // !USERMODE_DRIVER
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngSetLastError(ULONG);
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY EngGetLastError();
 
@@ -2335,6 +2378,7 @@ ULONG APIENTRY EngGetLastError();
 #define HOOK_FLAGS                      0x0003b5ef
 #endif
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HBITMAP APIENTRY EngCreateBitmap(
     SIZEL sizl,
@@ -2344,6 +2388,7 @@ HBITMAP APIENTRY EngCreateBitmap(
     __in_data_source(USER_MODE) PVOID pvBits
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HSURF APIENTRY EngCreateDeviceSurface(
     __in_data_source(USER_MODE) DHSURF dhsurf,
@@ -2351,6 +2396,7 @@ HSURF APIENTRY EngCreateDeviceSurface(
     ULONG iFormatCompat
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HBITMAP APIENTRY EngCreateDeviceBitmap(
     __in_data_source(USER_MODE) DHSURF dhsurf,
@@ -2367,21 +2413,25 @@ HBITMAP APIENTRY EngCreateRedirectionDeviceBitmap(
     );
 #endif // (NTDDI_VERSION >= NTDDI_WIN7)
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngDeleteSurface(
     __in_data_source(USER_MODE) HSURF hsurf
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 SURFOBJ * APIENTRY EngLockSurface(
     HSURF hsurf
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngUnlockSurface(
     __in_data_source(USER_MODE) SURFOBJ *pso
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngEraseSurface(
     SURFOBJ *pso,
@@ -2389,6 +2439,7 @@ BOOL APIENTRY EngEraseSurface(
     __in_data_source(USER_MODE) ULONG    iColor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngAssociateSurface(
     HSURF hsurf,
@@ -2403,6 +2454,7 @@ BOOL APIENTRY EngAssociateSurface(
 #define MS_REUSEDDEVICEBITMAP 0x0008
 #endif // (NTDDI_VERSION >= NTDDI_WIN10_RS2)
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngModifySurface(
     HSURF   hsurf,
@@ -2415,11 +2467,13 @@ BOOL APIENTRY EngModifySurface(
     VOID*   pvReserved
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngMarkBandingSurface(
     __in_data_source(USER_MODE) HSURF hsurf
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngCheckAbort(
     SURFOBJ *pso
@@ -2429,9 +2483,11 @@ BOOL APIENTRY EngCheckAbort(
  * Engine callbacks - Paths
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PATHOBJ * APIENTRY EngCreatePath();
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDeletePath(
     PATHOBJ *ppo
@@ -2441,6 +2497,7 @@ VOID APIENTRY EngDeletePath(
  * Engine callbacks - Palettes
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HPALETTE APIENTRY EngCreatePalette(
     __in_data_source(USER_MODE) ULONG  iMode,
@@ -2451,6 +2508,7 @@ HPALETTE APIENTRY EngCreatePalette(
     __in_data_source(USER_MODE) FLONG  flBlue
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY EngQueryPalette(
     HPALETTE    hpal,
@@ -2458,6 +2516,7 @@ ULONG APIENTRY EngQueryPalette(
     ULONG       cColors,
     ULONG      *pulColors);
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngDeletePalette(
     __in_data_source(USER_MODE) HPALETTE hpal
@@ -2467,9 +2526,11 @@ BOOL APIENTRY EngDeletePalette(
  * Engine callbacks - Clipping
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 CLIPOBJ * APIENTRY EngCreateClip();
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDeleteClip(
     _In_opt_ CLIPOBJ *pco
@@ -3391,6 +3452,7 @@ typedef VOID (CALLBACK * WNDOBJCHANGEPROC)(WNDOBJ *pwo, FLONG fl);
 #define WO_RGN_SPRITE           0x0200
 #endif // (NTDDI_VERSION >= NTDDI_VISTA)
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 WNDOBJ * APIENTRY EngCreateWnd(
     SURFOBJ         *pso,
@@ -3400,11 +3462,13 @@ WNDOBJ * APIENTRY EngCreateWnd(
     int              iPixelFormat
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDeleteWnd(
     WNDOBJ  *pwo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY WNDOBJ_cEnumStart(
     WNDOBJ  *pwo,
@@ -3413,6 +3477,7 @@ ULONG APIENTRY WNDOBJ_cEnumStart(
     ULONG    cLimit
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY WNDOBJ_bEnum(
     WNDOBJ  *pwo,
@@ -3420,6 +3485,7 @@ BOOL APIENTRY WNDOBJ_bEnum(
     ULONG   *pul
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY WNDOBJ_vSetConsumer(
     WNDOBJ  *pwo,
@@ -3450,6 +3516,7 @@ VOID APIENTRY EngRenderHint(
 #define EHN_RESTORED        0
 #define EHN_ERROR           1
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 ULONG APIENTRY EngHangNotification(
     HDEV hdev,
@@ -3461,6 +3528,7 @@ ULONG APIENTRY EngHangNotification(
  * Engine callbacks - tracking driver managed resources
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HDRVOBJ APIENTRY EngCreateDriverObj(
     PVOID pvObj,
@@ -3468,6 +3536,7 @@ HDRVOBJ APIENTRY EngCreateDriverObj(
     HDEV hdev
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngDeleteDriverObj(
     HDRVOBJ hdo,
@@ -3475,11 +3544,13 @@ BOOL APIENTRY EngDeleteDriverObj(
     BOOL bLocked
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 DRIVEROBJ* APIENTRY EngLockDriverObj(
     HDRVOBJ hdo
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngUnlockDriverObj(
     HDRVOBJ hdo
@@ -3489,6 +3560,7 @@ BOOL APIENTRY EngUnlockDriverObj(
  * Engine callback - return current process handle.
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY EngGetProcessHandle();
 
@@ -3496,6 +3568,7 @@ HANDLE APIENTRY EngGetProcessHandle();
  * Engine callback - return current thread id
  */
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 HANDLE APIENTRY EngGetCurrentThreadId();
 
@@ -3503,6 +3576,7 @@ HANDLE APIENTRY EngGetCurrentThreadId();
  * Engine callback - return current process id
  */
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 HANDLE APIENTRY EngGetCurrentProcessId();
 
@@ -3536,6 +3610,7 @@ BOOL APIENTRY DrvSwapBuffers(
  * Function prototypes - Engine Simulations
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngBitBlt(
     _In_ SURFOBJ  *psoTrg,
@@ -3551,6 +3626,7 @@ BOOL APIENTRY EngBitBlt(
     __in_data_source(USER_MODE) ROP4      rop4
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngLineTo(
     SURFOBJ   *pso,
@@ -3564,6 +3640,7 @@ BOOL APIENTRY EngLineTo(
     __in_data_source(USER_MODE) MIX        mix
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngStretchBlt(
     SURFOBJ         *psoDest,
@@ -3579,6 +3656,7 @@ BOOL APIENTRY EngStretchBlt(
     __in_data_source(USER_MODE) ULONG            iMode
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngStretchBltROP(
     SURFOBJ         *psoDest,
@@ -3596,6 +3674,7 @@ BOOL APIENTRY EngStretchBltROP(
     __in_data_source(USER_MODE) DWORD            rop4
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngAlphaBlend(
     SURFOBJ       *psoDest,
@@ -3607,6 +3686,7 @@ BOOL APIENTRY EngAlphaBlend(
     BLENDOBJ      *pBlendObj
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngGradientFill(
     SURFOBJ         *psoDest,
@@ -3621,6 +3701,7 @@ BOOL APIENTRY EngGradientFill(
     __in_data_source(USER_MODE) ULONG            ulMode
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngTransparentBlt(
     _In_ SURFOBJ    *psoDst,
@@ -3633,6 +3714,7 @@ BOOL APIENTRY EngTransparentBlt(
     __in_data_source(USER_MODE) ULONG      bCalledFromBitBlt
     ); 
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngTextOut(
     SURFOBJ  *pso,
@@ -3647,6 +3729,7 @@ BOOL APIENTRY EngTextOut(
     __in_data_source(USER_MODE) MIX       mix
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngStrokePath(
     SURFOBJ   *pso,
@@ -3659,6 +3742,7 @@ BOOL APIENTRY EngStrokePath(
     __in_data_source(USER_MODE) MIX        mix
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngFillPath(
     SURFOBJ  *pso,
@@ -3670,6 +3754,7 @@ BOOL APIENTRY EngFillPath(
     __in_data_source(USER_MODE) FLONG     flOptions
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngStrokeAndFillPath(
     SURFOBJ   *pso,
@@ -3684,6 +3769,7 @@ BOOL APIENTRY EngStrokeAndFillPath(
     __in_data_source(USER_MODE) FLONG      flOptions
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngPaint(
     SURFOBJ  *pso,
@@ -3693,6 +3779,7 @@ BOOL APIENTRY EngPaint(
     __in_data_source(USER_MODE) MIX       mix
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngCopyBits(
     SURFOBJ  *psoDest,
@@ -3703,6 +3790,7 @@ BOOL APIENTRY EngCopyBits(
     __in_data_source(USER_MODE) POINTL   *pptlSrc
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY EngSetPointerShape(
     SURFOBJ  *pso,
@@ -3717,6 +3805,7 @@ ULONG APIENTRY EngSetPointerShape(
     FLONG     fl
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngMovePointer(
     SURFOBJ  *pso,
@@ -3725,6 +3814,7 @@ VOID APIENTRY EngMovePointer(
     RECTL    *prcl
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngPlgBlt(
     SURFOBJ         *psoTrg,
@@ -3740,6 +3830,7 @@ BOOL APIENTRY EngPlgBlt(
     __in_data_source(USER_MODE) ULONG            iMode
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 ULONG APIENTRY EngDitherColor(
     HDEV   hdev,
@@ -3775,6 +3866,7 @@ ULONG APIENTRY EngCTGetCurrentGamma(
 #define ECS_TEARDOWN    0x0001
 #define ECS_REDRAW      0x0002
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngControlSprites(
     WNDOBJ  *pwo,
@@ -3786,6 +3878,7 @@ BOOL APIENTRY EngControlSprites(
 // Halftone releated APIs
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 LONG APIENTRY HT_ComputeRGBGammaTable(
     _In_    USHORT  GammaTableEntries,
     _In_    USHORT  GammaTableType,
@@ -3795,6 +3888,7 @@ LONG APIENTRY HT_ComputeRGBGammaTable(
     _Out_writes_bytes_(GammaTableEntries * 3)   LPBYTE  pGammaTable
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LONG APIENTRY HT_Get8BPPFormatPalette(
     _Out_opt_   LPPALETTEENTRY  pPaletteEntry,
     _In_        USHORT          RedGamma,
@@ -3802,6 +3896,7 @@ LONG APIENTRY HT_Get8BPPFormatPalette(
     _In_        USHORT          BlueGamma
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 LONG APIENTRY HT_Get8BPPMaskPalette(
     _Out_opt_   LPPALETTEENTRY  pPaletteEntry,
     _In_        BOOL            Use8BPPMaskPal,
@@ -3829,6 +3924,7 @@ typedef struct _DEVHTADJDATA {
     PDEVHTINFO  pAdjHTInfo;
     } DEVHTADJDATA, *PDEVHTADJDATA;
 
+_Win32_metadata_supported_os_(windows5.0)
 LONG
 APIENTRY
 HTUI_DeviceColorAdjustment(
@@ -3841,6 +3937,7 @@ HTUI_DeviceColorAdjustment(
 // General support APIS
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 __analysis_noreturn
 ENGAPI
 VOID APIENTRY EngDebugBreak(
@@ -3858,6 +3955,7 @@ VOID APIENTRY EngBugCheckEx(
     );
 #endif // (NTDDI_VERSION >= NTDDI_WINXPSP2)
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDebugPrint(
     _In_z_              PCHAR StandardPrefix,
@@ -3865,16 +3963,19 @@ VOID APIENTRY EngDebugPrint(
     _In_                va_list ap
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngQueryPerformanceCounter(
     _Out_               LONGLONG  *pPerformanceCount
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngQueryPerformanceFrequency(
     _Out_               LONGLONG  *pFrequency
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 __drv_preferredFunction("(see documentation)", "Obsolete, always returns false. ") 
 ENGAPI
 BOOL APIENTRY EngSetPointerTag(
@@ -3908,6 +4009,7 @@ BOOL APIENTRY EngSetPointerTag(
 
 #else // !USERMODE_DRIVER
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 _Post_writable_byte_size_(MemSize)
 PVOID APIENTRY EngAllocMem(
@@ -3916,11 +4018,13 @@ PVOID APIENTRY EngAllocMem(
     _In_                ULONG Tag
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngFreeMem(
     _In_                PVOID Mem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 _Post_writable_byte_size_(cj)
 PVOID APIENTRY EngAllocUserMem(
@@ -3928,11 +4032,13 @@ PVOID APIENTRY EngAllocUserMem(
     _In_                ULONG tag
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngFreeUserMem(
     _In_                PVOID pv
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 _Post_writable_byte_size_(cj)
 PVOID APIENTRY EngAllocPrivateUserMem(
@@ -3941,6 +4047,7 @@ PVOID APIENTRY EngAllocPrivateUserMem(
     _In_                ULONG tag
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID
 EngFreePrivateUserMem(
@@ -3958,6 +4065,7 @@ EngDxIoctl(
     );
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 int APIENTRY EngMulDiv(
     _In_                int a,
@@ -3972,6 +4080,7 @@ int APIENTRY EngMulDiv(
 // User mode memory Operations
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngProbeForRead(
     _In_reads_bytes_(Length)     PVOID Address,
@@ -3983,6 +4092,7 @@ VOID APIENTRY EngProbeForRead(
     _In_                    ULONG Alignment
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngProbeForReadAndWrite(
     _Inout_updates_bytes_(Length)  PVOID Address,
@@ -3994,17 +4104,20 @@ VOID APIENTRY EngProbeForReadAndWrite(
     _In_                    ULONG Alignment
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY EngSecureMem(
     _In_reads_bytes_(Length)     PVOID Address,
     _In_                    ULONG Length
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngUnsecureMem(
     _In_                    HANDLE hSecure
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 DWORD APIENTRY EngDeviceIoControl(
     _In_                                HANDLE hDevice,
@@ -4028,22 +4141,26 @@ DWORD APIENTRY EngDeviceIoControl(
 
 #else // !USERMODE_DRIVER
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY EngLoadImage(
     _In_ LPWSTR pwszDriver
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PVOID APIENTRY EngFindImageProcAddress(
     HANDLE hModule,
     _In_ LPSTR lpProcName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngUnloadImage(
     _In_ HANDLE hModule
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PVOID APIENTRY EngMapModule(
     _In_ HANDLE h,
@@ -4057,11 +4174,13 @@ PVOID APIENTRY EngMapModule(
 // callback for extra PDEV information
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 LPWSTR APIENTRY EngGetPrinterDataFileName(
     _In_                    HDEV hdev
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 LPWSTR APIENTRY EngGetDriverName(
     _In_                    HDEV hdev
@@ -4074,6 +4193,7 @@ typedef struct _TYPE1_FONT
     ULONG   ulIdentifier;
 } TYPE1_FONT;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngGetType1FontList(
     _In_                            HDEV            hdev,
@@ -4088,11 +4208,13 @@ BOOL APIENTRY EngGetType1FontList(
 // Manipulating resource sections
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY EngLoadModule(
     _In_                    LPWSTR pwsz
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PVOID APIENTRY EngFindResource(
     _In_                    HANDLE h,
@@ -4101,11 +4223,13 @@ PVOID APIENTRY EngFindResource(
     _Out_                   PULONG pulSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngFreeModule(
     _In_                    HANDLE h
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL
 APIENTRY
@@ -4113,6 +4237,7 @@ EngDeleteFile (
     _In_                    LPWSTR  pwszFileName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 _Success_(return != 0)
 _When_(cjSize != 0, _At_(return, _Out_writes_bytes_(cjSize)))
@@ -4124,6 +4249,7 @@ EngMapFile(
     _Out_    ULONG_PTR  *piFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL
 APIENTRY
@@ -4135,12 +4261,14 @@ EngUnmapFile (
 // FontFile Callbacks
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 __drv_preferredFunction("EngUnmapFontFileFD", "Obsolete") 
 ENGAPI
 VOID APIENTRY EngUnmapFontFile(
     _In_                    ULONG_PTR iFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 __drv_preferredFunction("EngMapFontFileFD", "Obsolete") 
 ENGAPI
 BOOL APIENTRY EngMapFontFile(
@@ -4149,11 +4277,13 @@ BOOL APIENTRY EngMapFontFile(
     _Out_                           ULONG  *pcjBuf
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngUnmapFontFileFD(
     _In_                    ULONG_PTR iFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngMapFontFileFD(
     _In_                            ULONG_PTR  iFile,
@@ -4161,6 +4291,7 @@ BOOL APIENTRY EngMapFontFileFD(
     _Out_                           ULONG  *pcjBuf
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngLpkInstalled();
 
@@ -4170,21 +4301,25 @@ BOOL APIENTRY EngLpkInstalled();
 
 DECLARE_HANDLE(HSEMAPHORE);
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HSEMAPHORE APIENTRY EngCreateSemaphore(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngAcquireSemaphore(
     _In_                    HSEMAPHORE hsem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngReleaseSemaphore(
     _In_                    HSEMAPHORE hsem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDeleteSemaphore(
     _In_                    HSEMAPHORE hsem
@@ -4209,11 +4344,13 @@ BOOL APIENTRY EngAcquireSemaphoreSharedNoWait(
 #endif
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngIsSemaphoreOwned(
     _In_                    HSEMAPHORE hsem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngIsSemaphoreOwnedByCurrentThread(
     _In_                    HSEMAPHORE hsem
@@ -4260,16 +4397,19 @@ typedef struct _ENGSAFESEMAPHORE
     LONG lCount;
 } ENGSAFESEMAPHORE;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngInitializeSafeSemaphore(
     _Out_                   ENGSAFESEMAPHORE *pssem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngDeleteSafeSemaphore(
     _Inout_                 ENGSAFESEMAPHORE *pssem
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngMultiByteToUnicodeN(
     _Out_writes_bytes_to_(MaxBytesInUnicodeString, *BytesInUnicodeString)   LPWSTR UnicodeString,
@@ -4279,6 +4419,7 @@ VOID APIENTRY EngMultiByteToUnicodeN(
     _In_                                                                ULONG BytesInMultiByteString
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngUnicodeToMultiByteN(
     _Out_writes_bytes_to_(MaxBytesInMultiByteString, *BytesInMultiByteString)   PCHAR MultiByteString,
@@ -4302,6 +4443,7 @@ VOID APIENTRY EngUnicodeToMultiByteN(
 
 #else // !USERMODE_DRIVER
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngGetPrinterDriver(
     _In_                    HANDLE  hPrinter,
@@ -4312,6 +4454,7 @@ BOOL APIENTRY EngGetPrinterDriver(
     _Out_                   DWORD  *pcbNeeded
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 DWORD APIENTRY EngGetPrinterData(
     _In_                    HANDLE   hPrinter,
@@ -4322,6 +4465,7 @@ DWORD APIENTRY EngGetPrinterData(
     _Out_                   LPDWORD  pcbNeeded
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 DWORD APIENTRY EngSetPrinterData(
     _In_                        HANDLE   hPrinter,
@@ -4331,6 +4475,7 @@ DWORD APIENTRY EngSetPrinterData(
     _In_                        DWORD    cjPrinterData
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngGetForm(
     _In_                    HANDLE  hPrinter,
@@ -4341,6 +4486,7 @@ BOOL APIENTRY EngGetForm(
     _Out_                   LPDWORD pcbNeeded
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngWritePrinter(
     _In_                    HANDLE  hPrinter,
@@ -4349,6 +4495,7 @@ BOOL APIENTRY EngWritePrinter(
     _Out_                   LPDWORD pcWritten
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngGetPrinter(
     _In_                    HANDLE  hPrinter,
@@ -4358,6 +4505,7 @@ BOOL APIENTRY EngGetPrinter(
     _Out_                   LPDWORD pcbNeeded
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngEnumForms(
     _In_                    HANDLE  hPrinter,
@@ -4379,36 +4527,59 @@ BOOL APIENTRY EngEnumForms(
         ULONG ul2;
     } FLOATOBJ, *PFLOATOBJ;
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_SetFloat(PFLOATOBJ,FLOATL);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_SetLong(PFLOATOBJ,LONG);
 
+_Win32_metadata_supported_os_(windows5.0)
     LONG APIENTRY FLOATOBJ_GetFloat(PFLOATOBJ);
+_Win32_metadata_supported_os_(windows5.0)
     LONG APIENTRY FLOATOBJ_GetLong(PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_AddFloat(PFLOATOBJ,FLOATL);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_AddLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_Add(PFLOATOBJ,PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_SubFloat(PFLOATOBJ,FLOATL);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_SubLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_Sub(PFLOATOBJ,PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_MulFloat(PFLOATOBJ,FLOATL);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_MulLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_Mul(PFLOATOBJ,PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_DivFloat(PFLOATOBJ,FLOATL);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_DivLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_Div(PFLOATOBJ,PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     VOID APIENTRY FLOATOBJ_Neg(PFLOATOBJ);
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_EqualLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_GreaterThanLong(PFLOATOBJ,LONG);
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_LessThanLong(PFLOATOBJ,LONG);
 
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_Equal(PFLOATOBJ,PFLOATOBJ);
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_GreaterThan(PFLOATOBJ,PFLOATOBJ);
+_Win32_metadata_supported_os_(windows5.0)
     BOOL APIENTRY FLOATOBJ_LessThan(PFLOATOBJ,PFLOATOBJ);
 
 #else
@@ -4472,6 +4643,7 @@ typedef struct  tagFLOATOBJ_XFORM
     FLOATOBJ eDy;
 } FLOATOBJ_XFORM, *PFLOATOBJ_XFORM, FAR *LPFLOATOBJ_XFORM;
 
+_Win32_metadata_supported_os_(windows5.0)
 ULONG APIENTRY XFORMOBJ_iGetFloatObjXform(
     XFORMOBJ *pxo,
     FLOATOBJ_XFORM * pfxo
@@ -4484,6 +4656,7 @@ ULONG APIENTRY XFORMOBJ_iGetFloatObjXform(
 
 typedef int (__cdecl *SORTCOMP)(const void *pv1, const void *pv2);
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngSort(
     _Inout_updates_bytes_(c * cjElem)  PBYTE pjBuf,
@@ -4503,11 +4676,13 @@ typedef struct _ENG_TIME_FIELDS {
     USHORT usWeekday;     // range [0..6] == [Sunday..Saturday]
 } ENG_TIME_FIELDS, *PENG_TIME_FIELDS;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngQueryLocalTime(
     _Out_                   PENG_TIME_FIELDS
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return != 0)
 ENGAPI
 FD_GLYPHSET* APIENTRY EngComputeGlyphSet(
@@ -4516,6 +4691,7 @@ FD_GLYPHSET* APIENTRY EngComputeGlyphSet(
     _In_                    INT cChars
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 INT APIENTRY EngMultiByteToWideChar(
     _In_                                    UINT CodePage,
@@ -4525,6 +4701,7 @@ INT APIENTRY EngMultiByteToWideChar(
     _In_                                    INT BytesInMultiByteString
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 INT APIENTRY EngWideCharToMultiByte(
     _In_                                    UINT CodePage,
@@ -4534,23 +4711,27 @@ INT APIENTRY EngWideCharToMultiByte(
     _In_                                    INT BytesInMultiByteString
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 VOID APIENTRY EngGetCurrentCodePage(
     _Out_   PUSHORT OemCodePage,
     _Out_   PUSHORT AnsiCodePage
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 HANDLE APIENTRY EngLoadModuleForWrite(
     _In_    LPWSTR pwsz,
     _In_    ULONG  cjSizeOfModule
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 LARGE_INTEGER APIENTRY EngQueryFileTimeStamp (
     _In_    LPWSTR  pwsz
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return)
 ENGAPI
 BOOL APIENTRY EngGetFileChangeTime(
@@ -4558,6 +4739,7 @@ BOOL APIENTRY EngGetFileChangeTime(
     _Out_   LARGE_INTEGER   *pChangeTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Success_(return)
 ENGAPI
 BOOL APIENTRY EngGetFilePath(
@@ -4578,6 +4760,7 @@ _On_failure_(_Post_satisfies_(return == 0))
 __forceinline
 ULONG
 #pragma warning(suppress: 6001 6101 28104) // Suppress PREfast warnings for not acquiring EngFloatState and filling out pBuffer
+_Win32_metadata_supported_os_(windows5.0)
 EngSaveFloatingPointState (
     _At_(*pBuffer, _Kernel_requires_resource_not_held_(EngFloatState)) 
 
@@ -4597,6 +4780,7 @@ _At_(*pBuffer, _Kernel_requires_resource_held_(EngFloatState)
 __forceinline
 BOOL
 #pragma warning(suppress: 28103) // Suppress PREfast warning for not releasing EngFloatState
+_Win32_metadata_supported_os_(windows5.0)
 EngRestoreFloatingPointState (
     _In_reads_(_Inexpressible_(statesize)) 
     VOID *pBuffer 
@@ -4608,6 +4792,7 @@ EngRestoreFloatingPointState (
 
 #else
 
+_Win32_metadata_supported_os_(windows5.0)
 _Check_return_ 
 _Success_(((pBuffer != NULL && cjBufferSize != 0) && return == 1) ||
           ((pBuffer == NULL || cjBufferSize == 0) && return > 0))
@@ -4623,6 +4808,7 @@ ULONG APIENTRY EngSaveFloatingPointState(
     _Inout_     ULONG   cjBufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 _Check_return_ 
 _Success_(return) 
 _Kernel_float_restored_
@@ -4640,11 +4826,13 @@ BOOL APIENTRY EngRestoreFloatingPointState(
 // DirectDraw surface locking
 //
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 PDD_SURFACE_LOCAL APIENTRY EngLockDirectDrawSurface(
         _In_    HANDLE hSurface
         );
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY EngUnlockDirectDrawSurface(
         _In_    PDD_SURFACE_LOCAL pSurface
@@ -4660,21 +4848,25 @@ BOOL APIENTRY EngUnlockDirectDrawSurface(
 
 typedef struct _ENG_EVENT *PEVENT;
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 BOOL APIENTRY EngDeleteEvent(
     _In_    PEVENT  pEvent
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 BOOL APIENTRY EngCreateEvent(
     _Out_   PEVENT *ppEvent
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 BOOL APIENTRY EngUnmapEvent(
     _In_    PEVENT pEvent
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 PEVENT APIENTRY EngMapEvent(
     _In_        HDEV            hDev,
@@ -4684,24 +4876,28 @@ PEVENT APIENTRY EngMapEvent(
     _Reserved_  PVOID           Reserved3
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 BOOL APIENTRY EngWaitForSingleObject(
     _In_    PEVENT          pEvent,
     _In_    PLARGE_INTEGER  pTimeOut
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 LONG APIENTRY EngSetEvent(
     _In_    PEVENT pEvent
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 VOID APIENTRY
 EngClearEvent (
     _In_    PEVENT pEvent
 );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 LONG APIENTRY
 EngReadStateEvent (
@@ -4732,6 +4928,7 @@ typedef enum _ENG_SYSTEM_ATTRIBUTE {
 #endif // (NTDDI_VERSION >= NTDDI_LONGHORN)
 
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY
 EngQuerySystemAttribute(
@@ -4741,6 +4938,7 @@ EngQuerySystemAttribute(
 #define ENG_FNT_CACHE_READ_FAULT    0x1
 #define ENG_FNT_CACHE_WRITE_FAULT   0x2
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 _Success_(return != nullptr)
 ENGAPI
 _Post_writable_byte_size_(*pulSize)
@@ -4755,12 +4953,14 @@ BOOL WINAPI EngFntCacheFlush(
     _In_  BOOL  DiscardContent);
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 _Post_writable_byte_size_(ulSize)
 PVOID APIENTRY  EngFntCacheAlloc(
     _In_    ULONG FastCheckSum, 
     _In_    ULONG ulSize);
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 ENGAPI
 VOID  APIENTRY  EngFntCacheFault(
     _In_    ULONG ulFastCheckSum, 
@@ -4771,6 +4971,7 @@ typedef enum _ENG_DEVICE_ATTRIBUTE {
     QDA_ACCELERATION_LEVEL = 1
 } ENG_DEVICE_ATTRIBUTE;
 
+_Win32_metadata_supported_os_(windows5.0)
 ENGAPI
 BOOL APIENTRY
 EngQueryDeviceAttribute(
