diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
index 57c59bce..782f2e16 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
@@ -17,6 +17,9 @@ Abstract:
 #endif // _MSC_VER
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -27,6 +30,7 @@ extern "C" {
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
+_Win32_metadata_supported_os_(windows8.1)
 BOOL
 WINAPI
 CeipIsOptedIn();
