diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winefs.h b/generation/WinSDK/RecompiledIdlHeaders/um/winefs.h
index 30e1a95a..3710dcb1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winefs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winefs.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 //+---------------------------------------------------------------------------
 //
 //  Microsoft Windows
@@ -318,6 +319,7 @@ typedef struct _ENCRYPTION_PROTECTOR_LIST {
 //----------------------------------------------------------------------------------/
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -327,6 +329,7 @@ QueryUsersOnEncryptedFile(
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -336,6 +339,7 @@ QueryRecoveryAgentsOnEncryptedFile(
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -344,6 +348,7 @@ RemoveUsersFromEncryptedFile(
     _In_    PENCRYPTION_CERTIFICATE_HASH_LIST   pHashes
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -358,6 +363,7 @@ AddUsersToEncryptedFile(
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -367,6 +373,7 @@ SetUserFileEncryptionKey(
 
 #else
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -391,6 +398,7 @@ SetUserFileEncryptionKeyEx(
 
 #endif // #if (NTDDI_VERSION >= NTDDI_VISTA) 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 VOID
 WINAPI
@@ -398,6 +406,8 @@ FreeEncryptionCertificateHashList(
     _In_    PENCRYPTION_CERTIFICATE_HASH_LIST       pUsers
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 WINAPI
@@ -412,6 +422,7 @@ EncryptionDisable(
 // in WS03 to use a CONST SECURITY_ATTRIBUTES structure
 #if (NTDDI_VERSION >= NTDDI_WS03)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
@@ -427,6 +438,7 @@ DuplicateEncryptionInfoFile(
 #elif (NTDDI_VERSION >= NTDDI_WINXP)
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 DWORD
 WINAPI
