diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WinFax.h b/generation/WinSDK/RecompiledIdlHeaders/um/WinFax.h
index f0c668af..d641e19a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WinFax.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WinFax.h
@@ -19,6 +19,7 @@ Abstract:
 #define _FAXAPI_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -825,19 +826,25 @@ typedef PFAX_CONTEXT_INFOA PFAX_CONTEXT_INFO;
 // prototypes
 //
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
 FaxConnectFaxServerA(
     IN  LPCSTR MachineName OPTIONAL,
     OUT LPHANDLE FaxHandle
+        _Win32_metadata_raii_free_(FaxClose)
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
 FaxConnectFaxServerW(
     IN  LPCWSTR MachineName OPTIONAL,
     OUT LPHANDLE FaxHandle
+        _Win32_metadata_raii_free_(FaxClose)
     );
 #ifdef UNICODE
 #define FaxConnectFaxServer  FaxConnectFaxServerW
@@ -887,6 +894,7 @@ FaxOpenPort(
     IN  DWORD DeviceId,
     IN  DWORD Flags,
     OUT LPHANDLE FaxPortHandle
+        _Win32_metadata_raii_free_(FaxClose)
     );
 
 typedef BOOL
@@ -897,6 +905,7 @@ typedef BOOL
     OUT LPHANDLE FaxPortHandle
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINFAXAPI
 BOOL
 WINAPI
@@ -904,6 +913,7 @@ FaxCompleteJobParamsA(
     IN OUT PFAX_JOB_PARAMA *JobParams,
     IN OUT PFAX_COVERPAGE_INFOA *CoverpageInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINFAXAPI
 BOOL
 WINAPI
@@ -935,6 +945,8 @@ typedef BOOL
 
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -945,6 +957,8 @@ FaxSendDocumentA(
     IN const FAX_COVERPAGE_INFOA *CoverpageInfo, OPTIONAL
     OUT LPDWORD FaxJobId OPTIONAL
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1005,6 +1019,8 @@ typedef BOOL
 #define PFAX_RECIPIENT_CALLBACK  PFAX_RECIPIENT_CALLBACKA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1015,6 +1031,8 @@ FaxSendDocumentForBroadcastA(
     IN PFAX_RECIPIENT_CALLBACKA FaxRecipientCallback,
     IN LPVOID Context
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1053,6 +1071,8 @@ typedef BOOL
 #define PFAXSENDDOCUMENTFORBROADCAST  PFAXSENDDOCUMENTFORBROADCASTA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1061,6 +1081,8 @@ FaxEnumJobsA(
     OUT PFAX_JOB_ENTRYA *JobEntry,
     OUT LPDWORD JobsReturned
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1094,6 +1116,8 @@ typedef BOOL
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1102,6 +1126,8 @@ FaxGetJobA(
    IN  DWORD JobId,
    OUT PFAX_JOB_ENTRYA *JobEntry
    );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1134,6 +1160,8 @@ typedef BOOL
 #define PFAXGETJOB  PFAXGETJOBA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1143,6 +1171,8 @@ FaxSetJobA(
    IN DWORD Command,
    IN const FAX_JOB_ENTRYA *JobEntry
    );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1200,6 +1230,8 @@ typedef BOOL
    OUT LPDWORD ImageHeight
    );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1207,6 +1239,8 @@ FaxGetDeviceStatusA(
     IN  HANDLE FaxPortHandle,
     OUT PFAX_DEVICE_STATUSA *DeviceStatus
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1251,6 +1285,8 @@ typedef BOOL
     IN DWORD JobId
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1258,6 +1294,8 @@ FaxGetConfigurationA(
     IN  HANDLE FaxHandle,
     OUT PFAX_CONFIGURATIONA *FaxConfig
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1287,6 +1325,8 @@ typedef BOOL
 #define PFAXGETCONFIGURATION  PFAXGETCONFIGURATIONA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1294,6 +1334,8 @@ FaxSetConfigurationA(
     IN  HANDLE FaxHandle,
     IN  const FAX_CONFIGURATIONA *FaxConfig
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1323,6 +1365,8 @@ typedef BOOL
 #define PFAXSETCONFIGURATION  PFAXSETCONFIGURATIONA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1331,6 +1375,8 @@ FaxGetLoggingCategoriesA(
     OUT PFAX_LOG_CATEGORYA *Categories,
     OUT LPDWORD NumberCategories
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1363,6 +1409,8 @@ typedef BOOL
 #define PFAXGETLOGGINGCATEGORIES  PFAXGETLOGGINGCATEGORIESA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1371,6 +1419,8 @@ FaxSetLoggingCategoriesA(
     IN  const FAX_LOG_CATEGORYA *Categories,
     IN  DWORD NumberCategories
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1403,6 +1453,8 @@ typedef BOOL
 #define PFAXSETLOGGINGCATEGORIES  PFAXSETLOGGINGCATEGORIESA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1411,6 +1463,8 @@ FaxEnumPortsA(
     OUT PFAX_PORT_INFOA *PortInfo,
     OUT LPDWORD PortsReturned
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1443,6 +1497,8 @@ typedef BOOL
 #define PFAXENUMPORTS  PFAXENUMPORTSA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1450,6 +1506,8 @@ FaxGetPortA(
     IN  HANDLE FaxPortHandle,
     OUT PFAX_PORT_INFOA *PortInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1479,6 +1537,8 @@ typedef BOOL
 #define PFAXGETPORT  PFAXGETPORTA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1486,6 +1546,8 @@ FaxSetPortA(
     IN  HANDLE FaxPortHandle,
     IN  const FAX_PORT_INFOA *PortInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1515,6 +1577,8 @@ typedef BOOL
 #define PFAXSETPORT  PFAXSETPORTA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1523,6 +1587,8 @@ FaxEnumRoutingMethodsA(
     OUT PFAX_ROUTING_METHODA *RoutingMethod,
     OUT LPDWORD MethodsReturned
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1555,6 +1621,8 @@ typedef BOOL
 #define PFAXENUMROUTINGMETHODS  PFAXENUMROUTINGMETHODSA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1563,6 +1631,8 @@ FaxEnableRoutingMethodA(
     IN  LPCSTR RoutingGuid,
     IN  BOOL Enabled
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1596,6 +1666,8 @@ typedef BOOL
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1604,6 +1676,8 @@ FaxEnumGlobalRoutingInfoA(
     OUT PFAX_GLOBAL_ROUTING_INFOA *RoutingInfo,
     OUT LPDWORD MethodsReturned
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1636,6 +1710,8 @@ typedef BOOL
 #define PFAXENUMGLOBALROUTINGINFO  PFAXENUMGLOBALROUTINGINFOA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1643,6 +1719,8 @@ FaxSetGlobalRoutingInfoA(
     IN  HANDLE FaxHandle,
     IN  const FAX_GLOBAL_ROUTING_INFOA *RoutingInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1672,6 +1750,8 @@ typedef BOOL
 #define PFAXSETGLOBALROUTINGINFO  PFAXSETGLOBALROUTINGINFOA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1681,6 +1761,8 @@ FaxGetRoutingInfoA(
     OUT LPBYTE *RoutingInfoBuffer,
     OUT LPDWORD RoutingInfoBufferSize
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1717,6 +1799,8 @@ typedef BOOL
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1726,6 +1810,8 @@ FaxSetRoutingInfoA(
     IN  const BYTE *RoutingInfoBuffer,
     IN  DWORD RoutingInfoBufferSize
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1794,6 +1880,8 @@ typedef VOID
     LPVOID Buffer
     );
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1803,6 +1891,8 @@ FaxStartPrintJobA(
     OUT LPDWORD FaxJobId,
     OUT PFAX_CONTEXT_INFOA FaxContextInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1838,6 +1928,8 @@ typedef BOOL
 #define PFAXSTARTPRINTJOB  PFAXSTARTPRINTJOBA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1845,6 +1937,8 @@ FaxPrintCoverPageA(
     IN const FAX_CONTEXT_INFOA *FaxContextInfo,
     IN const FAX_COVERPAGE_INFOA *CoverPageInfo
     );
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1875,6 +1969,8 @@ typedef BOOL
 #endif // !UNICODE
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
@@ -1927,6 +2023,8 @@ typedef BOOL
 #define PFAX_ROUTING_INSTALLATION_CALLBACK PFAX_ROUTING_INSTALLATION_CALLBACKW
 
 
+_Win32_metadata_supported_os_(windows5.0)
+_Win32_metadata_set_last_error_
 WINFAXAPI
 BOOL
 WINAPI
