diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winnetwk.h b/generation/WinSDK/RecompiledIdlHeaders/um/winnetwk.h
index 35e5990c..b58a8960 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winnetwk.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winnetwk.h
@@ -1,4 +1,7 @@
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 /*++
 
@@ -540,6 +543,7 @@ WNetOpenEnumA(
     _In_  DWORD          dwUsage,
     _In_opt_ LPNETRESOURCEA lpNetResource,
     _Out_ LPHANDLE       lphEnum
+        _Win32_metadata_raii_free_(WNetCloseEnum)
     );
 _Check_return_
 DWORD APIENTRY
@@ -549,6 +553,7 @@ WNetOpenEnumW(
     _In_  DWORD          dwUsage,
     _In_opt_ LPNETRESOURCEW lpNetResource,
     _Out_ LPHANDLE       lphEnum
+        _Win32_metadata_raii_free_(WNetCloseEnum)
     );
 #ifdef UNICODE
 #define WNetOpenEnum  WNetOpenEnumW
@@ -789,6 +794,7 @@ WNetGetNetworkInformationW(
 //  Error handling.
 //
 
+_Win32_metadata_set_last_error_
 _Check_return_
 DWORD APIENTRY
 WNetGetLastErrorA(
@@ -798,6 +804,7 @@ WNetGetLastErrorA(
     _Out_writes_(nNameBufSize) LPSTR  lpNameBuf,
     _In_ DWORD      nNameBufSize
     );
+_Win32_metadata_set_last_error_
 _Check_return_
 DWORD APIENTRY
 WNetGetLastErrorW(
