diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h b/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
index 00e500ba..1a90be51 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
@@ -17,6 +17,7 @@ Abstract:
 
 #ifndef _WINPPI_
 #define _WINPPI_
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -28,7 +29,8 @@ typedef int (CALLBACK* EMFPLAYPROC)( HDC, INT, HANDLE );
 HANDLE WINAPI GdiGetSpoolFileHandle(
     LPWSTR     pwszPrinterName,
     LPDEVMODEW pDevmode,
-    LPWSTR     pwszDocName);
+    LPWSTR     pwszDocName)
+    _Win32_metadata_raii_free_(GdiDeleteSpoolFileHandle);
 
 BOOL WINAPI GdiDeleteSpoolFileHandle(
     HANDLE     SpoolFileHandle);
