diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winreg.h b/generation/WinSDK/RecompiledIdlHeaders/um/winreg.h
index e48c86d7..61f88f46 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winreg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winreg.h
@@ -17,6 +17,7 @@ Abstract:
 #define _WINREG_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #include <apiset.h>
 #include <apisetcconv.h>
@@ -238,6 +239,7 @@ typedef PVALENTA PVALENT;
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -251,6 +253,7 @@ RegCloseKey(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -259,6 +262,7 @@ RegOverridePredefKey (
     _In_opt_ HKEY hNewHKey
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -269,6 +273,7 @@ RegOpenUserClassesRoot(
     _Out_ PHKEY phkResult
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -277,6 +282,7 @@ RegOpenCurrentUser(
     _Out_ PHKEY phkResult
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -284,6 +290,7 @@ RegDisablePredefinedCache(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -291,6 +298,7 @@ RegDisablePredefinedCacheEx(
     VOID
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -299,6 +307,7 @@ RegConnectRegistryA (
     _In_ HKEY hKey,
     _Out_ PHKEY phkResult
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -349,6 +358,7 @@ RegConnectRegistryExW (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -357,6 +367,7 @@ RegCreateKeyA (
     _In_opt_ LPCSTR lpSubKey,
     _Out_ PHKEY phkResult
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -371,6 +382,7 @@ RegCreateKeyW (
 #define RegCreateKey  RegCreateKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -386,6 +398,7 @@ RegCreateKeyExA(
     _Out_opt_ LPDWORD lpdwDisposition
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -413,6 +426,7 @@ RegCreateKeyExW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -429,6 +443,7 @@ RegCreateKeyTransactedA (
     _In_        HANDLE hTransaction,
     _Reserved_ PVOID  pExtendedParemeter
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -457,6 +472,7 @@ RegCreateKeyTransactedW (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -464,6 +480,7 @@ RegDeleteKeyA (
     _In_ HKEY hKey,
     _In_ LPCSTR lpSubKey
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -477,6 +494,7 @@ RegDeleteKeyW (
 #define RegDeleteKey  RegDeleteKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -487,6 +505,7 @@ RegDeleteKeyExA(
     _Reserved_ DWORD Reserved
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -509,6 +528,7 @@ RegDeleteKeyExW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -520,6 +540,7 @@ RegDeleteKeyTransactedA (
     _In_        HANDLE hTransaction,
     _Reserved_ PVOID  pExtendedParameter
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -543,6 +564,7 @@ RegDeleteKeyTransactedW (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LONG
 APIENTRY
@@ -550,6 +572,7 @@ RegDisableReflectionKey (
     _In_ HKEY hBase
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LONG
 APIENTRY
@@ -557,6 +580,7 @@ RegEnableReflectionKey (
     _In_ HKEY hBase
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LONG
 APIENTRY
@@ -571,6 +595,7 @@ RegQueryReflectionKey (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -579,6 +604,7 @@ RegDeleteValueA(
     _In_opt_ LPCSTR lpValueName
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -593,6 +619,7 @@ RegDeleteValueW(
 #define RegDeleteValue  RegDeleteValueA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -602,6 +629,7 @@ RegEnumKeyA (
     _Out_writes_opt_(cchName) LPSTR lpName,
     _In_ DWORD cchName
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -617,6 +645,7 @@ RegEnumKeyW (
 #define RegEnumKey  RegEnumKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -631,6 +660,7 @@ RegEnumKeyExA(
     _Out_opt_ PFILETIME lpftLastWriteTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -651,6 +681,7 @@ RegEnumKeyExW(
 #define RegEnumKeyEx  RegEnumKeyExA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -665,6 +696,7 @@ RegEnumValueA(
     _Inout_opt_ LPDWORD lpcbData
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -691,6 +723,7 @@ RegEnumValueW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -698,6 +731,7 @@ RegFlushKey(
     _In_ HKEY hKey
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -708,6 +742,7 @@ RegGetKeySecurity(
     _Inout_ LPDWORD lpcbSecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -717,6 +752,7 @@ RegLoadKeyA(
     _In_ LPCSTR lpFile
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -732,6 +768,7 @@ RegLoadKeyW(
 #define RegLoadKey  RegLoadKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -749,6 +786,7 @@ RegNotifyChangeKeyValue(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -757,6 +795,7 @@ RegOpenKeyA (
     _In_opt_ LPCSTR lpSubKey,
     _Out_ PHKEY phkResult
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -771,6 +810,7 @@ RegOpenKeyW (
 #define RegOpenKey  RegOpenKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -782,6 +822,7 @@ RegOpenKeyExA(
     _Out_ PHKEY phkResult
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -805,6 +846,7 @@ RegOpenKeyExW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -817,6 +859,7 @@ RegOpenKeyTransactedA (
     _In_        HANDLE hTransaction,
     _Reserved_ PVOID  pExtendedParemeter
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -841,6 +884,7 @@ RegOpenKeyTransactedW (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -859,6 +903,7 @@ RegQueryInfoKeyA(
     _Out_opt_ PFILETIME lpftLastWriteTime
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -889,6 +934,7 @@ RegQueryInfoKeyW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -898,6 +944,7 @@ RegQueryValueA (
     _Out_writes_bytes_to_opt_(*lpcbData, *lpcbData) __out_data_source(REGISTRY) LPSTR lpData,
     _Inout_opt_ PLONG lpcbData
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -915,6 +962,7 @@ RegQueryValueW (
 
 #if(WINVER >= 0x0400)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -926,6 +974,7 @@ RegQueryMultipleValuesA(
     _Inout_opt_ LPDWORD ldwTotsize
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -951,6 +1000,7 @@ RegQueryMultipleValuesW(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -963,6 +1013,7 @@ RegQueryValueExA(
     _When_(lpData == NULL, _Out_opt_) _When_(lpData != NULL, _Inout_opt_) LPDWORD lpcbData
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -987,6 +1038,7 @@ RegQueryValueExW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -996,6 +1048,7 @@ RegReplaceKeyA (
     _In_ LPCSTR lpNewFile,
     _In_ LPCSTR lpOldFile
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1011,6 +1064,7 @@ RegReplaceKeyW (
 #define RegReplaceKey  RegReplaceKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1020,6 +1074,7 @@ RegRestoreKeyA(
     _In_ DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1060,6 +1115,7 @@ RegRenameKey(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1068,6 +1124,7 @@ RegSaveKeyA (
     _In_ LPCSTR lpFile,
     _In_opt_ CONST LPSECURITY_ATTRIBUTES lpSecurityAttributes
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1082,6 +1139,7 @@ RegSaveKeyW (
 #define RegSaveKey  RegSaveKeyA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1091,6 +1149,7 @@ RegSetKeySecurity(
     _In_ PSECURITY_DESCRIPTOR pSecurityDescriptor
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1101,6 +1160,7 @@ RegSetValueA (
     _In_reads_bytes_opt_(cbData) LPCSTR lpData,
     _In_ DWORD cbData
     );
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1123,6 +1183,7 @@ RegSetValueW (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1135,6 +1196,7 @@ RegSetValueExA(
     _In_ DWORD cbData
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1159,6 +1221,7 @@ RegSetValueExW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1167,6 +1230,7 @@ RegUnLoadKeyA(
     _In_opt_ LPCSTR lpSubKey
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1186,6 +1250,7 @@ RegUnLoadKeyW(
 //
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1195,6 +1260,7 @@ RegDeleteKeyValueA(
     _In_opt_ LPCSTR lpValueName
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1210,6 +1276,7 @@ RegDeleteKeyValueW(
 #define RegDeleteKeyValue  RegDeleteKeyValueA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1222,6 +1289,7 @@ RegSetKeyValueA(
     _In_ DWORD cbData
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1240,6 +1308,7 @@ RegSetKeyValueW(
 #define RegSetKeyValue  RegSetKeyValueA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1248,6 +1317,7 @@ RegDeleteTreeA(
     _In_opt_ LPCSTR lpSubKey
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1262,6 +1332,7 @@ RegDeleteTreeW(
 #define RegDeleteTree  RegDeleteTreeA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1278,6 +1349,7 @@ RegCopyTreeA (
 
 #if (_WIN32_WINNT >= 0x0502)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1298,6 +1370,7 @@ RegGetValueA(
     _Inout_opt_ LPDWORD pcbData
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1328,6 +1401,7 @@ RegGetValueW(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1341,6 +1415,7 @@ RegCopyTreeW(
 #define RegCopyTree  RegCopyTreeW
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1354,6 +1429,7 @@ RegLoadMUIStringA(
     _In_opt_ LPCSTR pszDirectory
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1373,6 +1449,7 @@ RegLoadMUIStringW(
 #define RegLoadMUIString  RegLoadMUIStringA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1384,6 +1461,7 @@ RegLoadAppKeyA(
     _Reserved_ DWORD Reserved
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1413,6 +1491,8 @@ RegLoadAppKeyW(
 // Remoteable System Shutdown APIs
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 __drv_preferredFunction("InitiateSystemShutdownEx", "Legacy API. Rearchitect to avoid Reboot")
 WINADVAPI
 BOOL
@@ -1424,6 +1504,8 @@ InitiateSystemShutdownA(
     _In_ BOOL bForceAppsClosed,
     _In_ BOOL bRebootAfterShutdown
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 __drv_preferredFunction("InitiateSystemShutdownEx", "Legacy API. Rearchitect to avoid Reboot")
 WINADVAPI
 BOOL
@@ -1441,12 +1523,16 @@ InitiateSystemShutdownW(
 #define InitiateSystemShutdown  InitiateSystemShutdownA
 #endif // !UNICODE
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
 AbortSystemShutdownA(
     _In_opt_ LPSTR lpMachineName
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 WINADVAPI
 BOOL
 APIENTRY
@@ -1482,6 +1568,8 @@ AbortSystemShutdownW(
 //
 #define MAX_SHUTDOWN_TIMEOUT (10*365*24*60*60)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 __drv_preferredFunction("a design alternative", "Rearchitect to avoid Reboot")
 _When_(((dwReason==0 && lpMessage==0)) || dwReason>=0xd0000000,
     __drv_reportError("Requires a valid dwReason or lpMessage"))
@@ -1496,6 +1584,8 @@ InitiateSystemShutdownExA(
     _In_ BOOL bRebootAfterShutdown,
     _In_ DWORD dwReason
     );
+_Win32_metadata_supported_os_(windows5.1.2600)
+_Win32_metadata_set_last_error_
 __drv_preferredFunction("a design alternative", "Rearchitect to avoid Reboot")
 _When_(((dwReason==0 && lpMessage==0)) || dwReason>=0xd0000000,
     __drv_reportError("Requires a valid dwReason or lpMessage"))
@@ -1539,6 +1629,7 @@ InitiateSystemShutdownExW(
 #define SHUTDOWN_SYSTEM_INITIATED       0x00010000
 #define SHUTDOWN_UPDATE_POWEROFF        0x00020000
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 DWORD
 APIENTRY
@@ -1549,6 +1640,7 @@ InitiateShutdownA(
     _In_     DWORD dwShutdownFlags,
     _In_     DWORD dwReason
     );
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINADVAPI
 DWORD
 APIENTRY
@@ -1579,6 +1671,7 @@ CheckForHiberboot(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LSTATUS
 APIENTRY
@@ -1589,6 +1682,7 @@ RegSaveKeyExA(
     _In_ DWORD Flags
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 WINADVAPI
 LSTATUS
 APIENTRY
