diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
index 25d49f69..bbd3c988 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
@@ -19,6 +19,10 @@ Revision History:
 #define _WINSPOOL_
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 
 #ifdef _WINUSER_
 #include <prsht.h>
@@ -1334,18 +1338,26 @@ _In_    HANDLE  hPrinter,
 _In_    HANDLE  hSpoolFile
 );
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 OpenPrinterA(
-_In_opt_    LPSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_In_opt_    LPSTR             pPrinterName _Win32_metadata_const_,
+_Out_       LPHANDLE            phPrinter
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0)
+    _Win32_metadata_raii_free_(ClosePrinter),
 _In_opt_    LPPRINTER_DEFAULTSA pDefault
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 OpenPrinterW(
-_In_opt_    LPWSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_In_opt_    LPWSTR             pPrinterName _Win32_metadata_const_,
+_Out_       LPHANDLE            phPrinter
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0)
+    _Win32_metadata_raii_free_(ClosePrinter),
 _In_opt_    LPPRINTER_DEFAULTSW pDefault
 );
 #ifdef UNICODE
@@ -2569,7 +2581,10 @@ _In_     HANDLE hPrinter,
          DWORD  fdwFilter,
          DWORD  fdwOptions,
 _In_opt_ PVOID  pPrinterNotifyOptions
-    );
+    )
+    _Win32_metadata_raii_free_(FindClosePrinterChangeNotification)
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0);
 
 
 BOOL
@@ -2665,6 +2680,7 @@ _In_   LPWSTR   pCaption,
 #define PRINTER_ERROR_JAM           0x00000002
 #define PRINTER_ERROR_OUTOFTONER    0x00000004
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 ClosePrinter(
@@ -3441,19 +3457,27 @@ typedef PPRINTER_OPTIONSA PPRINTER_OPTIONS;
 typedef LPPRINTER_OPTIONSA LPPRINTER_OPTIONS;
 #endif // UNICODE
 
+    _Win32_metadata_set_last_error_
     BOOL
     WINAPI
     OpenPrinter2A(
-    _In_opt_      LPCSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _In_opt_      LPCSTR                pPrinterName _Win32_metadata_const_,
+    _Out_         LPHANDLE                phPrinter
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(ClosePrinter),
     _In_opt_      PPRINTER_DEFAULTSA      pDefault,
     _In_opt_      PPRINTER_OPTIONSA       pOptions
     );
+    _Win32_metadata_set_last_error_
     BOOL
     WINAPI
     OpenPrinter2W(
-    _In_opt_      LPCWSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _In_opt_      LPCWSTR                pPrinterName _Win32_metadata_const_,
+    _Out_         LPHANDLE                phPrinter
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(ClosePrinter),
     _In_opt_      PPRINTER_DEFAULTSW      pDefault,
     _In_opt_      PPRINTER_OPTIONSW       pOptions
     );
