diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winsafer.h b/generation/WinSDK/RecompiledIdlHeaders/um/winsafer.h
index 1c0433f9..a54c48f1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winsafer.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winsafer.h
@@ -656,7 +656,10 @@ SaferCreateLevel(
     _In_ DWORD                dwScopeId,
     _In_ DWORD                dwLevelId,
     _In_ DWORD                OpenFlags,
-    _Outptr_ SAFER_LEVEL_HANDLE * pLevelHandle,
+    _Outptr_ SAFER_LEVEL_HANDLE * pLevelHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(SaferCloseLevel),
     _Reserved_ LPVOID               lpReserved
     );
 
@@ -673,7 +676,10 @@ BOOL WINAPI
 SaferIdentifyLevel(
     _In_ DWORD              dwNumProperties,
     _In_reads_opt_(dwNumProperties) PSAFER_CODE_PROPERTIES pCodeProperties,
-    _Outptr_ SAFER_LEVEL_HANDLE  * pLevelHandle,
+    _Outptr_ SAFER_LEVEL_HANDLE  * pLevelHandle
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(SaferCloseLevel),
     _In_opt_ LPVOID         lpReserved
     );
 
@@ -683,7 +689,10 @@ BOOL WINAPI
 SaferComputeTokenFromLevel(
     _In_ SAFER_LEVEL_HANDLE LevelHandle,
     _In_opt_ HANDLE         InAccessToken,
-    _Out_ PHANDLE           OutAccessToken,
+    _Out_ PHANDLE           OutAccessToken
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(CloseHandle),
     _In_ DWORD              dwFlags,
     _Inout_opt_ LPVOID      lpReserved
     );
