diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
index 25d49f69..728a1227 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
@@ -19,6 +19,10 @@ Revision History:
 #define _WINSPOOL_
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 
 #ifdef _WINUSER_
 #include <prsht.h>
@@ -1249,6 +1253,7 @@ typedef PPRINTER_ENUM_VALUESA PPRINTER_ENUM_VALUES;
 typedef LPPRINTER_ENUM_VALUESA LPPRINTER_ENUM_VALUES;
 #endif // UNICODE
 
+_Win32_metadata_set_last_error_
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1262,6 +1267,7 @@ _Out_writes_bytes_to_opt_(cbBuf, *pcbNeeded)
 _Out_         LPDWORD pcbNeeded,
 _Out_         LPDWORD pcReturned
 );
+_Win32_metadata_set_last_error_
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1313,6 +1319,7 @@ _Out_         LPDWORD pcReturned
 #define SPOOL_FILE_PERSISTENT    0x00000001
 #define SPOOL_FILE_TEMPORARY     0x00000002
 
+_Win32_metadata_set_last_error_
 HANDLE
 WINAPI
 GetSpoolFileHandle(
@@ -1334,18 +1341,26 @@ _In_    HANDLE  hPrinter,
 _In_    HANDLE  hSpoolFile
 );
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 OpenPrinterA(
-_In_opt_    LPSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_In_opt_    LPSTR             pPrinterName _Win32_metadata_const_,
+_Out_       LPHANDLE            phPrinter
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0)
+    _Win32_metadata_raii_free_(ClosePrinter),
 _In_opt_    LPPRINTER_DEFAULTSA pDefault
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 OpenPrinterW(
-_In_opt_    LPWSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_In_opt_    LPWSTR             pPrinterName _Win32_metadata_const_,
+_Out_       LPHANDLE            phPrinter
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0)
+    _Win32_metadata_raii_free_(ClosePrinter),
 _In_opt_    LPPRINTER_DEFAULTSW pDefault
 );
 #ifdef UNICODE
@@ -1434,6 +1449,7 @@ _Out_ LPDWORD  pcbNeeded
 #define GetJob  GetJobA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 EnumJobsA(
@@ -1447,6 +1463,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_ LPDWORD pcbNeeded,
 _Out_ LPDWORD pcReturned
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 EnumJobsW(
@@ -1466,6 +1483,7 @@ _Out_ LPDWORD pcReturned
 #define EnumJobs  EnumJobsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 HANDLE
 WINAPI
 AddPrinterA(
@@ -1476,6 +1494,7 @@ _When_(Level == 1, _In_reads_bytes_(sizeof(PRINTER_INFO_1)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(PRINTER_INFO_2)))
             LPBYTE     pPrinter
 );
+_Win32_metadata_set_last_error_
 HANDLE
 WINAPI
 AddPrinterW(
@@ -1492,12 +1511,14 @@ _When_(Level == 2, _In_reads_bytes_(sizeof(PRINTER_INFO_2)))
 #define AddPrinter  AddPrinterA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DeletePrinter(
 _Inout_ HANDLE   hPrinter
 );
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetPrinterA(
@@ -1517,6 +1538,7 @@ SetPrinterA(
                      LPBYTE  pPrinter,
     _In_             DWORD   Command
     );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetPrinterW(
@@ -1542,6 +1564,7 @@ SetPrinterW(
 #define SetPrinter  SetPrinterA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 GetPrinterA(
@@ -1552,6 +1575,7 @@ _Out_writes_bytes_opt_(cbBuf)
       DWORD   cbBuf,
 _Out_ LPDWORD pcbNeeded
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 GetPrinterW(
@@ -1568,6 +1592,7 @@ _Out_ LPDWORD pcbNeeded
 #define GetPrinter  GetPrinterA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddPrinterDriverA(
@@ -1575,6 +1600,7 @@ _In_opt_    LPSTR   pName,
             DWORD     Level,
 _In_        LPBYTE    pDriverInfo
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddPrinterDriverW(
@@ -1620,6 +1646,7 @@ _In_             DWORD    dwFileCopyFlags
 #define AddPrinterDriverEx  AddPrinterDriverExA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 EnumPrinterDriversA(
@@ -1632,6 +1659,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_       LPDWORD        pcbNeeded,
 _Out_       LPDWORD        pcReturned
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 EnumPrinterDriversW(
@@ -2569,7 +2597,10 @@ _In_     HANDLE hPrinter,
          DWORD  fdwFilter,
          DWORD  fdwOptions,
 _In_opt_ PVOID  pPrinterNotifyOptions
-    );
+    )
+    _Win32_metadata_raii_free_(FindClosePrinterChangeNotification)
+    _Win32_metadata_invalid_handle_(-1)
+    _Win32_metadata_invalid_handle_(0);
 
 
 BOOL
@@ -2665,6 +2696,7 @@ _In_   LPWSTR   pCaption,
 #define PRINTER_ERROR_JAM           0x00000002
 #define PRINTER_ERROR_OUTOFTONER    0x00000004
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 ClosePrinter(
@@ -2830,6 +2862,7 @@ _Out_       LPDWORD    pcReturned
 
 
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddMonitorA(
@@ -2838,6 +2871,7 @@ _In_opt_    LPSTR pName,
 _In_reads_bytes_opt_(sizeof(MONITOR_INFO_2))
             LPBYTE  pMonitors
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddMonitorW(
@@ -2854,6 +2888,7 @@ _In_reads_bytes_opt_(sizeof(MONITOR_INFO_2))
 
 
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DeleteMonitorA(
@@ -2861,6 +2896,7 @@ _In_opt_    LPSTR     pName,
 _In_opt_    LPSTR     pEnvironment,
 _In_        LPSTR     pMonitorName
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DeleteMonitorW(
@@ -2905,6 +2941,7 @@ _Out_       LPDWORD pcReturned
 #endif // !UNICODE
 
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddPortA(
@@ -2912,6 +2949,7 @@ _In_opt_    LPSTR   pName,
 _In_        HWND      hWnd,
 _In_        LPSTR   pMonitorName
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 AddPortW(
@@ -2947,6 +2985,7 @@ _In_        LPWSTR   pPortName
 #define ConfigurePort  ConfigurePortA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DeletePortA(
@@ -2954,6 +2993,7 @@ _In_opt_    LPSTR pName,
 _In_        HWND    hWnd,
 _In_        LPSTR pPortName
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 DeletePortW(
@@ -3018,6 +3058,7 @@ _In_opt_    LPCWSTR pszPrinter
 #endif // !UNICODE
 
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetPortA(
@@ -3027,6 +3068,7 @@ _In_        LPSTR     pPortName,
 _In_reads_bytes_(sizeof(PORT_INFO_3))
             LPBYTE      pPortInfo
 );
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetPortW(
@@ -3441,19 +3483,27 @@ typedef PPRINTER_OPTIONSA PPRINTER_OPTIONS;
 typedef LPPRINTER_OPTIONSA LPPRINTER_OPTIONS;
 #endif // UNICODE
 
+    _Win32_metadata_set_last_error_
     BOOL
     WINAPI
     OpenPrinter2A(
-    _In_opt_      LPCSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _In_opt_      LPCSTR                pPrinterName _Win32_metadata_const_,
+    _Out_         LPHANDLE                phPrinter
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(ClosePrinter),
     _In_opt_      PPRINTER_DEFAULTSA      pDefault,
     _In_opt_      PPRINTER_OPTIONSA       pOptions
     );
+    _Win32_metadata_set_last_error_
     BOOL
     WINAPI
     OpenPrinter2W(
-    _In_opt_      LPCWSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _In_opt_      LPCWSTR                pPrinterName _Win32_metadata_const_,
+    _Out_         LPHANDLE                phPrinter
+        _Win32_metadata_invalid_handle_(-1)
+        _Win32_metadata_invalid_handle_(0)
+        _Win32_metadata_raii_free_(ClosePrinter),
     _In_opt_      PPRINTER_DEFAULTSW      pDefault,
     _In_opt_      PPRINTER_OPTIONSW       pOptions
     );
