diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winsxs.h b/generation/WinSDK/RecompiledIdlHeaders/um/winsxs.h
index 2abaff60..6b2c8a47 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winsxs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winsxs.h
@@ -92,6 +92,7 @@ extern "C"{
 //=--------------------------------------------------------------------------=
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #pragma comment(lib,"uuid.lib")
@@ -235,6 +236,7 @@ EXTERN_C const IID IID_IAssemblyName;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("CD193BC0-B4BC-11d2-9833-00C04FC31D2E")
     IAssemblyName : public IUnknown
     {
@@ -445,6 +447,7 @@ EXTERN_C const IID IID_IAssemblyCacheItem;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9e3aaeb4-d1cd-11d2-bab9-00c04f8eceae")
     IAssemblyCacheItem : public IUnknown
     {
@@ -561,6 +564,7 @@ EXTERN_C const IID IID_IAssemblyCache;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("e707dcde-d1cd-11d2-bab9-00c04f8eceae")
     IAssemblyCache : public IUnknown
     {
@@ -708,7 +712,9 @@ enum __MIDL___MIDL_itf_winsxs_0000_0003_0001
         CANOF_SET_DEFAULT_VALUES	= 0x2
     } 	CREATE_ASM_NAME_OBJ_FLAGS;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI CreateAssemblyNameObject(LPASSEMBLYNAME *ppAssemblyNameObj, LPCWSTR szAssemblyName, DWORD dwFlags, LPVOID pvReserved);             
+_Win32_metadata_supported_os_(windows6.0.6000)
 STDAPI CreateAssemblyCache(IAssemblyCache **ppAsmCache, DWORD dwReserved); 
 #endif
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
