diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WinUser.h b/generation/WinSDK/RecompiledIdlHeaders/um/WinUser.h
index 23e34d2f..55d1d9f7 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WinUser.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WinUser.h
@@ -1,4 +1,7 @@
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 /****************************************************************************
 *                                                                           *
@@ -300,6 +303,7 @@ typedef DESKTOPENUMPROCA    DESKTOPENUMPROC;
 #pragma warning(disable:4995)
 #endif
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -307,6 +311,7 @@ wvsprintfA(
     _Out_ LPSTR,
     _In_ _Printf_format_string_ LPCSTR,
     _In_ va_list arglist);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -320,6 +325,7 @@ wvsprintfW(
 #define wvsprintf  wvsprintfA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPIV
@@ -327,6 +333,7 @@ wsprintfA(
     _Out_ LPSTR,
     _In_ _Printf_format_string_ LPCSTR,
     ...);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPIV
@@ -1235,12 +1242,14 @@ typedef struct tagHARDWAREHOOKSTRUCT {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HKL
 WINAPI
 LoadKeyboardLayoutA(
     _In_ LPCSTR pwszKLID,
     _In_ UINT Flags);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HKL
 WINAPI
@@ -1255,6 +1264,7 @@ LoadKeyboardLayoutW(
 
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HKL
 WINAPI
@@ -1262,6 +1272,7 @@ ActivateKeyboardLayout(
     _In_ HKL hkl,
     _In_ UINT Flags);
 #else
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1284,17 +1295,20 @@ ToUnicodeEx(
     _In_opt_ HKL dwhkl);
 #endif /* WINVER >= 0x0400 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 UnloadKeyboardLayout(
     _In_ HKL hkl);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 GetKeyboardLayoutNameA(
     _Out_writes_(KL_NAMELENGTH) LPSTR pwszKLID);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1307,6 +1321,7 @@ GetKeyboardLayoutNameW(
 #endif // !UNICODE
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -1350,6 +1365,7 @@ typedef struct tagMOUSEMOVEPOINT {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -1390,6 +1406,7 @@ GetMouseMovePointsEx(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1400,6 +1417,7 @@ CreateDesktopA(
     _In_ DWORD dwFlags,
     _In_ ACCESS_MASK dwDesiredAccess,
     _In_opt_ LPSECURITY_ATTRIBUTES lpsa);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1416,6 +1434,7 @@ CreateDesktopW(
 #define CreateDesktop  CreateDesktopA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1428,6 +1447,7 @@ CreateDesktopExA(
     _In_opt_ LPSECURITY_ATTRIBUTES lpsa,
     _In_ ULONG ulHeapSize,
     _Reserved_ PVOID pvoid);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1455,6 +1475,7 @@ CreateDesktopExW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1463,6 +1484,7 @@ OpenDesktopA(
     _In_ DWORD dwFlags,
     _In_ BOOL fInherit,
     _In_ ACCESS_MASK dwDesiredAccess);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1477,6 +1499,7 @@ OpenDesktopW(
 #define OpenDesktop  OpenDesktopA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1486,6 +1509,7 @@ OpenInputDesktop(
     _In_ ACCESS_MASK dwDesiredAccess);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1493,6 +1517,7 @@ EnumDesktopsA(
     _In_opt_ HWINSTA hwinsta,
     _In_ DESKTOPENUMPROCA lpEnumFunc,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1506,6 +1531,7 @@ EnumDesktopsW(
 #define EnumDesktops  EnumDesktopsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1515,6 +1541,7 @@ EnumDesktopWindows(
     _In_ LPARAM lParam);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1522,18 +1549,21 @@ SwitchDesktop(
     _In_ HDESK hDesktop);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 SetThreadDesktop(
      _In_ HDESK hDesktop);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 CloseDesktop(
     _In_ HDESK hDesktop);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDESK
 WINAPI
@@ -1578,6 +1608,7 @@ GetThreadDesktop(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWINSTA
 WINAPI
@@ -1586,6 +1617,7 @@ CreateWindowStationA(
     _In_ DWORD dwFlags,
     _In_ ACCESS_MASK dwDesiredAccess,
     _In_opt_ LPSECURITY_ATTRIBUTES lpsa);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWINSTA
 WINAPI
@@ -1600,6 +1632,7 @@ CreateWindowStationW(
 #define CreateWindowStation  CreateWindowStationA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWINSTA
 WINAPI
@@ -1607,6 +1640,7 @@ OpenWindowStationA(
     _In_ LPCSTR lpszWinSta,
     _In_ BOOL fInherit,
     _In_ ACCESS_MASK dwDesiredAccess);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWINSTA
 WINAPI
@@ -1620,12 +1654,14 @@ OpenWindowStationW(
 #define OpenWindowStation  OpenWindowStationA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 EnumWindowStationsA(
     _In_ WINSTAENUMPROCA lpEnumFunc,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1638,18 +1674,21 @@ EnumWindowStationsW(
 #define EnumWindowStations  EnumWindowStationsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 CloseWindowStation(
     _In_ HWINSTA hWinSta);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 SetProcessWindowStation(
     _In_ HWINSTA hWinSta);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWINSTA
 WINAPI
@@ -1666,6 +1705,7 @@ GetProcessWindowStation(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1674,6 +1714,7 @@ SetUserObjectSecurity(
     _In_ PSECURITY_INFORMATION pSIRequested,
     _In_ PSECURITY_DESCRIPTOR pSID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1706,6 +1747,7 @@ typedef struct tagUSEROBJECTFLAGS {
     DWORD dwFlags;
 } USEROBJECTFLAGS, *PUSEROBJECTFLAGS;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1715,6 +1757,7 @@ GetUserObjectInformationA(
     _Out_writes_bytes_opt_(nLength) PVOID pvInfo,
     _In_ DWORD nLength,
     _Out_opt_ LPDWORD lpnLengthNeeded);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1730,6 +1773,7 @@ GetUserObjectInformationW(
 #define GetUserObjectInformation  GetUserObjectInformationA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -1738,6 +1782,7 @@ SetUserObjectInformationA(
     _In_ int nIndex,
     _In_reads_bytes_(nLength) PVOID pvInfo,
     _In_ DWORD nLength);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -2643,11 +2688,13 @@ typedef struct {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
 RegisterWindowMessageA(
     _In_ LPCSTR lpString);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -2768,6 +2815,7 @@ typedef struct tagTRACKMOUSEEVENT {
     DWORD dwHoverTime;
 } TRACKMOUSEEVENT, *LPTRACKMOUSEEVENT;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3402,6 +3450,7 @@ typedef struct tagCOMPAREITEMSTRUCT {
  * Message Function Templates
  */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3410,6 +3459,7 @@ GetMessageA(
     _In_opt_ HWND hWnd,
     _In_ UINT wMsgFilterMin,
     _In_ UINT wMsgFilterMax);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3555,6 +3605,7 @@ PeekMessageW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3564,6 +3615,7 @@ RegisterHotKey(
     _In_ UINT fsModifiers,
     _In_ UINT vk);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3634,6 +3686,7 @@ UnregisterHotKey(
 _When_((uFlags&(EWX_POWEROFF|EWX_SHUTDOWN|EWX_FORCE))!=0,
     __drv_preferredFunction("InitiateSystemShutdownEx",
         "Legacy API. Rearchitect to avoid Reboot"))
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3695,6 +3748,7 @@ SetMessageExtraInfo(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LRESULT
 WINAPI
@@ -3703,6 +3757,7 @@ SendMessageA(
     _In_ UINT Msg,
     _Pre_maybenull_ _Post_valid_ WPARAM wParam,
     _Pre_maybenull_ _Post_valid_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LRESULT
 WINAPI
@@ -3747,6 +3802,7 @@ SendMessage(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LRESULT
 WINAPI
@@ -3758,6 +3814,7 @@ SendMessageTimeoutA(
     _In_ UINT fuFlags,
     _In_ UINT uTimeout,
     _Out_opt_ PDWORD_PTR lpdwResult);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LRESULT
 WINAPI
@@ -3775,6 +3832,7 @@ SendMessageTimeoutW(
 #define SendMessageTimeout  SendMessageTimeoutA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3783,6 +3841,7 @@ SendNotifyMessageA(
     _In_ UINT Msg,
     _In_ WPARAM wParam,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3797,6 +3856,7 @@ SendNotifyMessageW(
 #define SendNotifyMessage  SendNotifyMessageA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3807,6 +3867,7 @@ SendMessageCallbackA(
     _In_ LPARAM lParam,
     _In_ SENDASYNCPROC lpResultCallBack,
     _In_ ULONG_PTR dwData);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3831,6 +3892,7 @@ typedef struct {
     LUID  luid;
 } BSMINFO, *PBSMINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 long
 WINAPI
@@ -3841,6 +3903,7 @@ BroadcastSystemMessageExA(
     _In_ WPARAM wParam,
     _In_ LPARAM lParam,
     _Out_opt_ PBSMINFO pbsmInfo);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 long
 WINAPI
@@ -3876,6 +3939,7 @@ BroadcastSystemMessageA(
     _In_ UINT Msg,
     _In_ WPARAM wParam,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 long
 WINAPI
@@ -3892,6 +3956,7 @@ BroadcastSystemMessageW(
 #endif // !UNICODE
 #elif defined(_WIN32_WINDOWS)
 // The Win95 version isn't A/W decorated
+_Win32_metadata_set_last_error_
 WINUSERAPI
 long
 WINAPI
@@ -3950,6 +4015,7 @@ typedef  HDEVNOTIFY     *PHDEVNOTIFY;
 #define DEVICE_NOTIFY_ALL_INTERFACE_CLASSES  0x00000004
 #endif /* _WIN32_WINNT >= 0x0501 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDEVNOTIFY
 WINAPI
@@ -3957,6 +4023,7 @@ RegisterDeviceNotificationA(
     _In_ HANDLE hRecipient,
     _In_ LPVOID NotificationFilter,
     _In_ DWORD Flags);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDEVNOTIFY
 WINAPI
@@ -3970,6 +4037,7 @@ RegisterDeviceNotificationW(
 #define RegisterDeviceNotification  RegisterDeviceNotificationA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -3988,6 +4056,7 @@ typedef  HPOWERNOTIFY   *PHPOWERNOTIFY;
 
 #endif
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HPOWERNOTIFY
 WINAPI
@@ -3997,6 +4066,7 @@ RegisterPowerSettingNotification(
     IN DWORD Flags
     );
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4004,6 +4074,7 @@ UnregisterPowerSettingNotification(
     IN HPOWERNOTIFY Handle
     );
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HPOWERNOTIFY
 WINAPI
@@ -4012,6 +4083,7 @@ RegisterSuspendResumeNotification (
     IN DWORD Flags
     );
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4029,6 +4101,7 @@ UnregisterSuspendResumeNotification (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4037,6 +4110,7 @@ PostMessageA(
     _In_ UINT Msg,
     _In_ WPARAM wParam,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4051,6 +4125,7 @@ PostMessageW(
 #define PostMessage  PostMessageA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4059,6 +4134,7 @@ PostThreadMessageA(
     _In_ UINT Msg,
     _In_ WPARAM wParam,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4113,6 +4189,7 @@ WINAPI
 ReplyMessage(
     _In_ LRESULT lResult);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4277,6 +4354,7 @@ WINAPI
 GetDoubleClickTime(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4289,11 +4367,13 @@ SetDoubleClickTime(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ATOM
 WINAPI
 RegisterClassA(
     _In_ CONST WNDCLASSA *lpWndClass);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ATOM
 WINAPI
@@ -4305,12 +4385,14 @@ RegisterClassW(
 #define RegisterClass  RegisterClassA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 UnregisterClassA(
     _In_ LPCSTR lpClassName,
     _In_opt_ HINSTANCE hInstance);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4330,6 +4412,7 @@ UnregisterClassW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 _Success_(return)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4338,6 +4421,7 @@ GetClassInfoA(
     _In_ LPCSTR lpClassName,
     _Out_ LPWNDCLASSA lpWndClass);
 _Success_(return)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4358,11 +4442,13 @@ GetClassInfoW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ATOM
 WINAPI
 RegisterClassExA(
     _In_ CONST WNDCLASSEXA *);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ATOM
 WINAPI
@@ -4381,6 +4467,7 @@ RegisterClassExW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 _Success_(return)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4389,6 +4476,7 @@ GetClassInfoExA(
     _In_ LPCSTR lpszClass,
     _Out_ LPWNDCLASSEXA lpwcx);
 _Success_(return)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4421,6 +4509,7 @@ GetClassInfoExW(
 typedef BOOLEAN (WINAPI * PREGISTERCLASSNAMEW)(LPCWSTR);
 #endif /* _WIN32_WINNT >= 0x0501 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -4437,6 +4526,7 @@ CreateWindowExA(
     _In_opt_ HMENU hMenu,
     _In_opt_ HINSTANCE hInstance,
     _In_opt_ LPVOID lpParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -4506,6 +4596,7 @@ IsChild(
 #pragma region Desktop or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4526,6 +4617,7 @@ ShowWindow(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if(WINVER >= 0x0500)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4544,6 +4636,7 @@ AnimateWindow(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4596,6 +4689,7 @@ UpdateLayeredWindowIndirect(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4628,6 +4722,7 @@ PrintWindow(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4709,6 +4804,7 @@ FlashWindowEx(
 
 #endif /* WINVER >= 0x0500 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4716,12 +4812,14 @@ ShowOwnedPopups(
     _In_  HWND hWnd,
     _In_  BOOL fShow);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 OpenIcon(
     _In_  HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4734,6 +4832,7 @@ CloseWindow(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4745,6 +4844,7 @@ MoveWindow(
     _In_ int nHeight,
     _In_ BOOL bRepaint);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4763,6 +4863,7 @@ SetWindowPos(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4770,6 +4871,7 @@ GetWindowPlacement(
     _In_ HWND hWnd,
     _Inout_ WINDOWPLACEMENT *lpwndpl);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4783,6 +4885,7 @@ SetWindowPlacement(
 #define WDA_EXCLUDEFROMCAPTURE 0x00000011
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4790,6 +4893,7 @@ GetWindowDisplayAffinity(
     _In_ HWND hWnd,
     _Out_ DWORD* pdwAffinity);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4807,12 +4911,14 @@ SetWindowDisplayAffinity(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDWP
 WINAPI
 BeginDeferWindowPos(
     _In_ int nNumWindows);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HDWP
 WINAPI
@@ -4827,6 +4933,7 @@ DeferWindowPos(
     _In_ UINT uFlags);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -4865,6 +4972,7 @@ WINAPI
 AnyPopup(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5006,6 +5114,7 @@ typedef LPDLGITEMTEMPLATEA LPDLGITEMTEMPLATE;
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5015,6 +5124,7 @@ CreateDialogParamA(
     _In_opt_ HWND hWndParent,
     _In_opt_ DLGPROC lpDialogFunc,
     _In_ LPARAM dwInitParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5030,6 +5140,7 @@ CreateDialogParamW(
 #define CreateDialogParam  CreateDialogParamA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5039,6 +5150,7 @@ CreateDialogIndirectParamA(
     _In_opt_ HWND hWndParent,
     _In_opt_ DLGPROC lpDialogFunc,
     _In_ LPARAM dwInitParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5074,6 +5186,7 @@ CreateDialogIndirectParamW(hInstance, lpTemplate, hWndParent, lpDialogFunc, 0L)
 #define CreateDialogIndirect  CreateDialogIndirectA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 INT_PTR
 WINAPI
@@ -5083,6 +5196,7 @@ DialogBoxParamA(
     _In_opt_ HWND hWndParent,
     _In_opt_ DLGPROC lpDialogFunc,
     _In_ LPARAM dwInitParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 INT_PTR
 WINAPI
@@ -5098,6 +5212,7 @@ DialogBoxParamW(
 #define DialogBoxParam  DialogBoxParamA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 INT_PTR
 WINAPI
@@ -5107,6 +5222,7 @@ DialogBoxIndirectParamA(
     _In_opt_ HWND hWndParent,
     _In_opt_ DLGPROC lpDialogFunc,
     _In_ LPARAM dwInitParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 INT_PTR
 WINAPI
@@ -5142,6 +5258,7 @@ DialogBoxIndirectParamW(hInstance, lpTemplate, hWndParent, lpDialogFunc, 0L)
 #define DialogBoxIndirect  DialogBoxIndirectA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5149,6 +5266,7 @@ EndDialog(
     _In_ HWND hDlg,
     _In_ INT_PTR nResult);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5156,6 +5274,7 @@ GetDlgItem(
     _In_opt_ HWND hDlg,
     _In_ int nIDDlgItem);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5165,6 +5284,7 @@ SetDlgItemInt(
     _In_ UINT uValue,
     _In_ BOOL bSigned);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -5174,6 +5294,7 @@ GetDlgItemInt(
     _Out_opt_ BOOL *lpTranslated,
     _In_ BOOL bSigned);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5181,6 +5302,7 @@ SetDlgItemTextA(
     _In_ HWND hDlg,
     _In_ int nIDDlgItem,
     _In_ LPCSTR lpString);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5195,6 +5317,7 @@ SetDlgItemTextW(
 #endif // !UNICODE
 
 _Ret_range_(0, cchMax)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -5204,6 +5327,7 @@ GetDlgItemTextA(
     _Out_writes_(cchMax) LPSTR lpString,
     _In_ int cchMax);
 _Ret_range_(0, cchMax)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -5218,6 +5342,7 @@ GetDlgItemTextW(
 #define GetDlgItemText  GetDlgItemTextA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5226,6 +5351,7 @@ CheckDlgButton(
     _In_ int nIDButton,
     _In_ UINT uCheck);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5266,6 +5392,7 @@ SendDlgItemMessageW(
 #define SendDlgItemMessage  SendDlgItemMessageA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5274,6 +5401,7 @@ GetNextDlgGroupItem(
     _In_opt_ HWND hCtl,
     _In_ BOOL bPrevious);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5282,6 +5410,7 @@ GetNextDlgTabItem(
     _In_opt_ HWND hCtl,
     _In_ BOOL bPrevious);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5336,6 +5465,7 @@ typedef enum DIALOG_CONTROL_DPI_CHANGE_BEHAVIORS {
 DEFINE_ENUM_FLAG_OPERATORS(DIALOG_CONTROL_DPI_CHANGE_BEHAVIORS);
 #endif
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetDialogControlDpiChangeBehavior(
@@ -5343,6 +5473,7 @@ SetDialogControlDpiChangeBehavior(
     _In_ DIALOG_CONTROL_DPI_CHANGE_BEHAVIORS mask,
     _In_ DIALOG_CONTROL_DPI_CHANGE_BEHAVIORS values);
 
+_Win32_metadata_set_last_error_
 DIALOG_CONTROL_DPI_CHANGE_BEHAVIORS
 WINAPI
 GetDialogControlDpiChangeBehavior(
@@ -5359,6 +5490,7 @@ typedef enum DIALOG_DPI_CHANGE_BEHAVIORS {
 DEFINE_ENUM_FLAG_OPERATORS(DIALOG_DPI_CHANGE_BEHAVIORS);
 #endif
 
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 SetDialogDpiChangeBehavior(
@@ -5366,6 +5498,7 @@ SetDialogDpiChangeBehavior(
     _In_ DIALOG_DPI_CHANGE_BEHAVIORS mask,
     _In_ DIALOG_DPI_CHANGE_BEHAVIORS values);
 
+_Win32_metadata_set_last_error_
 DIALOG_DPI_CHANGE_BEHAVIORS
 WINAPI
 GetDialogDpiChangeBehavior(
@@ -5416,12 +5549,14 @@ CallMsgFilterW(
  * Clipboard Manager Functions
  */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 OpenClipboard(
     _In_opt_ HWND hWndNewOwner);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5439,18 +5574,21 @@ GetClipboardSequenceNumber(
 
 #endif /* WINVER >= 0x0500 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
 GetClipboardOwner(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
 SetClipboardViewer(
     _In_ HWND hWndNewViewer);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5464,6 +5602,7 @@ ChangeClipboardChain(
     _In_ HWND hWndRemove,
     _In_ HWND hWndNewNext);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -5471,6 +5610,7 @@ SetClipboardData(
     _In_ UINT uFormat,
     _In_opt_ HANDLE hMem);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -5492,11 +5632,13 @@ GetClipboardMetadata(
     UINT format,
     _Inout_ PGETCLIPBMETADATA metadata);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
 RegisterClipboardFormatA(
     _In_ LPCSTR lpszFormat);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -5508,18 +5650,21 @@ RegisterClipboardFormatW(
 #define RegisterClipboardFormat  RegisterClipboardFormatA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
 CountClipboardFormats(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
 EnumClipboardFormats(
     _In_ UINT format);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5527,6 +5672,7 @@ GetClipboardFormatNameA(
     _In_ UINT format,
     _Out_writes_(cchMaxCount) LPSTR lpszFormatName,
     _In_ int cchMaxCount);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5540,18 +5686,21 @@ GetClipboardFormatNameW(
 #define GetClipboardFormatName  GetClipboardFormatNameA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 EmptyClipboard(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsClipboardFormatAvailable(
     _In_ UINT format);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5559,6 +5708,7 @@ GetPriorityClipboardFormat(
     _In_reads_(cFormats) UINT *paFormatPriorityList,
     _In_ int cFormats);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5566,18 +5716,21 @@ GetOpenClipboardWindow(
     VOID);
 
 #if(WINVER >= 0x0600)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 AddClipboardFormatListener(
     _In_ HWND hwnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 RemoveClipboardFormatListener(
     _In_ HWND hwnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5593,12 +5746,14 @@ GetUpdatedClipboardFormats(
  * Character Translation Routines
  */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 CharToOemA(
     _In_ LPCSTR pSrc,
     _Out_writes_(_Inexpressible_(strlen(pSrc) + 1)) LPSTR pDst);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5612,6 +5767,7 @@ CharToOemW(
 #endif // !UNICODE
 
 __drv_preferredFunction("OemToCharBuff","Does not validate buffer size")
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5619,6 +5775,7 @@ OemToCharA(
     _In_ LPCSTR pSrc,
     _Out_writes_(_Inexpressible_(strlen(pSrc) + 1)) LPSTR pDst);
 __drv_preferredFunction("OemToCharBuff","Does not validate buffer size")
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5631,6 +5788,7 @@ OemToCharW(
 #define OemToChar  OemToCharA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5638,6 +5796,7 @@ CharToOemBuffA(
     _In_ LPCSTR lpszSrc,
     _Out_writes_(cchDstLength) LPSTR lpszDst,
     _In_ DWORD cchDstLength);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5651,6 +5810,7 @@ CharToOemBuffW(
 #define CharToOemBuff  CharToOemBuffA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5658,6 +5818,7 @@ OemToCharBuffA(
     _In_ LPCSTR lpszSrc,
     _Out_writes_(cchDstLength) LPSTR lpszDst,
     _In_ DWORD cchDstLength);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5677,11 +5838,13 @@ OemToCharBuffW(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LPSTR
 WINAPI
 CharUpperA(
     _Inout_ LPSTR lpsz);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LPWSTR
 WINAPI
@@ -5711,11 +5874,13 @@ CharUpperBuffW(
 #define CharUpperBuff  CharUpperBuffA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LPSTR
 WINAPI
 CharLowerA(
     _Inout_ LPSTR lpsz);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LPWSTR
 WINAPI
@@ -5823,11 +5988,13 @@ CharPrevExA(
  * Language dependent Routines
  */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsCharAlphaA(
     _In_ CHAR ch);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5839,11 +6006,13 @@ IsCharAlphaW(
 #define IsCharAlpha  IsCharAlphaA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsCharAlphaNumericA(
     _In_ CHAR ch);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5855,11 +6024,13 @@ IsCharAlphaNumericW(
 #define IsCharAlphaNumeric  IsCharAlphaNumericA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsCharUpperA(
     _In_ CHAR ch);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5871,11 +6042,13 @@ IsCharUpperW(
 #define IsCharUpper  IsCharUpperA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsCharLowerA(
     _In_ CHAR ch);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5895,6 +6068,7 @@ IsCharLowerW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -5931,6 +6105,7 @@ WINAPI
 GetAsyncKeyState(
     _In_ int vKey);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 _Check_return_
 BOOL
@@ -5938,6 +6113,7 @@ WINAPI
 GetKeyboardState(
     _Out_writes_(256) PBYTE lpKeyState);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -5951,6 +6127,7 @@ SetKeyboardState(
 #pragma region  Desktop or PC Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5958,6 +6135,7 @@ GetKeyNameTextA(
     _In_ LONG lParam,
     _Out_writes_(cchSize) LPSTR lpString,
     _In_ int cchSize);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -5978,6 +6156,7 @@ GetKeyNameTextW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -6166,6 +6345,7 @@ typedef struct tagINPUT {
     } DUMMYUNIONNAME;
 } INPUT, *PINPUT, FAR* LPINPUT;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -6238,6 +6418,7 @@ typedef TOUCHINPUT const * PCTOUCHINPUT;
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6247,6 +6428,7 @@ GetTouchInputInfo(
     _Out_writes_(cInputs) PTOUCHINPUT pInputs,  // array of touch inputs
     _In_ int cbSize);                           // sizeof(TOUCHINPUT)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6267,6 +6449,7 @@ CloseTouchInputHandle(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6274,6 +6457,7 @@ RegisterTouchWindow(
     _In_ HWND hwnd,
     _In_ ULONG ulFlags);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6480,6 +6664,7 @@ typedef enum {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6487,6 +6672,7 @@ InitializeTouchInjection(
     _In_ UINT32 maxCount,
     _In_ DWORD dwMode);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6523,6 +6709,7 @@ typedef struct tagINPUT_INJECTION_VALUE {
     USHORT index;
 }INPUT_INJECTION_VALUE, *PINPUT_INJECTION_VALUE;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6530,6 +6717,7 @@ GetPointerType(
     _In_ UINT32 pointerId,
     _Out_ POINTER_INPUT_TYPE *pointerType);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6537,6 +6725,7 @@ GetPointerCursorId(
     _In_ UINT32 pointerId,
     _Out_ UINT32 *cursorId);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6544,6 +6733,7 @@ GetPointerInfo(
     _In_ UINT32 pointerId,
     _Out_writes_(1) POINTER_INFO *pointerInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6552,6 +6742,7 @@ GetPointerInfoHistory(
     _Inout_ UINT32 *entriesCount,
     _Out_writes_opt_(*entriesCount) POINTER_INFO *pointerInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6560,6 +6751,7 @@ GetPointerFrameInfo(
     _Inout_ UINT32 *pointerCount,
     _Out_writes_opt_(*pointerCount) POINTER_INFO *pointerInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6569,6 +6761,7 @@ GetPointerFrameInfoHistory(
     _Inout_ UINT32 *pointerCount,
     _Out_writes_opt_(*entriesCount * *pointerCount) POINTER_INFO *pointerInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6576,6 +6769,7 @@ GetPointerTouchInfo(
     _In_ UINT32 pointerId,
     _Out_writes_(1) POINTER_TOUCH_INFO *touchInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6584,6 +6778,7 @@ GetPointerTouchInfoHistory(
     _Inout_ UINT32 *entriesCount,
     _Out_writes_opt_(*entriesCount) POINTER_TOUCH_INFO *touchInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6592,6 +6787,7 @@ GetPointerFrameTouchInfo(
     _Inout_ UINT32 *pointerCount,
     _Out_writes_opt_(*pointerCount) POINTER_TOUCH_INFO *touchInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6601,6 +6797,7 @@ GetPointerFrameTouchInfoHistory(
     _Inout_ UINT32 *pointerCount,
     _Out_writes_opt_(*entriesCount * *pointerCount) POINTER_TOUCH_INFO *touchInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6608,6 +6805,7 @@ GetPointerPenInfo(
     _In_ UINT32 pointerId,
     _Out_writes_(1) POINTER_PEN_INFO *penInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6616,6 +6814,7 @@ GetPointerPenInfoHistory(
     _Inout_ UINT32 *entriesCount,
     _Out_writes_opt_(*entriesCount) POINTER_PEN_INFO *penInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6624,6 +6823,7 @@ GetPointerFramePenInfo(
     _Inout_ UINT32 *pointerCount,
     _Out_writes_opt_(*pointerCount) POINTER_PEN_INFO *penInfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6635,12 +6835,14 @@ GetPointerFramePenInfoHistory(
 
 // TODO(47499024): Make public when Feature_TouchpadPublicApis3 is enabled
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 SkipPointerFrameMessages(
     _In_ UINT32 pointerId);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6648,6 +6850,7 @@ RegisterPointerInputTarget(
     _In_ HWND hwnd,
     _In_ POINTER_INPUT_TYPE pointerType);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6672,6 +6875,7 @@ UnregisterPointerInputTargetEx(
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS5)
 DECLARE_HANDLE(HSYNTHETICPOINTERDEVICE);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HSYNTHETICPOINTERDEVICE
 WINAPI
@@ -6680,6 +6884,7 @@ CreateSyntheticPointerDevice(
     _In_ ULONG maxCount,
     _In_ POINTER_FEEDBACK_MODE mode);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6697,6 +6902,7 @@ DestroySyntheticPointerDevice(
 
 // TODO(47499024): Make public when Feature_TouchpadPublicApis3 is enabled
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6721,6 +6927,7 @@ EnableMouseInPointerForThread(VOID);
 #define TOUCH_HIT_TESTING_CLIENT  0x1
 #define TOUCH_HIT_TESTING_NONE    0x2
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6751,6 +6958,7 @@ typedef struct tagTOUCH_HIT_TESTING_INPUT
 #define TOUCH_HIT_TESTING_PROXIMITY_CLOSEST  0x0
 #define TOUCH_HIT_TESTING_PROXIMITY_FARTHEST  0xFFF
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6759,6 +6967,7 @@ EvaluateProximityToRect(
     _In_ const TOUCH_HIT_TESTING_INPUT *pHitTestingInput,
     _Out_ TOUCH_HIT_TESTING_PROXIMITY_EVALUATION *pProximityEval);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6768,6 +6977,7 @@ EvaluateProximityToPolygon(
     _In_ const TOUCH_HIT_TESTING_INPUT *pHitTestingInput,
     _Out_ TOUCH_HIT_TESTING_PROXIMITY_EVALUATION *pProximityEval);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LRESULT
 WINAPI
@@ -6849,6 +7059,7 @@ typedef struct tagINPUT_TRANSFORM {
 #endif
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -6971,12 +7182,14 @@ WINAPI
 SetCapture(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 ReleaseCapture(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -6987,6 +7200,7 @@ MsgWaitForMultipleObjects(
     _In_ DWORD dwMilliseconds,
     _In_ DWORD dwWakeMask);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -7074,6 +7288,7 @@ MsgWaitForMultipleObjectsEx(
  * Windows Functions
  */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT_PTR
 WINAPI
@@ -7101,6 +7316,7 @@ SetTimer(
 
 #if(WINVER >= 0x0601)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT_PTR
 WINAPI
@@ -7119,6 +7335,7 @@ SetCoalescableTimer(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7151,12 +7368,14 @@ WINAPI
 IsWindowEnabled(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HACCEL
 WINAPI
 LoadAcceleratorsA(
     _In_opt_ HINSTANCE hInstance,
     _In_ LPCSTR lpTableName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HACCEL
 WINAPI
@@ -7169,12 +7388,14 @@ LoadAcceleratorsW(
 #define LoadAccelerators  LoadAcceleratorsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HACCEL
 WINAPI
 CreateAcceleratorTableA(
     _In_reads_(cAccel) LPACCEL paccel,
     _In_ int cAccel);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HACCEL
 WINAPI
@@ -7215,6 +7436,7 @@ CopyAcceleratorTableW(
 
 #ifndef NOMSG
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -7222,6 +7444,7 @@ TranslateAcceleratorA(
     _In_ HWND hWnd,
     _In_ HACCEL hAccTable,
     _In_ LPMSG lpMsg);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -7413,6 +7636,7 @@ GetSystemMetrics(
 
 
 #if(WINVER >= 0x0605)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -7467,12 +7691,14 @@ GetSystemMetricsForDpi(
 
 #define MENU_ALL_ACCESS (STANDARD_RIGHTS_ALL | MENU_READ_ACCESS | MENU_WRITE_ACCESS | MENU_EXECUTE_ACCESS)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
 LoadMenuA(
     _In_opt_ HINSTANCE hInstance,
     _In_ LPCSTR lpMenuName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
@@ -7485,11 +7711,13 @@ LoadMenuW(
 #define LoadMenu  LoadMenuA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
 LoadMenuIndirectA(
     _In_ CONST MENUTEMPLATEA *lpMenuTemplate);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
@@ -7507,6 +7735,7 @@ WINAPI
 GetMenu(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7579,6 +7808,7 @@ GetMenuState(
     _In_ UINT uId,
     _In_ UINT uFlags);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7599,18 +7829,21 @@ GetSystemMenu(
     _In_ BOOL bRevert);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
 CreateMenu(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HMENU
 WINAPI
 CreatePopupMenu(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7647,12 +7880,14 @@ GetMenuItemID(
     _In_ HMENU hMenu,
     _In_ int nPos);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
 GetMenuItemCount(
     _In_opt_ HMENU hMenu);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7662,6 +7897,7 @@ InsertMenuA(
     _In_ UINT uFlags,
     _In_ UINT_PTR uIDNewItem,
     _In_opt_ LPCSTR lpNewItem);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7677,6 +7913,7 @@ InsertMenuW(
 #define InsertMenu  InsertMenuA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7685,6 +7922,7 @@ AppendMenuA(
     _In_ UINT uFlags,
     _In_ UINT_PTR uIDNewItem,
     _In_opt_ LPCSTR lpNewItem);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7699,6 +7937,7 @@ AppendMenuW(
 #define AppendMenu  AppendMenuA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7708,6 +7947,7 @@ ModifyMenuA(
     _In_ UINT uFlags,
     _In_ UINT_PTR uIDNewItem,
     _In_opt_ LPCSTR lpNewItem);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7723,6 +7963,7 @@ ModifyMenuW(
 #define ModifyMenu  ModifyMenuA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI RemoveMenu(
@@ -7730,6 +7971,7 @@ WINAPI RemoveMenu(
     _In_ UINT uPosition,
     _In_ UINT uFlags);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7738,6 +7980,7 @@ DeleteMenu(
     _In_ UINT uPosition,
     _In_ UINT uFlags);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7754,6 +7997,7 @@ WINAPI
 GetMenuCheckMarkDimensions(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7780,6 +8024,7 @@ typedef struct tagTPMPARAMS
 }   TPMPARAMS;
 typedef TPMPARAMS FAR *LPTPMPARAMS;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7793,6 +8038,7 @@ TrackPopupMenuEx(
 #endif /* WINVER >= 0x0400 */
 
 #if(_WIN32_WINNT >= 0x0601)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7833,6 +8079,7 @@ typedef struct tagMENUINFO
 }   MENUINFO, FAR *LPMENUINFO;
 typedef MENUINFO CONST FAR *LPCMENUINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7840,6 +8087,7 @@ GetMenuInfo(
     _In_ HMENU,
     _Inout_ LPMENUINFO);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7847,6 +8095,7 @@ SetMenuInfo(
     _In_ HMENU,
     _In_ LPCMENUINFO);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7959,6 +8208,7 @@ typedef LPCMENUITEMINFOA LPCMENUITEMINFO;
 #endif // UNICODE
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7967,6 +8217,7 @@ InsertMenuItemA(
     _In_ UINT item,
     _In_ BOOL fByPosition,
     _In_ LPCMENUITEMINFOA lpmi);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7981,6 +8232,7 @@ InsertMenuItemW(
 #define InsertMenuItem  InsertMenuItemA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -7989,6 +8241,7 @@ GetMenuItemInfoA(
     _In_ UINT item,
     _In_ BOOL fByPosition,
     _Inout_ LPMENUITEMINFOA lpmii);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8003,6 +8256,7 @@ GetMenuItemInfoW(
 #define GetMenuItemInfo  GetMenuItemInfoA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8011,6 +8265,7 @@ SetMenuItemInfoA(
     _In_ UINT item,
     _In_ BOOL fByPositon,
     _In_ LPCMENUITEMINFOA lpmii);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8029,6 +8284,7 @@ SetMenuItemInfoW(
 #define GMDI_USEDISABLED    0x0001L
 #define GMDI_GOINTOPOPUPS   0x0002L
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -8037,6 +8293,7 @@ GetMenuDefaultItem(
     _In_ UINT fByPos,
     _In_ UINT gmdiFlags);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8045,6 +8302,7 @@ SetMenuDefaultItem(
     _In_ UINT uItem,
     _In_ UINT fByPos);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8165,6 +8423,7 @@ DragDetect(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8503,6 +8762,7 @@ WINAPI
 UpdateWindow(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -8539,6 +8799,7 @@ SetForegroundWindow(
     _In_ HWND hWnd);
 
 #if(_WIN32_WINNT >= 0x0500)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8547,6 +8808,7 @@ AllowSetForegroundWindow(
 
 #define ASFW_ANY    ((DWORD)-1)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8761,6 +9023,7 @@ WINAPI
 LockWindowUpdate(
     _In_opt_ HWND hWndLock);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8771,6 +9034,7 @@ ScrollWindow(
     _In_opt_ CONST RECT *lpRect,
     _In_opt_ CONST RECT *lpClipRect);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8783,6 +9047,7 @@ ScrollDC(
     _In_opt_ HRGN hrgnUpdate,
     _Out_opt_ LPRECT lprcUpdate);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -8811,6 +9076,7 @@ ScrollWindowEx(
 
 #ifndef NOSCROLL
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -8820,6 +9086,7 @@ SetScrollPos(
     _In_ int nPos,
     _In_ BOOL bRedraw);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -8827,6 +9094,7 @@ GetScrollPos(
     _In_ HWND hWnd,
     _In_ int nBar);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8837,6 +9105,7 @@ SetScrollRange(
     _In_ int nMaxPos,
     _In_ BOOL bRedraw);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8846,6 +9115,7 @@ GetScrollRange(
     _Out_ LPINT lpMinPos,
     _Out_ LPINT lpMaxPos);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8854,6 +9124,7 @@ ShowScrollBar(
     _In_ int wBar,
     _In_ BOOL bShow);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8882,6 +9153,7 @@ EnableScrollBar(
 #endif  /* !NOSCROLL */
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8889,6 +9161,7 @@ SetPropA(
     _In_ HWND hWnd,
     _In_ LPCSTR lpString,
     _In_opt_ HANDLE hData);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8920,12 +9193,14 @@ GetPropW(
 #define GetProp  GetPropA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
 RemovePropA(
     _In_ HWND hWnd,
     _In_ LPCSTR lpString);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -8976,12 +9251,14 @@ EnumPropsW(
 #define EnumProps  EnumPropsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 SetWindowTextA(
     _In_ HWND hWnd,
     _In_opt_ LPCSTR lpString);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -8995,6 +9272,7 @@ SetWindowTextW(
 #endif // !UNICODE
 
 _Ret_range_(0, nMaxCount)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9003,6 +9281,7 @@ GetWindowTextA(
     _Out_writes_(nMaxCount) LPSTR lpString,
     _In_ int nMaxCount);
 _Ret_range_(0, nMaxCount)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9016,11 +9295,13 @@ GetWindowTextW(
 #define GetWindowText  GetWindowTextA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
 GetWindowTextLengthA(
     _In_ HWND hWnd);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9038,6 +9319,7 @@ GetWindowTextLengthW(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9051,6 +9333,7 @@ GetClientRect(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9064,6 +9347,7 @@ GetWindowRect(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9072,6 +9356,7 @@ AdjustWindowRect(
     _In_ DWORD dwStyle,
     _In_ BOOL bMenu);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9088,6 +9373,7 @@ AdjustWindowRectEx(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if(WINVER >= 0x0605)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9120,6 +9406,7 @@ typedef struct tagHELPINFO      /* Structure pointed to by lParam of WM_HELP */
     POINT   MousePos;           /* Mouse Position in screen co-ordinates */
 }  HELPINFO, FAR *LPHELPINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9133,6 +9420,7 @@ WINAPI
 GetWindowContextHelpId(
     _In_ HWND);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9225,6 +9513,7 @@ GetMenuContextHelpId(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9233,6 +9522,7 @@ MessageBoxA(
     _In_opt_ LPCSTR lpText,
     _In_opt_ LPCSTR lpCaption,
     _In_ UINT uType);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9271,6 +9561,7 @@ MessageBox(
 }
 #endif  /* _M_CEE */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9280,6 +9571,7 @@ MessageBoxExA(
     _In_opt_ LPCSTR lpCaption,
     _In_ UINT uType,
     _In_ WORD wLanguageId);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -9360,6 +9652,7 @@ MessageBoxIndirectW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9393,6 +9686,7 @@ ShowCursor(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9401,6 +9695,7 @@ SetCursorPos(
     _In_ int Y);
 
 #if(WINVER >= 0x0600)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9421,6 +9716,7 @@ WINAPI
 SetCursor(
     _In_opt_ HCURSOR hCursor);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9434,6 +9730,7 @@ GetCursorPos(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if(WINVER >= 0x0600)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9442,6 +9739,7 @@ GetPhysicalCursorPos(
 #endif /* WINVER >= 0x0600 */
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9454,6 +9752,7 @@ WINAPI
 GetCursor(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9463,36 +9762,42 @@ CreateCaret(
     _In_ int nWidth,
     _In_ int nHeight);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
 GetCaretBlinkTime(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 SetCaretBlinkTime(
     _In_ UINT uMSeconds);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 DestroyCaret(
     VOID);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 HideCaret(
     _In_opt_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 ShowCaret(
     _In_opt_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9500,6 +9805,7 @@ SetCaretPos(
     _In_ int X,
     _In_ int Y);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9589,6 +9895,7 @@ ChildWindowFromPoint(
 
 #pragma region Desktop or PC Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9701,6 +10008,7 @@ GetSysColorBrush(
 
 #endif /* WINVER >= 0x0400 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -9857,12 +10165,14 @@ SetWindowWord(
 
 #ifndef NOWINOFFSETS
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG
 WINAPI
 GetWindowLongA(
     _In_ HWND hWnd,
     _In_ int nIndex);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG
 WINAPI
@@ -9875,6 +10185,7 @@ GetWindowLongW(
 #define GetWindowLong  GetWindowLongA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG
 WINAPI
@@ -9882,6 +10193,7 @@ SetWindowLongA(
     _In_ HWND hWnd,
     _In_ int nIndex,
     _In_ LONG dwNewLong);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG
 WINAPI
@@ -9897,12 +10209,14 @@ SetWindowLongW(
 
 #ifdef _WIN64
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG_PTR
 WINAPI
 GetWindowLongPtrA(
     _In_ HWND hWnd,
     _In_ int nIndex);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG_PTR
 WINAPI
@@ -9915,6 +10229,7 @@ GetWindowLongPtrW(
 #define GetWindowLongPtr  GetWindowLongPtrA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG_PTR
 WINAPI
@@ -9922,6 +10237,7 @@ SetWindowLongPtrA(
     _In_ HWND hWnd,
     _In_ int nIndex,
     _In_ LONG_PTR dwNewLong);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 LONG_PTR
 WINAPI
@@ -9965,6 +10281,7 @@ SetWindowLongPtrW(
 
 #ifndef NOWINOFFSETS
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 WORD
 WINAPI
@@ -9972,6 +10289,7 @@ GetClassWord(
     _In_ HWND hWnd,
     _In_ int nIndex);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 WORD
 WINAPI
@@ -9980,12 +10298,14 @@ SetClassWord(
     _In_ int nIndex,
     _In_ WORD wNewWord);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
 GetClassLongA(
     _In_ HWND hWnd,
     _In_ int nIndex);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -9998,6 +10318,7 @@ GetClassLongW(
 #define GetClassLong  GetClassLongA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -10005,6 +10326,7 @@ SetClassLongA(
     _In_ HWND hWnd,
     _In_ int nIndex,
     _In_ LONG dwNewLong);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -10020,12 +10342,14 @@ SetClassLongW(
 
 #ifdef _WIN64
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ULONG_PTR
 WINAPI
 GetClassLongPtrA(
     _In_ HWND hWnd,
     _In_ int nIndex);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ULONG_PTR
 WINAPI
@@ -10038,6 +10362,7 @@ GetClassLongPtrW(
 #define GetClassLongPtr  GetClassLongPtrA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ULONG_PTR
 WINAPI
@@ -10045,6 +10370,7 @@ SetClassLongPtrA(
     _In_ HWND hWnd,
     _In_ int nIndex,
     _In_ LONG_PTR dwNewLong);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 ULONG_PTR
 WINAPI
@@ -10081,12 +10407,14 @@ SetClassLongPtrW(
 #endif /* !NOWINOFFSETS */
 
 #if(WINVER >= 0x0500)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 GetProcessDefaultLayout(
     _Out_ DWORD *pdwDefaultLayout);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10101,12 +10429,14 @@ GetDesktopWindow(
     VOID);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
 GetParent(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10123,12 +10453,14 @@ EnumChildWindows(
     _In_ LPARAM lParam);
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
 FindWindowA(
     _In_opt_ LPCSTR lpClassName,
     _In_opt_ LPCSTR lpWindowName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10142,6 +10474,7 @@ FindWindowW(
 #endif // !UNICODE
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10150,6 +10483,7 @@ FindWindowExA(
     _In_opt_ HWND hWndChildAfter,
     _In_opt_ LPCSTR lpszClass,
     _In_opt_ LPCSTR lpszWindow);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10185,6 +10519,7 @@ WINAPI
 DeregisterShellHookWindow(
     _In_ HWND hwnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10209,6 +10544,7 @@ EnumThreadWindows(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -10217,6 +10553,7 @@ GetClassNameA(
     _Out_writes_to_(nMaxCount, return) LPSTR lpClassName,
     _In_ int nMaxCount
     );
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -10259,6 +10596,7 @@ GetClassName(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10312,6 +10650,7 @@ GetLastActivePopup(
 #define GW_MAX              6
 #endif
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -10371,6 +10710,7 @@ UnhookWindowsHook(
     _In_ int nCode,
     _In_ HOOKPROC pfnFilterProc);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HHOOK
 WINAPI
@@ -10379,6 +10719,7 @@ SetWindowsHookExA(
     _In_ HOOKPROC lpfn,
     _In_opt_ HINSTANCE hmod,
     _In_ DWORD dwThreadId);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HHOOK
 WINAPI
@@ -10394,6 +10735,7 @@ SetWindowsHookExW(
 #endif // !UNICODE
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10504,6 +10846,7 @@ CallNextHookEx(
 
 #if(WINVER >= 0x0400)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10608,12 +10951,14 @@ LoadBitmapW(
 #pragma region Desktop or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HCURSOR
 WINAPI
 LoadCursorA(
     _In_opt_ HINSTANCE hInstance,
     _In_ LPCSTR lpCursorName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HCURSOR
 WINAPI
@@ -10632,11 +10977,13 @@ LoadCursorW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HCURSOR
 WINAPI
 LoadCursorFromFileA(
     _In_ LPCSTR lpFileName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HCURSOR
 WINAPI
@@ -10654,6 +11001,7 @@ LoadCursorFromFileW(
 #pragma region Desktop or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HCURSOR
 WINAPI
@@ -10666,6 +11014,7 @@ CreateCursor(
     _In_ CONST VOID *pvANDPlane,
     _In_ CONST VOID *pvXORPlane);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10723,6 +11072,7 @@ CopyCursor(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -10739,12 +11089,14 @@ typedef struct _ICONINFO {
 } ICONINFO;
 typedef ICONINFO *PICONINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
 LoadIconA(
     _In_opt_ HINSTANCE hInstance,
     _In_ LPCSTR lpIconName);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
@@ -10788,6 +11140,7 @@ PrivateExtractIconsW(
 #define PrivateExtractIcons  PrivateExtractIconsA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
@@ -10800,12 +11153,14 @@ CreateIcon(
     _In_ CONST BYTE *lpbANDbits,
     _In_ CONST BYTE *lpbXORbits);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 DestroyIcon(
     _In_ HICON hIcon);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -10814,6 +11169,7 @@ LookupIconIdFromDirectory(
     _In_ BOOL fIcon);
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -10825,6 +11181,7 @@ LookupIconIdFromDirectoryEx(
     _In_ UINT Flags);
 #endif /* WINVER >= 0x0400 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
@@ -10835,6 +11192,7 @@ CreateIconFromResource(
     _In_ DWORD dwVer);
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
@@ -10901,6 +11259,7 @@ SetThreadCursorCreationScaling(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -10911,6 +11270,7 @@ LoadImageA(
     _In_ int cx,
     _In_ int cy,
     _In_ UINT fuLoad);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -10927,6 +11287,7 @@ LoadImageW(
 #define LoadImage  LoadImageA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HANDLE
 WINAPI
@@ -10946,6 +11307,7 @@ CopyImage(
 #define DI_NOMIRROR     0x0010
 #endif /* _WIN32_WINNT >= 0x0501 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI BOOL WINAPI DrawIconEx(
     _In_ HDC hdc,
     _In_ int xLeft,
@@ -10965,18 +11327,21 @@ WINUSERAPI BOOL WINAPI DrawIconEx(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
 CreateIconIndirect(
     _In_ PICONINFO piconinfo);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HICON
 WINAPI
 CopyIcon(
     _In_ HICON hIcon);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -11176,6 +11541,7 @@ GetIconInfoExW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -11184,6 +11550,7 @@ LoadStringA(
     _In_ UINT uID,
     _Out_writes_to_(cchBufferMax, return + 1) LPSTR lpBuffer,
     _In_ int cchBufferMax);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -11568,6 +11935,7 @@ IsDialogMessageW(
 
 #endif /* !NOMSG */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -11619,6 +11987,7 @@ DlgDirListW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -11627,6 +11996,7 @@ DlgDirSelectExA(
     _Out_writes_(chCount) LPSTR lpString,
     _In_ int chCount,
     _In_ int idListBox);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -11641,6 +12011,7 @@ DlgDirSelectExW(
 #define DlgDirSelectEx  DlgDirSelectExA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -11650,6 +12021,7 @@ DlgDirListComboBoxA(
     _In_ int nIDComboBox,
     _In_ int nIDStaticPath,
     _In_ UINT uFiletype);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -11665,6 +12037,7 @@ DlgDirListComboBoxW(
 #define DlgDirListComboBox  DlgDirListComboBoxA
 #endif // !UNICODE
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -11673,6 +12046,7 @@ DlgDirSelectComboBoxExA(
     _Out_writes_(cchOut) LPSTR lpString,
     _In_ int cchOut,
     _In_ int idComboBox);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -12056,6 +12430,7 @@ SetScrollInfo(
     _In_ LPCSCROLLINFO lpsi,
     _In_ BOOL redraw);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -12194,12 +12569,14 @@ TranslateMDISysAccel(
 
 #endif /* !NOMSG */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
 ArrangeIconicWindows(
     _In_ HWND hWnd);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -12214,6 +12591,7 @@ CreateMDIWindowA(
     _In_opt_ HWND hWndParent,
     _In_opt_ HINSTANCE hInstance,
     _In_ LPARAM lParam);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 HWND
 WINAPI
@@ -12235,6 +12613,7 @@ CreateMDIWindowW(
 #endif // !UNICODE
 
 #if(WINVER >= 0x0400)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 WORD
 WINAPI
@@ -12245,6 +12624,7 @@ TileWindows(
     _In_ UINT cKids,
     _In_reads_opt_(cKids) const HWND FAR * lpKids);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 WORD
 WINAPI CascadeWindows(
@@ -12367,6 +12747,7 @@ typedef LPHELPWININFOA LPHELPWININFO;
 
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -12375,6 +12756,7 @@ WinHelpA(
     _In_opt_ LPCSTR lpszHelp,
     _In_ UINT uCommand,
     _In_ ULONG_PTR dwData);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -12412,6 +12794,7 @@ WinHelpW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if(WINVER >= 0x0500)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 DWORD
 WINAPI
@@ -13251,6 +13634,7 @@ DisplayConfigSetDeviceInfo(
 #endif /* _WINGDI_ */
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 _Success_(return != FALSE)
 BOOL
@@ -13260,6 +13644,7 @@ SystemParametersInfoA(
     _In_ UINT uiParam,
     _Pre_maybenull_ _Post_valid_ PVOID pvParam,
     _In_ UINT fWinIni);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 _Success_(return != FALSE)
 BOOL
@@ -13277,6 +13662,7 @@ SystemParametersInfoW(
 
 
 #if(WINVER >= 0x0605)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 _Success_(return != FALSE)
 BOOL
@@ -13559,6 +13945,7 @@ SetDebugErrorLevel(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 VOID
 WINAPI
@@ -13566,6 +13953,7 @@ SetLastErrorEx(
     _In_ DWORD dwErrCode,
     _In_ DWORD dwType);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 int
 WINAPI
@@ -13576,6 +13964,7 @@ InternalGetWindowText(
 
 
 #if defined(WINNT)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14428,6 +14817,7 @@ typedef struct tagGUITHREADINFO
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14435,6 +14825,7 @@ GetGUIThreadInfo(
     _In_ DWORD idThread,
     _Inout_ PGUITHREADINFO pgui);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14521,6 +14912,7 @@ WINAPI
 GetSystemDpiForProcess(
     _In_ HANDLE hProcess);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14537,6 +14929,7 @@ InheritWindowMonitor(
 #endif /* WINVER >= 0x0605 */
 
 #if(WINVER >= 0x0605)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14659,6 +15052,7 @@ typedef struct tagCURSORINFO
 #define CURSOR_SUPPRESSED  0x00000002
 #endif /* WINVER >= 0x0602 */
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14690,6 +15084,7 @@ typedef struct tagWINDOWINFO
 
 #define WS_ACTIVECAPTION    0x0001
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14707,6 +15102,7 @@ typedef struct tagTITLEBARINFO
     DWORD rgstate[CCHILDREN_TITLEBAR + 1];
 } TITLEBARINFO, *PTITLEBARINFO, *LPTITLEBARINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14738,6 +15134,7 @@ typedef struct tagMENUBARINFO
     BOOL fUnused:30;     // reserved
 } MENUBARINFO, *PMENUBARINFO, *LPMENUBARINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14761,6 +15158,7 @@ typedef struct tagSCROLLBARINFO
     DWORD rgstate[CCHILDREN_SCROLLBAR + 1];
 } SCROLLBARINFO, *PSCROLLBARINFO, *LPSCROLLBARINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14783,6 +15181,7 @@ typedef struct tagCOMBOBOXINFO
     HWND hwndList;
 } COMBOBOXINFO, *PCOMBOBOXINFO, *LPCOMBOBOXINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14840,6 +15239,7 @@ RealGetWindowClassA(
  * This gets the name of the window TYPE, not class.  This allows us to
  * recognize ThunderButton32 et al.
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -14869,6 +15269,7 @@ typedef struct tagALTTABINFO
     POINT ptStart;
 } ALTTABINFO, *PALTTABINFO, *LPALTTABINFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14878,6 +15279,7 @@ GetAltTabInfoA(
     _Inout_ PALTTABINFO pati,
     _Out_writes_opt_(cchItemText) LPSTR pszItemText,
     _In_ UINT cchItemText);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14914,6 +15316,7 @@ GetListBoxInfo(
 
 
 #if(_WIN32_WINNT >= 0x0500)
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -14923,6 +15326,7 @@ LockWorkStation(
 
 #if(_WIN32_WINNT >= 0x0500)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15265,6 +15669,7 @@ typedef struct tagRID_DEVICE_INFO {
     } DUMMYUNIONNAME;
 } RID_DEVICE_INFO, *PRID_DEVICE_INFO, *LPRID_DEVICE_INFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -15273,6 +15678,7 @@ GetRawInputDeviceInfoA(
     _In_ UINT uiCommand,
     _Inout_updates_bytes_to_opt_(*pcbSize, *pcbSize) LPVOID pData,
     _Inout_ PUINT pcbSize);
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -15291,6 +15697,7 @@ GetRawInputDeviceInfoW(
 /*
  * Raw Input Bulk Read: GetRawInputBuffer
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -15347,6 +15754,7 @@ typedef CONST RAWINPUTDEVICE* PCRAWINPUTDEVICE;
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15355,6 +15763,7 @@ RegisterRawInputDevices(
     _In_ UINT uiNumDevices,
     _In_ UINT cbSize);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -15369,6 +15778,7 @@ typedef struct tagRAWINPUTDEVICELIST {
     DWORD dwType;
 } RAWINPUTDEVICELIST, *PRAWINPUTDEVICELIST;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 UINT
 WINAPI
@@ -15456,6 +15866,7 @@ typedef struct tagPOINTER_DEVICE_CURSOR_INFO {
     POINTER_DEVICE_CURSOR_TYPE cursor;
 } POINTER_DEVICE_CURSOR_INFO;
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15463,6 +15874,7 @@ GetPointerDevices(
     _Inout_ UINT32* deviceCount,
     _Out_writes_opt_(*deviceCount) POINTER_DEVICE_INFO *pointerDevices);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15470,6 +15882,7 @@ GetPointerDevice(
     _In_ HANDLE device,
     _Out_writes_(1) POINTER_DEVICE_INFO *pointerDevice);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15478,6 +15891,7 @@ GetPointerDeviceProperties(
     _Inout_ UINT32* propertyCount,
     _Out_writes_opt_(*propertyCount) POINTER_DEVICE_PROPERTY *pointerProperties);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15485,6 +15899,7 @@ RegisterPointerDeviceNotifications(
     _In_ HWND window,
     _In_ BOOL notifyRange);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15493,6 +15908,7 @@ GetPointerDeviceRects(
     _Out_writes_(1) RECT* pointerDeviceRect,
     _Out_writes_(1) RECT* displayRect);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15501,6 +15917,7 @@ GetPointerDeviceCursors(
     _Inout_ UINT32* cursorCount,
     _Out_writes_opt_(*cursorCount) POINTER_DEVICE_CURSOR_INFO *deviceCursors);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15625,6 +16042,7 @@ typedef struct tagTOUCHPAD_PARAMETERS_V2 {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15668,6 +16086,7 @@ typedef struct tagCHANGEFILTERSTRUCT {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15773,6 +16192,7 @@ typedef struct tagGESTURENOTIFYSTRUCT {
  * Gesture information retrieval
  *   - HGESTUREINFO is received by a window in the lParam of a WM_GESTURE message.
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15786,6 +16206,7 @@ GetGestureInfo(
  *   - Size, in bytes, of the extra argument data is available in the cbExtraArgs
  *     field of the GESTUREINFO structure retrieved using the GetGestureInfo function.
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15804,6 +16225,7 @@ GetGestureExtraArgs(
  *     one of the PostMessage or SendMessage class of API functions, the handle
  *     is transfered with the message and need not be closed by the application.
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15872,6 +16294,7 @@ typedef struct tagGESTURECONFIG {
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15887,6 +16310,7 @@ SetGestureConfig(
 #define GCF_INCLUDE_ANCESTORS           0x00000001      // If specified, GetGestureConfig returns consolidated configuration
                                                         // for the specified window and it's parent window chain
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15927,6 +16351,7 @@ GetGestureConfig(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15934,6 +16359,7 @@ ShutdownBlockReasonCreate(
     _In_ HWND hWnd,
     _In_ LPCWSTR pwszReason);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15942,6 +16368,7 @@ ShutdownBlockReasonQuery(
     _Out_writes_opt_(*pcchBuff) LPWSTR pwszBuff,
     _Inout_ DWORD *pcchBuff);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -15990,6 +16417,7 @@ typedef enum tagINPUT_MESSAGE_ORIGIN_ID {
 /*
  * API to determine the input source of the current messsage.
  */
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
@@ -16097,12 +16525,14 @@ SetDisplayAutoRotationPreferences(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
 IsImmersiveProcess(
     _In_ HANDLE hProcess);
 
+_Win32_metadata_set_last_error_
 WINUSERAPI
 BOOL
 WINAPI
