diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WinWlx.h b/generation/WinSDK/RecompiledIdlHeaders/um/WinWlx.h
index cdb335d1..6166a155 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WinWlx.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WinWlx.h
@@ -31,6 +31,7 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -594,6 +595,7 @@ typedef struct _WLX_CONSOLESWITCH_CREDENTIALS_INFO {
 //                                                                    //
 ////////////////////////////////////////////////////////////////////////
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxNegotiate(
@@ -601,6 +603,7 @@ WlxNegotiate(
     PDWORD                  pdwDllVersion
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxInitialize(
@@ -611,6 +614,7 @@ WlxInitialize(
     PVOID *                 pWlxContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 VOID
 WINAPI
 WlxDisplaySASNotice(
@@ -618,6 +622,7 @@ WlxDisplaySASNotice(
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 int
 WINAPI
 WlxLoggedOutSAS(
@@ -631,6 +636,7 @@ WlxLoggedOutSAS(
     PVOID *                 pProfile
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxActivateUserShell(
@@ -640,6 +646,7 @@ WlxActivateUserShell(
     PVOID                   pEnvironment
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 int
 WINAPI
 WlxLoggedOnSAS(
@@ -648,12 +655,14 @@ WlxLoggedOnSAS(
     PVOID                   pReserved
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 VOID
 WINAPI
 WlxDisplayLockedNotice(
     PVOID                   pWlxContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 int
 WINAPI
 WlxWkstaLockedSAS(
@@ -661,18 +670,21 @@ WlxWkstaLockedSAS(
     DWORD                   dwSasType
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxIsLockOk(
     PVOID                   pWlxContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxIsLogoffOk(
     PVOID                   pWlxContext
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 VOID
 WINAPI
 WlxLogoff(
@@ -680,6 +692,7 @@ WlxLogoff(
     );
 
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 VOID
 WINAPI
 WlxShutdown(
@@ -691,12 +704,14 @@ WlxShutdown(
 //
 // NEW for version 1.1
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxScreenSaverNotify(
     PVOID                   pWlxContext,
     BOOL *                  pSecure);
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxStartApplication(
@@ -710,6 +725,7 @@ WlxStartApplication(
 // New for 1.3
 //
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxNetworkProviderLoad(
@@ -721,6 +737,7 @@ WlxNetworkProviderLoad(
 #define STATUSMSG_OPTION_NOANIMATION    0x00000001
 #define STATUSMSG_OPTION_SETFOREGROUND  0x00000002
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxDisplayStatusMessage(
@@ -731,6 +748,7 @@ WlxDisplayStatusMessage(
     PWSTR                   pMessage
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxGetStatusMessage(
@@ -740,6 +758,7 @@ WlxGetStatusMessage(
     DWORD                   dwBufferSize
     );
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxRemoveStatusMessage(
@@ -750,6 +769,7 @@ WlxRemoveStatusMessage(
 //
 // New for 1.4
 //
+_Win32_metadata_supported_os_(windows5.1.2600)
 BOOL
 WINAPI
 WlxGetConsoleSwitchCredentials (
