diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wlanapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/wlanapi.h
index 3b444d9b..d3c6e777 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wlanapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wlanapi.h
@@ -25,6 +25,9 @@ Revision History:
 
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -1224,20 +1227,24 @@ typedef struct _WFD_GROUP_ID
 
 
 // public APIs
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanOpenHandle(
     _In_ DWORD dwClientVersion,
     _Reserved_ PVOID pReserved,
     _Out_ PDWORD pdwNegotiatedVersion,
     _Out_ PHANDLE phClientHandle
+        _Win32_metadata_raii_free_(WlanCloseHandle)
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanCloseHandle(
     _In_ HANDLE hClientHandle,
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanEnumInterfaces(
     _In_ HANDLE hClientHandle,
@@ -1245,6 +1252,7 @@ WlanEnumInterfaces(
     _Outptr_ PWLAN_INTERFACE_INFO_LIST *ppInterfaceList
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetAutoConfigParameter(
     _In_ HANDLE hClientHandle,
@@ -1254,6 +1262,7 @@ WlanSetAutoConfigParameter(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanQueryAutoConfigParameter(
     _In_ HANDLE hClientHandle,
@@ -1264,6 +1273,7 @@ WlanQueryAutoConfigParameter(
     _Out_opt_ PWLAN_OPCODE_VALUE_TYPE pWlanOpcodeValueType
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetInterfaceCapability(
     _In_ HANDLE hClientHandle,
@@ -1272,6 +1282,7 @@ WlanGetInterfaceCapability(
     _Outptr_ PWLAN_INTERFACE_CAPABILITY *ppCapability
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetInterface(
     _In_ HANDLE hClientHandle,
@@ -1282,6 +1293,7 @@ WlanSetInterface(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanQueryInterface(
     _In_ HANDLE hClientHandle,
@@ -1293,6 +1305,7 @@ WlanQueryInterface(
     _Out_opt_ PWLAN_OPCODE_VALUE_TYPE pWlanOpcodeValueType
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanIhvControl(
     _In_ HANDLE hClientHandle,
@@ -1305,6 +1318,7 @@ WlanIhvControl(
     _Out_ PDWORD pdwBytesReturned
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanScan(
     _In_ HANDLE hClientHandle,
@@ -1314,6 +1328,7 @@ WlanScan(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetAvailableNetworkList(
     _In_ HANDLE hClientHandle,
@@ -1332,6 +1347,7 @@ WlanGetAvailableNetworkList2(
     _Outptr_ PWLAN_AVAILABLE_NETWORK_LIST_V2 *ppAvailableNetworkList
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetNetworkBssList(
     _In_ HANDLE hClientHandle,
@@ -1343,6 +1359,7 @@ WlanGetNetworkBssList(
     _Outptr_ PWLAN_BSS_LIST *ppWlanBssList
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanConnect(
     _In_ HANDLE hClientHandle,
@@ -1359,6 +1376,7 @@ WlanConnect2(
     _Reserved_  PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanDisconnect(
     _In_ HANDLE hClientHandle,
@@ -1366,6 +1384,7 @@ WlanDisconnect(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanRegisterNotification(
     _In_ HANDLE hClientHandle,
@@ -1378,6 +1397,7 @@ WlanRegisterNotification(
 );
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetProfile(
     _In_ HANDLE hClientHandle,
@@ -1389,6 +1409,7 @@ WlanGetProfile(
     _Out_opt_ DWORD *pdwGrantedAccess
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfileEapUserData(
     _In_ HANDLE hClientHandle,
@@ -1401,6 +1422,7 @@ WlanSetProfileEapUserData(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfileEapXmlUserData(
     _In_ HANDLE hClientHandle,
@@ -1411,6 +1433,7 @@ WlanSetProfileEapXmlUserData(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfile(
     _In_ HANDLE hClientHandle,
@@ -1423,6 +1446,7 @@ WlanSetProfile(
     _Out_ DWORD *pdwReasonCode
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanDeleteProfile(
     _In_ HANDLE hClientHandle,
@@ -1431,6 +1455,7 @@ WlanDeleteProfile(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanRenameProfile(
     _In_ HANDLE hClientHandle,
@@ -1440,6 +1465,7 @@ WlanRenameProfile(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetProfileList(
     _In_ HANDLE hClientHandle,
@@ -1448,6 +1474,7 @@ WlanGetProfileList(
     _Outptr_ PWLAN_PROFILE_INFO_LIST *ppProfileList
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfileList(
     _In_ HANDLE hClientHandle,
@@ -1457,6 +1484,7 @@ WlanSetProfileList(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfilePosition(
     _In_ HANDLE hClientHandle,
@@ -1466,6 +1494,7 @@ WlanSetProfilePosition(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetProfileCustomUserData(
     _In_ HANDLE hClientHandle,
@@ -1476,6 +1505,7 @@ WlanSetProfileCustomUserData(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetProfileCustomUserData(
     _In_ HANDLE hClientHandle,
@@ -1486,6 +1516,7 @@ WlanGetProfileCustomUserData(
     _Outptr_result_bytebuffer_(*pdwDataSize) PBYTE *ppData
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetFilterList(
     _In_ HANDLE hClientHandle,
@@ -1494,6 +1525,7 @@ WlanSetFilterList(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetFilterList(
     _In_ HANDLE hClientHandle,
@@ -1502,6 +1534,7 @@ WlanGetFilterList(
     _Outptr_ PDOT11_NETWORK_LIST *ppNetworkList
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetPsdIEDataList(
     _In_ HANDLE hClientHandle,
@@ -1510,6 +1543,7 @@ WlanSetPsdIEDataList(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSaveTemporaryProfile(
     _In_ HANDLE hClientHandle,
@@ -1551,6 +1585,7 @@ WlanRegisterDeviceServiceNotification(
 #if !defined(__midl)
 
 // client side APIs
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanExtractPsdIEDataList(
     _In_ HANDLE hClientHandle,
@@ -1562,6 +1597,7 @@ WlanExtractPsdIEDataList(
 );
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanReasonCodeToString(
     _In_ DWORD dwReasonCode,
@@ -1570,16 +1606,20 @@ WlanReasonCodeToString(
     _Reserved_ PVOID pReserved
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 PVOID WINAPI
 WlanAllocateMemory(
     _In_ DWORD dwMemorySize
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 VOID WINAPI
 WlanFreeMemory(
     _In_ PVOID pMemory
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanSetSecuritySettings(
     _In_ HANDLE hClientHandle,
@@ -1587,6 +1627,7 @@ WlanSetSecuritySettings(
     _In_ LPCWSTR strModifiedSDDL
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanGetSecuritySettings(
     _In_ HANDLE hClientHandle,
@@ -1615,6 +1656,7 @@ typedef enum _WL_DISPLAY_PAGES
     WLAdvPage
 } WL_DISPLAY_PAGES, *PWL_DISPLAY_PAGES;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 DWORD WINAPI
 WlanUIEditProfile(
     _In_ DWORD dwClientVersion,
@@ -1702,6 +1744,7 @@ WLAN_HOSTED_NETWORK_PEER_AUTH_STATE, *PWLAN_HOSTED_NETWORK_PEER_AUTH_STATE;
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkStartUsing
@@ -1713,6 +1756,7 @@ WlanHostedNetworkStartUsing
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkStopUsing
@@ -1725,6 +1769,7 @@ WlanHostedNetworkStopUsing
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkForceStart
@@ -1736,6 +1781,7 @@ WlanHostedNetworkForceStart
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkForceStop
@@ -1878,6 +1924,7 @@ WLAN_HOSTED_NETWORK_SECURITY_SETTINGS, *PWLAN_HOSTED_NETWORK_SECURITY_SETTINGS;
 //
 // This function queries the static properties of the hosted network
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkQueryProperty
@@ -1896,6 +1943,7 @@ WlanHostedNetworkQueryProperty
 //
 // This function sets the static properties of the hosted network
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkSetProperty
@@ -1915,6 +1963,7 @@ WlanHostedNetworkSetProperty
 // on a machine. There are no effects if an initial
 // configuration has already been created.
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkInitSettings
@@ -1927,6 +1976,7 @@ WlanHostedNetworkInitSettings
 
 
 
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkRefreshSecuritySettings
@@ -1959,6 +2009,7 @@ WLAN_HOSTED_NETWORK_STATUS, *PWLAN_HOSTED_NETWORK_STATUS;
 //
 // This function queries the runtime status of the hosted network
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkQueryStatus
@@ -1977,6 +2028,7 @@ WlanHostedNetworkQueryStatus
 // if it is passphrase, key length includes the terminating '\0',
 // if not, key length is the number of bytes in the key data array.
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkSetSecondaryKey
@@ -1997,6 +2049,7 @@ WlanHostedNetworkSetSecondaryKey
 // If it is passphrase, key length includes the terminating '\0',
 // if not, key length is the number of bytes in the key data array.
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanHostedNetworkQuerySecondaryKey
@@ -2013,6 +2066,7 @@ WlanHostedNetworkQuerySecondaryKey
 //
 // This function is used to register and unregister notifications on virtual station.
 //
+_Win32_metadata_supported_os_(windows6.1)
 DWORD
 WINAPI
 WlanRegisterVirtualStationNotification
@@ -2564,6 +2618,7 @@ DEFINE_DEVPROPKEY(
     0x07
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDOpenHandle(
@@ -2572,6 +2627,7 @@ WFDOpenHandle(
     _Out_ PHANDLE           phClientHandle
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDCloseHandle(
@@ -2586,6 +2642,7 @@ typedef VOID (WINAPI *WFD_OPEN_SESSION_COMPLETE_CALLBACK) (
     _In_ DWORD          dwReasonCode
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDStartOpenSession(
@@ -2596,12 +2653,14 @@ WFDStartOpenSession(
     _Out_ PHANDLE                               phSessionHandle
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDCancelOpenSession(
     _In_ HANDLE                         hSessionHandle
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDOpenLegacySession(
@@ -2611,12 +2670,14 @@ WFDOpenLegacySession(
     _Out_ GUID* pGuidSessionInterface
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDCloseSession(
     _In_ HANDLE                         hSessionHandle
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 DWORD
 WINAPI
 WFDUpdateDeviceVisibility(
