diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h b/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
index 3e853a9c..a3b4dc58 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
@@ -113,6 +113,7 @@ extern "C"{
 /*                                                                             */
 /*******************************************************************************/
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family or WinMgmt Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WINMGMT)
 
@@ -192,6 +193,7 @@ EXTERN_C const IID IID_IWbemPathKeyList;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9AE62877-7544-4bb0-AA26-A13824659ED6")
     IWbemPathKeyList : public IUnknown
     {
@@ -480,6 +482,7 @@ EXTERN_C const IID IID_IWbemPath;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("3BC15AF2-736C-477e-9E51-238AF8667DCC")
     IWbemPath : public IUnknown
     {
@@ -972,6 +975,7 @@ EXTERN_C const IID IID_IWbemQuery;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("81166f58-dd98-11d3-a120-00105a1f515a")
     IWbemQuery : public IUnknown
     {
