diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wmsdkidl.h b/generation/WinSDK/RecompiledIdlHeaders/um/wmsdkidl.h
index 6132846e..9e44f140 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wmsdkidl.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wmsdkidl.h
@@ -717,6 +717,7 @@ extern "C"{
 //
 //=========================================================================
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 typedef unsigned __int64 QWORD;
@@ -2193,20 +2194,34 @@ EXTERN_GUID( WMT_DMOCATEGORY_AUDIO_WATERMARK, 0x65221c5a, 0xfa75, 0x4b39, 0xb5,
 EXTERN_GUID( WMT_DMOCATEGORY_VIDEO_WATERMARK, 0x187cc922, 0x8efc, 0x4404, 0x9d, 0xaf, 0x63, 0xf4, 0x83, 0x0d, 0xf1, 0xbc );
 #define WM_MAX_VIDEO_STREAMS            0x3f
 #define WM_MAX_STREAMS                  0x3f
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMIsContentProtected( const WCHAR *pwszFileName, BOOL *pfIsProtected );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateWriter( IUnknown* pUnkCert, IWMWriter **ppWriter );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateReader( IUnknown* pUnkCert, DWORD dwRights, IWMReader **ppReader );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateSyncReader( IUnknown* pUnkCert, DWORD dwRights, IWMSyncReader **ppSyncReader );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateEditor( IWMMetadataEditor **ppEditor );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateIndexer( IWMIndexer **ppIndexer );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateBackupRestorer( IUnknown *pCallback, IWMLicenseBackup **ppBackup );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateProfileManager( IWMProfileManager **ppProfileManager );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateWriterFileSink( IWMWriterFileSink **ppSink );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateWriterNetworkSink( IWMWriterNetworkSink **ppSink );
+_Win32_metadata_supported_os_(windows5.0)
 HRESULT STDMETHODCALLTYPE WMCreateWriterPushSink( IWMWriterPushSink **ppSink );
+_Win32_metadata_supported_os_(windows5.1.2600)
 HRESULT STDMETHODCALLTYPE WMCreateDeviceRegistration( IWMDeviceRegistration **ppDevReg );
+_Win32_metadata_supported_os_(windows5.1.2600)
 HRESULT STDMETHODCALLTYPE WMCreateDRMTranscryptor( IWMDRMTranscryptor **ppTranscryptor );
 HRESULT STDMETHODCALLTYPE WMCreateDRMTranscryptionManager( IWMDRMTranscryptionManager **ppTranscryptionManager );
+_Win32_metadata_supported_os_(windows5.1.2600)
 HRESULT STDMETHODCALLTYPE WMCreateLicenseRevocationAgent( IUnknown *pCallback, IWMLicenseRevocationAgent ** ppLicenseRevocationAgent );
 
 
@@ -2764,6 +2779,7 @@ EXTERN_C const IID IID_IWMDRMWriter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("d6ea5dd0-12a0-43f4-90ab-a3fd451e6a07")
     IWMDRMWriter : public IUnknown
     {
@@ -2946,6 +2962,7 @@ EXTERN_C const IID IID_IWMDRMWriter2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("38ee7a94-40e2-4e10-aa3f-33fd3210ed5b")
     IWMDRMWriter2 : public IWMDRMWriter
     {
@@ -3096,6 +3113,7 @@ EXTERN_C const IID IID_IWMDRMWriter3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("a7184082-a4aa-4dde-ac9c-e75dbd1117ce")
     IWMDRMWriter3 : public IWMDRMWriter2
     {
@@ -5532,6 +5550,7 @@ EXTERN_C const IID IID_IWMDRMEditor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("FF130EBC-A6C3-42A6-B401-C3382C3E08B3")
     IWMDRMEditor : public IUnknown
     {
@@ -16372,6 +16391,7 @@ EXTERN_C const IID IID_IWMDRMReader;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("D2827540-3EE7-432c-B14C-DC17F085D3B3")
     IWMDRMReader : public IUnknown
     {
@@ -16612,6 +16632,7 @@ EXTERN_C const IID IID_IWMDRMReader2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("befe7a75-9f1d-4075-b9d9-a3c37bda49a0")
     IWMDRMReader2 : public IWMDRMReader
     {
@@ -16827,6 +16848,7 @@ EXTERN_C const IID IID_IWMDRMReader3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("e08672de-f1e7-4ff4-a0a3-fc4b08e4caf8")
     IWMDRMReader3 : public IWMDRMReader2
     {
@@ -18661,6 +18683,7 @@ EXTERN_C const IID IID_IWMLicenseBackup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("05E5AC9F-3FB6-4508-BB43-A4067BA1EBE8")
     IWMLicenseBackup : public IUnknown
     {
@@ -18761,6 +18784,7 @@ EXTERN_C const IID IID_IWMLicenseRestore;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("C70B6334-A22E-4efb-A245-15E65A004A13")
     IWMLicenseRestore : public IUnknown
     {
@@ -18861,6 +18885,7 @@ EXTERN_C const IID IID_IWMBackupRestoreProps;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.0)
     MIDL_INTERFACE("3C8E0DA6-996F-4ff3-A1AF-4838F9377E2E")
     IWMBackupRestoreProps : public IUnknown
     {
@@ -19931,6 +19956,7 @@ EXTERN_C const IID IID_IWMDeviceRegistration;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("f6211f03-8d21-4e94-93e6-8510805f2d99")
     IWMDeviceRegistration : public IUnknown
     {
@@ -20399,6 +20425,7 @@ EXTERN_C const IID IID_IWMProximityDetection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("6A9FD8EE-B651-4bf0-B849-7D4ECE79A2B1")
     IWMProximityDetection : public IUnknown
     {
@@ -20514,6 +20541,7 @@ EXTERN_C const IID IID_IWMDRMMessageParser;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("A73A0072-25A0-4c99-B4A5-EDE8101A6C39")
     IWMDRMMessageParser : public IUnknown
     {
@@ -20642,6 +20670,7 @@ EXTERN_C const IID IID_IWMDRMTranscryptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows5.1.2600)
     MIDL_INTERFACE("69059850-6E6F-4bb2-806F-71863DDFC471")
     IWMDRMTranscryptor : public IUnknown
     {
