diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
index 594858a0..0378446e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
@@ -19,6 +19,7 @@ Abstract:
 #endif  // _MSC_VER
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or WNV Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WNV)
@@ -114,11 +115,14 @@ typedef struct _WNV_REDIRECT_PARAM {
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
+_Win32_metadata_supported_os_(windowsserver2012)
 HANDLE
 WINAPI
 WnvOpen(
-    );
+    )
+    _Win32_metadata_raii_free_(CloseHandle);
 
+_Win32_metadata_supported_os_(windowsserver2012)
 ULONG
 WINAPI
 WnvRequestNotification (
