diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WS2spi.h b/generation/WinSDK/RecompiledIdlHeaders/um/WS2spi.h
index 1785a208..6f008be4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WS2spi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WS2spi.h
@@ -20,6 +20,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -615,6 +616,7 @@ typedef struct _WSPUPCALLTABLE {
  *  WinSock 2 SPI socket function prototypes
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 _Must_inspect_result_
 int
 WSPAPI
@@ -641,6 +643,7 @@ int
  * Installation and configuration entrypoints.
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCEnumProtocols(
@@ -666,6 +669,7 @@ int
  * separate 64-bit and 32-bit catalogs.  API with '32' prefix
  * allow 32 bit catalog manipulations by 64 bit process.
  */
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCEnumProtocols32(
@@ -676,6 +680,7 @@ WSCEnumProtocols32(
     );
 #endif //(defined(_WIN64) && _WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCDeinstallProvider(
@@ -692,6 +697,7 @@ int
 
 #if(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCDeinstallProvider32(
@@ -700,6 +706,7 @@ WSCDeinstallProvider32(
     );
 #endif //(defined(_WIN64) && _WIN32_WINNT >= 0x0501)
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCInstallProvider(
@@ -727,6 +734,7 @@ int
  * It is needed to guarantee the same catalog IDs for provider catalog
  * entries in both 64-bit and 32-bit catalogs.
  */
+_Win32_metadata_supported_os_(windows5.1.2600)
 int
 WSPAPI
 WSCInstallProvider64_32(
@@ -738,6 +746,7 @@ WSCInstallProvider64_32(
     );
 #endif //(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCGetProviderPath(
@@ -758,6 +767,7 @@ int
 
 #if(_WIN32_WINNT >= 0x0501)
 #if defined(_WIN64)
+_Win32_metadata_supported_os_(windows6.0.6000)
 _Success_(return == 0)
 int
 WSPAPI
@@ -769,6 +779,7 @@ WSCGetProviderPath32(
     );
 #endif
 
+_Win32_metadata_supported_os_(windows5.1.2600)
 int
 WSPAPI
 WSCUpdateProvider(
@@ -790,6 +801,7 @@ int
     );
 
 #if defined(_WIN64)
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCUpdateProvider32(
@@ -803,6 +815,7 @@ WSCUpdateProvider32(
 #endif //(_WIN32_WINNT >= 0x0501)
 
 #if (_WIN32_WINNT < 0x0600)
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCInstallQOSTemplate (
@@ -819,6 +832,7 @@ int
     _In_ LPQOS           Qos
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WSCRemoveQOSTemplate (
@@ -870,6 +884,7 @@ typedef struct _WSC_PROVIDER_AUDIT_INFO {
     // Full Path of EXE performing install
 } WSC_PROVIDER_AUDIT_INFO;
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCSetProviderInfo( 
@@ -881,6 +896,7 @@ WSCSetProviderInfo(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCGetProviderInfo( 
@@ -894,6 +910,7 @@ WSCGetProviderInfo(
 
 #if defined(_WIN64)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCSetProviderInfo32( 
@@ -905,6 +922,7 @@ WSCSetProviderInfo32(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCGetProviderInfo32( 
@@ -922,6 +940,7 @@ WSCGetProviderInfo32(
 // App Permitted LSP Categorization APIs
 //
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCSetApplicationCategory(
@@ -934,6 +953,7 @@ WSCSetApplicationCategory(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 int
 WSPAPI
 WSCGetApplicationCategory(
@@ -953,6 +973,7 @@ WSCGetApplicationCategory(
  *  should not be used by any service providers.
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 WSPAPI
 WPUCloseEvent(
@@ -960,6 +981,7 @@ WPUCloseEvent(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUCloseSocketHandle(
@@ -967,12 +989,14 @@ WPUCloseSocketHandle(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 WSAEVENT
 WSPAPI
 WPUCreateEvent(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 SOCKET
 WSPAPI
 WPUCreateSocketHandle(
@@ -981,6 +1005,7 @@ WPUCreateSocketHandle(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUFDIsSet(
@@ -988,6 +1013,7 @@ WPUFDIsSet(
     _In_ fd_set FAR * fdset
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUGetProviderPath(
@@ -997,6 +1023,7 @@ WPUGetProviderPath(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 SOCKET
 WSPAPI
 WPUModifyIFSHandle(
@@ -1005,6 +1032,7 @@ WPUModifyIFSHandle(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 WSPAPI
 WPUPostMessage(
@@ -1014,6 +1042,7 @@ WPUPostMessage(
     _In_ LPARAM lParam
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUQueryBlockingCallback(
@@ -1023,6 +1052,7 @@ WPUQueryBlockingCallback(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUQuerySocketHandleContext(
@@ -1031,6 +1061,7 @@ WPUQuerySocketHandleContext(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUQueueApc(
@@ -1040,6 +1071,7 @@ WPUQueueApc(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 WSPAPI
 WPUResetEvent(
@@ -1047,6 +1079,7 @@ WPUResetEvent(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 BOOL
 WSPAPI
 WPUSetEvent(
@@ -1054,6 +1087,7 @@ WPUSetEvent(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUCompleteOverlappedRequest (
@@ -1065,6 +1099,7 @@ WPUCompleteOverlappedRequest (
 );
 
 #if(_WIN32_WINNT >= 0x0501)
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUOpenCurrentThread(
@@ -1072,6 +1107,7 @@ WPUOpenCurrentThread(
     _Out_ LPINT lpErrno
     );
 
+_Win32_metadata_supported_os_(windows5.0)
 int
 WSPAPI
 WPUCloseThread(
@@ -1094,6 +1130,7 @@ WPUCloseThread(
 #define LPFN_WSCENUMNAMESPACEPROVIDERS LPFN_WSAENUMNAMESPACEPROVIDERSW
 
 #if defined(_WIN64)
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSAAPI
 WSCEnumNameSpaceProviders32(
@@ -1108,6 +1145,7 @@ WSCEnumNameSpaceProviders32(
 #define LPFN_WSCENUMNAMESPACEPROVIDERSEX LPFN_WSAENUMNAMESPACEPROVIDERSEXW
 
 #if defined(_WIN64)
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSAAPI
 WSCEnumNameSpaceProvidersEx32(
@@ -1117,6 +1155,7 @@ WSCEnumNameSpaceProvidersEx32(
 #endif //(_WIN64)
 #endif //(_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows5.0)
 INT
 WSPAPI
 WSCInstallNameSpace (
@@ -1138,6 +1177,7 @@ INT
     );
 
 #if(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSPAPI
 WSCInstallNameSpace32 (
@@ -1149,6 +1189,7 @@ WSCInstallNameSpace32 (
     );
 #endif //(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
 
+_Win32_metadata_supported_os_(windows5.0)
 INT
 WSPAPI
 WSCUnInstallNameSpace (
@@ -1162,6 +1203,7 @@ INT
     );
 
 #if (_WIN32_WINNT >= 0x0600 )
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSPAPI
 WSCInstallNameSpaceEx(
@@ -1175,6 +1217,7 @@ WSCInstallNameSpaceEx(
 #endif //(defined(_WIN32_WINNT >= 0x0600)
 
 #if(defined(_WIN64) && (_WIN32_WINNT >= 0x0600))
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSPAPI
 WSCInstallNameSpaceEx32(
@@ -1188,6 +1231,7 @@ WSCInstallNameSpaceEx32(
 #endif //(defined(_WIN64) && (_WIN32_WINNT >= 0x0600))
 
 #if(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSPAPI
 WSCUnInstallNameSpace32 (
@@ -1195,6 +1239,7 @@ WSCUnInstallNameSpace32 (
     );
 #endif //(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
 
+_Win32_metadata_supported_os_(windows5.0)
 INT
 WSPAPI
 WSCEnableNSProvider (
@@ -1210,6 +1255,7 @@ INT
     );
 
 #if(defined(_WIN64) && (_WIN32_WINNT >= 0x0501))
+_Win32_metadata_supported_os_(windows6.0.6000)
 INT
 WSPAPI
 WSCEnableNSProvider32 (
@@ -1221,6 +1267,7 @@ WSCEnableNSProvider32 (
 #if (_WIN32_WINNT >= 0x0600)
 
 #if defined(_WIN64)
+_Win32_metadata_supported_os_(windows6.0.6000)
 int WSPAPI WSCInstallProviderAndChains64_32(
 #else
 int WSPAPI WSCInstallProviderAndChains(
@@ -1359,6 +1406,7 @@ typedef struct _NSP_ROUTINE {
  * Startup procedures.
  */
 
+_Win32_metadata_supported_os_(windows5.0)
 _Must_inspect_result_
 INT
 WSAAPI
@@ -1450,13 +1498,19 @@ typedef const NSPV2_ROUTINE * PCNSPV2_ROUTINE, * LPCNSPV2_ROUTINE;
 
 #if(_WIN32_WINNT >= 0x0600)
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 INT WSAAPI WSAAdvertiseProvider(
         _In_                const GUID *                    puuidProviderId,
         _In_                const LPCNSPV2_ROUTINE          pNSPv2Routine);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 INT WSAAPI WSAUnadvertiseProvider(
         _In_                const GUID *                    puuidProviderId);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 INT WSAAPI WSAProviderCompleteAsyncCall(
         _In_                HANDLE                          hAsyncCall,
         _In_                INT                             iRetCode);
