diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WS2tcpip.h b/generation/WinSDK/RecompiledIdlHeaders/um/WS2tcpip.h
index 0808320a..747c319f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WS2tcpip.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WS2tcpip.h
@@ -23,6 +23,9 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if _MSC_VER >= 1200
 #pragma warning(push)
@@ -115,6 +118,8 @@ typedef ADDRINFOEXA     ADDRINFOEX, *PADDRINFOEX;
 extern "C" {
 #endif
 
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -126,6 +131,7 @@ getaddrinfo(
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINXPSP2) || (_WIN32_WINNT >= 0x0502)
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -185,6 +191,7 @@ void
 
 #pragma region Desktop Family or OneCore Family or Game Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows6.0.6000)
 _WINSOCK_DEPRECATED_BY("GetAddrInfoExW()")
 WINSOCK_API_LINKAGE
 INT
@@ -204,6 +211,7 @@ GetAddrInfoExA(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -220,6 +228,7 @@ GetAddrInfoExW(
     _Out_opt_   LPHANDLE        lpHandle
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -227,6 +236,7 @@ GetAddrInfoExCancel(
     _In_        LPHANDLE        lpHandle
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -300,6 +310,7 @@ INT
 #if (_WIN32_WINNT >= 0x0600)
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows8.1)
 _WINSOCK_DEPRECATED_BY("SetAddrInfoExW()")
 WINSOCK_API_LINKAGE
 INT
@@ -321,6 +332,7 @@ SetAddrInfoExA(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -393,6 +405,8 @@ INT
 
 #endif
 
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 VOID
 WSAAPI
@@ -401,6 +415,7 @@ freeaddrinfo(
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINXPSP2) || (_WIN32_WINNT >= 0x0502)
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 VOID
 WSAAPI
@@ -443,6 +458,7 @@ VOID
 
 #pragma region App Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Win32_metadata_supported_os_(windows8.1)
 _WINSOCK_DEPRECATED_BY("FreeAddrInfoExW()")
 WINSOCK_API_LINKAGE
 void
@@ -453,6 +469,7 @@ FreeAddrInfoEx(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 void
 WSAAPI
@@ -495,6 +512,8 @@ void
 
 typedef int socklen_t;
 
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -509,6 +528,7 @@ getnameinfo(
     );
 
 #if (NTDDI_VERSION >= NTDDI_WINXPSP2) || (_WIN32_WINNT >= 0x0502)
+_Win32_metadata_supported_os_(windows8.1)
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -567,6 +587,8 @@ INT
 
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -578,6 +600,7 @@ inet_pton(
                                               PVOID           pAddrBuf
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 INT
 WSAAPI
 InetPtonW(
@@ -588,6 +611,8 @@ InetPtonW(
                                               PVOID           pAddrBuf
     );
 
+_Win32_metadata_supported_os_(windows8.1)
+_Win32_metadata_set_last_error_
 PCSTR
 WSAAPI
 inet_ntop(
@@ -597,6 +622,7 @@ inet_ntop(
     _In_                                size_t          StringBufSize
     );
 
+_Win32_metadata_supported_os_(windows8.1)
 PCWSTR
 WSAAPI
 InetNtopW(
@@ -679,6 +705,7 @@ PCWSTR
 // used by getaddrinfo().
 #define GAI_STRERROR_BUFFER_SIZE 1024
 
+_Win32_metadata_supported_os_(windows5.0)
 WS2TCPIP_INLINE
 char *
 gai_strerrorA(
@@ -686,6 +713,7 @@ gai_strerrorA(
 {
     static char buff[GAI_STRERROR_BUFFER_SIZE + 1];
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     (void)FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM
                              |FORMAT_MESSAGE_IGNORE_INSERTS
                              |FORMAT_MESSAGE_MAX_WIDTH_MASK,
@@ -699,6 +727,7 @@ gai_strerrorA(
     return buff;
 }
 
+_Win32_metadata_supported_os_(windows5.0)
 WS2TCPIP_INLINE
 WCHAR *
 gai_strerrorW(
@@ -707,6 +736,7 @@ gai_strerrorW(
 {
     static WCHAR buff[GAI_STRERROR_BUFFER_SIZE + 1];
 
+_Win32_metadata_supported_os_(windows5.1.2600)
     (void)FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM
                              |FORMAT_MESSAGE_IGNORE_INSERTS
                              |FORMAT_MESSAGE_MAX_WIDTH_MASK,
@@ -724,6 +754,7 @@ gai_strerrorW(
 
 /* Multicast source filter APIs from RFC 3678. */
 
+_Win32_metadata_supported_os_(windows8.1)
 WS2TCPIP_INLINE
 int
 setipv4sourcefilter(
@@ -769,6 +800,7 @@ setipv4sourcefilter(
     return Error;
 }
 
+_Win32_metadata_supported_os_(windows8.1)
 _Success_(return == 0)
 WS2TCPIP_INLINE
 int
@@ -820,6 +852,7 @@ getipv4sourcefilter(
 }
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
+_Win32_metadata_supported_os_(windows8.1)
 WS2TCPIP_INLINE
 int
 setsourcefilter(
@@ -867,6 +900,7 @@ setsourcefilter(
     return Error;
 }
 
+_Win32_metadata_supported_os_(windows8.1)
 _Success_(return == 0)
 WS2TCPIP_INLINE
 int
@@ -963,6 +997,7 @@ idealsendbacklognotify(
 // Wrapper functions for the IP_USER_MTU/IPV6_USER_MTU socket option.
 //
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSAGetIPUserMtu(
@@ -978,10 +1013,12 @@ WSAGetIPUserMtu(
     Error = getsockopt(Socket, SOL_SOCKET, SO_PROTOCOL_INFO, (PCHAR)&Info, &InfoSize);
     if (Error != SOCKET_ERROR) {
         if (Info.iAddressFamily == AF_INET) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 getsockopt(Socket, IPPROTO_IP, IP_USER_MTU, (PCHAR)Mtu, &OptSize);
 #if(_WIN32_WINNT >= 0x0501)
         } else if (Info.iAddressFamily == AF_INET6) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 getsockopt(Socket, IPPROTO_IPV6, IPV6_USER_MTU, (PCHAR)Mtu, &OptSize);
 #endif //(_WIN32_WINNT >= 0x0501)
@@ -994,6 +1031,7 @@ WSAGetIPUserMtu(
     return Error;
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSASetIPUserMtu(
@@ -1008,10 +1046,12 @@ WSASetIPUserMtu(
     Error = getsockopt(Socket, SOL_SOCKET, SO_PROTOCOL_INFO, (PCHAR)&Info, &InfoSize);
     if (Error != SOCKET_ERROR) {
         if (Info.iAddressFamily == AF_INET) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 setsockopt(Socket, IPPROTO_IP, IP_USER_MTU, (PCHAR)&Mtu, sizeof(Mtu));
 #if(_WIN32_WINNT >= 0x0501)
         } else if (Info.iAddressFamily == AF_INET6) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 setsockopt(Socket, IPPROTO_IPV6, IPV6_USER_MTU, (PCHAR)&Mtu, sizeof(Mtu));
 #endif //(_WIN32_WINNT >= 0x0501)
@@ -1029,6 +1069,7 @@ WSASetIPUserMtu(
 // TCP_ICMP_ERROR_SOURCE_ADDRESS socket options.
 //
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSAGetFailConnectOnIcmpError(
@@ -1041,6 +1082,7 @@ WSAGetFailConnectOnIcmpError(
                       (CHAR*)Enabled, &OptSize);
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSASetFailConnectOnIcmpError(
@@ -1052,6 +1094,7 @@ WSASetFailConnectOnIcmpError(
                       (CHAR*)&Enabled, sizeof(Enabled));
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSAGetIcmpErrorInfo(
@@ -1077,6 +1120,7 @@ WSAGetIcmpErrorInfo(
 // socket options.
 //
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSAGetUdpSendMessageSize(
@@ -1090,6 +1134,7 @@ WSAGetUdpSendMessageSize(
                       &OptSize);
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSASetUdpSendMessageSize(
@@ -1101,6 +1146,7 @@ WSASetUdpSendMessageSize(
                       sizeof(MsgSize));
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSAGetUdpRecvMaxCoalescedSize(
@@ -1114,6 +1160,7 @@ WSAGetUdpRecvMaxCoalescedSize(
                       (PCHAR)MaxCoalescedMsgSize, &OptSize);
 }
 
+_Win32_metadata_supported_os_(windows10.0.19041)
 WS2TCPIP_INLINE
 INT
 WSASetUdpRecvMaxCoalescedSize(
@@ -1146,10 +1193,12 @@ WSAGetRecvIPEcn(
     Error = getsockopt(Socket, SOL_SOCKET, SO_PROTOCOL_INFO, (PCHAR)&Info, &InfoSize);
     if (Error != SOCKET_ERROR) {
         if (Info.iAddressFamily == AF_INET) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 getsockopt(Socket, IPPROTO_IP, IP_RECVECN, (PCHAR)Enabled, &OptSize);
 #if(_WIN32_WINNT >= 0x0501)
         } else if (Info.iAddressFamily == AF_INET6) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 getsockopt(Socket, IPPROTO_IPV6, IPV6_RECVECN, (PCHAR)Enabled, &OptSize);
 #endif //(_WIN32_WINNT >= 0x0501)
@@ -1176,10 +1225,12 @@ WSASetRecvIPEcn(
     Error = getsockopt(Socket, SOL_SOCKET, SO_PROTOCOL_INFO, (PCHAR)&Info, &InfoSize);
     if (Error != SOCKET_ERROR) {
         if (Info.iAddressFamily == AF_INET) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 setsockopt(Socket, IPPROTO_IP, IP_RECVECN, (PCHAR)&Enabled, sizeof(Enabled));
 #if(_WIN32_WINNT >= 0x0501)
         } else if (Info.iAddressFamily == AF_INET6) {
+_Win32_metadata_supported_os_(windows8.1)
             Error =
                 setsockopt(Socket, IPPROTO_IPV6, IPV6_RECVECN, (PCHAR)&Enabled, sizeof(Enabled));
 #endif //(_WIN32_WINNT >= 0x0501)
@@ -1229,6 +1280,8 @@ WSASetReceivedProcessorOption(
 // Secure socket API definitions
 //
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -1240,6 +1293,8 @@ WSASetSocketSecurity (
    _In_opt_ LPWSAOVERLAPPED_COMPLETION_ROUTINE CompletionRoutine
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -1253,6 +1308,8 @@ WSAQuerySocketSecurity (
    _In_opt_ LPWSAOVERLAPPED_COMPLETION_ROUTINE CompletionRoutine
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -1264,6 +1321,8 @@ WSASetSocketPeerTargetName (
    _In_opt_ LPWSAOVERLAPPED_COMPLETION_ROUTINE CompletionRoutine
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -1275,6 +1334,8 @@ WSADeleteSocketPeerTargetName (
    _In_opt_ LPWSAOVERLAPPED_COMPLETION_ROUTINE CompletionRoutine
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
@@ -1284,6 +1345,8 @@ WSAImpersonateSocketPeer (
    _In_ ULONG PeerAddrLen
 );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 WINSOCK_API_LINKAGE
 INT
 WSAAPI
