diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdbase.h b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdbase.h
index 68f97032..4b949da6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdbase.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdbase.h
@@ -130,6 +130,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -245,6 +246,7 @@ EXTERN_C const IID IID_IWSDAddress;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("b9574c6c-12a6-4f74-93a1-3318ff605759")
     IWSDAddress : public IUnknown
     {
@@ -353,6 +355,7 @@ EXTERN_C const IID IID_IWSDTransportAddress;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("70d23498-4ee6-4340-a3df-d845d2235467")
     IWSDTransportAddress : public IWSDAddress
     {
@@ -519,6 +522,7 @@ EXTERN_C const IID IID_IWSDMessageParameters;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("1fafe8a2-e6fc-4b80-b6cf-b7d45c416d7c")
     IWSDMessageParameters : public IUnknown
     {
@@ -650,6 +654,7 @@ EXTERN_C const IID IID_IWSDMessageParameters;
 /* interface __MIDL_itf_wsdbase_0000_0003 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT WINAPI
 WSDCreateUdpMessageParameters(
     IWSDUdpMessageParameters** ppTxParams);
@@ -678,6 +683,7 @@ EXTERN_C const IID IID_IWSDUdpMessageParameters;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("9934149f-8f0c-447b-aa0b-73124b0ca7f0")
     IWSDUdpMessageParameters : public IWSDMessageParameters
     {
@@ -816,6 +822,7 @@ EXTERN_C const IID IID_IWSDUdpMessageParameters;
 /* interface __MIDL_itf_wsdbase_0000_0004 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT WINAPI
 WSDCreateUdpAddress(
     _Outptr_ IWSDUdpAddress** ppAddress);
@@ -851,6 +858,7 @@ EXTERN_C const IID IID_IWSDUdpAddress;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("74d6124a-a441-4f78-a1eb-97a8d1996893")
     IWSDUdpAddress : public IWSDTransportAddress
     {
@@ -1114,6 +1122,7 @@ EXTERN_C const IID IID_IWSDUdpAddress;
 /* interface __MIDL_itf_wsdbase_0000_0005 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT WINAPI
 WSDCreateHttpMessageParameters(
     _Outptr_ IWSDHttpMessageParameters** ppTxParams);
@@ -1133,6 +1142,7 @@ EXTERN_C const IID IID_IWSDHttpMessageParameters;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("540bd122-5c83-4dec-b396-ea62a2697fdf")
     IWSDHttpMessageParameters : public IWSDMessageParameters
     {
@@ -1358,6 +1368,7 @@ EXTERN_C const IID IID_IWSDHttpMessageParameters;
 /* interface __MIDL_itf_wsdbase_0000_0006 */
 /* [local] */ 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HRESULT WINAPI
 WSDCreateHttpAddress(
     _Outptr_ IWSDHttpAddress** ppAddress);
@@ -1377,6 +1388,7 @@ EXTERN_C const IID IID_IWSDHttpAddress;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.0.6000)
     MIDL_INTERFACE("d09ac7bd-2a3e-4b85-8605-2737ff3e4ea0")
     IWSDHttpAddress : public IWSDTransportAddress
     {
@@ -1579,6 +1591,7 @@ EXTERN_C const IID IID_IWSDSSLClientCertificate;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("de105e87-a0da-418e-98ad-27b9eed87bdc")
     IWSDSSLClientCertificate : public IUnknown
     {
@@ -1679,6 +1692,7 @@ EXTERN_C const IID IID_IWSDHttpAuthParameters;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows8.0)
     MIDL_INTERFACE("0b476df0-8dac-480d-b05c-99781a5884aa")
     IWSDHttpAuthParameters : public IUnknown
     {
@@ -1779,6 +1793,7 @@ EXTERN_C const IID IID_IWSDSignatureProperty;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("03ce20aa-71c4-45e2-b32e-3766c61c790f")
     IWSDSignatureProperty : public IUnknown
     {
