diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WtsApi32.h b/generation/WinSDK/RecompiledIdlHeaders/um/WtsApi32.h
index 77ba67e2..506f3932 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WtsApi32.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WtsApi32.h
@@ -15,6 +15,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -800,11 +801,15 @@ typedef struct _WTS_SESSION_ADDRESS {
 ==   Windows Terminal Server public APIs
 =====================================================================*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSStopRemoteControlSession(
     IN ULONG   LogonId
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSStartRemoteControlSessionW(
     _In_ LPWSTR  pTargetServerName,
@@ -813,6 +818,8 @@ WTSStartRemoteControlSessionW(
     IN USHORT  HotkeyModifiers
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSStartRemoteControlSessionA(
     _In_ LPSTR  pTargetServerName,
@@ -827,6 +834,8 @@ WTSStartRemoteControlSessionA(
 #define WTSStartRemoteControlSession WTSStartRemoteControlSessionA
 #endif /* UNICODE */
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSConnectSessionA(
@@ -836,6 +845,8 @@ WTSConnectSessionA(
     IN BOOL bWait
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSConnectSessionW(
@@ -851,6 +862,8 @@ WTSConnectSessionW(
 #define WTSConnectSession WTSConnectSessionA
 #endif
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateServersW(
@@ -861,6 +874,8 @@ WTSEnumerateServersW(
     _Out_ DWORD * pCount
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateServersA(
@@ -879,29 +894,37 @@ WTSEnumerateServersA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HANDLE
 WINAPI
 WTSOpenServerW(
     _In_ LPWSTR pServerName
-    );
+    )
+    _Win32_metadata_raii_free_(WTSCloseServer);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 HANDLE
 WINAPI
 WTSOpenServerA(
     _In_ LPSTR pServerName
-    );
+    )
+    _Win32_metadata_raii_free_(WTSCloseServer);
 
+_Win32_metadata_supported_os_(windows6.1)
 HANDLE
 WINAPI
 WTSOpenServerExW(
     _In_ LPWSTR pServerName
-    );
+    )
+    _Win32_metadata_raii_free_(WTSCloseServer);
 
+_Win32_metadata_supported_os_(windows6.1)
 HANDLE
 WINAPI
 WTSOpenServerExA(
     _In_ LPSTR pServerName
-    );
+    )
+    _Win32_metadata_raii_free_(WTSCloseServer);
 
 #ifdef UNICODE
 #define WTSOpenServer WTSOpenServerW
@@ -913,6 +936,7 @@ WTSOpenServerExA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 VOID
 WINAPI
 WTSCloseServer(
@@ -921,6 +945,8 @@ WTSCloseServer(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateSessionsW(
@@ -931,6 +957,8 @@ WTSEnumerateSessionsW(
     _Out_ DWORD * pCount
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateSessionsA(
@@ -941,6 +969,8 @@ WTSEnumerateSessionsA(
     _Out_ DWORD * pCount
     );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateSessionsExW(
@@ -950,6 +980,8 @@ WTSEnumerateSessionsExW(
     _Outptr_result_buffer_(*pCount) PWTS_SESSION_INFO_1W * ppSessionInfo,
     _Out_ DWORD * pCount );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateSessionsExA(
@@ -969,6 +1001,8 @@ WTSEnumerateSessionsExA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateProcessesW(
@@ -979,6 +1013,8 @@ WTSEnumerateProcessesW(
     OUT DWORD * pCount
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSEnumerateProcessesA(
@@ -997,6 +1033,8 @@ WTSEnumerateProcessesA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSTerminateProcess(
@@ -1008,6 +1046,8 @@ WTSTerminateProcess(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSQuerySessionInformationW(
@@ -1018,6 +1058,8 @@ WTSQuerySessionInformationW(
     _Out_ DWORD * pBytesReturned
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSQuerySessionInformationA(
@@ -1036,6 +1078,8 @@ WTSQuerySessionInformationA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSQueryUserConfigW(
@@ -1046,6 +1090,8 @@ WTSQueryUserConfigW(
     _Out_ DWORD * pBytesReturned
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSQueryUserConfigA(
@@ -1064,6 +1110,8 @@ WTSQueryUserConfigA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSSetUserConfigW(
@@ -1074,6 +1122,8 @@ WTSSetUserConfigW(
     IN DWORD DataLength
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSSetUserConfigA(
@@ -1092,6 +1142,8 @@ WTSSetUserConfigA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSSendMessageW(
@@ -1107,6 +1159,8 @@ WTSSendMessageW(
     IN BOOL bWait
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSSendMessageA(
@@ -1130,6 +1184,8 @@ WTSSendMessageA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSDisconnectSession(
@@ -1140,6 +1196,8 @@ WTSDisconnectSession(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSLogoffSession(
@@ -1150,6 +1208,8 @@ WTSLogoffSession(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSShutdownSystem(
@@ -1159,6 +1219,8 @@ WTSShutdownSystem(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSWaitSystemEvent(
@@ -1169,13 +1231,16 @@ WTSWaitSystemEvent(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 HANDLE
 WINAPI
 WTSVirtualChannelOpen(
     IN HANDLE hServer,
     IN DWORD SessionId,
     _In_ LPSTR pVirtualName   /* ascii name */
-    );
+    )
+    _Win32_metadata_raii_free_(WTSVirtualChannelClose);
 
 #define WTS_CHANNEL_OPTION_DYNAMIC          0x00000001       // dynamic channel
 #define WTS_CHANNEL_OPTION_DYNAMIC_PRI_LOW  0x00000000   // priorities
@@ -1184,20 +1249,27 @@ WTSVirtualChannelOpen(
 #define WTS_CHANNEL_OPTION_DYNAMIC_PRI_REAL 0x00000006
 #define WTS_CHANNEL_OPTION_DYNAMIC_NO_COMPRESS 0x00000008
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 HANDLE
 WINAPI
 WTSVirtualChannelOpenEx(
                      IN DWORD SessionId,
                      _In_ LPSTR pVirtualName,   /* ascii name */
                      IN DWORD flags
-                     );
+                     )
+    _Win32_metadata_raii_free_(WTSVirtualChannelClose);
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelClose(
     IN HANDLE hChannelHandle
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelRead(
@@ -1208,6 +1280,8 @@ WTSVirtualChannelRead(
     _Out_ PULONG pBytesRead
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelWrite(
@@ -1217,12 +1291,16 @@ WTSVirtualChannelWrite(
     _Out_ PULONG pBytesWritten
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelPurgeInput(
     IN HANDLE hChannelHandle
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelPurgeOutput(
@@ -1230,6 +1308,8 @@ WTSVirtualChannelPurgeOutput(
     );
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSVirtualChannelQuery(
@@ -1242,6 +1322,7 @@ WTSVirtualChannelQuery(
 /*------------------------------------------------*/
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
 VOID
 WINAPI
 WTSFreeMemory(
@@ -1254,6 +1335,8 @@ WTSFreeMemory(
 #define NOTIFY_FOR_THIS_SESSION     0
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSRegisterSessionNotification(
@@ -1261,6 +1344,8 @@ WTSRegisterSessionNotification(
     DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSUnRegisterSessionNotification(
@@ -1268,6 +1353,8 @@ WTSUnRegisterSessionNotification(
     );
 
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSRegisterSessionNotificationEx(
@@ -1276,6 +1363,8 @@ WTSRegisterSessionNotificationEx(
     IN DWORD dwFlags
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSUnRegisterSessionNotificationEx(
@@ -1283,6 +1372,8 @@ WTSUnRegisterSessionNotificationEx(
     IN HWND hWnd
     );
 
+_Win32_metadata_supported_os_(windows6.0.6000)
+_Win32_metadata_set_last_error_
 BOOL
 WINAPI
 WTSQueryUserToken(
@@ -1347,6 +1438,8 @@ typedef enum _WTS_TYPE_CLASS {
 #endif
 } WTS_TYPE_CLASS;
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSFreeMemoryExW(
     _In_ WTS_TYPE_CLASS WTSTypeClass,
@@ -1354,6 +1447,8 @@ WTSFreeMemoryExW(
     _In_ ULONG   NumberOfEntries
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSFreeMemoryExA(
     _In_ WTS_TYPE_CLASS WTSTypeClass,
@@ -1369,6 +1464,8 @@ WTSFreeMemoryExA(
 
 /*------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSEnumerateProcessesExW
 (
@@ -1379,6 +1476,8 @@ WTSEnumerateProcessesExW
     _Out_ DWORD * pCount
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSEnumerateProcessesExA
 (
@@ -1412,6 +1511,8 @@ typedef WTSLISTENERNAMEA *PWTSLISTENERNAMEA;
 #define PWTSLISTENERNAME PWTSLISTENERNAMEA
 #endif
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSEnumerateListenersW (
   _In_                  HANDLE hServer,
@@ -1421,6 +1522,8 @@ WTSEnumerateListenersW (
   _Inout_       DWORD*  pCount
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI
 WTSEnumerateListenersA (
   _In_                  HANDLE hServer,
@@ -1521,6 +1624,8 @@ typedef struct _WTSLISTENERCONFIGA{
     CHAR InitialProgram[ MAX_PATH + 1 ];
 } WTSLISTENERCONFIGA, *PWTSLISTENERCONFIGA;
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSQueryListenerConfigW (
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1529,6 +1634,8 @@ BOOL WINAPI WTSQueryListenerConfigW (
   _Out_         PWTSLISTENERCONFIGW pBuffer
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSQueryListenerConfigA (
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1537,6 +1644,8 @@ BOOL WINAPI WTSQueryListenerConfigA (
   _Out_         PWTSLISTENERCONFIGA pBuffer
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSCreateListenerW (
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1546,6 +1655,8 @@ BOOL WINAPI WTSCreateListenerW (
   _In_          DWORD flag
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSCreateListenerA (
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1555,6 +1666,8 @@ BOOL WINAPI WTSCreateListenerA (
   _In_          DWORD flag
 );
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSSetListenerSecurityW(
   _In_          HANDLE hServer, 
   _In_          PVOID pReserved,  
@@ -1563,6 +1676,8 @@ BOOL WINAPI WTSSetListenerSecurityW(
   _In_          SECURITY_INFORMATION SecurityInformation,
   _In_          PSECURITY_DESCRIPTOR pSecurityDescriptor);
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSSetListenerSecurityA(
   _In_          HANDLE hServer, 
   _In_          PVOID pReserved,  
@@ -1571,6 +1686,8 @@ BOOL WINAPI WTSSetListenerSecurityA(
   _In_          SECURITY_INFORMATION SecurityInformation,
   _In_          PSECURITY_DESCRIPTOR pSecurityDescriptor);
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSGetListenerSecurityW(
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1582,6 +1699,8 @@ BOOL WINAPI WTSGetListenerSecurityW(
   _Out_         LPDWORD lpnLengthNeeded);
 
 
+_Win32_metadata_supported_os_(windows6.1)
+_Win32_metadata_set_last_error_
 BOOL WINAPI WTSGetListenerSecurityA(
   _In_          HANDLE hServer,
   _In_          PVOID pReserved,
@@ -1646,16 +1765,19 @@ BOOL WINAPI WTSCloudAuthDuplicateSerializedUserCredential(
 #define WTSGetListenerSecurity WTSGetListenerSecurityA
 #endif
 
+_Win32_metadata_supported_os_(windows8.0)
 BOOL
 WTSEnableChildSessions(
     BOOL bEnable
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 BOOL
 WTSIsChildSessionsEnabled(
     _Out_ PBOOL pbEnabled
     );
 
+_Win32_metadata_supported_os_(windows8.0)
 BOOL
 WTSGetChildSessionId(
     _Out_ PULONG pSessionId
