diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wtshintapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/wtshintapi.h
index da70dd26..16869e2a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wtshintapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wtshintapi.h
@@ -17,6 +17,7 @@
 #endif
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -86,6 +87,7 @@ extern "C" {
 //                 The hintDataSize MUST be sizeof(RECT)
 /*------------------------------------------------------------------*/
 
+_Win32_metadata_supported_os_(windows8.0)
 HRESULT WINAPI WTSSetRenderHint(
   _Inout_  UINT64*             pRenderHintID,
   _In_     HWND  	       hwndOwner,
