diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wtsprotocol.h b/generation/WinSDK/RecompiledIdlHeaders/um/wtsprotocol.h
index 6dbf2ba8..13ff2f96 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wtsprotocol.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wtsprotocol.h
@@ -229,6 +229,9 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -255,6 +258,7 @@ EXTERN_C const IID IID_IWTSProtocolManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("F9EAF6CC-ED79-4f01-821D-1F881B9F66CC")
     IWTSProtocolManager : public IUnknown
     {
@@ -402,6 +406,7 @@ EXTERN_C const IID IID_IWTSProtocolListener;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("23083765-45f0-4394-8f69-32b2bc0ef4ca")
     IWTSProtocolListener : public IUnknown
     {
@@ -498,6 +503,7 @@ EXTERN_C const IID IID_IWTSProtocolListenerCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("23083765-1a2d-4de2-97de-4a35f260f0b3")
     IWTSProtocolListenerCallback : public IUnknown
     {
@@ -589,6 +595,7 @@ EXTERN_C const IID IID_IWTSProtocolConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("23083765-9095-4648-98bf-ef81c914032d")
     IWTSProtocolConnection : public IUnknown
     {
@@ -1030,6 +1037,7 @@ EXTERN_C const IID IID_IWTSProtocolConnectionCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("23083765-75eb-41fe-b4fb-e086242afa0f")
     IWTSProtocolConnectionCallback : public IUnknown
     {
@@ -1163,6 +1171,7 @@ EXTERN_C const IID IID_IWTSProtocolShadowConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("ee3b0c14-37fb-456b-bab3-6d6cd51e13bf")
     IWTSProtocolShadowConnection : public IUnknown
     {
@@ -1320,6 +1329,7 @@ EXTERN_C const IID IID_IWTSProtocolShadowCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("503a2504-aae5-4ab1-93e0-6d1c4bc6f71a")
     IWTSProtocolShadowCallback : public IUnknown
     {
@@ -1454,6 +1464,7 @@ EXTERN_C const IID IID_IWTSProtocolLicenseConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("23083765-178c-4079-8e4a-fea6496a4d70")
     IWTSProtocolLicenseConnection : public IUnknown
     {
@@ -1600,6 +1611,7 @@ EXTERN_C const IID IID_IWTSProtocolLogonErrorRedirector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2008)
     MIDL_INTERFACE("FD9B61A7-2916-4627-8DEE-4328711AD6CB")
     IWTSProtocolLogonErrorRedirector : public IUnknown
     {
@@ -1776,6 +1788,7 @@ EXTERN_C const IID IID_IWRdsProtocolSettings;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("654A5A6A-2550-47EB-B6F7-EBD637475265")
     IWRdsProtocolSettings : public IUnknown
     {
@@ -1886,6 +1899,7 @@ EXTERN_C const IID IID_IWRdsProtocolManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("DC796967-3ABB-40CD-A446-105276B58950")
     IWRdsProtocolManager : public IUnknown
     {
@@ -2072,6 +2086,7 @@ EXTERN_C const IID IID_IWRdsProtocolListener;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("FCBC131B-C686-451D-A773-E279E230F540")
     IWRdsProtocolListener : public IUnknown
     {
@@ -2183,6 +2198,7 @@ EXTERN_C const IID IID_IWRdsProtocolListenerCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("3AB27E5B-4449-4DC1-B74A-91621D4FE984")
     IWRdsProtocolListenerCallback : public IUnknown
     {
@@ -2278,6 +2294,7 @@ EXTERN_C const IID IID_IWRdsProtocolConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("324ED94F-FDAF-4FF6-81A8-42ABE755830B")
     IWRdsProtocolConnection : public IUnknown
     {
@@ -2732,6 +2749,7 @@ EXTERN_C const IID IID_IWRdsProtocolConnectionCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("F1D70332-D070-4EF1-A088-78313536C2D6")
     IWRdsProtocolConnectionCallback : public IUnknown
     {
@@ -2865,6 +2883,7 @@ EXTERN_C const IID IID_IWRdsProtocolShadowConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("9AE85CE6-CADE-4548-8FEB-99016597F60A")
     IWRdsProtocolShadowConnection : public IUnknown
     {
@@ -3022,6 +3041,7 @@ EXTERN_C const IID IID_IWRdsProtocolShadowCallback;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("E0667CE0-0372-40D6-ADB2-A0F3322674D6")
     IWRdsProtocolShadowCallback : public IUnknown
     {
@@ -3156,6 +3176,7 @@ EXTERN_C const IID IID_IWRdsProtocolLicenseConnection;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("1D6A145F-D095-4424-957A-407FAE822D84")
     IWRdsProtocolLicenseConnection : public IUnknown
     {
@@ -3302,6 +3323,7 @@ EXTERN_C const IID IID_IWRdsProtocolLogonErrorRedirector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windowsserver2012)
     MIDL_INTERFACE("519FE83B-142A-4120-A3D5-A405D315281A")
     IWRdsProtocolLogonErrorRedirector : public IUnknown
     {
