diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/wtypes.h b/generation/WinSDK/RecompiledIdlHeaders/shared/wtypes.h
index de2c4220..b442e72c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/wtypes.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/wtypes.h
@@ -29,6 +29,10 @@
 #ifndef __wtypes_h__
 #define __wtypes_h__
 
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #if defined(_MSC_VER) && (_MSC_VER >= 1020)
 #pragma once
 #endif
@@ -734,9 +738,11 @@ typedef DECIMAL *LPDECIMAL;
 typedef /* [unique] */  __RPC_unique_pointer FLAGGED_WORD_BLOB *wireBSTR;
 
 #ifndef _PREFAST_
+_Win32_metadata_free_with_(SysFreeString)
 typedef /* [wire_marshal] */ OLECHAR *BSTR;
 
 #else // _PREFAST_
+_Win32_metadata_free_with_(SysFreeString)
 typedef _Null_terminated_ OLECHAR *BSTR;
 #endif
 typedef BSTR *LPBSTR;
