diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/xapobase.h b/generation/WinSDK/RecompiledIdlHeaders/um/xapobase.h
index 928f08bb..71cc0e20 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/xapobase.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/xapobase.h
@@ -13,6 +13,7 @@
     1.  See XAPO.h for the rules governing XAPO interface behaviour.        */
 
 #pragma once
+#include <win32metadata_annotations.h>
 
 //--------------<D-E-F-I-N-I-T-I-O-N-S>-------------------------------------//
 #include "XAPO.h"
