diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/xpsprint.h b/generation/WinSDK/RecompiledIdlHeaders/um/xpsprint.h
index 1a87e0d8..95c02d47 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/xpsprint.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/xpsprint.h
@@ -77,6 +77,9 @@ extern "C"{
 //
 //--------------------------------------------------------------------------
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #if (NTDDI_VERSION >= NTDDI_WIN7)
@@ -127,6 +130,7 @@ EXTERN_C const IID IID_IXpsPrintJobStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("7a77dc5f-45d6-4dff-9307-d8cb846347ca")
     IXpsPrintJobStream : public ISequentialStream
     {
@@ -236,6 +240,7 @@ EXTERN_C const IID IID_IXpsPrintJob;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+_Win32_metadata_supported_os_(windows6.1)
     MIDL_INTERFACE("5ab89b06-8194-425f-ab3b-d7a96e350161")
     IXpsPrintJob : public IUnknown
     {
@@ -326,6 +331,7 @@ EXTERN_C const IID IID_IXpsPrintJob;
 /* module XpsPrint */
 /* [dllname][uuid] */ 
 
+_Win32_metadata_supported_os_(windows6.1)
 /* [entry][helpstring] */ HRESULT __stdcall StartXpsPrintJob( 
     /* [string][in] */ __RPC__in_string LPCWSTR printerName,
     /* [string][in] */ __RPC__in_string LPCWSTR jobName,
@@ -338,6 +344,7 @@ EXTERN_C const IID IID_IXpsPrintJob;
     /* [out] */ __RPC__deref_out_opt IXpsPrintJobStream **documentStream,
     /* [out] */ __RPC__deref_out_opt IXpsPrintJobStream **printTicketStream);
 
+_Win32_metadata_supported_os_(windows6.1)
 /* [entry][helpstring] */ HRESULT __stdcall StartXpsPrintJob1( 
     /* [string][in] */ __RPC__in_string LPCWSTR printerName,
     /* [string][in] */ __RPC__in_string LPCWSTR jobName,
